{
 "cells": [
  {
   "cell_type": "markdown",
   "id": "e6c79191-7c4b-4a43-96d6-7b469a88884e",
   "metadata": {},
   "source": [
    "Cutadapt \n",
    "- using SYM_VAR primers to extract symbiodinium (symbiont)seqs from original raw sequences \n",
    "- https://cutadapt.readthedocs.io/en/stable/guide.html#five-prime-adapters\n",
    "\n",
    "Parameter Notes:\n",
    "- Used assembled contig file as input file\n",
    "- SYM_VAR primer seqs from (Hume et al 2018)\n",
    "- paired primers...if using paired end seqs as input files, would include the reverse complements of both \n",
    "- linked primers are apparently only for specific scenarios like barcoding or circulization etc\n",
    "- output as fasta file since 1) input is fasta format (.fa) and 2) no quality trimming or info "
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "66281f66-8ec2-4d74-bc4f-5fb45f97f2cd",
   "metadata": {},
   "outputs": [],
   "source": [
    "conda create -n cutadaptenv\n",
    "conda install -c bioconda cutadapt                                                        \n",
    "#install cutadapt to new env"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "0af365d5-950b-42b4-b8df-aacb9c5bc18b",
   "metadata": {},
   "outputs": [],
   "source": [
    "#!/bin/bash\n",
    "#SBATCH -c 24  # Number of Cores per Task\n",
    "#SBATCH --mem=180G  # Requested Memory\n",
    "#SBATCH -p cpu-long  # Partition\n",
    "#SBATCH -t 24:00:00  # Job time limit\n",
    "#SBATCH --mail-type=ALL\n",
    "#SBATCH -o /project/pi_sarah_gignouxwolfsohn_uml_edu/brooke/bash_scripts/slurm-%j.out  # %j = job ID\n",
    "\n",
    "# Set your input and output files\n",
    "SAMPLENAME=\"mcav\"\n",
    "INPUTDIR=\"/project/pi_sarah_gignouxwolfsohn_uml_edu/brooke/mcav/assembly/mcav_assembly3\"\n",
    "OUTPUTDIR=\"/project/pi_sarah_gignouxwolfsohn_uml_edu/brooke/mcav/ITS2/redo_1_2024\"\n",
    "\n",
    "input_fasta=\"$SAMPLENAME.contigs.fa\"\n",
    "#output_fasta=\"$SAMPLENAME_symb.contigs.fasta\"\n",
    "#couldn't get this to work?? just typed it out for the -o parameter in cutadapt script\n",
    "\n",
    "#load modules\n",
    "module load miniconda/22.11.1-1\n",
    "conda activate cutadaptenv\n",
    "\n",
    "# Set your primer sequences\n",
    "forward_primer=\"GAATTGCAGAACTCCGTGAACC\"\n",
    "reverse_primer=\"CGGGTTCWCTTGTYTGACTTCATGC\"\n",
    "\n",
    "# Verify path and input dir - only need for troubleshooting\n",
    "echo \"Working Directory: $(pwd)\"\n",
    "#ls -l $INPUTDIR\n",
    "\n",
    "# Run cutadapt\n",
    "cutadapt \\\n",
    "  -g \"$forward_primer\" \\\n",
    "  -a \"$reverse_primer\" \\\n",
    "  --discard-untrimmed \\\n",
    "  --action=retain \\\n",
    "  -o $OUTPUTDIR/\"mcav_ITS2.contigs.fasta\" \\\n",
    "  $INPUTDIR/\"$input_fasta\"\n",
    "  \n",
    "ls -l $OUTPUTDIR\n",
    "#check results dir to see if it was successful in creating output file \n",
    "\n",
    "#script file name: ITS2_trimming\n",
    "#JOB ID: 18137275\n"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "7078bcf0-96fc-4c13-8fc8-e936b4e9c87a",
   "metadata": {},
   "source": [
    "Loading miniconda version 22.11.1-1\n",
    "Working Directory: //project/pi_sarah_gignouxwolfsohn_uml_edu/brooke\n",
    "This is cutadapt 4.4 with Python 3.7.12\n",
    "Command line parameters: -g GAATTGCAGAACTCCGTGAACC -a CGGGTTCWCTTGTYTGACTTCATGC --discard-untrimmed --action=retain -o /project/pi_sarah_gignouxwolfsohn_uml_edu/brooke/mcav/ITS2/redo_1_2024/mcav_ITS2.contigs.fasta /project/pi_sarah_gignouxwolfsohn_uml_edu/brooke/mcav/assembly/mcav_assembly3/mcav.contigs.fa\n",
    "Processing single-end reads on 1 core ...\n",
    "Finished in 6.469 s (7.703 Âµs/read; 7.79 M reads/minute).\n",
    "\n",
    "=== Summary ===\n",
    "\n",
    "Total reads processed:                 839,850\n",
    "Reads with adapters:                    23,485 (2.8%)\n",
    "\n",
    "== Read fate breakdown ==\n",
    "Reads discarded as untrimmed:          816,365 (97.2%)\n",
    "Reads written (passing filters):        23,485 (2.8%)\n",
    "\n",
    "Total basepairs processed:   435,829,081 bp\n",
    "Total written (filtered):     12,156,332 bp (2.8%)\n",
    "\n",
    "=== Adapter 1 ===\n",
    "\n",
    "Sequence: GAATTGCAGAACTCCGTGAACC; Type: regular 5'; Length: 22; Trimmed: 13528 times\n",
    "\n",
    "Minimum overlap: 3\n",
    "No. of allowed errors:\n",
    "1-9 bp: 0; 10-19 bp: 1; 20-22 bp: 2\n",
    "\n",
    "Overview of removed sequences\n",
    "length\tcount\texpect\tmax.err\terror counts\n",
    "3\t9739\t13122.7\t0\t9739\n",
    "4\t2849\t3280.7\t0\t2849\n",
    "5\t694\t820.2\t0\t694\n",
    "6\t146\t205.0\t0\t146\n",
    "7\t62\t51.3\t0\t62\n",
    "8\t9\t12.8\t0\t9\n",
    "9\t3\t3.2\t0\t0 3\n",
    "10\t14\t0.8\t1\t0 14\n",
    "11\t5\t0.2\t1\t0 5\n",
    "12\t3\t0.1\t1\t0 3\n",
    "13\t1\t0.0\t1\t0 1\n",
    "24\t1\t0.0\t2\t1\n",
    "119\t1\t0.0\t2\t1\n",
    "2819\t1\t0.0\t2\t1\n",
    "\n",
    "\n",
    "=== Adapter 2 ===\n",
    "\n",
    "Sequence: CGGGTTCWCTTGTYTGACTTCATGC; Type: regular 3'; Length: 25; Trimmed: 9957 times\n",
    "\n",
    "Minimum overlap: 3\n",
    "No. of allowed errors:\n",
    "1-9 bp: 0; 10-19 bp: 1; 20-25 bp: 2\n",
    "\n",
    "Bases preceding removed adapters:\n",
    "  A: 23.2%\n",
    "  C: 23.8%\n",
    "  G: 25.1%\n",
    "  T: 27.9%\n",
    "  none/other: 0.0%\n",
    "\n",
    "Overview of removed sequences\n",
    "length\tcount\texpect\tmax.err\terror counts\n",
    "3\t7883\t13122.7\t0\t7883\n",
    "4\t1639\t3280.7\t0\t1639\n",
    "5\t258\t820.2\t0\t258\n",
    "6\t65\t205.0\t0\t65\n",
    "7\t31\t51.3\t0\t31\n",
    "8\t11\t12.8\t0\t11\n",
    "9\t20\t3.2\t0\t4 16\n",
    "10\t24\t0.8\t1\t2 22\n",
    "11\t14\t0.2\t1\t1 13\n",
    "12\t6\t0.1\t1\t0 6\n",
    "13\t4\t0.0\t1\t0 4\n",
    "14\t1\t0.0\t1\t0 1\n",
    "17\t1\t0.0\t1\t1\n",
    "total 4113\n",
    "-rw-rw-r-- 1 brooke_sienkiewicz_student_uml_edu pi_sarah_gignouxwolfsohn_uml_edu 13140771 Jan 31 16:49 mcav_ITS2.contigs.fasta"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "455514d3-218c-47c1-8ec4-293653dedc71",
   "metadata": {},
   "outputs": [],
   "source": [
    "#!/bin/bash\n",
    "#SBATCH -c 24  # Number of Cores per Task\n",
    "#SBATCH --mem=50G  # Requested Memory\n",
    "#SBATCH -p cpu  # Partition\n",
    "#SBATCH -t 16:00:00  # Job time limit\n",
    "#SBATCH --mail-type=ALL\n",
    "#SBATCH -o /project/pi_sarah_gignouxwolfsohn_uml_edu/brooke/bash_scripts/slurm-%j.out  # %j = job ID\n",
    "\n",
    "module load miniconda/22.11.1-1\n",
    "conda activate anvio-8\n",
    "\n",
    "#set variables\n",
    "SAMPLENAME=mcav\n",
    "READSPATH=\"//project/pi_sarah_gignouxwolfsohn_uml_edu/brooke/mcav/trimmed/redo_auto_detect_01312024\"\n",
    "CONTIGPATH=\"/project/pi_sarah_gignouxwolfsohn_uml_edu/brooke/mcav/ITS2/redo_1_2024\"\n",
    "CONTIGFILE=\"$SAMPLENAME\"_ITS2.contigs.fasta\n",
    "IDPATH=\"/project/pi_sarah_gignouxwolfsohn_uml_edu/brooke/mcav/trimmed/01312024_redo\"\n",
    "IDFILE=\"mcav_sampleids.txt\"\n",
    "#No FIXEDCON - Skipping defline fix and contig bp filtering step \n",
    "\n",
    "#build an index of your contigs, which only needs to happen once\n",
    "bowtie2-build $CONTIGPATH/$CONTIGFILE $CONTIGPATH/\"$SAMPLENAME\"_ITS2_contigs\n",
    "\n",
    "#align reads to your contigs and collect in a .sam file\n",
    "LINES=$(cat $IDPATH/$IDFILE)                                                                                                                                                                            \n",
    "for f in $LINES \n",
    "do\n",
    "bowtie2 --threads 11 -x $CONTIGPATH/\"$SAMPLENAME\"_ITS2_contigs -1 $READSPATH/\"$f\"_R1_001_val_1.fq -2 $READSPATH/\"$f\"_R2_001_val_2.fq -S $CONTIGPATH/\"$f\".sam\n",
    "\n",
    "#convert sam files to a bam files \n",
    "samtools view -F 4 -b -S $CONTIGPATH/\"$f\".sam -o $CONTIGPATH/\"$f\"-RAW.bam\n",
    "\n",
    "#index and sort bam files using anvio\n",
    "anvi-init-bam $CONTIGPATH/\"$f\"-RAW.bam -o $CONTIGPATH/index/\"$f\".bam\n",
    "\n",
    "#remove raw bam files to cleanup \n",
    "rm $CONTIGPATH/\"$f\"-RAW.bam\n",
    "\n",
    "#convert bam files to fastq format for symbio import\n",
    "samtools bam2fq $CONTIGPATH/index/\"$f\".bam > $CONTIGPATH/seqs/\"$f\".fastq\n",
    "\n",
    "done\n",
    "#generates BAM files from each sample sequence, aligns, indexes...need output bam for downstream analysis\n",
    "\n",
    "#bash script: ITS2_mapping\n",
    "#JOB ID: 18332770"
   ]
  }
 ],
 "metadata": {
  "kernelspec": {
   "display_name": "Python [conda env:jupyterlab]",
   "language": "python",
   "name": "conda-env-jupyterlab-py"
  },
  "language_info": {
   "codemirror_mode": {
    "name": "ipython",
    "version": 3
   },
   "file_extension": ".py",
   "mimetype": "text/x-python",
   "name": "python",
   "nbconvert_exporter": "python",
   "pygments_lexer": "ipython3",
   "version": "3.11.0"
  }
 },
 "nbformat": 4,
 "nbformat_minor": 5
}
