{
 "cells": [
  {
   "cell_type": "markdown",
   "id": "8947d045-0751-4f6e-841b-0e22c113f119",
   "metadata": {},
   "source": [
    "# Read Counts "
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 273,
   "id": "720f842f-be43-441a-8ab7-6cfedbf1e3ba",
   "metadata": {},
   "outputs": [],
   "source": [
    "# !pip install biopython"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 274,
   "id": "a12ddfa3-8a40-43d6-8928-b7f24d6505eb",
   "metadata": {},
   "outputs": [],
   "source": [
    "import os\n",
    "import gzip\n",
    "from Bio import SeqIO\n",
    "import pandas as pd\n",
    "import re\n",
    "import numpy as np\n",
    "# change directory to location of raw seqs\n",
    "os.chdir(\"/project/pi_sarah_gignouxwolfsohn_uml_edu/Raw_sequences/SCTLD_raw/BEL/renamed\")"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 275,
   "id": "211673fc-2593-439f-9fcc-233be4177e53",
   "metadata": {},
   "outputs": [],
   "source": [
    "os.chdir('/project/pi_sarah_gignouxwolfsohn_uml_edu/brooke/mcav/trimmed/redo_auto_detect_01312024/')"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 276,
   "id": "3faa6e40-54b0-485b-b837-dd1991741cb6",
   "metadata": {},
   "outputs": [],
   "source": [
    "cutadapt=pd.read_csv(\"multiqc_data_1/multiqc_cutadapt.txt\", sep=\"\\t\")"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 277,
   "id": "83e38e9e-e46e-4770-9af0-e78eca1d93dc",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>Sample</th>\n",
       "      <th>cutadapt_version</th>\n",
       "      <th>r_processed</th>\n",
       "      <th>r_with_adapters</th>\n",
       "      <th>r_written</th>\n",
       "      <th>bp_processed</th>\n",
       "      <th>quality_trimmed</th>\n",
       "      <th>bp_written</th>\n",
       "      <th>percent_trimmed</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>052022_BEL_CBC_T1_12_MCAV_R1_001</td>\n",
       "      <td>2.6</td>\n",
       "      <td>20754538</td>\n",
       "      <td>7948694</td>\n",
       "      <td>20754538</td>\n",
       "      <td>3133935238</td>\n",
       "      <td>7112099</td>\n",
       "      <td>3094510566</td>\n",
       "      <td>1.257993</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>052022_BEL_CBC_T1_12_MCAV_R2_001</td>\n",
       "      <td>2.6</td>\n",
       "      <td>20754538</td>\n",
       "      <td>7879397</td>\n",
       "      <td>20754538</td>\n",
       "      <td>3133935238</td>\n",
       "      <td>8334277</td>\n",
       "      <td>3092034192</td>\n",
       "      <td>1.337011</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>052022_BEL_CBC_T1_13_MCAV_R1_001</td>\n",
       "      <td>2.6</td>\n",
       "      <td>20870396</td>\n",
       "      <td>8052009</td>\n",
       "      <td>20870396</td>\n",
       "      <td>3151429796</td>\n",
       "      <td>5025643</td>\n",
       "      <td>3102979311</td>\n",
       "      <td>1.537413</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>052022_BEL_CBC_T1_13_MCAV_R2_001</td>\n",
       "      <td>2.6</td>\n",
       "      <td>20870396</td>\n",
       "      <td>8007983</td>\n",
       "      <td>20870396</td>\n",
       "      <td>3151429796</td>\n",
       "      <td>5313551</td>\n",
       "      <td>3101964385</td>\n",
       "      <td>1.569618</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>052022_BEL_CBC_T1_40_MCAV_R1_001</td>\n",
       "      <td>2.6</td>\n",
       "      <td>15425540</td>\n",
       "      <td>6062769</td>\n",
       "      <td>15425540</td>\n",
       "      <td>2329256540</td>\n",
       "      <td>5763468</td>\n",
       "      <td>2296536812</td>\n",
       "      <td>1.404728</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "                             Sample  cutadapt_version  r_processed  \\\n",
       "0  052022_BEL_CBC_T1_12_MCAV_R1_001               2.6     20754538   \n",
       "1  052022_BEL_CBC_T1_12_MCAV_R2_001               2.6     20754538   \n",
       "2  052022_BEL_CBC_T1_13_MCAV_R1_001               2.6     20870396   \n",
       "3  052022_BEL_CBC_T1_13_MCAV_R2_001               2.6     20870396   \n",
       "4  052022_BEL_CBC_T1_40_MCAV_R1_001               2.6     15425540   \n",
       "\n",
       "   r_with_adapters  r_written  bp_processed  quality_trimmed  bp_written  \\\n",
       "0          7948694   20754538    3133935238          7112099  3094510566   \n",
       "1          7879397   20754538    3133935238          8334277  3092034192   \n",
       "2          8052009   20870396    3151429796          5025643  3102979311   \n",
       "3          8007983   20870396    3151429796          5313551  3101964385   \n",
       "4          6062769   15425540    2329256540          5763468  2296536812   \n",
       "\n",
       "   percent_trimmed  \n",
       "0         1.257993  \n",
       "1         1.337011  \n",
       "2         1.537413  \n",
       "3         1.569618  \n",
       "4         1.404728  "
      ]
     },
     "execution_count": 277,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "cutadapt.head()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 278,
   "id": "3e5ee7ba-967e-4069-9edd-4690d6adce93",
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "/project/pi_sarah_gignouxwolfsohn_uml_edu/brooke/mcav/trimmed/redo_auto_detect_01312024/multiqc_data_1\n"
     ]
    }
   ],
   "source": [
    "cd multiqc_data_1/"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 279,
   "id": "84f4a220-4e42-4b81-af20-db491df160d7",
   "metadata": {},
   "outputs": [],
   "source": [
    "fastqc=pd.read_csv(\"multiqc_fastqc.txt\", sep=\"\\t\")"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 280,
   "id": "3dfb8d6c-bddd-4ab4-a7be-8ce653443433",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>Sample</th>\n",
       "      <th>Filename</th>\n",
       "      <th>File type</th>\n",
       "      <th>Encoding</th>\n",
       "      <th>Total Sequences</th>\n",
       "      <th>Total Bases</th>\n",
       "      <th>Sequences flagged as poor quality</th>\n",
       "      <th>Sequence length</th>\n",
       "      <th>%GC</th>\n",
       "      <th>total_deduplicated_percentage</th>\n",
       "      <th>...</th>\n",
       "      <th>per_base_sequence_quality</th>\n",
       "      <th>per_tile_sequence_quality</th>\n",
       "      <th>per_sequence_quality_scores</th>\n",
       "      <th>per_base_sequence_content</th>\n",
       "      <th>per_sequence_gc_content</th>\n",
       "      <th>per_base_n_content</th>\n",
       "      <th>sequence_length_distribution</th>\n",
       "      <th>sequence_duplication_levels</th>\n",
       "      <th>overrepresented_sequences</th>\n",
       "      <th>adapter_content</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>052022_BEL_CBC_T1_12_MCAV_R1_001_val_1</td>\n",
       "      <td>052022_BEL_CBC_T1_12_MCAV_R1_001_val_1.fq</td>\n",
       "      <td>Conventional base calls</td>\n",
       "      <td>Sanger / Illumina 1.9</td>\n",
       "      <td>20742228.0</td>\n",
       "      <td>3 Gbp</td>\n",
       "      <td>0.0</td>\n",
       "      <td>20-151</td>\n",
       "      <td>41.0</td>\n",
       "      <td>85.446362</td>\n",
       "      <td>...</td>\n",
       "      <td>pass</td>\n",
       "      <td>warn</td>\n",
       "      <td>pass</td>\n",
       "      <td>fail</td>\n",
       "      <td>pass</td>\n",
       "      <td>pass</td>\n",
       "      <td>warn</td>\n",
       "      <td>pass</td>\n",
       "      <td>pass</td>\n",
       "      <td>pass</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>052022_BEL_CBC_T1_12_MCAV_R2_001_val_2</td>\n",
       "      <td>052022_BEL_CBC_T1_12_MCAV_R2_001_val_2.fq</td>\n",
       "      <td>Conventional base calls</td>\n",
       "      <td>Sanger / Illumina 1.9</td>\n",
       "      <td>20742228.0</td>\n",
       "      <td>3 Gbp</td>\n",
       "      <td>0.0</td>\n",
       "      <td>20-151</td>\n",
       "      <td>41.0</td>\n",
       "      <td>86.714436</td>\n",
       "      <td>...</td>\n",
       "      <td>pass</td>\n",
       "      <td>pass</td>\n",
       "      <td>pass</td>\n",
       "      <td>fail</td>\n",
       "      <td>pass</td>\n",
       "      <td>pass</td>\n",
       "      <td>warn</td>\n",
       "      <td>pass</td>\n",
       "      <td>pass</td>\n",
       "      <td>pass</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>052022_BEL_CBC_T1_13_MCAV_R1_001_val_1</td>\n",
       "      <td>052022_BEL_CBC_T1_13_MCAV_R1_001_val_1.fq</td>\n",
       "      <td>Conventional base calls</td>\n",
       "      <td>Sanger / Illumina 1.9</td>\n",
       "      <td>20861960.0</td>\n",
       "      <td>3.1 Gbp</td>\n",
       "      <td>0.0</td>\n",
       "      <td>20-151</td>\n",
       "      <td>44.0</td>\n",
       "      <td>87.619706</td>\n",
       "      <td>...</td>\n",
       "      <td>pass</td>\n",
       "      <td>warn</td>\n",
       "      <td>pass</td>\n",
       "      <td>fail</td>\n",
       "      <td>pass</td>\n",
       "      <td>pass</td>\n",
       "      <td>warn</td>\n",
       "      <td>pass</td>\n",
       "      <td>pass</td>\n",
       "      <td>pass</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>052022_BEL_CBC_T1_13_MCAV_R2_001_val_2</td>\n",
       "      <td>052022_BEL_CBC_T1_13_MCAV_R2_001_val_2.fq</td>\n",
       "      <td>Conventional base calls</td>\n",
       "      <td>Sanger / Illumina 1.9</td>\n",
       "      <td>20861960.0</td>\n",
       "      <td>3.1 Gbp</td>\n",
       "      <td>0.0</td>\n",
       "      <td>20-151</td>\n",
       "      <td>44.0</td>\n",
       "      <td>88.271117</td>\n",
       "      <td>...</td>\n",
       "      <td>pass</td>\n",
       "      <td>pass</td>\n",
       "      <td>pass</td>\n",
       "      <td>fail</td>\n",
       "      <td>pass</td>\n",
       "      <td>pass</td>\n",
       "      <td>warn</td>\n",
       "      <td>pass</td>\n",
       "      <td>pass</td>\n",
       "      <td>pass</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>052022_BEL_CBC_T1_40_MCAV_R1_001_val_1</td>\n",
       "      <td>052022_BEL_CBC_T1_40_MCAV_R1_001_val_1.fq</td>\n",
       "      <td>Conventional base calls</td>\n",
       "      <td>Sanger / Illumina 1.9</td>\n",
       "      <td>15419779.0</td>\n",
       "      <td>2.2 Gbp</td>\n",
       "      <td>0.0</td>\n",
       "      <td>20-151</td>\n",
       "      <td>41.0</td>\n",
       "      <td>84.377874</td>\n",
       "      <td>...</td>\n",
       "      <td>pass</td>\n",
       "      <td>warn</td>\n",
       "      <td>pass</td>\n",
       "      <td>fail</td>\n",
       "      <td>pass</td>\n",
       "      <td>pass</td>\n",
       "      <td>warn</td>\n",
       "      <td>pass</td>\n",
       "      <td>pass</td>\n",
       "      <td>pass</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>5 rows × 23 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "                                   Sample  \\\n",
       "0  052022_BEL_CBC_T1_12_MCAV_R1_001_val_1   \n",
       "1  052022_BEL_CBC_T1_12_MCAV_R2_001_val_2   \n",
       "2  052022_BEL_CBC_T1_13_MCAV_R1_001_val_1   \n",
       "3  052022_BEL_CBC_T1_13_MCAV_R2_001_val_2   \n",
       "4  052022_BEL_CBC_T1_40_MCAV_R1_001_val_1   \n",
       "\n",
       "                                    Filename                File type  \\\n",
       "0  052022_BEL_CBC_T1_12_MCAV_R1_001_val_1.fq  Conventional base calls   \n",
       "1  052022_BEL_CBC_T1_12_MCAV_R2_001_val_2.fq  Conventional base calls   \n",
       "2  052022_BEL_CBC_T1_13_MCAV_R1_001_val_1.fq  Conventional base calls   \n",
       "3  052022_BEL_CBC_T1_13_MCAV_R2_001_val_2.fq  Conventional base calls   \n",
       "4  052022_BEL_CBC_T1_40_MCAV_R1_001_val_1.fq  Conventional base calls   \n",
       "\n",
       "                Encoding  Total Sequences Total Bases  \\\n",
       "0  Sanger / Illumina 1.9       20742228.0       3 Gbp   \n",
       "1  Sanger / Illumina 1.9       20742228.0       3 Gbp   \n",
       "2  Sanger / Illumina 1.9       20861960.0     3.1 Gbp   \n",
       "3  Sanger / Illumina 1.9       20861960.0     3.1 Gbp   \n",
       "4  Sanger / Illumina 1.9       15419779.0     2.2 Gbp   \n",
       "\n",
       "   Sequences flagged as poor quality Sequence length   %GC  \\\n",
       "0                                0.0          20-151  41.0   \n",
       "1                                0.0          20-151  41.0   \n",
       "2                                0.0          20-151  44.0   \n",
       "3                                0.0          20-151  44.0   \n",
       "4                                0.0          20-151  41.0   \n",
       "\n",
       "   total_deduplicated_percentage  ...  per_base_sequence_quality  \\\n",
       "0                      85.446362  ...                       pass   \n",
       "1                      86.714436  ...                       pass   \n",
       "2                      87.619706  ...                       pass   \n",
       "3                      88.271117  ...                       pass   \n",
       "4                      84.377874  ...                       pass   \n",
       "\n",
       "   per_tile_sequence_quality per_sequence_quality_scores  \\\n",
       "0                       warn                        pass   \n",
       "1                       pass                        pass   \n",
       "2                       warn                        pass   \n",
       "3                       pass                        pass   \n",
       "4                       warn                        pass   \n",
       "\n",
       "  per_base_sequence_content per_sequence_gc_content per_base_n_content  \\\n",
       "0                      fail                    pass               pass   \n",
       "1                      fail                    pass               pass   \n",
       "2                      fail                    pass               pass   \n",
       "3                      fail                    pass               pass   \n",
       "4                      fail                    pass               pass   \n",
       "\n",
       "  sequence_length_distribution sequence_duplication_levels  \\\n",
       "0                         warn                        pass   \n",
       "1                         warn                        pass   \n",
       "2                         warn                        pass   \n",
       "3                         warn                        pass   \n",
       "4                         warn                        pass   \n",
       "\n",
       "  overrepresented_sequences adapter_content  \n",
       "0                      pass            pass  \n",
       "1                      pass            pass  \n",
       "2                      pass            pass  \n",
       "3                      pass            pass  \n",
       "4                      pass            pass  \n",
       "\n",
       "[5 rows x 23 columns]"
      ]
     },
     "execution_count": 280,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "fastqc.head()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 281,
   "id": "4bca9f6f-e2a4-4ca8-8569-6038d706ccbe",
   "metadata": {},
   "outputs": [],
   "source": [
    "# take num of reads processed from cutadapt as raw reads (cutadapt was first step, so these are raw # reads)\n",
    "    # all reads were kept in cutadapt so not including that in the table for now\n",
    "# num of reads post qc filtering: 'Total Sequences' from fastqc (multiqc output txt file) which is run on RAW seqs \n",
    "\n",
    "\n",
    "# for F & R for each sample, extract fastqc$Total Sequences and cutadapt$r_written\n",
    "# edit fastqc sample names (remove val_1)\n",
    "fastqc['Sample'] = fastqc['Sample'].str.replace(r'_val_.*', '', regex=True)\n",
    "# add r_processed and r_written to fastqc df \n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 282,
   "id": "f8031058-e0f3-42d3-8b2a-487e80897532",
   "metadata": {},
   "outputs": [],
   "source": [
    "fastqc['Sample'] = fastqc['Sample'].astype(str)\n",
    "cutadapt['Sample'] = cutadapt['Sample'].astype(str)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 283,
   "id": "2dda7284-a219-49ef-8033-cbe0ec60b674",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "Index(['Sample', 'Filename', 'File type', 'Encoding', 'Total Sequences',\n",
       "       'Total Bases', 'Sequences flagged as poor quality', 'Sequence length',\n",
       "       '%GC', 'total_deduplicated_percentage', 'avg_sequence_length',\n",
       "       'median_sequence_length', 'basic_statistics',\n",
       "       'per_base_sequence_quality', 'per_tile_sequence_quality',\n",
       "       'per_sequence_quality_scores', 'per_base_sequence_content',\n",
       "       'per_sequence_gc_content', 'per_base_n_content',\n",
       "       'sequence_length_distribution', 'sequence_duplication_levels',\n",
       "       'overrepresented_sequences', 'adapter_content'],\n",
       "      dtype='object')"
      ]
     },
     "execution_count": 283,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "fastqc.columns"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 284,
   "id": "27ac01a4-f024-4697-bf38-c45e4eb751aa",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>Sample</th>\n",
       "      <th>Filename</th>\n",
       "      <th>File type</th>\n",
       "      <th>Encoding</th>\n",
       "      <th>Total Sequences</th>\n",
       "      <th>Total Bases</th>\n",
       "      <th>Sequences flagged as poor quality</th>\n",
       "      <th>Sequence length</th>\n",
       "      <th>%GC</th>\n",
       "      <th>total_deduplicated_percentage</th>\n",
       "      <th>...</th>\n",
       "      <th>per_tile_sequence_quality</th>\n",
       "      <th>per_sequence_quality_scores</th>\n",
       "      <th>per_base_sequence_content</th>\n",
       "      <th>per_sequence_gc_content</th>\n",
       "      <th>per_base_n_content</th>\n",
       "      <th>sequence_length_distribution</th>\n",
       "      <th>sequence_duplication_levels</th>\n",
       "      <th>overrepresented_sequences</th>\n",
       "      <th>adapter_content</th>\n",
       "      <th>r_processed</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>052022_BEL_CBC_T1_12_MCAV_R1_001</td>\n",
       "      <td>052022_BEL_CBC_T1_12_MCAV_R1_001_val_1.fq</td>\n",
       "      <td>Conventional base calls</td>\n",
       "      <td>Sanger / Illumina 1.9</td>\n",
       "      <td>20742228.0</td>\n",
       "      <td>3 Gbp</td>\n",
       "      <td>0.0</td>\n",
       "      <td>20-151</td>\n",
       "      <td>41.0</td>\n",
       "      <td>85.446362</td>\n",
       "      <td>...</td>\n",
       "      <td>warn</td>\n",
       "      <td>pass</td>\n",
       "      <td>fail</td>\n",
       "      <td>pass</td>\n",
       "      <td>pass</td>\n",
       "      <td>warn</td>\n",
       "      <td>pass</td>\n",
       "      <td>pass</td>\n",
       "      <td>pass</td>\n",
       "      <td>20754538</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>052022_BEL_CBC_T1_12_MCAV_R2_001</td>\n",
       "      <td>052022_BEL_CBC_T1_12_MCAV_R2_001_val_2.fq</td>\n",
       "      <td>Conventional base calls</td>\n",
       "      <td>Sanger / Illumina 1.9</td>\n",
       "      <td>20742228.0</td>\n",
       "      <td>3 Gbp</td>\n",
       "      <td>0.0</td>\n",
       "      <td>20-151</td>\n",
       "      <td>41.0</td>\n",
       "      <td>86.714436</td>\n",
       "      <td>...</td>\n",
       "      <td>pass</td>\n",
       "      <td>pass</td>\n",
       "      <td>fail</td>\n",
       "      <td>pass</td>\n",
       "      <td>pass</td>\n",
       "      <td>warn</td>\n",
       "      <td>pass</td>\n",
       "      <td>pass</td>\n",
       "      <td>pass</td>\n",
       "      <td>20754538</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>052022_BEL_CBC_T1_13_MCAV_R1_001</td>\n",
       "      <td>052022_BEL_CBC_T1_13_MCAV_R1_001_val_1.fq</td>\n",
       "      <td>Conventional base calls</td>\n",
       "      <td>Sanger / Illumina 1.9</td>\n",
       "      <td>20861960.0</td>\n",
       "      <td>3.1 Gbp</td>\n",
       "      <td>0.0</td>\n",
       "      <td>20-151</td>\n",
       "      <td>44.0</td>\n",
       "      <td>87.619706</td>\n",
       "      <td>...</td>\n",
       "      <td>warn</td>\n",
       "      <td>pass</td>\n",
       "      <td>fail</td>\n",
       "      <td>pass</td>\n",
       "      <td>pass</td>\n",
       "      <td>warn</td>\n",
       "      <td>pass</td>\n",
       "      <td>pass</td>\n",
       "      <td>pass</td>\n",
       "      <td>20870396</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>052022_BEL_CBC_T1_13_MCAV_R2_001</td>\n",
       "      <td>052022_BEL_CBC_T1_13_MCAV_R2_001_val_2.fq</td>\n",
       "      <td>Conventional base calls</td>\n",
       "      <td>Sanger / Illumina 1.9</td>\n",
       "      <td>20861960.0</td>\n",
       "      <td>3.1 Gbp</td>\n",
       "      <td>0.0</td>\n",
       "      <td>20-151</td>\n",
       "      <td>44.0</td>\n",
       "      <td>88.271117</td>\n",
       "      <td>...</td>\n",
       "      <td>pass</td>\n",
       "      <td>pass</td>\n",
       "      <td>fail</td>\n",
       "      <td>pass</td>\n",
       "      <td>pass</td>\n",
       "      <td>warn</td>\n",
       "      <td>pass</td>\n",
       "      <td>pass</td>\n",
       "      <td>pass</td>\n",
       "      <td>20870396</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>052022_BEL_CBC_T1_40_MCAV_R1_001</td>\n",
       "      <td>052022_BEL_CBC_T1_40_MCAV_R1_001_val_1.fq</td>\n",
       "      <td>Conventional base calls</td>\n",
       "      <td>Sanger / Illumina 1.9</td>\n",
       "      <td>15419779.0</td>\n",
       "      <td>2.2 Gbp</td>\n",
       "      <td>0.0</td>\n",
       "      <td>20-151</td>\n",
       "      <td>41.0</td>\n",
       "      <td>84.377874</td>\n",
       "      <td>...</td>\n",
       "      <td>warn</td>\n",
       "      <td>pass</td>\n",
       "      <td>fail</td>\n",
       "      <td>pass</td>\n",
       "      <td>pass</td>\n",
       "      <td>warn</td>\n",
       "      <td>pass</td>\n",
       "      <td>pass</td>\n",
       "      <td>pass</td>\n",
       "      <td>15425540</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>5 rows × 24 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "                             Sample  \\\n",
       "0  052022_BEL_CBC_T1_12_MCAV_R1_001   \n",
       "1  052022_BEL_CBC_T1_12_MCAV_R2_001   \n",
       "2  052022_BEL_CBC_T1_13_MCAV_R1_001   \n",
       "3  052022_BEL_CBC_T1_13_MCAV_R2_001   \n",
       "4  052022_BEL_CBC_T1_40_MCAV_R1_001   \n",
       "\n",
       "                                    Filename                File type  \\\n",
       "0  052022_BEL_CBC_T1_12_MCAV_R1_001_val_1.fq  Conventional base calls   \n",
       "1  052022_BEL_CBC_T1_12_MCAV_R2_001_val_2.fq  Conventional base calls   \n",
       "2  052022_BEL_CBC_T1_13_MCAV_R1_001_val_1.fq  Conventional base calls   \n",
       "3  052022_BEL_CBC_T1_13_MCAV_R2_001_val_2.fq  Conventional base calls   \n",
       "4  052022_BEL_CBC_T1_40_MCAV_R1_001_val_1.fq  Conventional base calls   \n",
       "\n",
       "                Encoding  Total Sequences Total Bases  \\\n",
       "0  Sanger / Illumina 1.9       20742228.0       3 Gbp   \n",
       "1  Sanger / Illumina 1.9       20742228.0       3 Gbp   \n",
       "2  Sanger / Illumina 1.9       20861960.0     3.1 Gbp   \n",
       "3  Sanger / Illumina 1.9       20861960.0     3.1 Gbp   \n",
       "4  Sanger / Illumina 1.9       15419779.0     2.2 Gbp   \n",
       "\n",
       "   Sequences flagged as poor quality Sequence length   %GC  \\\n",
       "0                                0.0          20-151  41.0   \n",
       "1                                0.0          20-151  41.0   \n",
       "2                                0.0          20-151  44.0   \n",
       "3                                0.0          20-151  44.0   \n",
       "4                                0.0          20-151  41.0   \n",
       "\n",
       "   total_deduplicated_percentage  ...  per_tile_sequence_quality  \\\n",
       "0                      85.446362  ...                       warn   \n",
       "1                      86.714436  ...                       pass   \n",
       "2                      87.619706  ...                       warn   \n",
       "3                      88.271117  ...                       pass   \n",
       "4                      84.377874  ...                       warn   \n",
       "\n",
       "   per_sequence_quality_scores per_base_sequence_content  \\\n",
       "0                         pass                      fail   \n",
       "1                         pass                      fail   \n",
       "2                         pass                      fail   \n",
       "3                         pass                      fail   \n",
       "4                         pass                      fail   \n",
       "\n",
       "  per_sequence_gc_content per_base_n_content sequence_length_distribution  \\\n",
       "0                    pass               pass                         warn   \n",
       "1                    pass               pass                         warn   \n",
       "2                    pass               pass                         warn   \n",
       "3                    pass               pass                         warn   \n",
       "4                    pass               pass                         warn   \n",
       "\n",
       "  sequence_duplication_levels overrepresented_sequences adapter_content  \\\n",
       "0                        pass                      pass            pass   \n",
       "1                        pass                      pass            pass   \n",
       "2                        pass                      pass            pass   \n",
       "3                        pass                      pass            pass   \n",
       "4                        pass                      pass            pass   \n",
       "\n",
       "  r_processed  \n",
       "0    20754538  \n",
       "1    20754538  \n",
       "2    20870396  \n",
       "3    20870396  \n",
       "4    15425540  \n",
       "\n",
       "[5 rows x 24 columns]"
      ]
     },
     "execution_count": 284,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "merged = fastqc.merge(cutadapt[['Sample', 'r_processed']], on='Sample', how='left')\n",
    "# merged.drop(,axis=1\n",
    "merged.head()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 285,
   "id": "7b5bc357-8b9c-4e56-8a0d-7c9ba86f9b1d",
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "[0.]\n"
     ]
    },
    {
     "data": {
      "text/plain": [
       "array(['pass'], dtype=object)"
      ]
     },
     "execution_count": 285,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "print(merged['Sequences flagged as poor quality'].unique())\n",
    "merged['per_sequence_quality_scores'].unique()\n",
    "# all pass qc "
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 286,
   "id": "e15d9f6d-1eed-4c87-921e-f5eca0105096",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>Sample</th>\n",
       "      <th>Filename</th>\n",
       "      <th>Total Sequences</th>\n",
       "      <th>r_processed</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>052022_BEL_CBC_T1_12_MCAV_R1_001</td>\n",
       "      <td>052022_BEL_CBC_T1_12_MCAV_R1_001_val_1.fq</td>\n",
       "      <td>20742228.0</td>\n",
       "      <td>20754538</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>052022_BEL_CBC_T1_12_MCAV_R2_001</td>\n",
       "      <td>052022_BEL_CBC_T1_12_MCAV_R2_001_val_2.fq</td>\n",
       "      <td>20742228.0</td>\n",
       "      <td>20754538</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>052022_BEL_CBC_T1_13_MCAV_R1_001</td>\n",
       "      <td>052022_BEL_CBC_T1_13_MCAV_R1_001_val_1.fq</td>\n",
       "      <td>20861960.0</td>\n",
       "      <td>20870396</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>052022_BEL_CBC_T1_13_MCAV_R2_001</td>\n",
       "      <td>052022_BEL_CBC_T1_13_MCAV_R2_001_val_2.fq</td>\n",
       "      <td>20861960.0</td>\n",
       "      <td>20870396</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>052022_BEL_CBC_T1_40_MCAV_R1_001</td>\n",
       "      <td>052022_BEL_CBC_T1_40_MCAV_R1_001_val_1.fq</td>\n",
       "      <td>15419779.0</td>\n",
       "      <td>15425540</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "                             Sample  \\\n",
       "0  052022_BEL_CBC_T1_12_MCAV_R1_001   \n",
       "1  052022_BEL_CBC_T1_12_MCAV_R2_001   \n",
       "2  052022_BEL_CBC_T1_13_MCAV_R1_001   \n",
       "3  052022_BEL_CBC_T1_13_MCAV_R2_001   \n",
       "4  052022_BEL_CBC_T1_40_MCAV_R1_001   \n",
       "\n",
       "                                    Filename  Total Sequences  r_processed  \n",
       "0  052022_BEL_CBC_T1_12_MCAV_R1_001_val_1.fq       20742228.0     20754538  \n",
       "1  052022_BEL_CBC_T1_12_MCAV_R2_001_val_2.fq       20742228.0     20754538  \n",
       "2  052022_BEL_CBC_T1_13_MCAV_R1_001_val_1.fq       20861960.0     20870396  \n",
       "3  052022_BEL_CBC_T1_13_MCAV_R2_001_val_2.fq       20861960.0     20870396  \n",
       "4  052022_BEL_CBC_T1_40_MCAV_R1_001_val_1.fq       15419779.0     15425540  "
      ]
     },
     "execution_count": 286,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "reads=merged[['Sample','Filename','Total Sequences','r_processed']]\n",
    "reads.head()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 287,
   "id": "845a0b20-34d8-41f0-ad23-bbe7e1f5681b",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>Sample</th>\n",
       "      <th>Filename</th>\n",
       "      <th>post_qc</th>\n",
       "      <th>raw_seqs</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>052022_BEL_CBC_T1_12_MCAV</td>\n",
       "      <td>052022_BEL_CBC_T1_12_MCAV_R1_001_val_1.fq</td>\n",
       "      <td>20742228.0</td>\n",
       "      <td>20754538</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>052022_BEL_CBC_T1_12_MCAV</td>\n",
       "      <td>052022_BEL_CBC_T1_12_MCAV_R2_001_val_2.fq</td>\n",
       "      <td>20742228.0</td>\n",
       "      <td>20754538</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>052022_BEL_CBC_T1_13_MCAV</td>\n",
       "      <td>052022_BEL_CBC_T1_13_MCAV_R1_001_val_1.fq</td>\n",
       "      <td>20861960.0</td>\n",
       "      <td>20870396</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>052022_BEL_CBC_T1_13_MCAV</td>\n",
       "      <td>052022_BEL_CBC_T1_13_MCAV_R2_001_val_2.fq</td>\n",
       "      <td>20861960.0</td>\n",
       "      <td>20870396</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>052022_BEL_CBC_T1_40_MCAV</td>\n",
       "      <td>052022_BEL_CBC_T1_40_MCAV_R1_001_val_1.fq</td>\n",
       "      <td>15419779.0</td>\n",
       "      <td>15425540</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "                      Sample                                   Filename  \\\n",
       "0  052022_BEL_CBC_T1_12_MCAV  052022_BEL_CBC_T1_12_MCAV_R1_001_val_1.fq   \n",
       "1  052022_BEL_CBC_T1_12_MCAV  052022_BEL_CBC_T1_12_MCAV_R2_001_val_2.fq   \n",
       "2  052022_BEL_CBC_T1_13_MCAV  052022_BEL_CBC_T1_13_MCAV_R1_001_val_1.fq   \n",
       "3  052022_BEL_CBC_T1_13_MCAV  052022_BEL_CBC_T1_13_MCAV_R2_001_val_2.fq   \n",
       "4  052022_BEL_CBC_T1_40_MCAV  052022_BEL_CBC_T1_40_MCAV_R1_001_val_1.fq   \n",
       "\n",
       "      post_qc  raw_seqs  \n",
       "0  20742228.0  20754538  \n",
       "1  20742228.0  20754538  \n",
       "2  20861960.0  20870396  \n",
       "3  20861960.0  20870396  \n",
       "4  15419779.0  15425540  "
      ]
     },
     "execution_count": 287,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "# rename columns \n",
    "reads=reads.rename(columns={'Total Sequences':'post_qc',\n",
    "                      'r_processed':'raw_seqs'})\n",
    "                      #'Sample':'sample'\n",
    "# make sample id base id\n",
    "reads['Sample'] = reads['Sample'].str.replace(r'_R[12]_001', '', regex=True)\n",
    "reads.head()"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "e5c09560-9ccc-4755-a5a7-026388fd84c8",
   "metadata": {},
   "source": [
    "## Get reads post-filtering: host, symbiont, human\n",
    "- loop through samples in each set "
   ]
  },
  {
   "cell_type": "markdown",
   "id": "46335538-3c84-404f-9eac-ab230f3e599f",
   "metadata": {
    "tags": []
   },
   "source": [
    "### host: slurm output (job name 'removal' in 1Assembly nb)\n",
    "- pull out num of unmatched reads "
   ]
  },
  {
   "cell_type": "raw",
   "id": "535dd38c-d138-474b-822f-cea629a697d3",
   "metadata": {},
   "source": [
    "# example output in slurm file\n",
    "79877382 reads; of these:\n",
    "  79877382 (100.00%) were paired; of these:\n",
    "    79448919 (99.46%) aligned concordantly 0 times\n",
    "    72904 (0.09%) aligned concordantly exactly 1 time\n",
    "    355559 (0.45%) aligned concordantly >1 times\n",
    "    ----\n",
    "    79448919 pairs aligned concordantly 0 times; of these:\n",
    "      6809 (0.01%) aligned discordantly 1 time\n",
    "    ----\n",
    "    79442110 pairs aligned 0 times concordantly or discordantly; of these:\n",
    "      158884220 mates make up the pairs; of these:\n",
    "        158309389 (99.64%) aligned 0 times # this line is the leftover reads (host-removed)\n",
    "        171335 (0.11%) aligned exactly 1 time\n",
    "        403496 (0.25%) aligned >1 times\n",
    "0.90% overall alignment rate"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 288,
   "id": "143f2ac6-51b4-49b1-be33-e7818095cbfd",
   "metadata": {},
   "outputs": [],
   "source": [
    "# set dir\n",
    "os.chdir('/project/pi_sarah_gignouxwolfsohn_uml_edu/brooke/mcav/healthy_2019_mcav')"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 289,
   "id": "baa69009-c189-4e75-94f8-516e13e4da93",
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "ITS2_trim                                 \u001b[0m\u001b[01;34mmapping\u001b[0m/\n",
      "\u001b[01;34manvio\u001b[0m/                                    \u001b[01;34mmegahit_host_removed\u001b[0m/\n",
      "assembly                                  \u001b[01;34mquast_output\u001b[0m/\n",
      "\u001b[01;34mbinning\u001b[0m/                                  removal\n",
      "fastq_screen                              resis_mcav_otu.csv\n",
      "\u001b[01;34mfinal_reads_filtered\u001b[0m/                     slurm-19420673.out\n",
      "healthy_2019_MCAV_sampleids               slurm-assembly-19388626.out\n",
      "healthy_2019_mcav_filtered.contigs.fasta  slurm-fastq_screen-22809172.out\n",
      "healthy_2019_mcav_reads_R1_ALL.fastq      slurm-removal-19358781.out\n",
      "healthy_2019_mcav_reads_R2_ALL.fastq      \u001b[01;34mtaxonomy\u001b[0m/\n",
      "\u001b[01;34mhost_removed\u001b[0m/\n"
     ]
    }
   ],
   "source": [
    "ls"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 290,
   "id": "f26ad9d1-2654-4cfb-8d30-991cf32613cf",
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "                       Sample  percent_mapped_host  Unmapped_Reads\n",
      "0    062019_BEL_CBC_T3_8_MCAV                 0.90      79154694.5\n",
      "1   062019_BEL_CBC_T2_16_MCAV                84.17       3151347.5\n",
      "2   062019_BEL_CBC_T1_20_MCAV                 0.70       8458232.5\n",
      "3   062019_BEL_CBC_T1_24_MCAV                76.09       4516033.5\n",
      "4    062019_BEL_CBC_T3_9_MCAV                67.44       4568457.5\n",
      "5   062019_BEL_CBC_T3_14_MCAV                79.74       4335512.5\n",
      "6    062019_BEL_CBC_T3_1_MCAV                74.96       7505458.5\n",
      "7   062019_BEL_CBC_T3_15_MCAV                84.49       3078684.5\n",
      "8   062019_BEL_CBC_T1_16_MCAV                68.49       8131957.5\n",
      "9   062019_BEL_CBC_T3_16_MCAV                67.97       1134331.5\n",
      "10   062019_BEL_CBC_T1_3_MCAV                32.46       3190834.0\n",
      "11   062019_BEL_CBC_T2_5_MCAV                62.04       9711853.0\n",
      "12   062019_BEL_CBC_T3_6_MCAV                23.19       1876312.0\n"
     ]
    }
   ],
   "source": [
    "sample_file='healthy_2019_MCAV_sampleids'\n",
    "slurm_output='slurm-removal-19358781.out'\n",
    "\n",
    "# Read sample IDs in order\n",
    "with open(sample_file, \"r\") as f:\n",
    "    sample_ids = [line.strip() for line in f if line.strip()]\n",
    "\n",
    "# Read SLURM output\n",
    "with open(slurm_output, \"r\") as f:\n",
    "    slurm_data = f.read()\n",
    "\n",
    "# Split logs to get read counts for each sample\n",
    "log_blocks = slurm_data.split(\"reads; of these:\")\n",
    "\n",
    "# first split chunk is SLURM noise, so discard it\n",
    "log_blocks = log_blocks[1:]\n",
    "\n",
    "# Prepend \"reads; of these:\" back to the actual logs\n",
    "log_blocks = [\"reads; of these:\" + block for block in log_blocks]\n",
    "\n",
    "# if # of samples doesn't match log outputs, print and compare\n",
    "if len(log_blocks) != len(sample_ids):\n",
    "    print(\"Warning: Number of logs and samples do not match!\")\n",
    "    print(f\"Samples: {len(sample_ids)}\")\n",
    "    print(f\"Logs:    {len(log_blocks)}\\n\")\n",
    "\n",
    "    # Show side-by-side preview\n",
    "    for i, (sample, log) in enumerate(zip(sample_ids, log_blocks)):\n",
    "        print(f\"{i}: Sample: {sample} | Log starts with: {log[:50]}\")\n",
    "\n",
    "# Extract unmapped reads \n",
    "unmapped_counts = []\n",
    "for block in log_blocks:\n",
    "    match = re.search(r\"(\\d[\\d,]*) \\([\\d.]+%\\) aligned 0 times\", block)\n",
    "    #print(match)\n",
    "    if match:\n",
    "        seqs = int(match.group(1).replace(\",\", \"\"))\n",
    "        unmapped_counts.append(seqs)\n",
    "    else:\n",
    "        unmapped_counts.append(None)\n",
    "\n",
    "# Extract % mapped \n",
    "mapped_percent = []\n",
    "for block in log_blocks:\n",
    "    match = re.search(r\"([\\d.]+)% overall alignment rate\", block)\n",
    "    #print(match)\n",
    "    if match:\n",
    "        percent = float(match.group(1))\n",
    "        mapped_percent.append(percent)\n",
    "    else:\n",
    "        mapped_percent.append(None)\n",
    "\n",
    "\n",
    "# Create dataframe\n",
    "df = pd.DataFrame({\n",
    "    \"Sample\": sample_ids,\n",
    "    \"percent_mapped_host\": mapped_percent,\n",
    "    \"Unmapped_Reads\": unmapped_counts\n",
    "})\n",
    "\n",
    "# divide numbers in half since they are calculating with F & R together \n",
    "df['Unmapped_Reads']=df['Unmapped_Reads']/2\n",
    "# don't know if need to round..\n",
    "\n",
    "print(df)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 291,
   "id": "d810b672-9cde-44c0-b89c-a69655fae284",
   "metadata": {},
   "outputs": [],
   "source": [
    "# append to reads df \n",
    "reads = reads.merge(df[['Sample','Unmapped_Reads','percent_mapped_host']], on='Sample', how='left')"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 292,
   "id": "07d65a93-2d34-4e5e-ab50-7df59fa46721",
   "metadata": {},
   "outputs": [],
   "source": [
    "# repeat for next sample set \n",
    "#"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 293,
   "id": "5146af1e-3a79-497e-81ec-59f68cff30cb",
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "                      Sample  percent_mapped_host  Unmapped_Reads\n",
      "0  052022_BEL_CBC_T3_13_MCAV                78.10       2945179.5\n",
      "1   052022_BEL_CBC_T3_1_MCAV                22.89       4495655.5\n",
      "2  052022_BEL_CBC_T3_19_MCAV                77.85       2003485.5\n",
      "3  052022_BEL_CBC_T2_10_MCAV                30.69      11697348.5\n",
      "4  052022_BEL_CBC_T1_57_MCAV                71.43       9131401.0\n"
     ]
    }
   ],
   "source": [
    "sample_path='/project/pi_sarah_gignouxwolfsohn_uml_edu/brooke/mcav/healthy_2022_mcav'\n",
    "sample_file='healthy_2022_MCAV'\n",
    "sample_file = os.path.join(sample_path, sample_file)\n",
    "\n",
    "slurm_path='/work/pi_sarah_gignouxwolfsohn_uml_edu/brooke/job_files'\n",
    "slurm_output='slurm-assembly-23178009.out'\n",
    "slurm_output = os.path.join(slurm_path, slurm_output)\n",
    "\n",
    "# Read sample IDs in order\n",
    "with open(sample_file, \"r\") as f:\n",
    "    sample_ids = [line.strip() for line in f if line.strip()]\n",
    "\n",
    "# Read SLURM output\n",
    "with open(slurm_output, \"r\") as f:\n",
    "    slurm_data = f.read()\n",
    "\n",
    "# Split logs to get read counts for each sample\n",
    "log_blocks = slurm_data.split(\"reads; of these:\")\n",
    "\n",
    "# first split chunk is SLURM noise, so discard it\n",
    "log_blocks = log_blocks[1:]\n",
    "\n",
    "# Prepend \"reads; of these:\" back to the actual logs\n",
    "log_blocks = [\"reads; of these:\" + block for block in log_blocks]\n",
    "\n",
    "# if # of samples doesn't match log outputs, print and compare\n",
    "if len(log_blocks) != len(sample_ids):\n",
    "    print(\"Warning: Number of logs and samples do not match!\")\n",
    "    print(f\"Samples: {len(sample_ids)}\")\n",
    "    print(f\"Logs:    {len(log_blocks)}\\n\")\n",
    "\n",
    "    # Show side-by-side preview\n",
    "    for i, (sample, log) in enumerate(zip(sample_ids, log_blocks)):\n",
    "        print(f\"{i}: Sample: {sample} | Log starts with: {log[:50]}\")\n",
    "\n",
    "# Extract unmapped reads \n",
    "unmapped_counts = []\n",
    "for block in log_blocks:\n",
    "    match = re.search(r\"(\\d[\\d,]*) \\([\\d.]+%\\) aligned 0 times\", block)\n",
    "    #print(match)\n",
    "    if match:\n",
    "        seqs = int(match.group(1).replace(\",\", \"\"))\n",
    "        unmapped_counts.append(seqs)\n",
    "    else:\n",
    "        unmapped_counts.append(None)\n",
    "\n",
    "# Extract % mapped \n",
    "mapped_percent = []\n",
    "for block in log_blocks:\n",
    "    match = re.search(r\"([\\d.]+)% overall alignment rate\", block)\n",
    "    #print(match)\n",
    "    if match:\n",
    "        percent = float(match.group(1))\n",
    "        mapped_percent.append(percent)\n",
    "    else:\n",
    "        mapped_percent.append(None)\n",
    "\n",
    "\n",
    "# Create dataframe\n",
    "df = pd.DataFrame({\n",
    "    \"Sample\": sample_ids,\n",
    "    \"percent_mapped_host\": mapped_percent,\n",
    "    \"Unmapped_Reads\": unmapped_counts\n",
    "})\n",
    "\n",
    "# divide numbers in half since they are calculating with F & R together \n",
    "df['Unmapped_Reads']=df['Unmapped_Reads']/2\n",
    "\n",
    "print(df)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 294,
   "id": "cd55fee4-7643-4c63-a686-94f06a50d99a",
   "metadata": {},
   "outputs": [],
   "source": [
    "# append to reads df \n",
    "# fill in NAs of unampped reads in reads df with new values by matching sample id \n",
    "reads['Unmapped_Reads'] = reads['Sample'].map(df.set_index('Sample')\n",
    "                                              ['Unmapped_Reads']).fillna(reads['Unmapped_Reads'])             "
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 295,
   "id": "67c0c19f-8b45-40d8-84a7-4e5aa19bafa3",
   "metadata": {},
   "outputs": [],
   "source": [
    "# append to reads df \n",
    "# fill in NAs of unampped reads in reads df with new values by matching sample id \n",
    "reads['percent_mapped_host'] = reads['Sample'].map(df.set_index('Sample')\n",
    "                                              ['percent_mapped_host']).fillna(reads['percent_mapped_host'])"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 296,
   "id": "51933e3c-c4ba-487c-a566-3bedb24b33a0",
   "metadata": {},
   "outputs": [],
   "source": [
    "# # repeat with diseased \n",
    "# sample_path='/project/pi_sarah_gignouxwolfsohn_uml_edu/brooke/mcav/diseased_mcav'\n",
    "# sample_file='diseased_MCAV'\n",
    "# sample_file = os.path.join(sample_path, sample_file)\n",
    "\n",
    "# slurm_path='/project/pi_sarah_gignouxwolfsohn_uml_edu/brooke/mcav/diseased_mcav/assembly'\n",
    "# slurm_output='slurm-removal-21639875.out'\n",
    "# slurm_output = os.path.join(slurm_path, slurm_output)\n",
    "\n",
    "# # Read sample IDs in order\n",
    "# with open(sample_file, \"r\") as f:\n",
    "#     sample_ids = [line.strip() for line in f if line.strip()]\n",
    "\n",
    "# # Read SLURM output\n",
    "# with open(slurm_output, \"r\") as f:\n",
    "#     slurm_data = f.read()\n",
    "\n",
    "# # Split logs to get read counts for each sample\n",
    "# log_blocks = slurm_data.split(\"reads; of these:\")\n",
    "\n",
    "# # first split chunk is SLURM noise, so discard it\n",
    "# log_blocks = log_blocks[1:]\n",
    "\n",
    "# # Prepend \"reads; of these:\" back to the actual logs\n",
    "# log_blocks = [\"reads; of these:\" + block for block in log_blocks]\n",
    "\n",
    "# # if # of samples doesn't match log outputs, print and compare\n",
    "# if len(log_blocks) != len(sample_ids):\n",
    "#     print(\"Warning: Number of logs and samples do not match!\")\n",
    "#     print(f\"Samples: {len(sample_ids)}\")\n",
    "#     print(f\"Logs:    {len(log_blocks)}\\n\")\n",
    "\n",
    "#     # Show side-by-side preview\n",
    "#     for i, (sample, log) in enumerate(zip(sample_ids, log_blocks)):\n",
    "#         print(f\"{i}: Sample: {sample} | Log starts with: {log[:50]}\")\n",
    "\n",
    "# # Extract unmapped reads \n",
    "# unmapped_counts = []\n",
    "# for block in log_blocks:\n",
    "#     match = re.search(r\"(\\d[\\d,]*) \\([\\d.]+%\\) aligned 0 times\", block)\n",
    "#     #print(match)\n",
    "#     if match:\n",
    "#         seqs = int(match.group(1).replace(\",\", \"\"))\n",
    "#         unmapped_counts.append(seqs)\n",
    "#     else:\n",
    "#         unmapped_counts.append(None)\n",
    "\n",
    "# # Extract % mapped \n",
    "# mapped_percent = []\n",
    "# for block in log_blocks:\n",
    "#     match = re.search(r\"([\\d.]+)% overall alignment rate\", block)\n",
    "#     #print(match)\n",
    "#     if match:\n",
    "#         percent = float(match.group(1))\n",
    "#         mapped_percent.append(percent)\n",
    "#     else:\n",
    "#         mapped_percent.append(None)\n",
    "\n",
    "\n",
    "# # Create dataframe\n",
    "# df = pd.DataFrame({\n",
    "#     \"Sample\": sample_ids,\n",
    "#     \"percent_mapped_host\": mapped_percent,\n",
    "#     \"Unmapped_Reads\": unmapped_counts\n",
    "# })\n",
    "\n",
    "# # divide numbers in half since they are calculating with F & R together \n",
    "# df['Unmapped_Reads']=df['Unmapped_Reads']/2\n",
    "\n",
    "# print(df)\n",
    "\n",
    "# i must've messed this up or deleted somehting on accident..see previous versions of repo to see this data. \n",
    "# downloading this sample set seperately  "
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 297,
   "id": "fac0f3d7-6992-4579-ac7c-119e7a75ba5b",
   "metadata": {},
   "outputs": [],
   "source": [
    "# append to reads df \n",
    "# fill in NAs of unampped reads in reads df with new values by matching sample id \n",
    "# reads['Unmapped_Reads'] = reads['Sample'].map(df.set_index('Sample')\n",
    "#                                               ['Unmapped_Reads']).fillna(reads['Unmapped_Reads'])"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 298,
   "id": "9715a875-1954-40ef-a1b5-740739c15736",
   "metadata": {},
   "outputs": [],
   "source": [
    "# append to reads df \n",
    "# fill in NAs of unampped reads in reads df with new values by matching sample id \n",
    "# reads['percent_mapped_host'] = reads['Sample'].map(df.set_index('Sample')\n",
    "#                                               ['percent_mapped_host']).fillna(reads['percent_mapped_host'])"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 299,
   "id": "00a07c7a-8748-4c95-af07-53bed258e177",
   "metadata": {},
   "outputs": [],
   "source": [
    "# cant find margin mcav so using multiqc results from host removed"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 300,
   "id": "8b0587f2-44a4-43de-a339-5476451d7497",
   "metadata": {},
   "outputs": [],
   "source": [
    "os.chdir('/project/pi_sarah_gignouxwolfsohn_uml_edu/brooke/mcav/margin_mcav/assembly/host_removed/multiqc_data')"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 301,
   "id": "2b1a71f9-d71c-4d87-b348-7efbf55e8dcc",
   "metadata": {},
   "outputs": [],
   "source": [
    "# view sample list"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 302,
   "id": "a28f3aae-d59e-4a00-9556-4ee564c80753",
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "052022_BEL_CBC_T1_12_MCAV\n",
      "052022_BEL_CBC_T1_40_MCAV\n",
      "052022_BEL_CBC_T3_48_MCAV\n",
      "052022_BEL_CBC_T3_40_MCAV\n"
     ]
    }
   ],
   "source": [
    "cat /project/pi_sarah_gignouxwolfsohn_uml_edu/brooke/mcav/margin_mcav/margin_MCAV"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 303,
   "id": "b8ec0994-8c19-4101-9795-8a494bf76af3",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>Sample</th>\n",
       "      <th>Filename</th>\n",
       "      <th>post_qc</th>\n",
       "      <th>raw_seqs</th>\n",
       "      <th>Unmapped_Reads</th>\n",
       "      <th>percent_mapped_host</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>052022_BEL_CBC_T1_12_MCAV</td>\n",
       "      <td>052022_BEL_CBC_T1_12_MCAV_R1_001_val_1.fq</td>\n",
       "      <td>20742228.0</td>\n",
       "      <td>20754538</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>052022_BEL_CBC_T1_12_MCAV</td>\n",
       "      <td>052022_BEL_CBC_T1_12_MCAV_R2_001_val_2.fq</td>\n",
       "      <td>20742228.0</td>\n",
       "      <td>20754538</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>052022_BEL_CBC_T1_13_MCAV</td>\n",
       "      <td>052022_BEL_CBC_T1_13_MCAV_R1_001_val_1.fq</td>\n",
       "      <td>20861960.0</td>\n",
       "      <td>20870396</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>052022_BEL_CBC_T1_13_MCAV</td>\n",
       "      <td>052022_BEL_CBC_T1_13_MCAV_R2_001_val_2.fq</td>\n",
       "      <td>20861960.0</td>\n",
       "      <td>20870396</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>052022_BEL_CBC_T1_40_MCAV</td>\n",
       "      <td>052022_BEL_CBC_T1_40_MCAV_R1_001_val_1.fq</td>\n",
       "      <td>15419779.0</td>\n",
       "      <td>15425540</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "                      Sample                                   Filename  \\\n",
       "0  052022_BEL_CBC_T1_12_MCAV  052022_BEL_CBC_T1_12_MCAV_R1_001_val_1.fq   \n",
       "1  052022_BEL_CBC_T1_12_MCAV  052022_BEL_CBC_T1_12_MCAV_R2_001_val_2.fq   \n",
       "2  052022_BEL_CBC_T1_13_MCAV  052022_BEL_CBC_T1_13_MCAV_R1_001_val_1.fq   \n",
       "3  052022_BEL_CBC_T1_13_MCAV  052022_BEL_CBC_T1_13_MCAV_R2_001_val_2.fq   \n",
       "4  052022_BEL_CBC_T1_40_MCAV  052022_BEL_CBC_T1_40_MCAV_R1_001_val_1.fq   \n",
       "\n",
       "      post_qc  raw_seqs  Unmapped_Reads  percent_mapped_host  \n",
       "0  20742228.0  20754538             NaN                  NaN  \n",
       "1  20742228.0  20754538             NaN                  NaN  \n",
       "2  20861960.0  20870396             NaN                  NaN  \n",
       "3  20861960.0  20870396             NaN                  NaN  \n",
       "4  15419779.0  15425540             NaN                  NaN  "
      ]
     },
     "execution_count": 303,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "reads.head()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 304,
   "id": "0bac6503-2073-4c33-9384-417347ef8425",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>Sample</th>\n",
       "      <th>Filename</th>\n",
       "      <th>File type</th>\n",
       "      <th>Encoding</th>\n",
       "      <th>Total Sequences</th>\n",
       "      <th>Total Bases</th>\n",
       "      <th>Sequences flagged as poor quality</th>\n",
       "      <th>Sequence length</th>\n",
       "      <th>%GC</th>\n",
       "      <th>total_deduplicated_percentage</th>\n",
       "      <th>...</th>\n",
       "      <th>per_base_sequence_quality</th>\n",
       "      <th>per_tile_sequence_quality</th>\n",
       "      <th>per_sequence_quality_scores</th>\n",
       "      <th>per_base_sequence_content</th>\n",
       "      <th>per_sequence_gc_content</th>\n",
       "      <th>per_base_n_content</th>\n",
       "      <th>sequence_length_distribution</th>\n",
       "      <th>sequence_duplication_levels</th>\n",
       "      <th>overrepresented_sequences</th>\n",
       "      <th>adapter_content</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>052022_BEL_CBC_T1_12_MCAV</td>\n",
       "      <td>052022_BEL_CBC_T1_12_MCAV_host_removed_R1.fastq</td>\n",
       "      <td>Conventional base calls</td>\n",
       "      <td>Sanger / Illumina 1.9</td>\n",
       "      <td>6447936.0</td>\n",
       "      <td>963.4 Mbp</td>\n",
       "      <td>0.0</td>\n",
       "      <td>20-151</td>\n",
       "      <td>43.0</td>\n",
       "      <td>91.781848</td>\n",
       "      <td>...</td>\n",
       "      <td>pass</td>\n",
       "      <td>warn</td>\n",
       "      <td>pass</td>\n",
       "      <td>fail</td>\n",
       "      <td>pass</td>\n",
       "      <td>pass</td>\n",
       "      <td>warn</td>\n",
       "      <td>pass</td>\n",
       "      <td>pass</td>\n",
       "      <td>pass</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>052022_BEL_CBC_T1_40_MCAV</td>\n",
       "      <td>052022_BEL_CBC_T1_40_MCAV_host_removed_R1.fastq</td>\n",
       "      <td>Conventional base calls</td>\n",
       "      <td>Sanger / Illumina 1.9</td>\n",
       "      <td>1843935.0</td>\n",
       "      <td>273.8 Mbp</td>\n",
       "      <td>0.0</td>\n",
       "      <td>20-151</td>\n",
       "      <td>41.0</td>\n",
       "      <td>92.151268</td>\n",
       "      <td>...</td>\n",
       "      <td>pass</td>\n",
       "      <td>pass</td>\n",
       "      <td>pass</td>\n",
       "      <td>fail</td>\n",
       "      <td>pass</td>\n",
       "      <td>pass</td>\n",
       "      <td>warn</td>\n",
       "      <td>pass</td>\n",
       "      <td>pass</td>\n",
       "      <td>pass</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>052022_BEL_CBC_T3_40_MCAV</td>\n",
       "      <td>052022_BEL_CBC_T3_40_MCAV_host_removed_R1.fastq</td>\n",
       "      <td>Conventional base calls</td>\n",
       "      <td>Sanger / Illumina 1.9</td>\n",
       "      <td>7061478.0</td>\n",
       "      <td>1 Gbp</td>\n",
       "      <td>0.0</td>\n",
       "      <td>20-151</td>\n",
       "      <td>44.0</td>\n",
       "      <td>88.423478</td>\n",
       "      <td>...</td>\n",
       "      <td>pass</td>\n",
       "      <td>warn</td>\n",
       "      <td>pass</td>\n",
       "      <td>fail</td>\n",
       "      <td>pass</td>\n",
       "      <td>pass</td>\n",
       "      <td>warn</td>\n",
       "      <td>pass</td>\n",
       "      <td>pass</td>\n",
       "      <td>pass</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>6</th>\n",
       "      <td>052022_BEL_CBC_T3_48_MCAV</td>\n",
       "      <td>052022_BEL_CBC_T3_48_MCAV_host_removed_R1.fastq</td>\n",
       "      <td>Conventional base calls</td>\n",
       "      <td>Sanger / Illumina 1.9</td>\n",
       "      <td>2525681.0</td>\n",
       "      <td>373.3 Mbp</td>\n",
       "      <td>0.0</td>\n",
       "      <td>20-151</td>\n",
       "      <td>40.0</td>\n",
       "      <td>91.624805</td>\n",
       "      <td>...</td>\n",
       "      <td>pass</td>\n",
       "      <td>warn</td>\n",
       "      <td>pass</td>\n",
       "      <td>fail</td>\n",
       "      <td>warn</td>\n",
       "      <td>pass</td>\n",
       "      <td>warn</td>\n",
       "      <td>pass</td>\n",
       "      <td>pass</td>\n",
       "      <td>pass</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>4 rows × 23 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "                      Sample                                         Filename  \\\n",
       "0  052022_BEL_CBC_T1_12_MCAV  052022_BEL_CBC_T1_12_MCAV_host_removed_R1.fastq   \n",
       "2  052022_BEL_CBC_T1_40_MCAV  052022_BEL_CBC_T1_40_MCAV_host_removed_R1.fastq   \n",
       "4  052022_BEL_CBC_T3_40_MCAV  052022_BEL_CBC_T3_40_MCAV_host_removed_R1.fastq   \n",
       "6  052022_BEL_CBC_T3_48_MCAV  052022_BEL_CBC_T3_48_MCAV_host_removed_R1.fastq   \n",
       "\n",
       "                 File type               Encoding  Total Sequences  \\\n",
       "0  Conventional base calls  Sanger / Illumina 1.9        6447936.0   \n",
       "2  Conventional base calls  Sanger / Illumina 1.9        1843935.0   \n",
       "4  Conventional base calls  Sanger / Illumina 1.9        7061478.0   \n",
       "6  Conventional base calls  Sanger / Illumina 1.9        2525681.0   \n",
       "\n",
       "  Total Bases  Sequences flagged as poor quality Sequence length   %GC  \\\n",
       "0   963.4 Mbp                                0.0          20-151  43.0   \n",
       "2   273.8 Mbp                                0.0          20-151  41.0   \n",
       "4       1 Gbp                                0.0          20-151  44.0   \n",
       "6   373.3 Mbp                                0.0          20-151  40.0   \n",
       "\n",
       "   total_deduplicated_percentage  ...  per_base_sequence_quality  \\\n",
       "0                      91.781848  ...                       pass   \n",
       "2                      92.151268  ...                       pass   \n",
       "4                      88.423478  ...                       pass   \n",
       "6                      91.624805  ...                       pass   \n",
       "\n",
       "   per_tile_sequence_quality per_sequence_quality_scores  \\\n",
       "0                       warn                        pass   \n",
       "2                       pass                        pass   \n",
       "4                       warn                        pass   \n",
       "6                       warn                        pass   \n",
       "\n",
       "  per_base_sequence_content per_sequence_gc_content per_base_n_content  \\\n",
       "0                      fail                    pass               pass   \n",
       "2                      fail                    pass               pass   \n",
       "4                      fail                    pass               pass   \n",
       "6                      fail                    warn               pass   \n",
       "\n",
       "  sequence_length_distribution sequence_duplication_levels  \\\n",
       "0                         warn                        pass   \n",
       "2                         warn                        pass   \n",
       "4                         warn                        pass   \n",
       "6                         warn                        pass   \n",
       "\n",
       "  overrepresented_sequences adapter_content  \n",
       "0                      pass            pass  \n",
       "2                      pass            pass  \n",
       "4                      pass            pass  \n",
       "6                      pass            pass  \n",
       "\n",
       "[4 rows x 23 columns]"
      ]
     },
     "execution_count": 304,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "margin=pd.read_csv(\"multiqc_fastqc.txt\", sep=\"\\t\")\n",
    "# fix sample names \n",
    "margin['Sample'] = margin['Sample'].str.replace(r'_host_removed_R.*', '', regex=True)\n",
    "# remove every other row \n",
    "margin = margin.iloc[::2]\n",
    "margin\n",
    "# checked and f and r are the same here, so just keeping one for simplicity "
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 305,
   "id": "5c413c69-7003-409d-840d-4051f7f61911",
   "metadata": {},
   "outputs": [],
   "source": [
    "# add margin to reads df \n",
    "# fill in NAs of unampped reads in reads df with new values by matching sample id \n",
    "reads['Unmapped_Reads'] = reads['Sample'].map(margin.set_index('Sample')\n",
    "                                              ['Total Sequences']).fillna(reads['Unmapped_Reads'])\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 307,
   "id": "5d9e387f-f343-4721-bc04-02c495a78ba0",
   "metadata": {},
   "outputs": [],
   "source": [
    "# fix cols \n",
    "reads.rename(columns={'bowtie_percent_mapped_host': 'percent_mapped_host(slurm)'}, inplace=True)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 308,
   "id": "d416d475-1472-4f05-b2eb-f799d4f2a0d4",
   "metadata": {},
   "outputs": [],
   "source": [
    "# calculate percent reads mapped \n",
    "reads['percent_mapped_calculated'] = reads['percent_mapped_calculated'] = (((reads['post_qc'] - reads['Unmapped_Reads']) / reads['post_qc']) * 100).round(2)\n",
    "\n",
    "# calculate percent unmapped reads - see if it matches slurm output % \n",
    "reads['percent_unmapped_calculated'] = ((reads['Unmapped_Reads'] / reads['post_qc']) * 100).round(2)\n",
    "\n",
    "# do they add up? \n",
    "# total percent added from calculations \n",
    "reads['total_percent_calculated'] = reads['percent_mapped_calculated'] + reads['percent_unmapped_calculated']\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 309,
   "id": "a8d88d64-25b1-4ba0-a05e-b0f8d28e8d4d",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>Sample</th>\n",
       "      <th>Filename</th>\n",
       "      <th>post_qc</th>\n",
       "      <th>raw_seqs</th>\n",
       "      <th>Unmapped_Reads</th>\n",
       "      <th>percent_mapped_host</th>\n",
       "      <th>percent_mapped_calculated</th>\n",
       "      <th>percent_unmapped_calculated</th>\n",
       "      <th>total_percent_calculated</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>052022_BEL_CBC_T1_12_MCAV</td>\n",
       "      <td>052022_BEL_CBC_T1_12_MCAV_R1_001_val_1.fq</td>\n",
       "      <td>20742228.0</td>\n",
       "      <td>20754538</td>\n",
       "      <td>6447936.0</td>\n",
       "      <td>NaN</td>\n",
       "      <td>68.91</td>\n",
       "      <td>31.09</td>\n",
       "      <td>100.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>052022_BEL_CBC_T1_12_MCAV</td>\n",
       "      <td>052022_BEL_CBC_T1_12_MCAV_R2_001_val_2.fq</td>\n",
       "      <td>20742228.0</td>\n",
       "      <td>20754538</td>\n",
       "      <td>6447936.0</td>\n",
       "      <td>NaN</td>\n",
       "      <td>68.91</td>\n",
       "      <td>31.09</td>\n",
       "      <td>100.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>052022_BEL_CBC_T1_13_MCAV</td>\n",
       "      <td>052022_BEL_CBC_T1_13_MCAV_R1_001_val_1.fq</td>\n",
       "      <td>20861960.0</td>\n",
       "      <td>20870396</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>052022_BEL_CBC_T1_13_MCAV</td>\n",
       "      <td>052022_BEL_CBC_T1_13_MCAV_R2_001_val_2.fq</td>\n",
       "      <td>20861960.0</td>\n",
       "      <td>20870396</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>052022_BEL_CBC_T1_40_MCAV</td>\n",
       "      <td>052022_BEL_CBC_T1_40_MCAV_R1_001_val_1.fq</td>\n",
       "      <td>15419779.0</td>\n",
       "      <td>15425540</td>\n",
       "      <td>1843935.0</td>\n",
       "      <td>NaN</td>\n",
       "      <td>88.04</td>\n",
       "      <td>11.96</td>\n",
       "      <td>100.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>5</th>\n",
       "      <td>052022_BEL_CBC_T1_40_MCAV</td>\n",
       "      <td>052022_BEL_CBC_T1_40_MCAV_R2_001_val_2.fq</td>\n",
       "      <td>15419779.0</td>\n",
       "      <td>15425540</td>\n",
       "      <td>1843935.0</td>\n",
       "      <td>NaN</td>\n",
       "      <td>88.04</td>\n",
       "      <td>11.96</td>\n",
       "      <td>100.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>6</th>\n",
       "      <td>052022_BEL_CBC_T1_57_MCAV</td>\n",
       "      <td>052022_BEL_CBC_T1_57_MCAV_R1_001_val_1.fq</td>\n",
       "      <td>31964707.0</td>\n",
       "      <td>31976704</td>\n",
       "      <td>9131401.0</td>\n",
       "      <td>71.43</td>\n",
       "      <td>71.43</td>\n",
       "      <td>28.57</td>\n",
       "      <td>100.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>7</th>\n",
       "      <td>052022_BEL_CBC_T1_57_MCAV</td>\n",
       "      <td>052022_BEL_CBC_T1_57_MCAV_R2_001_val_2.fq</td>\n",
       "      <td>31964707.0</td>\n",
       "      <td>31976704</td>\n",
       "      <td>9131401.0</td>\n",
       "      <td>71.43</td>\n",
       "      <td>71.43</td>\n",
       "      <td>28.57</td>\n",
       "      <td>100.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>8</th>\n",
       "      <td>052022_BEL_CBC_T1_70_MCAV</td>\n",
       "      <td>052022_BEL_CBC_T1_70_MCAV_R1_001_val_1.fq</td>\n",
       "      <td>15776674.0</td>\n",
       "      <td>15782392</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>9</th>\n",
       "      <td>052022_BEL_CBC_T1_70_MCAV</td>\n",
       "      <td>052022_BEL_CBC_T1_70_MCAV_R2_001_val_2.fq</td>\n",
       "      <td>15776674.0</td>\n",
       "      <td>15782392</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "                      Sample                                   Filename  \\\n",
       "0  052022_BEL_CBC_T1_12_MCAV  052022_BEL_CBC_T1_12_MCAV_R1_001_val_1.fq   \n",
       "1  052022_BEL_CBC_T1_12_MCAV  052022_BEL_CBC_T1_12_MCAV_R2_001_val_2.fq   \n",
       "2  052022_BEL_CBC_T1_13_MCAV  052022_BEL_CBC_T1_13_MCAV_R1_001_val_1.fq   \n",
       "3  052022_BEL_CBC_T1_13_MCAV  052022_BEL_CBC_T1_13_MCAV_R2_001_val_2.fq   \n",
       "4  052022_BEL_CBC_T1_40_MCAV  052022_BEL_CBC_T1_40_MCAV_R1_001_val_1.fq   \n",
       "5  052022_BEL_CBC_T1_40_MCAV  052022_BEL_CBC_T1_40_MCAV_R2_001_val_2.fq   \n",
       "6  052022_BEL_CBC_T1_57_MCAV  052022_BEL_CBC_T1_57_MCAV_R1_001_val_1.fq   \n",
       "7  052022_BEL_CBC_T1_57_MCAV  052022_BEL_CBC_T1_57_MCAV_R2_001_val_2.fq   \n",
       "8  052022_BEL_CBC_T1_70_MCAV  052022_BEL_CBC_T1_70_MCAV_R1_001_val_1.fq   \n",
       "9  052022_BEL_CBC_T1_70_MCAV  052022_BEL_CBC_T1_70_MCAV_R2_001_val_2.fq   \n",
       "\n",
       "      post_qc  raw_seqs  Unmapped_Reads  percent_mapped_host  \\\n",
       "0  20742228.0  20754538       6447936.0                  NaN   \n",
       "1  20742228.0  20754538       6447936.0                  NaN   \n",
       "2  20861960.0  20870396             NaN                  NaN   \n",
       "3  20861960.0  20870396             NaN                  NaN   \n",
       "4  15419779.0  15425540       1843935.0                  NaN   \n",
       "5  15419779.0  15425540       1843935.0                  NaN   \n",
       "6  31964707.0  31976704       9131401.0                71.43   \n",
       "7  31964707.0  31976704       9131401.0                71.43   \n",
       "8  15776674.0  15782392             NaN                  NaN   \n",
       "9  15776674.0  15782392             NaN                  NaN   \n",
       "\n",
       "   percent_mapped_calculated  percent_unmapped_calculated  \\\n",
       "0                      68.91                        31.09   \n",
       "1                      68.91                        31.09   \n",
       "2                        NaN                          NaN   \n",
       "3                        NaN                          NaN   \n",
       "4                      88.04                        11.96   \n",
       "5                      88.04                        11.96   \n",
       "6                      71.43                        28.57   \n",
       "7                      71.43                        28.57   \n",
       "8                        NaN                          NaN   \n",
       "9                        NaN                          NaN   \n",
       "\n",
       "   total_percent_calculated  \n",
       "0                     100.0  \n",
       "1                     100.0  \n",
       "2                       NaN  \n",
       "3                       NaN  \n",
       "4                     100.0  \n",
       "5                     100.0  \n",
       "6                     100.0  \n",
       "7                     100.0  \n",
       "8                       NaN  \n",
       "9                       NaN  "
      ]
     },
     "execution_count": 309,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "reads.head(10)"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "27a556ed-c950-408e-842a-18c15d79d253",
   "metadata": {},
   "source": [
    "### symbiont and/or human?"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 310,
   "id": "633c3fe2-f1b7-4f72-96c1-b872be586a43",
   "metadata": {},
   "outputs": [],
   "source": [
    "# healthy 2019 \n",
    "fastqc_screen=pd.read_csv('/project/pi_sarah_gignouxwolfsohn_uml_edu/brooke/mcav/healthy_2019_mcav/final_reads_filtered/multiqc_data/multiqc_fastq_screen.txt', sep=\"\\t\")"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 311,
   "id": "e36727c6-5eec-4eb0-8195-cb0d8a6e2bf3",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "<bound method NDFrame.head of                                               Sample  total_reads  \\\n",
       "0   062019_BEL_CBC_T1_16_MCAV_host_removed_R1_screen      7533935   \n",
       "1   062019_BEL_CBC_T1_16_MCAV_host_removed_R2_screen      7533935   \n",
       "2   062019_BEL_CBC_T1_20_MCAV_host_removed_R1_screen      8436904   \n",
       "3   062019_BEL_CBC_T1_20_MCAV_host_removed_R2_screen      8436904   \n",
       "4   062019_BEL_CBC_T1_24_MCAV_host_removed_R1_screen      3891938   \n",
       "5   062019_BEL_CBC_T1_24_MCAV_host_removed_R2_screen      3891938   \n",
       "6    062019_BEL_CBC_T1_3_MCAV_host_removed_R1_screen      3141837   \n",
       "7    062019_BEL_CBC_T1_3_MCAV_host_removed_R2_screen      3141837   \n",
       "8   062019_BEL_CBC_T2_16_MCAV_host_removed_R1_screen      2427227   \n",
       "9   062019_BEL_CBC_T2_16_MCAV_host_removed_R2_screen      2427227   \n",
       "10   062019_BEL_CBC_T2_5_MCAV_host_removed_R1_screen      9068181   \n",
       "11   062019_BEL_CBC_T2_5_MCAV_host_removed_R2_screen      9068181   \n",
       "12  062019_BEL_CBC_T3_14_MCAV_host_removed_R1_screen      3722331   \n",
       "13  062019_BEL_CBC_T3_14_MCAV_host_removed_R2_screen      3722331   \n",
       "14  062019_BEL_CBC_T3_15_MCAV_host_removed_R1_screen      2321387   \n",
       "15  062019_BEL_CBC_T3_15_MCAV_host_removed_R2_screen      2321387   \n",
       "16  062019_BEL_CBC_T3_16_MCAV_host_removed_R1_screen      1094028   \n",
       "17  062019_BEL_CBC_T3_16_MCAV_host_removed_R2_screen      1094028   \n",
       "18   062019_BEL_CBC_T3_1_MCAV_host_removed_R1_screen      6548146   \n",
       "19   062019_BEL_CBC_T3_1_MCAV_host_removed_R2_screen      6548146   \n",
       "20   062019_BEL_CBC_T3_6_MCAV_host_removed_R1_screen      1861982   \n",
       "21   062019_BEL_CBC_T3_6_MCAV_host_removed_R2_screen      1861982   \n",
       "22   062019_BEL_CBC_T3_8_MCAV_host_removed_R1_screen     78898318   \n",
       "23   062019_BEL_CBC_T3_8_MCAV_host_removed_R2_screen     78898318   \n",
       "24   062019_BEL_CBC_T3_9_MCAV_host_removed_R1_screen      4166616   \n",
       "25   062019_BEL_CBC_T3_9_MCAV_host_removed_R2_screen      4166616   \n",
       "\n",
       "    Human counts  Human percentage  Symbionts counts  Symbionts percentage  \\\n",
       "0       170887.0              2.27          383926.0                  5.09   \n",
       "1       177877.0              2.36          386350.0                  5.12   \n",
       "2       105832.0              1.26          110162.0                  1.30   \n",
       "3       109660.0              1.30          111038.0                  1.31   \n",
       "4       105879.0              2.72          215443.0                  5.54   \n",
       "5       108590.0              2.78          215611.0                  5.54   \n",
       "6        53206.0              1.70          126162.0                  4.01   \n",
       "7        50527.0              1.60          116814.0                  3.72   \n",
       "8        55935.0              2.31           48577.0                  2.00   \n",
       "9        59799.0              2.46           51478.0                  2.12   \n",
       "10      273309.0              3.01          535098.0                  5.90   \n",
       "11      257407.0              2.84          494059.0                  5.44   \n",
       "12       81496.0              2.18          142390.0                  3.82   \n",
       "13       82479.0              2.21          140301.0                  3.77   \n",
       "14       41716.0              1.80           64053.0                  2.76   \n",
       "15       42435.0              1.83           62702.0                  2.70   \n",
       "16       38436.0              3.51           58396.0                  5.34   \n",
       "17       37126.0              3.40           56295.0                  5.14   \n",
       "18      170958.0              2.60          334458.0                  5.11   \n",
       "19      172449.0              2.63          327658.0                  5.00   \n",
       "20       23348.0              1.26           38422.0                  2.06   \n",
       "21       22849.0              1.23           35950.0                  1.94   \n",
       "22      808807.0              1.02          739757.0                  0.94   \n",
       "23      905121.0              1.15          792071.0                  1.00   \n",
       "24      100433.0              2.40          224059.0                  5.38   \n",
       "25      103910.0              2.50          222983.0                  5.34   \n",
       "\n",
       "    Symbiont2 counts  Symbiont2 percentage  Symbiont3 counts  \\\n",
       "0           355086.0                  4.71          251497.0   \n",
       "1           356912.0                  4.74          258480.0   \n",
       "2           123935.0                  1.47          326038.0   \n",
       "3           122653.0                  1.46          323381.0   \n",
       "4           211889.0                  5.44          151653.0   \n",
       "5           211754.0                  5.44          154135.0   \n",
       "6           112295.0                  3.57           78944.0   \n",
       "7           104516.0                  3.32           75720.0   \n",
       "8            62608.0                  2.59           58429.0   \n",
       "9            63493.0                  2.62           62135.0   \n",
       "10          470988.0                  5.19          355495.0   \n",
       "11          441108.0                  4.86          341251.0   \n",
       "12          135829.0                  3.64          105852.0   \n",
       "13          133202.0                  3.58          105719.0   \n",
       "14           65855.0                  2.83           53454.0   \n",
       "15           63822.0                  2.75           53339.0   \n",
       "16           52481.0                  4.79           42042.0   \n",
       "17           50557.0                  4.62           41242.0   \n",
       "18          318111.0                  4.86          235143.0   \n",
       "19          311664.0                  4.77          235352.0   \n",
       "20           29609.0                  1.59           23303.0   \n",
       "21           28514.0                  1.52           22945.0   \n",
       "22          878212.0                  1.11         2673314.0   \n",
       "23          882631.0                  1.12         2737750.0   \n",
       "24          213103.0                  5.11          149998.0   \n",
       "25          211975.0                  5.09          152783.0   \n",
       "\n",
       "    Symbiont3 percentage  ...  Symbiont5 counts  Symbiont5 percentage  \\\n",
       "0                   3.34  ...          460588.0                  6.12   \n",
       "1                   3.44  ...          465440.0                  6.17   \n",
       "2                   3.87  ...          139682.0                  1.66   \n",
       "3                   3.83  ...          142415.0                  1.69   \n",
       "4                   3.90  ...          289321.0                  7.43   \n",
       "5                   3.97  ...          289537.0                  7.44   \n",
       "6                   2.51  ...          137016.0                  4.36   \n",
       "7                   2.41  ...          128455.0                  4.10   \n",
       "8                   2.41  ...           63705.0                  2.63   \n",
       "9                   2.55  ...           67383.0                  2.78   \n",
       "10                  3.92  ...          555125.0                  6.12   \n",
       "11                  3.75  ...          520787.0                  5.75   \n",
       "12                  2.84  ...          153337.0                  4.12   \n",
       "13                  2.84  ...          151993.0                  4.08   \n",
       "14                  2.30  ...           76441.0                  3.30   \n",
       "15                  2.30  ...           75423.0                  3.25   \n",
       "16                  3.84  ...           60564.0                  5.54   \n",
       "17                  3.76  ...           58345.0                  5.34   \n",
       "18                  3.59  ...          359624.0                  5.50   \n",
       "19                  3.60  ...          356142.0                  5.45   \n",
       "20                  1.25  ...           32822.0                  1.77   \n",
       "21                  1.23  ...           31568.0                  1.70   \n",
       "22                  3.39  ...          967584.0                  1.22   \n",
       "23                  3.47  ...         1045870.0                  1.33   \n",
       "24                  3.59  ...          253410.0                  6.08   \n",
       "25                  3.67  ...          253504.0                  6.08   \n",
       "\n",
       "    Symbiont6 counts  Symbiont6 percentage  Symbiont7 counts  \\\n",
       "0           246410.0                  3.27          254458.0   \n",
       "1           252927.0                  3.36          260684.0   \n",
       "2           320850.0                  3.80          635729.0   \n",
       "3           317875.0                  3.77          629783.0   \n",
       "4           145869.0                  3.75          166964.0   \n",
       "5           148316.0                  3.81          169217.0   \n",
       "6            77758.0                  2.48           79209.0   \n",
       "7            75281.0                  2.39           75642.0   \n",
       "8            55328.0                  2.27           61103.0   \n",
       "9            59102.0                  2.43           64585.0   \n",
       "10          350782.0                  3.86          354837.0   \n",
       "11          338391.0                  3.73          340166.0   \n",
       "12          103263.0                  2.78          105682.0   \n",
       "13          103219.0                  2.78          105307.0   \n",
       "14           52486.0                  2.26           55296.0   \n",
       "15           52191.0                  2.25           54879.0   \n",
       "16           41847.0                  3.82           41642.0   \n",
       "17           41189.0                  3.77           40735.0   \n",
       "18          230286.0                  3.52          241133.0   \n",
       "19          230772.0                  3.52          241158.0   \n",
       "20           22756.0                  1.22           23200.0   \n",
       "21           23536.0                  1.26           22697.0   \n",
       "22         2626226.0                  3.32         5400438.0   \n",
       "23         2685342.0                  3.39         5442089.0   \n",
       "24          145655.0                  3.49          156626.0   \n",
       "25          148677.0                  3.56          159257.0   \n",
       "\n",
       "    Symbiont7 percentage  Symbiont8 counts  Symbiont8 percentage  \\\n",
       "0                   3.37         4986820.0                 66.19   \n",
       "1                   3.46         4965001.0                 65.90   \n",
       "2                   7.53          139845.0                  1.66   \n",
       "3                   7.47          143132.0                  1.70   \n",
       "4                   4.28         2056350.0                 52.84   \n",
       "5                   4.35         2038522.0                 52.38   \n",
       "6                   2.53         1807998.0                 57.55   \n",
       "7                   2.41         1692589.0                 53.87   \n",
       "8                   2.52          150979.0                  6.22   \n",
       "9                   2.67          154508.0                  6.37   \n",
       "10                  3.90         6352780.0                 70.04   \n",
       "11                  3.75         5898062.0                 65.04   \n",
       "12                  2.84         1656380.0                 44.50   \n",
       "13                  2.83         1636028.0                 43.95   \n",
       "14                  2.38          557074.0                 23.99   \n",
       "15                  2.36          544944.0                 23.48   \n",
       "16                  3.80          678595.0                 62.01   \n",
       "17                  3.71          656009.0                 59.96   \n",
       "18                  3.68         3790960.0                 57.90   \n",
       "19                  3.68         3707773.0                 56.63   \n",
       "20                  1.25          363028.0                 19.49   \n",
       "21                  1.22          349640.0                 18.78   \n",
       "22                  6.84         1030880.0                  1.31   \n",
       "23                  6.90         1118202.0                  1.41   \n",
       "24                  3.76         2875922.0                 69.02   \n",
       "25                  3.83         2827803.0                 67.87   \n",
       "\n",
       "    No hits counts  No hits percentage  \n",
       "0          2494485               33.11  \n",
       "1          2515580               33.39  \n",
       "2          7740015               91.74  \n",
       "3          7744234               91.79  \n",
       "4          1777058               45.66  \n",
       "5          1794183               46.10  \n",
       "6          1317372               41.93  \n",
       "7          1432049               45.58  \n",
       "8          2251981               92.78  \n",
       "9          2248583               92.64  \n",
       "10         2676927               29.52  \n",
       "11         3128522               34.50  \n",
       "12         2047282               55.00  \n",
       "13         2066638               55.52  \n",
       "14         1748700               75.33  \n",
       "15         1760307               75.83  \n",
       "16          405227               37.04  \n",
       "17          427436               39.07  \n",
       "18         2711587               41.41  \n",
       "19         2792784               42.65  \n",
       "20         1485116               79.76  \n",
       "21         1498709               80.49  \n",
       "22        73067732               92.61  \n",
       "23        72988833               92.51  \n",
       "24         1257901               30.19  \n",
       "25         1304984               31.32  \n",
       "\n",
       "[26 rows x 22 columns]>"
      ]
     },
     "execution_count": 311,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "fastqc_screen.head\n",
    "\n",
    "# why do all the reads still slightly differ!?\n",
    "    # t1 16 2019 \n",
    "# reads from host removal slurm output - 8,131,957\n",
    "# fastqc screen input reads - 7,533,935\n",
    "# leftover 2,494,485"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 315,
   "id": "ab685ca5-047c-4f90-b8ef-776966110f14",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>Sample</th>\n",
       "      <th>Filename</th>\n",
       "      <th>post_qc</th>\n",
       "      <th>raw_seqs</th>\n",
       "      <th>Unmapped_Reads</th>\n",
       "      <th>percent_mapped_host</th>\n",
       "      <th>percent_mapped_calculated</th>\n",
       "      <th>percent_unmapped_calculated</th>\n",
       "      <th>total_percent_calculated</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>28</th>\n",
       "      <td>062019_BEL_CBC_T1_16_MCAV</td>\n",
       "      <td>062019_BEL_CBC_T1_16_MCAV_R1_001_val_1.fq</td>\n",
       "      <td>25807031.0</td>\n",
       "      <td>25812745</td>\n",
       "      <td>8131957.5</td>\n",
       "      <td>68.49</td>\n",
       "      <td>68.49</td>\n",
       "      <td>31.51</td>\n",
       "      <td>100.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>29</th>\n",
       "      <td>062019_BEL_CBC_T1_16_MCAV</td>\n",
       "      <td>062019_BEL_CBC_T1_16_MCAV_R2_001_val_2.fq</td>\n",
       "      <td>25807031.0</td>\n",
       "      <td>25812745</td>\n",
       "      <td>8131957.5</td>\n",
       "      <td>68.49</td>\n",
       "      <td>68.49</td>\n",
       "      <td>31.51</td>\n",
       "      <td>100.0</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "                       Sample                                   Filename  \\\n",
       "28  062019_BEL_CBC_T1_16_MCAV  062019_BEL_CBC_T1_16_MCAV_R1_001_val_1.fq   \n",
       "29  062019_BEL_CBC_T1_16_MCAV  062019_BEL_CBC_T1_16_MCAV_R2_001_val_2.fq   \n",
       "\n",
       "       post_qc  raw_seqs  Unmapped_Reads  percent_mapped_host  \\\n",
       "28  25807031.0  25812745       8131957.5                68.49   \n",
       "29  25807031.0  25812745       8131957.5                68.49   \n",
       "\n",
       "    percent_mapped_calculated  percent_unmapped_calculated  \\\n",
       "28                      68.49                        31.51   \n",
       "29                      68.49                        31.51   \n",
       "\n",
       "    total_percent_calculated  \n",
       "28                     100.0  \n",
       "29                     100.0  "
      ]
     },
     "execution_count": 315,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "reads[reads['Sample']=='062019_BEL_CBC_T1_16_MCAV']"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 316,
   "id": "c51c1cea-6f27-4bda-88ec-9523ad24a599",
   "metadata": {},
   "outputs": [],
   "source": [
    "# upload rest of data sets and concat together \n",
    "\n",
    "# mcav 2022 \n",
    "fastqscreen_2022=pd.read_csv('/project/pi_sarah_gignouxwolfsohn_uml_edu/brooke/mcav/healthy_2022_mcav/assembly/final_reads_filtered/multiqc_data/multiqc_fastq_screen.txt', sep=\"\\t\")\n",
    "\n",
    "# margin\n",
    "fastqscreen_margin=pd.read_csv('/project/pi_sarah_gignouxwolfsohn_uml_edu/brooke/mcav/margin_mcav/assembly/final_reads_filtered/multiqc_data/multiqc_fastq_screen.txt',sep=\"\\t\")\n",
    "\n",
    "# diseased mcav was done differently?\n",
    "# tried to rerun fastqc screen, but there are errors so just ignoring this set for now "
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 317,
   "id": "f21bb045-1375-4dcd-958d-9ab6a5eda461",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>Sample</th>\n",
       "      <th>Filename</th>\n",
       "      <th>post_qc</th>\n",
       "      <th>raw_seqs</th>\n",
       "      <th>Unmapped_Reads</th>\n",
       "      <th>percent_mapped_host</th>\n",
       "      <th>percent_mapped_calculated</th>\n",
       "      <th>percent_unmapped_calculated</th>\n",
       "      <th>total_percent_calculated</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>052022_BEL_CBC_T1_12_MCAV</td>\n",
       "      <td>052022_BEL_CBC_T1_12_MCAV_R1_001_val_1.fq</td>\n",
       "      <td>20742228.0</td>\n",
       "      <td>20754538</td>\n",
       "      <td>6447936.0</td>\n",
       "      <td>NaN</td>\n",
       "      <td>68.91</td>\n",
       "      <td>31.09</td>\n",
       "      <td>100.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>052022_BEL_CBC_T1_12_MCAV</td>\n",
       "      <td>052022_BEL_CBC_T1_12_MCAV_R2_001_val_2.fq</td>\n",
       "      <td>20742228.0</td>\n",
       "      <td>20754538</td>\n",
       "      <td>6447936.0</td>\n",
       "      <td>NaN</td>\n",
       "      <td>68.91</td>\n",
       "      <td>31.09</td>\n",
       "      <td>100.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>052022_BEL_CBC_T1_13_MCAV</td>\n",
       "      <td>052022_BEL_CBC_T1_13_MCAV_R1_001_val_1.fq</td>\n",
       "      <td>20861960.0</td>\n",
       "      <td>20870396</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>052022_BEL_CBC_T1_13_MCAV</td>\n",
       "      <td>052022_BEL_CBC_T1_13_MCAV_R2_001_val_2.fq</td>\n",
       "      <td>20861960.0</td>\n",
       "      <td>20870396</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>052022_BEL_CBC_T1_40_MCAV</td>\n",
       "      <td>052022_BEL_CBC_T1_40_MCAV_R1_001_val_1.fq</td>\n",
       "      <td>15419779.0</td>\n",
       "      <td>15425540</td>\n",
       "      <td>1843935.0</td>\n",
       "      <td>NaN</td>\n",
       "      <td>88.04</td>\n",
       "      <td>11.96</td>\n",
       "      <td>100.0</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "                      Sample                                   Filename  \\\n",
       "0  052022_BEL_CBC_T1_12_MCAV  052022_BEL_CBC_T1_12_MCAV_R1_001_val_1.fq   \n",
       "1  052022_BEL_CBC_T1_12_MCAV  052022_BEL_CBC_T1_12_MCAV_R2_001_val_2.fq   \n",
       "2  052022_BEL_CBC_T1_13_MCAV  052022_BEL_CBC_T1_13_MCAV_R1_001_val_1.fq   \n",
       "3  052022_BEL_CBC_T1_13_MCAV  052022_BEL_CBC_T1_13_MCAV_R2_001_val_2.fq   \n",
       "4  052022_BEL_CBC_T1_40_MCAV  052022_BEL_CBC_T1_40_MCAV_R1_001_val_1.fq   \n",
       "\n",
       "      post_qc  raw_seqs  Unmapped_Reads  percent_mapped_host  \\\n",
       "0  20742228.0  20754538       6447936.0                  NaN   \n",
       "1  20742228.0  20754538       6447936.0                  NaN   \n",
       "2  20861960.0  20870396             NaN                  NaN   \n",
       "3  20861960.0  20870396             NaN                  NaN   \n",
       "4  15419779.0  15425540       1843935.0                  NaN   \n",
       "\n",
       "   percent_mapped_calculated  percent_unmapped_calculated  \\\n",
       "0                      68.91                        31.09   \n",
       "1                      68.91                        31.09   \n",
       "2                        NaN                          NaN   \n",
       "3                        NaN                          NaN   \n",
       "4                      88.04                        11.96   \n",
       "\n",
       "   total_percent_calculated  \n",
       "0                     100.0  \n",
       "1                     100.0  \n",
       "2                       NaN  \n",
       "3                       NaN  \n",
       "4                     100.0  "
      ]
     },
     "execution_count": 317,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "reads.head()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 318,
   "id": "65a7b268-09b7-4efb-a113-52192a7d4a62",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>Sample</th>\n",
       "      <th>host_removed_reads(fqscreen_input)</th>\n",
       "      <th>Human counts</th>\n",
       "      <th>Human percentage</th>\n",
       "      <th>Symbionts counts</th>\n",
       "      <th>Symbionts percentage</th>\n",
       "      <th>Symbiont2 counts</th>\n",
       "      <th>Symbiont2 percentage</th>\n",
       "      <th>Symbiont3 counts</th>\n",
       "      <th>Symbiont3 percentage</th>\n",
       "      <th>...</th>\n",
       "      <th>Symbiont5 percentage</th>\n",
       "      <th>Symbiont6 counts</th>\n",
       "      <th>Symbiont6 percentage</th>\n",
       "      <th>Symbiont7 counts</th>\n",
       "      <th>Symbiont7 percentage</th>\n",
       "      <th>Symbiont8 counts</th>\n",
       "      <th>Symbiont8 percentage</th>\n",
       "      <th>No hits counts</th>\n",
       "      <th>No hits percentage</th>\n",
       "      <th>Filename</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>062019_BEL_CBC_T1_16_MCAV</td>\n",
       "      <td>7533935</td>\n",
       "      <td>170887.0</td>\n",
       "      <td>2.27</td>\n",
       "      <td>383926.0</td>\n",
       "      <td>5.09</td>\n",
       "      <td>355086.0</td>\n",
       "      <td>4.71</td>\n",
       "      <td>251497.0</td>\n",
       "      <td>3.34</td>\n",
       "      <td>...</td>\n",
       "      <td>6.12</td>\n",
       "      <td>246410.0</td>\n",
       "      <td>3.27</td>\n",
       "      <td>254458.0</td>\n",
       "      <td>3.37</td>\n",
       "      <td>4986820.0</td>\n",
       "      <td>66.19</td>\n",
       "      <td>2494485</td>\n",
       "      <td>33.11</td>\n",
       "      <td>062019_BEL_CBC_T1_16_MCAV_R1_001_val_1.fq</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>062019_BEL_CBC_T1_16_MCAV</td>\n",
       "      <td>7533935</td>\n",
       "      <td>177877.0</td>\n",
       "      <td>2.36</td>\n",
       "      <td>386350.0</td>\n",
       "      <td>5.12</td>\n",
       "      <td>356912.0</td>\n",
       "      <td>4.74</td>\n",
       "      <td>258480.0</td>\n",
       "      <td>3.44</td>\n",
       "      <td>...</td>\n",
       "      <td>6.17</td>\n",
       "      <td>252927.0</td>\n",
       "      <td>3.36</td>\n",
       "      <td>260684.0</td>\n",
       "      <td>3.46</td>\n",
       "      <td>4965001.0</td>\n",
       "      <td>65.90</td>\n",
       "      <td>2515580</td>\n",
       "      <td>33.39</td>\n",
       "      <td>062019_BEL_CBC_T1_16_MCAV_R2_001_val_2.fq</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>062019_BEL_CBC_T1_20_MCAV</td>\n",
       "      <td>8436904</td>\n",
       "      <td>105832.0</td>\n",
       "      <td>1.26</td>\n",
       "      <td>110162.0</td>\n",
       "      <td>1.30</td>\n",
       "      <td>123935.0</td>\n",
       "      <td>1.47</td>\n",
       "      <td>326038.0</td>\n",
       "      <td>3.87</td>\n",
       "      <td>...</td>\n",
       "      <td>1.66</td>\n",
       "      <td>320850.0</td>\n",
       "      <td>3.80</td>\n",
       "      <td>635729.0</td>\n",
       "      <td>7.53</td>\n",
       "      <td>139845.0</td>\n",
       "      <td>1.66</td>\n",
       "      <td>7740015</td>\n",
       "      <td>91.74</td>\n",
       "      <td>062019_BEL_CBC_T1_20_MCAV_R1_001_val_1.fq</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>062019_BEL_CBC_T1_20_MCAV</td>\n",
       "      <td>8436904</td>\n",
       "      <td>109660.0</td>\n",
       "      <td>1.30</td>\n",
       "      <td>111038.0</td>\n",
       "      <td>1.31</td>\n",
       "      <td>122653.0</td>\n",
       "      <td>1.46</td>\n",
       "      <td>323381.0</td>\n",
       "      <td>3.83</td>\n",
       "      <td>...</td>\n",
       "      <td>1.69</td>\n",
       "      <td>317875.0</td>\n",
       "      <td>3.77</td>\n",
       "      <td>629783.0</td>\n",
       "      <td>7.47</td>\n",
       "      <td>143132.0</td>\n",
       "      <td>1.70</td>\n",
       "      <td>7744234</td>\n",
       "      <td>91.79</td>\n",
       "      <td>062019_BEL_CBC_T1_20_MCAV_R2_001_val_2.fq</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>062019_BEL_CBC_T1_24_MCAV</td>\n",
       "      <td>3891938</td>\n",
       "      <td>105879.0</td>\n",
       "      <td>2.72</td>\n",
       "      <td>215443.0</td>\n",
       "      <td>5.54</td>\n",
       "      <td>211889.0</td>\n",
       "      <td>5.44</td>\n",
       "      <td>151653.0</td>\n",
       "      <td>3.90</td>\n",
       "      <td>...</td>\n",
       "      <td>7.43</td>\n",
       "      <td>145869.0</td>\n",
       "      <td>3.75</td>\n",
       "      <td>166964.0</td>\n",
       "      <td>4.28</td>\n",
       "      <td>2056350.0</td>\n",
       "      <td>52.84</td>\n",
       "      <td>1777058</td>\n",
       "      <td>45.66</td>\n",
       "      <td>062019_BEL_CBC_T1_24_MCAV_R1_001_val_1.fq</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>5 rows × 23 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "                      Sample  host_removed_reads(fqscreen_input)  \\\n",
       "0  062019_BEL_CBC_T1_16_MCAV                             7533935   \n",
       "1  062019_BEL_CBC_T1_16_MCAV                             7533935   \n",
       "2  062019_BEL_CBC_T1_20_MCAV                             8436904   \n",
       "3  062019_BEL_CBC_T1_20_MCAV                             8436904   \n",
       "4  062019_BEL_CBC_T1_24_MCAV                             3891938   \n",
       "\n",
       "   Human counts  Human percentage  Symbionts counts  Symbionts percentage  \\\n",
       "0      170887.0              2.27          383926.0                  5.09   \n",
       "1      177877.0              2.36          386350.0                  5.12   \n",
       "2      105832.0              1.26          110162.0                  1.30   \n",
       "3      109660.0              1.30          111038.0                  1.31   \n",
       "4      105879.0              2.72          215443.0                  5.54   \n",
       "\n",
       "   Symbiont2 counts  Symbiont2 percentage  Symbiont3 counts  \\\n",
       "0          355086.0                  4.71          251497.0   \n",
       "1          356912.0                  4.74          258480.0   \n",
       "2          123935.0                  1.47          326038.0   \n",
       "3          122653.0                  1.46          323381.0   \n",
       "4          211889.0                  5.44          151653.0   \n",
       "\n",
       "   Symbiont3 percentage  ...  Symbiont5 percentage  Symbiont6 counts  \\\n",
       "0                  3.34  ...                  6.12          246410.0   \n",
       "1                  3.44  ...                  6.17          252927.0   \n",
       "2                  3.87  ...                  1.66          320850.0   \n",
       "3                  3.83  ...                  1.69          317875.0   \n",
       "4                  3.90  ...                  7.43          145869.0   \n",
       "\n",
       "   Symbiont6 percentage  Symbiont7 counts  Symbiont7 percentage  \\\n",
       "0                  3.27          254458.0                  3.37   \n",
       "1                  3.36          260684.0                  3.46   \n",
       "2                  3.80          635729.0                  7.53   \n",
       "3                  3.77          629783.0                  7.47   \n",
       "4                  3.75          166964.0                  4.28   \n",
       "\n",
       "   Symbiont8 counts  Symbiont8 percentage  No hits counts  No hits percentage  \\\n",
       "0         4986820.0                 66.19         2494485               33.11   \n",
       "1         4965001.0                 65.90         2515580               33.39   \n",
       "2          139845.0                  1.66         7740015               91.74   \n",
       "3          143132.0                  1.70         7744234               91.79   \n",
       "4         2056350.0                 52.84         1777058               45.66   \n",
       "\n",
       "                                    Filename  \n",
       "0  062019_BEL_CBC_T1_16_MCAV_R1_001_val_1.fq  \n",
       "1  062019_BEL_CBC_T1_16_MCAV_R2_001_val_2.fq  \n",
       "2  062019_BEL_CBC_T1_20_MCAV_R1_001_val_1.fq  \n",
       "3  062019_BEL_CBC_T1_20_MCAV_R2_001_val_2.fq  \n",
       "4  062019_BEL_CBC_T1_24_MCAV_R1_001_val_1.fq  \n",
       "\n",
       "[5 rows x 23 columns]"
      ]
     },
     "execution_count": 318,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "# concat \n",
    "fastq_screen=pd.concat([fastqc_screen,fastqscreen_2022, fastqscreen_margin])\n",
    "\n",
    "#rename 'total reads'\n",
    "fastq_screen = fastq_screen.rename(columns={'total_reads': 'host_removed_reads(fqscreen_input)'})\n",
    "\n",
    "#fix sample id to match the other ones \n",
    "# remove 'host_removed_' & '_screen' from sample name \n",
    "fastq_screen['Sample']=fastq_screen['Sample'].str.replace(r'host_removed_','',regex=True)\n",
    "\n",
    "# add extension 'R1_001_val_1.fq' or 'R2_001_val_2.fq to match up f vs r seq file in other df \n",
    "#np.where(condition, value if condition is true, value if condition is false)\n",
    "fastq_screen['Filename'] = np.where(\n",
    "    fastq_screen['Sample'].str.contains('R1'),\n",
    "    fastq_screen['Sample'].str.replace(r'screen', '001_val_1.fq', regex=True),\n",
    "    fastq_screen['Sample'].str.replace(r'screen', '001_val_2.fq', regex=True)\n",
    ")\n",
    "\n",
    "# finish fixing sample name\n",
    "fastq_screen['Sample']=fastq_screen['Sample'].str.replace(r'_R[12]_screen','',regex=True)\n",
    "fastq_screen.head()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 319,
   "id": "4e1fb209-1cdf-4980-846b-49328e9de484",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>Sample</th>\n",
       "      <th>Filename</th>\n",
       "      <th>post_qc</th>\n",
       "      <th>raw_seqs</th>\n",
       "      <th>Unmapped_Reads</th>\n",
       "      <th>percent_mapped_host</th>\n",
       "      <th>percent_mapped_calculated</th>\n",
       "      <th>percent_unmapped_calculated</th>\n",
       "      <th>total_percent_calculated</th>\n",
       "      <th>host_removed_reads(fqscreen_input)</th>\n",
       "      <th>...</th>\n",
       "      <th>Symbiont5 counts</th>\n",
       "      <th>Symbiont5 percentage</th>\n",
       "      <th>Symbiont6 counts</th>\n",
       "      <th>Symbiont6 percentage</th>\n",
       "      <th>Symbiont7 counts</th>\n",
       "      <th>Symbiont7 percentage</th>\n",
       "      <th>Symbiont8 counts</th>\n",
       "      <th>Symbiont8 percentage</th>\n",
       "      <th>No hits counts</th>\n",
       "      <th>No hits percentage</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>052022_BEL_CBC_T1_12_MCAV</td>\n",
       "      <td>052022_BEL_CBC_T1_12_MCAV_R1_001_val_1.fq</td>\n",
       "      <td>20742228.0</td>\n",
       "      <td>20754538</td>\n",
       "      <td>6447936.0</td>\n",
       "      <td>NaN</td>\n",
       "      <td>68.91</td>\n",
       "      <td>31.09</td>\n",
       "      <td>100.0</td>\n",
       "      <td>6447936.0</td>\n",
       "      <td>...</td>\n",
       "      <td>341064.0</td>\n",
       "      <td>5.29</td>\n",
       "      <td>184526.0</td>\n",
       "      <td>2.86</td>\n",
       "      <td>199007.0</td>\n",
       "      <td>3.08</td>\n",
       "      <td>4455857.0</td>\n",
       "      <td>69.10</td>\n",
       "      <td>1944697.0</td>\n",
       "      <td>30.16</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>052022_BEL_CBC_T1_12_MCAV</td>\n",
       "      <td>052022_BEL_CBC_T1_12_MCAV_R2_001_val_2.fq</td>\n",
       "      <td>20742228.0</td>\n",
       "      <td>20754538</td>\n",
       "      <td>6447936.0</td>\n",
       "      <td>NaN</td>\n",
       "      <td>68.91</td>\n",
       "      <td>31.09</td>\n",
       "      <td>100.0</td>\n",
       "      <td>6447936.0</td>\n",
       "      <td>...</td>\n",
       "      <td>327436.0</td>\n",
       "      <td>5.08</td>\n",
       "      <td>181482.0</td>\n",
       "      <td>2.82</td>\n",
       "      <td>195345.0</td>\n",
       "      <td>3.03</td>\n",
       "      <td>4265102.0</td>\n",
       "      <td>66.14</td>\n",
       "      <td>2133622.0</td>\n",
       "      <td>33.09</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>052022_BEL_CBC_T1_13_MCAV</td>\n",
       "      <td>052022_BEL_CBC_T1_13_MCAV_R1_001_val_1.fq</td>\n",
       "      <td>20861960.0</td>\n",
       "      <td>20870396</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>052022_BEL_CBC_T1_13_MCAV</td>\n",
       "      <td>052022_BEL_CBC_T1_13_MCAV_R2_001_val_2.fq</td>\n",
       "      <td>20861960.0</td>\n",
       "      <td>20870396</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>052022_BEL_CBC_T1_40_MCAV</td>\n",
       "      <td>052022_BEL_CBC_T1_40_MCAV_R1_001_val_1.fq</td>\n",
       "      <td>15419779.0</td>\n",
       "      <td>15425540</td>\n",
       "      <td>1843935.0</td>\n",
       "      <td>NaN</td>\n",
       "      <td>88.04</td>\n",
       "      <td>11.96</td>\n",
       "      <td>100.0</td>\n",
       "      <td>1843935.0</td>\n",
       "      <td>...</td>\n",
       "      <td>58574.0</td>\n",
       "      <td>3.18</td>\n",
       "      <td>43354.0</td>\n",
       "      <td>2.35</td>\n",
       "      <td>48012.0</td>\n",
       "      <td>2.60</td>\n",
       "      <td>449731.0</td>\n",
       "      <td>24.39</td>\n",
       "      <td>1378710.0</td>\n",
       "      <td>74.77</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>5</th>\n",
       "      <td>052022_BEL_CBC_T1_40_MCAV</td>\n",
       "      <td>052022_BEL_CBC_T1_40_MCAV_R2_001_val_2.fq</td>\n",
       "      <td>15419779.0</td>\n",
       "      <td>15425540</td>\n",
       "      <td>1843935.0</td>\n",
       "      <td>NaN</td>\n",
       "      <td>88.04</td>\n",
       "      <td>11.96</td>\n",
       "      <td>100.0</td>\n",
       "      <td>1843935.0</td>\n",
       "      <td>...</td>\n",
       "      <td>58466.0</td>\n",
       "      <td>3.17</td>\n",
       "      <td>43680.0</td>\n",
       "      <td>2.37</td>\n",
       "      <td>48003.0</td>\n",
       "      <td>2.60</td>\n",
       "      <td>440527.0</td>\n",
       "      <td>23.89</td>\n",
       "      <td>1387929.0</td>\n",
       "      <td>75.27</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>6</th>\n",
       "      <td>052022_BEL_CBC_T1_57_MCAV</td>\n",
       "      <td>052022_BEL_CBC_T1_57_MCAV_R1_001_val_1.fq</td>\n",
       "      <td>31964707.0</td>\n",
       "      <td>31976704</td>\n",
       "      <td>9131401.0</td>\n",
       "      <td>71.43</td>\n",
       "      <td>71.43</td>\n",
       "      <td>28.57</td>\n",
       "      <td>100.0</td>\n",
       "      <td>7263034.0</td>\n",
       "      <td>...</td>\n",
       "      <td>391796.0</td>\n",
       "      <td>5.40</td>\n",
       "      <td>271562.0</td>\n",
       "      <td>3.74</td>\n",
       "      <td>275939.0</td>\n",
       "      <td>3.80</td>\n",
       "      <td>4233566.0</td>\n",
       "      <td>58.29</td>\n",
       "      <td>2988012.0</td>\n",
       "      <td>41.14</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>7</th>\n",
       "      <td>052022_BEL_CBC_T1_57_MCAV</td>\n",
       "      <td>052022_BEL_CBC_T1_57_MCAV_R2_001_val_2.fq</td>\n",
       "      <td>31964707.0</td>\n",
       "      <td>31976704</td>\n",
       "      <td>9131401.0</td>\n",
       "      <td>71.43</td>\n",
       "      <td>71.43</td>\n",
       "      <td>28.57</td>\n",
       "      <td>100.0</td>\n",
       "      <td>7263034.0</td>\n",
       "      <td>...</td>\n",
       "      <td>392323.0</td>\n",
       "      <td>5.40</td>\n",
       "      <td>274510.0</td>\n",
       "      <td>3.78</td>\n",
       "      <td>278633.0</td>\n",
       "      <td>3.84</td>\n",
       "      <td>4183791.0</td>\n",
       "      <td>57.60</td>\n",
       "      <td>3035948.0</td>\n",
       "      <td>41.80</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>8</th>\n",
       "      <td>052022_BEL_CBC_T1_70_MCAV</td>\n",
       "      <td>052022_BEL_CBC_T1_70_MCAV_R1_001_val_1.fq</td>\n",
       "      <td>15776674.0</td>\n",
       "      <td>15782392</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>9</th>\n",
       "      <td>052022_BEL_CBC_T1_70_MCAV</td>\n",
       "      <td>052022_BEL_CBC_T1_70_MCAV_R2_001_val_2.fq</td>\n",
       "      <td>15776674.0</td>\n",
       "      <td>15782392</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>10</th>\n",
       "      <td>052022_BEL_CBC_T2_10_MCAV</td>\n",
       "      <td>052022_BEL_CBC_T2_10_MCAV_R1_001_val_1.fq</td>\n",
       "      <td>16876812.0</td>\n",
       "      <td>16900082</td>\n",
       "      <td>11697348.5</td>\n",
       "      <td>30.69</td>\n",
       "      <td>30.69</td>\n",
       "      <td>69.31</td>\n",
       "      <td>100.0</td>\n",
       "      <td>11241452.0</td>\n",
       "      <td>...</td>\n",
       "      <td>348530.0</td>\n",
       "      <td>3.11</td>\n",
       "      <td>225358.0</td>\n",
       "      <td>2.00</td>\n",
       "      <td>234412.0</td>\n",
       "      <td>2.08</td>\n",
       "      <td>3614590.0</td>\n",
       "      <td>32.16</td>\n",
       "      <td>7574490.0</td>\n",
       "      <td>67.38</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>11</th>\n",
       "      <td>052022_BEL_CBC_T2_10_MCAV</td>\n",
       "      <td>052022_BEL_CBC_T2_10_MCAV_R2_001_val_2.fq</td>\n",
       "      <td>16876812.0</td>\n",
       "      <td>16900082</td>\n",
       "      <td>11697348.5</td>\n",
       "      <td>30.69</td>\n",
       "      <td>30.69</td>\n",
       "      <td>69.31</td>\n",
       "      <td>100.0</td>\n",
       "      <td>11241452.0</td>\n",
       "      <td>...</td>\n",
       "      <td>352138.0</td>\n",
       "      <td>3.14</td>\n",
       "      <td>232470.0</td>\n",
       "      <td>2.06</td>\n",
       "      <td>240629.0</td>\n",
       "      <td>2.14</td>\n",
       "      <td>3575910.0</td>\n",
       "      <td>31.80</td>\n",
       "      <td>7611587.0</td>\n",
       "      <td>67.71</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>12</th>\n",
       "      <td>052022_BEL_CBC_T3_13_MCAV</td>\n",
       "      <td>052022_BEL_CBC_T3_13_MCAV_R1_001_val_1.fq</td>\n",
       "      <td>13449647.0</td>\n",
       "      <td>13459995</td>\n",
       "      <td>2945179.5</td>\n",
       "      <td>78.10</td>\n",
       "      <td>78.10</td>\n",
       "      <td>21.90</td>\n",
       "      <td>100.0</td>\n",
       "      <td>2238983.0</td>\n",
       "      <td>...</td>\n",
       "      <td>88990.0</td>\n",
       "      <td>3.98</td>\n",
       "      <td>63594.0</td>\n",
       "      <td>2.84</td>\n",
       "      <td>65184.0</td>\n",
       "      <td>2.91</td>\n",
       "      <td>886635.0</td>\n",
       "      <td>39.60</td>\n",
       "      <td>1337120.0</td>\n",
       "      <td>59.72</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>13</th>\n",
       "      <td>052022_BEL_CBC_T3_13_MCAV</td>\n",
       "      <td>052022_BEL_CBC_T3_13_MCAV_R2_001_val_2.fq</td>\n",
       "      <td>13449647.0</td>\n",
       "      <td>13459995</td>\n",
       "      <td>2945179.5</td>\n",
       "      <td>78.10</td>\n",
       "      <td>78.10</td>\n",
       "      <td>21.90</td>\n",
       "      <td>100.0</td>\n",
       "      <td>2238983.0</td>\n",
       "      <td>...</td>\n",
       "      <td>88776.0</td>\n",
       "      <td>3.97</td>\n",
       "      <td>64063.0</td>\n",
       "      <td>2.86</td>\n",
       "      <td>65645.0</td>\n",
       "      <td>2.92</td>\n",
       "      <td>875471.0</td>\n",
       "      <td>39.10</td>\n",
       "      <td>1348091.0</td>\n",
       "      <td>60.21</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>14</th>\n",
       "      <td>052022_BEL_CBC_T3_19_MCAV</td>\n",
       "      <td>052022_BEL_CBC_T3_19_MCAV_R1_001_val_1.fq</td>\n",
       "      <td>9044371.0</td>\n",
       "      <td>9049169</td>\n",
       "      <td>2003485.5</td>\n",
       "      <td>77.85</td>\n",
       "      <td>77.85</td>\n",
       "      <td>22.15</td>\n",
       "      <td>100.0</td>\n",
       "      <td>1403471.0</td>\n",
       "      <td>...</td>\n",
       "      <td>51578.0</td>\n",
       "      <td>3.67</td>\n",
       "      <td>37688.0</td>\n",
       "      <td>2.68</td>\n",
       "      <td>41372.0</td>\n",
       "      <td>2.95</td>\n",
       "      <td>289974.0</td>\n",
       "      <td>20.66</td>\n",
       "      <td>1098496.0</td>\n",
       "      <td>78.27</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>15</th>\n",
       "      <td>052022_BEL_CBC_T3_19_MCAV</td>\n",
       "      <td>052022_BEL_CBC_T3_19_MCAV_R2_001_val_2.fq</td>\n",
       "      <td>9044371.0</td>\n",
       "      <td>9049169</td>\n",
       "      <td>2003485.5</td>\n",
       "      <td>77.85</td>\n",
       "      <td>77.85</td>\n",
       "      <td>22.15</td>\n",
       "      <td>100.0</td>\n",
       "      <td>1403471.0</td>\n",
       "      <td>...</td>\n",
       "      <td>52671.0</td>\n",
       "      <td>3.74</td>\n",
       "      <td>38795.0</td>\n",
       "      <td>2.77</td>\n",
       "      <td>42563.0</td>\n",
       "      <td>3.04</td>\n",
       "      <td>287009.0</td>\n",
       "      <td>20.45</td>\n",
       "      <td>1101303.0</td>\n",
       "      <td>78.47</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>16</th>\n",
       "      <td>052022_BEL_CBC_T3_1_MCAV</td>\n",
       "      <td>052022_BEL_CBC_T3_1_MCAV_R1_001_val_1.fq</td>\n",
       "      <td>5829824.0</td>\n",
       "      <td>5833913</td>\n",
       "      <td>4495655.5</td>\n",
       "      <td>22.89</td>\n",
       "      <td>22.89</td>\n",
       "      <td>77.11</td>\n",
       "      <td>100.0</td>\n",
       "      <td>4359957.0</td>\n",
       "      <td>...</td>\n",
       "      <td>299183.0</td>\n",
       "      <td>6.86</td>\n",
       "      <td>188593.0</td>\n",
       "      <td>4.32</td>\n",
       "      <td>190214.0</td>\n",
       "      <td>4.37</td>\n",
       "      <td>3810755.0</td>\n",
       "      <td>87.41</td>\n",
       "      <td>534966.0</td>\n",
       "      <td>12.27</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>17</th>\n",
       "      <td>052022_BEL_CBC_T3_1_MCAV</td>\n",
       "      <td>052022_BEL_CBC_T3_1_MCAV_R2_001_val_2.fq</td>\n",
       "      <td>5829824.0</td>\n",
       "      <td>5833913</td>\n",
       "      <td>4495655.5</td>\n",
       "      <td>22.89</td>\n",
       "      <td>22.89</td>\n",
       "      <td>77.11</td>\n",
       "      <td>100.0</td>\n",
       "      <td>4359957.0</td>\n",
       "      <td>...</td>\n",
       "      <td>295852.0</td>\n",
       "      <td>6.79</td>\n",
       "      <td>190195.0</td>\n",
       "      <td>4.36</td>\n",
       "      <td>191720.0</td>\n",
       "      <td>4.39</td>\n",
       "      <td>3730898.0</td>\n",
       "      <td>85.58</td>\n",
       "      <td>613881.0</td>\n",
       "      <td>14.08</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>18</th>\n",
       "      <td>052022_BEL_CBC_T3_40_MCAV</td>\n",
       "      <td>052022_BEL_CBC_T3_40_MCAV_R1_001_val_1.fq</td>\n",
       "      <td>26755227.0</td>\n",
       "      <td>26765873</td>\n",
       "      <td>7061478.0</td>\n",
       "      <td>NaN</td>\n",
       "      <td>73.61</td>\n",
       "      <td>26.39</td>\n",
       "      <td>100.0</td>\n",
       "      <td>7061478.0</td>\n",
       "      <td>...</td>\n",
       "      <td>46222.0</td>\n",
       "      <td>0.66</td>\n",
       "      <td>39584.0</td>\n",
       "      <td>0.55</td>\n",
       "      <td>43496.0</td>\n",
       "      <td>0.62</td>\n",
       "      <td>161318.0</td>\n",
       "      <td>2.28</td>\n",
       "      <td>6875054.0</td>\n",
       "      <td>97.36</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>19</th>\n",
       "      <td>052022_BEL_CBC_T3_40_MCAV</td>\n",
       "      <td>052022_BEL_CBC_T3_40_MCAV_R2_001_val_2.fq</td>\n",
       "      <td>26755227.0</td>\n",
       "      <td>26765873</td>\n",
       "      <td>7061478.0</td>\n",
       "      <td>NaN</td>\n",
       "      <td>73.61</td>\n",
       "      <td>26.39</td>\n",
       "      <td>100.0</td>\n",
       "      <td>7061478.0</td>\n",
       "      <td>...</td>\n",
       "      <td>48935.0</td>\n",
       "      <td>0.70</td>\n",
       "      <td>42842.0</td>\n",
       "      <td>0.60</td>\n",
       "      <td>46371.0</td>\n",
       "      <td>0.66</td>\n",
       "      <td>161074.0</td>\n",
       "      <td>2.29</td>\n",
       "      <td>6874348.0</td>\n",
       "      <td>97.35</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>20</th>\n",
       "      <td>052022_BEL_CBC_T3_48_MCAV</td>\n",
       "      <td>052022_BEL_CBC_T3_48_MCAV_R1_001_val_1.fq</td>\n",
       "      <td>19490564.0</td>\n",
       "      <td>19499503</td>\n",
       "      <td>2525681.0</td>\n",
       "      <td>NaN</td>\n",
       "      <td>87.04</td>\n",
       "      <td>12.96</td>\n",
       "      <td>100.0</td>\n",
       "      <td>2525681.0</td>\n",
       "      <td>...</td>\n",
       "      <td>66444.0</td>\n",
       "      <td>2.63</td>\n",
       "      <td>60419.0</td>\n",
       "      <td>2.40</td>\n",
       "      <td>69629.0</td>\n",
       "      <td>2.75</td>\n",
       "      <td>224309.0</td>\n",
       "      <td>8.88</td>\n",
       "      <td>2273112.0</td>\n",
       "      <td>90.00</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>21</th>\n",
       "      <td>052022_BEL_CBC_T3_48_MCAV</td>\n",
       "      <td>052022_BEL_CBC_T3_48_MCAV_R2_001_val_2.fq</td>\n",
       "      <td>19490564.0</td>\n",
       "      <td>19499503</td>\n",
       "      <td>2525681.0</td>\n",
       "      <td>NaN</td>\n",
       "      <td>87.04</td>\n",
       "      <td>12.96</td>\n",
       "      <td>100.0</td>\n",
       "      <td>2525681.0</td>\n",
       "      <td>...</td>\n",
       "      <td>67032.0</td>\n",
       "      <td>2.65</td>\n",
       "      <td>60888.0</td>\n",
       "      <td>2.42</td>\n",
       "      <td>69914.0</td>\n",
       "      <td>2.76</td>\n",
       "      <td>221405.0</td>\n",
       "      <td>8.76</td>\n",
       "      <td>2275638.0</td>\n",
       "      <td>90.10</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>22</th>\n",
       "      <td>052022_BEL_CBC_T3_49_MCAV</td>\n",
       "      <td>052022_BEL_CBC_T3_49_MCAV_R1_001_val_1.fq</td>\n",
       "      <td>15749618.0</td>\n",
       "      <td>15754799</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>23</th>\n",
       "      <td>052022_BEL_CBC_T3_49_MCAV</td>\n",
       "      <td>052022_BEL_CBC_T3_49_MCAV_R2_001_val_2.fq</td>\n",
       "      <td>15749618.0</td>\n",
       "      <td>15754799</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>24</th>\n",
       "      <td>052022_BEL_CBC_T3_51_MCAV</td>\n",
       "      <td>052022_BEL_CBC_T3_51_MCAV_R1_001_val_1.fq</td>\n",
       "      <td>13877944.0</td>\n",
       "      <td>13893014</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>25</th>\n",
       "      <td>052022_BEL_CBC_T3_51_MCAV</td>\n",
       "      <td>052022_BEL_CBC_T3_51_MCAV_R2_001_val_2.fq</td>\n",
       "      <td>13877944.0</td>\n",
       "      <td>13893014</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>26</th>\n",
       "      <td>052022_BEL_CBC_T3_60_MCAV</td>\n",
       "      <td>052022_BEL_CBC_T3_60_MCAV_R1_001_val_1.fq</td>\n",
       "      <td>76712696.0</td>\n",
       "      <td>76770478</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>27</th>\n",
       "      <td>052022_BEL_CBC_T3_60_MCAV</td>\n",
       "      <td>052022_BEL_CBC_T3_60_MCAV_R2_001_val_2.fq</td>\n",
       "      <td>76712696.0</td>\n",
       "      <td>76770478</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>28</th>\n",
       "      <td>062019_BEL_CBC_T1_16_MCAV</td>\n",
       "      <td>062019_BEL_CBC_T1_16_MCAV_R1_001_val_1.fq</td>\n",
       "      <td>25807031.0</td>\n",
       "      <td>25812745</td>\n",
       "      <td>8131957.5</td>\n",
       "      <td>68.49</td>\n",
       "      <td>68.49</td>\n",
       "      <td>31.51</td>\n",
       "      <td>100.0</td>\n",
       "      <td>7533935.0</td>\n",
       "      <td>...</td>\n",
       "      <td>460588.0</td>\n",
       "      <td>6.12</td>\n",
       "      <td>246410.0</td>\n",
       "      <td>3.27</td>\n",
       "      <td>254458.0</td>\n",
       "      <td>3.37</td>\n",
       "      <td>4986820.0</td>\n",
       "      <td>66.19</td>\n",
       "      <td>2494485.0</td>\n",
       "      <td>33.11</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>29</th>\n",
       "      <td>062019_BEL_CBC_T1_16_MCAV</td>\n",
       "      <td>062019_BEL_CBC_T1_16_MCAV_R2_001_val_2.fq</td>\n",
       "      <td>25807031.0</td>\n",
       "      <td>25812745</td>\n",
       "      <td>8131957.5</td>\n",
       "      <td>68.49</td>\n",
       "      <td>68.49</td>\n",
       "      <td>31.51</td>\n",
       "      <td>100.0</td>\n",
       "      <td>7533935.0</td>\n",
       "      <td>...</td>\n",
       "      <td>465440.0</td>\n",
       "      <td>6.17</td>\n",
       "      <td>252927.0</td>\n",
       "      <td>3.36</td>\n",
       "      <td>260684.0</td>\n",
       "      <td>3.46</td>\n",
       "      <td>4965001.0</td>\n",
       "      <td>65.90</td>\n",
       "      <td>2515580.0</td>\n",
       "      <td>33.39</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>30</th>\n",
       "      <td>062019_BEL_CBC_T1_20_MCAV</td>\n",
       "      <td>062019_BEL_CBC_T1_20_MCAV_R1_001_val_1.fq</td>\n",
       "      <td>8517445.0</td>\n",
       "      <td>8521485</td>\n",
       "      <td>8458232.5</td>\n",
       "      <td>0.70</td>\n",
       "      <td>0.70</td>\n",
       "      <td>99.30</td>\n",
       "      <td>100.0</td>\n",
       "      <td>8436904.0</td>\n",
       "      <td>...</td>\n",
       "      <td>139682.0</td>\n",
       "      <td>1.66</td>\n",
       "      <td>320850.0</td>\n",
       "      <td>3.80</td>\n",
       "      <td>635729.0</td>\n",
       "      <td>7.53</td>\n",
       "      <td>139845.0</td>\n",
       "      <td>1.66</td>\n",
       "      <td>7740015.0</td>\n",
       "      <td>91.74</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>31</th>\n",
       "      <td>062019_BEL_CBC_T1_20_MCAV</td>\n",
       "      <td>062019_BEL_CBC_T1_20_MCAV_R2_001_val_2.fq</td>\n",
       "      <td>8517445.0</td>\n",
       "      <td>8521485</td>\n",
       "      <td>8458232.5</td>\n",
       "      <td>0.70</td>\n",
       "      <td>0.70</td>\n",
       "      <td>99.30</td>\n",
       "      <td>100.0</td>\n",
       "      <td>8436904.0</td>\n",
       "      <td>...</td>\n",
       "      <td>142415.0</td>\n",
       "      <td>1.69</td>\n",
       "      <td>317875.0</td>\n",
       "      <td>3.77</td>\n",
       "      <td>629783.0</td>\n",
       "      <td>7.47</td>\n",
       "      <td>143132.0</td>\n",
       "      <td>1.70</td>\n",
       "      <td>7744234.0</td>\n",
       "      <td>91.79</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>32</th>\n",
       "      <td>062019_BEL_CBC_T1_24_MCAV</td>\n",
       "      <td>062019_BEL_CBC_T1_24_MCAV_R1_001_val_1.fq</td>\n",
       "      <td>18885388.0</td>\n",
       "      <td>18892465</td>\n",
       "      <td>4516033.5</td>\n",
       "      <td>76.09</td>\n",
       "      <td>76.09</td>\n",
       "      <td>23.91</td>\n",
       "      <td>100.0</td>\n",
       "      <td>3891938.0</td>\n",
       "      <td>...</td>\n",
       "      <td>289321.0</td>\n",
       "      <td>7.43</td>\n",
       "      <td>145869.0</td>\n",
       "      <td>3.75</td>\n",
       "      <td>166964.0</td>\n",
       "      <td>4.28</td>\n",
       "      <td>2056350.0</td>\n",
       "      <td>52.84</td>\n",
       "      <td>1777058.0</td>\n",
       "      <td>45.66</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>33</th>\n",
       "      <td>062019_BEL_CBC_T1_24_MCAV</td>\n",
       "      <td>062019_BEL_CBC_T1_24_MCAV_R2_001_val_2.fq</td>\n",
       "      <td>18885388.0</td>\n",
       "      <td>18892465</td>\n",
       "      <td>4516033.5</td>\n",
       "      <td>76.09</td>\n",
       "      <td>76.09</td>\n",
       "      <td>23.91</td>\n",
       "      <td>100.0</td>\n",
       "      <td>3891938.0</td>\n",
       "      <td>...</td>\n",
       "      <td>289537.0</td>\n",
       "      <td>7.44</td>\n",
       "      <td>148316.0</td>\n",
       "      <td>3.81</td>\n",
       "      <td>169217.0</td>\n",
       "      <td>4.35</td>\n",
       "      <td>2038522.0</td>\n",
       "      <td>52.38</td>\n",
       "      <td>1794183.0</td>\n",
       "      <td>46.10</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>34</th>\n",
       "      <td>062019_BEL_CBC_T1_3_MCAV</td>\n",
       "      <td>062019_BEL_CBC_T1_3_MCAV_R1_001_val_1.fq</td>\n",
       "      <td>4724478.0</td>\n",
       "      <td>4755741</td>\n",
       "      <td>3190834.0</td>\n",
       "      <td>32.46</td>\n",
       "      <td>32.46</td>\n",
       "      <td>67.54</td>\n",
       "      <td>100.0</td>\n",
       "      <td>3141837.0</td>\n",
       "      <td>...</td>\n",
       "      <td>137016.0</td>\n",
       "      <td>4.36</td>\n",
       "      <td>77758.0</td>\n",
       "      <td>2.48</td>\n",
       "      <td>79209.0</td>\n",
       "      <td>2.53</td>\n",
       "      <td>1807998.0</td>\n",
       "      <td>57.55</td>\n",
       "      <td>1317372.0</td>\n",
       "      <td>41.93</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>35</th>\n",
       "      <td>062019_BEL_CBC_T1_3_MCAV</td>\n",
       "      <td>062019_BEL_CBC_T1_3_MCAV_R2_001_val_2.fq</td>\n",
       "      <td>4724478.0</td>\n",
       "      <td>4755741</td>\n",
       "      <td>3190834.0</td>\n",
       "      <td>32.46</td>\n",
       "      <td>32.46</td>\n",
       "      <td>67.54</td>\n",
       "      <td>100.0</td>\n",
       "      <td>3141837.0</td>\n",
       "      <td>...</td>\n",
       "      <td>128455.0</td>\n",
       "      <td>4.10</td>\n",
       "      <td>75281.0</td>\n",
       "      <td>2.39</td>\n",
       "      <td>75642.0</td>\n",
       "      <td>2.41</td>\n",
       "      <td>1692589.0</td>\n",
       "      <td>53.87</td>\n",
       "      <td>1432049.0</td>\n",
       "      <td>45.58</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>36</th>\n",
       "      <td>062019_BEL_CBC_T2_16_MCAV</td>\n",
       "      <td>062019_BEL_CBC_T2_16_MCAV_R1_001_val_1.fq</td>\n",
       "      <td>19910121.0</td>\n",
       "      <td>19924257</td>\n",
       "      <td>3151347.5</td>\n",
       "      <td>84.17</td>\n",
       "      <td>84.17</td>\n",
       "      <td>15.83</td>\n",
       "      <td>100.0</td>\n",
       "      <td>2427227.0</td>\n",
       "      <td>...</td>\n",
       "      <td>63705.0</td>\n",
       "      <td>2.63</td>\n",
       "      <td>55328.0</td>\n",
       "      <td>2.27</td>\n",
       "      <td>61103.0</td>\n",
       "      <td>2.52</td>\n",
       "      <td>150979.0</td>\n",
       "      <td>6.22</td>\n",
       "      <td>2251981.0</td>\n",
       "      <td>92.78</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>37</th>\n",
       "      <td>062019_BEL_CBC_T2_16_MCAV</td>\n",
       "      <td>062019_BEL_CBC_T2_16_MCAV_R2_001_val_2.fq</td>\n",
       "      <td>19910121.0</td>\n",
       "      <td>19924257</td>\n",
       "      <td>3151347.5</td>\n",
       "      <td>84.17</td>\n",
       "      <td>84.17</td>\n",
       "      <td>15.83</td>\n",
       "      <td>100.0</td>\n",
       "      <td>2427227.0</td>\n",
       "      <td>...</td>\n",
       "      <td>67383.0</td>\n",
       "      <td>2.78</td>\n",
       "      <td>59102.0</td>\n",
       "      <td>2.43</td>\n",
       "      <td>64585.0</td>\n",
       "      <td>2.67</td>\n",
       "      <td>154508.0</td>\n",
       "      <td>6.37</td>\n",
       "      <td>2248583.0</td>\n",
       "      <td>92.64</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>38</th>\n",
       "      <td>062019_BEL_CBC_T2_5_MCAV</td>\n",
       "      <td>062019_BEL_CBC_T2_5_MCAV_R1_001_val_1.fq</td>\n",
       "      <td>25584829.0</td>\n",
       "      <td>25736116</td>\n",
       "      <td>9711853.0</td>\n",
       "      <td>62.04</td>\n",
       "      <td>62.04</td>\n",
       "      <td>37.96</td>\n",
       "      <td>100.0</td>\n",
       "      <td>9068181.0</td>\n",
       "      <td>...</td>\n",
       "      <td>555125.0</td>\n",
       "      <td>6.12</td>\n",
       "      <td>350782.0</td>\n",
       "      <td>3.86</td>\n",
       "      <td>354837.0</td>\n",
       "      <td>3.90</td>\n",
       "      <td>6352780.0</td>\n",
       "      <td>70.04</td>\n",
       "      <td>2676927.0</td>\n",
       "      <td>29.52</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>39</th>\n",
       "      <td>062019_BEL_CBC_T2_5_MCAV</td>\n",
       "      <td>062019_BEL_CBC_T2_5_MCAV_R2_001_val_2.fq</td>\n",
       "      <td>25584829.0</td>\n",
       "      <td>25736116</td>\n",
       "      <td>9711853.0</td>\n",
       "      <td>62.04</td>\n",
       "      <td>62.04</td>\n",
       "      <td>37.96</td>\n",
       "      <td>100.0</td>\n",
       "      <td>9068181.0</td>\n",
       "      <td>...</td>\n",
       "      <td>520787.0</td>\n",
       "      <td>5.75</td>\n",
       "      <td>338391.0</td>\n",
       "      <td>3.73</td>\n",
       "      <td>340166.0</td>\n",
       "      <td>3.75</td>\n",
       "      <td>5898062.0</td>\n",
       "      <td>65.04</td>\n",
       "      <td>3128522.0</td>\n",
       "      <td>34.50</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>40</th>\n",
       "      <td>062019_BEL_CBC_T3_14_MCAV</td>\n",
       "      <td>062019_BEL_CBC_T3_14_MCAV_R1_001_val_1.fq</td>\n",
       "      <td>21400877.0</td>\n",
       "      <td>21409896</td>\n",
       "      <td>4335512.5</td>\n",
       "      <td>79.74</td>\n",
       "      <td>79.74</td>\n",
       "      <td>20.26</td>\n",
       "      <td>100.0</td>\n",
       "      <td>3722331.0</td>\n",
       "      <td>...</td>\n",
       "      <td>153337.0</td>\n",
       "      <td>4.12</td>\n",
       "      <td>103263.0</td>\n",
       "      <td>2.78</td>\n",
       "      <td>105682.0</td>\n",
       "      <td>2.84</td>\n",
       "      <td>1656380.0</td>\n",
       "      <td>44.50</td>\n",
       "      <td>2047282.0</td>\n",
       "      <td>55.00</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>41</th>\n",
       "      <td>062019_BEL_CBC_T3_14_MCAV</td>\n",
       "      <td>062019_BEL_CBC_T3_14_MCAV_R2_001_val_2.fq</td>\n",
       "      <td>21400877.0</td>\n",
       "      <td>21409896</td>\n",
       "      <td>4335512.5</td>\n",
       "      <td>79.74</td>\n",
       "      <td>79.74</td>\n",
       "      <td>20.26</td>\n",
       "      <td>100.0</td>\n",
       "      <td>3722331.0</td>\n",
       "      <td>...</td>\n",
       "      <td>151993.0</td>\n",
       "      <td>4.08</td>\n",
       "      <td>103219.0</td>\n",
       "      <td>2.78</td>\n",
       "      <td>105307.0</td>\n",
       "      <td>2.83</td>\n",
       "      <td>1636028.0</td>\n",
       "      <td>43.95</td>\n",
       "      <td>2066638.0</td>\n",
       "      <td>55.52</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>42</th>\n",
       "      <td>062019_BEL_CBC_T3_15_MCAV</td>\n",
       "      <td>062019_BEL_CBC_T3_15_MCAV_R1_001_val_1.fq</td>\n",
       "      <td>19854618.0</td>\n",
       "      <td>19866857</td>\n",
       "      <td>3078684.5</td>\n",
       "      <td>84.49</td>\n",
       "      <td>84.49</td>\n",
       "      <td>15.51</td>\n",
       "      <td>100.0</td>\n",
       "      <td>2321387.0</td>\n",
       "      <td>...</td>\n",
       "      <td>76441.0</td>\n",
       "      <td>3.30</td>\n",
       "      <td>52486.0</td>\n",
       "      <td>2.26</td>\n",
       "      <td>55296.0</td>\n",
       "      <td>2.38</td>\n",
       "      <td>557074.0</td>\n",
       "      <td>23.99</td>\n",
       "      <td>1748700.0</td>\n",
       "      <td>75.33</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>43</th>\n",
       "      <td>062019_BEL_CBC_T3_15_MCAV</td>\n",
       "      <td>062019_BEL_CBC_T3_15_MCAV_R2_001_val_2.fq</td>\n",
       "      <td>19854618.0</td>\n",
       "      <td>19866857</td>\n",
       "      <td>3078684.5</td>\n",
       "      <td>84.49</td>\n",
       "      <td>84.49</td>\n",
       "      <td>15.51</td>\n",
       "      <td>100.0</td>\n",
       "      <td>2321387.0</td>\n",
       "      <td>...</td>\n",
       "      <td>75423.0</td>\n",
       "      <td>3.25</td>\n",
       "      <td>52191.0</td>\n",
       "      <td>2.25</td>\n",
       "      <td>54879.0</td>\n",
       "      <td>2.36</td>\n",
       "      <td>544944.0</td>\n",
       "      <td>23.48</td>\n",
       "      <td>1760307.0</td>\n",
       "      <td>75.83</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>44</th>\n",
       "      <td>062019_BEL_CBC_T3_16_MCAV</td>\n",
       "      <td>062019_BEL_CBC_T3_16_MCAV_R1_001_val_1.fq</td>\n",
       "      <td>3541481.0</td>\n",
       "      <td>3548005</td>\n",
       "      <td>1134331.5</td>\n",
       "      <td>67.97</td>\n",
       "      <td>67.97</td>\n",
       "      <td>32.03</td>\n",
       "      <td>100.0</td>\n",
       "      <td>1094028.0</td>\n",
       "      <td>...</td>\n",
       "      <td>60564.0</td>\n",
       "      <td>5.54</td>\n",
       "      <td>41847.0</td>\n",
       "      <td>3.82</td>\n",
       "      <td>41642.0</td>\n",
       "      <td>3.80</td>\n",
       "      <td>678595.0</td>\n",
       "      <td>62.01</td>\n",
       "      <td>405227.0</td>\n",
       "      <td>37.04</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>45</th>\n",
       "      <td>062019_BEL_CBC_T3_16_MCAV</td>\n",
       "      <td>062019_BEL_CBC_T3_16_MCAV_R2_001_val_2.fq</td>\n",
       "      <td>3541481.0</td>\n",
       "      <td>3548005</td>\n",
       "      <td>1134331.5</td>\n",
       "      <td>67.97</td>\n",
       "      <td>67.97</td>\n",
       "      <td>32.03</td>\n",
       "      <td>100.0</td>\n",
       "      <td>1094028.0</td>\n",
       "      <td>...</td>\n",
       "      <td>58345.0</td>\n",
       "      <td>5.34</td>\n",
       "      <td>41189.0</td>\n",
       "      <td>3.77</td>\n",
       "      <td>40735.0</td>\n",
       "      <td>3.71</td>\n",
       "      <td>656009.0</td>\n",
       "      <td>59.96</td>\n",
       "      <td>427436.0</td>\n",
       "      <td>39.07</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>46</th>\n",
       "      <td>062019_BEL_CBC_T3_1_MCAV</td>\n",
       "      <td>062019_BEL_CBC_T3_1_MCAV_R1_001_val_1.fq</td>\n",
       "      <td>29977945.0</td>\n",
       "      <td>30002521</td>\n",
       "      <td>7505458.5</td>\n",
       "      <td>74.96</td>\n",
       "      <td>74.96</td>\n",
       "      <td>25.04</td>\n",
       "      <td>100.0</td>\n",
       "      <td>6548146.0</td>\n",
       "      <td>...</td>\n",
       "      <td>359624.0</td>\n",
       "      <td>5.50</td>\n",
       "      <td>230286.0</td>\n",
       "      <td>3.52</td>\n",
       "      <td>241133.0</td>\n",
       "      <td>3.68</td>\n",
       "      <td>3790960.0</td>\n",
       "      <td>57.90</td>\n",
       "      <td>2711587.0</td>\n",
       "      <td>41.41</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>47</th>\n",
       "      <td>062019_BEL_CBC_T3_1_MCAV</td>\n",
       "      <td>062019_BEL_CBC_T3_1_MCAV_R2_001_val_2.fq</td>\n",
       "      <td>29977945.0</td>\n",
       "      <td>30002521</td>\n",
       "      <td>7505458.5</td>\n",
       "      <td>74.96</td>\n",
       "      <td>74.96</td>\n",
       "      <td>25.04</td>\n",
       "      <td>100.0</td>\n",
       "      <td>6548146.0</td>\n",
       "      <td>...</td>\n",
       "      <td>356142.0</td>\n",
       "      <td>5.45</td>\n",
       "      <td>230772.0</td>\n",
       "      <td>3.52</td>\n",
       "      <td>241158.0</td>\n",
       "      <td>3.68</td>\n",
       "      <td>3707773.0</td>\n",
       "      <td>56.63</td>\n",
       "      <td>2792784.0</td>\n",
       "      <td>42.65</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>48</th>\n",
       "      <td>062019_BEL_CBC_T3_6_MCAV</td>\n",
       "      <td>062019_BEL_CBC_T3_6_MCAV_R1_001_val_1.fq</td>\n",
       "      <td>2442857.0</td>\n",
       "      <td>2472061</td>\n",
       "      <td>1876312.0</td>\n",
       "      <td>23.19</td>\n",
       "      <td>23.19</td>\n",
       "      <td>76.81</td>\n",
       "      <td>100.0</td>\n",
       "      <td>1861982.0</td>\n",
       "      <td>...</td>\n",
       "      <td>32822.0</td>\n",
       "      <td>1.77</td>\n",
       "      <td>22756.0</td>\n",
       "      <td>1.22</td>\n",
       "      <td>23200.0</td>\n",
       "      <td>1.25</td>\n",
       "      <td>363028.0</td>\n",
       "      <td>19.49</td>\n",
       "      <td>1485116.0</td>\n",
       "      <td>79.76</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>49</th>\n",
       "      <td>062019_BEL_CBC_T3_6_MCAV</td>\n",
       "      <td>062019_BEL_CBC_T3_6_MCAV_R2_001_val_2.fq</td>\n",
       "      <td>2442857.0</td>\n",
       "      <td>2472061</td>\n",
       "      <td>1876312.0</td>\n",
       "      <td>23.19</td>\n",
       "      <td>23.19</td>\n",
       "      <td>76.81</td>\n",
       "      <td>100.0</td>\n",
       "      <td>1861982.0</td>\n",
       "      <td>...</td>\n",
       "      <td>31568.0</td>\n",
       "      <td>1.70</td>\n",
       "      <td>23536.0</td>\n",
       "      <td>1.26</td>\n",
       "      <td>22697.0</td>\n",
       "      <td>1.22</td>\n",
       "      <td>349640.0</td>\n",
       "      <td>18.78</td>\n",
       "      <td>1498709.0</td>\n",
       "      <td>80.49</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>50</th>\n",
       "      <td>062019_BEL_CBC_T3_8_MCAV</td>\n",
       "      <td>062019_BEL_CBC_T3_8_MCAV_R1_001_val_1.fq</td>\n",
       "      <td>79877382.0</td>\n",
       "      <td>79936588</td>\n",
       "      <td>79154694.5</td>\n",
       "      <td>0.90</td>\n",
       "      <td>0.90</td>\n",
       "      <td>99.10</td>\n",
       "      <td>100.0</td>\n",
       "      <td>78898318.0</td>\n",
       "      <td>...</td>\n",
       "      <td>967584.0</td>\n",
       "      <td>1.22</td>\n",
       "      <td>2626226.0</td>\n",
       "      <td>3.32</td>\n",
       "      <td>5400438.0</td>\n",
       "      <td>6.84</td>\n",
       "      <td>1030880.0</td>\n",
       "      <td>1.31</td>\n",
       "      <td>73067732.0</td>\n",
       "      <td>92.61</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>51</th>\n",
       "      <td>062019_BEL_CBC_T3_8_MCAV</td>\n",
       "      <td>062019_BEL_CBC_T3_8_MCAV_R2_001_val_2.fq</td>\n",
       "      <td>79877382.0</td>\n",
       "      <td>79936588</td>\n",
       "      <td>79154694.5</td>\n",
       "      <td>0.90</td>\n",
       "      <td>0.90</td>\n",
       "      <td>99.10</td>\n",
       "      <td>100.0</td>\n",
       "      <td>78898318.0</td>\n",
       "      <td>...</td>\n",
       "      <td>1045870.0</td>\n",
       "      <td>1.33</td>\n",
       "      <td>2685342.0</td>\n",
       "      <td>3.39</td>\n",
       "      <td>5442089.0</td>\n",
       "      <td>6.90</td>\n",
       "      <td>1118202.0</td>\n",
       "      <td>1.41</td>\n",
       "      <td>72988833.0</td>\n",
       "      <td>92.51</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>52</th>\n",
       "      <td>062019_BEL_CBC_T3_9_MCAV</td>\n",
       "      <td>062019_BEL_CBC_T3_9_MCAV_R1_001_val_1.fq</td>\n",
       "      <td>14032207.0</td>\n",
       "      <td>14036963</td>\n",
       "      <td>4568457.5</td>\n",
       "      <td>67.44</td>\n",
       "      <td>67.44</td>\n",
       "      <td>32.56</td>\n",
       "      <td>100.0</td>\n",
       "      <td>4166616.0</td>\n",
       "      <td>...</td>\n",
       "      <td>253410.0</td>\n",
       "      <td>6.08</td>\n",
       "      <td>145655.0</td>\n",
       "      <td>3.49</td>\n",
       "      <td>156626.0</td>\n",
       "      <td>3.76</td>\n",
       "      <td>2875922.0</td>\n",
       "      <td>69.02</td>\n",
       "      <td>1257901.0</td>\n",
       "      <td>30.19</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>53</th>\n",
       "      <td>062019_BEL_CBC_T3_9_MCAV</td>\n",
       "      <td>062019_BEL_CBC_T3_9_MCAV_R2_001_val_2.fq</td>\n",
       "      <td>14032207.0</td>\n",
       "      <td>14036963</td>\n",
       "      <td>4568457.5</td>\n",
       "      <td>67.44</td>\n",
       "      <td>67.44</td>\n",
       "      <td>32.56</td>\n",
       "      <td>100.0</td>\n",
       "      <td>4166616.0</td>\n",
       "      <td>...</td>\n",
       "      <td>253504.0</td>\n",
       "      <td>6.08</td>\n",
       "      <td>148677.0</td>\n",
       "      <td>3.56</td>\n",
       "      <td>159257.0</td>\n",
       "      <td>3.83</td>\n",
       "      <td>2827803.0</td>\n",
       "      <td>67.87</td>\n",
       "      <td>1304984.0</td>\n",
       "      <td>31.32</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>54 rows × 30 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "                       Sample                                   Filename  \\\n",
       "0   052022_BEL_CBC_T1_12_MCAV  052022_BEL_CBC_T1_12_MCAV_R1_001_val_1.fq   \n",
       "1   052022_BEL_CBC_T1_12_MCAV  052022_BEL_CBC_T1_12_MCAV_R2_001_val_2.fq   \n",
       "2   052022_BEL_CBC_T1_13_MCAV  052022_BEL_CBC_T1_13_MCAV_R1_001_val_1.fq   \n",
       "3   052022_BEL_CBC_T1_13_MCAV  052022_BEL_CBC_T1_13_MCAV_R2_001_val_2.fq   \n",
       "4   052022_BEL_CBC_T1_40_MCAV  052022_BEL_CBC_T1_40_MCAV_R1_001_val_1.fq   \n",
       "5   052022_BEL_CBC_T1_40_MCAV  052022_BEL_CBC_T1_40_MCAV_R2_001_val_2.fq   \n",
       "6   052022_BEL_CBC_T1_57_MCAV  052022_BEL_CBC_T1_57_MCAV_R1_001_val_1.fq   \n",
       "7   052022_BEL_CBC_T1_57_MCAV  052022_BEL_CBC_T1_57_MCAV_R2_001_val_2.fq   \n",
       "8   052022_BEL_CBC_T1_70_MCAV  052022_BEL_CBC_T1_70_MCAV_R1_001_val_1.fq   \n",
       "9   052022_BEL_CBC_T1_70_MCAV  052022_BEL_CBC_T1_70_MCAV_R2_001_val_2.fq   \n",
       "10  052022_BEL_CBC_T2_10_MCAV  052022_BEL_CBC_T2_10_MCAV_R1_001_val_1.fq   \n",
       "11  052022_BEL_CBC_T2_10_MCAV  052022_BEL_CBC_T2_10_MCAV_R2_001_val_2.fq   \n",
       "12  052022_BEL_CBC_T3_13_MCAV  052022_BEL_CBC_T3_13_MCAV_R1_001_val_1.fq   \n",
       "13  052022_BEL_CBC_T3_13_MCAV  052022_BEL_CBC_T3_13_MCAV_R2_001_val_2.fq   \n",
       "14  052022_BEL_CBC_T3_19_MCAV  052022_BEL_CBC_T3_19_MCAV_R1_001_val_1.fq   \n",
       "15  052022_BEL_CBC_T3_19_MCAV  052022_BEL_CBC_T3_19_MCAV_R2_001_val_2.fq   \n",
       "16   052022_BEL_CBC_T3_1_MCAV   052022_BEL_CBC_T3_1_MCAV_R1_001_val_1.fq   \n",
       "17   052022_BEL_CBC_T3_1_MCAV   052022_BEL_CBC_T3_1_MCAV_R2_001_val_2.fq   \n",
       "18  052022_BEL_CBC_T3_40_MCAV  052022_BEL_CBC_T3_40_MCAV_R1_001_val_1.fq   \n",
       "19  052022_BEL_CBC_T3_40_MCAV  052022_BEL_CBC_T3_40_MCAV_R2_001_val_2.fq   \n",
       "20  052022_BEL_CBC_T3_48_MCAV  052022_BEL_CBC_T3_48_MCAV_R1_001_val_1.fq   \n",
       "21  052022_BEL_CBC_T3_48_MCAV  052022_BEL_CBC_T3_48_MCAV_R2_001_val_2.fq   \n",
       "22  052022_BEL_CBC_T3_49_MCAV  052022_BEL_CBC_T3_49_MCAV_R1_001_val_1.fq   \n",
       "23  052022_BEL_CBC_T3_49_MCAV  052022_BEL_CBC_T3_49_MCAV_R2_001_val_2.fq   \n",
       "24  052022_BEL_CBC_T3_51_MCAV  052022_BEL_CBC_T3_51_MCAV_R1_001_val_1.fq   \n",
       "25  052022_BEL_CBC_T3_51_MCAV  052022_BEL_CBC_T3_51_MCAV_R2_001_val_2.fq   \n",
       "26  052022_BEL_CBC_T3_60_MCAV  052022_BEL_CBC_T3_60_MCAV_R1_001_val_1.fq   \n",
       "27  052022_BEL_CBC_T3_60_MCAV  052022_BEL_CBC_T3_60_MCAV_R2_001_val_2.fq   \n",
       "28  062019_BEL_CBC_T1_16_MCAV  062019_BEL_CBC_T1_16_MCAV_R1_001_val_1.fq   \n",
       "29  062019_BEL_CBC_T1_16_MCAV  062019_BEL_CBC_T1_16_MCAV_R2_001_val_2.fq   \n",
       "30  062019_BEL_CBC_T1_20_MCAV  062019_BEL_CBC_T1_20_MCAV_R1_001_val_1.fq   \n",
       "31  062019_BEL_CBC_T1_20_MCAV  062019_BEL_CBC_T1_20_MCAV_R2_001_val_2.fq   \n",
       "32  062019_BEL_CBC_T1_24_MCAV  062019_BEL_CBC_T1_24_MCAV_R1_001_val_1.fq   \n",
       "33  062019_BEL_CBC_T1_24_MCAV  062019_BEL_CBC_T1_24_MCAV_R2_001_val_2.fq   \n",
       "34   062019_BEL_CBC_T1_3_MCAV   062019_BEL_CBC_T1_3_MCAV_R1_001_val_1.fq   \n",
       "35   062019_BEL_CBC_T1_3_MCAV   062019_BEL_CBC_T1_3_MCAV_R2_001_val_2.fq   \n",
       "36  062019_BEL_CBC_T2_16_MCAV  062019_BEL_CBC_T2_16_MCAV_R1_001_val_1.fq   \n",
       "37  062019_BEL_CBC_T2_16_MCAV  062019_BEL_CBC_T2_16_MCAV_R2_001_val_2.fq   \n",
       "38   062019_BEL_CBC_T2_5_MCAV   062019_BEL_CBC_T2_5_MCAV_R1_001_val_1.fq   \n",
       "39   062019_BEL_CBC_T2_5_MCAV   062019_BEL_CBC_T2_5_MCAV_R2_001_val_2.fq   \n",
       "40  062019_BEL_CBC_T3_14_MCAV  062019_BEL_CBC_T3_14_MCAV_R1_001_val_1.fq   \n",
       "41  062019_BEL_CBC_T3_14_MCAV  062019_BEL_CBC_T3_14_MCAV_R2_001_val_2.fq   \n",
       "42  062019_BEL_CBC_T3_15_MCAV  062019_BEL_CBC_T3_15_MCAV_R1_001_val_1.fq   \n",
       "43  062019_BEL_CBC_T3_15_MCAV  062019_BEL_CBC_T3_15_MCAV_R2_001_val_2.fq   \n",
       "44  062019_BEL_CBC_T3_16_MCAV  062019_BEL_CBC_T3_16_MCAV_R1_001_val_1.fq   \n",
       "45  062019_BEL_CBC_T3_16_MCAV  062019_BEL_CBC_T3_16_MCAV_R2_001_val_2.fq   \n",
       "46   062019_BEL_CBC_T3_1_MCAV   062019_BEL_CBC_T3_1_MCAV_R1_001_val_1.fq   \n",
       "47   062019_BEL_CBC_T3_1_MCAV   062019_BEL_CBC_T3_1_MCAV_R2_001_val_2.fq   \n",
       "48   062019_BEL_CBC_T3_6_MCAV   062019_BEL_CBC_T3_6_MCAV_R1_001_val_1.fq   \n",
       "49   062019_BEL_CBC_T3_6_MCAV   062019_BEL_CBC_T3_6_MCAV_R2_001_val_2.fq   \n",
       "50   062019_BEL_CBC_T3_8_MCAV   062019_BEL_CBC_T3_8_MCAV_R1_001_val_1.fq   \n",
       "51   062019_BEL_CBC_T3_8_MCAV   062019_BEL_CBC_T3_8_MCAV_R2_001_val_2.fq   \n",
       "52   062019_BEL_CBC_T3_9_MCAV   062019_BEL_CBC_T3_9_MCAV_R1_001_val_1.fq   \n",
       "53   062019_BEL_CBC_T3_9_MCAV   062019_BEL_CBC_T3_9_MCAV_R2_001_val_2.fq   \n",
       "\n",
       "       post_qc  raw_seqs  Unmapped_Reads  percent_mapped_host  \\\n",
       "0   20742228.0  20754538       6447936.0                  NaN   \n",
       "1   20742228.0  20754538       6447936.0                  NaN   \n",
       "2   20861960.0  20870396             NaN                  NaN   \n",
       "3   20861960.0  20870396             NaN                  NaN   \n",
       "4   15419779.0  15425540       1843935.0                  NaN   \n",
       "5   15419779.0  15425540       1843935.0                  NaN   \n",
       "6   31964707.0  31976704       9131401.0                71.43   \n",
       "7   31964707.0  31976704       9131401.0                71.43   \n",
       "8   15776674.0  15782392             NaN                  NaN   \n",
       "9   15776674.0  15782392             NaN                  NaN   \n",
       "10  16876812.0  16900082      11697348.5                30.69   \n",
       "11  16876812.0  16900082      11697348.5                30.69   \n",
       "12  13449647.0  13459995       2945179.5                78.10   \n",
       "13  13449647.0  13459995       2945179.5                78.10   \n",
       "14   9044371.0   9049169       2003485.5                77.85   \n",
       "15   9044371.0   9049169       2003485.5                77.85   \n",
       "16   5829824.0   5833913       4495655.5                22.89   \n",
       "17   5829824.0   5833913       4495655.5                22.89   \n",
       "18  26755227.0  26765873       7061478.0                  NaN   \n",
       "19  26755227.0  26765873       7061478.0                  NaN   \n",
       "20  19490564.0  19499503       2525681.0                  NaN   \n",
       "21  19490564.0  19499503       2525681.0                  NaN   \n",
       "22  15749618.0  15754799             NaN                  NaN   \n",
       "23  15749618.0  15754799             NaN                  NaN   \n",
       "24  13877944.0  13893014             NaN                  NaN   \n",
       "25  13877944.0  13893014             NaN                  NaN   \n",
       "26  76712696.0  76770478             NaN                  NaN   \n",
       "27  76712696.0  76770478             NaN                  NaN   \n",
       "28  25807031.0  25812745       8131957.5                68.49   \n",
       "29  25807031.0  25812745       8131957.5                68.49   \n",
       "30   8517445.0   8521485       8458232.5                 0.70   \n",
       "31   8517445.0   8521485       8458232.5                 0.70   \n",
       "32  18885388.0  18892465       4516033.5                76.09   \n",
       "33  18885388.0  18892465       4516033.5                76.09   \n",
       "34   4724478.0   4755741       3190834.0                32.46   \n",
       "35   4724478.0   4755741       3190834.0                32.46   \n",
       "36  19910121.0  19924257       3151347.5                84.17   \n",
       "37  19910121.0  19924257       3151347.5                84.17   \n",
       "38  25584829.0  25736116       9711853.0                62.04   \n",
       "39  25584829.0  25736116       9711853.0                62.04   \n",
       "40  21400877.0  21409896       4335512.5                79.74   \n",
       "41  21400877.0  21409896       4335512.5                79.74   \n",
       "42  19854618.0  19866857       3078684.5                84.49   \n",
       "43  19854618.0  19866857       3078684.5                84.49   \n",
       "44   3541481.0   3548005       1134331.5                67.97   \n",
       "45   3541481.0   3548005       1134331.5                67.97   \n",
       "46  29977945.0  30002521       7505458.5                74.96   \n",
       "47  29977945.0  30002521       7505458.5                74.96   \n",
       "48   2442857.0   2472061       1876312.0                23.19   \n",
       "49   2442857.0   2472061       1876312.0                23.19   \n",
       "50  79877382.0  79936588      79154694.5                 0.90   \n",
       "51  79877382.0  79936588      79154694.5                 0.90   \n",
       "52  14032207.0  14036963       4568457.5                67.44   \n",
       "53  14032207.0  14036963       4568457.5                67.44   \n",
       "\n",
       "    percent_mapped_calculated  percent_unmapped_calculated  \\\n",
       "0                       68.91                        31.09   \n",
       "1                       68.91                        31.09   \n",
       "2                         NaN                          NaN   \n",
       "3                         NaN                          NaN   \n",
       "4                       88.04                        11.96   \n",
       "5                       88.04                        11.96   \n",
       "6                       71.43                        28.57   \n",
       "7                       71.43                        28.57   \n",
       "8                         NaN                          NaN   \n",
       "9                         NaN                          NaN   \n",
       "10                      30.69                        69.31   \n",
       "11                      30.69                        69.31   \n",
       "12                      78.10                        21.90   \n",
       "13                      78.10                        21.90   \n",
       "14                      77.85                        22.15   \n",
       "15                      77.85                        22.15   \n",
       "16                      22.89                        77.11   \n",
       "17                      22.89                        77.11   \n",
       "18                      73.61                        26.39   \n",
       "19                      73.61                        26.39   \n",
       "20                      87.04                        12.96   \n",
       "21                      87.04                        12.96   \n",
       "22                        NaN                          NaN   \n",
       "23                        NaN                          NaN   \n",
       "24                        NaN                          NaN   \n",
       "25                        NaN                          NaN   \n",
       "26                        NaN                          NaN   \n",
       "27                        NaN                          NaN   \n",
       "28                      68.49                        31.51   \n",
       "29                      68.49                        31.51   \n",
       "30                       0.70                        99.30   \n",
       "31                       0.70                        99.30   \n",
       "32                      76.09                        23.91   \n",
       "33                      76.09                        23.91   \n",
       "34                      32.46                        67.54   \n",
       "35                      32.46                        67.54   \n",
       "36                      84.17                        15.83   \n",
       "37                      84.17                        15.83   \n",
       "38                      62.04                        37.96   \n",
       "39                      62.04                        37.96   \n",
       "40                      79.74                        20.26   \n",
       "41                      79.74                        20.26   \n",
       "42                      84.49                        15.51   \n",
       "43                      84.49                        15.51   \n",
       "44                      67.97                        32.03   \n",
       "45                      67.97                        32.03   \n",
       "46                      74.96                        25.04   \n",
       "47                      74.96                        25.04   \n",
       "48                      23.19                        76.81   \n",
       "49                      23.19                        76.81   \n",
       "50                       0.90                        99.10   \n",
       "51                       0.90                        99.10   \n",
       "52                      67.44                        32.56   \n",
       "53                      67.44                        32.56   \n",
       "\n",
       "    total_percent_calculated  host_removed_reads(fqscreen_input)  ...  \\\n",
       "0                      100.0                           6447936.0  ...   \n",
       "1                      100.0                           6447936.0  ...   \n",
       "2                        NaN                                 NaN  ...   \n",
       "3                        NaN                                 NaN  ...   \n",
       "4                      100.0                           1843935.0  ...   \n",
       "5                      100.0                           1843935.0  ...   \n",
       "6                      100.0                           7263034.0  ...   \n",
       "7                      100.0                           7263034.0  ...   \n",
       "8                        NaN                                 NaN  ...   \n",
       "9                        NaN                                 NaN  ...   \n",
       "10                     100.0                          11241452.0  ...   \n",
       "11                     100.0                          11241452.0  ...   \n",
       "12                     100.0                           2238983.0  ...   \n",
       "13                     100.0                           2238983.0  ...   \n",
       "14                     100.0                           1403471.0  ...   \n",
       "15                     100.0                           1403471.0  ...   \n",
       "16                     100.0                           4359957.0  ...   \n",
       "17                     100.0                           4359957.0  ...   \n",
       "18                     100.0                           7061478.0  ...   \n",
       "19                     100.0                           7061478.0  ...   \n",
       "20                     100.0                           2525681.0  ...   \n",
       "21                     100.0                           2525681.0  ...   \n",
       "22                       NaN                                 NaN  ...   \n",
       "23                       NaN                                 NaN  ...   \n",
       "24                       NaN                                 NaN  ...   \n",
       "25                       NaN                                 NaN  ...   \n",
       "26                       NaN                                 NaN  ...   \n",
       "27                       NaN                                 NaN  ...   \n",
       "28                     100.0                           7533935.0  ...   \n",
       "29                     100.0                           7533935.0  ...   \n",
       "30                     100.0                           8436904.0  ...   \n",
       "31                     100.0                           8436904.0  ...   \n",
       "32                     100.0                           3891938.0  ...   \n",
       "33                     100.0                           3891938.0  ...   \n",
       "34                     100.0                           3141837.0  ...   \n",
       "35                     100.0                           3141837.0  ...   \n",
       "36                     100.0                           2427227.0  ...   \n",
       "37                     100.0                           2427227.0  ...   \n",
       "38                     100.0                           9068181.0  ...   \n",
       "39                     100.0                           9068181.0  ...   \n",
       "40                     100.0                           3722331.0  ...   \n",
       "41                     100.0                           3722331.0  ...   \n",
       "42                     100.0                           2321387.0  ...   \n",
       "43                     100.0                           2321387.0  ...   \n",
       "44                     100.0                           1094028.0  ...   \n",
       "45                     100.0                           1094028.0  ...   \n",
       "46                     100.0                           6548146.0  ...   \n",
       "47                     100.0                           6548146.0  ...   \n",
       "48                     100.0                           1861982.0  ...   \n",
       "49                     100.0                           1861982.0  ...   \n",
       "50                     100.0                          78898318.0  ...   \n",
       "51                     100.0                          78898318.0  ...   \n",
       "52                     100.0                           4166616.0  ...   \n",
       "53                     100.0                           4166616.0  ...   \n",
       "\n",
       "    Symbiont5 counts  Symbiont5 percentage  Symbiont6 counts  \\\n",
       "0           341064.0                  5.29          184526.0   \n",
       "1           327436.0                  5.08          181482.0   \n",
       "2                NaN                   NaN               NaN   \n",
       "3                NaN                   NaN               NaN   \n",
       "4            58574.0                  3.18           43354.0   \n",
       "5            58466.0                  3.17           43680.0   \n",
       "6           391796.0                  5.40          271562.0   \n",
       "7           392323.0                  5.40          274510.0   \n",
       "8                NaN                   NaN               NaN   \n",
       "9                NaN                   NaN               NaN   \n",
       "10          348530.0                  3.11          225358.0   \n",
       "11          352138.0                  3.14          232470.0   \n",
       "12           88990.0                  3.98           63594.0   \n",
       "13           88776.0                  3.97           64063.0   \n",
       "14           51578.0                  3.67           37688.0   \n",
       "15           52671.0                  3.74           38795.0   \n",
       "16          299183.0                  6.86          188593.0   \n",
       "17          295852.0                  6.79          190195.0   \n",
       "18           46222.0                  0.66           39584.0   \n",
       "19           48935.0                  0.70           42842.0   \n",
       "20           66444.0                  2.63           60419.0   \n",
       "21           67032.0                  2.65           60888.0   \n",
       "22               NaN                   NaN               NaN   \n",
       "23               NaN                   NaN               NaN   \n",
       "24               NaN                   NaN               NaN   \n",
       "25               NaN                   NaN               NaN   \n",
       "26               NaN                   NaN               NaN   \n",
       "27               NaN                   NaN               NaN   \n",
       "28          460588.0                  6.12          246410.0   \n",
       "29          465440.0                  6.17          252927.0   \n",
       "30          139682.0                  1.66          320850.0   \n",
       "31          142415.0                  1.69          317875.0   \n",
       "32          289321.0                  7.43          145869.0   \n",
       "33          289537.0                  7.44          148316.0   \n",
       "34          137016.0                  4.36           77758.0   \n",
       "35          128455.0                  4.10           75281.0   \n",
       "36           63705.0                  2.63           55328.0   \n",
       "37           67383.0                  2.78           59102.0   \n",
       "38          555125.0                  6.12          350782.0   \n",
       "39          520787.0                  5.75          338391.0   \n",
       "40          153337.0                  4.12          103263.0   \n",
       "41          151993.0                  4.08          103219.0   \n",
       "42           76441.0                  3.30           52486.0   \n",
       "43           75423.0                  3.25           52191.0   \n",
       "44           60564.0                  5.54           41847.0   \n",
       "45           58345.0                  5.34           41189.0   \n",
       "46          359624.0                  5.50          230286.0   \n",
       "47          356142.0                  5.45          230772.0   \n",
       "48           32822.0                  1.77           22756.0   \n",
       "49           31568.0                  1.70           23536.0   \n",
       "50          967584.0                  1.22         2626226.0   \n",
       "51         1045870.0                  1.33         2685342.0   \n",
       "52          253410.0                  6.08          145655.0   \n",
       "53          253504.0                  6.08          148677.0   \n",
       "\n",
       "    Symbiont6 percentage  Symbiont7 counts  Symbiont7 percentage  \\\n",
       "0                   2.86          199007.0                  3.08   \n",
       "1                   2.82          195345.0                  3.03   \n",
       "2                    NaN               NaN                   NaN   \n",
       "3                    NaN               NaN                   NaN   \n",
       "4                   2.35           48012.0                  2.60   \n",
       "5                   2.37           48003.0                  2.60   \n",
       "6                   3.74          275939.0                  3.80   \n",
       "7                   3.78          278633.0                  3.84   \n",
       "8                    NaN               NaN                   NaN   \n",
       "9                    NaN               NaN                   NaN   \n",
       "10                  2.00          234412.0                  2.08   \n",
       "11                  2.06          240629.0                  2.14   \n",
       "12                  2.84           65184.0                  2.91   \n",
       "13                  2.86           65645.0                  2.92   \n",
       "14                  2.68           41372.0                  2.95   \n",
       "15                  2.77           42563.0                  3.04   \n",
       "16                  4.32          190214.0                  4.37   \n",
       "17                  4.36          191720.0                  4.39   \n",
       "18                  0.55           43496.0                  0.62   \n",
       "19                  0.60           46371.0                  0.66   \n",
       "20                  2.40           69629.0                  2.75   \n",
       "21                  2.42           69914.0                  2.76   \n",
       "22                   NaN               NaN                   NaN   \n",
       "23                   NaN               NaN                   NaN   \n",
       "24                   NaN               NaN                   NaN   \n",
       "25                   NaN               NaN                   NaN   \n",
       "26                   NaN               NaN                   NaN   \n",
       "27                   NaN               NaN                   NaN   \n",
       "28                  3.27          254458.0                  3.37   \n",
       "29                  3.36          260684.0                  3.46   \n",
       "30                  3.80          635729.0                  7.53   \n",
       "31                  3.77          629783.0                  7.47   \n",
       "32                  3.75          166964.0                  4.28   \n",
       "33                  3.81          169217.0                  4.35   \n",
       "34                  2.48           79209.0                  2.53   \n",
       "35                  2.39           75642.0                  2.41   \n",
       "36                  2.27           61103.0                  2.52   \n",
       "37                  2.43           64585.0                  2.67   \n",
       "38                  3.86          354837.0                  3.90   \n",
       "39                  3.73          340166.0                  3.75   \n",
       "40                  2.78          105682.0                  2.84   \n",
       "41                  2.78          105307.0                  2.83   \n",
       "42                  2.26           55296.0                  2.38   \n",
       "43                  2.25           54879.0                  2.36   \n",
       "44                  3.82           41642.0                  3.80   \n",
       "45                  3.77           40735.0                  3.71   \n",
       "46                  3.52          241133.0                  3.68   \n",
       "47                  3.52          241158.0                  3.68   \n",
       "48                  1.22           23200.0                  1.25   \n",
       "49                  1.26           22697.0                  1.22   \n",
       "50                  3.32         5400438.0                  6.84   \n",
       "51                  3.39         5442089.0                  6.90   \n",
       "52                  3.49          156626.0                  3.76   \n",
       "53                  3.56          159257.0                  3.83   \n",
       "\n",
       "    Symbiont8 counts  Symbiont8 percentage  No hits counts  No hits percentage  \n",
       "0          4455857.0                 69.10       1944697.0               30.16  \n",
       "1          4265102.0                 66.14       2133622.0               33.09  \n",
       "2                NaN                   NaN             NaN                 NaN  \n",
       "3                NaN                   NaN             NaN                 NaN  \n",
       "4           449731.0                 24.39       1378710.0               74.77  \n",
       "5           440527.0                 23.89       1387929.0               75.27  \n",
       "6          4233566.0                 58.29       2988012.0               41.14  \n",
       "7          4183791.0                 57.60       3035948.0               41.80  \n",
       "8                NaN                   NaN             NaN                 NaN  \n",
       "9                NaN                   NaN             NaN                 NaN  \n",
       "10         3614590.0                 32.16       7574490.0               67.38  \n",
       "11         3575910.0                 31.80       7611587.0               67.71  \n",
       "12          886635.0                 39.60       1337120.0               59.72  \n",
       "13          875471.0                 39.10       1348091.0               60.21  \n",
       "14          289974.0                 20.66       1098496.0               78.27  \n",
       "15          287009.0                 20.45       1101303.0               78.47  \n",
       "16         3810755.0                 87.41        534966.0               12.27  \n",
       "17         3730898.0                 85.58        613881.0               14.08  \n",
       "18          161318.0                  2.28       6875054.0               97.36  \n",
       "19          161074.0                  2.29       6874348.0               97.35  \n",
       "20          224309.0                  8.88       2273112.0               90.00  \n",
       "21          221405.0                  8.76       2275638.0               90.10  \n",
       "22               NaN                   NaN             NaN                 NaN  \n",
       "23               NaN                   NaN             NaN                 NaN  \n",
       "24               NaN                   NaN             NaN                 NaN  \n",
       "25               NaN                   NaN             NaN                 NaN  \n",
       "26               NaN                   NaN             NaN                 NaN  \n",
       "27               NaN                   NaN             NaN                 NaN  \n",
       "28         4986820.0                 66.19       2494485.0               33.11  \n",
       "29         4965001.0                 65.90       2515580.0               33.39  \n",
       "30          139845.0                  1.66       7740015.0               91.74  \n",
       "31          143132.0                  1.70       7744234.0               91.79  \n",
       "32         2056350.0                 52.84       1777058.0               45.66  \n",
       "33         2038522.0                 52.38       1794183.0               46.10  \n",
       "34         1807998.0                 57.55       1317372.0               41.93  \n",
       "35         1692589.0                 53.87       1432049.0               45.58  \n",
       "36          150979.0                  6.22       2251981.0               92.78  \n",
       "37          154508.0                  6.37       2248583.0               92.64  \n",
       "38         6352780.0                 70.04       2676927.0               29.52  \n",
       "39         5898062.0                 65.04       3128522.0               34.50  \n",
       "40         1656380.0                 44.50       2047282.0               55.00  \n",
       "41         1636028.0                 43.95       2066638.0               55.52  \n",
       "42          557074.0                 23.99       1748700.0               75.33  \n",
       "43          544944.0                 23.48       1760307.0               75.83  \n",
       "44          678595.0                 62.01        405227.0               37.04  \n",
       "45          656009.0                 59.96        427436.0               39.07  \n",
       "46         3790960.0                 57.90       2711587.0               41.41  \n",
       "47         3707773.0                 56.63       2792784.0               42.65  \n",
       "48          363028.0                 19.49       1485116.0               79.76  \n",
       "49          349640.0                 18.78       1498709.0               80.49  \n",
       "50         1030880.0                  1.31      73067732.0               92.61  \n",
       "51         1118202.0                  1.41      72988833.0               92.51  \n",
       "52         2875922.0                 69.02       1257901.0               30.19  \n",
       "53         2827803.0                 67.87       1304984.0               31.32  \n",
       "\n",
       "[54 rows x 30 columns]"
      ]
     },
     "execution_count": 319,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "# concat with reads df \n",
    "reads = reads.merge(\n",
    "    fastq_screen,\n",
    "    on=['Sample','Filename'],\n",
    "    how='left'\n",
    ")\n",
    "reads"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 320,
   "id": "6f2da843-47b5-476c-9295-d5ee5542fbdf",
   "metadata": {},
   "outputs": [],
   "source": [
    "# export\n",
    "reads.to_csv('/project/pi_sarah_gignouxwolfsohn_uml_edu/brooke/mcav/raw_reads.csv')"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "4f1c3f3d-4a33-4fc8-bc5d-b32b78eaf20f",
   "metadata": {},
   "outputs": [],
   "source": []
  }
 ],
 "metadata": {
  "kernelspec": {
   "display_name": "Python 3 (ipykernel)",
   "language": "python",
   "name": "python3"
  },
  "language_info": {
   "codemirror_mode": {
    "name": "ipython",
    "version": 3
   },
   "file_extension": ".py",
   "mimetype": "text/x-python",
   "name": "python",
   "nbconvert_exporter": "python",
   "pygments_lexer": "ipython3",
   "version": "3.11.0"
  }
 },
 "nbformat": 4,
 "nbformat_minor": 5
}
