{
 "cells": [
  {
   "cell_type": "markdown",
   "id": "79398e90-e3f0-4004-9436-3e8b5a76c780",
   "metadata": {},
   "source": [
    "# Assembly"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "20751f43-626d-418c-ae3b-4c188ff39ddd",
   "metadata": {},
   "source": [
    "**Megahit**\n",
    "https://www.metagenomics.wiki/tools/assembly/megahit\n",
    "- de novo assembly (w/o reference genome)\n",
    "- aligns/assembles short reads together to reconstruct one 'metagenome'\n",
    "- assembled contigs are stored in fasta file"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "f946edea-5aaf-446b-ad3e-6b9409431b1a",
   "metadata": {
    "tags": []
   },
   "source": [
    "### Installation"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "f75e5450",
   "metadata": {},
   "outputs": [],
   "source": [
    "module load miniconda/22.11.1-1"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "7a7f31b5",
   "metadata": {},
   "outputs": [],
   "source": [
    "conda create -n assembly\n",
    "#dir=/home/brooke_sienkiewicz_student_uml_edu/.conda/envs/assembly"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "54f855c4",
   "metadata": {},
   "outputs": [],
   "source": [
    "#conda info --env\n",
    "##lists all ur conda envs \n",
    "conda activate assembly"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "68e28340-4982-45d5-ac35-1ca32cdac04e",
   "metadata": {},
   "outputs": [],
   "source": [
    "#installation - just do the first time upon creating assembly env\n",
    "conda install -c bioconda megahit\n",
    "conda install -c bioconda quast python=2.7"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "986c622d-867b-4f5d-b284-2ee31306b852",
   "metadata": {
    "tags": []
   },
   "source": [
    "### MCAV"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "7e8f985a-c043-406d-b661-427dc9e7b6a9",
   "metadata": {},
   "source": [
    "#### MCAV - healthy, 2019"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "b3151628-fa03-4497-a3dc-5fafd7a7bad1",
   "metadata": {},
   "outputs": [],
   "source": [
    "# Using trimmed, qc seqs from redo_auto_detect_01312024 folder\n",
    "# 1)remove host from sample reads\n",
    "# 2)concatenate all f and r seqs into single file (1 for f, 1 for r)\n",
    "# 3)ASSEMBLE reads into contigs (contiguous sequence - joins them together based on read overlap, and ensures there are no gaps - larger portions of genomes if not all are now together in one sequence)\n",
    "# 4)remove ITS2 seqs from assembled contigs (& remove adapters) ... should try to perform on raw reads so we end up with just one final contig file \n",
    "\n",
    "# can definitely combine these into 1 or 2 batch scripts"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "7d360ec7-096a-4884-9753-38028d7a0b4d",
   "metadata": {},
   "outputs": [],
   "source": [
    "#!/bin/bash\n",
    "#SBATCH -c 24  # Number of Cores per Task\n",
    "#SBATCH --mem=180G  # Requested Memory\n",
    "#SBATCH -p cpu-long  # Partition\n",
    "#SBATCH -t 24:00:00  # Job time limit\n",
    "#SBATCH --mail-type=ALL\n",
    "#SBATCH -o /project/pi_sarah_gignouxwolfsohn_uml_edu/brooke/mcav/assembly/healthy_2019_mcav/slurm-removal-%j.out  # %j = job ID\n",
    "\n",
    "module load miniconda/22.11.1-1\n",
    "conda activate anvio-8\n",
    "\n",
    "# 1)remove host from sample reads\n",
    "# Host seq removal - Thij's script https://github.com/ThijsSt/SCTLD-metagenomes/blob/main/Quality_control_metagenomes.ipynb\n",
    "\n",
    "#set parameters:\n",
    "SAMPLENAME=\"healthy_2019_mcav\"\n",
    "GENOME=\"mcav\"\n",
    "READSPATH=/project/pi_sarah_gignouxwolfsohn_uml_edu/brooke/mcav/trimmed/redo_auto_detect_01312024\n",
    "INDEX=\"$GENOME\"_DB\n",
    "INPUTPATH=\"/project/pi_sarah_gignouxwolfsohn_uml_edu/Reference_genomes/Mcav_genome\"\n",
    "WORKINGPATH='/project/pi_sarah_gignouxwolfsohn_uml_edu/brooke/mcav/assembly/healthy_2019_mcav/host_removed'\n",
    "\n",
    "#build a bowtie2 index from a known genome\n",
    "bowtie2-build $INPUTPATH/Mcavernosa_July2018.fasta $INPUTPATH/\"$INDEX\"\n",
    "\n",
    "#loop through samples\n",
    "while IFS= read -r SAMPLEID; do\n",
    "\n",
    "#re-align reads back to the index\n",
    "bowtie2 -p 8 -x $INPUTPATH/$INDEX -1 \"$READSPATH\"/\"${SAMPLEID}_R1_001_val_1.fq\" -2 \"$READSPATH\"/\"${SAMPLEID}_R2_001_val_2.fq\" -S $WORKINGPATH/\"${SAMPLEID}\"_mapped_and_unmapped.sam\n",
    "\n",
    "#convert sam file from bowtie to a bam file for processing\n",
    "samtools view -bS $WORKINGPATH/\"${SAMPLEID}\"_mapped_and_unmapped.sam > $WORKINGPATH/\"${SAMPLEID}\"_mapped_and_unmapped.bam\n",
    "\n",
    "#extract only the reads of which both do not match against the host genome\n",
    "samtools view -b -f 12 -F 256 $WORKINGPATH/\"${SAMPLEID}\"_mapped_and_unmapped.bam > $WORKINGPATH/\"${SAMPLEID}\"_bothReadsUnmapped.bam\n",
    "#ask thijs what flags mean \n",
    "\n",
    "# sorts the file so both mates are together and then extracts them back as .fastq files\n",
    "samtools sort -n -m 5G -@ 2 $WORKINGPATH/\"${SAMPLEID}\"_bothReadsUnmapped.bam -o $WORKINGPATH/\"${SAMPLEID}\"_bothReadsUnmapped_sorted.bam\n",
    "samtools fastq -@ 8 $WORKINGPATH/\"${SAMPLEID}\"_bothReadsUnmapped_sorted.bam \\\n",
    "    -1 \"${SAMPLEID}\"_host_removed_R1.fastq \\\n",
    "    -2 \"${SAMPLEID}\"_host_removed_R2.fastq \\\n",
    "    -0 /dev/null -s /dev/null -n\n",
    "#can i direct these to a diff folder?\n",
    "\n",
    "done < \"healthy_2019_MCAV_sampleids\"\n",
    "#run in dir with sampleids txt file (~/mcav/assembly/healthy_2019_mcav)\n",
    "\n",
    "# JOB-ID: 19358781\n",
    "# bash script file name: brooke/mcav/assembly/healthy_2019_mcav/removal"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "169dddb5-8b1f-4f42-bf94-ffa5d19a8197",
   "metadata": {},
   "outputs": [],
   "source": [
    "#!/bin/bash\n",
    "#SBATCH -c 24  # Number of Cores per Task\n",
    "#SBATCH --mem=180G  # Requested Memory\n",
    "#SBATCH -p cpu-long  # Partition\n",
    "#SBATCH -t 56:00:00  # Job time limit\n",
    "#SBATCH --mail-type=ALL\n",
    "#SBATCH -o /project/pi_sarah_gignouxwolfsohn_uml_edu/brooke/mcav/assembly/healthy_2019_mcav/slurm-assembly-%j.out  # %j = job ID\n",
    "\n",
    "module load miniconda/22.11.1-1\n",
    "conda activate assembly\n",
    "# 2)concatenate all f and r seqs into single file (1 for f, 1 for r)\n",
    "\n",
    "READSPATH='/project/pi_sarah_gignouxwolfsohn_uml_edu/brooke/mcav/assembly/healthy_2019_mcav/host_removed'\n",
    "SAMPLENAME=\"healthy_2019_mcav\"\n",
    "OUTDIR=/project/pi_sarah_gignouxwolfsohn_uml_edu/brooke/mcav/assembly/healthy_2019_mcav\n",
    "\n",
    "# Read the sample IDs from the file\n",
    "while IFS= read -r SAMPLEID; do\n",
    "    # Construct the file paths for forward and reverse reads\n",
    "    FORWARD_READ=\"$READSPATH/${SAMPLEID}_host_removed_R1.fastq\"\n",
    "    REVERSE_READ=\"$READSPATH/${SAMPLEID}_host_removed_R2.fastq\"\n",
    "\n",
    "    # Check if the files exist before concatenating\n",
    "    if [ -e \"$FORWARD_READ\" ]; then\n",
    "        cat \"$FORWARD_READ\" >> \"$OUTDIR/${SAMPLENAME}_reads_R1_ALL.fastq\"\n",
    "    else\n",
    "        echo \"Forward read file not found for sample $SAMPLEID\"\n",
    "    fi\n",
    "\n",
    "    if [ -e \"$REVERSE_READ\" ]; then\n",
    "        cat \"$REVERSE_READ\" >> \"$OUTDIR/${SAMPLENAME}_reads_R2_ALL.fastq\"\n",
    "    else\n",
    "        echo \"Reverse read file not found for sample $SAMPLEID\"\n",
    "    fi\n",
    "done < \"$OUTDIR/healthy_2019_MCAV_sampleids\"\n",
    "\n",
    "# 3)ASSEMBLE reads into contigs (contiguous sequence - joins them together based on read overlap, and ensures there are no gaps\n",
    "megahit --presets meta-large \\\n",
    "-1 \"$OUTDIR\"/\"$SAMPLENAME\"_reads_R1_ALL.fastq \\\n",
    "-2 \"$OUTDIR\"/\"$SAMPLENAME\"_reads_R2_ALL.fastq \\\n",
    "--keep-tmp-files \\\n",
    "-o megahit_host_removed --out-prefix $SAMPLENAME \\\n",
    "#--continue\n",
    "#this one has to make the directory; will fail if it already exists\n",
    "\n",
    "# try metavelvet next? \n",
    "\n",
    "# JOB-ID: 19388626\n",
    "# bash script file name: $OUTDIR/assembly"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "949f9385",
   "metadata": {},
   "source": [
    "Total time elapsed: 46 hrs \n",
    "looks like 190 GB is enough... and 24 CPU - 90% efficiency on last part of run "
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "361054d6-4259-41f5-9087-1f86b96f4729",
   "metadata": {},
   "outputs": [],
   "source": [
    "# NEED TO: Rename concatenated seqs so it can just be done in 1 step instead of renaming all indiividual files \n",
    "# (qc step it adds \"val_1/2\" to each seq file)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "695644db-c7c0-4f10-9571-aac1199a1c4a",
   "metadata": {},
   "outputs": [],
   "source": [
    "#!/bin/bash\n",
    "#SBATCH -c 24  # Number of Cores per Task\n",
    "#SBATCH --mem=180G  # Requested Memory\n",
    "#SBATCH -p cpu-long  # Partition\n",
    "#SBATCH -t 24:00:00  # Job time limit\n",
    "#SBATCH --mail-type=ALL\n",
    "#SBATCH -o /project/pi_sarah_gignouxwolfsohn_uml_edu/brooke/mcav/assembly/healthy_2019_mcav/slurm-%j.out  # %j = job ID\n",
    "\n",
    "\n",
    "#load modules\n",
    "module load miniconda/22.11.1-1\n",
    "conda activate cutadaptenv\n",
    "\n",
    "# 4)remove ITS2 seqs from assembled contigs (& remove adapters)\n",
    "\n",
    "# Set your input and output files\n",
    "SAMPLENAME=\"healthy_2019_mcav\"\n",
    "INPUTDIR=\"/project/pi_sarah_gignouxwolfsohn_uml_edu/brooke/mcav/assembly/healthy_2019_mcav/megahit_host_removed\"\n",
    "OUTPUTDIR=\"/project/pi_sarah_gignouxwolfsohn_uml_edu/brooke/mcav/assembly/healthy_2019_mcav\"\n",
    "\n",
    "\n",
    "#READSPATH='/project/pi_sarah_gignouxwolfsohn_uml_edu/brooke/mcav/trimmed/redo_auto_detect_01312024'\n",
    "\n",
    "input_fasta=\"$SAMPLENAME.contigs.fa\"\n",
    "#output_fasta=\"${SAMPLENAME}_filtered.contigs.fasta\"\n",
    "#couldn't get this to work?? just typed it out for the -o parameter in cutadapt script\n",
    "\n",
    "# Set your primer sequences\n",
    "forward_primer=\"GAATTGCAGAACTCCGTGAACC\"\n",
    "reverse_primer=\"CGGGTTCWCTTGTYTGACTTCATGC\"\n",
    "\n",
    "# Verify path and input dir - only need for troubleshooting\n",
    "echo \"Working Directory: $(pwd)\"\n",
    "#ls -l $INPUTDIR\n",
    "\n",
    "# Run cutadapt\n",
    "cutadapt \\\n",
    "  -g \"$forward_primer\" \\\n",
    "  -a \"$reverse_primer\" \\\n",
    "  --discard-trimmed \\\n",
    "  -o $OUTPUTDIR/\"{$SAMPLENAME}_filtered\" \\\n",
    "  $INPUTDIR/\"$input_fasta\"\n",
    "  \n",
    "ls -l $OUTPUTDIR\n",
    "#check results dir to see if it was successful in creating output file \n",
    "\n",
    "# JOB-ID: 19420673\n",
    "# bash script file name: brooke/mcav/assembly/healthy_2019_mcav/ITS2_trim\n",
    "\n",
    "\n",
    "#final assembled contigs: healthy_2019_mcav_filtered.contigs.fasta in ~/brooke/mcav/assembly/healthy_2019_mcav"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "8bf878e7-66d2-46c3-8c8f-91cbf394c1e0",
   "metadata": {},
   "source": [
    "#### MCAV - diseased"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "4c5ee932-c2c7-4a2a-8b11-96451bddaa52",
   "metadata": {},
   "outputs": [],
   "source": [
    "#!/bin/bash\n",
    "#SBATCH -c 24  # Number of Cores per Task\n",
    "#SBATCH --mem=180G  # Requested Memory\n",
    "#SBATCH -p cpu-long  # Partition\n",
    "#SBATCH -t 24:00:00  # Job time limit\n",
    "#SBATCH --mail-type=ALL\n",
    "#SBATCH -o /project/pi_sarah_gignouxwolfsohn_uml_edu/brooke/mcav/diseased_mcav/assembly/slurm-removal-%j.out  # %j = job ID\n",
    "\n",
    "module load miniconda/22.11.1-1\n",
    "conda activate anvio-8\n",
    "\n",
    "# 1)remove host from sample reads\n",
    "# Host seq removal - Thij's script https://github.com/ThijsSt/SCTLD-metagenomes/blob/main/Quality_control_metagenomes.ipynb\n",
    "\n",
    "#set parameters:\n",
    "SAMPLENAME=\"diseased_mcav\"\n",
    "GENOME=\"mcav\"\n",
    "READSPATH=/project/pi_sarah_gignouxwolfsohn_uml_edu/brooke/mcav/trimmed/redo_auto_detect_01312024\n",
    "INDEX=\"$GENOME\"_DB\n",
    "INPUTPATH=\"/project/pi_sarah_gignouxwolfsohn_uml_edu/Reference_genomes/Mcav_genome\"\n",
    "WORKINGPATH='/project/pi_sarah_gignouxwolfsohn_uml_edu/brooke/mcav/diseased_mcav/assembly/host_removed'\n",
    "\n",
    "mkdir $WORKINGPATH\n",
    "#build a bowtie2 index from a known genome\n",
    "bowtie2-build $INPUTPATH/Mcavernosa_July2018.fasta $INPUTPATH/\"$INDEX\"\n",
    "\n",
    "#loop through samples\n",
    "while IFS= read -r SAMPLEID; do\n",
    "\n",
    "#re-align reads back to the index\n",
    "bowtie2 -p 8 -x $INPUTPATH/$INDEX -1 \"$READSPATH\"/\"${SAMPLEID}_R1_001_val_1.fq\" -2 \"$READSPATH\"/\"${SAMPLEID}_R2_001_val_2.fq\" -S $WORKINGPATH/\"${SAMPLEID}\"_mapped_and_unmapped.sam\n",
    "\n",
    "#convert sam file from bowtie to a bam file for processing\n",
    "samtools view -bS $WORKINGPATH/\"${SAMPLEID}\"_mapped_and_unmapped.sam > $WORKINGPATH/\"${SAMPLEID}\"_mapped_and_unmapped.bam\n",
    "\n",
    "#extract only the reads of which both do not match against the host genome\n",
    "samtools view -b -f 12 -F 256 $WORKINGPATH/\"${SAMPLEID}\"_mapped_and_unmapped.bam > $WORKINGPATH/\"${SAMPLEID}\"_bothReadsUnmapped.bam\n",
    "#ask thijs what flags mean \n",
    "\n",
    "# sorts the file so both mates are together and then extracts them back as .fastq files\n",
    "samtools sort -n -m 5G -@ 2 $WORKINGPATH/\"${SAMPLEID}\"_bothReadsUnmapped.bam -o $WORKINGPATH/\"${SAMPLEID}\"_bothReadsUnmapped_sorted.bam\n",
    "samtools fastq -@ 8 $WORKINGPATH/\"${SAMPLEID}\"_bothReadsUnmapped_sorted.bam \\\n",
    "    -1 \"${SAMPLEID}\"_host_removed_R1.fastq \\\n",
    "    -2 \"${SAMPLEID}\"_host_removed_R2.fastq \\\n",
    "    -0 /dev/null -s /dev/null -n\n",
    "#can i direct these to a diff folder?\n",
    "\n",
    "done < \"diseased_MCAV\"\n",
    "#run in dir with sampleids txt file (~/mcav/diseased_mcav)\n",
    "\n",
    "# JOB-ID: 21639875\n",
    "# bash script file name: brooke/mcav/diseased_mcav/assembly/host_removal"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "af314e7d-8880-49be-8e94-9b8a6c0ea569",
   "metadata": {},
   "outputs": [],
   "source": [
    "#!/bin/bash\n",
    "#SBATCH -c 24  # Number of Cores per Task\n",
    "#SBATCH --mem=180G  # Requested Memory\n",
    "#SBATCH -p cpu-long  # Partition\n",
    "#SBATCH -t 56:00:00  # Job time limit\n",
    "#SBATCH --mail-type=ALL\n",
    "#SBATCH -o /project/pi_sarah_gignouxwolfsohn_uml_edu/brooke/mcav/diseased_mcav/assembly/slurm-assembly-%j.out  # %j = job ID\n",
    "\n",
    "module load miniconda/22.11.1-1\n",
    "conda activate assembly\n",
    "# 2)concatenate all f and r seqs into single file (1 for f, 1 for r)\n",
    "\n",
    "READSPATH='/project/pi_sarah_gignouxwolfsohn_uml_edu/brooke/mcav/diseased_mcav/assembly/host_removed'\n",
    "SAMPLENAME=\"diseased_mcav\"\n",
    "OUTDIR='/project/pi_sarah_gignouxwolfsohn_uml_edu/brooke/mcav/diseased_mcav/assembly'\n",
    "\n",
    "# Read the sample IDs from the file\n",
    "while IFS= read -r SAMPLEID; do\n",
    "    # Construct the file paths for forward and reverse reads\n",
    "    FORWARD_READ=\"$READSPATH/${SAMPLEID}_host_removed_R1.fastq\"\n",
    "    REVERSE_READ=\"$READSPATH/${SAMPLEID}_host_removed_R2.fastq\"\n",
    "\n",
    "    # Check if the files exist before concatenating\n",
    "    if [ -e \"$FORWARD_READ\" ]; then\n",
    "        cat \"$FORWARD_READ\" >> \"$OUTDIR/${SAMPLENAME}_reads_R1_ALL.fastq\"\n",
    "    else\n",
    "        echo \"Forward read file not found for sample $SAMPLEID\"\n",
    "    fi\n",
    "\n",
    "    if [ -e \"$REVERSE_READ\" ]; then\n",
    "        cat \"$REVERSE_READ\" >> \"$OUTDIR/${SAMPLENAME}_reads_R2_ALL.fastq\"\n",
    "    else\n",
    "        echo \"Reverse read file not found for sample $SAMPLEID\"\n",
    "    fi\n",
    "done < \"diseased_MCAV\"\n",
    "\n",
    "# 3)ASSEMBLE reads into contigs (contiguous sequence - joins them together based on read overlap, and ensures there are no gaps\n",
    "megahit --presets meta-large \\\n",
    "-1 \"$OUTDIR\"/\"$SAMPLENAME\"_reads_R1_ALL.fastq \\\n",
    "-2 \"$OUTDIR\"/\"$SAMPLENAME\"_reads_R2_ALL.fastq \\\n",
    "--keep-tmp-files \\\n",
    "-o $OUTDIR/contigs --out-prefix $SAMPLENAME \\\n",
    "#--continue\n",
    "#this one has to make the directory; will fail if it already exists\n",
    "\n",
    "\n",
    "# 4)remove ITS2 seqs from assembled contigs (& remove adapters)\n",
    "conda deactivate assembly \n",
    "conda activate cutadaptenv\n",
    "\n",
    "# Set your input and output files\n",
    "INPUTDIR=\"/project/pi_sarah_gignouxwolfsohn_uml_edu/brooke/mcav/diseased_mcav/assembly/contigs\"\n",
    "OUTPUTDIR=\"/project/pi_sarah_gignouxwolfsohn_uml_edu/brooke/mcav/diseased_mcav/assembly\"\n",
    "\n",
    "input_fasta=\"$SAMPLENAME.contigs.fa\"\n",
    "#output_fasta=\"${SAMPLENAME}_filtered.contigs.fasta\"\n",
    "#couldn't get this to work?? just typed it out for the -o parameter in cutadapt script\n",
    "\n",
    "# Set your primer sequences\n",
    "forward_primer=\"GAATTGCAGAACTCCGTGAACC\"\n",
    "reverse_primer=\"CGGGTTCWCTTGTYTGACTTCATGC\"\n",
    "\n",
    "# Verify path and input dir - only need for troubleshooting\n",
    "echo \"Working Directory: $(pwd)\"\n",
    "#ls -l $INPUTDIR\n",
    "\n",
    "# Run cutadapt\n",
    "cutadapt \\\n",
    "  -g \"$forward_primer\" \\\n",
    "  -a \"$reverse_primer\" \\\n",
    "  --discard-trimmed \\\n",
    "  -o $OUTPUTDIR/\"{$SAMPLENAME}_filtered\" \\\n",
    "  $INPUTDIR/\"$input_fasta\"\n",
    "  \n",
    "ls -l $OUTPUTDIR\n",
    "#check results dir to see if it was successful in creating output file \n",
    "\n",
    "\n",
    "# JOB-ID: 21644151\n",
    "# bash script file name: brooke/mcav/diseased_mcav/assembly/assembly\n",
    "\n",
    "\n",
    "#cutadapt script failed so redoing in own bash script:\n",
    "# JOB-ID: 21680100\n",
    "# bash script file name: brooke/mcav/diseased_mcav/assembly/ITS2_filtering\n",
    "\n",
    "#final assembled contigs: diseased_mcav_filtered.contigs.fasta in ~/brooke/mcav/diseased_mcav/assembly/diseased_mcav_filtered.contigs.fasta\n"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "e74d081e-43a5-4c60-831c-35ec91e1728e",
   "metadata": {},
   "source": [
    "=== Summary ===\n",
    "\n",
    "Total reads processed:               2,122,290\n",
    "Reads with adapters:                    63,973 (3.0%)\n",
    "\n",
    "== Read fate breakdown ==\n",
    "Reads discarded as trimmed:             63,973 (3.0%)\n",
    "Reads written (passing filters):     2,058,317 (97.0%)\n",
    "\n",
    "Total basepairs processed: 1,821,385,837 bp\n",
    "Total written (filtered):  1,769,144,061 bp (97.1%)\n",
    "\n",
    "=== Adapter 1 ===\n",
    "\n",
    "Sequence: GAATTGCAGAACTCCGTGAACC; Type: regular 5'; Length: 22; Trimmed: 39801 times\n",
    "\n",
    "Minimum overlap: 3\n",
    "No. of allowed errors:\n",
    "1-9 bp: 0; 10-19 bp: 1; 20-22 bp: 2\n",
    "\n",
    "Overview of removed sequences\n",
    "length\tcount\texpect\tmax.err\terror counts\n",
    "3\t27460\t33160.8\t0\t27460\n",
    "4\t9697\t8290.2\t0\t9697\n",
    "5\t1947\t2072.5\t0\t1947\n",
    "6\t469\t518.1\t0\t469\n",
    "7\t106\t129.5\t0\t106\n",
    "8\t23\t32.4\t0\t23\n",
    "9\t30\t8.1\t0\t3 27\n",
    "10\t43\t2.0\t1\t0 43\n",
    "11\t12\t0.5\t1\t0 12\n",
    "12\t4\t0.1\t1\t0 4\n",
    "13\t2\t0.0\t1\t0 2\n",
    "14\t1\t0.0\t1\t0 1\n",
    "40\t2\t0.0\t2\t2\n",
    "316\t1\t0.0\t2\t0 1\n",
    "443\t1\t0.0\t2\t0 0 1\n",
    "548\t1\t0.0\t2\t0 1\n",
    "580\t1\t0.0\t2\t1\n",
    "727\t1\t0.0\t2\t1\n",
    "\n",
    "\n",
    "=== Adapter 2 ===\n",
    "\n",
    "Sequence: CGGGTTCWCTTGTYTGACTTCATGC; Type: regular 3'; Length: 25; Trimmed: 24172 times\n",
    "\n",
    "Minimum overlap: 3\n",
    "No. of allowed errors:\n",
    "1-9 bp: 0; 10-19 bp: 1; 20-25 bp: 2\n",
    "\n",
    "Bases preceding removed adapters:\n",
    "  A: 22.4%\n",
    "  C: 23.5%\n",
    "  G: 24.4%\n",
    "  T: 29.7%\n",
    "  none/other: 0.0%\n",
    "\n",
    "Overview of removed sequences\n",
    "length\tcount\texpect\tmax.err\terror counts\n",
    "3\t18071\t33160.8\t0\t18071\n",
    "4\t4756\t8290.2\t0\t4756\n",
    "5\t799\t2072.5\t0\t799\n",
    "6\t296\t518.1\t0\t296\n",
    "7\t64\t129.5\t0\t64\n",
    "8\t30\t32.4\t0\t30\n",
    "9\t34\t8.1\t0\t12 22\n",
    "10\t73\t2.0\t1\t2 71\n",
    "11\t25\t0.5\t1\t0 25\n",
    "12\t10\t0.1\t1\t0 10\n",
    "13\t2\t0.0\t1\t0 2\n",
    "14\t2\t0.0\t1\t0 2\n",
    "15\t1\t0.0\t1\t0 1\n",
    "19\t2\t0.0\t1\t0 0 2\n",
    "74\t1\t0.0\t2\t0 1\n",
    "76\t1\t0.0\t2\t1\n",
    "163\t1\t0.0\t2\t1\n",
    "204\t1\t0.0\t2\t1\n",
    "311\t1\t0.0\t2\t0 0 1\n",
    "337\t1\t0.0\t2\t0 0 1\n",
    "1314\t1\t0.0\t2\t0 1"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "5b10f894-3196-4e33-a63a-9388abebdb58",
   "metadata": {
    "tags": []
   },
   "source": [
    "# Quality Check\n",
    "**Metaquast**\n",
    "- quality assessment of metagenomic reads, no reference genome included here"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "42c2d37f-0089-47c5-9d73-1fc630fc7cfa",
   "metadata": {},
   "source": [
    "https://quast.sourceforge.net/docs/manual.html#sec1\n",
    "https://www.ncbi.nlm.nih.gov/pmc/articles/PMC3624806/\n",
    "..how to interpret quality results?\n",
    "- check how many large contigs you have (>1000 bp)\n",
    "- did not map to reference genome.\n",
    "- right now just helpful to see length and quality of contigs, maybe can reassess after mapping back to metagenome?\n",
    "Cite metaquast: https://quast.sourceforge.net/publications.html"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "d7763870-3ec9-488c-90b5-1c02cc9cc3d7",
   "metadata": {},
   "outputs": [],
   "source": [
    "module load miniconda/22.11.1-1\n",
    "conda activate assembly"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "c0e4621d-194a-4c0c-8f7c-9a35fc55a80e",
   "metadata": {},
   "outputs": [],
   "source": [
    "metaquast healthy_2019_mcav_filtered.contigs.fasta -o quast_output\n",
    "# find way to streamline deleting excess files because we only want to look at contig stats unaligned without reference genomes"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "f3187a15-e518-42d6-aee7-59d1ff898d2e",
   "metadata": {},
   "outputs": [],
   "source": [
    "# mcav, healthy, 2019 sample assembly (host and ITS2 removal)\n",
    "# /project/pi_sarah_gignouxwolfsohn_uml_edu/brooke/mcav/healthy_2019_mcav/quast_output/not_aligned/report.txt\n",
    "\n",
    "All statistics are based on contigs of size >= 500 bp, unless otherwise noted (e.g., \"# contigs (>= 0 bp)\" and \"Total length (>= 0 bp)\" include all contigs).\n",
    "\n",
    "Assembly                    healthy_2019_mcav_filtered.contigs\n",
    "# contigs (>= 1000 bp)      447572                            \n",
    "# contigs (>= 5000 bp)      42741                             \n",
    "# contigs (>= 10000 bp)     9795                              \n",
    "# contigs (>= 25000 bp)     681                               \n",
    "# contigs (>= 50000 bp)     46                                \n",
    "Total length (>= 1000 bp)   1185480790                        \n",
    "Total length (>= 5000 bp)   373241850                         \n",
    "Total length (>= 10000 bp)  153130609                         \n",
    "Total length (>= 25000 bp)  24831349                          \n",
    "Total length (>= 50000 bp)  5071479                           \n",
    "# contigs                   1039025                           \n",
    "Largest contig              437957                            \n",
    "Total length                1589447609                        \n",
    "GC (%)                      42.72                             \n",
    "N50                         2169                              \n",
    "N75                         987                               \n",
    "L50                         176118                            \n",
    "L75                         454223                            \n",
    "# N's per 100 kbp           0.00      \n",
    " "
   ]
  },
  {
   "cell_type": "raw",
   "id": "eacde40f-be94-496d-9ec4-947715029118",
   "metadata": {
    "tags": []
   },
   "source": [
    "# MCAV assembly before seperating by sample type\n",
    "\n",
    "\n",
    "All statistics are based on contigs of size >= 500 bp, unless otherwise noted (e.g., \"# contigs (>= 0 bp)\" and \"Total length (>= 0 bp)\" include all contigs).\n",
    "\n",
    "### Assembly                    mcav.contigs\n",
    "#### contigs (>= 0 bp)         839850      \n",
    "#### contigs (>= 1000 bp)      39414       \n",
    "#### contigs (>= 5000 bp)      370         \n",
    "#### contigs (>= 10000 bp)     35          \n",
    "#### contigs (>= 25000 bp)     0           \n",
    "#### contigs (>= 50000 bp)     0           \n",
    "Total length (>= 0 bp)      435829081   \n",
    "Total length (>= 1000 bp)   57097694    \n",
    "Total length (>= 5000 bp)   2623425     \n",
    "Total length (>= 10000 bp)  443897      \n",
    "Total length (>= 25000 bp)  0           \n",
    "Total length (>= 50000 bp)  0           \n",
    "#### contigs                   312757      \n",
    "Largest contig              24857       \n",
    "Total length                237368824   \n",
    "GC (%)                      42.80       \n",
    "N50                         729         \n",
    "N75                         595         \n",
    "L50                         113047      \n",
    "L75                         203599      \n",
    "N's per 100 kbp           0.00  "
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "4141405b-cec4-481e-aac9-1dbf033bfcd7",
   "metadata": {},
   "outputs": [],
   "source": [
    "#!/bin/bash\n",
    "#SBATCH -c 24  # Number of Cores per Task\n",
    "#SBATCH --mem=180G  # Requested Memory\n",
    "#SBATCH -p cpu-long  # Partition\n",
    "#SBATCH -t 56:00:00  # Job time limit\n",
    "#SBATCH --mail-type=ALL\n",
    "#SBATCH -o /project/pi_sarah_gignouxwolfsohn_uml_edu/brooke/mcav/diseased_mcav/assembly/slurm-metaquast-%j.out  # %j = job ID\n",
    "\n",
    "#### MCAV - diseased\n",
    "module load miniconda/22.11.1-1\n",
    "conda activate assembly\n",
    "metaquast diseased_mcav_filtered.contigs.fasta -t 12 -o quast_output\n",
    "\n",
    "#JOB ID: 21682141"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "b2a5935e-cf5c-4adb-b1da-d4f75f159f01",
   "metadata": {},
   "outputs": [],
   "source": [
    "cd //project/pi_sarah_gignouxwolfsohn_uml_edu/brooke/mcav/diseased_mcav/assembly/quast_output\n",
    "rm -r krona_charts quast_downloaded_references runs_per_reference icarus_viewers summary/TEX summary/TSV combined_reference/icarus_viewers combined_reference/aligned_stats\n",
    "rm combined_reference/*.tex combined_reference/*.tsv\n",
    "rm not_aligned/*.tex not_aligned/*.tsv"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "2d8848bb-6176-4c04-b595-abc21e8a2dcb",
   "metadata": {},
   "outputs": [],
   "source": [
    "# mcav diseased sample assembly (host and ITS2 removal)\n",
    "# /project/pi_sarah_gignouxwolfsohn_uml_edu/brooke/mcav/diseased_mcav/assembly/quast_output/not_aligned/report.txt\n",
    "\n",
    "All statistics are based on contigs of size >= 500 bp, unless otherwise noted (e.g., \"# contigs (>= 0 bp)\" and \"Total length (>= 0 bp)\" include all contigs).\n",
    "\n",
    "Assembly                    diseased_mcav_filtered.contigs\n",
    "# contigs (>= 1000 bp)      386921                        \n",
    "# contigs (>= 5000 bp)      43604                         \n",
    "# contigs (>= 10000 bp)     9279                          \n",
    "# contigs (>= 25000 bp)     557                           \n",
    "# contigs (>= 50000 bp)     10                            \n",
    "Total length (>= 1000 bp)   1089301896                    \n",
    "Total length (>= 5000 bp)   369050266                     \n",
    "Total length (>= 10000 bp)  139814218                     \n",
    "Total length (>= 25000 bp)  17768633                      \n",
    "Total length (>= 50000 bp)  597200                        \n",
    "# contigs                   787579                        \n",
    "Largest contig              76539                         \n",
    "Total length                1363359090                    \n",
    "GC (%)                      42.38                         \n",
    "N50                         2620                          \n",
    "N75                         1197                          \n",
    "L50                         132007                        \n",
    "L75                         325806                        \n",
    "# N's per 100 kbp           0.00         \n",
    "\n",
    "#smaller sample size - looks comparable to healthy mcav "
   ]
  }
 ],
 "metadata": {
  "kernelspec": {
   "display_name": "Python 3 (ipykernel)",
   "language": "python",
   "name": "python3"
  },
  "language_info": {
   "codemirror_mode": {
    "name": "ipython",
    "version": 3
   },
   "file_extension": ".py",
   "mimetype": "text/x-python",
   "name": "python",
   "nbconvert_exporter": "python",
   "pygments_lexer": "ipython3",
   "version": "3.11.0"
  }
 },
 "nbformat": 4,
 "nbformat_minor": 5
}
