{
 "cells": [
  {
   "cell_type": "markdown",
   "id": "00899c6b-9865-4ce8-810a-3d1db2c3b5b2",
   "metadata": {},
   "source": [
    "# Mapping \n",
    "- maps the reads onto to the MAG (metagenome-assembled genome) (contigs.fa)\n",
    "- allows you to quantify genes/taxa in each sample by matching the sequences to the MAG\n",
    "\n",
    "## Anvio\n",
    "\n",
    "https://anvio.org/\n",
    "- used for further analysis \n",
    "- Here, we use it for mapping, ..."
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "ba679cd6-4a9c-42f8-ab23-c836bbeb6e62",
   "metadata": {},
   "outputs": [],
   "source": [
    "#create anvio environment\n",
    "conda create -n anvio-8\n",
    "#dir=/home/brooke_sienkiewicz_student_uml_edu/.conda/envs/anvio-7.1"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "a89b912f-1e44-4f59-9689-5d0c66213938",
   "metadata": {},
   "source": [
    "Anvio\n",
    "https://merenlab.org/tutorials/assembly-based-metagenomics/\n",
    "- reformats fasta file, filters contigs >1000bp\n",
    "- creates index of contigs ('reference genome') in order to map reads against it "
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "612967b6-6520-472e-bfc3-5a4a61ca15fc",
   "metadata": {},
   "outputs": [],
   "source": [
    "#!/bin/bash\n",
    "#SBATCH -c 24  # Number of Cores per Task\n",
    "#SBATCH --mem=180G  # Requested Memory\n",
    "#SBATCH -p cpu-preempt  # Partition\n",
    "#SBATCH -t 20:00:00  # Job time limit\n",
    "#SBATCH --mail-type=ALL\n",
    "#SBATCH -o /project/pi_sarah_gignouxwolfsohn_uml_edu/brooke/mcav/diseased_mcav/mapping/slurm-mapping-%j.out  # %j = job ID\n",
    "\n",
    "module load miniconda/22.11.1-1\n",
    "conda activate anvio-8\n",
    "#filters contigs and reformats so naming is cleaner ('fixing deflines')\n",
    "#aligns each sample seq to the contigs ... need output bam for downstream analysis to determine how many of each taxa are in the samples \n",
    "\n",
    "\n",
    "SAMPLENAME=\"diseased_mcav\"\n",
    "READSPATH='/project/pi_sarah_gignouxwolfsohn_uml_edu/brooke/mcav/diseased_mcav/assembly/host_removed'\n",
    "CONTIGPATH='/project/pi_sarah_gignouxwolfsohn_uml_edu/brooke/mcav/diseased_mcav/assembly'\n",
    "CONTIGFILE=\"$SAMPLENAME\"_filtered.contigs.fasta\n",
    "newCONTIGPATH='/project/pi_sarah_gignouxwolfsohn_uml_edu/brooke/mcav/diseased_mcav/mapping'\n",
    "FILEPATH='/project/pi_sarah_gignouxwolfsohn_uml_edu/brooke/mcav/diseased_mcav'\n",
    "\n",
    "#fixes deflines for later and filters on size \n",
    "anvi-script-reformat-fasta $CONTIGPATH/$CONTIGFILE -o $newCONTIGPATH/\"${SAMPLENAME}_filtered.contigs-fixed.fsa\" -l 1000 --simplify-names --report-file contig-rename-report-txt\n",
    "# filtering seq length 1000bp...need to play around with filtering based on bp length\n",
    "#deflines = sequence definition line. comes directly before its associated sequence in a fasta file\n",
    "\n",
    "\n",
    "FIXEDCON=\"${SAMPLENAME}_filtered.contigs-fixed.fsa\"\n",
    "\n",
    "cd $newCONTIGPATH\n",
    "#this builds an index of your contigs, which only needs to happen once\n",
    "bowtie2-build $FIXEDCON \"$SAMPLENAME\"_contigs\n",
    "# will not accept path before contigs file - must be in the correct dir \n",
    "# or error is just that there's abug so it won't accept .fa extension - renamed to .fsa\n",
    "ls #debugging\n",
    "\n",
    "while IFS= read -r SAMPLEID; do\n",
    "    #align reads to your contigs and collects that in a .sam file\n",
    "    bowtie2 --threads 11 -x \"$SAMPLENAME\"_contigs -1 $READSPATH/\"${SAMPLEID}\"_host_removed_R1.fastq -2 $READSPATH/\"${SAMPLEID}\"_host_removed_R2.fastq -S index/\"${SAMPLEID}\".sam\n",
    "    #note that these reads have host removed but NOT symbionts (which were removed from contigs, so there may be a bit of mismatch there (ie: low alignment rate))\n",
    "    #make sure to point it to the index not the FIXEDCON file (-x parameter)\n",
    "    \n",
    "    #converts your sam file to a bam file, but its neither sorted nor indexed, so we use an Anvi'O script to do so:\n",
    "    samtools view -F 4 -b -S index/\"${SAMPLEID}\".sam -o index/\"${SAMPLEID}\"-RAW.bam\n",
    "   \n",
    "    #index and sort your bam file\n",
    "    anvi-init-bam index/\"${SAMPLEID}\"-RAW.bam -o index/\"${SAMPLEID}\".bam\n",
    "    \n",
    "    rm index/\"${SAMPLEID}\"-RAW.bam\n",
    "done < $FILEPATH/\"diseased_MCAV\"\n",
    "\n",
    "#bash script: ~/mcav/diseased_mcav/mapping\n",
    "#JOB ID: 21753567\n",
    "#Time elapsed:"
   ]
  },
  {
   "cell_type": "raw",
   "id": "8561bace-07db-4151-aa64-82cc58dd0f8c",
   "metadata": {},
   "source": [
    "Total time for backward call to driver() for mirror index: 00:09:24\n",
    "Renaming healthy_2019_mcav_contigs.3.bt2.tmp to healthy_2019_mcav_contigs.3.bt2\n",
    "Renaming healthy_2019_mcav_contigs.4.bt2.tmp to healthy_2019_mcav_contigs.4.bt2\n",
    "Renaming healthy_2019_mcav_contigs.1.bt2.tmp to healthy_2019_mcav_contigs.1.bt2\n",
    "Renaming healthy_2019_mcav_contigs.2.bt2.tmp to healthy_2019_mcav_contigs.2.bt2\n",
    "Renaming healthy_2019_mcav_contigs.rev.1.bt2.tmp to healthy_2019_mcav_contigs.rev.1.bt2\n",
    "Renaming healthy_2019_mcav_contigs.rev.2.bt2.tmp to healthy_2019_mcav_contigs.rev.2.bt2\n",
    "78898318 reads; of these:\n",
    "  78898318 (100.00%) were paired; of these:\n",
    "    25483422 (32.30%) aligned concordantly 0 times\n",
    "    30987826 (39.28%) aligned concordantly exactly 1 time\n",
    "    22427070 (28.43%) aligned concordantly >1 times\n",
    "    ----\n",
    "    25483422 pairs aligned concordantly 0 times; of these:\n",
    "      3401917 (13.35%) aligned discordantly 1 time\n",
    "    ----\n",
    "    22081505 pairs aligned 0 times concordantly or discordantly; of these:\n",
    "      44163010 mates make up the pairs; of these:\n",
    "        27055105 (61.26%) aligned 0 times\n",
    "        8006167 (18.13%) aligned exactly 1 time\n",
    "        9101738 (20.61%) aligned >1 times\n",
    "82.85% overall alignment rate\n",
    "samtools: /home/brooke_sienkiewicz_student_uml_edu/.conda/envs/anvio-8/bin/../lib/libtinfow.so.6: no version information available (required by samtools)\n",
    "samtools: /home/brooke_sienkiewicz_student_uml_edu/.conda/envs/anvio-8/bin/../lib/libncursesw.so.6: no version information available (required by samtools)\n",
    "samtools: /home/brooke_sienkiewicz_student_uml_edu/.conda/envs/anvio-8/bin/../lib/libncursesw.so.6: no version information available (required by samtools)\n",
    "Sorted BAM File ..............................: /project/pi_sarah_gignouxwolfsohn_uml_edu/brooke/mcav/assembly/healthy_2019_mcav/mapping/index/062019_BEL_CBC_T3_8_MCAV.bam\n",
    "BAM File Index ...............................: /project/pi_sarah_gignouxwolfsohn_uml_edu/brooke/mcav/assembly/healthy_2019_mcav/mapping/index/062019_BEL_CBC_T3_8_MCAV.bam.bai\n",
    "2427227 reads; of these:\n",
    "  2427227 (100.00%) were paired; of these:\n",
    "    1142926 (47.09%) aligned concordantly 0 times\n",
    "    979385 (40.35%) aligned concordantly exactly 1 time\n",
    "    304916 (12.56%) aligned concordantly >1 times\n",
    "    ----\n",
    "    1142926 pairs aligned concordantly 0 times; of these:\n",
    "      59337 (5.19%) aligned discordantly 1 time\n",
    "    ----\n",
    "    1083589 pairs aligned 0 times concordantly or discordantly; of these:\n",
    "      2167178 mates make up the pairs; of these:\n",
    "        1971186 (90.96%) aligned 0 times\n",
    "        135563 (6.26%) aligned exactly 1 time\n",
    "        60429 (2.79%) aligned >1 times\n",
    "59.39% overall alignment rate\n",
    "samtools: /home/brooke_sienkiewicz_student_uml_edu/.conda/envs/anvio-8/bin/../lib/libtinfow.so.6: no version information available (required by samtools)\n",
    "samtools: /home/brooke_sienkiewicz_student_uml_edu/.conda/envs/anvio-8/bin/../lib/libncursesw.so.6: no version information available (required by samtools)\n",
    "samtools: /home/brooke_sienkiewicz_student_uml_edu/.conda/envs/anvio-8/bin/../lib/libncursesw.so.6: no version information available (required by samtools)\n",
    "Sorted BAM File ..............................: /project/pi_sarah_gignouxwolfsohn_uml_edu/brooke/mcav/assembly/healthy_2019_mcav/mapping/index/062019_BEL_CBC_T2_16_MCAV.bam\n",
    "BAM File Index ...............................: /project/pi_sarah_gignouxwolfsohn_uml_edu/brooke/mcav/assembly/healthy_2019_mcav/mapping/index/062019_BEL_CBC_T2_16_MCAV.bam.bai\n",
    "8436904 reads; of these:\n",
    "  8436904 (100.00%) were paired; of these:\n",
    "    4582561 (54.32%) aligned concordantly 0 times\n",
    "    2188671 (25.94%) aligned concordantly exactly 1 time\n",
    "    1665672 (19.74%) aligned concordantly >1 times\n",
    "    ----\n",
    "    4582561 pairs aligned concordantly 0 times; of these:\n",
    "      520726 (11.36%) aligned discordantly 1 time\n",
    "    ----\n",
    "    4061835 pairs aligned 0 times concordantly or discordantly; of these:\n",
    "      8123670 mates make up the pairs; of these:\n",
    "        6396763 (78.74%) aligned 0 times\n",
    "        652011 (8.03%) aligned exactly 1 time\n",
    "        1074896 (13.23%) aligned >1 times\n",
    "62.09% overall alignment rate\n",
    "samtools: /home/brooke_sienkiewicz_student_uml_edu/.conda/envs/anvio-8/bin/../lib/libtinfow.so.6: no version information available (required by samtools)\n",
    "samtools: /home/brooke_sienkiewicz_student_uml_edu/.conda/envs/anvio-8/bin/../lib/libncursesw.so.6: no version information available (required by samtools)\n",
    "samtools: /home/brooke_sienkiewicz_student_uml_edu/.conda/envs/anvio-8/bin/../lib/libncursesw.so.6: no version information available (required by samtools)\n",
    "Sorted BAM File ..............................: /project/pi_sarah_gignouxwolfsohn_uml_edu/brooke/mcav/assembly/healthy_2019_mcav/mapping/index/062019_BEL_CBC_T1_20_MCAV.bam\n",
    "BAM File Index ...............................: /project/pi_sarah_gignouxwolfsohn_uml_edu/brooke/mcav/assembly/healthy_2019_mcav/mapping/index/062019_BEL_CBC_T1_20_MCAV.bam.bai\n",
    "3891938 reads; of these:\n",
    "  3891938 (100.00%) were paired; of these:\n",
    "    1568894 (40.31%) aligned concordantly 0 times\n",
    "    1739313 (44.69%) aligned concordantly exactly 1 time\n",
    "    583731 (15.00%) aligned concordantly >1 times\n",
    "    ----\n",
    "    1568894 pairs aligned concordantly 0 times; of these:\n",
    "      336330 (21.44%) aligned discordantly 1 time\n",
    "    ----\n",
    "    1232564 pairs aligned 0 times concordantly or discordantly; of these:\n",
    "      2465128 mates make up the pairs; of these:\n",
    "        1878014 (76.18%) aligned 0 times\n",
    "        326397 (13.24%) aligned exactly 1 time\n",
    "        260717 (10.58%) aligned >1 times\n",
    "75.87% overall alignment rate\n",
    "samtools: /home/brooke_sienkiewicz_student_uml_edu/.conda/envs/anvio-8/bin/../lib/libtinfow.so.6: no version information available (required by samtools)\n",
    "samtools: /home/brooke_sienkiewicz_student_uml_edu/.conda/envs/anvio-8/bin/../lib/libncursesw.so.6: no version information available (required by samtools)\n",
    "samtools: /home/brooke_sienkiewicz_student_uml_edu/.conda/envs/anvio-8/bin/../lib/libncursesw.so.6: no version information available (required by samtools)\n",
    "Sorted BAM File ..............................: /project/pi_sarah_gignouxwolfsohn_uml_edu/brooke/mcav/assembly/healthy_2019_mcav/mapping/index/062019_BEL_CBC_T1_24_MCAV.bam\n",
    "BAM File Index ...............................: /project/pi_sarah_gignouxwolfsohn_uml_edu/brooke/mcav/assembly/healthy_2019_mcav/mapping/index/062019_BEL_CBC_T1_24_MCAV.bam.bai\n",
    "4166616 reads; of these:\n",
    "  4166616 (100.00%) were paired; of these:\n",
    "    1975762 (47.42%) aligned concordantly 0 times\n",
    "    1609999 (38.64%) aligned concordantly exactly 1 time\n",
    "    580855 (13.94%) aligned concordantly >1 times\n",
    "    ----\n",
    "    1975762 pairs aligned concordantly 0 times; of these:\n",
    "      612466 (31.00%) aligned discordantly 1 time\n",
    "    ----\n",
    "    1363296 pairs aligned 0 times concordantly or discordantly; of these:\n",
    "      2726592 mates make up the pairs; of these:\n",
    "        1820626 (66.77%) aligned 0 times\n",
    "        435846 (15.99%) aligned exactly 1 time\n",
    "        470120 (17.24%) aligned >1 times\n",
    "78.15% overall alignment rate\n",
    "samtools: /home/brooke_sienkiewicz_student_uml_edu/.conda/envs/anvio-8/bin/../lib/libtinfow.so.6: no version information available (required by samtools)\n",
    "samtools: /home/brooke_sienkiewicz_student_uml_edu/.conda/envs/anvio-8/bin/../lib/libncursesw.so.6: no version information available (required by samtools)\n",
    "samtools: /home/brooke_sienkiewicz_student_uml_edu/.conda/envs/anvio-8/bin/../lib/libncursesw.so.6: no version information available (required by samtools)\n",
    "Sorted BAM File ..............................: /project/pi_sarah_gignouxwolfsohn_uml_edu/brooke/mcav/assembly/healthy_2019_mcav/mapping/index/062019_BEL_CBC_T3_9_MCAV.bam\n",
    "BAM File Index ...............................: /project/pi_sarah_gignouxwolfsohn_uml_edu/brooke/mcav/assembly/healthy_2019_mcav/mapping/index/062019_BEL_CBC_T3_9_MCAV.bam.bai\n",
    "3722331 reads; of these:\n",
    "  3722331 (100.00%) were paired; of these:\n",
    "    1419935 (38.15%) aligned concordantly 0 times\n",
    "    1769687 (47.54%) aligned concordantly exactly 1 time\n",
    "    532709 (14.31%) aligned concordantly >1 times\n",
    "    ----\n",
    "    1419935 pairs aligned concordantly 0 times; of these:\n",
    "      207262 (14.60%) aligned discordantly 1 time\n",
    "    ----\n",
    "    1212673 pairs aligned 0 times concordantly or discordantly; of these:\n",
    "      2425346 mates make up the pairs; of these:\n",
    "        1992481 (82.15%) aligned 0 times\n",
    "        244223 (10.07%) aligned exactly 1 time\n",
    "        188642 (7.78%) aligned >1 times\n",
    "73.24% overall alignment rate\n",
    "samtools: /home/brooke_sienkiewicz_student_uml_edu/.conda/envs/anvio-8/bin/../lib/libtinfow.so.6: no version information available (required by samtools)\n",
    "samtools: /home/brooke_sienkiewicz_student_uml_edu/.conda/envs/anvio-8/bin/../lib/libncursesw.so.6: no version information available (required by samtools)\n",
    "samtools: /home/brooke_sienkiewicz_student_uml_edu/.conda/envs/anvio-8/bin/../lib/libncursesw.so.6: no version information available (required by samtools)\n",
    "Sorted BAM File ..............................: /project/pi_sarah_gignouxwolfsohn_uml_edu/brooke/mcav/assembly/healthy_2019_mcav/mapping/index/062019_BEL_CBC_T3_14_MCAV.bam\n",
    "BAM File Index ...............................: /project/pi_sarah_gignouxwolfsohn_uml_edu/brooke/mcav/assembly/healthy_2019_mcav/mapping/index/062019_BEL_CBC_T3_14_MCAV.bam.bai\n",
    "6548146 reads; of these:\n",
    "  6548146 (100.00%) were paired; of these:\n",
    "    2864951 (43.75%) aligned concordantly 0 times\n",
    "    2772876 (42.35%) aligned concordantly exactly 1 time\n",
    "    910319 (13.90%) aligned concordantly >1 times\n",
    "    ----\n",
    "    2864951 pairs aligned concordantly 0 times; of these:\n",
    "      757416 (26.44%) aligned discordantly 1 time\n",
    "    ----\n",
    "    2107535 pairs aligned 0 times concordantly or discordantly; of these:\n",
    "      4215070 mates make up the pairs; of these:\n",
    "        2999564 (71.16%) aligned 0 times\n",
    "        617293 (14.64%) aligned exactly 1 time\n",
    "        598213 (14.19%) aligned >1 times\n",
    "77.10% overall alignment rate\n",
    "samtools: /home/brooke_sienkiewicz_student_uml_edu/.conda/envs/anvio-8/bin/../lib/libtinfow.so.6: no version information available (required by samtools)\n",
    "samtools: /home/brooke_sienkiewicz_student_uml_edu/.conda/envs/anvio-8/bin/../lib/libncursesw.so.6: no version information available (required by samtools)\n",
    "samtools: /home/brooke_sienkiewicz_student_uml_edu/.conda/envs/anvio-8/bin/../lib/libncursesw.so.6: no version information available (required by samtools)\n",
    "Sorted BAM File ..............................: /project/pi_sarah_gignouxwolfsohn_uml_edu/brooke/mcav/assembly/healthy_2019_mcav/mapping/index/062019_BEL_CBC_T3_1_MCAV.bam\n",
    "BAM File Index ...............................: /project/pi_sarah_gignouxwolfsohn_uml_edu/brooke/mcav/assembly/healthy_2019_mcav/mapping/index/062019_BEL_CBC_T3_1_MCAV.bam.bai\n",
    "2321387 reads; of these:\n",
    "  2321387 (100.00%) were paired; of these:\n",
    "    1049950 (45.23%) aligned concordantly 0 times\n",
    "    984568 (42.41%) aligned concordantly exactly 1 time\n",
    "    286869 (12.36%) aligned concordantly >1 times\n",
    "    ----\n",
    "    1049950 pairs aligned concordantly 0 times; of these:\n",
    "      254746 (24.26%) aligned discordantly 1 time\n",
    "    ----\n",
    "    795204 pairs aligned 0 times concordantly or discordantly; of these:\n",
    "      1590408 mates make up the pairs; of these:\n",
    "        1198784 (75.38%) aligned 0 times\n",
    "        223412 (14.05%) aligned exactly 1 time\n",
    "        168212 (10.58%) aligned >1 times\n",
    "74.18% overall alignment rate\n",
    "samtools: /home/brooke_sienkiewicz_student_uml_edu/.conda/envs/anvio-8/bin/../lib/libtinfow.so.6: no version information available (required by samtools)\n",
    "samtools: /home/brooke_sienkiewicz_student_uml_edu/.conda/envs/anvio-8/bin/../lib/libncursesw.so.6: no version information available (required by samtools)\n",
    "samtools: /home/brooke_sienkiewicz_student_uml_edu/.conda/envs/anvio-8/bin/../lib/libncursesw.so.6: no version information available (required by samtools)\n",
    "Sorted BAM File ..............................: /project/pi_sarah_gignouxwolfsohn_uml_edu/brooke/mcav/assembly/healthy_2019_mcav/mapping/index/062019_BEL_CBC_T3_15_MCAV.bam\n",
    "BAM File Index ...............................: /project/pi_sarah_gignouxwolfsohn_uml_edu/brooke/mcav/assembly/healthy_2019_mcav/mapping/index/062019_BEL_CBC_T3_15_MCAV.bam.bai\n",
    "7533935 reads; of these:\n",
    "  7533935 (100.00%) were paired; of these:\n",
    "    2569664 (34.11%) aligned concordantly 0 times\n",
    "    3747553 (49.74%) aligned concordantly exactly 1 time\n",
    "    1216718 (16.15%) aligned concordantly >1 times\n",
    "    ----\n",
    "    2569664 pairs aligned concordantly 0 times; of these:\n",
    "      402544 (15.67%) aligned discordantly 1 time\n",
    "    ----\n",
    "    2167120 pairs aligned 0 times concordantly or discordantly; of these:\n",
    "      4334240 mates make up the pairs; of these:\n",
    "        3434037 (79.23%) aligned 0 times\n",
    "        477616 (11.02%) aligned exactly 1 time\n",
    "        422587 (9.75%) aligned >1 times\n",
    "77.21% overall alignment rate\n",
    "samtools: /home/brooke_sienkiewicz_student_uml_edu/.conda/envs/anvio-8/bin/../lib/libtinfow.so.6: no version information available (required by samtools)\n",
    "samtools: /home/brooke_sienkiewicz_student_uml_edu/.conda/envs/anvio-8/bin/../lib/libncursesw.so.6: no version information available (required by samtools)\n",
    "samtools: /home/brooke_sienkiewicz_student_uml_edu/.conda/envs/anvio-8/bin/../lib/libncursesw.so.6: no version information available (required by samtools)\n",
    "Sorted BAM File ..............................: /project/pi_sarah_gignouxwolfsohn_uml_edu/brooke/mcav/assembly/healthy_2019_mcav/mapping/index/062019_BEL_CBC_T1_16_MCAV.bam\n",
    "BAM File Index ...............................: /project/pi_sarah_gignouxwolfsohn_uml_edu/brooke/mcav/assembly/healthy_2019_mcav/mapping/index/062019_BEL_CBC_T1_16_MCAV.bam.bai\n",
    "1094028 reads; of these:\n",
    "  1094028 (100.00%) were paired; of these:\n",
    "    336444 (30.75%) aligned concordantly 0 times\n",
    "    561693 (51.34%) aligned concordantly exactly 1 time\n",
    "    195891 (17.91%) aligned concordantly >1 times\n",
    "    ----\n",
    "    336444 pairs aligned concordantly 0 times; of these:\n",
    "      17680 (5.25%) aligned discordantly 1 time\n",
    "    ----\n",
    "    318764 pairs aligned 0 times concordantly or discordantly; of these:\n",
    "      637528 mates make up the pairs; of these:\n",
    "        576929 (90.49%) aligned 0 times\n",
    "        34252 (5.37%) aligned exactly 1 time\n",
    "        26347 (4.13%) aligned >1 times\n",
    "73.63% overall alignment rate\n",
    "samtools: /home/brooke_sienkiewicz_student_uml_edu/.conda/envs/anvio-8/bin/../lib/libtinfow.so.6: no version information available (required by samtools)\n",
    "samtools: /home/brooke_sienkiewicz_student_uml_edu/.conda/envs/anvio-8/bin/../lib/libncursesw.so.6: no version information available (required by samtools)\n",
    "samtools: /home/brooke_sienkiewicz_student_uml_edu/.conda/envs/anvio-8/bin/../lib/libncursesw.so.6: no version information available (required by samtools)\n",
    "Sorted BAM File ..............................: /project/pi_sarah_gignouxwolfsohn_uml_edu/brooke/mcav/assembly/healthy_2019_mcav/mapping/index/062019_BEL_CBC_T3_16_MCAV.bam\n",
    "BAM File Index ...............................: /project/pi_sarah_gignouxwolfsohn_uml_edu/brooke/mcav/assembly/healthy_2019_mcav/mapping/index/062019_BEL_CBC_T3_16_MCAV.bam.bai\n",
    "3141837 reads; of these:\n",
    "  3141837 (100.00%) were paired; of these:\n",
    "    1440687 (45.85%) aligned concordantly 0 times\n",
    "    1218005 (38.77%) aligned concordantly exactly 1 time\n",
    "    483145 (15.38%) aligned concordantly >1 times\n",
    "    ----\n",
    "    1440687 pairs aligned concordantly 0 times; of these:\n",
    "      137909 (9.57%) aligned discordantly 1 time\n",
    "    ----\n",
    "    1302778 pairs aligned 0 times concordantly or discordantly; of these:\n",
    "      2605556 mates make up the pairs; of these:\n",
    "        2230751 (85.62%) aligned 0 times\n",
    "        198736 (7.63%) aligned exactly 1 time\n",
    "        176069 (6.76%) aligned >1 times\n",
    "64.50% overall alignment rate\n",
    "samtools: /home/brooke_sienkiewicz_student_uml_edu/.conda/envs/anvio-8/bin/../lib/libtinfow.so.6: no version information available (required by samtools)\n",
    "samtools: /home/brooke_sienkiewicz_student_uml_edu/.conda/envs/anvio-8/bin/../lib/libncursesw.so.6: no version information available (required by samtools)\n",
    "samtools: /home/brooke_sienkiewicz_student_uml_edu/.conda/envs/anvio-8/bin/../lib/libncursesw.so.6: no version information available (required by samtools)\n",
    "Sorted BAM File ..............................: /project/pi_sarah_gignouxwolfsohn_uml_edu/brooke/mcav/assembly/healthy_2019_mcav/mapping/index/062019_BEL_CBC_T1_3_MCAV.bam\n",
    "BAM File Index ...............................: /project/pi_sarah_gignouxwolfsohn_uml_edu/brooke/mcav/assembly/healthy_2019_mcav/mapping/index/062019_BEL_CBC_T1_3_MCAV.bam.bai\n",
    "9068181 reads; of these:\n",
    "  9068181 (100.00%) were paired; of these:\n",
    "    3397764 (37.47%) aligned concordantly 0 times\n",
    "    4217847 (46.51%) aligned concordantly exactly 1 time\n",
    "    1452570 (16.02%) aligned concordantly >1 times\n",
    "    ----\n",
    "    3397764 pairs aligned concordantly 0 times; of these:\n",
    "      626219 (18.43%) aligned discordantly 1 time\n",
    "    ----\n",
    "    2771545 pairs aligned 0 times concordantly or discordantly; of these:\n",
    "      5543090 mates make up the pairs; of these:\n",
    "        4178641 (75.38%) aligned 0 times\n",
    "        716242 (12.92%) aligned exactly 1 time\n",
    "        648207 (11.69%) aligned >1 times\n",
    "76.96% overall alignment rate\n",
    "samtools: /home/brooke_sienkiewicz_student_uml_edu/.conda/envs/anvio-8/bin/../lib/libtinfow.so.6: no version information available (required by samtools)\n",
    "samtools: /home/brooke_sienkiewicz_student_uml_edu/.conda/envs/anvio-8/bin/../lib/libncursesw.so.6: no version information available (required by samtools)\n",
    "samtools: /home/brooke_sienkiewicz_student_uml_edu/.conda/envs/anvio-8/bin/../lib/libncursesw.so.6: no version information available (required by samtools)\n",
    "Sorted BAM File ..............................: /project/pi_sarah_gignouxwolfsohn_uml_edu/brooke/mcav/assembly/healthy_2019_mcav/mapping/index/062019_BEL_CBC_T2_5_MCAV.bam\n",
    "BAM File Index ...............................: /project/pi_sarah_gignouxwolfsohn_uml_edu/brooke/mcav/assembly/healthy_2019_mcav/mapping/index/062019_BEL_CBC_T2_5_MCAV.bam.bai\n",
    "1861982 reads; of these:\n",
    "  1861982 (100.00%) were paired; of these:\n",
    "    605992 (32.55%) aligned concordantly 0 times\n",
    "    1131930 (60.79%) aligned concordantly exactly 1 time\n",
    "    124060 (6.66%) aligned concordantly >1 times\n",
    "    ----\n",
    "    605992 pairs aligned concordantly 0 times; of these:\n",
    "      109151 (18.01%) aligned discordantly 1 time\n",
    "    ----\n",
    "    496841 pairs aligned 0 times concordantly or discordantly; of these:\n",
    "      993682 mates make up the pairs; of these:\n",
    "        864819 (87.03%) aligned 0 times\n",
    "        110104 (11.08%) aligned exactly 1 time\n",
    "        18759 (1.89%) aligned >1 times\n",
    "76.78% overall alignment rate\n",
    "samtools: /home/brooke_sienkiewicz_student_uml_edu/.conda/envs/anvio-8/bin/../lib/libtinfow.so.6: no version information available (required by samtools)\n",
    "samtools: /home/brooke_sienkiewicz_student_uml_edu/.conda/envs/anvio-8/bin/../lib/libncursesw.so.6: no version information available (required by samtools)\n",
    "samtools: /home/brooke_sienkiewicz_student_uml_edu/.conda/envs/anvio-8/bin/../lib/libncursesw.so.6: no version information available (required by samtools)\n",
    "Sorted BAM File ..............................: /project/pi_sarah_gignouxwolfsohn_uml_edu/brooke/mcav/assembly/healthy_2019_mcav/mapping/index/062019_BEL_CBC_T3_6_MCAV.bam\n",
    "BAM File Index ...............................: /project/pi_sarah_gignouxwolfsohn_uml_edu/brooke/mcav/assembly/healthy_2019_mcav/mapping/index/062019_BEL_CBC_T3_6_MCAV.bam.bai"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "da026c81-1e8d-49e0-94eb-966f1c8bb8d1",
   "metadata": {},
   "source": [
    "### Diseased mcav "
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "30084a1a-becb-4fca-9653-0f70430d3523",
   "metadata": {},
   "outputs": [],
   "source": [
    "#!/bin/bash\n",
    "#SBATCH -c 24  # Number of Cores per Task\n",
    "#SBATCH --mem=180G  # Requested Memory\n",
    "#SBATCH -p cpu  # Partition\n",
    "#SBATCH -t 20:00:00  # Job time limit\n",
    "#SBATCH --mail-type=ALL\n",
    "#SBATCH -o /project/pi_sarah_gignouxwolfsohn_uml_edu/brooke/mcav/diseased_mcav/mapping/slurm-mapping-%j.out  # %j = job ID\n",
    "\n",
    "module load miniconda/22.11.1-1\n",
    "conda activate anvio-8\n",
    "#filters contigs and reformats so naming is cleaner ('fixing deflines')\n",
    "#aligns each sample seq to the contigs ... need output bam for downstream analysis to determine how many of each taxa are in the samples \n",
    "\n",
    "\n",
    "SAMPLENAME=\"diseased_mcav\"\n",
    "READSPATH='/project/pi_sarah_gignouxwolfsohn_uml_edu/brooke/mcav/diseased_mcav/assembly/host_removed'\n",
    "CONTIGPATH='/project/pi_sarah_gignouxwolfsohn_uml_edu/brooke/mcav/diseased_mcav/assembly'\n",
    "CONTIGFILE=\"$SAMPLENAME\"_filtered.contigs.fasta\n",
    "newCONTIGPATH='/project/pi_sarah_gignouxwolfsohn_uml_edu/brooke/mcav/diseased_mcav/mapping'\n",
    "FILEPATH='/project/pi_sarah_gignouxwolfsohn_uml_edu/brooke/mcav/diseased_mcav'\n",
    "\n",
    "#fixes deflines for later and filters on size \n",
    "#anvi-script-reformat-fasta $CONTIGPATH/$CONTIGFILE -o $newCONTIGPATH/\"${SAMPLENAME}_filtered.contigs-fixed.fsa\" -l 1000 --simplify-names --report-file contig-rename-report-txt\n",
    "# filtering seq length 1000bp...need to play around with filtering based on bp length\n",
    "#deflines = sequence definition line. comes directly before its associated sequence in a fasta file\n",
    "\n",
    "\n",
    "FIXEDCON=\"${SAMPLENAME}_filtered.contigs-fixed.fa\"\n",
    "\n",
    "cd $newCONTIGPATH\n",
    "mkdir index\n",
    "#this builds an index of your contigs, which only needs to happen once\n",
    "bowtie2-build $FIXEDCON \"$SAMPLENAME\"_contigs\n",
    "# will not accept path before contigs file - must be in the correct dir \n",
    "# or error is just that there's abug so it won't accept .fa extension - renamed to .fsa\n",
    "ls #debugging\n",
    "\n",
    "while IFS= read -r SAMPLEID; do\n",
    "    #align reads to your contigs and collects that in a .sam file\n",
    "    bowtie2 --threads 11 -x \"$SAMPLENAME\"_contigs -1 $READSPATH/\"${SAMPLEID}\"_host_removed_R1.fastq -2 $READSPATH/\"${SAMPLEID}\"_host_removed_R2.fastq -S index/\"${SAMPLEID}\".sam\n",
    "    #note that these reads have host removed but NOT symbionts (which were removed from contigs, so there may be a bit of mismatch there (ie: low alignment rate))\n",
    "    #make sure to point it to the index not the FIXEDCON file (-x parameter)\n",
    "    \n",
    "    #converts your sam file to a bam file, but its neither sorted nor indexed, so we use an Anvi'O script to do so:\n",
    "    samtools view -F 4 -b -S index/\"${SAMPLEID}\".sam -o index/\"${SAMPLEID}\"-RAW.bam\n",
    "   \n",
    "    #index and sort your bam file\n",
    "    anvi-init-bam index/\"${SAMPLEID}\"-RAW.bam -o index/\"${SAMPLEID}\".bam\n",
    "    \n",
    "    rm index/\"${SAMPLEID}\"-RAW.bam\n",
    "done < $FILEPATH/\"diseased_MCAV\"\n",
    "\n",
    "#bash script: ~/mcav/diseased_mcav/mapping\n",
    "#JOB ID: 21753440, 21753947\n",
    "#Time elapsed:"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "a50e66fa-ee14-406c-ab54-7812446f5452",
   "metadata": {},
   "outputs": [],
   "source": [
    "WHAT WAS THERE\n",
    "===============================================\n",
    "Total num contigs ............................: 2,058,317\n",
    "Total num nucleotides ........................: 1,769,144,061\n",
    "\n",
    "WHAT WAS ASKED\n",
    "===============================================\n",
    "Simplify deflines? ...........................: Yes\n",
    "Add prefix to sequence names? ................: No\n",
    "Minimum length of contigs to keep ............: 1,000\n",
    "Max % gaps allowed ...........................: 100.00%\n",
    "Max num gaps allowed .........................: 1,000,000\n",
    "Exclude specific sequences? ..................: No\n",
    "Keep specific sequences? .....................: No\n",
    "Enforce sequence type? .......................: No\n",
    "\n",
    "WHAT HAPPENED\n",
    "===============================================\n",
    "Contigs removed ..............................: 1,671,251 (81.20% of all)\n",
    "Nucleotides removed ..........................: 679,593,620 (38.41% of all)\n",
    "Nucleotides modified .........................: 0 (0.00000% of all)\n",
    "Deflines simplified ..........................: True"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "12eeb65c-b779-4717-ad96-f830612f167f",
   "metadata": {},
   "outputs": [],
   "source": [
    "1837012 reads; of these:\n",
    "  1837012 (100.00%) were paired; of these:\n",
    "    1005586 (54.74%) aligned concordantly 0 times\n",
    "    688840 (37.50%) aligned concordantly exactly 1 time\n",
    "    142586 (7.76%) aligned concordantly >1 times\n",
    "    ----\n",
    "    1005586 pairs aligned concordantly 0 times; of these:\n",
    "      113638 (11.30%) aligned discordantly 1 time\n",
    "    ----\n",
    "    891948 pairs aligned 0 times concordantly or discordantly; of these:\n",
    "      1783896 mates make up the pairs; of these:\n",
    "        1564287 (87.69%) aligned 0 times\n",
    "        156888 (8.79%) aligned exactly 1 time\n",
    "        62721 (3.52%) aligned >1 times\n",
    "57.42% overall alignment rate\n",
    "samtools: /home/brooke_sienkiewicz_student_uml_edu/.conda/envs/anvio-8/bin/../lib/libtinfow.so.6: no version information available (required by samtools)\n",
    "samtools: /home/brooke_sienkiewicz_student_uml_edu/.conda/envs/anvio-8/bin/../lib/libncursesw.so.6: no version information available (required by samtools)\n",
    "samtools: /home/brooke_sienkiewicz_student_uml_edu/.conda/envs/anvio-8/bin/../lib/libncursesw.so.6: no version information available (required by samtools)\n",
    "Sorted BAM File ..............................: /project/pi_sarah_gignouxwolfsohn_uml_edu/brooke/mcav/diseased_mcav/mapping/index/052022_BEL_CBC_T1_70_MCAV.bam\n",
    "BAM File Index ...............................: /project/pi_sarah_gignouxwolfsohn_uml_edu/brooke/mcav/diseased_mcav/mapping/index/052022_BEL_CBC_T1_70_MCAV.bam.bai\n",
    "2419949 reads; of these:\n",
    "  2419949 (100.00%) were paired; of these:\n",
    "    1100682 (45.48%) aligned concordantly 0 times\n",
    "    1065716 (44.04%) aligned concordantly exactly 1 time\n",
    "    253551 (10.48%) aligned concordantly >1 times\n",
    "    ----\n",
    "    1100682 pairs aligned concordantly 0 times; of these:\n",
    "      101806 (9.25%) aligned discordantly 1 time\n",
    "    ----\n",
    "    998876 pairs aligned 0 times concordantly or discordantly; of these:\n",
    "      1997752 mates make up the pairs; of these:\n",
    "        1711757 (85.68%) aligned 0 times\n",
    "        196600 (9.84%) aligned exactly 1 time\n",
    "        89395 (4.47%) aligned >1 times\n",
    "64.63% overall alignment rate\n",
    "samtools: /home/brooke_sienkiewicz_student_uml_edu/.conda/envs/anvio-8/bin/../lib/libtinfow.so.6: no version information available (required by samtools)\n",
    "samtools: /home/brooke_sienkiewicz_student_uml_edu/.conda/envs/anvio-8/bin/../lib/libncursesw.so.6: no version information available (required by samtools)\n",
    "samtools: /home/brooke_sienkiewicz_student_uml_edu/.conda/envs/anvio-8/bin/../lib/libncursesw.so.6: no version information available (required by samtools)\n",
    "Sorted BAM File ..............................: /project/pi_sarah_gignouxwolfsohn_uml_edu/brooke/mcav/diseased_mcav/mapping/index/052022_BEL_CBC_T3_49_MCAV.bam\n",
    "BAM File Index ...............................: /project/pi_sarah_gignouxwolfsohn_uml_edu/brooke/mcav/diseased_mcav/mapping/index/052022_BEL_CBC_T3_49_MCAV.bam.bai\n",
    "2223749 reads; of these:\n",
    "  2223749 (100.00%) were paired; of these:\n",
    "    1170497 (52.64%) aligned concordantly 0 times\n",
    "    869677 (39.11%) aligned concordantly exactly 1 time\n",
    "    183575 (8.26%) aligned concordantly >1 times\n",
    "    ----\n",
    "    1170497 pairs aligned concordantly 0 times; of these:\n",
    "      268715 (22.96%) aligned discordantly 1 time\n",
    "    ----\n",
    "    901782 pairs aligned 0 times concordantly or discordantly; of these:\n",
    "      1803564 mates make up the pairs; of these:\n",
    "        1454932 (80.67%) aligned 0 times\n",
    "        212458 (11.78%) aligned exactly 1 time\n",
    "        136174 (7.55%) aligned >1 times\n",
    "67.29% overall alignment rate\n",
    "samtools: /home/brooke_sienkiewicz_student_uml_edu/.conda/envs/anvio-8/bin/../lib/libtinfow.so.6: no version information available (required by samtools)\n",
    "samtools: /home/brooke_sienkiewicz_student_uml_edu/.conda/envs/anvio-8/bin/../lib/libncursesw.so.6: no version information available (required by samtools)\n",
    "samtools: /home/brooke_sienkiewicz_student_uml_edu/.conda/envs/anvio-8/bin/../lib/libncursesw.so.6: no version information available (required by samtools)\n",
    "Sorted BAM File ..............................: /project/pi_sarah_gignouxwolfsohn_uml_edu/brooke/mcav/diseased_mcav/mapping/index/052022_BEL_CBC_T3_51_MCAV.bam\n",
    "BAM File Index ...............................: /project/pi_sarah_gignouxwolfsohn_uml_edu/brooke/mcav/diseased_mcav/mapping/index/052022_BEL_CBC_T3_51_MCAV.bam.bai\n",
    "75625794 reads; of these:\n",
    "  75625794 (100.00%) were paired; of these:\n",
    "    21352018 (28.23%) aligned concordantly 0 times\n",
    "    30095005 (39.79%) aligned concordantly exactly 1 time\n",
    "    24178771 (31.97%) aligned concordantly >1 times\n",
    "    ----\n",
    "    21352018 pairs aligned concordantly 0 times; of these:\n",
    "      3825538 (17.92%) aligned discordantly 1 time\n",
    "    ----\n",
    "    17526480 pairs aligned 0 times concordantly or discordantly; of these:\n",
    "      35052960 mates make up the pairs; of these:\n",
    "        20762952 (59.23%) aligned 0 times\n",
    "        5197142 (14.83%) aligned exactly 1 time\n",
    "        9092866 (25.94%) aligned >1 times\n",
    "86.27% overall alignment rate\n",
    "samtools: /home/brooke_sienkiewicz_student_uml_edu/.conda/envs/anvio-8/bin/../lib/libtinfow.so.6: no version information available (required by samtools)\n",
    "samtools: /home/brooke_sienkiewicz_student_uml_edu/.conda/envs/anvio-8/bin/../lib/libncursesw.so.6: no version information available (required by samtools)\n",
    "samtools: /home/brooke_sienkiewicz_student_uml_edu/.conda/envs/anvio-8/bin/../lib/libncursesw.so.6: no version information available (required by samtools)\n",
    "Sorted BAM File ..............................: /project/pi_sarah_gignouxwolfsohn_uml_edu/brooke/mcav/diseased_mcav/mapping/index/052022_BEL_CBC_T3_60_MCAV.bam\n",
    "BAM File Index ...............................: /project/pi_sarah_gignouxwolfsohn_uml_edu/brooke/mcav/diseased_mcav/mapping/index/052022_BEL_CBC_T3_60_MCAV.bam.bai\n",
    "19946893 reads; of these:\n",
    "  19946893 (100.00%) were paired; of these:\n",
    "    7558356 (37.89%) aligned concordantly 0 times\n",
    "    9600843 (48.13%) aligned concordantly exactly 1 time\n",
    "    2787694 (13.98%) aligned concordantly >1 times\n",
    "    ----\n",
    "    7558356 pairs aligned concordantly 0 times; of these:\n",
    "      1962157 (25.96%) aligned discordantly 1 time\n",
    "    ----\n",
    "    5596199 pairs aligned 0 times concordantly or discordantly; of these:\n",
    "      11192398 mates make up the pairs; of these:\n",
    "        7787485 (69.58%) aligned 0 times\n",
    "        1804624 (16.12%) aligned exactly 1 time\n",
    "        1600289 (14.30%) aligned >1 times\n",
    "80.48% overall alignment rate\n",
    "samtools: /home/brooke_sienkiewicz_student_uml_edu/.conda/envs/anvio-8/bin/../lib/libtinfow.so.6: no version information available (required by samtools)\n",
    "samtools: /home/brooke_sienkiewicz_student_uml_edu/.conda/envs/anvio-8/bin/../lib/libncursesw.so.6: no version information available (required by samtools)\n",
    "samtools: /home/brooke_sienkiewicz_student_uml_edu/.conda/envs/anvio-8/bin/../lib/libncursesw.so.6: no version information available (required by samtools)\n",
    "Sorted BAM File ..............................: /project/pi_sarah_gignouxwolfsohn_uml_edu/brooke/mcav/diseased_mcav/mapping/index/052022_BEL_CBC_T1_13_MCAV.bam\n",
    "BAM File Index ...............................: /project/pi_sarah_gignouxwolfsohn_uml_edu/brooke/mcav/diseased_mcav/mapping/index/052022_BEL_CBC_T1_13_MCAV.bam.bai"
   ]
  }
 ],
 "metadata": {
  "kernelspec": {
   "display_name": "Python 3 (ipykernel)",
   "language": "python",
   "name": "python3"
  },
  "language_info": {
   "codemirror_mode": {
    "name": "ipython",
    "version": 3
   },
   "file_extension": ".py",
   "mimetype": "text/x-python",
   "name": "python",
   "nbconvert_exporter": "python",
   "pygments_lexer": "ipython3",
   "version": "3.11.0"
  }
 },
 "nbformat": 4,
 "nbformat_minor": 5
}
