{
 "cells": [
  {
   "cell_type": "markdown",
   "id": "f2b24a3a",
   "metadata": {},
   "source": [
    "### Binning Metagenomes\n",
    "\n",
    "- clustering-based binning\n",
    "- Grouping contigs based on similar sequences into bins...essentially creating metagenome-assembled metagenomes (MAGs). The goal is that these contigs (sequences) will come together to form a single genome that represents a microbial species. These MAGs can be matched to a reference database and a species can be assigned, or the MAG represents a novel species.  \n",
    "\n",
    "Software documentations:\n",
    "\n",
    "MetaBAT: https://bitbucket.org/berkeleylab/metabat/src/master/README.md\n",
    "\n",
    "CONCOCT: https://github.com/BinPro/CONCOCT\n",
    "\n",
    "CheckM: https://github.com/Ecogenomics/CheckM/wiki\n",
    "\n",
    "Das_tool: https://github.com/cmks/DAS_Tool"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "a85590d3-5b61-4e66-9951-f94e95f687db",
   "metadata": {
    "tags": []
   },
   "outputs": [],
   "source": [
    "#Create new env and install software \n",
    "conda create -n binning python=3.7\n",
    "conda activate binning\n",
    "conda install -c bioconda metabat2\n",
    "\n",
    "conda install -c bioconda checkm-genome\n",
    "conda install -c bioconda das_tool"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "25da1fd1",
   "metadata": {
    "tags": []
   },
   "source": [
    "#### MetaBat2\n",
    "The first piece of code here generates a fairly simple text file for the coverage of these files. The next set of code runs MetaBat2  (v2.10.2) using minContig 1500, minCV 1.0, minCVSum 1.0, maxP 95%, minS 60, lcuster size 50000 and maxEdges 200. It sets the minimum size for a bin to 200000 basepairs, which is fairly low, so you can keep it. It gathers all mapping information into a single depth file, so you can use your 1 file in the next analysis. An important parameter to play around with is the minimum bin size. When set to 200000, this will severely limit the amount of bins you gain, especially if your samples aren't perfect. Therefore, it is wise to run MetaBAT several times with slight alterations to the -s flag to find your optimal setting (you don't want 3 bins, you also don't want 1000).\n",
    "\n",
    "For a reference on how to do this accurately, use: https://bitbucket.org/berkeleylab/metabat/wiki/Best%20Binning%20Practices"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "04840474",
   "metadata": {},
   "outputs": [],
   "source": [
    "#!/bin/bash\n",
    "#SBATCH -c 24  # Number of Cores per Task\n",
    "#SBATCH --mem=50G  # Requested Memory\n",
    "#SBATCH -p cpu  # Partition\n",
    "#SBATCH -t 20:00:00  # Job time limit\n",
    "#SBATCH --mail-type=ALL\n",
    "#SBATCH -o /project/pi_sarah_gignouxwolfsohn_uml_edu/brooke/mcav/healthy_2019_mcav/binning/slurm-%j.out  # %j = job ID  # %j = job ID\n",
    "\n",
    "module load miniconda/22.11.1-1\n",
    "conda activate binning\n",
    "\n",
    "#set parameters for binning:\n",
    "SAMPLENAME=\"healthy_2019_mcav\"\n",
    "OUTDIR=MetaBAT_\"$SAMPLENAME\"_bins2\n",
    "DEPTHPATH=/project/pi_sarah_gignouxwolfsohn_uml_edu/brooke/mcav/healthy_2019_mcav/binning/MetaBAT_\"$SAMPLENAME\"_bins\n",
    "\n",
    "CONTIGPATH=/project/pi_sarah_gignouxwolfsohn_uml_edu/brooke/mcav/healthy_2019_mcav/mapping\n",
    "CONTIGFILE=\"healthy_2019_mcav_filtered.contigs-fixed.fsa\"\n",
    "\n",
    "#run in dir: brooke/mcav/healthy_2019_mcav\n",
    "mkdir ./binning/$OUTDIR\n",
    "\n",
    "#create depth file for MetaBat2\n",
    "#jgi_summarize_bam_contig_depths --outputDepth ./binning/$OUTDIR/MetaBAT_\"$SAMPLENAME\"_depth.txt $CONTIGPATH/index/*.bam\n",
    "\n",
    "#MetaBat2 script with verbose output, minimum length (m)(has to be >=1500) and no min bin size \n",
    "metabat2 -i $CONTIGPATH/$CONTIGFILE -a $DEPTHPATH/MetaBAT_\"$SAMPLENAME\"_depth.txt \\\n",
    "-o ./binning/$OUTDIR/Metabat/ \\\n",
    "-m 1500 \n",
    "-s 5000\n",
    "#MetaBAT 2 (v2.12.1) using minContig 1500, minCV 1.0, minCVSum 1.0, \n",
    "\n",
    "# default parameters:\n",
    "# -m [ --minContig ] arg (=2500), has to be at least 1500 \n",
    "# maxP 95% Percentage of 'good' contigs considered for binning decided by connection among contigs\n",
    "# minS 60 Minimum score of a edge for binning (should be between 1 and 99). The greater, the more specific.\n",
    "# maxEdges 200 Maximum number of edges per node. The greater, the more sensitive.\n",
    "#-x [ --minCV ] arg (=1)           Minimum mean coverage of a contig in each library for binning.\n",
    "#  --minCVSum arg (=1)               Minimum total effective mean coverage of a contig (sum of depth over minCV) for binning.\n",
    "# -s [ --minClsSize ] arg (=200000) Minimum size of a bin as the output.\n",
    "\n",
    "#this runs CheckM immediately after and puts the results alongside your bins\n",
    "checkm lineage_wf -x fa -t 3 ./binning/$OUTDIR/Metabat ./binning/$OUTDIR/Metabat/bins-stats\n",
    "\n",
    "#bash script: metabat\n",
    "#script location: brooke/mcav/healthy_2019_mcav/binning\n",
    "#JOB ID: 19498217"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "cae5e684-8c71-4ec1-ad04-bfa0ed258234",
   "metadata": {},
   "outputs": [],
   "source": [
    "MetaBAT_healthy_2019_mcav_bins: metabat results from min length 1500, min bin size 200000 (i think is the automatic value)\n",
    "\n",
    "-------------------------------------------------------------------------------------------------------------------------------------------------------------------------\n",
    "  Bin Id            Marker lineage            # genomes   # markers   # marker sets    0     1    2    3   4   5+   Completeness   Contamination   Strain heterogeneity  \n",
    "-------------------------------------------------------------------------------------------------------------------------------------------------------------------------\n",
    "  .6        o__Sphingomonadales (UID3310)         26         569           293         3    545   21   0   0   0       99.59            3.15               0.00          \n",
    "  .14      c__Gammaproteobacteria (UID4444)      263         507           232         43   451   13   0   0   0       85.98            2.83              38.46          \n",
    "  .22           k__Bacteria (UID3187)            2258        188           117         73   112   3    0   0   0       59.27            1.28               0.00          \n",
    "  .16             k__Archaea (UID2)              207         149           107         80    47   17   3   0   2       40.20           20.84              18.06          \n",
    "  .9              k__Archaea (UID2)              207         149           107         82    51   10   5   1   0       39.73            4.41               0.00          \n",
    "  .17         p__Cyanobacteria (UID2143)         129         472           368        345   127   0    0   0   0       26.70            0.00               0.00          \n",
    "  .19        o__Actinomycetales (UID1530)        622         259           152        210    48   1    0   0   0       15.33            0.11               0.00          \n",
    "  .18            k__Bacteria (UID203)            5449        104            58         93    9    2    0   0   0        9.72            0.73              50.00          \n",
    "  .10             k__Archaea (UID2)              207         149           107        142    7    0    0   0   0        5.45            0.00               0.00          \n",
    "  .8              k__Archaea (UID2)              207         145           103        142    3    0    0   0   0        2.43            0.00               0.00          \n",
    "  .3             k__Bacteria (UID203)            5449        104            58        101    3    0    0   0   0        2.04            0.00               0.00          \n",
    "  .1             k__Bacteria (UID203)            5449         96            51         94    2    0    0   0   0        1.31            0.00               0.00          \n",
    "  .7                 root (UID1)                 5656         56            24         56    0    0    0   0   0        0.00            0.00               0.00          \n",
    "  .5                 root (UID1)                 5656         56            24         56    0    0    0   0   0        0.00            0.00               0.00          \n",
    "  .4                 root (UID1)                 5656         56            24         56    0    0    0   0   0        0.00            0.00               0.00          \n",
    "  .21                root (UID1)                 5656         56            24         56    0    0    0   0   0        0.00            0.00               0.00          \n",
    "  .20                root (UID1)                 5656         56            24         56    0    0    0   0   0        0.00            0.00               0.00          \n",
    "  .2                 root (UID1)                 5656         56            24         56    0    0    0   0   0        0.00            0.00               0.00          \n",
    "  .15                root (UID1)                 5656         56            24         56    0    0    0   0   0        0.00            0.00               0.00          \n",
    "  .13                root (UID1)                 5656         56            24         56    0    0    0   0   0        0.00            0.00               0.00          \n",
    "  .12                root (UID1)                 5656         56            24         56    0    0    0   0   0        0.00            0.00               0.00          \n",
    "  .11                root (UID1)                 5656         56            24         56    0    0    0   0   0        0.00            0.00               0.00          \n",
    "-------------------------------------------------------------------------------------------------------------------------------------------------------------------------\n",
    "[2024-02-21 16:42:10] INFO: { Current stage: 0:00:01.755 || Total: 0:19:31.234 }\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "ff20a3bd-155e-4094-8296-699e625412b9",
   "metadata": {},
   "outputs": [],
   "source": [
    "MetaBAT_healthy_2019_mcav_bins2: metabat results EXACTLY THE SAME from min length 1500, min bin size 5000\n"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "02d33a97",
   "metadata": {},
   "source": [
    " #### CONCOCT\n",
    "https://concoct.readthedocs.io/en/latest/installation.html\n",
    "\n",
    "This set of commands runs CONCOCT in its standard mode. It first creates a depth/coverage file for itself to use and then runs CONCOCT, with the standard settings. This means k-mer value is set to 4, minimum contig length is 1000, and CONCOCT runs on the exact amount of slots given to it by Hydra. \n",
    " \n",
    "CONCOCT creates a depth file out of the coverance created in the mapping step. It is key that this is all in the correct places before proceeding with binning. It creates a single file, which is then used for the complete binning process. Do keep in mind that binning might take awhile, so be prepared to let this run overnight."
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "9b362215-4d95-4da0-9fcd-9e9bf9414230",
   "metadata": {},
   "outputs": [],
   "source": [
    "# Conda installation\n",
    "conda config --add channels defaults\n",
    "conda config --add channels bioconda\n",
    "conda config --add channels conda-forge\n",
    "\n",
    "conda create -n concoct_env python=3 concoct"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "34233edd",
   "metadata": {},
   "outputs": [],
   "source": [
    "#!/bin/bash\n",
    "#SBATCH -c 24  # Number of Cores per Task\n",
    "#SBATCH --mem=50G  # Requested Memory\n",
    "#SBATCH -p cpu  # Partition\n",
    "#SBATCH -t 24:00:00  # Job time limit\n",
    "#SBATCH --mail-type=ALL\n",
    "#SBATCH -o /project/pi_sarah_gignouxwolfsohn_uml_edu/brooke/mcav/healthy_2019_mcav/binning/slurm-%j-concoct.out  # %j = job ID  # %j = job ID\n",
    "\n",
    "module load miniconda/22.11.1-1\n",
    "conda activate concoct_env\n",
    "\n",
    "\n",
    "#set parameters\n",
    "SAMPLENAME=\"healthy_2019_mcav\"\n",
    "BINPATH=/project/pi_sarah_gignouxwolfsohn_uml_edu/brooke/mcav/healthy_2019_mcav/binning/Concoct_\"$SAMPLENAME\"_bins\n",
    "CONTIGPATH=/project/pi_sarah_gignouxwolfsohn_uml_edu/brooke/mcav/healthy_2019_mcav/mapping\n",
    "CONTIGFILE=\"healthy_2019_mcav_filtered.contigs-fixed.fsa\"\n",
    "BAMPATH='/project/pi_sarah_gignouxwolfsohn_uml_edu/brooke/mcav/healthy_2019_mcav/mapping/index'\n",
    "TEMPDIR=concoct_\"$SAMPLENAME\"_temp\n",
    "\n",
    "#mkdir $BINPATH\n",
    "#creates the CONCOCT depth file\n",
    "#this part cuts up the contigs into 10kb pieces for CONCOCT to use \n",
    "cut_up_fasta.py $CONTIGPATH/$CONTIGFILE -c 10000 -o 0 --merge_last -b $BINPATH/${SAMPLENAME}_contigs_cut.bed > $BINPATH/${SAMPLENAME}_contigs_cut.fa\n",
    "#estimate contig coverage\n",
    "\n",
    "concoct_coverage_table.py $BINPATH/${SAMPLENAME}_contigs_cut.bed $BAMPATH/*.bam > $BINPATH/coverage_table_${SAMPLENAME}.tsv || { echo 'Exit code 2: failed to create coverage file, exiting.' && exit; }\n",
    "\n",
    "#CONCOCT script\n",
    "\n",
    "#mkdir $BINPATH/$TEMPDIR\n",
    "\n",
    "#run CONCOCT\n",
    "concoct --composition_file $BINPATH/${SAMPLENAME}_contigs_cut.fa --coverage_file $BINPATH/coverage_table_${SAMPLENAME}.tsv -t 3 -b $BINPATH/$TEMPDIR || { echo 'Exit code 3: CONCOCT failed to run, exiting.' && exit; }\n",
    "merge_cutup_clustering.py $BINPATH/$TEMPDIR/clustering_gt1000.csv > $BINPATH/$TEMPDIR/${SAMPLENAME}_clustering_merged.csv || { echo 'Exit code 4: failed to merge clusters, exiting.' && exit; }\n",
    "extract_fasta_bins.py $CONTIGPATH/$CONTIGFILE $BINPATH/$TEMPDIR/${SAMPLENAME}_clustering_merged.csv --output_path $BINPATH || { echo 'Exit code 5: Bins were not extracted, exiting.' && exit; }\n",
    "\n",
    "# Checkm is in binning env \n",
    "conda deactivate concoct_env\n",
    "conda activate binning\n",
    "#this runs CheckM immediately after and puts the results alongside your bins\n",
    "checkm lineage_wf -x fa -t 3 $BINPATH  $BINPATH/CheckM\n",
    "#can you add '--tab_table' to make output easier to read?\n",
    "\n",
    "#bash script: concoct\n",
    "#script location: mcav/healthy_2019_mcav/binning\n",
    "#JOB ID: 21569803\n",
    "#checkm job id: 21563395"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "a9bbbe85-1268-4307-a181-9935d8e1bc5c",
   "metadata": {},
   "outputs": [],
   "source": [
    "Bin Id                                   Marker lineage            # genomes   # markers   # marker sets    0     1    2    3    4   5+   Completeness   Contamination   Strain heterogeneity  \n",
    "-------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------\n",
    "  healthy_2019_mcav_contigs_cut             root (UID1)                 5656         56            24         0     0    1    2    5   48      100.00          610.87              1.19          \n",
    "  42                               o__Sphingomonadales (UID3310)         26         569           293         3    556   10   0    0   0       99.59            2.30               0.00          \n",
    "  56                              c__Gammaproteobacteria (UID4444)      263         507           232         45   450   12   0    0   0       85.79            2.68              33.33          \n",
    "  109                                  k__Bacteria (UID3187)            2258        188           117         53   132   3    0    0   0       68.90            1.12               0.00          \n",
    "  68                                     k__Archaea (UID2)              207         149           107         61    62   18   7    1   0       53.82           13.76               0.00          \n",
    "  139                                p__Cyanobacteria (UID2143)         129         472           368        242   220   8    2    0   0       46.70            1.84              14.29          \n",
    "  50                                    k__Bacteria (UID203)            5449        103            58         39    30   8    11   5   10      41.74           32.07               6.25          \n",
    "  150                               o__Actinomycetales (UID1530)        622         259           152        174    83   2    0    0   0       28.65            0.16               0.00          \n",
    "  167                                    k__Archaea (UID2)              207         149           107        113    28   6    2    0   0       22.39            5.99              16.67          \n",
    "  98                                     k__Archaea (UID2)              207         149           107        117    24   5    2    0   1       18.04            7.31              26.92          \n",
    "  170                                   k__Bacteria (UID203)            5449        104            58         90    14   0    0    0   0       13.87            0.00               0.00          \n",
    "  83                                    k__Bacteria (UID203)            5449        104            58         94    9    1    0    0   0        8.73            0.16              100.00         \n",
    "  113                                    k__Archaea (UID2)              207         149           107        134    11   4    0    0   0        8.54            1.08               0.00          \n",
    "  84                                     k__Archaea (UID2)              207         149           107        138    9    1    1    0   0        7.94            1.87              50.00          \n",
    "  7                                    k__Bacteria (UID1453)            901         171           117        144    27   0    0    0   0        5.72            0.00               0.00          \n",
    "  105                                   k__Bacteria (UID203)            5449        103            58         91    10   2    0    0   0        5.58            1.72               0.00          \n",
    "  54                                     k__Archaea (UID2)              207         149           107        143    6    0    0    0   0        4.52            0.00               0.00          \n",
    "  76                                        root (UID1)                 5656         56            24         55    1    0    0    0   0        4.17            0.00               0.00          \n",
    "  49                                        root (UID1)                 5656         56            24         55    1    0    0    0   0        4.17            0.00               0.00          \n",
    "  169                                    k__Archaea (UID2)              207         145           103        139    6    0    0    0   0        4.05            0.00               0.00          \n",
    "  66                                    k__Bacteria (UID203)            5449        103            58         97    6    0    0    0   0        3.28            0.00               0.00          \n",
    "  152                                   k__Bacteria (UID203)            5449        103            58         94    8    1    0    0   0        2.82            0.16               0.00          \n",
    "  136                                    k__Archaea (UID2)              207         145           103        142    3    0    0    0   0        1.94            0.00               0.00          \n",
    "  97                                    k__Bacteria (UID203)            5449        104            58        103    1    0    0    0   0        1.72            0.00               0.00          \n",
    "  137                                   k__Bacteria (UID203)            5449        104            58        102    2    0    0    0   0        1.44            0.00               0.00          \n",
    "  154                                    k__Archaea (UID2)              207         145           103        141    3    1    0    0   0        1.36            0.07               0.00          \n",
    "  14                                     k__Archaea (UID2)              207         149           107        146    3    0    0    0   0        1.08            0.00               0.00          \n",
    "  90                                    k__Bacteria (UID203)            5449        104            58        100    4    0    0    0   0        1.00            0.00               0.00          \n",
    "  75                                    k__Bacteria (UID203)            5449        102            56        101    1    0    0    0   0        0.89            0.00               0.00          \n",
    "  160                                       root (UID1)                 5656         56            24         55    1    0    0    0   0        0.52            0.00               0.00          \n",
    "  156                                   k__Bacteria (UID203)            5449        104            58        103    1    0    0    0   0        0.34            0.00               0.00          \n",
    "  43                                     k__Archaea (UID2)              207         148           106        147    1    0    0    0   0        0.31            0.00               0.00          \n",
    "  188                                    k__Archaea (UID2)              207         149           107        148    1    0    0    0   0        0.31            0.00               0.00          \n",
    "  166                                    k__Archaea (UID2)              207         149           107        148    1    0    0    0   0        0.31            0.00               0.00          \n",
    "  134                                   k__Bacteria (UID203)            5449        104            58        103    1    0    0    0   0        0.16            0.00               0.00          \n",
    "  45                                     k__Archaea (UID2)              207         149           107        148    1    0    0    0   0        0.07            0.00               0.00          \n",
    "  99                                        root (UID1)                 5656         56            24         56    0    0    0    0   0        0.00            0.00               0.00          \n",
    "  #UID1 repeats after this   \n",
    "-------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------\n",
    "[2024-03-15 03:42:52] INFO: { Current stage: 0:00:08.272 || Total: 0:43:47.260 }\n",
    "\n",
    "\n",
    "#obtained from stdout file "
   ]
  },
  {
   "cell_type": "markdown",
   "id": "fe6e42f9-be5c-4207-a6ba-6dca46ac8760",
   "metadata": {
    "tags": []
   },
   "source": [
    "### DAS_Tool \n",
    "- combines & refines bins into non-redundant set\n",
    "https://github.com/cmks/DAS_Tool"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 33,
   "id": "62b19cf6-43b1-4b60-9cfb-70dd2bb2ce7f",
   "metadata": {},
   "outputs": [],
   "source": [
    "import pandas as pd\n",
    "import os"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 26,
   "id": "34c7a181-f789-4ac2-a22a-cfbc017ad80e",
   "metadata": {},
   "outputs": [],
   "source": [
    "os.chdir('/project/pi_sarah_gignouxwolfsohn_uml_edu/brooke/mcav/healthy_2019_mcav/binning/Concoct_healthy_2019_mcav_bins/concoct_healthy_2019_mcav_temp')"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 40,
   "id": "a6304371-403a-4922-b39e-22ae8c2846df",
   "metadata": {},
   "outputs": [],
   "source": [
    "# Convert contig-bin file from concoct to tab-delimited format\n",
    "concoct_bins=pd.read_csv('healthy_2019_mcav_clustering_merged.csv',index_col=0)\n",
    "concoct_bins\n",
    "concoct_bins.to_csv(\"concoct_bins.tsv\", sep='\\t',header=False)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 46,
   "id": "4fbe7360-37b0-49ed-badf-71d26989893e",
   "metadata": {},
   "outputs": [],
   "source": [
    "# Convert all contig-bin files from metabat to single tab-delimited file\n",
    "\n",
    "directory = '/project/pi_sarah_gignouxwolfsohn_uml_edu/brooke/mcav/healthy_2019_mcav/binning/MetaBAT_healthy_2019_mcav_bins/Metabat'\n",
    "output_file = os.path.join(directory, 'metabat_bins.tsv')\n",
    "# Define a dictionary to store the mapping of contig IDs to bin IDs\n",
    "contig_bin_mapping = {}\n",
    "\n",
    "# Iterate through each file in the directory\n",
    "for filename in os.listdir(directory):\n",
    "    # Check if the file is a .fa file and matches the expected pattern\n",
    "    if filename.endswith('.fa') and filename.startswith('.'):\n",
    "        # Extract the bin ID from the filename\n",
    "        bin_id = int(filename.split('.')[1])\n",
    "        \n",
    "        # Open the file and read the contig IDs\n",
    "        with open(os.path.join(directory, filename), 'r') as file:\n",
    "            for line in file:\n",
    "                # Check if the line starts with '>'\n",
    "                if line.startswith('>'):\n",
    "                    # Extract the contig ID and add it to the dictionary with the corresponding bin ID\n",
    "                    contig_id = line.strip().split()[0][1:]  # Remove '>' and any additional information\n",
    "                    contig_bin_mapping[contig_id] = bin_id\n",
    "\n",
    "# Write the contig IDs and corresponding bin IDs to a tab-delimited file\n",
    "with open(output_file, 'w') as outfile:\n",
    "    for contig_id, bin_id in contig_bin_mapping.items():\n",
    "        outfile.write(f\"{contig_id}\\t{bin_id}\\n\")"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "10416c1f-4991-4da0-8bdc-cee0fe146439",
   "metadata": {},
   "outputs": [],
   "source": [
    "#!/bin/bash\n",
    "#SBATCH -c 24  # Number of Cores per Task\n",
    "#SBATCH --mem=50G  # Requested Memory\n",
    "#SBATCH -p cpu  # Partition\n",
    "#SBATCH -t 24:00:00  # Job time limit\n",
    "#SBATCH --mail-type=ALL\n",
    "#SBATCH -o /project/pi_sarah_gignouxwolfsohn_uml_edu/brooke/mcav/healthy_2019_mcav/binning/slurm-%j-das_tool.out  # %j = job ID  # %j = job ID\n",
    "\n",
    "\n",
    "module load miniconda/22.11.1-1\n",
    "conda activate binning\n",
    "\n",
    "# Set parameters\n",
    "CONCOCTPATH='/project/pi_sarah_gignouxwolfsohn_uml_edu/brooke/mcav/healthy_2019_mcav/binning/Concoct_healthy_2019_mcav_bins/concoct_healthy_2019_mcav_temp/concoct_bins.tsv'  \n",
    "METABATPATH='/project/pi_sarah_gignouxwolfsohn_uml_edu/brooke/mcav/healthy_2019_mcav/binning/MetaBAT_healthy_2019_mcav_bins/Metabat/metabat_bins.tsv'\n",
    "CONTIGPATH='/project/pi_sarah_gignouxwolfsohn_uml_edu/brooke/mcav/healthy_2019_mcav/mapping'\n",
    "CONTIGFILE=\"healthy_2019_mcav_filtered.contigs-fixed.fsa\"\n",
    "\n",
    "\n",
    "DAS_Tool -i $CONCOCTPATH,$METABATPATH \\\n",
    "-l concoct,metabat \\\n",
    "-c $CONTIGPATH/$CONTIGFILE \\\n",
    "-t 11 \\\n",
    "--write_bin_evals \\\n",
    "--write_bins \\\n",
    "-o das_tool\n",
    "\n",
    "# -i input list: tab seperated table of contigs-bins \n",
    "#--score_threshold default is 0.5\n",
    "\n",
    "#bash script: das_tool\n",
    "#script location: mcav/healthy_2019_mcav/binning\n",
    "# JOBID: 21582072"
   ]
  },
  {
   "cell_type": "raw",
   "id": "38c25824-4213-43e7-899f-cd90a19e3781",
   "metadata": {},
   "source": [
    "bin bin_set\tunique_SCGs\tredundant_SCGs\tSCG_set\tsize\tcontigs\tN50\tbin_score\tSCG_completeness\tSCG_redundancy\n",
    "concoct_42\tconcoct\t51\t0\tbacteria\t4125652\t41\t166474\t1\t100\t0\n",
    "metabat_14\tmetabat\t49\t0\tbacteria\t3110921\t257\t21708\t0.96078431372549\t96\t0\n",
    "concoct_109\tconcoct\t41\t1\tbacteria\t2027633\t960\t2243\t0.77948350071736\t80\t2"
   ]
  },
  {
   "cell_type": "raw",
   "id": "69036f85-c2de-4adf-8cf9-246048d5f291",
   "metadata": {},
   "source": [
    "bin\tbin_set\tunique_SCGs\tredundant_SCGs\tSCG_set\tsize\tcontigs\tN50\tbin_score\tSCG_completeness\tSCG_redundancy\n",
    "concoct_42\tconcoct\t51\t0\tbacteria\t4125652\t41\t166474\t1\t100\t0\n",
    "metabat_6\tmetabat\t51\t0\tbacteria\t4223952\t64\t153761\t1\t100\t0\n",
    "metabat_14\tmetabat\t49\t0\tbacteria\t3110921\t257\t21708\t0.96078431372549\t96\t0\n",
    "concoct_56\tconcoct\t49\t0\tbacteria\t3552249\t331\t19979\t0.96078431372549\t96\t0\n",
    "concoct_109\tconcoct\t41\t1\tbacteria\t2027633\t960\t2243\t0.77948350071736\t80\t2\n",
    "metabat_22\tmetabat\t34\t1\tbacteria\t1836506\t718\t2669\t0.63921568627451\t67\t2\n",
    "concoct_139\tconcoct\t27\t1\tbacteria\t51077582\t36279\t1329\t0.497385620915033\t53\t2\n",
    "concoct_150\tconcoct\t24\t0\tbacteria\t1140479\t667\t1701\t0.470588235294118\t47\t0\n",
    "concoct_68\tconcoct\t16\t1\tarchaea\t13943708\t7504\t1921\t0.370394736842105\t42\t3\n",
    "concoct_7\tconcoct\t15\t0\tbacteria\t102772\t15\t16847\t0.294117647058824\t29\t0\n",
    "metabat_19\tmetabat\t14\t0\tbacteria\t675848\t295\t2218\t0.274509803921569\t27\t0\n",
    "metabat_17\tmetabat\t14\t0\tbacteria\t4307961\t2011\t2016\t0.274509803921569\t27\t0\n",
    "metabat_9\tmetabat\t10\t0\tarchaea\t9585409\t4007\t2392\t0.263157894736842\t26\t0\n",
    "concoct_152\tconcoct\t7\t0\tbacteria\t3301432\t2384\t1322\t0.137254901960784\t14\t0\n",
    "metabat_16\tmetabat\t5\t0\tbacteria\t388694235\t76893\t6735\t0.0980392156862745\t10\t0\n",
    "concoct_113\tconcoct\t3\t0\tarchaea\t3274827\t2377\t1307\t0.0789473684210526\t8\t0\n",
    "metabat_18\tmetabat\t4\t0\tbacteria\t248950413\t80173\t3309\t0.0784313725490196\t8\t0\n",
    "concoct_167\tconcoct\t3\t0\tbacteria\t188758408\t19789\t10485\t0.0588235294117647\t6\t0\n",
    "concoct_170\tconcoct\t3\t0\tbacteria\t217645\t148\t1413\t0.0588235294117647\t6\t0\n",
    "concoct_105\tconcoct\t10\t2\tbacteria\t2445288\t1613\t1438\t0.0564705882352941\t20\t4\n",
    "concoct_84\tconcoct\t2\t0\tarchaea\t128714994\t67929\t2022\t0.0526315789473684\t5\t0\n",
    "concoct_66\tconcoct\t2\t0\tbacteria\t24061\t1\t24061\t0.0392156862745098\t4\t0\n",
    "concoct_50\tconcoct\t36\t18\tbacteria\t4965234\t3399\t1329\t0.0333333333333334\t71\t35\n",
    "concoct_83\tconcoct\t1\t0\tarchaea\t66540710\t14988\t4598\t0.0263157894736842\t3\t0\n",
    "concoct_98\tconcoct\t1\t0\tarchaea\t197017240\t81472\t2832\t0.0263157894736842\t3\t0\n",
    "concoct_49\tconcoct\t1\t0\tarchaea\t26130711\t13511\t2074\t0.0263157894736842\t3\t0\n",
    "metabat_7\tmetabat\t1\t0\tarchaea\t703403\t385\t1722\t0.0263157894736842\t3\t0\n",
    "concoct_188\tconcoct\t1\t0\tarchaea\t10402123\t6485\t1609\t0.0263157894736842\t3\t0\n",
    "concoct_169\tconcoct\t1\t0\tarchaea\t29965340\t20267\t1465\t0.0263157894736842\t3\t0\n",
    "concoct_137\tconcoct\t1\t0\tbacteria\t28227993\t3209\t11128\t0.0196078431372549\t2\t0\n",
    "concoct_18\tconcoct\t1\t0\tbacteria\t1886561\t518\t4735\t0.0196078431372549\t2\t0\n",
    "concoct_94\tconcoct\t1\t0\tbacteria\t104032662\t22865\t4709\t0.0196078431372549\t2\t0\n",
    "concoct_134\tconcoct\t1\t0\tbacteria\t15556310\t5375\t3153\t0.0196078431372549\t2\t0\n",
    "concoct_90\tconcoct\t1\t0\tbacteria\t111131\t78\t1360\t0.0196078431372549\t2\t0\n",
    "concoct_156\tconcoct\t1\t0\tbacteria\t378693\t267\t1352\t0.0196078431372549\t2\t0"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "ca64f14c-451a-4d60-baea-7417b8f59ecc",
   "metadata": {},
   "source": [
    "### prep file to import as collection file to anvio \n",
    "- metabin produces fasta files containing contigs of each bin \n",
    "- collection artifact requires a txt file that contains list of contigs with their associated bins (2 columns) "
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "5bd6e348-b140-4a55-bcd1-d71285103985",
   "metadata": {},
   "outputs": [],
   "source": [
    "#!/bin/bash\n",
    "# A simple script to convert metabin results to anvio\n",
    "FILES=$(find *.fa)\n",
    "for f in $FILES; do\n",
    " NAME=$(basename $f .fasta)\n",
    " grep \">\" $f | sed 's/>//' | sed -e \"s/$/\\t$NAME/\" | sed 's/\\./_/' >> metabins4anvio.txt\n",
    "done"
   ]
  }
 ],
 "metadata": {
  "kernelspec": {
   "display_name": "Python 3 (ipykernel)",
   "language": "python",
   "name": "python3"
  },
  "language_info": {
   "codemirror_mode": {
    "name": "ipython",
    "version": 3
   },
   "file_extension": ".py",
   "mimetype": "text/x-python",
   "name": "python",
   "nbconvert_exporter": "python",
   "pygments_lexer": "ipython3",
   "version": "3.11.0"
  }
 },
 "nbformat": 4,
 "nbformat_minor": 5
}
