{
 "cells": [
  {
   "cell_type": "markdown",
   "id": "c8efe34a-b914-4865-afbf-82fd6d8476a9",
   "metadata": {},
   "source": [
    "### Anvio - Contig profiling\n",
    "\n",
    "https://merenlab.org/2016/06/22/anvio-tutorial-v2/\n",
    "\n",
    "Contig profiling creates a database of your contigs. It calculates k-mer frequencies for your sample (standard k-setting is 4, which you can change with the --kmer-size parameter (DON'T unless you have a good reason)), soft splits long contigs, and identifies open reading frames (which can be skipped using --skip-gene-calling). Run the following code to generate your database:\n",
    "\n",
    "Subsequently, you can add various elements of analysis to your contig profile. The following list is available:\n",
    "\n",
    "Augustus + Prodigal gene calls: adds open reading frames to your dataset from the genes from the Augustus database (eukaryotes) and Prodigal (bacteria + archaea) WORKING ON THIS: NOT SURE IF IT WORKS\n",
    "\n",
    "Hidden Markov Model (HMM): A widely used prediction model in bioinformatics software, which can offer great advantages in homology detection.\n",
    "\n",
    "NCBI's Cluster of Orthologous genes (NCBI COG): this allows you to annotate your database with gene functions from NCBI COG. Current version: 2020\n",
    "\n",
    "KoFAM Metabolism calls: Uses the KEGG database to call metabolic genes and estimate paths of your community. Currently used KEGG version is KEGG_build_2020-12-23.\n",
    "\n",
    "Kaiju Taxonomy calls:\n",
    "\n",
    "Each of these adds a new layer of information to your dataset, so might be very interesting to explore."
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "3e0062cf-ed54-4bcd-b9a5-1ca4d41f9012",
   "metadata": {},
   "outputs": [],
   "source": [
    "conda activate anvio-8\n",
    "\n",
    "#install taxonomy \n",
    "diamond --version\n",
    "conda install diamond=0.9.14\n",
    "anvi-setup-scg-taxonomy\n",
    "\n",
    "#download ncbi-cogs \n",
    "anvi-setup-ncbi-cogs --num-threads 11\n",
    "\n",
    "#metabolism database\n",
    "anvi-setup-kegg-data"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "7cd0be4a-4a34-4cb2-b3a6-72efd8f78a43",
   "metadata": {},
   "outputs": [],
   "source": [
    "#!/bin/bash\n",
    "#SBATCH -c 24  # Number of Cores per Task\n",
    "#SBATCH --mem=50G  # Requested Memory\n",
    "#SBATCH -p cpu  # Partition\n",
    "#SBATCH -t 20:00:00  # Job time limit\n",
    "#SBATCH --mail-type=ALL\n",
    "#SBATCH -o //project/pi_sarah_gignouxwolfsohn_uml_edu/brooke/mcav/healthy_2019_mcav/anvio/slurm-%j.out  # %j = job ID\n",
    "\n",
    "module load miniconda/22.11.1-1\n",
    "conda activate anvio-8\n",
    "\n",
    "#Contig database from assembled genomes. stores information related to your sequences: positions of open reading frames, k-mer frequencies for each contig, functional and taxonomic annotation of genes, etc.\n",
    "#set parameters:\n",
    "SAMPLENAME='healthy_2019_mcav'\n",
    "CONTIGPATH='/project/pi_sarah_gignouxwolfsohn_uml_edu/brooke/mcav/healthy_2019_mcav/mapping'\n",
    "CONTIGFILE=\"healthy_2019_mcav_filtered.contigs-fixed.fsa\"\n",
    "OUTDIR='/project/pi_sarah_gignouxwolfsohn_uml_edu/brooke/mcav/healthy_2019_mcav/anvio/profiles'\n",
    "BAMPATH='/project/pi_sarah_gignouxwolfsohn_uml_edu/brooke/mcav/healthy_2019_mcav/mapping/index'\n",
    "\n",
    "\n",
    "#excute from healthy_2019_mcav dir\n",
    "\n",
    "#generates contig database \n",
    "\n",
    "#from merged, fixed contig fasta file created in previous step..need for downstream analysis\n",
    "#default k-mer frequency is 4\n",
    "anvi-gen-contigs-database -f $CONTIGPATH/$CONTIGFILE --project-name $SAMPLENAME -o ./anvio/$SAMPLENAME.contigs.db  \n",
    "\n",
    "#integrate HMMs into the database:\n",
    "anvi-run-hmms -c ./anvio/$SAMPLENAME.contigs.db --num-threads 6\n",
    "\n",
    "#this runs NCBI COGs against your contigs.db, integrating gene functions.\n",
    "anvi-run-ncbi-cogs -c ./anvio/$SAMPLENAME.contigs.db -T 4\n",
    "\n",
    "#ADD KEGG-KOFAM\n",
    "anvi-run-kegg-kofams -c ./anvio/$SAMPLENAME.contigs.db \\\n",
    "                     -T 4 #these are the threads that Anvi'O is allowed to use\n",
    "#ADD CONTIG STATS\n",
    "anvi-display-contigs-stats ./anvio/$SAMPLENAME.contigs.db --report-as-text --as-markdown -o ./anvio/anvio_stats.txt\n",
    "\n",
    "\n",
    "#create sample profiles\n",
    "for f in `cat healthy_2019_MCAV_sampleids`;\n",
    "do\n",
    "anvi-profile -c ./anvio/$SAMPLENAME.contigs.db  \\\n",
    "            -i $BAMPATH/\"$f\".bam \\\n",
    "            --min-percent-identity 95 \\\n",
    "            --sample-name \"healthy\"_\"$f\" \\\n",
    "            --output-dir $OUTDIR\n",
    "# --output-dir doesn't seem to work. its putting it in bampath\n",
    "#use contig database and sample bam files to create single profiles. can specify to keep contigs of min length (min-contig-length) and 95% identity to database\n",
    "#keep parameters consistent in order to merge to larger profile \n",
    "#already filtered out min 1000bp contigs in 'anvi-script-reformat-fasta', so not adding specification.\n",
    "done\n",
    "\n",
    "#merge single sample profiles to one profile\n",
    "anvi-merge -c ./anvio/\"$SAMPLENAME\".contigs.db \\\n",
    "            $OUTDIR/*/PROFILE.db \\\n",
    "            -o $OUTDIR/\"$SAMPLENAME\"_profile_merged\n",
    "\n",
    "\n",
    "\n",
    "#JOB ID: 21599196\n",
    "#sample profiles: 21634556\n",
    "#merge profiles: 21649620\n",
    "#bash script: anvio_db_profiles & profiles_script & profiles"
   ]
  },
  {
   "cell_type": "raw",
   "id": "04c8cd5d-db79-427b-b372-d3f413b22f11",
   "metadata": {
    "tags": []
   },
   "source": [
    "CONTIGS DB CREATE REPORT\n",
    "===============================================\n",
    "Split Length .................................: 20,000\n",
    "K-mer size ...................................: 4\n",
    "Skip gene calling? ...........................: False\n",
    "External gene calls provided? ................: False\n",
    "Ignoring internal stop codons? ...............: False\n",
    "Splitting pays attention to gene calls? ......: True\n",
    "Contigs with at least one gene call ..........: 429196 of 448229 (95.8%)\n",
    "Contigs database .............................: A new database, ./anvio/healthy_2019_mcav.contigs.db, has been created.\n",
    "Number of contigs ............................: 448,229\n",
    "Number of splits .............................: 448,485\n",
    "Total number of nucleotides ..................: 1,186,762,994\n",
    "Gene calling step skipped ....................: False\n",
    "Splits broke genes (non-mindful mode) ........: False\n",
    "Desired split length (what the user wanted) ..: 20,000\n",
    "Average split length (what anvi'o gave back) .: 22,045\n",
    "\n",
    "Number of decent hits added back after relaxing bitscore threshold : 4,214\n",
    "Total number of hits in annotation dictionary after adding these back : 16,720\n",
    "Gene functions ...............................: 16,720 function calls from 1 source (KOfam) for 16,393 unique gene calls have been added to the contigs database.\n",
    "Gene functions ...............................: 2,873 function calls from 1 source (KEGG_Module) for 2,820 unique gene calls have been added to the contigs database.\n",
    "Gene functions ...............................: 2,873 function calls from 1 source (KEGG_Class) for 2,820 unique gene calls have been added to the contigs database.\n",
    "Gene functions ...............................: 16,629 function calls from 1 source (KEGG_BRITE) for 16,306 unique gene calls have been added to the contigs database."
   ]
  },
  {
   "cell_type": "markdown",
   "id": "6667269f-e909-4b31-ade7-2b5efc785ce5",
   "metadata": {
    "tags": []
   },
   "source": [
    "#### Results - contig db \n",
    "|contigs_db|healthy_2019_mcav|\n",
    "|:--|:--:|\n",
    "|Total Length|1186762994|\n",
    "|Num Contigs|448229|\n",
    "|Num Contigs > 100 kb|19|\n",
    "|Num Contigs > 50 kb|47|\n",
    "|Num Contigs > 20 kb|1501|\n",
    "|Num Contigs > 10 kb|9799|\n",
    "|Num Contigs > 5 kb|42749|\n",
    "|Num Contigs > 2.5 kb|144061|\n",
    "|Longest Contig|437957|\n",
    "|Shortest Contig|1000|\n",
    "|Num Genes (prodigal)|1058453|\n",
    "|L50|98859|\n",
    "|L75|224211|\n",
    "|L90|342407|\n",
    "|N50|3176|\n",
    "|N75|1806|\n",
    "|N90|1265|\n",
    "|Archaea_76|466|\n",
    "|Bacteria_71|607|\n",
    "|Protista_83|184|\n",
    "|Ribosomal_RNA_12S|0|\n",
    "|Ribosomal_RNA_16S|7|\n",
    "|Ribosomal_RNA_18S|9|\n",
    "|Ribosomal_RNA_23S|13|\n",
    "|Ribosomal_RNA_28S|11|\n",
    "|Ribosomal_RNA_5S|0|\n",
    "|archaea (Archaea_76)|0|\n",
    "|eukarya (Protista_83)|1|\n",
    "|bacteria (Bacteria_71)|6|\n",
    "\n",
    "\n",
    "##### last time results for comparison: improvement :)\n",
    "contigs_db\tmcav_contigs_fixed\n",
    "Total Length\t57097694\n",
    "Num Contigs\t39414\n",
    "Num Contigs > 100 kb\t0\n",
    "Num Contigs > 50 kb\t0\n",
    "Num Contigs > 20 kb\t2\n",
    "Num Contigs > 10 kb\t35\n",
    "Num Contigs > 5 kb\t370\n",
    "Num Contigs > 2.5 kb\t2156\n",
    "Longest Contig\t24857\n",
    "Shortest Contig\t1000\n",
    "Num Genes (prodigal)\t58029\n",
    "L50\t14211\n",
    "L75\t25911\n",
    "L90\t33821\n",
    "N50\t1339\n",
    "N75\t1127\n",
    "N90\t1044\n",
    "Archaea_76\t9\n",
    "Bacteria_71\t8\n",
    "Protista_83\t5\n",
    "Ribosomal_RNA_12S\t0\n",
    "Ribosomal_RNA_16S\t0\n",
    "Ribosomal_RNA_18S\t2\n",
    "Ribosomal_RNA_23S\t0\n",
    "Ribosomal_RNA_28S\t2\n",
    "Ribosomal_RNA_5S\t0\n",
    "bacteria (Bacteria_71)\t0\n",
    "eukarya (Protista_83)\t0\n",
    "archaea (Archaea_76)\t0"
   ]
  },
  {
   "cell_type": "raw",
   "id": "0751e1a7-3e89-47b8-8ce4-1a355bdf540d",
   "metadata": {},
   "source": [
    "##### Results - profile merge\n",
    "WARNING\n",
    "===============================================\n",
    "It seems you have more than 20,000 splits in your samples to be merged. This is\n",
    "the soft limit for anvi'o to attempt to create a hierarchical clustering of your\n",
    "splits (which becomes the center tree in all anvi'o displays). If you want a\n",
    "hierarchical clustering to be done anyway, please see the flag `--enforce-\n",
    "hierarchical-clustering`. But more importantly, please take a look at the anvi'o\n",
    "tutorial to make sure you know your better options to analyze large metagenomic\n",
    "datasets with anvi'o.\n",
    "\n",
    "profiler_version .............................: 38\n",
    "output_dir ...................................: /project/pi_sarah_gignouxwolfsohn_uml_edu/brooke/mcav/healthy_2019_mcav/anvio/profiles/healthy_2019_mcav_profile_merged\n",
    "sample_id ....................................: healthy_2019_mcav_profile_merged\n",
    "description ..................................: None\n",
    "profile_db ...................................: /project/pi_sarah_gignouxwolfsohn_uml_edu/brooke/mcav/healthy_2019_mcav/anvio/profiles/healthy_2019_mcav_profile_merged/PROFILE.db\n",
    "merged .......................................: True\n",
    "contigs_db_hash ..............................: hasha580e845\n",
    "num_runs_processed ...........................: 447\n",
    "merged_sample_ids ............................: healthy_062019_BEL_CBC_T1_16_MCAV, healthy_062019_BEL_CBC_T1_20_MCAV, healthy_062019_BEL_CBC_T1_24_MCAV, healthy_062019_BEL_CBC_T1_3_MCAV, healthy_062019_BEL_CBC_T2_16_MCAV, healthy_062019_BEL_CBC_T2_5_MCAV, healthy_062019_BEL_CBC_T3_14_MCAV, healthy_062019_BEL_CBC_T3_15_MCAV, healthy_062019_BEL_CBC_T3_16_MCAV, healthy_062019_BEL_CBC_T3_1_MCAV, healthy_062019_BEL_CBC_T3_6_MCAV, healthy_062019_BEL_CBC_T3_8_MCAV, healthy_062019_BEL_CBC_T3_9_MCAV\n",
    "fetch_filter .................................: None, None, None, None, None, None, None, None, None, None, None, None, None\n",
    "min_percent_identity .........................: 95.0, 95.0, 95.0, 95.0, 95.0, 95.0, 95.0, 95.0, 95.0, 95.0, 95.0, 95.0, 95.0\n",
    "Common layer additional data keys ............: default\n",
    "total_reads_mapped ...........................: 11633833, 10477045, 5905862, 4052923, 2883268, 13957721, 5452181, 3443990, 1611127, 10096728, 2859145, 130741531, 6512606\n",
    "cmd_line .....................................: \n",
    "\n",
    "WARNING\n",
    "===============================================\n",
    "Codon frequencies were not profiled, hence, these tables will be empty in the\n",
    "merged profile database.\n",
    "\n",
    "\n",
    "* Anvi'o hierarchical clustering of contigs...\n",
    "\n",
    "\n",
    "* Additional data and layer orders...\n",
    "\n",
    "Auxiliary Data ...............................: Found: /project/pi_sarah_gignouxwolfsohn_uml_edu/brooke/mcav/healthy_2019_mcav/anvio/profiles/healthy_2019_mcav_profile_merged/AUXILIARY-DATA.db (v. 2)\n",
    "Profile Super ................................: Initialized with all 448485 splits: /project/pi_sarah_gignouxwolfsohn_uml_edu/brooke/mcav/healthy_2019_mcav/anvio/profiles/healthy_2019_mcav_profile_merged/PROFILE.db (v. 38)\n",
    "\n",
    "Layer orders added\n",
    "===============================================\n",
    "* std_coverage\n",
    "* mean_coverage\n",
    "* mean_coverage_Q2Q3\n",
    "* detection\n",
    "* abundance\n",
    "* variability\n",
    "\n",
    "Data groups added\n",
    "===============================================\n",
    "* default (w/4 items)\n",
    "\n",
    "* Happy ☘\n"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "5863737e-b4fe-4825-b2f3-3bc9dd5357d4",
   "metadata": {},
   "source": [
    "**Do not use below code...still troubleshooting**"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "dca386f1-d8e3-4b74-ad2c-4f66d5e0cfce",
   "metadata": {},
   "outputs": [],
   "source": [
    "#!/bin/bash\n",
    "#SBATCH -c 24  # Number of Cores per Task\n",
    "#SBATCH --mem=50G  # Requested Memory\n",
    "#SBATCH -p cpu  # Partition\n",
    "#SBATCH -t 6:00:00  # Job time limit\n",
    "#SBATCH --mail-type=ALL\n",
    "#SBATCH -o //project/pi_sarah_gignouxwolfsohn_uml_edu/brooke/mcav/healthy_2019_mcav/anvio/slurm-%j.out  # %j = job ID\n",
    "\n",
    "module load miniconda/22.11.1-1\n",
    "conda activate anvio-8\n",
    "\n",
    "SAMPLENAME='healthy_2019_mcav'\n",
    "#DIR=//project/pi_sarah_gignouxwolfsohn_uml_edu/brooke/results/MetaBAT_mcav_bins\n",
    "BINFILE=\"/project/pi_sarah_gignouxwolfsohn_uml_edu/brooke/mcav/healthy_2019_mcav/binning/das_tool_DASTool_bins/das_tool_DASTool_contig2bin.tsv\"\n",
    "PROFILEDIR='/project/pi_sarah_gignouxwolfsohn_uml_edu/brooke/mcav/healthy_2019_mcav/anvio/profiles'\n",
    "\n",
    "\n",
    "#convert metabin results to anvio format\n",
    "\n",
    "#FILES=$(find $DIR/*.fa)\n",
    "#for f in $FILES; do\n",
    "# NAME=$(basename $f .fa)\n",
    "# grep \">\" $f | sed 's/>//' | sed -e \"s/$/\\t$NAME/\" | sed 's/\\./_/' >> metabins4anvio.txt\n",
    "#done\n",
    "#metabin produces fasta files containing contigs of each bin \n",
    "#collection artifact requires a txt file that contains list of contigs with their associated bins (2 columns) \n",
    "\n",
    "# Import bin results as anvio collection \n",
    "# run from anvio dir \n",
    "anvi-import-collection $BINFILE \\\n",
    "                       -p $PROFILEDIR/\"$SAMPLENAME\"_profile_merged/PROFILE.db \\\n",
    "                       -c $SAMPLENAME.contigs.db \\\n",
    "                       --contigs-mode \\\n",
    "                       -C \"$SAMPLENAME\"_collection\n",
    "#import binning results of dastool from 3Binning step as a collection artifact in anvio\n",
    "##contigs-mode specificies that input txt file describes contigs names not split names\n",
    "  \n",
    "\n",
    "#bash script: import_collection\n",
    "#job ID: 21663018"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "64b8204a-0f7d-4e82-a5a9-c5cd806ab311",
   "metadata": {},
   "source": [
    "**Troubleshooting code below**\n",
    "- Trying to figure out clustering in anvio itself\n",
    "- Skip for now"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "9c5eba26-b14e-4a96-bce9-a5fd5e96d5a5",
   "metadata": {},
   "outputs": [],
   "source": [
    "anvi-cluster-contigs -p $OUTDIR/\"$SAMPLENAME\"_contigs_merged \\\n",
    "                    -c ./working/$SAMPLENAME.contigs.db \\\n",
    "                    --log-file //project/pi_sarah_gignouxwolfsohn_uml_edu/brooke/bash_scripts/cluster_log \\\n",
    "                    -C \"$SAMPLENAME\"_collection \\\n",
    "                    -T 3 \\\n",
    "                    --driver 'metabat2'\n",
    "#binning: clusters contigs. can use various drivers: 'concoct, metabat2, maxbin2, dastool, or binsanity.'\n",
    "\n",
    "#error output that the module of clustering isnt fully developed and is better to just use own binning software. use flag '--just-do-it' if you want to try it out"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "415912cf-c306-4b58-85b6-4322c5d5a3b6",
   "metadata": {},
   "source": [
    "#### Results from importing collection:"
   ]
  },
  {
   "cell_type": "raw",
   "id": "59514fec-31c8-452e-a39c-bfe4a91ee540",
   "metadata": {},
   "source": [
    "Contig/split name conversion .................: 1258 contig names converted into 1426 split names.\n",
    "Item names in input ..........................: 1,426\n",
    "Num bins in input ............................: 3\n",
    "Items in profile database ....................: 448,485\n",
    "Item names shared between input and db .......: 1,426\n",
    "\n",
    "WARNING\n",
    "===============================================\n",
    "447059 item(s) that were in the database, but were not in the input file, will\n",
    "not be described by any bin in the collection healthy_2019_mcav_collection. That\n",
    "is totally fine, but anvi'o hopes that you are aware of that. This means you\n",
    "have more things in your database than the number of things your input file\n",
    "describes. Here is an example of something that is in your database but not in\n",
    "any bin in your input file: c_000000243786_split_00001.\n",
    "\n",
    "Collections ..................................: The collection 'healthy_2019_mcav_collection' that describes 1,426 items in 3 bins was successfully added to the to the database at '/project/pi_sarah_gignouxwolfsohn_uml_edu/brooke/mcav/healthy_2019_mcav/anvio/profiles/healthy_2019_mcav_profile_merged/PROFILE.db'. Here is a full list of the bin names in this collection: concoct_42, metabat_14, concoct_109."
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "cfa6bd90-a188-498a-bda7-9dad067a5d54",
   "metadata": {},
   "outputs": [],
   "source": [
    "anvi-migrate --migrate-safely ./working/\"$SAMPLENAME\".contigs.db\n",
    "#migrate anvio artifact after updating anvio to version 8 - only do once if updated"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "42b9ddf9-2114-470b-820f-35c8b8b8922a",
   "metadata": {},
   "source": [
    "#### Interactive mode\n",
    "https://merenlab.org/2016/02/27/the-anvio-interactive-interface/\n",
    "- read more into features and go through tutorial \n",
    "https://merenlab.org/2015/11/28/visualizing-from-a-server/"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "5f44f72f-3c49-4603-a566-e9fce4ed82b3",
   "metadata": {},
   "outputs": [],
   "source": [
    "# ssh unity like normal\n",
    "# launch interactive job \n",
    "salloc -p cpu --mem 150G\n",
    "# start new tab and relaunch unity specifying compute node\n",
    "ssh -L 8080:cpu022:8080 unity\n",
    "\n",
    "# initiate command in compute node, go to http://0.0.0.0:8080 in local chrome  \n",
    "anvi-interactive -p $PROFILEDIR/\"$SAMPLENAME\"_profile_merged/PROFILE.db -c $SAMPLENAME.contigs.db -C \"$SAMPLENAME\"_collection\n",
    "#need collection info since skipped clustering in the merging step \n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "53facb1a-fc92-441a-bacb-95a23fd4b3ba",
   "metadata": {},
   "outputs": [],
   "source": [
    "ssh -L 8080:localhost:8080 unity\n",
    "#connect to unity this way from local terminal if you want to use the interactive browser below\n",
    "# ex: ssh -L 8080:localhost:8080 meren@server.university.edu\n",
    "\n",
    "# now on public server: https://anvi-server.org/"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "0bc88c7e-602c-436a-a2b6-de66e4678c1c",
   "metadata": {},
   "outputs": [],
   "source": [
    "# summarize bins - creates html output that summarizes completion/coverage of contigs in bins \n",
    "# takes long time to run \n",
    "anvi-summarize -p $PROFILEDIR/\"$SAMPLENAME\"_profile_merged/PROFILE.db -c $SAMPLENAME.contigs.db -o \"$SAMPLENAME\"-SUMMARY -C \"$SAMPLENAME\"_collection"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "da6fed86-b9e5-4704-b902-5a3a3de9f952",
   "metadata": {},
   "source": [
    "#### Taxonomy \n",
    "https://merenlab.org/2019/10/08/anvio-scg-taxonomy/\n",
    "- single-copy core genes (SCGs) and the taxonomy of the genomes - as defined by the GTDB - from which these genes are coming from"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "9d463a43-8bd6-4638-8eff-b22bf165050f",
   "metadata": {},
   "outputs": [],
   "source": [
    "module load miniconda/22.11.1-1\n",
    "conda activate anvio-8"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "c9ddff96-b0e7-4763-aeeb-cf2be3e7dae3",
   "metadata": {},
   "outputs": [],
   "source": [
    "SAMPLENAME='healthy_2019_mcav'\n",
    "PROFILEDIR='/project/pi_sarah_gignouxwolfsohn_uml_edu/brooke/mcav/healthy_2019_mcav/anvio/profiles'\n",
    "OUTDIR='/project/pi_sarah_gignouxwolfsohn_uml_edu/brooke/mcav/healthy_2019_mcav/anvio/taxonomy'"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "1fd59a1a-514c-4893-b27b-72b2308f7fb7",
   "metadata": {},
   "outputs": [],
   "source": [
    "anvi-run-scg-taxonomy -c \"$SAMPLENAME\".contigs.db\n",
    "# adding this to the contig db creation step, but here it was done separately \n",
    "# default percent identity: 90"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "ff387659-6192-4664-9f32-3f4b90200777",
   "metadata": {},
   "outputs": [],
   "source": [
    "anvi-estimate-scg-taxonomy -c \"$SAMPLENAME\".contigs.db \\\n",
    "                           -p $PROFILEDIR/\"$SAMPLENAME\"_profile_merged/PROFILE.db \\\n",
    "                           --num-threads 12 \\\n",
    "                           --metagenome-mode \\\n",
    "                           --compute-scg-coverages \\\n",
    "                           --output-file $OUTDIR/\"$SAMPLENAME\"-taxa-abundance.tsv\n",
    "#taxa matching from assembled metagenome - will just match ASVs to SCG taxa and give percent identity \n",
    "#difference here is that it will calculate relative abundances of the matched SCG taxa across all samples as well as percent identity (like an OTU table)\n",
    "# a little unclear on what --compute-scg-coverages does: estimate coverage of bins ?"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "58d02b6c-0ce9-4ef0-b854-00248ffc7b23",
   "metadata": {},
   "outputs": [],
   "source": [
    "anvi-show-collections-and-bins -p $PROFILEDIR/\"$SAMPLENAME\"_profile_merged/PROFILE.db"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "56db908c-74a0-4252-a395-3bb3cf5990c5",
   "metadata": {},
   "outputs": [],
   "source": [
    "#calculate taxonomy of MAGs\n",
    "anvi-estimate-scg-taxonomy -c \"$SAMPLENAME\".contigs.db \\\n",
    "                           -p $PROFILEDIR/\"$SAMPLENAME\"_profile_merged/PROFILE.db \\\n",
    "                           -C \"$SAMPLENAME\"_collection \\\n",
    "                           --num-threads 12 \\\n",
    "                           --metagenome-mode \\\n",
    "                           --compute-scg-coverages \\\n",
    "                           --output-file $OUTDIR/\"$SAMPLENAME\"-taxa-abundance.tsv\n"
   ]
  },
  {
   "cell_type": "raw",
   "id": "f65c298e-1b42-450d-9a3f-6953923c65a6",
   "metadata": {
    "tags": []
   },
   "source": [
    "Num relevant SCGs in contigs db ..............: 204                             \n",
    "Taxonomy .....................................: /home/brooke_sienkiewicz_student_uml_edu/.conda/envs/anvio-8/lib/python3.10/site-packages/anvio/data/misc/SCG_TAXONOMY/GTDB/ACCESSION_TO_TAXONOMY.txt.gz\n",
    "Database reference ...........................: /home/brooke_sienkiewicz_student_uml_edu/.conda/envs/anvio-8/lib/python3.10/site-packages/anvio/data/misc/SCG_TAXONOMY/GTDB/SCG_SEARCH_DATABASES\n",
    "Number of SCGs ...............................: 22\n",
    "\n",
    "Parameters for search\n",
    "===============================================\n",
    "Max number of target sequences ...............: 20\n",
    "Max e-value to report alignments .............: 1e-05\n",
    "Min percent identity to report alignments ....: 90.0\n",
    "Num aligment tasks running in parallel .......: 1\n",
    "Num CPUs per aligment task ...................: 1\n",
    "Log file path ................................: /tmp/tmpgsx1hgb6\n",
    "                                                                                \n",
    "WARNING\n",
    "===============================================\n",
    "Please take a careful look at the table below. It shows the number of sequences\n",
    "found in the contigs database for a given SCG, and how many of them actually was\n",
    "annotated based on sequences in GTDB. The discrepancy between these two numbers\n",
    "can occur due to multiple reasons. You may have bacterial or archaeal sequences\n",
    "in your data that are way too novel to hit anything in the GTDB above the\n",
    "percent identity cutoff set by the parameter`--min-percent-identity` to report\n",
    "alignments (which is set to 90.0 in the current run). Or, you may have a lot of\n",
    "eukaryotic organisms, single-copy core gene sequences of which may have no\n",
    "representation in the GTDB, in which case decreasing `--min-percent-identity`\n",
    "will only give you erronous results.\n",
    "\n",
    "+----------------+-----------------+---------------------+----------------------+\n",
    "| SCG            |  in_contigs_db  |  in_search_results  |  percent_annotation  |\n",
    "+================+=================+=====================+======================+\n",
    "| Ribosomal_L1   |        7        |          3          |         42.9         |\n",
    "+----------------+-----------------+---------------------+----------------------+\n",
    "| Ribosomal_L13  |        8        |          3          |         37.5         |\n",
    "+----------------+-----------------+---------------------+----------------------+\n",
    "| Ribosomal_L16  |       16        |          5          |         31.2         |\n",
    "+----------------+-----------------+---------------------+----------------------+\n",
    "| Ribosomal_L17  |        7        |          3          |         42.9         |\n",
    "+----------------+-----------------+---------------------+----------------------+\n",
    "| Ribosomal_L2   |       12        |          3          |          25          |\n",
    "+----------------+-----------------+---------------------+----------------------+\n",
    "| Ribosomal_L20  |        8        |          2          |          25          |\n",
    "+----------------+-----------------+---------------------+----------------------+\n",
    "| Ribosomal_L21p |        7        |          3          |         42.9         |\n",
    "+----------------+-----------------+---------------------+----------------------+\n",
    "| Ribosomal_L22  |        7        |          4          |         57.1         |\n",
    "+----------------+-----------------+---------------------+----------------------+\n",
    "| Ribosomal_L27A |        4        |          1          |          25          |\n",
    "+----------------+-----------------+---------------------+----------------------+\n",
    "| Ribosomal_L3   |       13        |          2          |         15.4         |\n",
    "+----------------+-----------------+---------------------+----------------------+\n",
    "| Ribosomal_L4   |        9        |          3          |         33.3         |\n",
    "+----------------+-----------------+---------------------+----------------------+\n",
    "| Ribosomal_L6   |        7        |          2          |         28.6         |\n",
    "+----------------+-----------------+---------------------+----------------------+\n",
    "| Ribosomal_L9_C |        7        |          2          |         28.6         |\n",
    "+----------------+-----------------+---------------------+----------------------+\n",
    "| Ribosomal_S11  |       15        |          4          |         26.7         |\n",
    "+----------------+-----------------+---------------------+----------------------+\n",
    "| Ribosomal_S2   |       12        |          2          |         16.7         |\n",
    "+----------------+-----------------+---------------------+----------------------+\n",
    "| Ribosomal_S20p |        7        |          3          |         42.9         |\n",
    "+----------------+-----------------+---------------------+----------------------+\n",
    "| Ribosomal_S3_C |       11        |          3          |         27.3         |\n",
    "+----------------+-----------------+---------------------+----------------------+\n",
    "| Ribosomal_S6   |        6        |          3          |          50          |\n",
    "+----------------+-----------------+---------------------+----------------------+\n",
    "| Ribosomal_S7   |       15        |          4          |         26.7         |\n",
    "+----------------+-----------------+---------------------+----------------------+\n",
    "| Ribosomal_S8   |       11        |          2          |         18.2         |\n",
    "+----------------+-----------------+---------------------+----------------------+\n",
    "| Ribosomal_S9   |       11        |          3          |         27.3         |\n",
    "+----------------+-----------------+---------------------+----------------------+\n",
    "| ribosomal_L24  |        4        |          2          |          50          |\n",
    "+----------------+-----------------+---------------------+----------------------+"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "c98e3cdf-1fc5-4fce-b830-d414fc2c7769",
   "metadata": {},
   "source": [
    "### Results"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "707e163a-eeb5-40cf-b618-b6b6708a001d",
   "metadata": {},
   "outputs": [],
   "source": [
    "SAMPLENAME=mcav\n",
    "anvi-db-info working/mcav_assembly_redo/$SAMPLENAME.contigs.db"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "2217fca1-f5c2-496d-9e1c-9781ac49577b",
   "metadata": {},
   "outputs": [],
   "source": [
    "DB Info (no touch)\n",
    "===============================================\n",
    "Database Path ................................: working/mcav_assembly_redo/mcav.contigs.db\n",
    "description ..................................: [Not found, but it's OK]\n",
    "db_type ......................................: contigs (variant: unknown)\n",
    "version ......................................: 21\n",
    "\n",
    "\n",
    "DB Info (no touch also)\n",
    "===============================================\n",
    "project_name .................................: mcav\n",
    "contigs_db_hash ..............................: hash2fac8c34\n",
    "split_length .................................: 20000\n",
    "kmer_size ....................................: 4\n",
    "num_contigs ..................................: 839850\n",
    "total_length .................................: 435829081\n",
    "num_splits ...................................: 839850\n",
    "gene_level_taxonomy_source ...................: None\n",
    "genes_are_called .............................: 1\n",
    "external_gene_calls ..........................: 0\n",
    "external_gene_amino_acid_seqs ................: 0\n",
    "skip_predict_frame ...........................: 0\n",
    "splits_consider_gene_calls ...................: 1\n",
    "scg_taxonomy_was_run .........................: 0\n",
    "scg_taxonomy_database_version ................: None\n",
    "trna_taxonomy_was_run ........................: 0\n",
    "trna_taxonomy_database_version ...............: None\n",
    "creation_date ................................: 1701991494.31667\n",
    "gene_function_sources ........................: KEGG_Class,KEGG_Module,COG20_CATEGORY,COG20_FUNCTION,COG20_PATHWAY,KOfam,KEGG_BRITE\n",
    "modules_db_hash ..............................: a2b5bde358bb\n",
    "\n",
    "* Please remember that it is never a good idea to change these values. But in some\n",
    "  cases it may be absolutely necessary to update something here, and a\n",
    "  programmer may ask you to run this program and do it. But even then, you\n",
    "  should be extremely careful.\n",
    "\n",
    "\n",
    "AVAILABLE GENE CALLERS\n",
    "===============================================\n",
    "* 'prodigal' (707,308 gene calls)\n",
    "* 'Ribosomal_RNA_28S' (2 gene calls)\n",
    "* 'Ribosomal_RNA_18S' (2 gene calls)\n",
    "\n",
    "\n",
    "AVAILABLE FUNCTIONAL ANNOTATION SOURCES\n",
    "===============================================\n",
    "* COG20_CATEGORY (4,796 annotations)\n",
    "* COG20_FUNCTION (4,796 annotations)\n",
    "* COG20_PATHWAY (1,268 annotations)\n",
    "* KEGG_BRITE (2,346 annotations)\n",
    "* KEGG_Class (217 annotations)\n",
    "* KEGG_Module (217 annotations)\n",
    "* KOfam (2,362 annotations)\n",
    "\n",
    "\n",
    "AVAILABLE HMM SOURCES\n",
    "===============================================\n",
    "* 'Archaea_76' (76 models with 71 hits)\n",
    "* 'Bacteria_71' (71 models with 62 hits)\n",
    "* 'Protista_83' (83 models with 14 hits)\n",
    "* 'Ribosomal_RNA_12S' (1 model with 0 hits)\n",
    "* 'Ribosomal_RNA_16S' (3 models with 0 hits)\n",
    "* 'Ribosomal_RNA_18S' (1 model with 2 hits)\n",
    "* 'Ribosomal_RNA_23S' (2 models with 0 hits)\n",
    "* 'Ribosomal_RNA_28S' (1 model with 2 hits)\n",
    "* 'Ribosomal_RNA_5S' (5 models with 0 hits)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "453258e7-30f0-4717-a551-33ee34db1762",
   "metadata": {},
   "outputs": [],
   "source": [
    "OUTDIR=//project/pi_sarah_gignouxwolfsohn_uml_edu/brooke/working/mcav_assembly_redo/profiles\n",
    "TAXPATH=/project/pi_sarah_gignouxwolfsohn_uml_edu/brooke/working/mcav_assembly_redo/taxonomy\n",
    "mkdir $TAXPATH\n",
    "anvi-estimate-scg-taxonomy -c $DBPATH/$SAMPLENAME.contigs.db -p $OUTDIR/\"$SAMPLENAME\"_profile_merged/PROFILE.db -T 3 --metagenome-mode --compute-scg-coverages -o $TAXPATH/\"$SAMPLENAME\"-scg-taxonomy.tsv\n",
    "#taxa matching from assembled metagenome - will just match ASVs to SCG taxa and give percent identity \n",
    "anvi-run-scg-taxonomy -c $DBPATH/\"$SAMPLENAME\".contigs.db"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "a6c6785b-e1c8-4e35-904e-cf23715863fb",
   "metadata": {},
   "outputs": [],
   "source": [
    "#estimate metabolism using KEGGKOFOAMS annotation of db\n",
    "anvi-estimate-metabolism -c $DBPATH/$SAMPLENAME.contigs.db --metagenome-mode"
   ]
  }
 ],
 "metadata": {
  "kernelspec": {
   "display_name": "Python 3 (ipykernel)",
   "language": "python",
   "name": "python3"
  },
  "language_info": {
   "codemirror_mode": {
    "name": "ipython",
    "version": 3
   },
   "file_extension": ".py",
   "mimetype": "text/x-python",
   "name": "python",
   "nbconvert_exporter": "python",
   "pygments_lexer": "ipython3",
   "version": "3.11.0"
  }
 },
 "nbformat": 4,
 "nbformat_minor": 5
}
