{
 "cells": [
  {
   "cell_type": "markdown",
   "id": "2544cefe-41ce-4b50-a576-15893d2a718a",
   "metadata": {},
   "source": [
    "# Investigating MAG quality\n",
    "\n",
    "## Healthy mcav "
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 1,
   "id": "8d501023-ab95-405c-9536-0ba9acac4ecb",
   "metadata": {},
   "outputs": [],
   "source": [
    "import numpy as np\n",
    "import pandas as pd\n",
    "import os"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 2,
   "id": "f5022fa4-587b-45ca-ba9e-cb7636ac15e5",
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Concoct_healthy_2019_mcav_bins\tmetabat\n",
      "MetaBAT_healthy_2019_mcav_bins\tslurm-21569803-concoct.out\n",
      "concoct\t\t\t\tslurm-21582072-das_tool.out\n",
      "das_tool\t\t\tslurm-metabat-19497287.out\n",
      "das_tool_DASTool_bins\n"
     ]
    }
   ],
   "source": [
    "!ls //project/pi_sarah_gignouxwolfsohn_uml_edu/brooke/mcav/healthy_2019_mcav/binning/"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 3,
   "id": "bb4ae2ee-a02d-4785-a83f-3ccd755464e8",
   "metadata": {},
   "outputs": [],
   "source": [
    "os.chdir('//project/pi_sarah_gignouxwolfsohn_uml_edu/brooke/mcav/healthy_2019_mcav/binning/Concoct_healthy_2019_mcav_bins/CheckM')\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 4,
   "id": "69cbbd1f-9239-4698-b7a2-d87eeb8f8860",
   "metadata": {},
   "outputs": [],
   "source": [
    "binpath='//project/pi_sarah_gignouxwolfsohn_uml_edu/brooke/mcav/healthy_2019_mcav/binning/Concoct_healthy_2019_mcav_bins/CheckM'\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 5,
   "id": "c6f9ff89-43b7-42e9-bd80-cea86de23430",
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "\u001b[0m\u001b[01;34mbins\u001b[0m/  bins_stats.tsv  checkm.log  lineage.ms  \u001b[01;34mstorage\u001b[0m/\n"
     ]
    }
   ],
   "source": [
    "ls"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 6,
   "id": "4439395b-ae93-4c5c-a4ff-44c449061b94",
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "-------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------\n",
      "  Bin Id                                   Marker lineage            # genomes   # markers   # marker sets    0     1    2    3    4   5+   Completeness   Contamination   Strain heterogeneity  \n",
      "-------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------\n",
      "  healthy_2019_mcav_contigs_cut             root (UID1)                 5656         56            24         0     0    1    2    5   48      100.00          610.87              1.19          \n",
      "  42                               o__Sphingomonadales (UID3310)         26         569           293         3    556   10   0    0   0       99.59            2.30               0.00          \n",
      "  56                              c__Gammaproteobacteria (UID4444)      263         507           232         45   450   12   0    0   0       85.79            2.68              33.33          \n",
      "  109                                  k__Bacteria (UID3187)            2258        188           117         53   132   3    0    0   0       68.90            1.12               0.00          \n",
      "  68                                     k__Archaea (UID2)              207         149           107         61    62   18   7    1   0       53.82           13.76               0.00          \n",
      "  139                                p__Cyanobacteria (UID2143)         129         472           368        242   220   8    2    0   0       46.70            1.84              14.29          \n",
      "  50                                    k__Bacteria (UID203)            5449        103            58         39    30   8    11   5   10      41.74           32.07               6.25          \n"
     ]
    }
   ],
   "source": [
    "!head bins_stats.tsv"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 7,
   "id": "0d7a43b9-c58f-4b3c-8ec6-f8231a0d525d",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>0</th>\n",
       "      <th>1</th>\n",
       "      <th>2</th>\n",
       "      <th>3</th>\n",
       "      <th>4</th>\n",
       "      <th>5</th>\n",
       "      <th>6</th>\n",
       "      <th>7</th>\n",
       "      <th>8</th>\n",
       "      <th>9</th>\n",
       "      <th>10</th>\n",
       "      <th>11</th>\n",
       "      <th>12</th>\n",
       "      <th>13</th>\n",
       "      <th>14</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>42</td>\n",
       "      <td>o__Sphingomonadales</td>\n",
       "      <td>(UID3310)</td>\n",
       "      <td>26</td>\n",
       "      <td>569</td>\n",
       "      <td>293</td>\n",
       "      <td>3</td>\n",
       "      <td>556</td>\n",
       "      <td>10</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>99.59</td>\n",
       "      <td>2.30</td>\n",
       "      <td>0.00</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>56</td>\n",
       "      <td>c__Gammaproteobacteria</td>\n",
       "      <td>(UID4444)</td>\n",
       "      <td>263</td>\n",
       "      <td>507</td>\n",
       "      <td>232</td>\n",
       "      <td>45</td>\n",
       "      <td>450</td>\n",
       "      <td>12</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>85.79</td>\n",
       "      <td>2.68</td>\n",
       "      <td>33.33</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>109</td>\n",
       "      <td>k__Bacteria</td>\n",
       "      <td>(UID3187)</td>\n",
       "      <td>2258</td>\n",
       "      <td>188</td>\n",
       "      <td>117</td>\n",
       "      <td>53</td>\n",
       "      <td>132</td>\n",
       "      <td>3</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>68.90</td>\n",
       "      <td>1.12</td>\n",
       "      <td>0.00</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>68</td>\n",
       "      <td>k__Archaea</td>\n",
       "      <td>(UID2)</td>\n",
       "      <td>207</td>\n",
       "      <td>149</td>\n",
       "      <td>107</td>\n",
       "      <td>61</td>\n",
       "      <td>62</td>\n",
       "      <td>18</td>\n",
       "      <td>7</td>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "      <td>53.82</td>\n",
       "      <td>13.76</td>\n",
       "      <td>0.00</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>139</td>\n",
       "      <td>p__Cyanobacteria</td>\n",
       "      <td>(UID2143)</td>\n",
       "      <td>129</td>\n",
       "      <td>472</td>\n",
       "      <td>368</td>\n",
       "      <td>242</td>\n",
       "      <td>220</td>\n",
       "      <td>8</td>\n",
       "      <td>2</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>46.70</td>\n",
       "      <td>1.84</td>\n",
       "      <td>14.29</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>184</th>\n",
       "      <td>101</td>\n",
       "      <td>root</td>\n",
       "      <td>(UID1)</td>\n",
       "      <td>5656</td>\n",
       "      <td>56</td>\n",
       "      <td>24</td>\n",
       "      <td>56</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0.00</td>\n",
       "      <td>0.00</td>\n",
       "      <td>0.00</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>185</th>\n",
       "      <td>100</td>\n",
       "      <td>root</td>\n",
       "      <td>(UID1)</td>\n",
       "      <td>5656</td>\n",
       "      <td>56</td>\n",
       "      <td>24</td>\n",
       "      <td>56</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0.00</td>\n",
       "      <td>0.00</td>\n",
       "      <td>0.00</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>186</th>\n",
       "      <td>10</td>\n",
       "      <td>root</td>\n",
       "      <td>(UID1)</td>\n",
       "      <td>5656</td>\n",
       "      <td>56</td>\n",
       "      <td>24</td>\n",
       "      <td>56</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0.00</td>\n",
       "      <td>0.00</td>\n",
       "      <td>0.00</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>187</th>\n",
       "      <td>1</td>\n",
       "      <td>root</td>\n",
       "      <td>(UID1)</td>\n",
       "      <td>5656</td>\n",
       "      <td>56</td>\n",
       "      <td>24</td>\n",
       "      <td>56</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0.00</td>\n",
       "      <td>0.00</td>\n",
       "      <td>0.00</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>188</th>\n",
       "      <td>0</td>\n",
       "      <td>root</td>\n",
       "      <td>(UID1)</td>\n",
       "      <td>5656</td>\n",
       "      <td>56</td>\n",
       "      <td>24</td>\n",
       "      <td>56</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0.00</td>\n",
       "      <td>0.00</td>\n",
       "      <td>0.00</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>189 rows × 15 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "      0                       1          2     3    4    5    6    7   8   9   \\\n",
       "0     42     o__Sphingomonadales  (UID3310)    26  569  293    3  556  10   0   \n",
       "1     56  c__Gammaproteobacteria  (UID4444)   263  507  232   45  450  12   0   \n",
       "2    109             k__Bacteria  (UID3187)  2258  188  117   53  132   3   0   \n",
       "3     68              k__Archaea     (UID2)   207  149  107   61   62  18   7   \n",
       "4    139        p__Cyanobacteria  (UID2143)   129  472  368  242  220   8   2   \n",
       "..   ...                     ...        ...   ...  ...  ...  ...  ...  ..  ..   \n",
       "184  101                    root     (UID1)  5656   56   24   56    0   0   0   \n",
       "185  100                    root     (UID1)  5656   56   24   56    0   0   0   \n",
       "186   10                    root     (UID1)  5656   56   24   56    0   0   0   \n",
       "187    1                    root     (UID1)  5656   56   24   56    0   0   0   \n",
       "188    0                    root     (UID1)  5656   56   24   56    0   0   0   \n",
       "\n",
       "     10  11     12     13     14  \n",
       "0     0   0  99.59   2.30   0.00  \n",
       "1     0   0  85.79   2.68  33.33  \n",
       "2     0   0  68.90   1.12   0.00  \n",
       "3     1   0  53.82  13.76   0.00  \n",
       "4     0   0  46.70   1.84  14.29  \n",
       "..   ..  ..    ...    ...    ...  \n",
       "184   0   0   0.00   0.00   0.00  \n",
       "185   0   0   0.00   0.00   0.00  \n",
       "186   0   0   0.00   0.00   0.00  \n",
       "187   0   0   0.00   0.00   0.00  \n",
       "188   0   0   0.00   0.00   0.00  \n",
       "\n",
       "[189 rows x 15 columns]"
      ]
     },
     "execution_count": 7,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "# load bin stats\n",
    "concoct_bins=pd.read_csv('bins_stats.tsv',skiprows=4,delim_whitespace=True,header=None,skipfooter=1,engine='python')\n",
    "concoct_bins"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 8,
   "id": "fe4588f6-018b-46d1-b76f-2606ebdafa2b",
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "15\n",
      "(189, 15)\n"
     ]
    },
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead tr th {\n",
       "        text-align: left;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr>\n",
       "      <th></th>\n",
       "      <th>Bin_Id</th>\n",
       "      <th>Marker_lineage</th>\n",
       "      <th>UID</th>\n",
       "      <th>#genomes</th>\n",
       "      <th>#markers</th>\n",
       "      <th>#marker_sets</th>\n",
       "      <th>0</th>\n",
       "      <th>1</th>\n",
       "      <th>2</th>\n",
       "      <th>3</th>\n",
       "      <th>4</th>\n",
       "      <th>5+</th>\n",
       "      <th>Completeness</th>\n",
       "      <th>Contamination</th>\n",
       "      <th>Strain_heterogeneity</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>42</td>\n",
       "      <td>o__Sphingomonadales</td>\n",
       "      <td>(UID3310)</td>\n",
       "      <td>26</td>\n",
       "      <td>569</td>\n",
       "      <td>293</td>\n",
       "      <td>3</td>\n",
       "      <td>556</td>\n",
       "      <td>10</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>99.59</td>\n",
       "      <td>2.30</td>\n",
       "      <td>0.00</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>56</td>\n",
       "      <td>c__Gammaproteobacteria</td>\n",
       "      <td>(UID4444)</td>\n",
       "      <td>263</td>\n",
       "      <td>507</td>\n",
       "      <td>232</td>\n",
       "      <td>45</td>\n",
       "      <td>450</td>\n",
       "      <td>12</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>85.79</td>\n",
       "      <td>2.68</td>\n",
       "      <td>33.33</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>109</td>\n",
       "      <td>k__Bacteria</td>\n",
       "      <td>(UID3187)</td>\n",
       "      <td>2258</td>\n",
       "      <td>188</td>\n",
       "      <td>117</td>\n",
       "      <td>53</td>\n",
       "      <td>132</td>\n",
       "      <td>3</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>68.90</td>\n",
       "      <td>1.12</td>\n",
       "      <td>0.00</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>68</td>\n",
       "      <td>k__Archaea</td>\n",
       "      <td>(UID2)</td>\n",
       "      <td>207</td>\n",
       "      <td>149</td>\n",
       "      <td>107</td>\n",
       "      <td>61</td>\n",
       "      <td>62</td>\n",
       "      <td>18</td>\n",
       "      <td>7</td>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "      <td>53.82</td>\n",
       "      <td>13.76</td>\n",
       "      <td>0.00</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>139</td>\n",
       "      <td>p__Cyanobacteria</td>\n",
       "      <td>(UID2143)</td>\n",
       "      <td>129</td>\n",
       "      <td>472</td>\n",
       "      <td>368</td>\n",
       "      <td>242</td>\n",
       "      <td>220</td>\n",
       "      <td>8</td>\n",
       "      <td>2</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>46.70</td>\n",
       "      <td>1.84</td>\n",
       "      <td>14.29</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>184</th>\n",
       "      <td>101</td>\n",
       "      <td>root</td>\n",
       "      <td>(UID1)</td>\n",
       "      <td>5656</td>\n",
       "      <td>56</td>\n",
       "      <td>24</td>\n",
       "      <td>56</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0.00</td>\n",
       "      <td>0.00</td>\n",
       "      <td>0.00</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>185</th>\n",
       "      <td>100</td>\n",
       "      <td>root</td>\n",
       "      <td>(UID1)</td>\n",
       "      <td>5656</td>\n",
       "      <td>56</td>\n",
       "      <td>24</td>\n",
       "      <td>56</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0.00</td>\n",
       "      <td>0.00</td>\n",
       "      <td>0.00</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>186</th>\n",
       "      <td>10</td>\n",
       "      <td>root</td>\n",
       "      <td>(UID1)</td>\n",
       "      <td>5656</td>\n",
       "      <td>56</td>\n",
       "      <td>24</td>\n",
       "      <td>56</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0.00</td>\n",
       "      <td>0.00</td>\n",
       "      <td>0.00</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>187</th>\n",
       "      <td>1</td>\n",
       "      <td>root</td>\n",
       "      <td>(UID1)</td>\n",
       "      <td>5656</td>\n",
       "      <td>56</td>\n",
       "      <td>24</td>\n",
       "      <td>56</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0.00</td>\n",
       "      <td>0.00</td>\n",
       "      <td>0.00</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>188</th>\n",
       "      <td>0</td>\n",
       "      <td>root</td>\n",
       "      <td>(UID1)</td>\n",
       "      <td>5656</td>\n",
       "      <td>56</td>\n",
       "      <td>24</td>\n",
       "      <td>56</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0.00</td>\n",
       "      <td>0.00</td>\n",
       "      <td>0.00</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>189 rows × 15 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "    Bin_Id          Marker_lineage        UID #genomes #markers #marker_sets  \\\n",
       "0       42     o__Sphingomonadales  (UID3310)       26      569          293   \n",
       "1       56  c__Gammaproteobacteria  (UID4444)      263      507          232   \n",
       "2      109             k__Bacteria  (UID3187)     2258      188          117   \n",
       "3       68              k__Archaea     (UID2)      207      149          107   \n",
       "4      139        p__Cyanobacteria  (UID2143)      129      472          368   \n",
       "..     ...                     ...        ...      ...      ...          ...   \n",
       "184    101                    root     (UID1)     5656       56           24   \n",
       "185    100                    root     (UID1)     5656       56           24   \n",
       "186     10                    root     (UID1)     5656       56           24   \n",
       "187      1                    root     (UID1)     5656       56           24   \n",
       "188      0                    root     (UID1)     5656       56           24   \n",
       "\n",
       "       0    1   2  3  4 5+ Completeness Contamination Strain_heterogeneity  \n",
       "0      3  556  10  0  0  0        99.59          2.30                 0.00  \n",
       "1     45  450  12  0  0  0        85.79          2.68                33.33  \n",
       "2     53  132   3  0  0  0        68.90          1.12                 0.00  \n",
       "3     61   62  18  7  1  0        53.82         13.76                 0.00  \n",
       "4    242  220   8  2  0  0        46.70          1.84                14.29  \n",
       "..   ...  ...  .. .. .. ..          ...           ...                  ...  \n",
       "184   56    0   0  0  0  0         0.00          0.00                 0.00  \n",
       "185   56    0   0  0  0  0         0.00          0.00                 0.00  \n",
       "186   56    0   0  0  0  0         0.00          0.00                 0.00  \n",
       "187   56    0   0  0  0  0         0.00          0.00                 0.00  \n",
       "188   56    0   0  0  0  0         0.00          0.00                 0.00  \n",
       "\n",
       "[189 rows x 15 columns]"
      ]
     },
     "execution_count": 8,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "# add column names manually \n",
    "header=['Bin_Id','Marker_lineage','UID','#genomes','#markers','#marker_sets','0','1','2','3','4','5+','Completeness','Contamination','Strain_heterogeneity']\n",
    "concoct_bins.columns=[header]     \n",
    "print(len(header))\n",
    "print(concoct_bins.shape)\n",
    "concoct_bins"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 9,
   "id": "2677dc68-f531-4918-a3e3-5aa54e4ee4f2",
   "metadata": {},
   "outputs": [],
   "source": [
    "# load header '\\s{3}'\n",
    "#concoct_header=pd.read_csv('bins_stats.tsv',nrows=1,sep='\\s{3}',engine='python',na_filter=False,usecols=lambda column: column != 0)\n",
    "#concoct_header=pd.read_csv('bins_stats.tsv', index_col=0,skiprows=lambda x: x in [0, 2],delim_whitespace=True)\n",
    "#concoct_header"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 10,
   "id": "1f4a6097-8665-4be4-99c2-a3cdb6e37c6d",
   "metadata": {},
   "outputs": [],
   "source": [
    "# want to add ./storage/bin_stats.analyze.tsv to add genome size or i think bin_stats_ext already has both"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 11,
   "id": "c464d25b-94fe-44d2-b6d6-e8ef60ffba27",
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "10\t{'GC': 0.45383397689109817, 'GC std': 0.017741057662292045, 'Genome size': 67593, '# ambiguous bases': 0, '# scaffolds': 49, '# contigs': 49, 'Longest scaffold': 2600, 'Longest contig': 2600, 'N50 (scaffolds)': 1396, 'N50 (contigs)': 1396, 'Mean scaffold length': 1379.4489795918366, 'Mean contig length': 1379.4489795918366, 'Coding density': 0.4126018966461024, 'Translation table': 4, '# predicted genes': 60}\n",
      "1\t{'GC': 0.47662485462908316, 'GC std': 0.01813517400388053, 'Genome size': 401559, '# ambiguous bases': 0, '# scaffolds': 177, '# contigs': 177, 'Longest scaffold': 8573, 'Longest contig': 8573, 'N50 (scaffolds)': 2459, 'N50 (contigs)': 2459, 'Mean scaffold length': 2268.6949152542375, 'Mean contig length': 2268.6949152542375, 'Coding density': 0.8767927004499961, 'Translation table': 11, '# predicted genes': 538}\n",
      "101\t{'GC': 0.4231691853348191, 'GC std': 0.03386657292470408, 'Genome size': 182173, '# ambiguous bases': 0, '# scaffolds': 128, '# contigs': 128, 'Longest scaffold': 2808, 'Longest contig': 2808, 'N50 (scaffolds)': 1419, 'N50 (contigs)': 1419, 'Mean scaffold length': 1423.2265625, 'Mean contig length': 1423.2265625, 'Coding density': 0.49618220043584943, 'Translation table': 11, '# predicted genes': 339}\n",
      "102\t{'GC': 0.4045708465788093, 'GC std': 0.04412584721709139, 'Genome size': 194406, '# ambiguous bases': 0, '# scaffolds': 135, '# contigs': 135, 'Longest scaffold': 3275, 'Longest contig': 3275, 'N50 (scaffolds)': 1367, 'N50 (contigs)': 1367, 'Mean scaffold length': 1440.0444444444445, 'Mean contig length': 1440.0444444444445, 'Coding density': 0.7444266123473555, 'Translation table': 4, '# predicted genes': 365}\n",
      "100\t{'GC': 0.33128435782108945, 'GC std': 0.052705820586636025, 'Genome size': 80040, '# ambiguous bases': 0, '# scaffolds': 45, '# contigs': 45, 'Longest scaffold': 18131, 'Longest contig': 18131, 'N50 (scaffolds)': 1540, 'N50 (contigs)': 1540, 'Mean scaffold length': 1778.6666666666667, 'Mean contig length': 1778.6666666666667, 'Coding density': 0.6600949525237382, 'Translation table': 11, '# predicted genes': 115}\n",
      "104\t{'GC': 0.37397319498486814, 'GC std': 0.00484686845007604, 'Genome size': 2313, '# ambiguous bases': 0, '# scaffolds': 2, '# contigs': 2, 'Longest scaffold': 1191, 'Longest contig': 1191, 'N50 (scaffolds)': 1191, 'N50 (contigs)': 1191, 'Mean scaffold length': 1156.5, 'Mean contig length': 1156.5, 'Coding density': 0.5123216601815823, 'Translation table': 11, '# predicted genes': 4}\n",
      "103\t{'GC': 0.44307629162177414, 'GC std': 0.03908526949718335, 'Genome size': 19491, '# ambiguous bases': 0, '# scaffolds': 13, '# contigs': 13, 'Longest scaffold': 3774, 'Longest contig': 3774, 'N50 (scaffolds)': 1333, 'N50 (contigs)': 1333, 'Mean scaffold length': 1499.3076923076924, 'Mean contig length': 1499.3076923076924, 'Coding density': 0.4665230106202863, 'Translation table': 11, '# predicted genes': 26}\n",
      "106\t{'GC': 0.3993453355155483, 'GC std': 0.020422714801198213, 'Genome size': 5499, '# ambiguous bases': 0, '# scaffolds': 2, '# contigs': 2, 'Longest scaffold': 4048, 'Longest contig': 4048, 'N50 (scaffolds)': 4048, 'N50 (contigs)': 4048, 'Mean scaffold length': 2749.5, 'Mean contig length': 2749.5, 'Coding density': 0.5804691762138571, 'Translation table': 11, '# predicted genes': 6}\n",
      "107\t{'GC': 0.49224982984008475, 'GC std': 0.04015873709700588, 'Genome size': 427539, '# ambiguous bases': 0, '# scaffolds': 182, '# contigs': 182, 'Longest scaffold': 17381, 'Longest contig': 17381, 'N50 (scaffolds)': 2667, 'N50 (contigs)': 2667, 'Mean scaffold length': 2349.1153846153848, 'Mean contig length': 2349.1153846153848, 'Coding density': 0.5416301202931194, 'Translation table': 11, '# predicted genes': 708}\n",
      "105\t{'GC': 0.39102510624515396, 'GC std': 0.029902625530478594, 'Genome size': 2445288, '# ambiguous bases': 0, '# scaffolds': 1613, '# contigs': 1613, 'Longest scaffold': 9343, 'Longest contig': 9343, 'N50 (scaffolds)': 1438, 'N50 (contigs)': 1438, 'Mean scaffold length': 1515.9876007439555, 'Mean contig length': 1515.9876007439555, 'Coding density': 0.4201439666820432, 'Translation table': 11, '# predicted genes': 3863}\n"
     ]
    }
   ],
   "source": [
    "!head storage/bin_stats.analyze.tsv"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 12,
   "id": "51a82cc9-4856-4fdf-9104-14718ce44b6d",
   "metadata": {},
   "outputs": [],
   "source": [
    "# read in header of table with genome size \n",
    "#conc_bin_stats_firstcols=pd.read_csv('storage/bin_stats.analyze.tsv',header=None,sep='\\t',usecols=[0])\n",
    "#conc_bin_stats_firstcols.columns=['Bin_Id']"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 13,
   "id": "9b10f959-9ac9-47ee-9a18-9ea78cd2e0ed",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>1</th>\n",
       "      <th>2</th>\n",
       "      <th>3</th>\n",
       "      <th>4</th>\n",
       "      <th>5</th>\n",
       "      <th>6</th>\n",
       "      <th>7</th>\n",
       "      <th>8</th>\n",
       "      <th>9</th>\n",
       "      <th>10</th>\n",
       "      <th>11</th>\n",
       "      <th>12</th>\n",
       "      <th>13</th>\n",
       "      <th>14</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>10\\t{'GC': 0.45383397689109817</td>\n",
       "      <td>'GC std': 0.017741057662292045</td>\n",
       "      <td>'Genome size': 67593</td>\n",
       "      <td>'# ambiguous bases': 0</td>\n",
       "      <td>'# scaffolds': 49</td>\n",
       "      <td>'# contigs': 49</td>\n",
       "      <td>'Longest scaffold': 2600</td>\n",
       "      <td>'Longest contig': 2600</td>\n",
       "      <td>'N50 (scaffolds)': 1396</td>\n",
       "      <td>'N50 (contigs)': 1396</td>\n",
       "      <td>'Mean scaffold length': 1379.4489795918366</td>\n",
       "      <td>'Mean contig length': 1379.4489795918366</td>\n",
       "      <td>'Coding density': 0.4126018966461024</td>\n",
       "      <td>'Translation table': 4</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>1\\t{'GC': 0.47662485462908316</td>\n",
       "      <td>'GC std': 0.01813517400388053</td>\n",
       "      <td>'Genome size': 401559</td>\n",
       "      <td>'# ambiguous bases': 0</td>\n",
       "      <td>'# scaffolds': 177</td>\n",
       "      <td>'# contigs': 177</td>\n",
       "      <td>'Longest scaffold': 8573</td>\n",
       "      <td>'Longest contig': 8573</td>\n",
       "      <td>'N50 (scaffolds)': 2459</td>\n",
       "      <td>'N50 (contigs)': 2459</td>\n",
       "      <td>'Mean scaffold length': 2268.6949152542375</td>\n",
       "      <td>'Mean contig length': 2268.6949152542375</td>\n",
       "      <td>'Coding density': 0.8767927004499961</td>\n",
       "      <td>'Translation table': 11</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>101\\t{'GC': 0.4231691853348191</td>\n",
       "      <td>'GC std': 0.03386657292470408</td>\n",
       "      <td>'Genome size': 182173</td>\n",
       "      <td>'# ambiguous bases': 0</td>\n",
       "      <td>'# scaffolds': 128</td>\n",
       "      <td>'# contigs': 128</td>\n",
       "      <td>'Longest scaffold': 2808</td>\n",
       "      <td>'Longest contig': 2808</td>\n",
       "      <td>'N50 (scaffolds)': 1419</td>\n",
       "      <td>'N50 (contigs)': 1419</td>\n",
       "      <td>'Mean scaffold length': 1423.2265625</td>\n",
       "      <td>'Mean contig length': 1423.2265625</td>\n",
       "      <td>'Coding density': 0.49618220043584943</td>\n",
       "      <td>'Translation table': 11</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>102\\t{'GC': 0.4045708465788093</td>\n",
       "      <td>'GC std': 0.04412584721709139</td>\n",
       "      <td>'Genome size': 194406</td>\n",
       "      <td>'# ambiguous bases': 0</td>\n",
       "      <td>'# scaffolds': 135</td>\n",
       "      <td>'# contigs': 135</td>\n",
       "      <td>'Longest scaffold': 3275</td>\n",
       "      <td>'Longest contig': 3275</td>\n",
       "      <td>'N50 (scaffolds)': 1367</td>\n",
       "      <td>'N50 (contigs)': 1367</td>\n",
       "      <td>'Mean scaffold length': 1440.0444444444445</td>\n",
       "      <td>'Mean contig length': 1440.0444444444445</td>\n",
       "      <td>'Coding density': 0.7444266123473555</td>\n",
       "      <td>'Translation table': 4</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>100\\t{'GC': 0.33128435782108945</td>\n",
       "      <td>'GC std': 0.052705820586636025</td>\n",
       "      <td>'Genome size': 80040</td>\n",
       "      <td>'# ambiguous bases': 0</td>\n",
       "      <td>'# scaffolds': 45</td>\n",
       "      <td>'# contigs': 45</td>\n",
       "      <td>'Longest scaffold': 18131</td>\n",
       "      <td>'Longest contig': 18131</td>\n",
       "      <td>'N50 (scaffolds)': 1540</td>\n",
       "      <td>'N50 (contigs)': 1540</td>\n",
       "      <td>'Mean scaffold length': 1778.6666666666667</td>\n",
       "      <td>'Mean contig length': 1778.6666666666667</td>\n",
       "      <td>'Coding density': 0.6600949525237382</td>\n",
       "      <td>'Translation table': 11</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "                                1                                2   \\\n",
       "0   10\\t{'GC': 0.45383397689109817   'GC std': 0.017741057662292045   \n",
       "1    1\\t{'GC': 0.47662485462908316    'GC std': 0.01813517400388053   \n",
       "2   101\\t{'GC': 0.4231691853348191    'GC std': 0.03386657292470408   \n",
       "3   102\\t{'GC': 0.4045708465788093    'GC std': 0.04412584721709139   \n",
       "4  100\\t{'GC': 0.33128435782108945   'GC std': 0.052705820586636025   \n",
       "\n",
       "                       3                        4                    5   \\\n",
       "0    'Genome size': 67593   '# ambiguous bases': 0    '# scaffolds': 49   \n",
       "1   'Genome size': 401559   '# ambiguous bases': 0   '# scaffolds': 177   \n",
       "2   'Genome size': 182173   '# ambiguous bases': 0   '# scaffolds': 128   \n",
       "3   'Genome size': 194406   '# ambiguous bases': 0   '# scaffolds': 135   \n",
       "4    'Genome size': 80040   '# ambiguous bases': 0    '# scaffolds': 45   \n",
       "\n",
       "                  6                           7                         8   \\\n",
       "0    '# contigs': 49    'Longest scaffold': 2600    'Longest contig': 2600   \n",
       "1   '# contigs': 177    'Longest scaffold': 8573    'Longest contig': 8573   \n",
       "2   '# contigs': 128    'Longest scaffold': 2808    'Longest contig': 2808   \n",
       "3   '# contigs': 135    'Longest scaffold': 3275    'Longest contig': 3275   \n",
       "4    '# contigs': 45   'Longest scaffold': 18131   'Longest contig': 18131   \n",
       "\n",
       "                         9                       10  \\\n",
       "0   'N50 (scaffolds)': 1396   'N50 (contigs)': 1396   \n",
       "1   'N50 (scaffolds)': 2459   'N50 (contigs)': 2459   \n",
       "2   'N50 (scaffolds)': 1419   'N50 (contigs)': 1419   \n",
       "3   'N50 (scaffolds)': 1367   'N50 (contigs)': 1367   \n",
       "4   'N50 (scaffolds)': 1540   'N50 (contigs)': 1540   \n",
       "\n",
       "                                            11  \\\n",
       "0   'Mean scaffold length': 1379.4489795918366   \n",
       "1   'Mean scaffold length': 2268.6949152542375   \n",
       "2         'Mean scaffold length': 1423.2265625   \n",
       "3   'Mean scaffold length': 1440.0444444444445   \n",
       "4   'Mean scaffold length': 1778.6666666666667   \n",
       "\n",
       "                                          12  \\\n",
       "0   'Mean contig length': 1379.4489795918366   \n",
       "1   'Mean contig length': 2268.6949152542375   \n",
       "2         'Mean contig length': 1423.2265625   \n",
       "3   'Mean contig length': 1440.0444444444445   \n",
       "4   'Mean contig length': 1778.6666666666667   \n",
       "\n",
       "                                       13                        14  \n",
       "0    'Coding density': 0.4126018966461024    'Translation table': 4  \n",
       "1    'Coding density': 0.8767927004499961   'Translation table': 11  \n",
       "2   'Coding density': 0.49618220043584943   'Translation table': 11  \n",
       "3    'Coding density': 0.7444266123473555    'Translation table': 4  \n",
       "4    'Coding density': 0.6600949525237382   'Translation table': 11  "
      ]
     },
     "execution_count": 13,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "# read in table with genome size\n",
    "conc_bin_stats=pd.read_csv('storage/bin_stats.analyze.tsv',header=None,usecols=lambda x: x != 0)\n",
    "conc_bin_stats.head(5)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 14,
   "id": "776c65a7-5acf-4e5c-85cb-b0a747b3f6b9",
   "metadata": {},
   "outputs": [],
   "source": [
    "# split first column that contains bin id \n",
    "conc_bin_stats[['Bin_Id',1]]=conc_bin_stats[1].str.split('\\t{',expand=True)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 81,
   "id": "0ce8428e-9d44-4772-98a3-e110a7a943e6",
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "190\n"
     ]
    },
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>1</th>\n",
       "      <th>2</th>\n",
       "      <th>3</th>\n",
       "      <th>4</th>\n",
       "      <th>5</th>\n",
       "      <th>6</th>\n",
       "      <th>7</th>\n",
       "      <th>8</th>\n",
       "      <th>9</th>\n",
       "      <th>10</th>\n",
       "      <th>11</th>\n",
       "      <th>12</th>\n",
       "      <th>13</th>\n",
       "      <th>14</th>\n",
       "      <th>Bin_Id</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>187</th>\n",
       "      <td>'GC': 0.4343907050953209</td>\n",
       "      <td>'GC std': 0.03418430197601377</td>\n",
       "      <td>'Genome size': 128714994</td>\n",
       "      <td>'# ambiguous bases': 0</td>\n",
       "      <td>'# scaffolds': 67929</td>\n",
       "      <td>'# contigs': 67929</td>\n",
       "      <td>'Longest scaffold': 8240</td>\n",
       "      <td>'Longest contig': 8240</td>\n",
       "      <td>'N50 (scaffolds)': 2022</td>\n",
       "      <td>'N50 (contigs)': 2022</td>\n",
       "      <td>'Mean scaffold length': 1894.8460009716027</td>\n",
       "      <td>'Mean contig length': 1894.8460009716027</td>\n",
       "      <td>'Coding density': 0.5431238958842666</td>\n",
       "      <td>'Translation table': 11</td>\n",
       "      <td>84</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>188</th>\n",
       "      <td>'GC': 0.39014272558076646</td>\n",
       "      <td>'GC std': 0.027089530386056247</td>\n",
       "      <td>'Genome size': 197017240</td>\n",
       "      <td>'# ambiguous bases': 0</td>\n",
       "      <td>'# scaffolds': 81472</td>\n",
       "      <td>'# contigs': 81472</td>\n",
       "      <td>'Longest scaffold': 43574</td>\n",
       "      <td>'Longest contig': 43574</td>\n",
       "      <td>'N50 (scaffolds)': 2832</td>\n",
       "      <td>'N50 (contigs)': 2832</td>\n",
       "      <td>'Mean scaffold length': 2418.2202474469755</td>\n",
       "      <td>'Mean contig length': 2418.2202474469755</td>\n",
       "      <td>'Coding density': 0.40279689736796637</td>\n",
       "      <td>'Translation table': 11</td>\n",
       "      <td>98</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>189</th>\n",
       "      <td>'GC': 0.4204441792697152</td>\n",
       "      <td>'GC std': 0.05159295343068146</td>\n",
       "      <td>'Genome size': 1186762994</td>\n",
       "      <td>'# ambiguous bases': 0</td>\n",
       "      <td>'# scaffolds': 450486</td>\n",
       "      <td>'# contigs': 450486</td>\n",
       "      <td>'Longest scaffold': 19998</td>\n",
       "      <td>'Longest contig': 19998</td>\n",
       "      <td>'N50 (scaffolds)': 3176</td>\n",
       "      <td>'N50 (contigs)': 3176</td>\n",
       "      <td>'Mean scaffold length': 2634.4059393632656</td>\n",
       "      <td>'Mean contig length': 2634.4059393632656</td>\n",
       "      <td>'Coding density': 0.4967216276378095</td>\n",
       "      <td>'Translation table': 11</td>\n",
       "      <td>healthy_2019_mcav_contigs_cut</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "                             1                                2  \\\n",
       "187   'GC': 0.4343907050953209    'GC std': 0.03418430197601377   \n",
       "188  'GC': 0.39014272558076646   'GC std': 0.027089530386056247   \n",
       "189   'GC': 0.4204441792697152    'GC std': 0.05159295343068146   \n",
       "\n",
       "                              3                        4  \\\n",
       "187    'Genome size': 128714994   '# ambiguous bases': 0   \n",
       "188    'Genome size': 197017240   '# ambiguous bases': 0   \n",
       "189   'Genome size': 1186762994   '# ambiguous bases': 0   \n",
       "\n",
       "                          5                     6                           7  \\\n",
       "187    '# scaffolds': 67929    '# contigs': 67929    'Longest scaffold': 8240   \n",
       "188    '# scaffolds': 81472    '# contigs': 81472   'Longest scaffold': 43574   \n",
       "189   '# scaffolds': 450486   '# contigs': 450486   'Longest scaffold': 19998   \n",
       "\n",
       "                            8                         9  \\\n",
       "187    'Longest contig': 8240   'N50 (scaffolds)': 2022   \n",
       "188   'Longest contig': 43574   'N50 (scaffolds)': 2832   \n",
       "189   'Longest contig': 19998   'N50 (scaffolds)': 3176   \n",
       "\n",
       "                         10                                           11  \\\n",
       "187   'N50 (contigs)': 2022   'Mean scaffold length': 1894.8460009716027   \n",
       "188   'N50 (contigs)': 2832   'Mean scaffold length': 2418.2202474469755   \n",
       "189   'N50 (contigs)': 3176   'Mean scaffold length': 2634.4059393632656   \n",
       "\n",
       "                                            12  \\\n",
       "187   'Mean contig length': 1894.8460009716027   \n",
       "188   'Mean contig length': 2418.2202474469755   \n",
       "189   'Mean contig length': 2634.4059393632656   \n",
       "\n",
       "                                         13                        14  \\\n",
       "187    'Coding density': 0.5431238958842666   'Translation table': 11   \n",
       "188   'Coding density': 0.40279689736796637   'Translation table': 11   \n",
       "189    'Coding density': 0.4967216276378095   'Translation table': 11   \n",
       "\n",
       "                            Bin_Id  \n",
       "187                             84  \n",
       "188                             98  \n",
       "189  healthy_2019_mcav_contigs_cut  "
      ]
     },
     "execution_count": 81,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "print(len(conc_bin_stats))\n",
    "conc_bin_stats.tail(3)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 84,
   "id": "a372919c-2808-4b7c-90e8-05ec889fc697",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>1</th>\n",
       "      <th>2</th>\n",
       "      <th>3</th>\n",
       "      <th>4</th>\n",
       "      <th>5</th>\n",
       "      <th>6</th>\n",
       "      <th>7</th>\n",
       "      <th>8</th>\n",
       "      <th>9</th>\n",
       "      <th>10</th>\n",
       "      <th>11</th>\n",
       "      <th>12</th>\n",
       "      <th>13</th>\n",
       "      <th>14</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>'GC': 0.45383397689109817</td>\n",
       "      <td>'GC std': 0.017741057662292045</td>\n",
       "      <td>'Genome size': 67593</td>\n",
       "      <td>'# ambiguous bases': 0</td>\n",
       "      <td>'# scaffolds': 49</td>\n",
       "      <td>'# contigs': 49</td>\n",
       "      <td>'Longest scaffold': 2600</td>\n",
       "      <td>'Longest contig': 2600</td>\n",
       "      <td>'N50 (scaffolds)': 1396</td>\n",
       "      <td>'N50 (contigs)': 1396</td>\n",
       "      <td>'Mean scaffold length': 1379.4489795918366</td>\n",
       "      <td>'Mean contig length': 1379.4489795918366</td>\n",
       "      <td>'Coding density': 0.4126018966461024</td>\n",
       "      <td>'Translation table': 4</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>'GC': 0.47662485462908316</td>\n",
       "      <td>'GC std': 0.01813517400388053</td>\n",
       "      <td>'Genome size': 401559</td>\n",
       "      <td>'# ambiguous bases': 0</td>\n",
       "      <td>'# scaffolds': 177</td>\n",
       "      <td>'# contigs': 177</td>\n",
       "      <td>'Longest scaffold': 8573</td>\n",
       "      <td>'Longest contig': 8573</td>\n",
       "      <td>'N50 (scaffolds)': 2459</td>\n",
       "      <td>'N50 (contigs)': 2459</td>\n",
       "      <td>'Mean scaffold length': 2268.6949152542375</td>\n",
       "      <td>'Mean contig length': 2268.6949152542375</td>\n",
       "      <td>'Coding density': 0.8767927004499961</td>\n",
       "      <td>'Translation table': 11</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>'GC': 0.4231691853348191</td>\n",
       "      <td>'GC std': 0.03386657292470408</td>\n",
       "      <td>'Genome size': 182173</td>\n",
       "      <td>'# ambiguous bases': 0</td>\n",
       "      <td>'# scaffolds': 128</td>\n",
       "      <td>'# contigs': 128</td>\n",
       "      <td>'Longest scaffold': 2808</td>\n",
       "      <td>'Longest contig': 2808</td>\n",
       "      <td>'N50 (scaffolds)': 1419</td>\n",
       "      <td>'N50 (contigs)': 1419</td>\n",
       "      <td>'Mean scaffold length': 1423.2265625</td>\n",
       "      <td>'Mean contig length': 1423.2265625</td>\n",
       "      <td>'Coding density': 0.49618220043584943</td>\n",
       "      <td>'Translation table': 11</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>'GC': 0.4045708465788093</td>\n",
       "      <td>'GC std': 0.04412584721709139</td>\n",
       "      <td>'Genome size': 194406</td>\n",
       "      <td>'# ambiguous bases': 0</td>\n",
       "      <td>'# scaffolds': 135</td>\n",
       "      <td>'# contigs': 135</td>\n",
       "      <td>'Longest scaffold': 3275</td>\n",
       "      <td>'Longest contig': 3275</td>\n",
       "      <td>'N50 (scaffolds)': 1367</td>\n",
       "      <td>'N50 (contigs)': 1367</td>\n",
       "      <td>'Mean scaffold length': 1440.0444444444445</td>\n",
       "      <td>'Mean contig length': 1440.0444444444445</td>\n",
       "      <td>'Coding density': 0.7444266123473555</td>\n",
       "      <td>'Translation table': 4</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>'GC': 0.33128435782108945</td>\n",
       "      <td>'GC std': 0.052705820586636025</td>\n",
       "      <td>'Genome size': 80040</td>\n",
       "      <td>'# ambiguous bases': 0</td>\n",
       "      <td>'# scaffolds': 45</td>\n",
       "      <td>'# contigs': 45</td>\n",
       "      <td>'Longest scaffold': 18131</td>\n",
       "      <td>'Longest contig': 18131</td>\n",
       "      <td>'N50 (scaffolds)': 1540</td>\n",
       "      <td>'N50 (contigs)': 1540</td>\n",
       "      <td>'Mean scaffold length': 1778.6666666666667</td>\n",
       "      <td>'Mean contig length': 1778.6666666666667</td>\n",
       "      <td>'Coding density': 0.6600949525237382</td>\n",
       "      <td>'Translation table': 11</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>184</th>\n",
       "      <td>'GC': 0.41145448317168515</td>\n",
       "      <td>'GC std': 0.028191213036258277</td>\n",
       "      <td>'Genome size': 23152437</td>\n",
       "      <td>'# ambiguous bases': 0</td>\n",
       "      <td>'# scaffolds': 10584</td>\n",
       "      <td>'# contigs': 10584</td>\n",
       "      <td>'Longest scaffold': 8288</td>\n",
       "      <td>'Longest contig': 8288</td>\n",
       "      <td>'N50 (scaffolds)': 2330</td>\n",
       "      <td>'N50 (contigs)': 2330</td>\n",
       "      <td>'Mean scaffold length': 2187.4940476190477</td>\n",
       "      <td>'Mean contig length': 2187.4940476190477</td>\n",
       "      <td>'Coding density': 0.3832219044586969</td>\n",
       "      <td>'Translation table': 11</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>185</th>\n",
       "      <td>'GC': 0.41414145492114773</td>\n",
       "      <td>'GC std': 0.018241284840074274</td>\n",
       "      <td>'Genome size': 104032662</td>\n",
       "      <td>'# ambiguous bases': 0</td>\n",
       "      <td>'# scaffolds': 22865</td>\n",
       "      <td>'# contigs': 22865</td>\n",
       "      <td>'Longest scaffold': 29540</td>\n",
       "      <td>'Longest contig': 29540</td>\n",
       "      <td>'N50 (scaffolds)': 4709</td>\n",
       "      <td>'N50 (contigs)': 4709</td>\n",
       "      <td>'Mean scaffold length': 4549.864946424666</td>\n",
       "      <td>'Mean contig length': 4549.864946424666</td>\n",
       "      <td>'Coding density': 0.1972559444840506</td>\n",
       "      <td>'Translation table': 11</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>186</th>\n",
       "      <td>'GC': 0.35429501336742997</td>\n",
       "      <td>'GC std': 0.04238931071808634</td>\n",
       "      <td>'Genome size': 17206</td>\n",
       "      <td>'# ambiguous bases': 0</td>\n",
       "      <td>'# scaffolds': 12</td>\n",
       "      <td>'# contigs': 12</td>\n",
       "      <td>'Longest scaffold': 3072</td>\n",
       "      <td>'Longest contig': 3072</td>\n",
       "      <td>'N50 (scaffolds)': 1332</td>\n",
       "      <td>'N50 (contigs)': 1332</td>\n",
       "      <td>'Mean scaffold length': 1433.8333333333333</td>\n",
       "      <td>'Mean contig length': 1433.8333333333333</td>\n",
       "      <td>'Coding density': 0.25543415087760085</td>\n",
       "      <td>'Translation table': 11</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>187</th>\n",
       "      <td>'GC': 0.4343907050953209</td>\n",
       "      <td>'GC std': 0.03418430197601377</td>\n",
       "      <td>'Genome size': 128714994</td>\n",
       "      <td>'# ambiguous bases': 0</td>\n",
       "      <td>'# scaffolds': 67929</td>\n",
       "      <td>'# contigs': 67929</td>\n",
       "      <td>'Longest scaffold': 8240</td>\n",
       "      <td>'Longest contig': 8240</td>\n",
       "      <td>'N50 (scaffolds)': 2022</td>\n",
       "      <td>'N50 (contigs)': 2022</td>\n",
       "      <td>'Mean scaffold length': 1894.8460009716027</td>\n",
       "      <td>'Mean contig length': 1894.8460009716027</td>\n",
       "      <td>'Coding density': 0.5431238958842666</td>\n",
       "      <td>'Translation table': 11</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>188</th>\n",
       "      <td>'GC': 0.39014272558076646</td>\n",
       "      <td>'GC std': 0.027089530386056247</td>\n",
       "      <td>'Genome size': 197017240</td>\n",
       "      <td>'# ambiguous bases': 0</td>\n",
       "      <td>'# scaffolds': 81472</td>\n",
       "      <td>'# contigs': 81472</td>\n",
       "      <td>'Longest scaffold': 43574</td>\n",
       "      <td>'Longest contig': 43574</td>\n",
       "      <td>'N50 (scaffolds)': 2832</td>\n",
       "      <td>'N50 (contigs)': 2832</td>\n",
       "      <td>'Mean scaffold length': 2418.2202474469755</td>\n",
       "      <td>'Mean contig length': 2418.2202474469755</td>\n",
       "      <td>'Coding density': 0.40279689736796637</td>\n",
       "      <td>'Translation table': 11</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>189 rows × 14 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "                             1                                2  \\\n",
       "0    'GC': 0.45383397689109817   'GC std': 0.017741057662292045   \n",
       "1    'GC': 0.47662485462908316    'GC std': 0.01813517400388053   \n",
       "2     'GC': 0.4231691853348191    'GC std': 0.03386657292470408   \n",
       "3     'GC': 0.4045708465788093    'GC std': 0.04412584721709139   \n",
       "4    'GC': 0.33128435782108945   'GC std': 0.052705820586636025   \n",
       "..                         ...                              ...   \n",
       "184  'GC': 0.41145448317168515   'GC std': 0.028191213036258277   \n",
       "185  'GC': 0.41414145492114773   'GC std': 0.018241284840074274   \n",
       "186  'GC': 0.35429501336742997    'GC std': 0.04238931071808634   \n",
       "187   'GC': 0.4343907050953209    'GC std': 0.03418430197601377   \n",
       "188  'GC': 0.39014272558076646   'GC std': 0.027089530386056247   \n",
       "\n",
       "                             3                        4  \\\n",
       "0         'Genome size': 67593   '# ambiguous bases': 0   \n",
       "1        'Genome size': 401559   '# ambiguous bases': 0   \n",
       "2        'Genome size': 182173   '# ambiguous bases': 0   \n",
       "3        'Genome size': 194406   '# ambiguous bases': 0   \n",
       "4         'Genome size': 80040   '# ambiguous bases': 0   \n",
       "..                         ...                      ...   \n",
       "184    'Genome size': 23152437   '# ambiguous bases': 0   \n",
       "185   'Genome size': 104032662   '# ambiguous bases': 0   \n",
       "186       'Genome size': 17206   '# ambiguous bases': 0   \n",
       "187   'Genome size': 128714994   '# ambiguous bases': 0   \n",
       "188   'Genome size': 197017240   '# ambiguous bases': 0   \n",
       "\n",
       "                         5                    6                           7  \\\n",
       "0        '# scaffolds': 49      '# contigs': 49    'Longest scaffold': 2600   \n",
       "1       '# scaffolds': 177     '# contigs': 177    'Longest scaffold': 8573   \n",
       "2       '# scaffolds': 128     '# contigs': 128    'Longest scaffold': 2808   \n",
       "3       '# scaffolds': 135     '# contigs': 135    'Longest scaffold': 3275   \n",
       "4        '# scaffolds': 45      '# contigs': 45   'Longest scaffold': 18131   \n",
       "..                     ...                  ...                         ...   \n",
       "184   '# scaffolds': 10584   '# contigs': 10584    'Longest scaffold': 8288   \n",
       "185   '# scaffolds': 22865   '# contigs': 22865   'Longest scaffold': 29540   \n",
       "186      '# scaffolds': 12      '# contigs': 12    'Longest scaffold': 3072   \n",
       "187   '# scaffolds': 67929   '# contigs': 67929    'Longest scaffold': 8240   \n",
       "188   '# scaffolds': 81472   '# contigs': 81472   'Longest scaffold': 43574   \n",
       "\n",
       "                            8                         9  \\\n",
       "0      'Longest contig': 2600   'N50 (scaffolds)': 1396   \n",
       "1      'Longest contig': 8573   'N50 (scaffolds)': 2459   \n",
       "2      'Longest contig': 2808   'N50 (scaffolds)': 1419   \n",
       "3      'Longest contig': 3275   'N50 (scaffolds)': 1367   \n",
       "4     'Longest contig': 18131   'N50 (scaffolds)': 1540   \n",
       "..                        ...                       ...   \n",
       "184    'Longest contig': 8288   'N50 (scaffolds)': 2330   \n",
       "185   'Longest contig': 29540   'N50 (scaffolds)': 4709   \n",
       "186    'Longest contig': 3072   'N50 (scaffolds)': 1332   \n",
       "187    'Longest contig': 8240   'N50 (scaffolds)': 2022   \n",
       "188   'Longest contig': 43574   'N50 (scaffolds)': 2832   \n",
       "\n",
       "                         10                                           11  \\\n",
       "0     'N50 (contigs)': 1396   'Mean scaffold length': 1379.4489795918366   \n",
       "1     'N50 (contigs)': 2459   'Mean scaffold length': 2268.6949152542375   \n",
       "2     'N50 (contigs)': 1419         'Mean scaffold length': 1423.2265625   \n",
       "3     'N50 (contigs)': 1367   'Mean scaffold length': 1440.0444444444445   \n",
       "4     'N50 (contigs)': 1540   'Mean scaffold length': 1778.6666666666667   \n",
       "..                      ...                                          ...   \n",
       "184   'N50 (contigs)': 2330   'Mean scaffold length': 2187.4940476190477   \n",
       "185   'N50 (contigs)': 4709    'Mean scaffold length': 4549.864946424666   \n",
       "186   'N50 (contigs)': 1332   'Mean scaffold length': 1433.8333333333333   \n",
       "187   'N50 (contigs)': 2022   'Mean scaffold length': 1894.8460009716027   \n",
       "188   'N50 (contigs)': 2832   'Mean scaffold length': 2418.2202474469755   \n",
       "\n",
       "                                            12  \\\n",
       "0     'Mean contig length': 1379.4489795918366   \n",
       "1     'Mean contig length': 2268.6949152542375   \n",
       "2           'Mean contig length': 1423.2265625   \n",
       "3     'Mean contig length': 1440.0444444444445   \n",
       "4     'Mean contig length': 1778.6666666666667   \n",
       "..                                         ...   \n",
       "184   'Mean contig length': 2187.4940476190477   \n",
       "185    'Mean contig length': 4549.864946424666   \n",
       "186   'Mean contig length': 1433.8333333333333   \n",
       "187   'Mean contig length': 1894.8460009716027   \n",
       "188   'Mean contig length': 2418.2202474469755   \n",
       "\n",
       "                                         13                        14  \n",
       "0      'Coding density': 0.4126018966461024    'Translation table': 4  \n",
       "1      'Coding density': 0.8767927004499961   'Translation table': 11  \n",
       "2     'Coding density': 0.49618220043584943   'Translation table': 11  \n",
       "3      'Coding density': 0.7444266123473555    'Translation table': 4  \n",
       "4      'Coding density': 0.6600949525237382   'Translation table': 11  \n",
       "..                                      ...                       ...  \n",
       "184    'Coding density': 0.3832219044586969   'Translation table': 11  \n",
       "185    'Coding density': 0.1972559444840506   'Translation table': 11  \n",
       "186   'Coding density': 0.25543415087760085   'Translation table': 11  \n",
       "187    'Coding density': 0.5431238958842666   'Translation table': 11  \n",
       "188   'Coding density': 0.40279689736796637   'Translation table': 11  \n",
       "\n",
       "[189 rows x 14 columns]"
      ]
     },
     "execution_count": 84,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "# remove bin id col for now, and remove 'healthy_2019_mcav_contigs_cut' bin \n",
    "extract_df=conc_bin_stats.iloc[0:189,0:14]\n",
    "extract_df"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 86,
   "id": "5f30a12a-ec7f-44ca-a5eb-3f5e41c22bf4",
   "metadata": {},
   "outputs": [],
   "source": [
    "# in first row, split by : and keep left side for header names\n",
    "#first_row= extract_df.iloc[0].str.split(':', expand=True)\n",
    "#first_row"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 87,
   "id": "077ed373-4074-4116-9858-2350dadd0cd9",
   "metadata": {},
   "outputs": [],
   "source": [
    "#first_row[0]"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 88,
   "id": "676268c9-0c75-4104-8dbc-b366fb26a30b",
   "metadata": {},
   "outputs": [],
   "source": [
    "#first_row.to_dict()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 90,
   "id": "3c969b68-364c-4f34-bcf1-2d7b49ab397b",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "1                       'GC': 0.47662485462908316\n",
       "2                   'GC std': 0.01813517400388053\n",
       "3                           'Genome size': 401559\n",
       "4                          '# ambiguous bases': 0\n",
       "5                              '# scaffolds': 177\n",
       "6                                '# contigs': 177\n",
       "7                        'Longest scaffold': 8573\n",
       "8                          'Longest contig': 8573\n",
       "9                         'N50 (scaffolds)': 2459\n",
       "10                          'N50 (contigs)': 2459\n",
       "11     'Mean scaffold length': 2268.6949152542375\n",
       "12       'Mean contig length': 2268.6949152542375\n",
       "13           'Coding density': 0.8767927004499961\n",
       "14                        'Translation table': 11\n",
       "Name: 1, dtype: object"
      ]
     },
     "execution_count": 90,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "extract_df.iloc[1]"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 91,
   "id": "655a68f3-8857-4c25-8ee8-6ba30d9acf9b",
   "metadata": {},
   "outputs": [],
   "source": [
    "# in first col of new df: split by \":\" and get left side for col names \n",
    "new=extract_df.iloc[1].str.split(':', expand=True)\n",
    "index=new[0]\n",
    "cols=index.to_list()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 92,
   "id": "84fd72c4-c23e-43e1-8bb4-8fecbbbc33cd",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>'GC'</th>\n",
       "      <th>'GC std'</th>\n",
       "      <th>'Genome size'</th>\n",
       "      <th>'# ambiguous bases'</th>\n",
       "      <th>'# scaffolds'</th>\n",
       "      <th>'# contigs'</th>\n",
       "      <th>'Longest scaffold'</th>\n",
       "      <th>'Longest contig'</th>\n",
       "      <th>'N50 (scaffolds)'</th>\n",
       "      <th>'N50 (contigs)'</th>\n",
       "      <th>'Mean scaffold length'</th>\n",
       "      <th>'Mean contig length'</th>\n",
       "      <th>'Coding density'</th>\n",
       "      <th>'Translation table'</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>184</th>\n",
       "      <td>0.41145448317168515</td>\n",
       "      <td>0.028191213036258277</td>\n",
       "      <td>23152437</td>\n",
       "      <td>0</td>\n",
       "      <td>10584</td>\n",
       "      <td>10584</td>\n",
       "      <td>8288</td>\n",
       "      <td>8288</td>\n",
       "      <td>2330</td>\n",
       "      <td>2330</td>\n",
       "      <td>2187.4940476190477</td>\n",
       "      <td>2187.4940476190477</td>\n",
       "      <td>0.3832219044586969</td>\n",
       "      <td>11</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>185</th>\n",
       "      <td>0.41414145492114773</td>\n",
       "      <td>0.018241284840074274</td>\n",
       "      <td>104032662</td>\n",
       "      <td>0</td>\n",
       "      <td>22865</td>\n",
       "      <td>22865</td>\n",
       "      <td>29540</td>\n",
       "      <td>29540</td>\n",
       "      <td>4709</td>\n",
       "      <td>4709</td>\n",
       "      <td>4549.864946424666</td>\n",
       "      <td>4549.864946424666</td>\n",
       "      <td>0.1972559444840506</td>\n",
       "      <td>11</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>186</th>\n",
       "      <td>0.35429501336742997</td>\n",
       "      <td>0.04238931071808634</td>\n",
       "      <td>17206</td>\n",
       "      <td>0</td>\n",
       "      <td>12</td>\n",
       "      <td>12</td>\n",
       "      <td>3072</td>\n",
       "      <td>3072</td>\n",
       "      <td>1332</td>\n",
       "      <td>1332</td>\n",
       "      <td>1433.8333333333333</td>\n",
       "      <td>1433.8333333333333</td>\n",
       "      <td>0.25543415087760085</td>\n",
       "      <td>11</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>187</th>\n",
       "      <td>0.4343907050953209</td>\n",
       "      <td>0.03418430197601377</td>\n",
       "      <td>128714994</td>\n",
       "      <td>0</td>\n",
       "      <td>67929</td>\n",
       "      <td>67929</td>\n",
       "      <td>8240</td>\n",
       "      <td>8240</td>\n",
       "      <td>2022</td>\n",
       "      <td>2022</td>\n",
       "      <td>1894.8460009716027</td>\n",
       "      <td>1894.8460009716027</td>\n",
       "      <td>0.5431238958842666</td>\n",
       "      <td>11</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>188</th>\n",
       "      <td>0.39014272558076646</td>\n",
       "      <td>0.027089530386056247</td>\n",
       "      <td>197017240</td>\n",
       "      <td>0</td>\n",
       "      <td>81472</td>\n",
       "      <td>81472</td>\n",
       "      <td>43574</td>\n",
       "      <td>43574</td>\n",
       "      <td>2832</td>\n",
       "      <td>2832</td>\n",
       "      <td>2418.2202474469755</td>\n",
       "      <td>2418.2202474469755</td>\n",
       "      <td>0.40279689736796637</td>\n",
       "      <td>11</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "                     'GC'               'GC std'  'Genome size'  \\\n",
       "184   0.41145448317168515   0.028191213036258277       23152437   \n",
       "185   0.41414145492114773   0.018241284840074274      104032662   \n",
       "186   0.35429501336742997    0.04238931071808634          17206   \n",
       "187    0.4343907050953209    0.03418430197601377      128714994   \n",
       "188   0.39014272558076646   0.027089530386056247      197017240   \n",
       "\n",
       "     '# ambiguous bases'  '# scaffolds'  '# contigs'  'Longest scaffold'  \\\n",
       "184                    0          10584        10584                8288   \n",
       "185                    0          22865        22865               29540   \n",
       "186                    0             12           12                3072   \n",
       "187                    0          67929        67929                8240   \n",
       "188                    0          81472        81472               43574   \n",
       "\n",
       "     'Longest contig'  'N50 (scaffolds)'  'N50 (contigs)'  \\\n",
       "184              8288               2330             2330   \n",
       "185             29540               4709             4709   \n",
       "186              3072               1332             1332   \n",
       "187              8240               2022             2022   \n",
       "188             43574               2832             2832   \n",
       "\n",
       "     'Mean scaffold length'  'Mean contig length'      'Coding density'  \\\n",
       "184      2187.4940476190477    2187.4940476190477    0.3832219044586969   \n",
       "185       4549.864946424666     4549.864946424666    0.1972559444840506   \n",
       "186      1433.8333333333333    1433.8333333333333   0.25543415087760085   \n",
       "187      1894.8460009716027    1894.8460009716027    0.5431238958842666   \n",
       "188      2418.2202474469755    2418.2202474469755   0.40279689736796637   \n",
       "\n",
       "     'Translation table'  \n",
       "184                   11  \n",
       "185                   11  \n",
       "186                   11  \n",
       "187                   11  \n",
       "188                   11  "
      ]
     },
     "execution_count": 92,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "# loop through and extract the right of : in each row, add to dictionary, readd col names, covnert to df \n",
    "d={}\n",
    "for i in extract_df.index:\n",
    "    extracted_values = extract_df.iloc[i].str.extract(r'(?<=:)(.*)', expand=True)\n",
    "    #print(extracted_values)\n",
    "    list_of_values=extracted_values[0]\n",
    "    lists=list_of_values.to_list()\n",
    "    d[i] = lists\n",
    "d\n",
    "intermediate=pd.DataFrame.from_dict(d,orient='index')\n",
    "intermediate.columns=cols\n",
    "intermediate.tail()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 95,
   "id": "28b49153-a260-4b67-ad62-965645075ba4",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>'GC'</th>\n",
       "      <th>'GC std'</th>\n",
       "      <th>'Genome size'</th>\n",
       "      <th>'# ambiguous bases'</th>\n",
       "      <th>'# scaffolds'</th>\n",
       "      <th>'# contigs'</th>\n",
       "      <th>'Longest scaffold'</th>\n",
       "      <th>'Longest contig'</th>\n",
       "      <th>'N50 (scaffolds)'</th>\n",
       "      <th>'N50 (contigs)'</th>\n",
       "      <th>'Mean scaffold length'</th>\n",
       "      <th>'Mean contig length'</th>\n",
       "      <th>'Coding density'</th>\n",
       "      <th>'Translation table'</th>\n",
       "      <th>Bin_Id</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>0.45383397689109817</td>\n",
       "      <td>0.017741057662292045</td>\n",
       "      <td>67593</td>\n",
       "      <td>0</td>\n",
       "      <td>49</td>\n",
       "      <td>49</td>\n",
       "      <td>2600</td>\n",
       "      <td>2600</td>\n",
       "      <td>1396</td>\n",
       "      <td>1396</td>\n",
       "      <td>1379.4489795918366</td>\n",
       "      <td>1379.4489795918366</td>\n",
       "      <td>0.4126018966461024</td>\n",
       "      <td>4</td>\n",
       "      <td>10</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>0.47662485462908316</td>\n",
       "      <td>0.01813517400388053</td>\n",
       "      <td>401559</td>\n",
       "      <td>0</td>\n",
       "      <td>177</td>\n",
       "      <td>177</td>\n",
       "      <td>8573</td>\n",
       "      <td>8573</td>\n",
       "      <td>2459</td>\n",
       "      <td>2459</td>\n",
       "      <td>2268.6949152542375</td>\n",
       "      <td>2268.6949152542375</td>\n",
       "      <td>0.8767927004499961</td>\n",
       "      <td>11</td>\n",
       "      <td>1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>0.4231691853348191</td>\n",
       "      <td>0.03386657292470408</td>\n",
       "      <td>182173</td>\n",
       "      <td>0</td>\n",
       "      <td>128</td>\n",
       "      <td>128</td>\n",
       "      <td>2808</td>\n",
       "      <td>2808</td>\n",
       "      <td>1419</td>\n",
       "      <td>1419</td>\n",
       "      <td>1423.2265625</td>\n",
       "      <td>1423.2265625</td>\n",
       "      <td>0.49618220043584943</td>\n",
       "      <td>11</td>\n",
       "      <td>101</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>0.4045708465788093</td>\n",
       "      <td>0.04412584721709139</td>\n",
       "      <td>194406</td>\n",
       "      <td>0</td>\n",
       "      <td>135</td>\n",
       "      <td>135</td>\n",
       "      <td>3275</td>\n",
       "      <td>3275</td>\n",
       "      <td>1367</td>\n",
       "      <td>1367</td>\n",
       "      <td>1440.0444444444445</td>\n",
       "      <td>1440.0444444444445</td>\n",
       "      <td>0.7444266123473555</td>\n",
       "      <td>4</td>\n",
       "      <td>102</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>0.33128435782108945</td>\n",
       "      <td>0.052705820586636025</td>\n",
       "      <td>80040</td>\n",
       "      <td>0</td>\n",
       "      <td>45</td>\n",
       "      <td>45</td>\n",
       "      <td>18131</td>\n",
       "      <td>18131</td>\n",
       "      <td>1540</td>\n",
       "      <td>1540</td>\n",
       "      <td>1778.6666666666667</td>\n",
       "      <td>1778.6666666666667</td>\n",
       "      <td>0.6600949525237382</td>\n",
       "      <td>11</td>\n",
       "      <td>100</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>184</th>\n",
       "      <td>0.41145448317168515</td>\n",
       "      <td>0.028191213036258277</td>\n",
       "      <td>23152437</td>\n",
       "      <td>0</td>\n",
       "      <td>10584</td>\n",
       "      <td>10584</td>\n",
       "      <td>8288</td>\n",
       "      <td>8288</td>\n",
       "      <td>2330</td>\n",
       "      <td>2330</td>\n",
       "      <td>2187.4940476190477</td>\n",
       "      <td>2187.4940476190477</td>\n",
       "      <td>0.3832219044586969</td>\n",
       "      <td>11</td>\n",
       "      <td>97</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>185</th>\n",
       "      <td>0.41414145492114773</td>\n",
       "      <td>0.018241284840074274</td>\n",
       "      <td>104032662</td>\n",
       "      <td>0</td>\n",
       "      <td>22865</td>\n",
       "      <td>22865</td>\n",
       "      <td>29540</td>\n",
       "      <td>29540</td>\n",
       "      <td>4709</td>\n",
       "      <td>4709</td>\n",
       "      <td>4549.864946424666</td>\n",
       "      <td>4549.864946424666</td>\n",
       "      <td>0.1972559444840506</td>\n",
       "      <td>11</td>\n",
       "      <td>94</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>186</th>\n",
       "      <td>0.35429501336742997</td>\n",
       "      <td>0.04238931071808634</td>\n",
       "      <td>17206</td>\n",
       "      <td>0</td>\n",
       "      <td>12</td>\n",
       "      <td>12</td>\n",
       "      <td>3072</td>\n",
       "      <td>3072</td>\n",
       "      <td>1332</td>\n",
       "      <td>1332</td>\n",
       "      <td>1433.8333333333333</td>\n",
       "      <td>1433.8333333333333</td>\n",
       "      <td>0.25543415087760085</td>\n",
       "      <td>11</td>\n",
       "      <td>99</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>187</th>\n",
       "      <td>0.4343907050953209</td>\n",
       "      <td>0.03418430197601377</td>\n",
       "      <td>128714994</td>\n",
       "      <td>0</td>\n",
       "      <td>67929</td>\n",
       "      <td>67929</td>\n",
       "      <td>8240</td>\n",
       "      <td>8240</td>\n",
       "      <td>2022</td>\n",
       "      <td>2022</td>\n",
       "      <td>1894.8460009716027</td>\n",
       "      <td>1894.8460009716027</td>\n",
       "      <td>0.5431238958842666</td>\n",
       "      <td>11</td>\n",
       "      <td>84</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>188</th>\n",
       "      <td>0.39014272558076646</td>\n",
       "      <td>0.027089530386056247</td>\n",
       "      <td>197017240</td>\n",
       "      <td>0</td>\n",
       "      <td>81472</td>\n",
       "      <td>81472</td>\n",
       "      <td>43574</td>\n",
       "      <td>43574</td>\n",
       "      <td>2832</td>\n",
       "      <td>2832</td>\n",
       "      <td>2418.2202474469755</td>\n",
       "      <td>2418.2202474469755</td>\n",
       "      <td>0.40279689736796637</td>\n",
       "      <td>11</td>\n",
       "      <td>98</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>189 rows × 15 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "                     'GC'               'GC std'  'Genome size'  \\\n",
       "0     0.45383397689109817   0.017741057662292045          67593   \n",
       "1     0.47662485462908316    0.01813517400388053         401559   \n",
       "2      0.4231691853348191    0.03386657292470408         182173   \n",
       "3      0.4045708465788093    0.04412584721709139         194406   \n",
       "4     0.33128435782108945   0.052705820586636025          80040   \n",
       "..                    ...                    ...            ...   \n",
       "184   0.41145448317168515   0.028191213036258277       23152437   \n",
       "185   0.41414145492114773   0.018241284840074274      104032662   \n",
       "186   0.35429501336742997    0.04238931071808634          17206   \n",
       "187    0.4343907050953209    0.03418430197601377      128714994   \n",
       "188   0.39014272558076646   0.027089530386056247      197017240   \n",
       "\n",
       "     '# ambiguous bases'  '# scaffolds'  '# contigs'  'Longest scaffold'  \\\n",
       "0                      0             49           49                2600   \n",
       "1                      0            177          177                8573   \n",
       "2                      0            128          128                2808   \n",
       "3                      0            135          135                3275   \n",
       "4                      0             45           45               18131   \n",
       "..                   ...            ...          ...                 ...   \n",
       "184                    0          10584        10584                8288   \n",
       "185                    0          22865        22865               29540   \n",
       "186                    0             12           12                3072   \n",
       "187                    0          67929        67929                8240   \n",
       "188                    0          81472        81472               43574   \n",
       "\n",
       "     'Longest contig'  'N50 (scaffolds)'  'N50 (contigs)'  \\\n",
       "0                2600               1396             1396   \n",
       "1                8573               2459             2459   \n",
       "2                2808               1419             1419   \n",
       "3                3275               1367             1367   \n",
       "4               18131               1540             1540   \n",
       "..                ...                ...              ...   \n",
       "184              8288               2330             2330   \n",
       "185             29540               4709             4709   \n",
       "186              3072               1332             1332   \n",
       "187              8240               2022             2022   \n",
       "188             43574               2832             2832   \n",
       "\n",
       "     'Mean scaffold length'  'Mean contig length'      'Coding density'  \\\n",
       "0        1379.4489795918366    1379.4489795918366    0.4126018966461024   \n",
       "1        2268.6949152542375    2268.6949152542375    0.8767927004499961   \n",
       "2              1423.2265625          1423.2265625   0.49618220043584943   \n",
       "3        1440.0444444444445    1440.0444444444445    0.7444266123473555   \n",
       "4        1778.6666666666667    1778.6666666666667    0.6600949525237382   \n",
       "..                      ...                   ...                   ...   \n",
       "184      2187.4940476190477    2187.4940476190477    0.3832219044586969   \n",
       "185       4549.864946424666     4549.864946424666    0.1972559444840506   \n",
       "186      1433.8333333333333    1433.8333333333333   0.25543415087760085   \n",
       "187      1894.8460009716027    1894.8460009716027    0.5431238958842666   \n",
       "188      2418.2202474469755    2418.2202474469755   0.40279689736796637   \n",
       "\n",
       "     'Translation table' Bin_Id  \n",
       "0                      4     10  \n",
       "1                     11      1  \n",
       "2                     11    101  \n",
       "3                      4    102  \n",
       "4                     11    100  \n",
       "..                   ...    ...  \n",
       "184                   11     97  \n",
       "185                   11     94  \n",
       "186                   11     99  \n",
       "187                   11     84  \n",
       "188                   11     98  \n",
       "\n",
       "[189 rows x 15 columns]"
      ]
     },
     "execution_count": 95,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "#add bin ids back in \n",
    "intermediate['Bin_Id']=conc_bin_stats['Bin_Id']\n",
    "intermediate"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 101,
   "id": "3e3360c1-5d6f-4c56-88a4-326715ce4781",
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "['10' '1' '101' '102' '100' '104' '103' '106' '107' '105' '0' '11' '110'\n",
      " '109' '112' '113' '114' '115' '116' '117' '118' '119' '108' '120' '121'\n",
      " '122' '123' '111' '125' '126' '127' '128' '129' '12' '13' '131' '132'\n",
      " '130' '133' '135' '124' '136' '134' '138' '137' '140' '141' '142' '143'\n",
      " '144' '145' '146' '147' '148' '149' '15' '150' '151' '14' '153' '152'\n",
      " '155' '156' '157' '158' '154' '159' '16' '161' '162' '163' '164' '165'\n",
      " '166' '139' '168' '160' '17' '170' '171' '172' '173' '174' '175' '176'\n",
      " '177' '178' '179' '18' '180' '181' '182' '183' '184' '185' '186' '187'\n",
      " '188' '169' '19' '20' '2' '22' '21' '24' '25' '26' '27' '28' '29' '3'\n",
      " '30' '31' '32' '33' '34' '35' '23' '36' '38' '39' '4' '40' '41' '37' '42'\n",
      " '44' '45' '46' '47' '48' '49' '5' '43' '51' '52' '53' '50' '55' '56' '57'\n",
      " '58' '59' '6' '60' '61' '62' '63' '64' '65' '66' '67' '54' '69' '7' '70'\n",
      " '71' '72' '73' '74' '75' '68' '77' '78' '79' '8' '76' '81' '82' '80'\n",
      " '167' '85' '86' '87' '88' '89' '9' '90' '91' '92' '93' '83' '95' '96'\n",
      " '97' '94' '99' '84' '98']\n",
      "189\n"
     ]
    }
   ],
   "source": [
    "print(intermediate['Bin_Id'].unique())\n",
    "print(len(intermediate['Bin_Id'].unique()))"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 104,
   "id": "fe0f823c-595e-4c0e-bb8e-e1997eea2dcf",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead tr th {\n",
       "        text-align: left;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr>\n",
       "      <th></th>\n",
       "      <th>Bin_Id</th>\n",
       "      <th>Marker_lineage</th>\n",
       "      <th>UID</th>\n",
       "      <th>#genomes</th>\n",
       "      <th>#markers</th>\n",
       "      <th>#marker_sets</th>\n",
       "      <th>0</th>\n",
       "      <th>1</th>\n",
       "      <th>2</th>\n",
       "      <th>3</th>\n",
       "      <th>4</th>\n",
       "      <th>5+</th>\n",
       "      <th>Completeness</th>\n",
       "      <th>Contamination</th>\n",
       "      <th>Strain_heterogeneity</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>42</td>\n",
       "      <td>o__Sphingomonadales</td>\n",
       "      <td>(UID3310)</td>\n",
       "      <td>26</td>\n",
       "      <td>569</td>\n",
       "      <td>293</td>\n",
       "      <td>3</td>\n",
       "      <td>556</td>\n",
       "      <td>10</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>99.59</td>\n",
       "      <td>2.30</td>\n",
       "      <td>0.00</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>56</td>\n",
       "      <td>c__Gammaproteobacteria</td>\n",
       "      <td>(UID4444)</td>\n",
       "      <td>263</td>\n",
       "      <td>507</td>\n",
       "      <td>232</td>\n",
       "      <td>45</td>\n",
       "      <td>450</td>\n",
       "      <td>12</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>85.79</td>\n",
       "      <td>2.68</td>\n",
       "      <td>33.33</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>109</td>\n",
       "      <td>k__Bacteria</td>\n",
       "      <td>(UID3187)</td>\n",
       "      <td>2258</td>\n",
       "      <td>188</td>\n",
       "      <td>117</td>\n",
       "      <td>53</td>\n",
       "      <td>132</td>\n",
       "      <td>3</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>68.90</td>\n",
       "      <td>1.12</td>\n",
       "      <td>0.00</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>68</td>\n",
       "      <td>k__Archaea</td>\n",
       "      <td>(UID2)</td>\n",
       "      <td>207</td>\n",
       "      <td>149</td>\n",
       "      <td>107</td>\n",
       "      <td>61</td>\n",
       "      <td>62</td>\n",
       "      <td>18</td>\n",
       "      <td>7</td>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "      <td>53.82</td>\n",
       "      <td>13.76</td>\n",
       "      <td>0.00</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>139</td>\n",
       "      <td>p__Cyanobacteria</td>\n",
       "      <td>(UID2143)</td>\n",
       "      <td>129</td>\n",
       "      <td>472</td>\n",
       "      <td>368</td>\n",
       "      <td>242</td>\n",
       "      <td>220</td>\n",
       "      <td>8</td>\n",
       "      <td>2</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>46.70</td>\n",
       "      <td>1.84</td>\n",
       "      <td>14.29</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "  Bin_Id          Marker_lineage        UID #genomes #markers #marker_sets  \\\n",
       "0     42     o__Sphingomonadales  (UID3310)       26      569          293   \n",
       "1     56  c__Gammaproteobacteria  (UID4444)      263      507          232   \n",
       "2    109             k__Bacteria  (UID3187)     2258      188          117   \n",
       "3     68              k__Archaea     (UID2)      207      149          107   \n",
       "4    139        p__Cyanobacteria  (UID2143)      129      472          368   \n",
       "\n",
       "     0    1   2  3  4 5+ Completeness Contamination Strain_heterogeneity  \n",
       "0    3  556  10  0  0  0        99.59          2.30                 0.00  \n",
       "1   45  450  12  0  0  0        85.79          2.68                33.33  \n",
       "2   53  132   3  0  0  0        68.90          1.12                 0.00  \n",
       "3   61   62  18  7  1  0        53.82         13.76                 0.00  \n",
       "4  242  220   8  2  0  0        46.70          1.84                14.29  "
      ]
     },
     "execution_count": 104,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "concoct_bins.head()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 117,
   "id": "f3d0d9de-32d1-48be-969b-59f41fd6240b",
   "metadata": {},
   "outputs": [],
   "source": [
    "# col names became multiIndex, so reformatting\n",
    "concoct_bins.columns = concoct_bins.columns.get_level_values(0)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 122,
   "id": "ad6e7eb2-8395-4cfe-a581-46c0eccc50da",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "array([ 42,  56, 109,  68, 139,  50, 150, 167,  98, 170,  83, 113,  84,\n",
       "         7, 105,  54,  76,  49, 169,  66, 152, 136,  97, 137, 154,  14,\n",
       "        90,  75, 160, 156,  43, 188, 166, 134,  45,  99,  96,  95,  94,\n",
       "        93,  92,  91,   9,  89,  88,  87,  86,  85,  82,  81,  80,   8,\n",
       "        79,  78,  77,  74,  73,  72,  71,  70,  69,  67,  65,  64,  63,\n",
       "        62,  61,  60,   6,  59,  58,  57,  55,  53,  52,  51,   5,  48,\n",
       "        47,  46,  44,  41,  40,   4,  39,  38,  37,  36,  35,  34,  33,\n",
       "        32,  31,  30,   3,  29,  28,  27,  26,  25,  24,  23,  22,  21,\n",
       "        20,   2,  19, 187, 186, 185, 184, 183, 182, 181, 180,  18, 179,\n",
       "       178, 177, 176, 175, 174, 173, 172, 171,  17, 168, 165, 164, 163,\n",
       "       162, 161,  16, 159, 158, 157, 155, 153, 151,  15, 149, 148, 147,\n",
       "       146, 145, 144, 143, 142, 141, 140, 138, 135, 133, 132, 131, 130,\n",
       "        13, 129, 128, 127, 126, 125, 124, 123, 122, 121, 120,  12, 119,\n",
       "       118, 117, 116, 115, 114, 112, 111, 110,  11, 108, 107, 106, 104,\n",
       "       103, 102, 101, 100,  10,   1,   0])"
      ]
     },
     "execution_count": 122,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "concoct_bins.Bin_Id.values"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 124,
   "id": "b79a4c89-f2c1-4b74-b130-c6c7d2daf955",
   "metadata": {},
   "outputs": [],
   "source": [
    "# converting intermiate bin id values into integers\n",
    "intermediate[\"Bin_Id\"] = intermediate.Bin_Id.apply(int)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 129,
   "id": "da20fdb7-828c-43dc-8ad6-48a21023a017",
   "metadata": {},
   "outputs": [],
   "source": [
    "#merging dataframes \n",
    "concoct_stats_merged=concoct_bins.merge(intermediate,on='Bin_Id')"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 133,
   "id": "f0df422d-9206-4686-a098-b2fa69abe892",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "Index(['Bin_Id', 'Marker_lineage', 'UID', '#genomes', '#markers',\n",
       "       '#marker_sets', '0', '1', '2', '3', '4', '5+', 'Completeness',\n",
       "       'Contamination', 'Strain_heterogeneity', ''GC'', ' 'GC std'',\n",
       "       ' 'Genome size'', ' '# ambiguous bases'', ' '# scaffolds'',\n",
       "       ' '# contigs'', ' 'Longest scaffold'', ' 'Longest contig'',\n",
       "       ' 'N50 (scaffolds)'', ' 'N50 (contigs)'', ' 'Mean scaffold length'',\n",
       "       ' 'Mean contig length'', ' 'Coding density'', ' 'Translation table''],\n",
       "      dtype='object')"
      ]
     },
     "execution_count": 133,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "concoct_stats_merged.columns"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 135,
   "id": "da8a7272-7a00-4c8f-971a-6d4b27c62383",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>Bin_Id</th>\n",
       "      <th>Marker_lineage</th>\n",
       "      <th>UID</th>\n",
       "      <th>#genomes</th>\n",
       "      <th>#markers</th>\n",
       "      <th>#marker_sets</th>\n",
       "      <th>0</th>\n",
       "      <th>1</th>\n",
       "      <th>2</th>\n",
       "      <th>3</th>\n",
       "      <th>...</th>\n",
       "      <th>'# scaffolds'</th>\n",
       "      <th>'# contigs'</th>\n",
       "      <th>'Longest scaffold'</th>\n",
       "      <th>'Longest contig'</th>\n",
       "      <th>'N50 (scaffolds)'</th>\n",
       "      <th>'N50 (contigs)'</th>\n",
       "      <th>'Mean scaffold length'</th>\n",
       "      <th>'Mean contig length'</th>\n",
       "      <th>'Coding density'</th>\n",
       "      <th>'Translation table'</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>109</td>\n",
       "      <td>k__Bacteria</td>\n",
       "      <td>(UID3187)</td>\n",
       "      <td>2258</td>\n",
       "      <td>188</td>\n",
       "      <td>117</td>\n",
       "      <td>53</td>\n",
       "      <td>132</td>\n",
       "      <td>3</td>\n",
       "      <td>0</td>\n",
       "      <td>...</td>\n",
       "      <td>960</td>\n",
       "      <td>960</td>\n",
       "      <td>8661</td>\n",
       "      <td>8661</td>\n",
       "      <td>2243</td>\n",
       "      <td>2243</td>\n",
       "      <td>2112.117708333333</td>\n",
       "      <td>2112.117708333333</td>\n",
       "      <td>0.797705501932549</td>\n",
       "      <td>11</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>1 rows × 29 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "   Bin_Id Marker_lineage        UID  #genomes  #markers  #marker_sets   0  \\\n",
       "2     109    k__Bacteria  (UID3187)      2258       188           117  53   \n",
       "\n",
       "     1  2  3  ...   '# scaffolds'   '# contigs'   'Longest scaffold'  \\\n",
       "2  132  3  0  ...             960           960                 8661   \n",
       "\n",
       "    'Longest contig'   'N50 (scaffolds)'  'N50 (contigs)'  \\\n",
       "2               8661                2243             2243   \n",
       "\n",
       "   'Mean scaffold length'  'Mean contig length'    'Coding density'  \\\n",
       "2       2112.117708333333     2112.117708333333   0.797705501932549   \n",
       "\n",
       "   'Translation table'  \n",
       "2                   11  \n",
       "\n",
       "[1 rows x 29 columns]"
      ]
     },
     "execution_count": 135,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "concoct_109=concoct_stats_merged[concoct_stats_merged['Bin_Id'] == 109]\n",
    "concoct_109"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 155,
   "id": "0dcfc754-66c4-431e-8ec8-c7f3315978f1",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>Bin_Id</th>\n",
       "      <th>Marker_lineage</th>\n",
       "      <th>UID</th>\n",
       "      <th>#genomes</th>\n",
       "      <th>#markers</th>\n",
       "      <th>#marker_sets</th>\n",
       "      <th>0</th>\n",
       "      <th>1</th>\n",
       "      <th>2</th>\n",
       "      <th>3</th>\n",
       "      <th>4</th>\n",
       "      <th>5+</th>\n",
       "      <th>Completeness</th>\n",
       "      <th>Contamination</th>\n",
       "      <th>Strain_heterogeneity</th>\n",
       "      <th>'GC'</th>\n",
       "      <th>'GC std'</th>\n",
       "      <th>'Genome size'</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>109</td>\n",
       "      <td>k__Bacteria</td>\n",
       "      <td>(UID3187)</td>\n",
       "      <td>2258</td>\n",
       "      <td>188</td>\n",
       "      <td>117</td>\n",
       "      <td>53</td>\n",
       "      <td>132</td>\n",
       "      <td>3</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>68.9</td>\n",
       "      <td>1.12</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.3992117902993293</td>\n",
       "      <td>0.027524218532616485</td>\n",
       "      <td>2027633</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "   Bin_Id Marker_lineage        UID  #genomes  #markers  #marker_sets   0  \\\n",
       "2     109    k__Bacteria  (UID3187)      2258       188           117  53   \n",
       "\n",
       "     1  2  3  4  5+  Completeness  Contamination  Strain_heterogeneity  \\\n",
       "2  132  3  0  0   0          68.9           1.12                   0.0   \n",
       "\n",
       "                  'GC'               'GC std'  'Genome size'  \n",
       "2   0.3992117902993293   0.027524218532616485        2027633  "
      ]
     },
     "execution_count": 155,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "concoct_109.iloc[:, 0:18]"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "141f4e1f-50cb-4d8e-858e-0142083956df",
   "metadata": {
    "tags": []
   },
   "source": [
    "#### Results & Next Steps\n",
    "- anvio found 0 scg for concoct 109, based on # contigs, length, and N50 it may just be too fragmented\n",
    "- run against gtdb independently and import into anvio \n",
    "- look more into mags in both binners and both sample sets "
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 2,
   "id": "2c1a527f-cf10-4bc2-9498-473ae19a6a3d",
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Concoct_healthy_2019_mcav_bins\tmetabat\n",
      "MetaBAT_healthy_2019_mcav_bins\tslurm-21569803-concoct.out\n",
      "concoct\t\t\t\tslurm-21582072-das_tool.out\n",
      "das_tool\t\t\tslurm-metabat-19497287.out\n",
      "das_tool_DASTool_bins\n"
     ]
    }
   ],
   "source": [
    "!ls //project/pi_sarah_gignouxwolfsohn_uml_edu/brooke/mcav/healthy_2019_mcav/binning/"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 181,
   "id": "7d85d0bc-e898-4c41-8672-d1030205beef",
   "metadata": {},
   "outputs": [],
   "source": [
    "os.chdir('//project/pi_sarah_gignouxwolfsohn_uml_edu/brooke/mcav/healthy_2019_mcav/binning/MetaBAT_healthy_2019_mcav_bins/Metabat/bins-stats/storage')"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 186,
   "id": "b60bbcac-6358-47a9-a461-72230d83fcbc",
   "metadata": {},
   "outputs": [],
   "source": [
    "binname=\"meta\""
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 193,
   "id": "06821cbe-b6c9-4c0a-92ca-545b36150c81",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>1</th>\n",
       "      <th>2</th>\n",
       "      <th>3</th>\n",
       "      <th>4</th>\n",
       "      <th>5</th>\n",
       "      <th>6</th>\n",
       "      <th>7</th>\n",
       "      <th>8</th>\n",
       "      <th>9</th>\n",
       "      <th>10</th>\n",
       "      <th>11</th>\n",
       "      <th>12</th>\n",
       "      <th>13</th>\n",
       "      <th>14</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>17</th>\n",
       "      <td>.7\\t{'GC': 0.3846045581267069</td>\n",
       "      <td>'GC std': 0.03400637013380865</td>\n",
       "      <td>'Genome size': 703403</td>\n",
       "      <td>'# ambiguous bases': 0</td>\n",
       "      <td>'# scaffolds': 385</td>\n",
       "      <td>'# contigs': 385</td>\n",
       "      <td>'Longest scaffold': 6914</td>\n",
       "      <td>'Longest contig': 6914</td>\n",
       "      <td>'N50 (scaffolds)': 1722</td>\n",
       "      <td>'N50 (contigs)': 1722</td>\n",
       "      <td>'Mean scaffold length': 1827.0207792207793</td>\n",
       "      <td>'Mean contig length': 1827.0207792207793</td>\n",
       "      <td>'Coding density': 0.8291420423285087</td>\n",
       "      <td>'Translation table': 11</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>18</th>\n",
       "      <td>.8\\t{'GC': 0.37689090505296374</td>\n",
       "      <td>'GC std': 0.0240623411465306</td>\n",
       "      <td>'Genome size': 9595339</td>\n",
       "      <td>'# ambiguous bases': 0</td>\n",
       "      <td>'# scaffolds': 2995</td>\n",
       "      <td>'# contigs': 2995</td>\n",
       "      <td>'Longest scaffold': 20426</td>\n",
       "      <td>'Longest contig': 20426</td>\n",
       "      <td>'N50 (scaffolds)': 3535</td>\n",
       "      <td>'N50 (contigs)': 3535</td>\n",
       "      <td>'Mean scaffold length': 3203.785976627713</td>\n",
       "      <td>'Mean contig length': 3203.785976627713</td>\n",
       "      <td>'Coding density': 0.3965412790522565</td>\n",
       "      <td>'Translation table': 11</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>19</th>\n",
       "      <td>.9\\t{'GC': 0.3985752720619433</td>\n",
       "      <td>'GC std': 0.02356025011909446</td>\n",
       "      <td>'Genome size': 9585409</td>\n",
       "      <td>'# ambiguous bases': 0</td>\n",
       "      <td>'# scaffolds': 4007</td>\n",
       "      <td>'# contigs': 4007</td>\n",
       "      <td>'Longest scaffold': 12780</td>\n",
       "      <td>'Longest contig': 12780</td>\n",
       "      <td>'N50 (scaffolds)': 2392</td>\n",
       "      <td>'N50 (contigs)': 2392</td>\n",
       "      <td>'Mean scaffold length': 2392.165959570751</td>\n",
       "      <td>'Mean contig length': 2392.165959570751</td>\n",
       "      <td>'Coding density': 0.8042703237806546</td>\n",
       "      <td>'Translation table': 11</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>20</th>\n",
       "      <td>.18\\t{'GC': 0.4576707370234409</td>\n",
       "      <td>'GC std': 0.027942308920779816</td>\n",
       "      <td>'Genome size': 248950413</td>\n",
       "      <td>'# ambiguous bases': 0</td>\n",
       "      <td>'# scaffolds': 80173</td>\n",
       "      <td>'# contigs': 80173</td>\n",
       "      <td>'Longest scaffold': 29177</td>\n",
       "      <td>'Longest contig': 29177</td>\n",
       "      <td>'N50 (scaffolds)': 3309</td>\n",
       "      <td>'N50 (contigs)': 3309</td>\n",
       "      <td>'Mean scaffold length': 3105.165242662742</td>\n",
       "      <td>'Mean contig length': 3105.165242662742</td>\n",
       "      <td>'Coding density': 0.7416085949614392</td>\n",
       "      <td>'Translation table': 4</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>21</th>\n",
       "      <td>.16\\t{'GC': 0.38969172516798456</td>\n",
       "      <td>'GC std': 0.021485159645323455</td>\n",
       "      <td>'Genome size': 388694235</td>\n",
       "      <td>'# ambiguous bases': 0</td>\n",
       "      <td>'# scaffolds': 76893</td>\n",
       "      <td>'# contigs': 76893</td>\n",
       "      <td>'Longest scaffold': 69610</td>\n",
       "      <td>'Longest contig': 69610</td>\n",
       "      <td>'N50 (scaffolds)': 6735</td>\n",
       "      <td>'N50 (contigs)': 6735</td>\n",
       "      <td>'Mean scaffold length': 5055.001560610199</td>\n",
       "      <td>'Mean contig length': 5055.001560610199</td>\n",
       "      <td>'Coding density': 0.3805687393331162</td>\n",
       "      <td>'Translation table': 11</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "                                 1                                2   \\\n",
       "17    .7\\t{'GC': 0.3846045581267069    'GC std': 0.03400637013380865   \n",
       "18   .8\\t{'GC': 0.37689090505296374     'GC std': 0.0240623411465306   \n",
       "19    .9\\t{'GC': 0.3985752720619433    'GC std': 0.02356025011909446   \n",
       "20   .18\\t{'GC': 0.4576707370234409   'GC std': 0.027942308920779816   \n",
       "21  .16\\t{'GC': 0.38969172516798456   'GC std': 0.021485159645323455   \n",
       "\n",
       "                           3                        4                      5   \\\n",
       "17      'Genome size': 703403   '# ambiguous bases': 0     '# scaffolds': 385   \n",
       "18     'Genome size': 9595339   '# ambiguous bases': 0    '# scaffolds': 2995   \n",
       "19     'Genome size': 9585409   '# ambiguous bases': 0    '# scaffolds': 4007   \n",
       "20   'Genome size': 248950413   '# ambiguous bases': 0   '# scaffolds': 80173   \n",
       "21   'Genome size': 388694235   '# ambiguous bases': 0   '# scaffolds': 76893   \n",
       "\n",
       "                     6                           7                         8   \\\n",
       "17     '# contigs': 385    'Longest scaffold': 6914    'Longest contig': 6914   \n",
       "18    '# contigs': 2995   'Longest scaffold': 20426   'Longest contig': 20426   \n",
       "19    '# contigs': 4007   'Longest scaffold': 12780   'Longest contig': 12780   \n",
       "20   '# contigs': 80173   'Longest scaffold': 29177   'Longest contig': 29177   \n",
       "21   '# contigs': 76893   'Longest scaffold': 69610   'Longest contig': 69610   \n",
       "\n",
       "                          9                       10  \\\n",
       "17   'N50 (scaffolds)': 1722   'N50 (contigs)': 1722   \n",
       "18   'N50 (scaffolds)': 3535   'N50 (contigs)': 3535   \n",
       "19   'N50 (scaffolds)': 2392   'N50 (contigs)': 2392   \n",
       "20   'N50 (scaffolds)': 3309   'N50 (contigs)': 3309   \n",
       "21   'N50 (scaffolds)': 6735   'N50 (contigs)': 6735   \n",
       "\n",
       "                                             11  \\\n",
       "17   'Mean scaffold length': 1827.0207792207793   \n",
       "18    'Mean scaffold length': 3203.785976627713   \n",
       "19    'Mean scaffold length': 2392.165959570751   \n",
       "20    'Mean scaffold length': 3105.165242662742   \n",
       "21    'Mean scaffold length': 5055.001560610199   \n",
       "\n",
       "                                           12  \\\n",
       "17   'Mean contig length': 1827.0207792207793   \n",
       "18    'Mean contig length': 3203.785976627713   \n",
       "19    'Mean contig length': 2392.165959570751   \n",
       "20    'Mean contig length': 3105.165242662742   \n",
       "21    'Mean contig length': 5055.001560610199   \n",
       "\n",
       "                                       13                        14  \n",
       "17   'Coding density': 0.8291420423285087   'Translation table': 11  \n",
       "18   'Coding density': 0.3965412790522565   'Translation table': 11  \n",
       "19   'Coding density': 0.8042703237806546   'Translation table': 11  \n",
       "20   'Coding density': 0.7416085949614392    'Translation table': 4  \n",
       "21   'Coding density': 0.3805687393331162   'Translation table': 11  "
      ]
     },
     "execution_count": 193,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "# read in table with genome size\n",
    "binname_bin_stats=pd.read_csv('bin_stats.analyze.tsv',header=None,usecols=lambda x: x != 0)\n",
    "binname_bin_stats.tail(5)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 194,
   "id": "7584589b-3ab9-4bf0-84eb-0e35cbc68f80",
   "metadata": {},
   "outputs": [
    {
     "ename": "AttributeError",
     "evalue": "'DataFrame' object has no attribute 'str'",
     "output_type": "error",
     "traceback": [
      "\u001b[0;31m---------------------------------------------------------------------------\u001b[0m",
      "\u001b[0;31mAttributeError\u001b[0m                            Traceback (most recent call last)",
      "Cell \u001b[0;32mIn[194], line 3\u001b[0m\n\u001b[1;32m      1\u001b[0m \u001b[38;5;66;03m# split first column that contains bin id \u001b[39;00m\n\u001b[1;32m      2\u001b[0m binname_bin_stats[[\u001b[38;5;124m'\u001b[39m\u001b[38;5;124mBin_Id\u001b[39m\u001b[38;5;124m'\u001b[39m,\u001b[38;5;241m1\u001b[39m]]\u001b[38;5;241m=\u001b[39mbinname_bin_stats[\u001b[38;5;241m1\u001b[39m]\u001b[38;5;241m.\u001b[39mstr\u001b[38;5;241m.\u001b[39msplit(\u001b[38;5;124m'\u001b[39m\u001b[38;5;130;01m\\t\u001b[39;00m\u001b[38;5;124m{\u001b[39m\u001b[38;5;124m'\u001b[39m,expand\u001b[38;5;241m=\u001b[39m\u001b[38;5;28;01mTrue\u001b[39;00m)\n\u001b[0;32m----> 3\u001b[0m \u001b[43mbinname_bin_stats\u001b[49m\u001b[38;5;241;43m.\u001b[39;49m\u001b[43mstr\u001b[49m\u001b[38;5;241m.\u001b[39mextract(\u001b[38;5;124mr\u001b[39m\u001b[38;5;124m'\u001b[39m\u001b[38;5;124m(\u001b[39m\u001b[38;5;124m\\\u001b[39m\u001b[38;5;124m.\u001b[39m\u001b[38;5;124m\\\u001b[39m\u001b[38;5;124md)\u001b[39m\u001b[38;5;124m'\u001b[39m)\n",
      "File \u001b[0;32m~/.local/lib/python3.11/site-packages/pandas/core/generic.py:6204\u001b[0m, in \u001b[0;36mNDFrame.__getattr__\u001b[0;34m(self, name)\u001b[0m\n\u001b[1;32m   6197\u001b[0m \u001b[38;5;28;01mif\u001b[39;00m (\n\u001b[1;32m   6198\u001b[0m     name \u001b[38;5;129;01mnot\u001b[39;00m \u001b[38;5;129;01min\u001b[39;00m \u001b[38;5;28mself\u001b[39m\u001b[38;5;241m.\u001b[39m_internal_names_set\n\u001b[1;32m   6199\u001b[0m     \u001b[38;5;129;01mand\u001b[39;00m name \u001b[38;5;129;01mnot\u001b[39;00m \u001b[38;5;129;01min\u001b[39;00m \u001b[38;5;28mself\u001b[39m\u001b[38;5;241m.\u001b[39m_metadata\n\u001b[1;32m   6200\u001b[0m     \u001b[38;5;129;01mand\u001b[39;00m name \u001b[38;5;129;01mnot\u001b[39;00m \u001b[38;5;129;01min\u001b[39;00m \u001b[38;5;28mself\u001b[39m\u001b[38;5;241m.\u001b[39m_accessors\n\u001b[1;32m   6201\u001b[0m     \u001b[38;5;129;01mand\u001b[39;00m \u001b[38;5;28mself\u001b[39m\u001b[38;5;241m.\u001b[39m_info_axis\u001b[38;5;241m.\u001b[39m_can_hold_identifiers_and_holds_name(name)\n\u001b[1;32m   6202\u001b[0m ):\n\u001b[1;32m   6203\u001b[0m     \u001b[38;5;28;01mreturn\u001b[39;00m \u001b[38;5;28mself\u001b[39m[name]\n\u001b[0;32m-> 6204\u001b[0m \u001b[38;5;28;01mreturn\u001b[39;00m \u001b[38;5;28;43mobject\u001b[39;49m\u001b[38;5;241;43m.\u001b[39;49m\u001b[38;5;21;43m__getattribute__\u001b[39;49m\u001b[43m(\u001b[49m\u001b[38;5;28;43mself\u001b[39;49m\u001b[43m,\u001b[49m\u001b[43m \u001b[49m\u001b[43mname\u001b[49m\u001b[43m)\u001b[49m\n",
      "\u001b[0;31mAttributeError\u001b[0m: 'DataFrame' object has no attribute 'str'"
     ]
    }
   ],
   "source": [
    "# split first column that contains bin id \n",
    "binname_bin_stats[['Bin_Id',1]]=binname_bin_stats[1].str.split('\\t{',expand=True)\n",
    "binname_bin_stats.str.extract(r'(\\.\\d)')"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 190,
   "id": "b60d8e87-3be7-4258-8463-d79cf52ae58f",
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "22\n"
     ]
    },
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>1</th>\n",
       "      <th>2</th>\n",
       "      <th>3</th>\n",
       "      <th>4</th>\n",
       "      <th>5</th>\n",
       "      <th>6</th>\n",
       "      <th>7</th>\n",
       "      <th>8</th>\n",
       "      <th>9</th>\n",
       "      <th>10</th>\n",
       "      <th>11</th>\n",
       "      <th>12</th>\n",
       "      <th>13</th>\n",
       "      <th>14</th>\n",
       "      <th>Bin_Id</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>19</th>\n",
       "      <td>'GC': 0.3985752720619433</td>\n",
       "      <td>'GC std': 0.02356025011909446</td>\n",
       "      <td>'Genome size': 9585409</td>\n",
       "      <td>'# ambiguous bases': 0</td>\n",
       "      <td>'# scaffolds': 4007</td>\n",
       "      <td>'# contigs': 4007</td>\n",
       "      <td>'Longest scaffold': 12780</td>\n",
       "      <td>'Longest contig': 12780</td>\n",
       "      <td>'N50 (scaffolds)': 2392</td>\n",
       "      <td>'N50 (contigs)': 2392</td>\n",
       "      <td>'Mean scaffold length': 2392.165959570751</td>\n",
       "      <td>'Mean contig length': 2392.165959570751</td>\n",
       "      <td>'Coding density': 0.8042703237806546</td>\n",
       "      <td>'Translation table': 11</td>\n",
       "      <td>.9</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>20</th>\n",
       "      <td>'GC': 0.4576707370234409</td>\n",
       "      <td>'GC std': 0.027942308920779816</td>\n",
       "      <td>'Genome size': 248950413</td>\n",
       "      <td>'# ambiguous bases': 0</td>\n",
       "      <td>'# scaffolds': 80173</td>\n",
       "      <td>'# contigs': 80173</td>\n",
       "      <td>'Longest scaffold': 29177</td>\n",
       "      <td>'Longest contig': 29177</td>\n",
       "      <td>'N50 (scaffolds)': 3309</td>\n",
       "      <td>'N50 (contigs)': 3309</td>\n",
       "      <td>'Mean scaffold length': 3105.165242662742</td>\n",
       "      <td>'Mean contig length': 3105.165242662742</td>\n",
       "      <td>'Coding density': 0.7416085949614392</td>\n",
       "      <td>'Translation table': 4</td>\n",
       "      <td>.18</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>21</th>\n",
       "      <td>'GC': 0.38969172516798456</td>\n",
       "      <td>'GC std': 0.021485159645323455</td>\n",
       "      <td>'Genome size': 388694235</td>\n",
       "      <td>'# ambiguous bases': 0</td>\n",
       "      <td>'# scaffolds': 76893</td>\n",
       "      <td>'# contigs': 76893</td>\n",
       "      <td>'Longest scaffold': 69610</td>\n",
       "      <td>'Longest contig': 69610</td>\n",
       "      <td>'N50 (scaffolds)': 6735</td>\n",
       "      <td>'N50 (contigs)': 6735</td>\n",
       "      <td>'Mean scaffold length': 5055.001560610199</td>\n",
       "      <td>'Mean contig length': 5055.001560610199</td>\n",
       "      <td>'Coding density': 0.3805687393331162</td>\n",
       "      <td>'Translation table': 11</td>\n",
       "      <td>.16</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "                            1                                2  \\\n",
       "19   'GC': 0.3985752720619433    'GC std': 0.02356025011909446   \n",
       "20   'GC': 0.4576707370234409   'GC std': 0.027942308920779816   \n",
       "21  'GC': 0.38969172516798456   'GC std': 0.021485159645323455   \n",
       "\n",
       "                            3                        4                      5  \\\n",
       "19     'Genome size': 9585409   '# ambiguous bases': 0    '# scaffolds': 4007   \n",
       "20   'Genome size': 248950413   '# ambiguous bases': 0   '# scaffolds': 80173   \n",
       "21   'Genome size': 388694235   '# ambiguous bases': 0   '# scaffolds': 76893   \n",
       "\n",
       "                      6                           7                         8  \\\n",
       "19    '# contigs': 4007   'Longest scaffold': 12780   'Longest contig': 12780   \n",
       "20   '# contigs': 80173   'Longest scaffold': 29177   'Longest contig': 29177   \n",
       "21   '# contigs': 76893   'Longest scaffold': 69610   'Longest contig': 69610   \n",
       "\n",
       "                           9                      10  \\\n",
       "19   'N50 (scaffolds)': 2392   'N50 (contigs)': 2392   \n",
       "20   'N50 (scaffolds)': 3309   'N50 (contigs)': 3309   \n",
       "21   'N50 (scaffolds)': 6735   'N50 (contigs)': 6735   \n",
       "\n",
       "                                            11  \\\n",
       "19   'Mean scaffold length': 2392.165959570751   \n",
       "20   'Mean scaffold length': 3105.165242662742   \n",
       "21   'Mean scaffold length': 5055.001560610199   \n",
       "\n",
       "                                          12  \\\n",
       "19   'Mean contig length': 2392.165959570751   \n",
       "20   'Mean contig length': 3105.165242662742   \n",
       "21   'Mean contig length': 5055.001560610199   \n",
       "\n",
       "                                       13                        14 Bin_Id  \n",
       "19   'Coding density': 0.8042703237806546   'Translation table': 11     .9  \n",
       "20   'Coding density': 0.7416085949614392    'Translation table': 4    .18  \n",
       "21   'Coding density': 0.3805687393331162   'Translation table': 11    .16  "
      ]
     },
     "execution_count": 190,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "print(len(binname_bin_stats))\n",
    "binname_bin_stats.tail(3)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 84,
   "id": "f343a140-0de5-49da-ad49-57f13481b660",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>1</th>\n",
       "      <th>2</th>\n",
       "      <th>3</th>\n",
       "      <th>4</th>\n",
       "      <th>5</th>\n",
       "      <th>6</th>\n",
       "      <th>7</th>\n",
       "      <th>8</th>\n",
       "      <th>9</th>\n",
       "      <th>10</th>\n",
       "      <th>11</th>\n",
       "      <th>12</th>\n",
       "      <th>13</th>\n",
       "      <th>14</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>'GC': 0.45383397689109817</td>\n",
       "      <td>'GC std': 0.017741057662292045</td>\n",
       "      <td>'Genome size': 67593</td>\n",
       "      <td>'# ambiguous bases': 0</td>\n",
       "      <td>'# scaffolds': 49</td>\n",
       "      <td>'# contigs': 49</td>\n",
       "      <td>'Longest scaffold': 2600</td>\n",
       "      <td>'Longest contig': 2600</td>\n",
       "      <td>'N50 (scaffolds)': 1396</td>\n",
       "      <td>'N50 (contigs)': 1396</td>\n",
       "      <td>'Mean scaffold length': 1379.4489795918366</td>\n",
       "      <td>'Mean contig length': 1379.4489795918366</td>\n",
       "      <td>'Coding density': 0.4126018966461024</td>\n",
       "      <td>'Translation table': 4</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>'GC': 0.47662485462908316</td>\n",
       "      <td>'GC std': 0.01813517400388053</td>\n",
       "      <td>'Genome size': 401559</td>\n",
       "      <td>'# ambiguous bases': 0</td>\n",
       "      <td>'# scaffolds': 177</td>\n",
       "      <td>'# contigs': 177</td>\n",
       "      <td>'Longest scaffold': 8573</td>\n",
       "      <td>'Longest contig': 8573</td>\n",
       "      <td>'N50 (scaffolds)': 2459</td>\n",
       "      <td>'N50 (contigs)': 2459</td>\n",
       "      <td>'Mean scaffold length': 2268.6949152542375</td>\n",
       "      <td>'Mean contig length': 2268.6949152542375</td>\n",
       "      <td>'Coding density': 0.8767927004499961</td>\n",
       "      <td>'Translation table': 11</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>'GC': 0.4231691853348191</td>\n",
       "      <td>'GC std': 0.03386657292470408</td>\n",
       "      <td>'Genome size': 182173</td>\n",
       "      <td>'# ambiguous bases': 0</td>\n",
       "      <td>'# scaffolds': 128</td>\n",
       "      <td>'# contigs': 128</td>\n",
       "      <td>'Longest scaffold': 2808</td>\n",
       "      <td>'Longest contig': 2808</td>\n",
       "      <td>'N50 (scaffolds)': 1419</td>\n",
       "      <td>'N50 (contigs)': 1419</td>\n",
       "      <td>'Mean scaffold length': 1423.2265625</td>\n",
       "      <td>'Mean contig length': 1423.2265625</td>\n",
       "      <td>'Coding density': 0.49618220043584943</td>\n",
       "      <td>'Translation table': 11</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>'GC': 0.4045708465788093</td>\n",
       "      <td>'GC std': 0.04412584721709139</td>\n",
       "      <td>'Genome size': 194406</td>\n",
       "      <td>'# ambiguous bases': 0</td>\n",
       "      <td>'# scaffolds': 135</td>\n",
       "      <td>'# contigs': 135</td>\n",
       "      <td>'Longest scaffold': 3275</td>\n",
       "      <td>'Longest contig': 3275</td>\n",
       "      <td>'N50 (scaffolds)': 1367</td>\n",
       "      <td>'N50 (contigs)': 1367</td>\n",
       "      <td>'Mean scaffold length': 1440.0444444444445</td>\n",
       "      <td>'Mean contig length': 1440.0444444444445</td>\n",
       "      <td>'Coding density': 0.7444266123473555</td>\n",
       "      <td>'Translation table': 4</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>'GC': 0.33128435782108945</td>\n",
       "      <td>'GC std': 0.052705820586636025</td>\n",
       "      <td>'Genome size': 80040</td>\n",
       "      <td>'# ambiguous bases': 0</td>\n",
       "      <td>'# scaffolds': 45</td>\n",
       "      <td>'# contigs': 45</td>\n",
       "      <td>'Longest scaffold': 18131</td>\n",
       "      <td>'Longest contig': 18131</td>\n",
       "      <td>'N50 (scaffolds)': 1540</td>\n",
       "      <td>'N50 (contigs)': 1540</td>\n",
       "      <td>'Mean scaffold length': 1778.6666666666667</td>\n",
       "      <td>'Mean contig length': 1778.6666666666667</td>\n",
       "      <td>'Coding density': 0.6600949525237382</td>\n",
       "      <td>'Translation table': 11</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>184</th>\n",
       "      <td>'GC': 0.41145448317168515</td>\n",
       "      <td>'GC std': 0.028191213036258277</td>\n",
       "      <td>'Genome size': 23152437</td>\n",
       "      <td>'# ambiguous bases': 0</td>\n",
       "      <td>'# scaffolds': 10584</td>\n",
       "      <td>'# contigs': 10584</td>\n",
       "      <td>'Longest scaffold': 8288</td>\n",
       "      <td>'Longest contig': 8288</td>\n",
       "      <td>'N50 (scaffolds)': 2330</td>\n",
       "      <td>'N50 (contigs)': 2330</td>\n",
       "      <td>'Mean scaffold length': 2187.4940476190477</td>\n",
       "      <td>'Mean contig length': 2187.4940476190477</td>\n",
       "      <td>'Coding density': 0.3832219044586969</td>\n",
       "      <td>'Translation table': 11</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>185</th>\n",
       "      <td>'GC': 0.41414145492114773</td>\n",
       "      <td>'GC std': 0.018241284840074274</td>\n",
       "      <td>'Genome size': 104032662</td>\n",
       "      <td>'# ambiguous bases': 0</td>\n",
       "      <td>'# scaffolds': 22865</td>\n",
       "      <td>'# contigs': 22865</td>\n",
       "      <td>'Longest scaffold': 29540</td>\n",
       "      <td>'Longest contig': 29540</td>\n",
       "      <td>'N50 (scaffolds)': 4709</td>\n",
       "      <td>'N50 (contigs)': 4709</td>\n",
       "      <td>'Mean scaffold length': 4549.864946424666</td>\n",
       "      <td>'Mean contig length': 4549.864946424666</td>\n",
       "      <td>'Coding density': 0.1972559444840506</td>\n",
       "      <td>'Translation table': 11</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>186</th>\n",
       "      <td>'GC': 0.35429501336742997</td>\n",
       "      <td>'GC std': 0.04238931071808634</td>\n",
       "      <td>'Genome size': 17206</td>\n",
       "      <td>'# ambiguous bases': 0</td>\n",
       "      <td>'# scaffolds': 12</td>\n",
       "      <td>'# contigs': 12</td>\n",
       "      <td>'Longest scaffold': 3072</td>\n",
       "      <td>'Longest contig': 3072</td>\n",
       "      <td>'N50 (scaffolds)': 1332</td>\n",
       "      <td>'N50 (contigs)': 1332</td>\n",
       "      <td>'Mean scaffold length': 1433.8333333333333</td>\n",
       "      <td>'Mean contig length': 1433.8333333333333</td>\n",
       "      <td>'Coding density': 0.25543415087760085</td>\n",
       "      <td>'Translation table': 11</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>187</th>\n",
       "      <td>'GC': 0.4343907050953209</td>\n",
       "      <td>'GC std': 0.03418430197601377</td>\n",
       "      <td>'Genome size': 128714994</td>\n",
       "      <td>'# ambiguous bases': 0</td>\n",
       "      <td>'# scaffolds': 67929</td>\n",
       "      <td>'# contigs': 67929</td>\n",
       "      <td>'Longest scaffold': 8240</td>\n",
       "      <td>'Longest contig': 8240</td>\n",
       "      <td>'N50 (scaffolds)': 2022</td>\n",
       "      <td>'N50 (contigs)': 2022</td>\n",
       "      <td>'Mean scaffold length': 1894.8460009716027</td>\n",
       "      <td>'Mean contig length': 1894.8460009716027</td>\n",
       "      <td>'Coding density': 0.5431238958842666</td>\n",
       "      <td>'Translation table': 11</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>188</th>\n",
       "      <td>'GC': 0.39014272558076646</td>\n",
       "      <td>'GC std': 0.027089530386056247</td>\n",
       "      <td>'Genome size': 197017240</td>\n",
       "      <td>'# ambiguous bases': 0</td>\n",
       "      <td>'# scaffolds': 81472</td>\n",
       "      <td>'# contigs': 81472</td>\n",
       "      <td>'Longest scaffold': 43574</td>\n",
       "      <td>'Longest contig': 43574</td>\n",
       "      <td>'N50 (scaffolds)': 2832</td>\n",
       "      <td>'N50 (contigs)': 2832</td>\n",
       "      <td>'Mean scaffold length': 2418.2202474469755</td>\n",
       "      <td>'Mean contig length': 2418.2202474469755</td>\n",
       "      <td>'Coding density': 0.40279689736796637</td>\n",
       "      <td>'Translation table': 11</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>189 rows × 14 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "                             1                                2  \\\n",
       "0    'GC': 0.45383397689109817   'GC std': 0.017741057662292045   \n",
       "1    'GC': 0.47662485462908316    'GC std': 0.01813517400388053   \n",
       "2     'GC': 0.4231691853348191    'GC std': 0.03386657292470408   \n",
       "3     'GC': 0.4045708465788093    'GC std': 0.04412584721709139   \n",
       "4    'GC': 0.33128435782108945   'GC std': 0.052705820586636025   \n",
       "..                         ...                              ...   \n",
       "184  'GC': 0.41145448317168515   'GC std': 0.028191213036258277   \n",
       "185  'GC': 0.41414145492114773   'GC std': 0.018241284840074274   \n",
       "186  'GC': 0.35429501336742997    'GC std': 0.04238931071808634   \n",
       "187   'GC': 0.4343907050953209    'GC std': 0.03418430197601377   \n",
       "188  'GC': 0.39014272558076646   'GC std': 0.027089530386056247   \n",
       "\n",
       "                             3                        4  \\\n",
       "0         'Genome size': 67593   '# ambiguous bases': 0   \n",
       "1        'Genome size': 401559   '# ambiguous bases': 0   \n",
       "2        'Genome size': 182173   '# ambiguous bases': 0   \n",
       "3        'Genome size': 194406   '# ambiguous bases': 0   \n",
       "4         'Genome size': 80040   '# ambiguous bases': 0   \n",
       "..                         ...                      ...   \n",
       "184    'Genome size': 23152437   '# ambiguous bases': 0   \n",
       "185   'Genome size': 104032662   '# ambiguous bases': 0   \n",
       "186       'Genome size': 17206   '# ambiguous bases': 0   \n",
       "187   'Genome size': 128714994   '# ambiguous bases': 0   \n",
       "188   'Genome size': 197017240   '# ambiguous bases': 0   \n",
       "\n",
       "                         5                    6                           7  \\\n",
       "0        '# scaffolds': 49      '# contigs': 49    'Longest scaffold': 2600   \n",
       "1       '# scaffolds': 177     '# contigs': 177    'Longest scaffold': 8573   \n",
       "2       '# scaffolds': 128     '# contigs': 128    'Longest scaffold': 2808   \n",
       "3       '# scaffolds': 135     '# contigs': 135    'Longest scaffold': 3275   \n",
       "4        '# scaffolds': 45      '# contigs': 45   'Longest scaffold': 18131   \n",
       "..                     ...                  ...                         ...   \n",
       "184   '# scaffolds': 10584   '# contigs': 10584    'Longest scaffold': 8288   \n",
       "185   '# scaffolds': 22865   '# contigs': 22865   'Longest scaffold': 29540   \n",
       "186      '# scaffolds': 12      '# contigs': 12    'Longest scaffold': 3072   \n",
       "187   '# scaffolds': 67929   '# contigs': 67929    'Longest scaffold': 8240   \n",
       "188   '# scaffolds': 81472   '# contigs': 81472   'Longest scaffold': 43574   \n",
       "\n",
       "                            8                         9  \\\n",
       "0      'Longest contig': 2600   'N50 (scaffolds)': 1396   \n",
       "1      'Longest contig': 8573   'N50 (scaffolds)': 2459   \n",
       "2      'Longest contig': 2808   'N50 (scaffolds)': 1419   \n",
       "3      'Longest contig': 3275   'N50 (scaffolds)': 1367   \n",
       "4     'Longest contig': 18131   'N50 (scaffolds)': 1540   \n",
       "..                        ...                       ...   \n",
       "184    'Longest contig': 8288   'N50 (scaffolds)': 2330   \n",
       "185   'Longest contig': 29540   'N50 (scaffolds)': 4709   \n",
       "186    'Longest contig': 3072   'N50 (scaffolds)': 1332   \n",
       "187    'Longest contig': 8240   'N50 (scaffolds)': 2022   \n",
       "188   'Longest contig': 43574   'N50 (scaffolds)': 2832   \n",
       "\n",
       "                         10                                           11  \\\n",
       "0     'N50 (contigs)': 1396   'Mean scaffold length': 1379.4489795918366   \n",
       "1     'N50 (contigs)': 2459   'Mean scaffold length': 2268.6949152542375   \n",
       "2     'N50 (contigs)': 1419         'Mean scaffold length': 1423.2265625   \n",
       "3     'N50 (contigs)': 1367   'Mean scaffold length': 1440.0444444444445   \n",
       "4     'N50 (contigs)': 1540   'Mean scaffold length': 1778.6666666666667   \n",
       "..                      ...                                          ...   \n",
       "184   'N50 (contigs)': 2330   'Mean scaffold length': 2187.4940476190477   \n",
       "185   'N50 (contigs)': 4709    'Mean scaffold length': 4549.864946424666   \n",
       "186   'N50 (contigs)': 1332   'Mean scaffold length': 1433.8333333333333   \n",
       "187   'N50 (contigs)': 2022   'Mean scaffold length': 1894.8460009716027   \n",
       "188   'N50 (contigs)': 2832   'Mean scaffold length': 2418.2202474469755   \n",
       "\n",
       "                                            12  \\\n",
       "0     'Mean contig length': 1379.4489795918366   \n",
       "1     'Mean contig length': 2268.6949152542375   \n",
       "2           'Mean contig length': 1423.2265625   \n",
       "3     'Mean contig length': 1440.0444444444445   \n",
       "4     'Mean contig length': 1778.6666666666667   \n",
       "..                                         ...   \n",
       "184   'Mean contig length': 2187.4940476190477   \n",
       "185    'Mean contig length': 4549.864946424666   \n",
       "186   'Mean contig length': 1433.8333333333333   \n",
       "187   'Mean contig length': 1894.8460009716027   \n",
       "188   'Mean contig length': 2418.2202474469755   \n",
       "\n",
       "                                         13                        14  \n",
       "0      'Coding density': 0.4126018966461024    'Translation table': 4  \n",
       "1      'Coding density': 0.8767927004499961   'Translation table': 11  \n",
       "2     'Coding density': 0.49618220043584943   'Translation table': 11  \n",
       "3      'Coding density': 0.7444266123473555    'Translation table': 4  \n",
       "4      'Coding density': 0.6600949525237382   'Translation table': 11  \n",
       "..                                      ...                       ...  \n",
       "184    'Coding density': 0.3832219044586969   'Translation table': 11  \n",
       "185    'Coding density': 0.1972559444840506   'Translation table': 11  \n",
       "186   'Coding density': 0.25543415087760085   'Translation table': 11  \n",
       "187    'Coding density': 0.5431238958842666   'Translation table': 11  \n",
       "188   'Coding density': 0.40279689736796637   'Translation table': 11  \n",
       "\n",
       "[189 rows x 14 columns]"
      ]
     },
     "execution_count": 84,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "# remove bin id col for now, and remove 'healthy_2019_mcav_contigs_cut' bin \n",
    "extract_df=binname_bin_stats.iloc[0:189,0:14]\n",
    "extract_df"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 86,
   "id": "823d05f7-7010-453d-abec-112e2ce713e3",
   "metadata": {},
   "outputs": [],
   "source": [
    "# in first row, split by : and keep left side for header names\n",
    "#first_row= extract_df.iloc[0].str.split(':', expand=True)\n",
    "#first_row"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 87,
   "id": "1c5549c0-500e-4a02-a432-7b393bdc3740",
   "metadata": {},
   "outputs": [],
   "source": [
    "#first_row[0]"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 88,
   "id": "0f21cd1c-b3ea-4491-a366-3a2a1b3acd49",
   "metadata": {},
   "outputs": [],
   "source": [
    "#first_row.to_dict()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 90,
   "id": "eadceaf5-63d3-4211-a58c-ac560b256f07",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "1                       'GC': 0.47662485462908316\n",
       "2                   'GC std': 0.01813517400388053\n",
       "3                           'Genome size': 401559\n",
       "4                          '# ambiguous bases': 0\n",
       "5                              '# scaffolds': 177\n",
       "6                                '# contigs': 177\n",
       "7                        'Longest scaffold': 8573\n",
       "8                          'Longest contig': 8573\n",
       "9                         'N50 (scaffolds)': 2459\n",
       "10                          'N50 (contigs)': 2459\n",
       "11     'Mean scaffold length': 2268.6949152542375\n",
       "12       'Mean contig length': 2268.6949152542375\n",
       "13           'Coding density': 0.8767927004499961\n",
       "14                        'Translation table': 11\n",
       "Name: 1, dtype: object"
      ]
     },
     "execution_count": 90,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "extract_df.iloc[1]"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 91,
   "id": "a8ada1f6-1c3c-411f-9f2b-23d8e294e5a6",
   "metadata": {},
   "outputs": [],
   "source": [
    "# in first col of new df: split by \":\" and get left side for col names \n",
    "new=extract_df.iloc[1].str.split(':', expand=True)\n",
    "index=new[0]\n",
    "cols=index.to_list()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 92,
   "id": "9d6e4c4c-2b99-47eb-9020-29a9001e5cf8",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>'GC'</th>\n",
       "      <th>'GC std'</th>\n",
       "      <th>'Genome size'</th>\n",
       "      <th>'# ambiguous bases'</th>\n",
       "      <th>'# scaffolds'</th>\n",
       "      <th>'# contigs'</th>\n",
       "      <th>'Longest scaffold'</th>\n",
       "      <th>'Longest contig'</th>\n",
       "      <th>'N50 (scaffolds)'</th>\n",
       "      <th>'N50 (contigs)'</th>\n",
       "      <th>'Mean scaffold length'</th>\n",
       "      <th>'Mean contig length'</th>\n",
       "      <th>'Coding density'</th>\n",
       "      <th>'Translation table'</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>184</th>\n",
       "      <td>0.41145448317168515</td>\n",
       "      <td>0.028191213036258277</td>\n",
       "      <td>23152437</td>\n",
       "      <td>0</td>\n",
       "      <td>10584</td>\n",
       "      <td>10584</td>\n",
       "      <td>8288</td>\n",
       "      <td>8288</td>\n",
       "      <td>2330</td>\n",
       "      <td>2330</td>\n",
       "      <td>2187.4940476190477</td>\n",
       "      <td>2187.4940476190477</td>\n",
       "      <td>0.3832219044586969</td>\n",
       "      <td>11</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>185</th>\n",
       "      <td>0.41414145492114773</td>\n",
       "      <td>0.018241284840074274</td>\n",
       "      <td>104032662</td>\n",
       "      <td>0</td>\n",
       "      <td>22865</td>\n",
       "      <td>22865</td>\n",
       "      <td>29540</td>\n",
       "      <td>29540</td>\n",
       "      <td>4709</td>\n",
       "      <td>4709</td>\n",
       "      <td>4549.864946424666</td>\n",
       "      <td>4549.864946424666</td>\n",
       "      <td>0.1972559444840506</td>\n",
       "      <td>11</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>186</th>\n",
       "      <td>0.35429501336742997</td>\n",
       "      <td>0.04238931071808634</td>\n",
       "      <td>17206</td>\n",
       "      <td>0</td>\n",
       "      <td>12</td>\n",
       "      <td>12</td>\n",
       "      <td>3072</td>\n",
       "      <td>3072</td>\n",
       "      <td>1332</td>\n",
       "      <td>1332</td>\n",
       "      <td>1433.8333333333333</td>\n",
       "      <td>1433.8333333333333</td>\n",
       "      <td>0.25543415087760085</td>\n",
       "      <td>11</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>187</th>\n",
       "      <td>0.4343907050953209</td>\n",
       "      <td>0.03418430197601377</td>\n",
       "      <td>128714994</td>\n",
       "      <td>0</td>\n",
       "      <td>67929</td>\n",
       "      <td>67929</td>\n",
       "      <td>8240</td>\n",
       "      <td>8240</td>\n",
       "      <td>2022</td>\n",
       "      <td>2022</td>\n",
       "      <td>1894.8460009716027</td>\n",
       "      <td>1894.8460009716027</td>\n",
       "      <td>0.5431238958842666</td>\n",
       "      <td>11</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>188</th>\n",
       "      <td>0.39014272558076646</td>\n",
       "      <td>0.027089530386056247</td>\n",
       "      <td>197017240</td>\n",
       "      <td>0</td>\n",
       "      <td>81472</td>\n",
       "      <td>81472</td>\n",
       "      <td>43574</td>\n",
       "      <td>43574</td>\n",
       "      <td>2832</td>\n",
       "      <td>2832</td>\n",
       "      <td>2418.2202474469755</td>\n",
       "      <td>2418.2202474469755</td>\n",
       "      <td>0.40279689736796637</td>\n",
       "      <td>11</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "                     'GC'               'GC std'  'Genome size'  \\\n",
       "184   0.41145448317168515   0.028191213036258277       23152437   \n",
       "185   0.41414145492114773   0.018241284840074274      104032662   \n",
       "186   0.35429501336742997    0.04238931071808634          17206   \n",
       "187    0.4343907050953209    0.03418430197601377      128714994   \n",
       "188   0.39014272558076646   0.027089530386056247      197017240   \n",
       "\n",
       "     '# ambiguous bases'  '# scaffolds'  '# contigs'  'Longest scaffold'  \\\n",
       "184                    0          10584        10584                8288   \n",
       "185                    0          22865        22865               29540   \n",
       "186                    0             12           12                3072   \n",
       "187                    0          67929        67929                8240   \n",
       "188                    0          81472        81472               43574   \n",
       "\n",
       "     'Longest contig'  'N50 (scaffolds)'  'N50 (contigs)'  \\\n",
       "184              8288               2330             2330   \n",
       "185             29540               4709             4709   \n",
       "186              3072               1332             1332   \n",
       "187              8240               2022             2022   \n",
       "188             43574               2832             2832   \n",
       "\n",
       "     'Mean scaffold length'  'Mean contig length'      'Coding density'  \\\n",
       "184      2187.4940476190477    2187.4940476190477    0.3832219044586969   \n",
       "185       4549.864946424666     4549.864946424666    0.1972559444840506   \n",
       "186      1433.8333333333333    1433.8333333333333   0.25543415087760085   \n",
       "187      1894.8460009716027    1894.8460009716027    0.5431238958842666   \n",
       "188      2418.2202474469755    2418.2202474469755   0.40279689736796637   \n",
       "\n",
       "     'Translation table'  \n",
       "184                   11  \n",
       "185                   11  \n",
       "186                   11  \n",
       "187                   11  \n",
       "188                   11  "
      ]
     },
     "execution_count": 92,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "# loop through and extract the right of : in each row, add to dictionary, readd col names, covnert to df \n",
    "d={}\n",
    "for i in extract_df.index:\n",
    "    extracted_values = extract_df.iloc[i].str.extract(r'(?<=:)(.*)', expand=True)\n",
    "    #print(extracted_values)\n",
    "    list_of_values=extracted_values[0]\n",
    "    lists=list_of_values.to_list()\n",
    "    d[i] = lists\n",
    "d\n",
    "intermediate=pd.DataFrame.from_dict(d,orient='index')\n",
    "intermediate.columns=cols\n",
    "intermediate.tail()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 95,
   "id": "f93af724-b985-4189-9cf5-7e306e21201b",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>'GC'</th>\n",
       "      <th>'GC std'</th>\n",
       "      <th>'Genome size'</th>\n",
       "      <th>'# ambiguous bases'</th>\n",
       "      <th>'# scaffolds'</th>\n",
       "      <th>'# contigs'</th>\n",
       "      <th>'Longest scaffold'</th>\n",
       "      <th>'Longest contig'</th>\n",
       "      <th>'N50 (scaffolds)'</th>\n",
       "      <th>'N50 (contigs)'</th>\n",
       "      <th>'Mean scaffold length'</th>\n",
       "      <th>'Mean contig length'</th>\n",
       "      <th>'Coding density'</th>\n",
       "      <th>'Translation table'</th>\n",
       "      <th>Bin_Id</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>0.45383397689109817</td>\n",
       "      <td>0.017741057662292045</td>\n",
       "      <td>67593</td>\n",
       "      <td>0</td>\n",
       "      <td>49</td>\n",
       "      <td>49</td>\n",
       "      <td>2600</td>\n",
       "      <td>2600</td>\n",
       "      <td>1396</td>\n",
       "      <td>1396</td>\n",
       "      <td>1379.4489795918366</td>\n",
       "      <td>1379.4489795918366</td>\n",
       "      <td>0.4126018966461024</td>\n",
       "      <td>4</td>\n",
       "      <td>10</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>0.47662485462908316</td>\n",
       "      <td>0.01813517400388053</td>\n",
       "      <td>401559</td>\n",
       "      <td>0</td>\n",
       "      <td>177</td>\n",
       "      <td>177</td>\n",
       "      <td>8573</td>\n",
       "      <td>8573</td>\n",
       "      <td>2459</td>\n",
       "      <td>2459</td>\n",
       "      <td>2268.6949152542375</td>\n",
       "      <td>2268.6949152542375</td>\n",
       "      <td>0.8767927004499961</td>\n",
       "      <td>11</td>\n",
       "      <td>1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>0.4231691853348191</td>\n",
       "      <td>0.03386657292470408</td>\n",
       "      <td>182173</td>\n",
       "      <td>0</td>\n",
       "      <td>128</td>\n",
       "      <td>128</td>\n",
       "      <td>2808</td>\n",
       "      <td>2808</td>\n",
       "      <td>1419</td>\n",
       "      <td>1419</td>\n",
       "      <td>1423.2265625</td>\n",
       "      <td>1423.2265625</td>\n",
       "      <td>0.49618220043584943</td>\n",
       "      <td>11</td>\n",
       "      <td>101</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>0.4045708465788093</td>\n",
       "      <td>0.04412584721709139</td>\n",
       "      <td>194406</td>\n",
       "      <td>0</td>\n",
       "      <td>135</td>\n",
       "      <td>135</td>\n",
       "      <td>3275</td>\n",
       "      <td>3275</td>\n",
       "      <td>1367</td>\n",
       "      <td>1367</td>\n",
       "      <td>1440.0444444444445</td>\n",
       "      <td>1440.0444444444445</td>\n",
       "      <td>0.7444266123473555</td>\n",
       "      <td>4</td>\n",
       "      <td>102</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>0.33128435782108945</td>\n",
       "      <td>0.052705820586636025</td>\n",
       "      <td>80040</td>\n",
       "      <td>0</td>\n",
       "      <td>45</td>\n",
       "      <td>45</td>\n",
       "      <td>18131</td>\n",
       "      <td>18131</td>\n",
       "      <td>1540</td>\n",
       "      <td>1540</td>\n",
       "      <td>1778.6666666666667</td>\n",
       "      <td>1778.6666666666667</td>\n",
       "      <td>0.6600949525237382</td>\n",
       "      <td>11</td>\n",
       "      <td>100</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>184</th>\n",
       "      <td>0.41145448317168515</td>\n",
       "      <td>0.028191213036258277</td>\n",
       "      <td>23152437</td>\n",
       "      <td>0</td>\n",
       "      <td>10584</td>\n",
       "      <td>10584</td>\n",
       "      <td>8288</td>\n",
       "      <td>8288</td>\n",
       "      <td>2330</td>\n",
       "      <td>2330</td>\n",
       "      <td>2187.4940476190477</td>\n",
       "      <td>2187.4940476190477</td>\n",
       "      <td>0.3832219044586969</td>\n",
       "      <td>11</td>\n",
       "      <td>97</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>185</th>\n",
       "      <td>0.41414145492114773</td>\n",
       "      <td>0.018241284840074274</td>\n",
       "      <td>104032662</td>\n",
       "      <td>0</td>\n",
       "      <td>22865</td>\n",
       "      <td>22865</td>\n",
       "      <td>29540</td>\n",
       "      <td>29540</td>\n",
       "      <td>4709</td>\n",
       "      <td>4709</td>\n",
       "      <td>4549.864946424666</td>\n",
       "      <td>4549.864946424666</td>\n",
       "      <td>0.1972559444840506</td>\n",
       "      <td>11</td>\n",
       "      <td>94</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>186</th>\n",
       "      <td>0.35429501336742997</td>\n",
       "      <td>0.04238931071808634</td>\n",
       "      <td>17206</td>\n",
       "      <td>0</td>\n",
       "      <td>12</td>\n",
       "      <td>12</td>\n",
       "      <td>3072</td>\n",
       "      <td>3072</td>\n",
       "      <td>1332</td>\n",
       "      <td>1332</td>\n",
       "      <td>1433.8333333333333</td>\n",
       "      <td>1433.8333333333333</td>\n",
       "      <td>0.25543415087760085</td>\n",
       "      <td>11</td>\n",
       "      <td>99</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>187</th>\n",
       "      <td>0.4343907050953209</td>\n",
       "      <td>0.03418430197601377</td>\n",
       "      <td>128714994</td>\n",
       "      <td>0</td>\n",
       "      <td>67929</td>\n",
       "      <td>67929</td>\n",
       "      <td>8240</td>\n",
       "      <td>8240</td>\n",
       "      <td>2022</td>\n",
       "      <td>2022</td>\n",
       "      <td>1894.8460009716027</td>\n",
       "      <td>1894.8460009716027</td>\n",
       "      <td>0.5431238958842666</td>\n",
       "      <td>11</td>\n",
       "      <td>84</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>188</th>\n",
       "      <td>0.39014272558076646</td>\n",
       "      <td>0.027089530386056247</td>\n",
       "      <td>197017240</td>\n",
       "      <td>0</td>\n",
       "      <td>81472</td>\n",
       "      <td>81472</td>\n",
       "      <td>43574</td>\n",
       "      <td>43574</td>\n",
       "      <td>2832</td>\n",
       "      <td>2832</td>\n",
       "      <td>2418.2202474469755</td>\n",
       "      <td>2418.2202474469755</td>\n",
       "      <td>0.40279689736796637</td>\n",
       "      <td>11</td>\n",
       "      <td>98</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>189 rows × 15 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "                     'GC'               'GC std'  'Genome size'  \\\n",
       "0     0.45383397689109817   0.017741057662292045          67593   \n",
       "1     0.47662485462908316    0.01813517400388053         401559   \n",
       "2      0.4231691853348191    0.03386657292470408         182173   \n",
       "3      0.4045708465788093    0.04412584721709139         194406   \n",
       "4     0.33128435782108945   0.052705820586636025          80040   \n",
       "..                    ...                    ...            ...   \n",
       "184   0.41145448317168515   0.028191213036258277       23152437   \n",
       "185   0.41414145492114773   0.018241284840074274      104032662   \n",
       "186   0.35429501336742997    0.04238931071808634          17206   \n",
       "187    0.4343907050953209    0.03418430197601377      128714994   \n",
       "188   0.39014272558076646   0.027089530386056247      197017240   \n",
       "\n",
       "     '# ambiguous bases'  '# scaffolds'  '# contigs'  'Longest scaffold'  \\\n",
       "0                      0             49           49                2600   \n",
       "1                      0            177          177                8573   \n",
       "2                      0            128          128                2808   \n",
       "3                      0            135          135                3275   \n",
       "4                      0             45           45               18131   \n",
       "..                   ...            ...          ...                 ...   \n",
       "184                    0          10584        10584                8288   \n",
       "185                    0          22865        22865               29540   \n",
       "186                    0             12           12                3072   \n",
       "187                    0          67929        67929                8240   \n",
       "188                    0          81472        81472               43574   \n",
       "\n",
       "     'Longest contig'  'N50 (scaffolds)'  'N50 (contigs)'  \\\n",
       "0                2600               1396             1396   \n",
       "1                8573               2459             2459   \n",
       "2                2808               1419             1419   \n",
       "3                3275               1367             1367   \n",
       "4               18131               1540             1540   \n",
       "..                ...                ...              ...   \n",
       "184              8288               2330             2330   \n",
       "185             29540               4709             4709   \n",
       "186              3072               1332             1332   \n",
       "187              8240               2022             2022   \n",
       "188             43574               2832             2832   \n",
       "\n",
       "     'Mean scaffold length'  'Mean contig length'      'Coding density'  \\\n",
       "0        1379.4489795918366    1379.4489795918366    0.4126018966461024   \n",
       "1        2268.6949152542375    2268.6949152542375    0.8767927004499961   \n",
       "2              1423.2265625          1423.2265625   0.49618220043584943   \n",
       "3        1440.0444444444445    1440.0444444444445    0.7444266123473555   \n",
       "4        1778.6666666666667    1778.6666666666667    0.6600949525237382   \n",
       "..                      ...                   ...                   ...   \n",
       "184      2187.4940476190477    2187.4940476190477    0.3832219044586969   \n",
       "185       4549.864946424666     4549.864946424666    0.1972559444840506   \n",
       "186      1433.8333333333333    1433.8333333333333   0.25543415087760085   \n",
       "187      1894.8460009716027    1894.8460009716027    0.5431238958842666   \n",
       "188      2418.2202474469755    2418.2202474469755   0.40279689736796637   \n",
       "\n",
       "     'Translation table' Bin_Id  \n",
       "0                      4     10  \n",
       "1                     11      1  \n",
       "2                     11    101  \n",
       "3                      4    102  \n",
       "4                     11    100  \n",
       "..                   ...    ...  \n",
       "184                   11     97  \n",
       "185                   11     94  \n",
       "186                   11     99  \n",
       "187                   11     84  \n",
       "188                   11     98  \n",
       "\n",
       "[189 rows x 15 columns]"
      ]
     },
     "execution_count": 95,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "#add bin ids back in \n",
    "intermediate['Bin_Id']=conc_bin_stats['Bin_Id']\n",
    "intermediate"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 101,
   "id": "276cfe97-480b-48ce-a285-d97527fbad09",
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "['10' '1' '101' '102' '100' '104' '103' '106' '107' '105' '0' '11' '110'\n",
      " '109' '112' '113' '114' '115' '116' '117' '118' '119' '108' '120' '121'\n",
      " '122' '123' '111' '125' '126' '127' '128' '129' '12' '13' '131' '132'\n",
      " '130' '133' '135' '124' '136' '134' '138' '137' '140' '141' '142' '143'\n",
      " '144' '145' '146' '147' '148' '149' '15' '150' '151' '14' '153' '152'\n",
      " '155' '156' '157' '158' '154' '159' '16' '161' '162' '163' '164' '165'\n",
      " '166' '139' '168' '160' '17' '170' '171' '172' '173' '174' '175' '176'\n",
      " '177' '178' '179' '18' '180' '181' '182' '183' '184' '185' '186' '187'\n",
      " '188' '169' '19' '20' '2' '22' '21' '24' '25' '26' '27' '28' '29' '3'\n",
      " '30' '31' '32' '33' '34' '35' '23' '36' '38' '39' '4' '40' '41' '37' '42'\n",
      " '44' '45' '46' '47' '48' '49' '5' '43' '51' '52' '53' '50' '55' '56' '57'\n",
      " '58' '59' '6' '60' '61' '62' '63' '64' '65' '66' '67' '54' '69' '7' '70'\n",
      " '71' '72' '73' '74' '75' '68' '77' '78' '79' '8' '76' '81' '82' '80'\n",
      " '167' '85' '86' '87' '88' '89' '9' '90' '91' '92' '93' '83' '95' '96'\n",
      " '97' '94' '99' '84' '98']\n",
      "189\n"
     ]
    }
   ],
   "source": [
    "print(intermediate['Bin_Id'].unique())\n",
    "print(len(intermediate['Bin_Id'].unique()))"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 104,
   "id": "5a48076c-3cee-4d2f-bcdf-efd7d5615d56",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead tr th {\n",
       "        text-align: left;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr>\n",
       "      <th></th>\n",
       "      <th>Bin_Id</th>\n",
       "      <th>Marker_lineage</th>\n",
       "      <th>UID</th>\n",
       "      <th>#genomes</th>\n",
       "      <th>#markers</th>\n",
       "      <th>#marker_sets</th>\n",
       "      <th>0</th>\n",
       "      <th>1</th>\n",
       "      <th>2</th>\n",
       "      <th>3</th>\n",
       "      <th>4</th>\n",
       "      <th>5+</th>\n",
       "      <th>Completeness</th>\n",
       "      <th>Contamination</th>\n",
       "      <th>Strain_heterogeneity</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>42</td>\n",
       "      <td>o__Sphingomonadales</td>\n",
       "      <td>(UID3310)</td>\n",
       "      <td>26</td>\n",
       "      <td>569</td>\n",
       "      <td>293</td>\n",
       "      <td>3</td>\n",
       "      <td>556</td>\n",
       "      <td>10</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>99.59</td>\n",
       "      <td>2.30</td>\n",
       "      <td>0.00</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>56</td>\n",
       "      <td>c__Gammaproteobacteria</td>\n",
       "      <td>(UID4444)</td>\n",
       "      <td>263</td>\n",
       "      <td>507</td>\n",
       "      <td>232</td>\n",
       "      <td>45</td>\n",
       "      <td>450</td>\n",
       "      <td>12</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>85.79</td>\n",
       "      <td>2.68</td>\n",
       "      <td>33.33</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>109</td>\n",
       "      <td>k__Bacteria</td>\n",
       "      <td>(UID3187)</td>\n",
       "      <td>2258</td>\n",
       "      <td>188</td>\n",
       "      <td>117</td>\n",
       "      <td>53</td>\n",
       "      <td>132</td>\n",
       "      <td>3</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>68.90</td>\n",
       "      <td>1.12</td>\n",
       "      <td>0.00</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>68</td>\n",
       "      <td>k__Archaea</td>\n",
       "      <td>(UID2)</td>\n",
       "      <td>207</td>\n",
       "      <td>149</td>\n",
       "      <td>107</td>\n",
       "      <td>61</td>\n",
       "      <td>62</td>\n",
       "      <td>18</td>\n",
       "      <td>7</td>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "      <td>53.82</td>\n",
       "      <td>13.76</td>\n",
       "      <td>0.00</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>139</td>\n",
       "      <td>p__Cyanobacteria</td>\n",
       "      <td>(UID2143)</td>\n",
       "      <td>129</td>\n",
       "      <td>472</td>\n",
       "      <td>368</td>\n",
       "      <td>242</td>\n",
       "      <td>220</td>\n",
       "      <td>8</td>\n",
       "      <td>2</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>46.70</td>\n",
       "      <td>1.84</td>\n",
       "      <td>14.29</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "  Bin_Id          Marker_lineage        UID #genomes #markers #marker_sets  \\\n",
       "0     42     o__Sphingomonadales  (UID3310)       26      569          293   \n",
       "1     56  c__Gammaproteobacteria  (UID4444)      263      507          232   \n",
       "2    109             k__Bacteria  (UID3187)     2258      188          117   \n",
       "3     68              k__Archaea     (UID2)      207      149          107   \n",
       "4    139        p__Cyanobacteria  (UID2143)      129      472          368   \n",
       "\n",
       "     0    1   2  3  4 5+ Completeness Contamination Strain_heterogeneity  \n",
       "0    3  556  10  0  0  0        99.59          2.30                 0.00  \n",
       "1   45  450  12  0  0  0        85.79          2.68                33.33  \n",
       "2   53  132   3  0  0  0        68.90          1.12                 0.00  \n",
       "3   61   62  18  7  1  0        53.82         13.76                 0.00  \n",
       "4  242  220   8  2  0  0        46.70          1.84                14.29  "
      ]
     },
     "execution_count": 104,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "concoct_bins.head()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 117,
   "id": "b2575e71-df94-4650-b0b9-4eadfe97ab81",
   "metadata": {},
   "outputs": [],
   "source": [
    "# col names became multiIndex, so reformatting\n",
    "concoct_bins.columns = concoct_bins.columns.get_level_values(0)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 122,
   "id": "b291d218-950c-4969-9931-f36bade030da",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "array([ 42,  56, 109,  68, 139,  50, 150, 167,  98, 170,  83, 113,  84,\n",
       "         7, 105,  54,  76,  49, 169,  66, 152, 136,  97, 137, 154,  14,\n",
       "        90,  75, 160, 156,  43, 188, 166, 134,  45,  99,  96,  95,  94,\n",
       "        93,  92,  91,   9,  89,  88,  87,  86,  85,  82,  81,  80,   8,\n",
       "        79,  78,  77,  74,  73,  72,  71,  70,  69,  67,  65,  64,  63,\n",
       "        62,  61,  60,   6,  59,  58,  57,  55,  53,  52,  51,   5,  48,\n",
       "        47,  46,  44,  41,  40,   4,  39,  38,  37,  36,  35,  34,  33,\n",
       "        32,  31,  30,   3,  29,  28,  27,  26,  25,  24,  23,  22,  21,\n",
       "        20,   2,  19, 187, 186, 185, 184, 183, 182, 181, 180,  18, 179,\n",
       "       178, 177, 176, 175, 174, 173, 172, 171,  17, 168, 165, 164, 163,\n",
       "       162, 161,  16, 159, 158, 157, 155, 153, 151,  15, 149, 148, 147,\n",
       "       146, 145, 144, 143, 142, 141, 140, 138, 135, 133, 132, 131, 130,\n",
       "        13, 129, 128, 127, 126, 125, 124, 123, 122, 121, 120,  12, 119,\n",
       "       118, 117, 116, 115, 114, 112, 111, 110,  11, 108, 107, 106, 104,\n",
       "       103, 102, 101, 100,  10,   1,   0])"
      ]
     },
     "execution_count": 122,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "concoct_bins.Bin_Id.values"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 124,
   "id": "e61b0299-4f75-4211-a25d-48cd4bee50d3",
   "metadata": {},
   "outputs": [],
   "source": [
    "# converting intermiate bin id values into integers\n",
    "intermediate[\"Bin_Id\"] = intermediate.Bin_Id.apply(int)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 129,
   "id": "832009e4-8453-42be-8c3d-3e37dd624206",
   "metadata": {},
   "outputs": [],
   "source": [
    "#merging dataframes \n",
    "concoct_stats_merged=concoct_bins.merge(intermediate,on='Bin_Id')"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 133,
   "id": "5e03a8cd-e045-4ff5-81ca-ca22ffdf24f7",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "Index(['Bin_Id', 'Marker_lineage', 'UID', '#genomes', '#markers',\n",
       "       '#marker_sets', '0', '1', '2', '3', '4', '5+', 'Completeness',\n",
       "       'Contamination', 'Strain_heterogeneity', ''GC'', ' 'GC std'',\n",
       "       ' 'Genome size'', ' '# ambiguous bases'', ' '# scaffolds'',\n",
       "       ' '# contigs'', ' 'Longest scaffold'', ' 'Longest contig'',\n",
       "       ' 'N50 (scaffolds)'', ' 'N50 (contigs)'', ' 'Mean scaffold length'',\n",
       "       ' 'Mean contig length'', ' 'Coding density'', ' 'Translation table''],\n",
       "      dtype='object')"
      ]
     },
     "execution_count": 133,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "concoct_stats_merged.columns"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 135,
   "id": "3220a6e0-1d4e-4b37-ae1b-68b9a4bc5224",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>Bin_Id</th>\n",
       "      <th>Marker_lineage</th>\n",
       "      <th>UID</th>\n",
       "      <th>#genomes</th>\n",
       "      <th>#markers</th>\n",
       "      <th>#marker_sets</th>\n",
       "      <th>0</th>\n",
       "      <th>1</th>\n",
       "      <th>2</th>\n",
       "      <th>3</th>\n",
       "      <th>...</th>\n",
       "      <th>'# scaffolds'</th>\n",
       "      <th>'# contigs'</th>\n",
       "      <th>'Longest scaffold'</th>\n",
       "      <th>'Longest contig'</th>\n",
       "      <th>'N50 (scaffolds)'</th>\n",
       "      <th>'N50 (contigs)'</th>\n",
       "      <th>'Mean scaffold length'</th>\n",
       "      <th>'Mean contig length'</th>\n",
       "      <th>'Coding density'</th>\n",
       "      <th>'Translation table'</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>109</td>\n",
       "      <td>k__Bacteria</td>\n",
       "      <td>(UID3187)</td>\n",
       "      <td>2258</td>\n",
       "      <td>188</td>\n",
       "      <td>117</td>\n",
       "      <td>53</td>\n",
       "      <td>132</td>\n",
       "      <td>3</td>\n",
       "      <td>0</td>\n",
       "      <td>...</td>\n",
       "      <td>960</td>\n",
       "      <td>960</td>\n",
       "      <td>8661</td>\n",
       "      <td>8661</td>\n",
       "      <td>2243</td>\n",
       "      <td>2243</td>\n",
       "      <td>2112.117708333333</td>\n",
       "      <td>2112.117708333333</td>\n",
       "      <td>0.797705501932549</td>\n",
       "      <td>11</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>1 rows × 29 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "   Bin_Id Marker_lineage        UID  #genomes  #markers  #marker_sets   0  \\\n",
       "2     109    k__Bacteria  (UID3187)      2258       188           117  53   \n",
       "\n",
       "     1  2  3  ...   '# scaffolds'   '# contigs'   'Longest scaffold'  \\\n",
       "2  132  3  0  ...             960           960                 8661   \n",
       "\n",
       "    'Longest contig'   'N50 (scaffolds)'  'N50 (contigs)'  \\\n",
       "2               8661                2243             2243   \n",
       "\n",
       "   'Mean scaffold length'  'Mean contig length'    'Coding density'  \\\n",
       "2       2112.117708333333     2112.117708333333   0.797705501932549   \n",
       "\n",
       "   'Translation table'  \n",
       "2                   11  \n",
       "\n",
       "[1 rows x 29 columns]"
      ]
     },
     "execution_count": 135,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "concoct_109=concoct_stats_merged[concoct_stats_merged['Bin_Id'] == 109]\n",
    "concoct_109"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 155,
   "id": "42ff0e47-dffa-4079-9bba-b464dd6bcc0e",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>Bin_Id</th>\n",
       "      <th>Marker_lineage</th>\n",
       "      <th>UID</th>\n",
       "      <th>#genomes</th>\n",
       "      <th>#markers</th>\n",
       "      <th>#marker_sets</th>\n",
       "      <th>0</th>\n",
       "      <th>1</th>\n",
       "      <th>2</th>\n",
       "      <th>3</th>\n",
       "      <th>4</th>\n",
       "      <th>5+</th>\n",
       "      <th>Completeness</th>\n",
       "      <th>Contamination</th>\n",
       "      <th>Strain_heterogeneity</th>\n",
       "      <th>'GC'</th>\n",
       "      <th>'GC std'</th>\n",
       "      <th>'Genome size'</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>109</td>\n",
       "      <td>k__Bacteria</td>\n",
       "      <td>(UID3187)</td>\n",
       "      <td>2258</td>\n",
       "      <td>188</td>\n",
       "      <td>117</td>\n",
       "      <td>53</td>\n",
       "      <td>132</td>\n",
       "      <td>3</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>68.9</td>\n",
       "      <td>1.12</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.3992117902993293</td>\n",
       "      <td>0.027524218532616485</td>\n",
       "      <td>2027633</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "   Bin_Id Marker_lineage        UID  #genomes  #markers  #marker_sets   0  \\\n",
       "2     109    k__Bacteria  (UID3187)      2258       188           117  53   \n",
       "\n",
       "     1  2  3  4  5+  Completeness  Contamination  Strain_heterogeneity  \\\n",
       "2  132  3  0  0   0          68.9           1.12                   0.0   \n",
       "\n",
       "                  'GC'               'GC std'  'Genome size'  \n",
       "2   0.3992117902993293   0.027524218532616485        2027633  "
      ]
     },
     "execution_count": 155,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "concoct_109.iloc[:, 0:18]"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "a2e3f90e-44ca-4519-b545-626c5c60c5a7",
   "metadata": {},
   "source": [
    "#### Results & Next Steps\n",
    "- anvio found 0 scg for concoct 109, based on # contigs, length, and N50 it may just be too fragmented\n",
    "- run against gtdb independently and import into anvio \n",
    "- look more into mags in both binners and both sample sets "
   ]
  }
 ],
 "metadata": {
  "kernelspec": {
   "display_name": "Python 3 (ipykernel)",
   "language": "python",
   "name": "python3"
  },
  "language_info": {
   "codemirror_mode": {
    "name": "ipython",
    "version": 3
   },
   "file_extension": ".py",
   "mimetype": "text/x-python",
   "name": "python",
   "nbconvert_exporter": "python",
   "pygments_lexer": "ipython3",
   "version": "3.11.0"
  }
 },
 "nbformat": 4,
 "nbformat_minor": 5
}
