{
 "cells": [
  {
   "cell_type": "markdown",
   "id": "20751f43-626d-418c-ae3b-4c188ff39ddd",
   "metadata": {},
   "source": [
    "**More QC & Taxonomic Profiling**\n",
    "( rearranged steps a bit for now) \n",
    "- running all PSTR together \n",
    "\n",
    "- this follows part of 1Assembly steps and 4Taxonomy\n",
    "    - host coral, human, and symbiont removal \n",
    "    - kraken and bracken read recruitment (taxonomic abundances)\n",
    "    - will NOT be assembling for now (skipping megahit step)"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "08b44973-74ba-44cc-91bf-8a20ebcaee30",
   "metadata": {},
   "source": [
    "**See notes on QC in CBC_metagenomics/0QC_trimming-112023 nb**"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "c31048b5-7185-42b5-9427-54fccf76f6c3",
   "metadata": {},
   "outputs": [],
   "source": [
    "#!/bin/bash\n",
    "#SBATCH -c 24  # Number of Cores per Task\n",
    "#SBATCH --mem=180G  # Requested Memory\n",
    "#SBATCH -p cpu-long  # Partition\n",
    "#SBATCH -t 56:00:00  # Job time limit\n",
    "#SBATCH --mail-type=ALL\n",
    "#SBATCH -o /work/pi_sarah_gignouxwolfsohn_uml_edu/brooke/pstr/slurm-removaltax-pstr-%j.out  # %j = job ID\n",
    "\n",
    "cd /project/pi_sarah_gignouxwolfsohn_uml_edu/brooke/pstr || exit\n",
    "module load miniconda/22.11.1-1\n",
    "conda activate anvio-8\n",
    "\n",
    "# 1)remove host from sample reads\n",
    "\n",
    "#set general parameters:\n",
    "SAMPLENAME=\"pstr\"\n",
    "SAMPLELIST=\"filtered_sampleids\"\n",
    "RAWREADSPATH='/project/pi_sarah_gignouxwolfsohn_uml_edu/brooke/qc_112023'\n",
    "READSPATH=\"/project/pi_sarah_gignouxwolfsohn_uml_edu/brooke/${SAMPLENAME}/assembly/host_removed\"\n",
    "mkdir -p $READSPATH\n",
    "LISTPATH=\"/project/pi_sarah_gignouxwolfsohn_uml_edu/brooke/${SAMPLENAME}\"\n",
    "#set step parameters \n",
    "GENOME=\"Ofav\"\n",
    "INPUTPATH=\"/project/pi_sarah_gignouxwolfsohn_uml_edu/Reference_genomes/${GENOME}_genome\"\n",
    "INDEX=\"${GENOME}_DB\"\n",
    "\n",
    "#build a bowtie2 index from a known genome\n",
    "# **using ofav since pstr is more closely related to ofav than mcav - (was only built for histat so need to redo for bowtie)\n",
    "#bowtie2-build $INPUTPATH/GCF_002042975.1_ofav_dov_v1_genomic.fna $INPUTPATH/\"$INDEX\"\n",
    "\n",
    "#loop through samples\n",
    "while IFS= read -r SAMPLEID; do\n",
    "#re-align reads back to the index\n",
    "bowtie2 -p 8 -x $INPUTPATH/$INDEX -1 \"$RAWREADSPATH\"/\"${SAMPLEID}_R1_001_val_1.fq\" -2 \"$RAWREADSPATH\"/\"${SAMPLEID}_R2_001_val_2.fq\" -S $READSPATH/\"${SAMPLEID}\"_mapped_and_unmapped.sam\n",
    "#convert sam file from bowtie to a bam file for processing\n",
    "samtools view -bS $READSPATH/\"${SAMPLEID}\"_mapped_and_unmapped.sam > $READSPATH/\"${SAMPLEID}\"_mapped_and_unmapped.bam\n",
    "#extract only the reads of which both do not match against the host genome\n",
    "samtools view -b -f 12 -F 256 $READSPATH/\"${SAMPLEID}\"_mapped_and_unmapped.bam > $READSPATH/\"${SAMPLEID}\"_bothReadsUnmapped.bam\n",
    "# sorts the file so both mates are together and then extracts them back as .fastq files\n",
    "samtools sort -n -m 5G -@ 2 $READSPATH/\"${SAMPLEID}\"_bothReadsUnmapped.bam -o $READSPATH/\"${SAMPLEID}\"_bothReadsUnmapped_sorted.bam\n",
    "samtools fastq -@ 8 $READSPATH/\"${SAMPLEID}\"_bothReadsUnmapped_sorted.bam \\\n",
    "    -1 $READSPATH/\"${SAMPLEID}\"_host_removed_R1.fastq \\\n",
    "    -2 $READSPATH/\"${SAMPLEID}\"_host_removed_R2.fastq \\\n",
    "    -0 /dev/null -s /dev/null -n\n",
    " if [ $? -eq 0 ]; then\n",
    "        echo \"host removal completed successfully for sample: $SAMPLEID\"\n",
    "    else\n",
    "        echo \"host removal encountered an error for sample: $SAMPLEID\"\n",
    "        exit 1  \n",
    "    fi\n",
    "done < \"$LISTPATH/${SAMPLELIST}\"\n",
    "conda deactivate\n",
    "echo \"Host removal: All samples processed successfully.\"\n",
    "\n",
    "# 2)remove symbiont and human seqs using fastq screen \n",
    "module load bowtie/2.4.5\n",
    "conda activate fastq_screen\n",
    "FASTQSCREEN='/home/brooke_sienkiewicz_student_uml_edu/.conda/envs/fastq_screen/share/fastq-screen-0.15.3-0'\n",
    "OUTPUTDIR='/project/pi_sarah_gignouxwolfsohn_uml_edu/brooke/${SAMPLENAME}/assembly/final_reads_filtered'\n",
    "\n",
    "mkdir -p $OUTPUTDIR\n",
    "\n",
    "while IFS= read -r SAMPLEID; do\n",
    "$FASTQSCREEN/fastq_screen --nohits --aligner bowtie2 --conf $FASTQSCREEN/fastq_screen.conf --outdir $OUTPUTDIR \\\n",
    "$READSPATH/\"${SAMPLEID}\"_host_removed_R1.fastq $READSPATH/\"${SAMPLEID}\"_host_removed_R2.fastq;\n",
    " if [ $? -eq 0 ]; then\n",
    "        echo \"fastq_screen completed successfully for sample: $SAMPLEID\"\n",
    "    else\n",
    "        echo \"fastq_screen encountered an error for sample: $SAMPLEID\"\n",
    "        exit 1\n",
    "    fi\n",
    "# --nohits = output reads do not map to any genomes\n",
    "done < \"$LISTPATH/${SAMPLELIST}\"\n",
    "conda deactivate\n",
    "echo \"Symbiont, host removal: All samples processed successfully.\"\n",
    "\n",
    "# 3)concatenate all f and r seqs into single file (1 for f, 1 for r)\n",
    "conda activate assembly\n",
    "OUTDIR='/project/pi_sarah_gignouxwolfsohn_uml_edu/brooke/${SAMPLENAME}/assembly'\n",
    "\n",
    "# Read the sample IDs from the file\n",
    "while IFS= read -r SAMPLEID; do\n",
    "    # Construct the file paths for forward and reverse reads\n",
    "    FORWARD_READ=\"$READSPATH/${SAMPLEID}_host_removed_R1.fastq\"\n",
    "    REVERSE_READ=\"$READSPATH/${SAMPLEID}_host_removed_R2.fastq\"\n",
    "    # Check if the files exist before concatenating\n",
    "    if [ -e \"$FORWARD_READ\" ]; then\n",
    "        cat \"$FORWARD_READ\" >> \"$OUTDIR/${SAMPLENAME}_reads_R1_ALL.fastq\"\n",
    "    else\n",
    "        echo \"Forward read file not found for sample $SAMPLEID\"\n",
    "    fi\n",
    "    if [ -e \"$REVERSE_READ\" ]; then\n",
    "        cat \"$REVERSE_READ\" >> \"$OUTDIR/${SAMPLENAME}_reads_R2_ALL.fastq\"\n",
    "    else\n",
    "        echo \"Reverse read file not found for sample $SAMPLEID\"\n",
    "    fi\n",
    "done < \"$LISTPATH/${SAMPLELIST}\"\n",
    "conda deactivate \n",
    "\n",
    "# kraken and bracken \n",
    "# SKIPPING intermediate steps to go ahead and run kraken \n",
    "conda activate kraken\n",
    "\n",
    "# set variables\n",
    "DBNAME='//project/pi_sarah_gignouxwolfsohn_uml_edu/Reference_genomes/ref_databases/standard'\n",
    "OUTDIR='//project/pi_sarah_gignouxwolfsohn_uml_edu/brooke/${SAMPLENAME}/taxonomy/kraken_stddb'\n",
    "THREADS=24\n",
    "KMER_LEN=35 #(default)\n",
    "READ_LEN=150\n",
    "\n",
    "mkdir -p $OUTDIR\n",
    "\n",
    "# classify each set of paired end reads against silva taxonomic database\n",
    "while IFS= read -r SAMPLEID; do\n",
    "kraken2 --db $DBNAME --threads $THREADS --report $OUTDIR/${SAMPLEID}.kreport2 --report-zero-counts --paired $READSPATH/\"${SAMPLEID}\"_host_removed_R1.fastq $READSPATH/\"${SAMPLEID}\"_host_removed_R2.fastq > $OUTDIR/${SAMPLEID}.kraken2\n",
    "if [ $? -eq 0 ]; then\n",
    "        echo \"kraken2 completed successfully for sample: $SAMPLEID\"\n",
    "    else\n",
    "        echo \"kraken2 encountered an error for sample: $SAMPLEID\"\n",
    "        exit 1\n",
    "    fi\n",
    "done < \"$LISTPATH/${SAMPLELIST}\"\n",
    "echo \"Kraken2: All samples processed successfully.\"\n",
    "\n",
    "#'report-zero-counts' so that all reports have same taxa so they can be merged\n",
    "\n",
    "# run bracken\n",
    "# navigate to Bracken dir because can't figure out how to set up symlink \n",
    "cd //project/pi_sarah_gignouxwolfsohn_uml_edu/brooke/Bracken\n",
    "# Generate bracken database - just do once\n",
    "#bracken-build -d ${DBNAME} -t ${THREADS} -k ${KMER_LEN} -l ${READ_LEN} \n",
    "# Already Done\n",
    "\n",
    "# Abundance Estimation\n",
    "FILEPATH='//project/pi_sarah_gignouxwolfsohn_uml_edu/brooke/mcav/healthy_2022_mcav/taxonomy/kraken_stddb'\n",
    "LEVEL=\"S\"\n",
    "\n",
    "while IFS= read -r line; do\n",
    "    # Execute bracken command for each file\n",
    "    bracken -d \"$DBNAME\" -i $FILEPATH/\"$line\".kreport2 -o \"$FILEPATH/${line%.kreport2}_$LEVEL.bracken\" -r \"$READ_LEN\" -l \"$LEVEL\"\n",
    "if [ $? -eq 0 ]; then\n",
    "        echo \"bracken completed successfully for sample: $line\"\n",
    "    else\n",
    "        echo \"bracken encountered an error for sample: $line\"\n",
    "        exit 1\n",
    "    fi\n",
    "done < \"$LISTPATH/${SAMPLELIST}\"\n",
    "echo \"Bracken: All samples processed successfully.\"\n",
    "#${SAMPLE}.kreport - the kraken report generated for a given dataset\n",
    "#{BRACKEN_OUTPUT_FILE}.bracken - the desired name of the output file to be generated by the code\n",
    "#The following optional parameters may be specified:\n",
    "    #${LEVEL} - Default = 'S'. This specifies that abundance estimation will calculate estimated reads for each species. Other possible options are K (kingdom level), P (phylum), C (class), O (order), F (family), and G (genus).\n",
    "    #${THRESHOLD} - Default = 10. For species classification, any species with <= 10 (or otherwise specified) reads will not receive any additional reads from higher taxonomy levels when distributing reads for abundance estimation. \n",
    "    #If another classification level is specified, thresholding will occur at that level.\n",
    "\n",
    "\n",
    "# JOB-ID: 23190073, 23191990\n",
    "# bash script file name: /work/pi_sarah_gignouxwolfsohn/brooke/${SAMPLENAME}/removaltax"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "bc2f3621-fa09-4a5d-b00a-1ee0c1f812f5",
   "metadata": {},
   "outputs": [],
   "source": [
    "# HAVE NOT RUN YET \n",
    "\n",
    "# 4)ASSEMBLE reads into contigs (contiguous sequence - joins them together based on read overlap, and ensures there are no gaps\n",
    "megahit --presets meta-large \\\n",
    "-1 \"$OUTDIR\"/\"$SAMPLENAME\"_reads_R1_ALL.fastq \\\n",
    "-2 \"$OUTDIR\"/\"$SAMPLENAME\"_reads_R2_ALL.fastq \\\n",
    "--keep-tmp-files \\\n",
    "#--continue \\\n",
    "-o $OUTDIR/contigs --out-prefix $SAMPLENAME \n",
    "#this one has to make the directory; will fail if it already exists\n",
    "if [ $? -eq 0 ]; then\n",
    "        echo \"megahit completed successfully for samples: $SAMPLENAME\"\n",
    "    else\n",
    "        echo \"megahit encountered an error for samples: $SAMPLENAME\"\n",
    "        exit 1\n",
    "    fi\n",
    "conda deactivate \n",
    "echo \"Megahit: Contig file created successfully.\"\n",
    "#final assembled contigs: ${SAMPLENAME}/assembly/contigs\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "169dddb5-8b1f-4f42-bf94-ffa5d19a8197",
   "metadata": {},
   "outputs": [],
   "source": [
    "# this was done previously in 11/2023 with no host removal or anything..delete after redoing process\n",
    "\n",
    "#!/bin/bash\n",
    "#SBATCH -c 24  # Number of Cores per Task\n",
    "#SBATCH --mem=180G  # Requested Memory\n",
    "#SBATCH -p cpu-long  # Partition\n",
    "#SBATCH -t 56:00:00  # Job time limit\n",
    "#SBATCH --mail-type=ALL\n",
    "#SBATCH -o /project/pi_sarah_gignouxwolfsohn_uml_edu/brooke/bash_scripts/slurm-%j.out  # %j = job ID\n",
    "\n",
    "module load miniconda/22.11.1-1\n",
    "conda activate assembly\n",
    "\n",
    "#set up variables\n",
    "READSPATH=/project/pi_sarah_gignouxwolfsohn_uml_edu/brooke/working/11272023/trimmed\n",
    "SAMPLENAME=CBC\n",
    "OUTDIR=\"$SAMPLENAME\"_assembly\n",
    "\n",
    "#concatenate all forward reads to one fastq file\n",
    "cat \"$READSPATH\"/*_R1*.fq > \"$READSPATH\"/\"$SAMPLENAME\"_reads_R1_ALL.fastq\n",
    "\n",
    "#concatenate all reverse reads to one fastq file\n",
    "cat \"$READSPATH\"/*_R2*.fq > \"$READSPATH\"/\"$SAMPLENAME\"_reads_R2_ALL.fastq\n",
    "\n",
    "megahit --presets meta-large \\\n",
    "-1 \"$READSPATH\"/\"$SAMPLENAME\"_reads_R1_ALL.fastq \\\n",
    "-2 \"$READSPATH\"/\"$SAMPLENAME\"_reads_R2_ALL.fastq \\\n",
    "--keep-tmp-files \\\n",
    "-o /project/pi_sarah_gignouxwolfsohn_uml_edu/brooke/working/11272023/$OUTDIR --out-prefix $SAMPLENAME \n",
    "#makes dir or do you need to mkdir beforehand???\n",
    "#file: /project/pi_sarah_gignouxwolfsohn_uml_edu/brooke/11_2023bash_scripts/assembly\n",
    "#JOB ID: 15972157"
   ]
  }
 ],
 "metadata": {
  "kernelspec": {
   "display_name": "Python 3 (ipykernel)",
   "language": "python",
   "name": "python3"
  },
  "language_info": {
   "codemirror_mode": {
    "name": "ipython",
    "version": 3
   },
   "file_extension": ".py",
   "mimetype": "text/x-python",
   "name": "python",
   "nbconvert_exporter": "python",
   "pygments_lexer": "ipython3",
   "version": "3.11.0"
  }
 },
 "nbformat": 4,
 "nbformat_minor": 5
}
