{
 "cells": [
  {
   "cell_type": "markdown",
   "id": "20751f43-626d-418c-ae3b-4c188ff39ddd",
   "metadata": {},
   "source": [
    "**More QC & Taxonomic Profiling**\n",
    "( rearranged steps a bit for now) \n",
    "- running all PSTR together \n",
    "\n",
    "- this follows part of 1Assembly steps and 4Taxonomy\n",
    "    - host coral, human, and symbiont removal \n",
    "    - kraken and bracken read recruitment (taxonomic abundances)\n",
    "    - will NOT be assembling for now (skipping megahit step)"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "08b44973-74ba-44cc-91bf-8a20ebcaee30",
   "metadata": {},
   "source": [
    "**See notes on QC in CBC_metagenomics/0QC_trimming-112023 nb**"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "004e6db5-6408-4c01-955f-a1c1db2fa638",
   "metadata": {
    "tags": []
   },
   "source": [
    "### Running host, symbiont removal and kraken/bracken"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "c31048b5-7185-42b5-9427-54fccf76f6c3",
   "metadata": {},
   "outputs": [],
   "source": [
    "#!/bin/bash\n",
    "#SBATCH -c 24  # Number of Cores per Task\n",
    "#SBATCH --mem=180G  # Requested Memory\n",
    "#SBATCH -p cpu-long  # Partition\n",
    "#SBATCH -t 56:00:00  # Job time limit\n",
    "#SBATCH --mail-type=ALL\n",
    "#SBATCH -o /work/pi_sarah_gignouxwolfsohn_uml_edu/brooke/pstr/slurm-removaltax-pstr-%j.out  # %j = job ID\n",
    "\n",
    "cd /project/pi_sarah_gignouxwolfsohn_uml_edu/brooke/pstr || exit\n",
    "module load miniconda/22.11.1-1\n",
    "conda activate anvio-8\n",
    "\n",
    "# 1)remove host from sample reads\n",
    "\n",
    "#set general parameters:\n",
    "SAMPLENAME=\"pstr\"\n",
    "SAMPLELIST=\"filtered_sampleids\"\n",
    "RAWREADSPATH='/project/pi_sarah_gignouxwolfsohn_uml_edu/brooke/qc_112023'\n",
    "READSPATH=\"/project/pi_sarah_gignouxwolfsohn_uml_edu/brooke/${SAMPLENAME}/assembly/host_removed\"\n",
    "mkdir -p $READSPATH\n",
    "LISTPATH=\"/project/pi_sarah_gignouxwolfsohn_uml_edu/brooke/${SAMPLENAME}\"\n",
    "#set step parameters \n",
    "GENOME=\"Ofav\"\n",
    "INPUTPATH=\"/project/pi_sarah_gignouxwolfsohn_uml_edu/Reference_genomes/${GENOME}_genome\"\n",
    "INDEX=\"${GENOME}_DB\"\n",
    "\n",
    "#build a bowtie2 index from a known genome\n",
    "# **using ofav since pstr is more closely related to ofav than mcav - (was only built for histat so need to redo for bowtie)\n",
    "#bowtie2-build $INPUTPATH/GCF_002042975.1_ofav_dov_v1_genomic.fna $INPUTPATH/\"$INDEX\"\n",
    "\n",
    "#loop through samples\n",
    "while IFS= read -r SAMPLEID; do\n",
    "#re-align reads back to the index\n",
    "bowtie2 -p 8 -x $INPUTPATH/$INDEX -1 \"$RAWREADSPATH\"/\"${SAMPLEID}_R1_001_val_1.fq\" -2 \"$RAWREADSPATH\"/\"${SAMPLEID}_R2_001_val_2.fq\" -S $READSPATH/\"${SAMPLEID}\"_mapped_and_unmapped.sam\n",
    "#convert sam file from bowtie to a bam file for processing\n",
    "samtools view -bS $READSPATH/\"${SAMPLEID}\"_mapped_and_unmapped.sam > $READSPATH/\"${SAMPLEID}\"_mapped_and_unmapped.bam\n",
    "#extract only the reads of which both do not match against the host genome\n",
    "samtools view -b -f 12 -F 256 $READSPATH/\"${SAMPLEID}\"_mapped_and_unmapped.bam > $READSPATH/\"${SAMPLEID}\"_bothReadsUnmapped.bam\n",
    "# sorts the file so both mates are together and then extracts them back as .fastq files\n",
    "samtools sort -n -m 5G -@ 2 $READSPATH/\"${SAMPLEID}\"_bothReadsUnmapped.bam -o $READSPATH/\"${SAMPLEID}\"_bothReadsUnmapped_sorted.bam\n",
    "samtools fastq -@ 8 $READSPATH/\"${SAMPLEID}\"_bothReadsUnmapped_sorted.bam \\\n",
    "    -1 $READSPATH/\"${SAMPLEID}\"_host_removed_R1.fastq \\\n",
    "    -2 $READSPATH/\"${SAMPLEID}\"_host_removed_R2.fastq \\\n",
    "    -0 /dev/null -s /dev/null -n\n",
    " if [ $? -eq 0 ]; then\n",
    "        echo \"host removal completed successfully for sample: $SAMPLEID\"\n",
    "    else\n",
    "        echo \"host removal encountered an error for sample: $SAMPLEID\"\n",
    "        exit 1  \n",
    "    fi\n",
    "done < \"$LISTPATH/${SAMPLELIST}\"\n",
    "conda deactivate\n",
    "echo \"Host removal: All samples processed successfully.\"\n",
    "\n",
    "# 2)remove symbiont and human seqs using fastq screen \n",
    "module load bowtie/2.4.5\n",
    "conda activate fastq_screen\n",
    "FASTQSCREEN='/home/brooke_sienkiewicz_student_uml_edu/.conda/envs/fastq_screen/share/fastq-screen-0.15.3-0'\n",
    "OUTPUTDIR=\"/project/pi_sarah_gignouxwolfsohn_uml_edu/brooke/${SAMPLENAME}/assembly/final_reads_filtered\"\n",
    "\n",
    "mkdir -p \"$OUTPUTDIR\"\n",
    "if [ $? -ne 0 ]; then\n",
    "    echo \"Error: Failed to create output directory $OUTPUTDIR\"\n",
    "    exit 1\n",
    "fi\n",
    "\n",
    "while IFS= read -r SAMPLEID; do\n",
    "$FASTQSCREEN/fastq_screen --nohits --aligner bowtie2 --conf $FASTQSCREEN/fastq_screen.conf --outdir $OUTPUTDIR \\\n",
    "$READSPATH/\"${SAMPLEID}\"_host_removed_R1.fastq $READSPATH/\"${SAMPLEID}\"_host_removed_R2.fastq;\n",
    " if [ $? -eq 0 ]; then\n",
    "        echo \"fastq_screen completed successfully for sample: $SAMPLEID\"\n",
    "    else\n",
    "        echo \"fastq_screen encountered an error for sample: $SAMPLEID\"\n",
    "        exit 1\n",
    "    fi\n",
    "# --nohits = output reads do not map to any genomes\n",
    "done < \"$LISTPATH/${SAMPLELIST}\"\n",
    "conda deactivate\n",
    "echo \"Symbiont, host removal: All samples processed successfully.\"\n",
    "\n",
    "\n",
    "\n",
    "# 3)concatenate all f and r seqs into single file (1 for f, 1 for r)\n",
    "conda activate assembly\n",
    "\n",
    "FILTERREADS=\"/project/pi_sarah_gignouxwolfsohn_uml_edu/brooke/${SAMPLENAME}/assembly/final_reads_filtered\"\n",
    "\n",
    "\n",
    "# Read the sample IDs from the file\n",
    "while IFS= read -r SAMPLEID; do\n",
    "    # Construct the file paths for forward and reverse reads\n",
    "    FORWARD_READ=\"$FILTERREADS/${SAMPLEID}_host_removed_R1.tagged_filter.fastq\"\n",
    "    REVERSE_READ=\"$FILTERREADS/${SAMPLEID}_host_removed_R2.tagged_filter.fastq\"\n",
    "    # Check if the files exist before concatenating\n",
    "    if [ -e \"$FORWARD_READ\" ]; then\n",
    "        cat \"$FORWARD_READ\" >> \"$LISTPATH/assembly/${SAMPLENAME}_reads_R1_ALL.fastq\"\n",
    "    else\n",
    "        echo \"Forward read file not found for sample $SAMPLEID\"\n",
    "    fi\n",
    "    if [ -e \"$REVERSE_READ\" ]; then\n",
    "        cat \"$REVERSE_READ\" >> \"$LISTPATH/assembly/${SAMPLENAME}_reads_R2_ALL.fastq\"\n",
    "    else\n",
    "        echo \"Reverse read file not found for sample $SAMPLEID\"\n",
    "    fi\n",
    "done < \"$LISTPATH/${SAMPLELIST}\"\n",
    "conda deactivate \n",
    "\n",
    "# kraken and bracken \n",
    "# SKIPPING intermediate steps to go ahead and run kraken \n",
    "conda activate kraken\n",
    "SAMPLENAME=\"pstr\"\n",
    "\n",
    "# set variables\n",
    "DBNAME='/project/pi_sarah_gignouxwolfsohn_uml_edu/Reference_genomes/ref_databases/standard'\n",
    "OUTDIR='/project/pi_sarah_gignouxwolfsohn_uml_edu/brooke/${SAMPLENAME}/taxonomy/kraken_stddb'\n",
    "THREADS=24\n",
    "KMER_LEN=35 #(default)\n",
    "READ_LEN=150\n",
    "\n",
    "mkdir -p $OUTDIR\n",
    "if [ $? -ne 0 ]; then\n",
    "    echo \"Error: Failed to create output directory $OUTPUTDIR\"\n",
    "    exit 1\n",
    "fi\n",
    "\n",
    "# classify each set of paired end reads against silva taxonomic database\n",
    "while IFS= read -r SAMPLEID; do\n",
    "kraken2 --db $DBNAME --threads $THREADS --report $OUTDIR/${SAMPLEID}.kreport2 --report-zero-counts --paired $FILTERREADS/\"${SAMPLEID}\"_host_removed_R1.tagged_filter.fastq $FILTERREADS/\"${SAMPLEID}\"_host_removed_R2.tagged_filter.fastq > $OUTDIR/${SAMPLEID}.kraken2\n",
    "if [ $? -eq 0 ]; then\n",
    "        echo \"kraken2 completed successfully for sample: $SAMPLEID\"\n",
    "    else\n",
    "        echo \"kraken2 encountered an error for sample: $SAMPLEID\"\n",
    "        exit 1\n",
    "    fi\n",
    "done < \"$LISTPATH/${SAMPLELIST}\"\n",
    "echo \"Kraken2: All samples processed successfully.\"\n",
    "\n",
    "#'report-zero-counts' so that all reports have same taxa so they can be merged\n",
    "\n",
    "# run bracken\n",
    "# navigate to Bracken dir because can't figure out how to set up symlink \n",
    "cd //project/pi_sarah_gignouxwolfsohn_uml_edu/brooke/Bracken\n",
    "# Generate bracken database - just do once\n",
    "#bracken-build -d ${DBNAME} -t ${THREADS} -k ${KMER_LEN} -l ${READ_LEN} \n",
    "# Already Done\n",
    "\n",
    "# Abundance Estimation\n",
    "FILEPATH='/project/pi_sarah_gignouxwolfsohn_uml_edu/brooke/${SAMPLENAME}/taxonomy/kraken_stddb'\n",
    "LEVEL=\"S\"\n",
    "\n",
    "while IFS= read -r line; do\n",
    "    INPUT_FILE=\"$FILEPATH/$line.kreport2\"\n",
    "    OUTPUT_FILE=\"$FILEPATH/${line%.kreport2}_$LEVEL.bracken\"\n",
    "    \n",
    "    if [ -f \"$INPUT_FILE\" ]; then\n",
    "        echo \"Processing sample: $line\"\n",
    "        bracken -d \"$DBNAME\" -i \"$INPUT_FILE\" -o \"$OUTPUT_FILE\" -r \"$READ_LEN\" -l \"$LEVEL\"\n",
    "        \n",
    "        if [ $? -eq 0 ]; then\n",
    "            echo \"bracken completed successfully for sample: $line\"\n",
    "        else\n",
    "            echo \"bracken encountered an error for sample: $line\"\n",
    "            exit 1\n",
    "        fi\n",
    "    else\n",
    "        echo \"Error: Input file $INPUT_FILE does not exist\"\n",
    "        exit 1\n",
    "    fi\n",
    "done < \"$LISTPATH/$SAMPLELIST\"\n",
    "\n",
    "echo \"Bracken: All samples processed successfully.\"\n",
    "#${SAMPLE}.kreport - the kraken report generated for a given dataset\n",
    "#{BRACKEN_OUTPUT_FILE}.bracken - the desired name of the output file to be generated by the code\n",
    "#The following optional parameters may be specified:\n",
    "    #${LEVEL} - Default = 'S'. This specifies that abundance estimation will calculate estimated reads for each species. Other possible options are K (kingdom level), P (phylum), C (class), O (order), F (family), and G (genus).\n",
    "    #${THRESHOLD} - Default = 10. For species classification, any species with <= 10 (or otherwise specified) reads will not receive any additional reads from higher taxonomy levels when distributing reads for abundance estimation. \n",
    "    #If another classification level is specified, thresholding will occur at that level.\n",
    "\n",
    "\n",
    "# JOB-ID: 23190073, 23191990\n",
    "# 23350054 - final (hopefully) script (minus host-removed since that was completed with job 23191990)\n",
    "# ^didn't use filtered reads as input for kraken\n",
    "# 23397902: just redoing #3 and #4 w/ filtered seqs (script file 'kraken')\n",
    "# bash script file name: /work/pi_sarah_gignouxwolfsohn/brooke/${SAMPLENAME}/removaltax"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "bc2f3621-fa09-4a5d-b00a-1ee0c1f812f5",
   "metadata": {},
   "outputs": [],
   "source": [
    "# HAVE NOT RUN YET \n",
    "\n",
    "# 4)ASSEMBLE reads into contigs (contiguous sequence - joins them together based on read overlap, and ensures there are no gaps\n",
    "megahit --presets meta-large \\\n",
    "-1 \"$OUTDIR\"/\"$SAMPLENAME\"_reads_R1_ALL.fastq \\\n",
    "-2 \"$OUTDIR\"/\"$SAMPLENAME\"_reads_R2_ALL.fastq \\\n",
    "--keep-tmp-files \\\n",
    "#--continue \\\n",
    "-o $OUTDIR/contigs --out-prefix $SAMPLENAME \n",
    "#this one has to make the directory; will fail if it already exists\n",
    "if [ $? -eq 0 ]; then\n",
    "        echo \"megahit completed successfully for samples: $SAMPLENAME\"\n",
    "    else\n",
    "        echo \"megahit encountered an error for samples: $SAMPLENAME\"\n",
    "        exit 1\n",
    "    fi\n",
    "conda deactivate \n",
    "echo \"Megahit: Contig file created successfully.\"\n",
    "#final assembled contigs: ${SAMPLENAME}/assembly/contigs\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "169dddb5-8b1f-4f42-bf94-ffa5d19a8197",
   "metadata": {},
   "outputs": [],
   "source": [
    "# this was done previously in 11/2023 with no host removal or anything..delete after redoing process\n",
    "\n",
    "#!/bin/bash\n",
    "#SBATCH -c 24  # Number of Cores per Task\n",
    "#SBATCH --mem=180G  # Requested Memory\n",
    "#SBATCH -p cpu-long  # Partition\n",
    "#SBATCH -t 56:00:00  # Job time limit\n",
    "#SBATCH --mail-type=ALL\n",
    "#SBATCH -o /project/pi_sarah_gignouxwolfsohn_uml_edu/brooke/bash_scripts/slurm-%j.out  # %j = job ID\n",
    "\n",
    "module load miniconda/22.11.1-1\n",
    "conda activate assembly\n",
    "\n",
    "#set up variables\n",
    "READSPATH=/project/pi_sarah_gignouxwolfsohn_uml_edu/brooke/working/11272023/trimmed\n",
    "SAMPLENAME=CBC\n",
    "OUTDIR=\"$SAMPLENAME\"_assembly\n",
    "\n",
    "#concatenate all forward reads to one fastq file\n",
    "cat \"$READSPATH\"/*_R1*.fq > \"$READSPATH\"/\"$SAMPLENAME\"_reads_R1_ALL.fastq\n",
    "\n",
    "#concatenate all reverse reads to one fastq file\n",
    "cat \"$READSPATH\"/*_R2*.fq > \"$READSPATH\"/\"$SAMPLENAME\"_reads_R2_ALL.fastq\n",
    "\n",
    "megahit --presets meta-large \\\n",
    "-1 \"$READSPATH\"/\"$SAMPLENAME\"_reads_R1_ALL.fastq \\\n",
    "-2 \"$READSPATH\"/\"$SAMPLENAME\"_reads_R2_ALL.fastq \\\n",
    "--keep-tmp-files \\\n",
    "-o /project/pi_sarah_gignouxwolfsohn_uml_edu/brooke/working/11272023/$OUTDIR --out-prefix $SAMPLENAME \n",
    "#makes dir or do you need to mkdir beforehand???\n",
    "#file: /project/pi_sarah_gignouxwolfsohn_uml_edu/brooke/11_2023bash_scripts/assembly\n",
    "#JOB ID: 15972157"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "ec0eccd5-30b7-44de-88cf-a8eef2472ce5",
   "metadata": {
    "jp-MarkdownHeadingCollapsed": true,
    "tags": []
   },
   "source": [
    "### Kraken results \n",
    "- very low classification rates "
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "28c8b11e-73ad-43ba-9955-d4237008b381",
   "metadata": {},
   "outputs": [],
   "source": [
    "Loading database information... done.\n",
    "28989033 sequences (5475.14 Mbp) processed in 69.191s (25138.2 Kseq/m, 4747.84 Mbp/m).\n",
    "  833336 sequences classified (2.87%)\n",
    "  28155697 sequences unclassified (97.13%)\n",
    "kraken2 completed successfully for sample: 052022_BEL_CBC_T2_45_PSTR\n",
    "Loading database information... done.\n",
    "20360853 sequences (3611.43 Mbp) processed in 46.989s (25998.5 Kseq/m, 4611.38 Mbp/m).\n",
    "  503123 sequences classified (2.47%)\n",
    "  19857730 sequences unclassified (97.53%)\n",
    "kraken2 completed successfully for sample: 102019_BEL_CBC_T1_28_PSTR\n",
    "Loading database information... done.\n",
    "14495243 sequences (2695.94 Mbp) processed in 35.030s (24827.9 Kseq/m, 4617.70 Mbp/m).\n",
    "  2425666 sequences classified (16.73%)\n",
    "  12069577 sequences unclassified (83.27%)\n",
    "kraken2 completed successfully for sample: 102019_BEL_CBC_T1_29_PSTR\n",
    "Loading database information... done.\n",
    "4061599 sequences (789.19 Mbp) processed in 10.412s (23404.4 Kseq/m, 4547.60 Mbp/m).\n",
    "  146335 sequences classified (3.60%)\n",
    "  3915264 sequences unclassified (96.40%)\n",
    "kraken2 completed successfully for sample: 052022_BEL_CBC_T2_46_PSTR\n",
    "Loading database information... done.\n",
    "16725843 sequences (3058.18 Mbp) processed in 40.679s (24669.8 Kseq/m, 4510.67 Mbp/m).\n",
    "  340778 sequences classified (2.04%)\n",
    "  16385065 sequences unclassified (97.96%)\n",
    "kraken2 completed successfully for sample: 052022_BEL_CBC_T1_4_PSTR\n",
    "Loading database information... done.\n",
    "21577306 sequences (4067.04 Mbp) processed in 53.055s (24401.9 Kseq/m, 4599.43 Mbp/m).\n",
    "  464294 sequences classified (2.15%)\n",
    "  21113012 sequences unclassified (97.85%)\n",
    "kraken2 completed successfully for sample: 102019_BEL_CBC_T1_25_PSTR\n",
    "Loading database information... done.\n",
    "26340368 sequences (4843.28 Mbp) processed in 62.979s (25094.3 Kseq/m, 4614.15 Mbp/m).\n",
    "  589534 sequences classified (2.24%)\n",
    "  25750834 sequences unclassified (97.76%)\n",
    "kraken2 completed successfully for sample: 052022_BEL_CBC_T3_15_PSTR\n",
    "Loading database information... done.\n",
    "38649413 sequences (6797.43 Mbp) processed in 90.692s (25569.8 Kseq/m, 4497.07 Mbp/m).\n",
    "  924266 sequences classified (2.39%)\n",
    "  37725147 sequences unclassified (97.61%)\n",
    "kraken2 completed successfully for sample: 052022_BEL_CBC_T3_3_PSTR\n",
    "Loading database information... done.\n",
    "18883375 sequences (3625.39 Mbp) processed in 46.018s (24620.8 Kseq/m, 4726.91 Mbp/m).\n",
    "  410368 sequences classified (2.17%)\n",
    "  18473007 sequences unclassified (97.83%)\n",
    "kraken2 completed successfully for sample: 102019_BEL_CBC_T1_27_PSTR\n",
    "Loading database information... done.\n",
    "21038048 sequences (4130.87 Mbp) processed in 53.740s (23488.9 Kseq/m, 4612.10 Mbp/m).\n",
    "  539903 sequences classified (2.57%)\n",
    "  20498145 sequences unclassified (97.43%)\n",
    "kraken2 completed successfully for sample: 052022_BEL_CBC_T2_14_PSTR\n",
    "Loading database information... done.\n",
    "22885217 sequences (4066.17 Mbp) processed in 54.918s (25003.0 Kseq/m, 4442.44 Mbp/m).\n",
    "  539185 sequences classified (2.36%)\n",
    "  22346032 sequences unclassified (97.64%)\n",
    "kraken2 completed successfully for sample: 052022_BEL_CBC_T1_10_PSTR\n",
    "Loading database information... done.\n",
    "9650681 sequences (1662.86 Mbp) processed in 22.238s (26038.3 Kseq/m, 4486.52 Mbp/m).\n",
    "  180594 sequences classified (1.87%)\n",
    "  9470087 sequences unclassified (98.13%)\n",
    "kraken2 completed successfully for sample: 102019_BEL_CBC_T1_26_PSTR\n",
    "Loading database information... done.\n",
    "18532636 sequences (3353.83 Mbp) processed in 44.284s (25109.7 Kseq/m, 4544.07 Mbp/m).\n",
    "  682659 sequences classified (3.68%)\n",
    "  17849977 sequences unclassified (96.32%)\n",
    "kraken2 completed successfully for sample: 052022_BEL_CBC_T2_13_PSTR\n",
    "Loading database information... done.\n",
    "21546121 sequences (3664.31 Mbp) processed in 50.901s (25397.8 Kseq/m, 4319.36 Mbp/m).\n",
    "  476625 sequences classified (2.21%)\n",
    "  21069496 sequences unclassified (97.79%)\n",
    "kraken2 completed successfully for sample: 052022_BEL_CBC_T1_11_PSTR\n",
    "Loading database information... done.\n",
    "12691364 sequences (2082.93 Mbp) processed in 28.553s (26668.9 Kseq/m, 4376.94 Mbp/m).\n",
    "  229063 sequences classified (1.80%)\n",
    "  12462301 sequences unclassified (98.20%)\n",
    "kraken2 completed successfully for sample: 102019_BEL_CBC_T2_32_PSTR\n",
    "Loading database information... done.\n",
    "27584708 sequences (5076.90 Mbp) processed in 65.510s (25264.4 Kseq/m, 4649.86 Mbp/m).\n",
    "  597929 sequences classified (2.17%)\n",
    "  26986779 sequences unclassified (97.83%)\n",
    "kraken2 completed successfully for sample: 102019_BEL_CBC_T1_30_PSTR\n",
    "Loading database information... done.\n",
    "29442462 sequences (5713.92 Mbp) processed in 74.174s (23816.4 Kseq/m, 4622.06 Mbp/m).\n",
    "  653351 sequences classified (2.22%)\n",
    "  28789111 sequences unclassified (97.78%)\n",
    "kraken2 completed successfully for sample: 102019_BEL_CBC_T2_33_PSTR\n",
    "Loading database information... done.\n",
    "16178429 sequences (2853.96 Mbp) processed in 38.795s (25021.4 Kseq/m, 4413.92 Mbp/m).\n",
    "  348045 sequences classified (2.15%)\n",
    "  15830384 sequences unclassified (97.85%)\n",
    "kraken2 completed successfully for sample: 102019_BEL_CBC_T1_31_PSTR\n",
    "Loading database information... done.\n",
    "18967626 sequences (3539.32 Mbp) processed in 45.938s (24773.6 Kseq/m, 4622.71 Mbp/m).\n",
    "  405022 sequences classified (2.14%)\n",
    "  18562604 sequences unclassified (97.86%)\n",
    "kraken2 completed successfully for sample: 102019_BEL_CBC_T2_30_PSTR\n",
    "Loading database information... done.\n",
    "9488210 sequences (1668.36 Mbp) processed in 22.954s (24801.9 Kseq/m, 4361.03 Mbp/m).\n",
    "  186817 sequences classified (1.97%)\n",
    "  9301393 sequences unclassified (98.03%)\n",
    "kraken2 completed successfully for sample: 102019_BEL_CBC_T2_31_PSTR\n",
    "Loading database information... done.\n",
    "22180483 sequences (4159.86 Mbp) processed in 54.826s (24273.5 Kseq/m, 4552.39 Mbp/m).\n",
    "  465266 sequences classified (2.10%)\n",
    "  21715217 sequences unclassified (97.90%)\n",
    "kraken2 completed successfully for sample: 102019_BEL_CBC_T3_37_PSTR\n",
    "Loading database information... done.\n",
    "27869329 sequences (4919.97 Mbp) processed in 65.213s (25641.3 Kseq/m, 4526.65 Mbp/m).\n",
    "  568439 sequences classified (2.04%)\n",
    "  27300890 sequences unclassified (97.96%)\n",
    "kraken2 completed successfully for sample: 102019_BEL_CBC_T2_36_PSTR\n",
    "Loading database information... done.\n",
    "9824136 sequences (1938.33 Mbp) processed in 24.947s (23627.6 Kseq/m, 4661.79 Mbp/m).\n",
    "  583528 sequences classified (5.94%)\n",
    "  9240608 sequences unclassified (94.06%)\n",
    "kraken2 completed successfully for sample: 052022_BEL_CBC_T2_7_PSTR\n",
    "Loading database information... done.\n",
    "30301892 sequences (5284.22 Mbp) processed in 70.163s (25912.8 Kseq/m, 4518.83 Mbp/m).\n",
    "  595304 sequences classified (1.96%)\n",
    "  29706588 sequences unclassified (98.04%)\n",
    "kraken2 completed successfully for sample: 102019_BEL_CBC_T3_36_PSTR\n",
    "Loading database information... done.\n",
    "26415194 sequences (5130.78 Mbp) processed in 67.417s (23509.1 Kseq/m, 4566.32 Mbp/m).\n",
    "  564014 sequences classified (2.14%)\n",
    "  25851180 sequences unclassified (97.86%)\n",
    "kraken2 completed successfully for sample: 102019_BEL_CBC_T3_40_PSTR\n",
    "Loading database information... done.\n",
    "22883705 sequences (4330.27 Mbp) processed in 56.080s (24483.2 Kseq/m, 4632.95 Mbp/m).\n",
    "  483196 sequences classified (2.11%)\n",
    "  22400509 sequences unclassified (97.89%)\n",
    "kraken2 completed successfully for sample: 102019_BEL_CBC_T3_35_PSTR\n",
    "Loading database information... done.\n",
    "23757549 sequences (4019.58 Mbp) processed in 55.783s (25553.7 Kseq/m, 4323.48 Mbp/m).\n",
    "  519015 sequences classified (2.18%)\n",
    "  23238534 sequences unclassified (97.82%)\n",
    "kraken2 completed successfully for sample: 102019_BEL_CBC_T2_34_PSTR\n",
    "Loading database information... done.\n",
    "26782526 sequences (4751.13 Mbp) processed in 62.857s (25565.1 Kseq/m, 4535.15 Mbp/m).\n",
    "  548177 sequences classified (2.05%)\n",
    "  26234349 sequences unclassified (97.95%)\n",
    "kraken2 completed successfully for sample: 102019_BEL_CBC_T3_34_PSTR\n",
    "Loading database information... done.\n",
    "24716567 sequences (4734.90 Mbp) processed in 59.948s (24738.2 Kseq/m, 4739.05 Mbp/m).\n",
    "  584728 sequences classified (2.37%)\n",
    "  24131839 sequences unclassified (97.63%)\n",
    "kraken2 completed successfully for sample: 052022_BEL_CBC_T3_23_PSTR\n",
    "Loading database information... done.\n",
    "32038239 sequences (5581.09 Mbp) processed in 74.541s (25788.5 Kseq/m, 4492.39 Mbp/m).\n",
    "  593633 sequences classified (1.85%)\n",
    "  31444606 sequences unclassified (98.15%)\n",
    "kraken2 completed successfully for sample: 102019_BEL_CBC_T3_38_PSTR\n",
    "Loading database information... done.\n",
    "31619956 sequences (5815.12 Mbp) processed in 78.197s (24261.8 Kseq/m, 4461.91 Mbp/m).\n",
    "  732345 sequences classified (2.32%)\n",
    "  30887611 sequences unclassified (97.68%)\n",
    "kraken2 completed successfully for sample: 052022_BEL_CBC_T2_8_PSTR\n",
    "Loading database information... done.\n",
    "15034256 sequences (2714.37 Mbp) processed in 36.043s (25026.9 Kseq/m, 4518.50 Mbp/m).\n",
    "  323816 sequences classified (2.15%)\n",
    "  14710440 sequences unclassified (97.85%)\n",
    "kraken2 completed successfully for sample: 102019_BEL_CBC_T3_39_PSTR\n",
    "Loading database information... done.\n",
    "36404364 sequences (6443.79 Mbp) processed in 85.013s (25693.1 Kseq/m, 4547.84 Mbp/m).\n",
    "  831601 sequences classified (2.28%)\n",
    "  35572763 sequences unclassified (97.72%)\n",
    "kraken2 completed successfully for sample: 052022_BEL_CBC_T1_55_PSTR\n",
    "Kraken2: All samples processed successfully."
   ]
  },
  {
   "cell_type": "markdown",
   "id": "46ff788d-ebf0-48c8-910c-c0aa005ac200",
   "metadata": {
    "tags": []
   },
   "source": [
    "### Merge samples into single otu tables \n",
    "- goal: create taxa bar plots comparing composition and abundance between samples"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 1,
   "id": "8b4a6acf-b6dd-4c55-8040-a30b964fa863",
   "metadata": {},
   "outputs": [],
   "source": [
    "import numpy as np\n",
    "import pandas as pd\n",
    "import os"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "3213ef87-87d1-435b-af10-51bbaa57c8cd",
   "metadata": {},
   "source": [
    "##### all pstr"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 4,
   "id": "ae432ba5-3ba9-4804-baa8-28bc47666cb2",
   "metadata": {},
   "outputs": [],
   "source": [
    "os.chdir('//project/pi_sarah_gignouxwolfsohn_uml_edu/brooke/pstr/taxonomy/kraken_stddb')"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 2,
   "id": "637a3aab-b7f3-4bd8-abb0-2a8e1cd4fa20",
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "['052022_BEL_CBC_T2_45_PSTR', '102019_BEL_CBC_T1_28_PSTR', '102019_BEL_CBC_T1_29_PSTR', '052022_BEL_CBC_T2_46_PSTR', '052022_BEL_CBC_T1_4_PSTR', '102019_BEL_CBC_T1_25_PSTR', '052022_BEL_CBC_T3_15_PSTR', '052022_BEL_CBC_T3_3_PSTR', '102019_BEL_CBC_T1_27_PSTR', '052022_BEL_CBC_T2_14_PSTR', '052022_BEL_CBC_T1_10_PSTR', '102019_BEL_CBC_T1_26_PSTR', '052022_BEL_CBC_T2_13_PSTR', '052022_BEL_CBC_T1_11_PSTR', '102019_BEL_CBC_T2_32_PSTR', '102019_BEL_CBC_T1_30_PSTR', '102019_BEL_CBC_T2_33_PSTR', '102019_BEL_CBC_T1_31_PSTR', '102019_BEL_CBC_T2_30_PSTR', '102019_BEL_CBC_T2_31_PSTR', '102019_BEL_CBC_T3_37_PSTR', '102019_BEL_CBC_T2_36_PSTR', '052022_BEL_CBC_T2_7_PSTR', '102019_BEL_CBC_T3_36_PSTR', '102019_BEL_CBC_T3_40_PSTR', '102019_BEL_CBC_T3_35_PSTR', '102019_BEL_CBC_T2_34_PSTR', '102019_BEL_CBC_T3_34_PSTR', '052022_BEL_CBC_T3_23_PSTR', '102019_BEL_CBC_T3_38_PSTR', '052022_BEL_CBC_T2_8_PSTR', '102019_BEL_CBC_T3_39_PSTR', '052022_BEL_CBC_T1_55_PSTR']\n",
      "052022_BEL_CBC_T2_45_PSTR,102019_BEL_CBC_T1_28_PSTR,102019_BEL_CBC_T1_29_PSTR,052022_BEL_CBC_T2_46_PSTR,052022_BEL_CBC_T1_4_PSTR,102019_BEL_CBC_T1_25_PSTR,052022_BEL_CBC_T3_15_PSTR,052022_BEL_CBC_T3_3_PSTR,102019_BEL_CBC_T1_27_PSTR,052022_BEL_CBC_T2_14_PSTR,052022_BEL_CBC_T1_10_PSTR,102019_BEL_CBC_T1_26_PSTR,052022_BEL_CBC_T2_13_PSTR,052022_BEL_CBC_T1_11_PSTR,102019_BEL_CBC_T2_32_PSTR,102019_BEL_CBC_T1_30_PSTR,102019_BEL_CBC_T2_33_PSTR,102019_BEL_CBC_T1_31_PSTR,102019_BEL_CBC_T2_30_PSTR,102019_BEL_CBC_T2_31_PSTR,102019_BEL_CBC_T3_37_PSTR,102019_BEL_CBC_T2_36_PSTR,052022_BEL_CBC_T2_7_PSTR,102019_BEL_CBC_T3_36_PSTR,102019_BEL_CBC_T3_40_PSTR,102019_BEL_CBC_T3_35_PSTR,102019_BEL_CBC_T2_34_PSTR,102019_BEL_CBC_T3_34_PSTR,052022_BEL_CBC_T3_23_PSTR,102019_BEL_CBC_T3_38_PSTR,052022_BEL_CBC_T2_8_PSTR,102019_BEL_CBC_T3_39_PSTR,052022_BEL_CBC_T1_55_PSTR\n",
      "33\n"
     ]
    }
   ],
   "source": [
    "# Set path variables and create sample list \n",
    "\n",
    "SAMPLES=\"/project/pi_sarah_gignouxwolfsohn_uml_edu/brooke/pstr/filtered_sampleids\"\n",
    "BRACKENS='//project/pi_sarah_gignouxwolfsohn_uml_edu/brooke/pstr/taxonomy/kraken_stddb'\n",
    "samplelist=[]\n",
    "\n",
    "with open(SAMPLES) as samples:\n",
    "    for line in samples:\n",
    "        samplelist.append(line.strip())\n",
    "print(samplelist)\n",
    "\n",
    "sample_names_string = ','.join(samplelist)\n",
    "print(sample_names_string)\n",
    "print(len(samplelist))"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 11,
   "id": "b605569a-04fc-4689-a281-7891ac60d30c",
   "metadata": {},
   "outputs": [],
   "source": [
    "#create symlink\n",
    "!ln -s //project/pi_sarah_gignouxwolfsohn_uml_edu/brooke/Bracken/bracken  //project/pi_sarah_gignouxwolfsohn_uml_edu/brooke/pstr/taxonomy/bracken\n",
    "!ln -s //project/pi_sarah_gignouxwolfsohn_uml_edu/brooke/Bracken/bracken-build  //project/pi_sarah_gignouxwolfsohn_uml_edu/brooke/pstr/taxonomy/bracken-build\n",
    "!cp //project/pi_sarah_gignouxwolfsohn_uml_edu/brooke/Bracken/analysis_scripts/combine_bracken_outputs.py   //project/pi_sarah_gignouxwolfsohn_uml_edu/brooke/pstr/taxonomy/combine_bracken_outputs.py"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 14,
   "id": "38f2e771-9cf7-4d4b-b8f5-d2e6e5059934",
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "PROGRAM START TIME: 06-23-2024 18:48:52\n",
      "Processing Output File //project/pi_sarah_gignouxwolfsohn_uml_edu/brooke/pstr/taxonomy/kraken_stddb/052022_BEL_CBC_T2_13_PSTR_S.bracken:: Sample 052022_BEL_CBC_T2_45_PSTR\n",
      "Processing Output File //project/pi_sarah_gignouxwolfsohn_uml_edu/brooke/pstr/taxonomy/kraken_stddb/102019_BEL_CBC_T3_39_PSTR_S.bracken:: Sample 102019_BEL_CBC_T1_28_PSTR\n",
      "Processing Output File //project/pi_sarah_gignouxwolfsohn_uml_edu/brooke/pstr/taxonomy/kraken_stddb/052022_BEL_CBC_T2_14_PSTR_S.bracken:: Sample 102019_BEL_CBC_T1_29_PSTR\n",
      "Processing Output File //project/pi_sarah_gignouxwolfsohn_uml_edu/brooke/pstr/taxonomy/kraken_stddb/102019_BEL_CBC_T2_32_PSTR_S.bracken:: Sample 052022_BEL_CBC_T2_46_PSTR\n",
      "Processing Output File //project/pi_sarah_gignouxwolfsohn_uml_edu/brooke/pstr/taxonomy/kraken_stddb/102019_BEL_CBC_T3_37_PSTR_S.bracken:: Sample 052022_BEL_CBC_T1_4_PSTR\n",
      "Processing Output File //project/pi_sarah_gignouxwolfsohn_uml_edu/brooke/pstr/taxonomy/kraken_stddb/052022_BEL_CBC_T3_23_PSTR_S.bracken:: Sample 102019_BEL_CBC_T1_25_PSTR\n",
      "Processing Output File //project/pi_sarah_gignouxwolfsohn_uml_edu/brooke/pstr/taxonomy/kraken_stddb/102019_BEL_CBC_T1_31_PSTR_S.bracken:: Sample 052022_BEL_CBC_T3_15_PSTR\n",
      "Processing Output File //project/pi_sarah_gignouxwolfsohn_uml_edu/brooke/pstr/taxonomy/kraken_stddb/102019_BEL_CBC_T1_28_PSTR_S.bracken:: Sample 052022_BEL_CBC_T3_3_PSTR\n",
      "Processing Output File //project/pi_sarah_gignouxwolfsohn_uml_edu/brooke/pstr/taxonomy/kraken_stddb/052022_BEL_CBC_T2_7_PSTR_S.bracken:: Sample 102019_BEL_CBC_T1_27_PSTR\n",
      "Processing Output File //project/pi_sarah_gignouxwolfsohn_uml_edu/brooke/pstr/taxonomy/kraken_stddb/052022_BEL_CBC_T1_10_PSTR_S.bracken:: Sample 052022_BEL_CBC_T2_14_PSTR\n",
      "Processing Output File //project/pi_sarah_gignouxwolfsohn_uml_edu/brooke/pstr/taxonomy/kraken_stddb/052022_BEL_CBC_T2_45_PSTR_S.bracken:: Sample 052022_BEL_CBC_T1_10_PSTR\n",
      "Processing Output File //project/pi_sarah_gignouxwolfsohn_uml_edu/brooke/pstr/taxonomy/kraken_stddb/102019_BEL_CBC_T2_30_PSTR_S.bracken:: Sample 102019_BEL_CBC_T1_26_PSTR\n",
      "Processing Output File //project/pi_sarah_gignouxwolfsohn_uml_edu/brooke/pstr/taxonomy/kraken_stddb/102019_BEL_CBC_T1_26_PSTR_S.bracken:: Sample 052022_BEL_CBC_T2_13_PSTR\n",
      "Processing Output File //project/pi_sarah_gignouxwolfsohn_uml_edu/brooke/pstr/taxonomy/kraken_stddb/102019_BEL_CBC_T3_35_PSTR_S.bracken:: Sample 052022_BEL_CBC_T1_11_PSTR\n",
      "Processing Output File //project/pi_sarah_gignouxwolfsohn_uml_edu/brooke/pstr/taxonomy/kraken_stddb/102019_BEL_CBC_T2_36_PSTR_S.bracken:: Sample 102019_BEL_CBC_T2_32_PSTR\n",
      "Processing Output File //project/pi_sarah_gignouxwolfsohn_uml_edu/brooke/pstr/taxonomy/kraken_stddb/052022_BEL_CBC_T1_11_PSTR_S.bracken:: Sample 102019_BEL_CBC_T1_30_PSTR\n",
      "Processing Output File //project/pi_sarah_gignouxwolfsohn_uml_edu/brooke/pstr/taxonomy/kraken_stddb/102019_BEL_CBC_T2_31_PSTR_S.bracken:: Sample 102019_BEL_CBC_T2_33_PSTR\n",
      "Processing Output File //project/pi_sarah_gignouxwolfsohn_uml_edu/brooke/pstr/taxonomy/kraken_stddb/052022_BEL_CBC_T1_55_PSTR_S.bracken:: Sample 102019_BEL_CBC_T1_31_PSTR\n",
      "Processing Output File //project/pi_sarah_gignouxwolfsohn_uml_edu/brooke/pstr/taxonomy/kraken_stddb/052022_BEL_CBC_T2_8_PSTR_S.bracken:: Sample 102019_BEL_CBC_T2_30_PSTR\n",
      "Processing Output File //project/pi_sarah_gignouxwolfsohn_uml_edu/brooke/pstr/taxonomy/kraken_stddb/102019_BEL_CBC_T1_27_PSTR_S.bracken:: Sample 102019_BEL_CBC_T2_31_PSTR\n",
      "Processing Output File //project/pi_sarah_gignouxwolfsohn_uml_edu/brooke/pstr/taxonomy/kraken_stddb/102019_BEL_CBC_T3_34_PSTR_S.bracken:: Sample 102019_BEL_CBC_T3_37_PSTR\n",
      "Processing Output File //project/pi_sarah_gignouxwolfsohn_uml_edu/brooke/pstr/taxonomy/kraken_stddb/102019_BEL_CBC_T1_30_PSTR_S.bracken:: Sample 102019_BEL_CBC_T2_36_PSTR\n",
      "Processing Output File //project/pi_sarah_gignouxwolfsohn_uml_edu/brooke/pstr/taxonomy/kraken_stddb/102019_BEL_CBC_T1_29_PSTR_S.bracken:: Sample 052022_BEL_CBC_T2_7_PSTR\n",
      "Processing Output File //project/pi_sarah_gignouxwolfsohn_uml_edu/brooke/pstr/taxonomy/kraken_stddb/052022_BEL_CBC_T3_15_PSTR_S.bracken:: Sample 102019_BEL_CBC_T3_36_PSTR\n",
      "Processing Output File //project/pi_sarah_gignouxwolfsohn_uml_edu/brooke/pstr/taxonomy/kraken_stddb/102019_BEL_CBC_T2_33_PSTR_S.bracken:: Sample 102019_BEL_CBC_T3_40_PSTR\n",
      "Processing Output File //project/pi_sarah_gignouxwolfsohn_uml_edu/brooke/pstr/taxonomy/kraken_stddb/102019_BEL_CBC_T1_25_PSTR_S.bracken:: Sample 102019_BEL_CBC_T3_35_PSTR\n",
      "Processing Output File //project/pi_sarah_gignouxwolfsohn_uml_edu/brooke/pstr/taxonomy/kraken_stddb/102019_BEL_CBC_T3_36_PSTR_S.bracken:: Sample 102019_BEL_CBC_T2_34_PSTR\n",
      "Processing Output File //project/pi_sarah_gignouxwolfsohn_uml_edu/brooke/pstr/taxonomy/kraken_stddb/102019_BEL_CBC_T2_34_PSTR_S.bracken:: Sample 102019_BEL_CBC_T3_34_PSTR\n",
      "Processing Output File //project/pi_sarah_gignouxwolfsohn_uml_edu/brooke/pstr/taxonomy/kraken_stddb/052022_BEL_CBC_T1_4_PSTR_S.bracken:: Sample 052022_BEL_CBC_T3_23_PSTR\n",
      "Processing Output File //project/pi_sarah_gignouxwolfsohn_uml_edu/brooke/pstr/taxonomy/kraken_stddb/052022_BEL_CBC_T2_46_PSTR_S.bracken:: Sample 102019_BEL_CBC_T3_38_PSTR\n",
      "Processing Output File //project/pi_sarah_gignouxwolfsohn_uml_edu/brooke/pstr/taxonomy/kraken_stddb/102019_BEL_CBC_T3_38_PSTR_S.bracken:: Sample 052022_BEL_CBC_T2_8_PSTR\n",
      "Processing Output File //project/pi_sarah_gignouxwolfsohn_uml_edu/brooke/pstr/taxonomy/kraken_stddb/052022_BEL_CBC_T3_3_PSTR_S.bracken:: Sample 102019_BEL_CBC_T3_39_PSTR\n",
      "Processing Output File //project/pi_sarah_gignouxwolfsohn_uml_edu/brooke/pstr/taxonomy/kraken_stddb/102019_BEL_CBC_T3_40_PSTR_S.bracken:: Sample 052022_BEL_CBC_T1_55_PSTR\n",
      "PROGRAM END TIME: 06-23-2024 18:48:56\n"
     ]
    }
   ],
   "source": [
    "# combine individual kraken/bracken results to single otu table at species level\n",
    "\n",
    "%run combine_bracken_outputs.py \\\n",
    "--files $BRACKENS/*.bracken \\\n",
    "--names $sample_names_string \\\n",
    "--output $BRACKENS/merged_bracken_species.txt"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 5,
   "id": "ff1c7101-2eac-4df5-8636-ca43b25f7ba4",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>name</th>\n",
       "      <th>taxonomy_lvl</th>\n",
       "      <th>052022_BEL_CBC_T2_45_PSTR_num</th>\n",
       "      <th>052022_BEL_CBC_T2_45_PSTR_frac</th>\n",
       "      <th>102019_BEL_CBC_T1_28_PSTR_num</th>\n",
       "      <th>102019_BEL_CBC_T1_28_PSTR_frac</th>\n",
       "      <th>102019_BEL_CBC_T1_29_PSTR_num</th>\n",
       "      <th>102019_BEL_CBC_T1_29_PSTR_frac</th>\n",
       "      <th>052022_BEL_CBC_T2_46_PSTR_num</th>\n",
       "      <th>052022_BEL_CBC_T2_46_PSTR_frac</th>\n",
       "      <th>...</th>\n",
       "      <th>052022_BEL_CBC_T3_23_PSTR_num</th>\n",
       "      <th>052022_BEL_CBC_T3_23_PSTR_frac</th>\n",
       "      <th>102019_BEL_CBC_T3_38_PSTR_num</th>\n",
       "      <th>102019_BEL_CBC_T3_38_PSTR_frac</th>\n",
       "      <th>052022_BEL_CBC_T2_8_PSTR_num</th>\n",
       "      <th>052022_BEL_CBC_T2_8_PSTR_frac</th>\n",
       "      <th>102019_BEL_CBC_T3_39_PSTR_num</th>\n",
       "      <th>102019_BEL_CBC_T3_39_PSTR_frac</th>\n",
       "      <th>052022_BEL_CBC_T1_55_PSTR_num</th>\n",
       "      <th>052022_BEL_CBC_T1_55_PSTR_frac</th>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>taxonomy_id</th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>2775403</th>\n",
       "      <td>Rhizobium sp. TH2</td>\n",
       "      <td>S</td>\n",
       "      <td>57</td>\n",
       "      <td>0.00008</td>\n",
       "      <td>9</td>\n",
       "      <td>0.00003</td>\n",
       "      <td>18</td>\n",
       "      <td>0.00003</td>\n",
       "      <td>1</td>\n",
       "      <td>0.00000</td>\n",
       "      <td>...</td>\n",
       "      <td>3</td>\n",
       "      <td>0.00001</td>\n",
       "      <td>1</td>\n",
       "      <td>0.00001</td>\n",
       "      <td>12</td>\n",
       "      <td>0.00002</td>\n",
       "      <td>14</td>\n",
       "      <td>0.00002</td>\n",
       "      <td>4</td>\n",
       "      <td>0.00001</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2952477</th>\n",
       "      <td>Rhizobium sp. SSA_523</td>\n",
       "      <td>S</td>\n",
       "      <td>53</td>\n",
       "      <td>0.00008</td>\n",
       "      <td>3</td>\n",
       "      <td>0.00001</td>\n",
       "      <td>20</td>\n",
       "      <td>0.00004</td>\n",
       "      <td>3</td>\n",
       "      <td>0.00001</td>\n",
       "      <td>...</td>\n",
       "      <td>9</td>\n",
       "      <td>0.00003</td>\n",
       "      <td>1</td>\n",
       "      <td>0.00001</td>\n",
       "      <td>11</td>\n",
       "      <td>0.00002</td>\n",
       "      <td>11</td>\n",
       "      <td>0.00001</td>\n",
       "      <td>16</td>\n",
       "      <td>0.00003</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2978974</th>\n",
       "      <td>Rhizobium sp. CSC1952</td>\n",
       "      <td>S</td>\n",
       "      <td>55</td>\n",
       "      <td>0.00008</td>\n",
       "      <td>4</td>\n",
       "      <td>0.00001</td>\n",
       "      <td>22</td>\n",
       "      <td>0.00004</td>\n",
       "      <td>2</td>\n",
       "      <td>0.00001</td>\n",
       "      <td>...</td>\n",
       "      <td>5</td>\n",
       "      <td>0.00001</td>\n",
       "      <td>0</td>\n",
       "      <td>0.00000</td>\n",
       "      <td>9</td>\n",
       "      <td>0.00002</td>\n",
       "      <td>13</td>\n",
       "      <td>0.00001</td>\n",
       "      <td>11</td>\n",
       "      <td>0.00002</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3039156</th>\n",
       "      <td>Rhizobium sp. CB3090</td>\n",
       "      <td>S</td>\n",
       "      <td>49</td>\n",
       "      <td>0.00007</td>\n",
       "      <td>3</td>\n",
       "      <td>0.00001</td>\n",
       "      <td>9</td>\n",
       "      <td>0.00002</td>\n",
       "      <td>1</td>\n",
       "      <td>0.00000</td>\n",
       "      <td>...</td>\n",
       "      <td>3</td>\n",
       "      <td>0.00001</td>\n",
       "      <td>11</td>\n",
       "      <td>0.00008</td>\n",
       "      <td>4</td>\n",
       "      <td>0.00001</td>\n",
       "      <td>3</td>\n",
       "      <td>0.00000</td>\n",
       "      <td>1</td>\n",
       "      <td>0.00000</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2806346</th>\n",
       "      <td>Rhizobium sp. SL42</td>\n",
       "      <td>S</td>\n",
       "      <td>48</td>\n",
       "      <td>0.00007</td>\n",
       "      <td>0</td>\n",
       "      <td>0.00000</td>\n",
       "      <td>16</td>\n",
       "      <td>0.00003</td>\n",
       "      <td>1</td>\n",
       "      <td>0.00000</td>\n",
       "      <td>...</td>\n",
       "      <td>2</td>\n",
       "      <td>0.00001</td>\n",
       "      <td>0</td>\n",
       "      <td>0.00000</td>\n",
       "      <td>1</td>\n",
       "      <td>0.00000</td>\n",
       "      <td>5</td>\n",
       "      <td>0.00001</td>\n",
       "      <td>1</td>\n",
       "      <td>0.00000</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>5 rows Ã— 68 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "                              name taxonomy_lvl  \\\n",
       "taxonomy_id                                       \n",
       "2775403          Rhizobium sp. TH2            S   \n",
       "2952477      Rhizobium sp. SSA_523            S   \n",
       "2978974      Rhizobium sp. CSC1952            S   \n",
       "3039156       Rhizobium sp. CB3090            S   \n",
       "2806346         Rhizobium sp. SL42            S   \n",
       "\n",
       "             052022_BEL_CBC_T2_45_PSTR_num  052022_BEL_CBC_T2_45_PSTR_frac  \\\n",
       "taxonomy_id                                                                  \n",
       "2775403                                 57                         0.00008   \n",
       "2952477                                 53                         0.00008   \n",
       "2978974                                 55                         0.00008   \n",
       "3039156                                 49                         0.00007   \n",
       "2806346                                 48                         0.00007   \n",
       "\n",
       "             102019_BEL_CBC_T1_28_PSTR_num  102019_BEL_CBC_T1_28_PSTR_frac  \\\n",
       "taxonomy_id                                                                  \n",
       "2775403                                  9                         0.00003   \n",
       "2952477                                  3                         0.00001   \n",
       "2978974                                  4                         0.00001   \n",
       "3039156                                  3                         0.00001   \n",
       "2806346                                  0                         0.00000   \n",
       "\n",
       "             102019_BEL_CBC_T1_29_PSTR_num  102019_BEL_CBC_T1_29_PSTR_frac  \\\n",
       "taxonomy_id                                                                  \n",
       "2775403                                 18                         0.00003   \n",
       "2952477                                 20                         0.00004   \n",
       "2978974                                 22                         0.00004   \n",
       "3039156                                  9                         0.00002   \n",
       "2806346                                 16                         0.00003   \n",
       "\n",
       "             052022_BEL_CBC_T2_46_PSTR_num  052022_BEL_CBC_T2_46_PSTR_frac  \\\n",
       "taxonomy_id                                                                  \n",
       "2775403                                  1                         0.00000   \n",
       "2952477                                  3                         0.00001   \n",
       "2978974                                  2                         0.00001   \n",
       "3039156                                  1                         0.00000   \n",
       "2806346                                  1                         0.00000   \n",
       "\n",
       "             ...  052022_BEL_CBC_T3_23_PSTR_num  \\\n",
       "taxonomy_id  ...                                  \n",
       "2775403      ...                              3   \n",
       "2952477      ...                              9   \n",
       "2978974      ...                              5   \n",
       "3039156      ...                              3   \n",
       "2806346      ...                              2   \n",
       "\n",
       "             052022_BEL_CBC_T3_23_PSTR_frac  102019_BEL_CBC_T3_38_PSTR_num  \\\n",
       "taxonomy_id                                                                  \n",
       "2775403                             0.00001                              1   \n",
       "2952477                             0.00003                              1   \n",
       "2978974                             0.00001                              0   \n",
       "3039156                             0.00001                             11   \n",
       "2806346                             0.00001                              0   \n",
       "\n",
       "             102019_BEL_CBC_T3_38_PSTR_frac  052022_BEL_CBC_T2_8_PSTR_num  \\\n",
       "taxonomy_id                                                                 \n",
       "2775403                             0.00001                            12   \n",
       "2952477                             0.00001                            11   \n",
       "2978974                             0.00000                             9   \n",
       "3039156                             0.00008                             4   \n",
       "2806346                             0.00000                             1   \n",
       "\n",
       "             052022_BEL_CBC_T2_8_PSTR_frac  102019_BEL_CBC_T3_39_PSTR_num  \\\n",
       "taxonomy_id                                                                 \n",
       "2775403                            0.00002                             14   \n",
       "2952477                            0.00002                             11   \n",
       "2978974                            0.00002                             13   \n",
       "3039156                            0.00001                              3   \n",
       "2806346                            0.00000                              5   \n",
       "\n",
       "             102019_BEL_CBC_T3_39_PSTR_frac  052022_BEL_CBC_T1_55_PSTR_num  \\\n",
       "taxonomy_id                                                                  \n",
       "2775403                             0.00002                              4   \n",
       "2952477                             0.00001                             16   \n",
       "2978974                             0.00001                             11   \n",
       "3039156                             0.00000                              1   \n",
       "2806346                             0.00001                              1   \n",
       "\n",
       "             052022_BEL_CBC_T1_55_PSTR_frac  \n",
       "taxonomy_id                                  \n",
       "2775403                             0.00001  \n",
       "2952477                             0.00003  \n",
       "2978974                             0.00002  \n",
       "3039156                             0.00000  \n",
       "2806346                             0.00000  \n",
       "\n",
       "[5 rows x 68 columns]"
      ]
     },
     "execution_count": 5,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "otu_table=pd.read_csv(\"merged_bracken_species.txt\", sep='\\t')\n",
    "otu_table.set_index(\"taxonomy_id\",inplace=True)\n",
    "otu_table.head()\n",
    "\n",
    "# Columns: \n",
    "# Sample num: Number of reads estimated for that species  \n",
    "# Sample frac: Fraction of total reads in the sample estimated for this species\n",
    "#       [NOTE: Fractions are of total reads classified, unclassified reads not accounted for]"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 6,
   "id": "0a2550bc-b633-4ec8-8c95-2c05dd025f34",
   "metadata": {},
   "outputs": [],
   "source": [
    "# Make taxa otu id table \n",
    "otu_id=otu_table.iloc[:, 0:2]\n",
    "otu_id\n",
    "# export otu id \n",
    "otu_id.to_csv(\"otu_id_species\")"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 7,
   "id": "79019ada-deda-425e-b060-4e15d86f6b45",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>name</th>\n",
       "      <th>taxonomy_lvl</th>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>taxonomy_id</th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>9606</th>\n",
       "      <td>Homo sapiens</td>\n",
       "      <td>S</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "                     name taxonomy_lvl\n",
       "taxonomy_id                           \n",
       "9606         Homo sapiens            S"
      ]
     },
     "execution_count": 7,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "# ID OTUnum for homo sapien\n",
    "humanid=otu_id['name']==\"Homo sapiens\"\n",
    "otu_id.loc[humanid,:]"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 20,
   "id": "fce58b1e-b7d2-4ecf-8eef-f232ea3c2960",
   "metadata": {},
   "outputs": [],
   "source": [
    "# export raw reads file \n",
    "# drop name, taxa, and frac cols\n",
    "drop_columns = [col for col in otu_table.columns if 'frac' in col]\n",
    "drop_columns.extend(['name', 'taxonomy_lvl'])\n",
    "otu_table_species=otu_table.drop(columns=drop_columns)\n",
    "\n",
    "# rename to drop _num\n",
    "otu_table_species.columns.str.replace('_num', '')\n",
    "\n",
    "# remove homo sapien (9606)\n",
    "otu_table_species.drop(9606,axis=0,inplace=True)\n",
    "# export otu table to csv\n",
    "otu_table_species.to_csv(\"otu_table_species\")"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 19,
   "id": "3377d8a8-b54a-488a-90de-757939f59a48",
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "{'052022_BEL_CBC_T2_45_PSTR': 28989033, '102019_BEL_CBC_T1_28_PSTR': 20360853, '102019_BEL_CBC_T1_29_PSTR': 14495243, '052022_BEL_CBC_T2_46_PSTR': 4061599, '052022_BEL_CBC_T1_4_PSTR': 16725843, '102019_BEL_CBC_T1_25_PSTR': 21577306, '052022_BEL_CBC_T3_15_PSTR': 26340368, '052022_BEL_CBC_T3_3_PSTR': 38649413, '102019_BEL_CBC_T1_27_PSTR': 18883375, '052022_BEL_CBC_T2_14_PSTR': 21038048, '052022_BEL_CBC_T1_10_PSTR': 22885217, '102019_BEL_CBC_T1_26_PSTR': 9650681, '052022_BEL_CBC_T2_13_PSTR': 18532636, '052022_BEL_CBC_T1_11_PSTR': 21546121, '102019_BEL_CBC_T2_32_PSTR': 12691364, '102019_BEL_CBC_T1_30_PSTR': 27584708, '102019_BEL_CBC_T2_33_PSTR': 29442462, '102019_BEL_CBC_T1_31_PSTR': 16178429, '102019_BEL_CBC_T2_30_PSTR': 18967626, '102019_BEL_CBC_T2_31_PSTR': 9488210, '102019_BEL_CBC_T3_37_PSTR': 22180483, '102019_BEL_CBC_T2_36_PSTR': 27869329, '052022_BEL_CBC_T2_7_PSTR': 9824136, '102019_BEL_CBC_T3_36_PSTR': 30301892, '102019_BEL_CBC_T3_40_PSTR': 26415194, '102019_BEL_CBC_T3_35_PSTR': 22883705, '102019_BEL_CBC_T2_34_PSTR': 23757549, '102019_BEL_CBC_T3_34_PSTR': 26782526, '052022_BEL_CBC_T3_23_PSTR': 24716567, '102019_BEL_CBC_T3_38_PSTR': 32038239, '052022_BEL_CBC_T2_8_PSTR': 31619956, '102019_BEL_CBC_T3_39_PSTR': 15034256, '052022_BEL_CBC_T1_55_PSTR': 36404364}\n"
     ]
    }
   ],
   "source": [
    "# still need to normalize based on total number of reads???\n",
    "# add column for normalized abundances for TOTAL number of reads per sample\n",
    "\n",
    "import re\n",
    "slurmdir='/work/pi_sarah_gignouxwolfsohn_uml_edu/brooke/pstr/slurm-kraken-pstr-23397902.out'\n",
    "\n",
    "# Read the output report and extract the total number of reads\n",
    "with open(slurmdir, 'r') as file:\n",
    "    report_text = file.read()\n",
    "\n",
    "# Extract total number of reads using regular expression\n",
    "total_reads_pattern = re.compile(r'(\\d+) sequences \\(.* Mbp\\) processed')\n",
    "total_reads_matches = total_reads_pattern.findall(report_text)\n",
    "\n",
    "# Create a dictionary to map sample ID to total reads\n",
    "total_reads_dict = {samplelist[i]: int(total_reads_matches[i]) for i in range(len(samplelist))}\n",
    "\n",
    "\n",
    "# Display the updated DataFrame\n",
    "print(total_reads_dict)\n",
    "\n",
    "# code adapted from chatgpt"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 22,
   "id": "be8c13df-bc86-4fcd-981c-da29c00410a6",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>052022_BEL_CBC_T2_45_PSTR</th>\n",
       "      <th>102019_BEL_CBC_T1_28_PSTR</th>\n",
       "      <th>102019_BEL_CBC_T1_29_PSTR</th>\n",
       "      <th>052022_BEL_CBC_T2_46_PSTR</th>\n",
       "      <th>052022_BEL_CBC_T1_4_PSTR</th>\n",
       "      <th>102019_BEL_CBC_T1_25_PSTR</th>\n",
       "      <th>052022_BEL_CBC_T3_15_PSTR</th>\n",
       "      <th>052022_BEL_CBC_T3_3_PSTR</th>\n",
       "      <th>102019_BEL_CBC_T1_27_PSTR</th>\n",
       "      <th>052022_BEL_CBC_T2_14_PSTR</th>\n",
       "      <th>...</th>\n",
       "      <th>102019_BEL_CBC_T3_40_PSTR</th>\n",
       "      <th>102019_BEL_CBC_T3_35_PSTR</th>\n",
       "      <th>102019_BEL_CBC_T2_34_PSTR</th>\n",
       "      <th>102019_BEL_CBC_T3_34_PSTR</th>\n",
       "      <th>052022_BEL_CBC_T3_23_PSTR</th>\n",
       "      <th>102019_BEL_CBC_T3_38_PSTR</th>\n",
       "      <th>052022_BEL_CBC_T2_8_PSTR</th>\n",
       "      <th>102019_BEL_CBC_T3_39_PSTR</th>\n",
       "      <th>052022_BEL_CBC_T1_55_PSTR</th>\n",
       "      <th>Sum</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>Total_reads</th>\n",
       "      <td>2.898903e+07</td>\n",
       "      <td>2.036085e+07</td>\n",
       "      <td>1.449524e+07</td>\n",
       "      <td>4.061599e+06</td>\n",
       "      <td>1.672584e+07</td>\n",
       "      <td>2.157731e+07</td>\n",
       "      <td>2.634037e+07</td>\n",
       "      <td>3.864941e+07</td>\n",
       "      <td>1.888338e+07</td>\n",
       "      <td>2.103805e+07</td>\n",
       "      <td>...</td>\n",
       "      <td>2.641519e+07</td>\n",
       "      <td>2.288370e+07</td>\n",
       "      <td>2.375755e+07</td>\n",
       "      <td>2.678253e+07</td>\n",
       "      <td>2.471657e+07</td>\n",
       "      <td>3.203824e+07</td>\n",
       "      <td>3.161996e+07</td>\n",
       "      <td>1.503426e+07</td>\n",
       "      <td>3.640436e+07</td>\n",
       "      <td>7.279167e+08</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>9606</th>\n",
       "      <td>7.555857e+03</td>\n",
       "      <td>5.887916e+03</td>\n",
       "      <td>1.331044e+04</td>\n",
       "      <td>2.255688e+04</td>\n",
       "      <td>1.094193e+04</td>\n",
       "      <td>9.801641e+03</td>\n",
       "      <td>4.783380e+03</td>\n",
       "      <td>4.166143e+03</td>\n",
       "      <td>4.590864e+03</td>\n",
       "      <td>8.851582e+03</td>\n",
       "      <td>...</td>\n",
       "      <td>9.783990e+03</td>\n",
       "      <td>7.644304e+03</td>\n",
       "      <td>9.618585e+03</td>\n",
       "      <td>6.644482e+03</td>\n",
       "      <td>5.213426e+03</td>\n",
       "      <td>2.083666e+03</td>\n",
       "      <td>7.741250e+03</td>\n",
       "      <td>2.072826e+04</td>\n",
       "      <td>6.388273e+03</td>\n",
       "      <td>3.152261e+05</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>28901</th>\n",
       "      <td>7.211003e+02</td>\n",
       "      <td>2.143329e+02</td>\n",
       "      <td>1.781757e+03</td>\n",
       "      <td>1.621283e+03</td>\n",
       "      <td>1.891923e+03</td>\n",
       "      <td>3.995355e+03</td>\n",
       "      <td>3.615743e+02</td>\n",
       "      <td>1.275957e+03</td>\n",
       "      <td>1.024764e+03</td>\n",
       "      <td>9.711928e+02</td>\n",
       "      <td>...</td>\n",
       "      <td>3.772071e+02</td>\n",
       "      <td>8.007882e+02</td>\n",
       "      <td>1.435586e+03</td>\n",
       "      <td>1.474469e+02</td>\n",
       "      <td>8.316689e+02</td>\n",
       "      <td>7.800054e+01</td>\n",
       "      <td>6.569902e+02</td>\n",
       "      <td>1.497713e+03</td>\n",
       "      <td>8.479478e+02</td>\n",
       "      <td>4.716290e+04</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1280</th>\n",
       "      <td>1.556796e+02</td>\n",
       "      <td>1.809895e+03</td>\n",
       "      <td>6.838795e+02</td>\n",
       "      <td>9.909890e+02</td>\n",
       "      <td>2.862038e+02</td>\n",
       "      <td>1.157698e+02</td>\n",
       "      <td>2.335958e+02</td>\n",
       "      <td>8.543467e+01</td>\n",
       "      <td>1.610411e+02</td>\n",
       "      <td>4.377783e+02</td>\n",
       "      <td>...</td>\n",
       "      <td>1.507769e+03</td>\n",
       "      <td>2.752614e+02</td>\n",
       "      <td>6.912750e+02</td>\n",
       "      <td>2.809332e+03</td>\n",
       "      <td>3.176412e+02</td>\n",
       "      <td>4.366345e+02</td>\n",
       "      <td>1.316257e+02</td>\n",
       "      <td>9.032838e+03</td>\n",
       "      <td>2.284891e+02</td>\n",
       "      <td>3.226823e+04</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1872427</th>\n",
       "      <td>2.759664e+00</td>\n",
       "      <td>3.437970e-01</td>\n",
       "      <td>3.035479e+00</td>\n",
       "      <td>1.477251e+00</td>\n",
       "      <td>5.978772e-01</td>\n",
       "      <td>1.019590e+00</td>\n",
       "      <td>4.555745e-01</td>\n",
       "      <td>4.657251e-01</td>\n",
       "      <td>5.401577e+00</td>\n",
       "      <td>1.283389e+00</td>\n",
       "      <td>...</td>\n",
       "      <td>7.949970e-01</td>\n",
       "      <td>5.680898e-01</td>\n",
       "      <td>1.052297e+00</td>\n",
       "      <td>4.107156e-01</td>\n",
       "      <td>2.022935e-01</td>\n",
       "      <td>6.242540e-02</td>\n",
       "      <td>4.743840e-01</td>\n",
       "      <td>1.064236e+00</td>\n",
       "      <td>7.416693e-01</td>\n",
       "      <td>2.335709e+04</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1982266</th>\n",
       "      <td>0.000000e+00</td>\n",
       "      <td>0.000000e+00</td>\n",
       "      <td>0.000000e+00</td>\n",
       "      <td>0.000000e+00</td>\n",
       "      <td>0.000000e+00</td>\n",
       "      <td>0.000000e+00</td>\n",
       "      <td>0.000000e+00</td>\n",
       "      <td>0.000000e+00</td>\n",
       "      <td>0.000000e+00</td>\n",
       "      <td>0.000000e+00</td>\n",
       "      <td>...</td>\n",
       "      <td>0.000000e+00</td>\n",
       "      <td>0.000000e+00</td>\n",
       "      <td>0.000000e+00</td>\n",
       "      <td>0.000000e+00</td>\n",
       "      <td>0.000000e+00</td>\n",
       "      <td>0.000000e+00</td>\n",
       "      <td>0.000000e+00</td>\n",
       "      <td>0.000000e+00</td>\n",
       "      <td>0.000000e+00</td>\n",
       "      <td>0.000000e+00</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1982252</th>\n",
       "      <td>0.000000e+00</td>\n",
       "      <td>0.000000e+00</td>\n",
       "      <td>0.000000e+00</td>\n",
       "      <td>0.000000e+00</td>\n",
       "      <td>0.000000e+00</td>\n",
       "      <td>0.000000e+00</td>\n",
       "      <td>0.000000e+00</td>\n",
       "      <td>0.000000e+00</td>\n",
       "      <td>0.000000e+00</td>\n",
       "      <td>0.000000e+00</td>\n",
       "      <td>...</td>\n",
       "      <td>0.000000e+00</td>\n",
       "      <td>0.000000e+00</td>\n",
       "      <td>0.000000e+00</td>\n",
       "      <td>0.000000e+00</td>\n",
       "      <td>0.000000e+00</td>\n",
       "      <td>0.000000e+00</td>\n",
       "      <td>0.000000e+00</td>\n",
       "      <td>0.000000e+00</td>\n",
       "      <td>0.000000e+00</td>\n",
       "      <td>0.000000e+00</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>702738</th>\n",
       "      <td>0.000000e+00</td>\n",
       "      <td>0.000000e+00</td>\n",
       "      <td>0.000000e+00</td>\n",
       "      <td>0.000000e+00</td>\n",
       "      <td>0.000000e+00</td>\n",
       "      <td>0.000000e+00</td>\n",
       "      <td>0.000000e+00</td>\n",
       "      <td>0.000000e+00</td>\n",
       "      <td>0.000000e+00</td>\n",
       "      <td>0.000000e+00</td>\n",
       "      <td>...</td>\n",
       "      <td>0.000000e+00</td>\n",
       "      <td>0.000000e+00</td>\n",
       "      <td>0.000000e+00</td>\n",
       "      <td>0.000000e+00</td>\n",
       "      <td>0.000000e+00</td>\n",
       "      <td>0.000000e+00</td>\n",
       "      <td>0.000000e+00</td>\n",
       "      <td>0.000000e+00</td>\n",
       "      <td>0.000000e+00</td>\n",
       "      <td>0.000000e+00</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3050249</th>\n",
       "      <td>0.000000e+00</td>\n",
       "      <td>0.000000e+00</td>\n",
       "      <td>0.000000e+00</td>\n",
       "      <td>0.000000e+00</td>\n",
       "      <td>0.000000e+00</td>\n",
       "      <td>0.000000e+00</td>\n",
       "      <td>0.000000e+00</td>\n",
       "      <td>0.000000e+00</td>\n",
       "      <td>0.000000e+00</td>\n",
       "      <td>0.000000e+00</td>\n",
       "      <td>...</td>\n",
       "      <td>0.000000e+00</td>\n",
       "      <td>0.000000e+00</td>\n",
       "      <td>0.000000e+00</td>\n",
       "      <td>0.000000e+00</td>\n",
       "      <td>0.000000e+00</td>\n",
       "      <td>0.000000e+00</td>\n",
       "      <td>0.000000e+00</td>\n",
       "      <td>0.000000e+00</td>\n",
       "      <td>0.000000e+00</td>\n",
       "      <td>0.000000e+00</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1982272</th>\n",
       "      <td>0.000000e+00</td>\n",
       "      <td>0.000000e+00</td>\n",
       "      <td>0.000000e+00</td>\n",
       "      <td>0.000000e+00</td>\n",
       "      <td>0.000000e+00</td>\n",
       "      <td>0.000000e+00</td>\n",
       "      <td>0.000000e+00</td>\n",
       "      <td>0.000000e+00</td>\n",
       "      <td>0.000000e+00</td>\n",
       "      <td>0.000000e+00</td>\n",
       "      <td>...</td>\n",
       "      <td>0.000000e+00</td>\n",
       "      <td>0.000000e+00</td>\n",
       "      <td>0.000000e+00</td>\n",
       "      <td>0.000000e+00</td>\n",
       "      <td>0.000000e+00</td>\n",
       "      <td>0.000000e+00</td>\n",
       "      <td>0.000000e+00</td>\n",
       "      <td>0.000000e+00</td>\n",
       "      <td>0.000000e+00</td>\n",
       "      <td>0.000000e+00</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>25488 rows Ã— 34 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "             052022_BEL_CBC_T2_45_PSTR  102019_BEL_CBC_T1_28_PSTR  \\\n",
       "Total_reads               2.898903e+07               2.036085e+07   \n",
       "9606                      7.555857e+03               5.887916e+03   \n",
       "28901                     7.211003e+02               2.143329e+02   \n",
       "1280                      1.556796e+02               1.809895e+03   \n",
       "1872427                   2.759664e+00               3.437970e-01   \n",
       "...                                ...                        ...   \n",
       "1982266                   0.000000e+00               0.000000e+00   \n",
       "1982252                   0.000000e+00               0.000000e+00   \n",
       "702738                    0.000000e+00               0.000000e+00   \n",
       "3050249                   0.000000e+00               0.000000e+00   \n",
       "1982272                   0.000000e+00               0.000000e+00   \n",
       "\n",
       "             102019_BEL_CBC_T1_29_PSTR  052022_BEL_CBC_T2_46_PSTR  \\\n",
       "Total_reads               1.449524e+07               4.061599e+06   \n",
       "9606                      1.331044e+04               2.255688e+04   \n",
       "28901                     1.781757e+03               1.621283e+03   \n",
       "1280                      6.838795e+02               9.909890e+02   \n",
       "1872427                   3.035479e+00               1.477251e+00   \n",
       "...                                ...                        ...   \n",
       "1982266                   0.000000e+00               0.000000e+00   \n",
       "1982252                   0.000000e+00               0.000000e+00   \n",
       "702738                    0.000000e+00               0.000000e+00   \n",
       "3050249                   0.000000e+00               0.000000e+00   \n",
       "1982272                   0.000000e+00               0.000000e+00   \n",
       "\n",
       "             052022_BEL_CBC_T1_4_PSTR  102019_BEL_CBC_T1_25_PSTR  \\\n",
       "Total_reads              1.672584e+07               2.157731e+07   \n",
       "9606                     1.094193e+04               9.801641e+03   \n",
       "28901                    1.891923e+03               3.995355e+03   \n",
       "1280                     2.862038e+02               1.157698e+02   \n",
       "1872427                  5.978772e-01               1.019590e+00   \n",
       "...                               ...                        ...   \n",
       "1982266                  0.000000e+00               0.000000e+00   \n",
       "1982252                  0.000000e+00               0.000000e+00   \n",
       "702738                   0.000000e+00               0.000000e+00   \n",
       "3050249                  0.000000e+00               0.000000e+00   \n",
       "1982272                  0.000000e+00               0.000000e+00   \n",
       "\n",
       "             052022_BEL_CBC_T3_15_PSTR  052022_BEL_CBC_T3_3_PSTR  \\\n",
       "Total_reads               2.634037e+07              3.864941e+07   \n",
       "9606                      4.783380e+03              4.166143e+03   \n",
       "28901                     3.615743e+02              1.275957e+03   \n",
       "1280                      2.335958e+02              8.543467e+01   \n",
       "1872427                   4.555745e-01              4.657251e-01   \n",
       "...                                ...                       ...   \n",
       "1982266                   0.000000e+00              0.000000e+00   \n",
       "1982252                   0.000000e+00              0.000000e+00   \n",
       "702738                    0.000000e+00              0.000000e+00   \n",
       "3050249                   0.000000e+00              0.000000e+00   \n",
       "1982272                   0.000000e+00              0.000000e+00   \n",
       "\n",
       "             102019_BEL_CBC_T1_27_PSTR  052022_BEL_CBC_T2_14_PSTR  ...  \\\n",
       "Total_reads               1.888338e+07               2.103805e+07  ...   \n",
       "9606                      4.590864e+03               8.851582e+03  ...   \n",
       "28901                     1.024764e+03               9.711928e+02  ...   \n",
       "1280                      1.610411e+02               4.377783e+02  ...   \n",
       "1872427                   5.401577e+00               1.283389e+00  ...   \n",
       "...                                ...                        ...  ...   \n",
       "1982266                   0.000000e+00               0.000000e+00  ...   \n",
       "1982252                   0.000000e+00               0.000000e+00  ...   \n",
       "702738                    0.000000e+00               0.000000e+00  ...   \n",
       "3050249                   0.000000e+00               0.000000e+00  ...   \n",
       "1982272                   0.000000e+00               0.000000e+00  ...   \n",
       "\n",
       "             102019_BEL_CBC_T3_40_PSTR  102019_BEL_CBC_T3_35_PSTR  \\\n",
       "Total_reads               2.641519e+07               2.288370e+07   \n",
       "9606                      9.783990e+03               7.644304e+03   \n",
       "28901                     3.772071e+02               8.007882e+02   \n",
       "1280                      1.507769e+03               2.752614e+02   \n",
       "1872427                   7.949970e-01               5.680898e-01   \n",
       "...                                ...                        ...   \n",
       "1982266                   0.000000e+00               0.000000e+00   \n",
       "1982252                   0.000000e+00               0.000000e+00   \n",
       "702738                    0.000000e+00               0.000000e+00   \n",
       "3050249                   0.000000e+00               0.000000e+00   \n",
       "1982272                   0.000000e+00               0.000000e+00   \n",
       "\n",
       "             102019_BEL_CBC_T2_34_PSTR  102019_BEL_CBC_T3_34_PSTR  \\\n",
       "Total_reads               2.375755e+07               2.678253e+07   \n",
       "9606                      9.618585e+03               6.644482e+03   \n",
       "28901                     1.435586e+03               1.474469e+02   \n",
       "1280                      6.912750e+02               2.809332e+03   \n",
       "1872427                   1.052297e+00               4.107156e-01   \n",
       "...                                ...                        ...   \n",
       "1982266                   0.000000e+00               0.000000e+00   \n",
       "1982252                   0.000000e+00               0.000000e+00   \n",
       "702738                    0.000000e+00               0.000000e+00   \n",
       "3050249                   0.000000e+00               0.000000e+00   \n",
       "1982272                   0.000000e+00               0.000000e+00   \n",
       "\n",
       "             052022_BEL_CBC_T3_23_PSTR  102019_BEL_CBC_T3_38_PSTR  \\\n",
       "Total_reads               2.471657e+07               3.203824e+07   \n",
       "9606                      5.213426e+03               2.083666e+03   \n",
       "28901                     8.316689e+02               7.800054e+01   \n",
       "1280                      3.176412e+02               4.366345e+02   \n",
       "1872427                   2.022935e-01               6.242540e-02   \n",
       "...                                ...                        ...   \n",
       "1982266                   0.000000e+00               0.000000e+00   \n",
       "1982252                   0.000000e+00               0.000000e+00   \n",
       "702738                    0.000000e+00               0.000000e+00   \n",
       "3050249                   0.000000e+00               0.000000e+00   \n",
       "1982272                   0.000000e+00               0.000000e+00   \n",
       "\n",
       "             052022_BEL_CBC_T2_8_PSTR  102019_BEL_CBC_T3_39_PSTR  \\\n",
       "Total_reads              3.161996e+07               1.503426e+07   \n",
       "9606                     7.741250e+03               2.072826e+04   \n",
       "28901                    6.569902e+02               1.497713e+03   \n",
       "1280                     1.316257e+02               9.032838e+03   \n",
       "1872427                  4.743840e-01               1.064236e+00   \n",
       "...                               ...                        ...   \n",
       "1982266                  0.000000e+00               0.000000e+00   \n",
       "1982252                  0.000000e+00               0.000000e+00   \n",
       "702738                   0.000000e+00               0.000000e+00   \n",
       "3050249                  0.000000e+00               0.000000e+00   \n",
       "1982272                  0.000000e+00               0.000000e+00   \n",
       "\n",
       "             052022_BEL_CBC_T1_55_PSTR           Sum  \n",
       "Total_reads               3.640436e+07  7.279167e+08  \n",
       "9606                      6.388273e+03  3.152261e+05  \n",
       "28901                     8.479478e+02  4.716290e+04  \n",
       "1280                      2.284891e+02  3.226823e+04  \n",
       "1872427                   7.416693e-01  2.335709e+04  \n",
       "...                                ...           ...  \n",
       "1982266                   0.000000e+00  0.000000e+00  \n",
       "1982252                   0.000000e+00  0.000000e+00  \n",
       "702738                    0.000000e+00  0.000000e+00  \n",
       "3050249                   0.000000e+00  0.000000e+00  \n",
       "1982272                   0.000000e+00  0.000000e+00  \n",
       "\n",
       "[25488 rows x 34 columns]"
      ]
     },
     "execution_count": 22,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "# Convert main df to raw abundances \n",
    "normalized_otus=otu_table.loc[:,otu_table.columns.str.contains(\"num\")]\n",
    "normalized_otus.columns = normalized_otus.columns.str.replace('_num', '')\n",
    "\n",
    "# Convert the dictionary to a DataFrame and transpose it\n",
    "sample_counts_df = pd.DataFrame(total_reads_dict, index=['Total_reads'])\n",
    "\n",
    "# Concatenate the original DataFrame and the new DataFrame with the total reads row\n",
    "normalized_otus = pd.concat([normalized_otus, sample_counts_df],axis=0)\n",
    "normalized_otus\n",
    "\n",
    "# Normalize: divide reads by total reads and multiply by 1 million\n",
    "normalized_otu = (normalized_otus / normalized_otus.loc['Total_reads'] * 1E6)\n",
    "# replace total reads row again \n",
    "normalized_otu.loc['Total_reads',:]=normalized_otus.loc['Total_reads',:]\n",
    "normalized_otu[\"Sum\"]=normalized_otu.sum(axis=1)\n",
    "normalized_otu.sort_values(by=\"Sum\", ascending=False)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 30,
   "id": "99f50755-a2a3-460c-a79b-288fc3f72cff",
   "metadata": {},
   "outputs": [],
   "source": [
    "# remove taxa where sum is 0 and remove human??\n",
    "# homo sapien is 9606\n",
    "normalized_otu.drop(9606,axis=0,inplace=True)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 31,
   "id": "952a2b6d-3ccf-4a1b-88d7-482758dea872",
   "metadata": {},
   "outputs": [],
   "source": [
    "# remove total reads row and sum col\n",
    "normalized_otu.drop(\"Total_reads\",axis=0,inplace=True)\n",
    "normalized_otu.drop(\"Sum\",axis=1,inplace=True)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 32,
   "id": "943325e2-9767-4185-b882-153c242841a4",
   "metadata": {},
   "outputs": [],
   "source": [
    "# export otu table to csv\n",
    "normalized_otu.to_csv(\"otu_normtable_species\")"
   ]
  }
 ],
 "metadata": {
  "kernelspec": {
   "display_name": "Python 3 (ipykernel)",
   "language": "python",
   "name": "python3"
  },
  "language_info": {
   "codemirror_mode": {
    "name": "ipython",
    "version": 3
   },
   "file_extension": ".py",
   "mimetype": "text/x-python",
   "name": "python",
   "nbconvert_exporter": "python",
   "pygments_lexer": "ipython3",
   "version": "3.11.0"
  }
 },
 "nbformat": 4,
 "nbformat_minor": 5
}
