{
 "cells": [
  {
   "cell_type": "markdown",
   "id": "7b64ba0e-624e-404a-832e-de0ab81dc810",
   "metadata": {},
   "source": [
    "### Taxonomic Profiling on MAGS\n",
    "- Can't figure out on anvio so for now trying other options\n",
    "- option 1: sylph \n",
    "- option 2: kraken"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "131c9931-f55e-41d1-ab97-20fac3a466cc",
   "metadata": {
    "jp-MarkdownHeadingCollapsed": true,
    "tags": []
   },
   "source": [
    "#### Sylph\n",
    "https://github.com/bluenote-1577/sylph\n",
    "- average nucleic identity (ANI) querying or taxonomic profiling"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "3345e46d-9dde-4dea-88cd-4be610cc4260",
   "metadata": {},
   "outputs": [],
   "source": [
    "#activate unity interactive job..need high memory for installation\n",
    "salloc -p cpu --mem 150G"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "a7b025ed-0786-4283-a3be-b5343e54d059",
   "metadata": {},
   "outputs": [],
   "source": [
    "%conda init --all"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "dfb5beeb-938f-4145-897b-99a0c4d22899",
   "metadata": {},
   "outputs": [],
   "source": [
    "#module load miniconda/22.11.1-1\n",
    "#conda create -n taxa\n",
    "%conda activate taxa"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "238f2098-b303-43b1-8ad9-604d8216f719",
   "metadata": {},
   "outputs": [],
   "source": [
    "#install sylph package\n",
    "conda install -c bioconda sylph\n",
    "#if above doesnt work do: \n",
    "conda install -c bioconda -c conda-forge sylph"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "a75cf3d5-7ff1-47db-ba4a-b2faf6cff548",
   "metadata": {},
   "outputs": [],
   "source": [
    "#set parameters\n",
    "SAMPLENAME=mcav\n",
    "CONTIGPATH=/project/pi_sarah_gignouxwolfsohn_uml_edu/brooke/results/mcav_assembly3\n",
    "CONTIGFILE=mcav.contigs-fixed.fa"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "780258a3-bc5d-46fb-acce-3419e19713b3",
   "metadata": {},
   "outputs": [],
   "source": [
    "#sketch paired-end reads\n",
    "sylph sketch -1 ./trimmed/*_R1_001_val_1.fq -2 ./trimmed/*_R2_001_val_2.fq -d \"$SAMPLENAME\"_sketches -t 50 \n",
    "#decrease -c (default =200)..."
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "8e17431b-56de-4ee6-a3d8-9de32ae1c080",
   "metadata": {},
   "outputs": [],
   "source": [
    "#create database if contigs are all in one fasta file\n",
    "sylph sketch $CONTIGPATH/$CONTIGFILE -i -o ./working/\"$SAMPLENAME\"_contig_database"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "6bba0587-108a-410a-9efd-eca97bb47bdb",
   "metadata": {},
   "outputs": [],
   "source": [
    "# query for ANI \n",
    "sylph query \"$SAMPLENAME\"_sketches/*.sylsp ./working/\"$SAMPLENAME\"_contig_database.syldb\n",
    "#output is statistics for each genome against our reads\n",
    "#redirect output??\n",
    "#Thus sylph query is not a profiler. It does not tell you the abundance of the genomes in your sample, just how similar your query genome is to the genomes within the sample."
   ]
  },
  {
   "cell_type": "raw",
   "id": "452b0bcd-b16a-4749-9e2e-88e076317274",
   "metadata": {},
   "source": [
    "Sample_file\tGenome_file\tAdjusted_ANI\tEff_cov\tANI_5-95_percentile\tEff_lambda\tLambda_5-95_percentile\tMedian_cov\tMean_cov_geq1\tContainment_ind\tNaive_ANI\tContig_name\n",
    "./trimmed/T1_12_2022_MCAV_R1_001_val_1.fq\t/project/pi_sarah_gignouxwolfsohn_uml_edu/brooke/results/mcav_assembly3/mcav.contigs-fixed.fa\t100.00\t8.213\tNA-NA\tHIGH\tNA-NA\t8\t8.213\t61/61\t100.00\tc_000000078117\n",
    "./trimmed/T1_12_2022_MCAV_R1_001_val_1.fq\t/project/pi_sarah_gignouxwolfsohn_uml_edu/brooke/results/mcav_assembly3/mcav.contigs-fixed.fa\t100.00\t16.000\tNA-NA\tHIGH\tNA-NA\t16\t15.440\t50/50\t100.00\tc_000000159761\n",
    "./trimmed/T1_12_2022_MCAV_R1_001_val_1.fq\t/project/pi_sarah_gignouxwolfsohn_uml_edu/brooke/results/mcav_assembly3/mcav.contigs-fixed.fa\t100.00\t822.000\tNA-NA\tHIGH\tNA-NA\t822\t697.089\t56/56\t100.00\tc_000000263852\n",
    "./trimmed/T1_12_2022_MCAV_R1_001_val_1.fq\t/project/pi_sarah_gignouxwolfsohn_uml_edu/brooke/results/mcav_assembly3/mcav.contigs-fixed.fa\t100.00\t33.000\tNA-NA\tHIGH\tNA-NA\t33\t32.960\t50/50\t100.00\tc_000000432420\n",
    "./trimmed/T1_12_2022_MCAV_R1_001_val_1.fq\t/project/pi_sarah_gignouxwolfsohn_uml_edu/brooke/results/mcav_assembly3/mcav.contigs-fixed.fa\t100.00\t6.722\tNA-NA\tHIGH\tNA-NA\t7\t6.722\t90/90\t100.00\tc_000000537045\n",
    "./trimmed/T1_12_2022_MCAV_R1_001_val_1.fq\t/project/pi_sarah_gignouxwolfsohn_uml_edu/brooke/results/mcav_assembly3/mcav.contigs-fixed.fa\t100.00\t8.125\tNA-NA\tHIGH\tNA-NA\t8\t8.125\t56/56\t100.00\tc_000000694960\n",
    "./trimmed/T1_12_2022_MCAV_R1_001_val_1.fq\t/project/pi_sarah_gignouxwolfsohn_uml_edu/brooke/results/mcav_assembly3/mcav.contigs-fixed.fa\t100.00\t6.635\tNA-NA\tHIGH\tNA-NA\t7\t6.635\t52/52\t100.00\tc_000000696837\n",
    "./trimmed/T1_12_2022_MCAV_R1_001_val_1.fq\t/project/pi_sarah_gignouxwolfsohn_uml_edu/brooke/results/mcav_assembly3/mcav.contigs-fixed.fa\t100.00\t28.000\tNA-NA\tHIGH\tNA-NA\t28\t27.696\t56/56\t100.00\tc_000000711824\n",
    "./trimmed/T1_12_2022_MCAV_R1_001_val_1.fq\t/project/pi_sarah_gignouxwolfsohn_uml_edu/brooke/results/mcav_assembly3/mcav.contigs-fixed.fa\t99.97\t20.000\tNA-NA\tHIGH\tNA-NA\t20\t17.292\t96/97\t99.97\tc_000000243370\n",
    "./trimmed/T1_12_2022_MCAV_R1_001_val_1.fq\t/project/pi_sarah_gignouxwolfsohn_uml_edu/brooke/results/mcav_assembly3/mcav.contigs-fixed.fa\t99.95\t10.898\tNA-NA\tHIGH\tNA-NA\t11\t10.898\t59/60\t99.95\tc_000000716083\n",
    "./trimmed/T1_12_2022_MCAV_R1_001_val_1.fq\t/project/pi_sarah_gignouxwolfsohn_uml_edu/brooke/results/mcav_assembly3/mcav.contigs-fixed.fa\t99.94\t43.000\tNA-NA\tHIGH\tNA-NA\t43\t43.240\t50/51\t99.94\tc_000000045189\n",
    "./trimmed/T1_12_2022_MCAV_R1_001_val_1.fq\t/project/pi_sarah_gignouxwolfsohn_uml_edu/brooke/results/mcav_assembly3/mcav.contigs-fixed.fa\t99.90\t4.523\tNA-NA\tHIGH\tNA-NA\t4\t4.523\t65/67\t99.90\tc_000000500791\n",
    "./trimmed/T1_12_2022_MCAV_R1_001_val_1.fq\t/project/pi_sarah_gignouxwolfsohn_uml_edu/brooke/results/mcav_assembly3/mcav.contigs-fixed.fa\t99.88\t16.000\tNA-NA\tHIGH\tNA-NA\t16\t15.500\t52/54\t99.88\tc_000000078627"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "bc53dd73-5902-4c2c-a4ee-c3d403a4505d",
   "metadata": {},
   "outputs": [],
   "source": [
    "#Import taxa database: https://github.com/bluenote-1577/sylph/wiki/Pre%E2%80%90built-databases\n",
    "wget https://storage.googleapis.com/sylph-stuff/v0.3-c200-gtdb-r214.syldb -O gtdb_database.syldb"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "6be41ed2-31bd-49af-b653-01e8a859e441",
   "metadata": {},
   "outputs": [],
   "source": [
    "#move database into folder called gtdb_genomes_reps_r214\n",
    "mkdir working/gtdb_genomes_reps_r214\n",
    "mv gtdb_database.syldb working/gtdb_genomes_reps_r214/gtdb_database.syldb\n",
    "\n",
    "#set path to GTDB\n",
    "DB=working/gtdb_genomes_reps_r214/gtdb_database.syldb\n",
    "#set input & output path\n",
    "INDIR=mcav_sketches\n",
    "OUTDIR=results/taxonomy"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "f253d868-e4b9-41f3-99b5-25dcea6cea05",
   "metadata": {},
   "outputs": [],
   "source": [
    "#maps samples against contig database???\n",
    "#read more: https://github.com/bluenote-1577/sylph/wiki/sylph-cookbook#creating-a-database-of-fasta-files\n",
    "sylph profile \"$SAMPLENAME\"_sketches/*.sylsp ./working/\"$SAMPLENAME\"_contig_database.syldb -t 50 -o ./results/\"$SAMPLENAME\"_sylph_profile.tsv"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "db901a55-27b8-4104-b1fa-9ec1df93412e",
   "metadata": {},
   "outputs": [],
   "source": [
    "#map reads against GTDB \n",
    "sylph profile $INDIR/*_MCAV_R1_001_val_1.fq.paired.sylsp \"$DB\" -t 10 -o $OUTDIR/sylph_GTDB.tsv"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "79605a6c-c5c5-4053-ac3f-a186f5a7a49b",
   "metadata": {},
   "outputs": [],
   "source": [
    "#get taxonomic profile \n",
    "#clone repo (in \"working\" dir)\n",
    "git clone https://github.com/bluenote-1577/sylph\n",
    "python working/sylph/scripts/sylph_to_taxprof.py -s $OUTDIR/sylph_GTDB.tsv -m  working/sylph/scripts/gtdb_r214_metadata.tsv.gz -o $OUTDIR/prefix_.tsv\n",
    "\n",
    "#results: only 6 samples have hits on the reference genomes database"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "c3fdbc2b-71ca-4c59-aa2e-7a827289b1fb",
   "metadata": {},
   "outputs": [],
   "source": [
    "#convert to tsv\n",
    "\n",
    "# Loop over all files \n",
    "for input_file in prefix_*_R1_001_val_1.fq.sylphmpa; do\n",
    "    # Extract the unique identifier from the file name\n",
    "    unique_identifier=$(echo \"$input_file\" | sed 's/^prefix_\\(.*\\)_R1_001_val_1\\.fq\\.sylphmpa$/\\1/')\n",
    "\n",
    "    # Generate the output TSV file name\n",
    "    output_file=\"${unique_identifier}.tsv\"\n",
    "\n",
    "    # Process the input file and save the result to the output file\n",
    "    awk -F'\\t' '{for (i=1; i<=NF; i++) printf \"%s%s\", $i, (i<NF ? \"\\t\" : \"\\n\")}' \"$input_file\" > \"$output_file\"\n",
    "\n",
    "    echo \"Processed $input_file and saved as $output_file\"\n",
    "done\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "4e496080-46b1-46bd-a409-430893cd6c7e",
   "metadata": {},
   "outputs": [],
   "source": [
    "#repeating with OceanDNA database\n",
    "wget https://storage.googleapis.com/sylph-stuff/OceanDNA-c200-v0.3.syldb\n",
    "mv OceanDNA-c200-v0.3.syldb working/gtdb_genomes_reps_r214/OceanDNA-c200-v0.3.syldb\n",
    "oceanDB=working/gtdb_genomes_reps_r214/OceanDNA-c200-v0.3.syldb\n",
    "\n",
    "sylph profile $INDIR/*_MCAV_R1_001_val_1.fq.paired.sylsp \"$oceanDB\" -t 10 -o $OUTDIR/sylph_ocean.tsv"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "5a7bde5c-0abf-4a7b-85e9-1b6c8ad68166",
   "metadata": {
    "tags": []
   },
   "source": [
    "#### Kraken\n",
    "- https://ccb.jhu.edu/software/kraken/MANUAL.html"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "8b5db189-87bf-404d-a07b-b0a64c401772",
   "metadata": {},
   "source": [
    "having trouble building \"standard\" databases...using silva anyway\n",
    "- ask unity?\n",
    "https://bioinformaticsworkbook.org/dataAnalysis/Metagenomics/Kraken.html#gsc.tab=0\n",
    "\n",
    "https://bioinformatics.stackexchange.com/questions/20589/kraken2-standard-database-failing-to-build-unexpected-ftp-path?rq=1\n",
    "\n",
    "https://github.com/DerrickWood/kraken2/issues/292"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "e1589f46-cd88-4e3c-8609-e76b7e35ce86",
   "metadata": {},
   "outputs": [],
   "source": [
    "module load miniconda/22.11.1-1\n",
    "module load kraken2/2.1.2"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "9448ab64-f479-4ca5-9724-389867783026",
   "metadata": {},
   "outputs": [],
   "source": [
    "# download SILVA database\n",
    "wget -c url https://genome-idx.s3.amazonaws.com/kraken/16S_Silva132_20200326.tgz\n",
    "#more db options (thinking about 'standard': https://benlangmead.github.io/aws-indexes/k2\n",
    "# unzip \n",
    "tar -xvzf //project/pi_sarah_gignouxwolfsohn_uml_edu/Reference_genomes/ref_databases/16S_Silva132_20200326.tgz"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "1bad34aa-9f1a-4fb1-af72-68adaa5049b5",
   "metadata": {},
   "source": [
    "##### match paired reads to 16S database "
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "db934654-5e5f-427e-b888-675066ed0c6d",
   "metadata": {},
   "outputs": [],
   "source": [
    "#!/bin/bash\n",
    "#SBATCH -c 24  # Number of Cores per Task\n",
    "#SBATCH --mem=180G  # Requested Memory\n",
    "#SBATCH -p cpu  # Partition\n",
    "#SBATCH -t 24:00:00  # Job time limit\n",
    "#SBATCH --mail-type=ALL\n",
    "#SBATCH -o //project/pi_sarah_gignouxwolfsohn_uml_edu/brooke/bash_scripts/slurm-kraken-%j.out  # %j = job ID\n",
    "\n",
    "module load miniconda/22.11.1-1\n",
    "module load kraken2/2.1.2\n",
    "\n",
    "# i think this specifically downloads NCBI and RefSeq\n",
    "#kraken2-build --build --standard --threads 24 --db $DBNAME\n",
    "#error with std db: kraken2: database (\"//project/pi_sarah_gignouxwolfsohn_uml_edu/Reference_genomes/ref_databases/standard\") does not contain necessary file taxo.k2d\n",
    "#/var/spool/slurm/slurmd/job15982456/slurm_script: line 23: --report: command not found\n",
    "# see error notes above\n",
    "\n",
    "# set variables\n",
    "DBNAME=//project/pi_sarah_gignouxwolfsohn_uml_edu/Reference_genomes/ref_databases/16S_SILVA132_k2db\n",
    "READSPATH=/project/pi_sarah_gignouxwolfsohn_uml_edu/brooke/trimmed\n",
    "OUTDIR=//project/pi_sarah_gignouxwolfsohn_uml_edu/brooke/mcav_taxonomy/kraken\n",
    "\n",
    "#mkdir /project/pi_sarah_gignouxwolfsohn_uml_edu/brooke/mcav_taxonomy\n",
    "#mkdir /project/pi_sarah_gignouxwolfsohn_uml_edu/brooke/mcav_taxonomy/kraken\n",
    "# classify each set of paired end reads against silva taxonomic database\n",
    "for f in T1_12_2022 T1_13_2022 T1_16_2019 T1_20_2019 T1_24_2019 T1_40_2022 T1_57_2022 T1_70_2022 T2_10_2022 T2_16_2019 T3_13_2022 T3_14_2019 T3_15_2019 T3_19_2022 T3_1_2019 T3_40_2022 T3_48_2022 T3_49_2022 T3_51_2022 T3_60_2022 T3_8_2019 T3_9_2019\n",
    "do\n",
    "kraken2 --db $DBNAME --threads 4 --report $OUTDIR/${f}.kreport2 --report-zero-counts --paired $READSPATH/${f}_MCAV_R1_001_val_1.fq $READSPATH/${f}_MCAV_R2_001_val_2.fq > $OUTDIR/${f}.kraken2\n",
    "done\n",
    "#add 'report-zero-counts' so that all reports have same taxa so they can be merged\n",
    "\n",
    "#bash script: kraken\n",
    "#JOB ID: 16146847\n",
    "# SUCCESS!"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "47f2598b-1b3f-470b-9c24-647f1d48f7f0",
   "metadata": {},
   "source": [
    "##### BRACKEN\n",
    "- est species abundances\n",
    "- https://github.com/jenniferlu717/Bracken/blob/master/README.md"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "75ef3909-578a-4ab8-898b-ae38dcf61303",
   "metadata": {},
   "outputs": [],
   "source": [
    "# Installation\n",
    "\n",
    "#create symlink\n",
    "ln -s //project/pi_sarah_gignouxwolfsohn_uml_edu/brooke/bash_scripts  //project/pi_sarah_gignouxwolfsohn_uml_edu/brooke\n",
    "ln -s //project/pi_sarah_gignouxwolfsohn_uml_edu/brooke/bash_scripts //project/pi_sarah_gignouxwolfsohn_uml_edu/brooke\n",
    "#execute bracken scripts from second dir path "
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "8483c2db-36da-49f2-a146-b18f7233e439",
   "metadata": {},
   "outputs": [],
   "source": [
    "#!/bin/bash\n",
    "#SBATCH -c 24  # Number of Cores per Task\n",
    "#SBATCH --mem=180G  # Requested Memory\n",
    "#SBATCH -p cpu  # Partition\n",
    "#SBATCH -t 24:00:00  # Job time limit\n",
    "#SBATCH --mail-type=ALL\n",
    "#SBATCH -o //project/pi_sarah_gignouxwolfsohn_uml_edu/brooke/bash_scripts/slurm-bracken-%j.out  # %j = job ID\n",
    "\n",
    "module load miniconda/22.11.1-1\n",
    "module activate kraken\n",
    "module load kraken2/2.1.2\n",
    "\n",
    "DBNAME=//project/pi_sarah_gignouxwolfsohn_uml_edu/Reference_genomes/ref_databases/16S_SILVA132_k2db\n",
    "THREADS=3\n",
    "KMER_LEN=35\n",
    "READ_LEN=250\n",
    "BRACKEN_BUILD_PATH=/project/pi_sarah_gignouxwolfsohn_uml_edu/brooke/bash_scripts\n",
    "# Generate bracken database - just do once\n",
    "#$BRACKEN_BUILD_PATH/bracken-build.txt -d ${DBNAME} -t ${THREADS} -k ${KMER_LEN} -l ${READ_LEN} \n",
    "#bracken-build -d ${DBNAME} -k ${KMER_LEN} -l ${READ_LEN} \n",
    "#${KRAKEN_DB}`  = location of the built Kraken 1/Kraken 2/KrakenUniq database\n",
    "#${THREADS}`    = number of threads to use with Kraken and the Bracken scripts\n",
    "#${KMER_LEN}`   = length of kmer used to build the Kraken database \n",
    "    #Kraken 1/KrakenUniq default kmer length = 31\n",
    "    #Kraken 2 default kmer length = 35\n",
    "    #Default set in the script is 35. \n",
    "#${READ_LEN}`   = the read length of your data e.g., if you are using 100 bp reads, set it to `100`. \n",
    "#need taxo.k2d, hash.k2d, and opts.k2d. in kraken2 database\n",
    "\n",
    "# Abundance Estimation\n",
    "FILEPATH=//project/pi_sarah_gignouxwolfsohn_uml_edu/brooke/mcav/taxonomy/kraken\n",
    "#ls $FILEPATH | find *.kreport2 > kraken_list.txt\n",
    "#cat kraken_list.txt| sed 's/.kreport2/''/' > SAMPLEID\n",
    "SCRIPTPATH=/project/pi_sarah_gignouxwolfsohn_uml_edu/brooke/bash_scripts/KrakenTools-master/DiversityTools\n",
    "#$BRACKEN_BUILD_PATH/bash install_bracken.sh\n",
    "level='G'\n",
    "\n",
    "while IFS= read -r line; do\n",
    "    # Execute bracken command for each file\n",
    "    bracken -d \"$DBNAME\" -i $FILEPATH/\"$line\" -o \"$FILEPATH/${line%.kreport2}_$level.bracken\" -r \"$READ_LEN\" -l \"$level\"\n",
    "done < $FILEPATH/kraken_list.txt\n",
    "#${SAMPLE}.kreport - the kraken report generated for a given dataset\n",
    "#{BRACKEN_OUTPUT_FILE}.bracken - the desired name of the output file to be generated by the code\n",
    "#The following optional parameters may be specified:\n",
    "    #${LEVEL} - Default = 'S'. This specifies that abundance estimation will calculate estimated reads for each species. Other possible options are K (kingdom level), P (phylum), C (class), O (order), F (family), and G (genus).\n",
    "    #${THRESHOLD} - Default = 10. For species classification, any species with <= 10 (or otherwise specified) reads will not receive any additional reads from higher taxonomy levels when distributing reads for abundance estimation. \n",
    "    #If another classification level is specified, thresholding will occur at that level.\n",
    "\n",
    "#repeating for species just for funsies\n",
    "while IFS= read -r line; do\n",
    "    # Execute bracken command for each file\n",
    "    bracken -d \"$DBNAME\" -i $FILEPATH/\"$line\" -o \"$FILEPATH/${line%.kreport2}_species.bracken\" -r \"$READ_LEN\"\n",
    "done < kraken_list.txt\n",
    "\n",
    "#bash script: bracken_script \n",
    "#JOB ID: 16083669"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "839dd097-ecac-4a1e-aa5a-76d1e2363be8",
   "metadata": {
    "jp-MarkdownHeadingCollapsed": true,
    "tags": []
   },
   "source": [
    "###### Genus & Species output (results from Bracken script)"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "bbb2e625-b254-4980-8420-9b77bbd9d44d",
   "metadata": {
    "tags": []
   },
   "source": [
    "\n",
    "PROGRAM START TIME: 12-14-2023 02:40:25\n",
    "BRACKEN SUMMARY (Kraken report: T1_12_2022.kreport2)\n",
    "    >>> Threshold: 0 \n",
    "    >>> Number of species in sample: 0 \n",
    "          >> Number of species with reads > threshold: 0 \n",
    "          >> Number of species with reads < threshold: 0 \n",
    "    >>> Total reads in sample: 20742228\n",
    "          >> Total reads kept at species level (reads > threshold): 0\n",
    "          >> Total reads discarded (species reads < threshold): 0\n",
    "          >> Reads distributed: 0\n",
    "          >> Reads not distributed (eg. no species above threshold): 48519\n",
    "          >> Unclassified reads: 20693709\n",
    "BRACKEN OUTPUT PRODUCED: //project/pi_sarah_gignouxwolfsohn_uml_edu/brooke/working/mcav_assembly_redo/taxonomy/kraken/bracken/T1_12_2022_species.bracken\n",
    "PROGRAM END TIME: 12-14-2023 02:40:26\n",
    "  Bracken complete.\n",
    " >> Checking for Valid Options...\n",
    " >> Running Bracken \n",
    "      >> python src/est_abundance.py -i T1_13_2022.kreport2 -o //project/pi_sarah_gignouxwolfsohn_uml_edu/brooke/working/mcav_assembly_redo/taxonomy/kraken/bracken/T1_13_2022_species.bracken -k //project/pi_sarah_gignouxwolfsohn_uml_edu/Reference_genomes/ref_databases/16S_SILVA132_k2db/database250mers.kmer_distrib -l S -t 0\n",
    "PROGRAM START TIME: 12-14-2023 02:40:26\n",
    "BRACKEN SUMMARY (Kraken report: T1_13_2022.kreport2)\n",
    "    >>> Threshold: 0 \n",
    "    >>> Number of species in sample: 0 \n",
    "          >> Number of species with reads > threshold: 0 \n",
    "          >> Number of species with reads < threshold: 0 \n",
    "    >>> Total reads in sample: 20861960\n",
    "          >> Total reads kept at species level (reads > threshold): 0\n",
    "          >> Total reads discarded (species reads < threshold): 0\n",
    "          >> Reads distributed: 0\n",
    "          >> Reads not distributed (eg. no species above threshold): 50615\n",
    "          >> Unclassified reads: 20811345\n",
    "BRACKEN OUTPUT PRODUCED: //project/pi_sarah_gignouxwolfsohn_uml_edu/brooke/working/mcav_assembly_redo/taxonomy/kraken/bracken/T1_13_2022_species.bracken\n",
    "PROGRAM END TIME: 12-14-2023 02:40:26\n",
    "  Bracken complete.\n",
    " >> Checking for Valid Options...\n",
    " >> Running Bracken \n",
    "      >> python src/est_abundance.py -i T1_16_2019.kreport2 -o //project/pi_sarah_gignouxwolfsohn_uml_edu/brooke/working/mcav_assembly_redo/taxonomy/kraken/bracken/T1_16_2019_species.bracken -k //project/pi_sarah_gignouxwolfsohn_uml_edu/Reference_genomes/ref_databases/16S_SILVA132_k2db/database250mers.kmer_distrib -l S -t 0\n",
    "PROGRAM START TIME: 12-14-2023 02:40:26\n",
    "BRACKEN SUMMARY (Kraken report: T1_16_2019.kreport2)\n",
    "    >>> Threshold: 0 \n",
    "    >>> Number of species in sample: 0 \n",
    "          >> Number of species with reads > threshold: 0 \n",
    "          >> Number of species with reads < threshold: 0 \n",
    "    >>> Total reads in sample: 25807031\n",
    "          >> Total reads kept at species level (reads > threshold): 0\n",
    "          >> Total reads discarded (species reads < threshold): 0\n",
    "          >> Reads distributed: 0\n",
    "          >> Reads not distributed (eg. no species above threshold): 71664\n",
    "          >> Unclassified reads: 25735367\n",
    "BRACKEN OUTPUT PRODUCED: //project/pi_sarah_gignouxwolfsohn_uml_edu/brooke/working/mcav_assembly_redo/taxonomy/kraken/bracken/T1_16_2019_species.bracken\n",
    "PROGRAM END TIME: 12-14-2023 02:40:26\n",
    "  Bracken complete.\n",
    " >> Checking for Valid Options...\n",
    " >> Running Bracken \n",
    "      >> python src/est_abundance.py -i T1_20_2019.kreport2 -o //project/pi_sarah_gignouxwolfsohn_uml_edu/brooke/working/mcav_assembly_redo/taxonomy/kraken/bracken/T1_20_2019_species.bracken -k //project/pi_sarah_gignouxwolfsohn_uml_edu/Reference_genomes/ref_databases/16S_SILVA132_k2db/database250mers.kmer_distrib -l S -t 0\n",
    "PROGRAM START TIME: 12-14-2023 02:40:27\n",
    "BRACKEN SUMMARY (Kraken report: T1_20_2019.kreport2)\n",
    "    >>> Threshold: 0 \n",
    "    >>> Number of species in sample: 0 \n",
    "          >> Number of species with reads > threshold: 0 \n",
    "          >> Number of species with reads < threshold: 0 \n",
    "    >>> Total reads in sample: 8517445\n",
    "          >> Total reads kept at species level (reads > threshold): 0\n",
    "          >> Total reads discarded (species reads < threshold): 0\n",
    "          >> Reads distributed: 0\n",
    "          >> Reads not distributed (eg. no species above threshold): 44133\n",
    "          >> Unclassified reads: 8473312\n",
    "BRACKEN OUTPUT PRODUCED: //project/pi_sarah_gignouxwolfsohn_uml_edu/brooke/working/mcav_assembly_redo/taxonomy/kraken/bracken/T1_20_2019_species.bracken\n",
    "PROGRAM END TIME: 12-14-2023 02:40:27\n",
    "  Bracken complete.\n",
    " >> Checking for Valid Options...\n",
    " >> Running Bracken \n",
    "      >> python src/est_abundance.py -i T1_24_2019.kreport2 -o //project/pi_sarah_gignouxwolfsohn_uml_edu/brooke/working/mcav_assembly_redo/taxonomy/kraken/bracken/T1_24_2019_species.bracken -k //project/pi_sarah_gignouxwolfsohn_uml_edu/Reference_genomes/ref_databases/16S_SILVA132_k2db/database250mers.kmer_distrib -l S -t 0\n",
    "PROGRAM START TIME: 12-14-2023 02:40:27\n",
    "BRACKEN SUMMARY (Kraken report: T1_24_2019.kreport2)\n",
    "    >>> Threshold: 0 \n",
    "    >>> Number of species in sample: 0 \n",
    "          >> Number of species with reads > threshold: 0 \n",
    "          >> Number of species with reads < threshold: 0 \n",
    "    >>> Total reads in sample: 17135017\n",
    "          >> Total reads kept at species level (reads > threshold): 0\n",
    "          >> Total reads discarded (species reads < threshold): 0\n",
    "          >> Reads distributed: 0\n",
    "          >> Reads not distributed (eg. no species above threshold): 59096\n",
    "          >> Unclassified reads: 17075921\n",
    "BRACKEN OUTPUT PRODUCED: //project/pi_sarah_gignouxwolfsohn_uml_edu/brooke/working/mcav_assembly_redo/taxonomy/kraken/bracken/T1_24_2019_species.bracken\n",
    "PROGRAM END TIME: 12-14-2023 02:40:27\n",
    "  Bracken complete.\n",
    " >> Checking for Valid Options...\n",
    " >> Running Bracken \n",
    "      >> python src/est_abundance.py -i T1_40_2022.kreport2 -o //project/pi_sarah_gignouxwolfsohn_uml_edu/brooke/working/mcav_assembly_redo/taxonomy/kraken/bracken/T1_40_2022_species.bracken -k //project/pi_sarah_gignouxwolfsohn_uml_edu/Reference_genomes/ref_databases/16S_SILVA132_k2db/database250mers.kmer_distrib -l S -t 0\n",
    "PROGRAM START TIME: 12-14-2023 02:40:27\n",
    "BRACKEN SUMMARY (Kraken report: T1_40_2022.kreport2)\n",
    "    >>> Threshold: 0 \n",
    "    >>> Number of species in sample: 0 \n",
    "          >> Number of species with reads > threshold: 0 \n",
    "          >> Number of species with reads < threshold: 0 \n",
    "    >>> Total reads in sample: 15419779\n",
    "          >> Total reads kept at species level (reads > threshold): 0\n",
    "          >> Total reads discarded (species reads < threshold): 0\n",
    "          >> Reads distributed: 0\n",
    "          >> Reads not distributed (eg. no species above threshold): 110723\n",
    "          >> Unclassified reads: 15309056\n",
    "BRACKEN OUTPUT PRODUCED: //project/pi_sarah_gignouxwolfsohn_uml_edu/brooke/working/mcav_assembly_redo/taxonomy/kraken/bracken/T1_40_2022_species.bracken\n",
    "PROGRAM END TIME: 12-14-2023 02:40:28\n",
    "  Bracken complete.\n",
    " >> Checking for Valid Options...\n",
    " >> Running Bracken \n",
    "      >> python src/est_abundance.py -i T1_57_2022.kreport2 -o //project/pi_sarah_gignouxwolfsohn_uml_edu/brooke/working/mcav_assembly_redo/taxonomy/kraken/bracken/T1_57_2022_species.bracken -k //project/pi_sarah_gignouxwolfsohn_uml_edu/Reference_genomes/ref_databases/16S_SILVA132_k2db/database250mers.kmer_distrib -l S -t 0\n",
    "PROGRAM START TIME: 12-14-2023 02:40:28\n",
    "BRACKEN SUMMARY (Kraken report: T1_57_2022.kreport2)\n",
    "    >>> Threshold: 0 \n",
    "    >>> Number of species in sample: 0 \n",
    "          >> Number of species with reads > threshold: 0 \n",
    "          >> Number of species with reads < threshold: 0 \n",
    "    >>> Total reads in sample: 31964707\n",
    "          >> Total reads kept at species level (reads > threshold): 0\n",
    "          >> Total reads discarded (species reads < threshold): 0\n",
    "          >> Reads distributed: 0\n",
    "          >> Reads not distributed (eg. no species above threshold): 157041\n",
    "          >> Unclassified reads: 31807666\n",
    "BRACKEN OUTPUT PRODUCED: //project/pi_sarah_gignouxwolfsohn_uml_edu/brooke/working/mcav_assembly_redo/taxonomy/kraken/bracken/T1_57_2022_species.bracken\n",
    "PROGRAM END TIME: 12-14-2023 02:40:28\n",
    "  Bracken complete.\n",
    " >> Checking for Valid Options...\n",
    " >> Running Bracken \n",
    "      >> python src/est_abundance.py -i T1_70_2022.kreport2 -o //project/pi_sarah_gignouxwolfsohn_uml_edu/brooke/working/mcav_assembly_redo/taxonomy/kraken/bracken/T1_70_2022_species.bracken -k //project/pi_sarah_gignouxwolfsohn_uml_edu/Reference_genomes/ref_databases/16S_SILVA132_k2db/database250mers.kmer_distrib -l S -t 0\n",
    "PROGRAM START TIME: 12-14-2023 02:40:28\n",
    "BRACKEN SUMMARY (Kraken report: T1_70_2022.kreport2)\n",
    "    >>> Threshold: 0 \n",
    "    >>> Number of species in sample: 0 \n",
    "          >> Number of species with reads > threshold: 0 \n",
    "          >> Number of species with reads < threshold: 0 \n",
    "    >>> Total reads in sample: 15776674\n",
    "          >> Total reads kept at species level (reads > threshold): 0\n",
    "          >> Total reads discarded (species reads < threshold): 0\n",
    "          >> Reads distributed: 0\n",
    "          >> Reads not distributed (eg. no species above threshold): 121001\n",
    "          >> Unclassified reads: 15655673\n",
    "BRACKEN OUTPUT PRODUCED: //project/pi_sarah_gignouxwolfsohn_uml_edu/brooke/working/mcav_assembly_redo/taxonomy/kraken/bracken/T1_70_2022_species.bracken\n",
    "PROGRAM END TIME: 12-14-2023 02:40:29\n",
    "  Bracken complete.\n",
    " >> Checking for Valid Options...\n",
    " >> Running Bracken \n",
    "      >> python src/est_abundance.py -i T2_10_2022.kreport2 -o //project/pi_sarah_gignouxwolfsohn_uml_edu/brooke/working/mcav_assembly_redo/taxonomy/kraken/bracken/T2_10_2022_species.bracken -k //project/pi_sarah_gignouxwolfsohn_uml_edu/Reference_genomes/ref_databases/16S_SILVA132_k2db/database250mers.kmer_distrib -l S -t 0\n",
    "PROGRAM START TIME: 12-14-2023 02:40:29\n",
    "BRACKEN SUMMARY (Kraken report: T2_10_2022.kreport2)\n",
    "    >>> Threshold: 0 \n",
    "    >>> Number of species in sample: 0 \n",
    "          >> Number of species with reads > threshold: 0 \n",
    "          >> Number of species with reads < threshold: 0 \n",
    "    >>> Total reads in sample: 16876812\n",
    "          >> Total reads kept at species level (reads > threshold): 0\n",
    "          >> Total reads discarded (species reads < threshold): 0\n",
    "          >> Reads distributed: 0\n",
    "          >> Reads not distributed (eg. no species above threshold): 57221\n",
    "          >> Unclassified reads: 16819591\n",
    "BRACKEN OUTPUT PRODUCED: //project/pi_sarah_gignouxwolfsohn_uml_edu/brooke/working/mcav_assembly_redo/taxonomy/kraken/bracken/T2_10_2022_species.bracken\n",
    "PROGRAM END TIME: 12-14-2023 02:40:29\n",
    "  Bracken complete.\n",
    " >> Checking for Valid Options...\n",
    " >> Running Bracken \n",
    "      >> python src/est_abundance.py -i T2_16_2019.kreport2 -o //project/pi_sarah_gignouxwolfsohn_uml_edu/brooke/working/mcav_assembly_redo/taxonomy/kraken/bracken/T2_16_2019_species.bracken -k //project/pi_sarah_gignouxwolfsohn_uml_edu/Reference_genomes/ref_databases/16S_SILVA132_k2db/database250mers.kmer_distrib -l S -t 0\n",
    "PROGRAM START TIME: 12-14-2023 02:40:29\n",
    "BRACKEN SUMMARY (Kraken report: T2_16_2019.kreport2)\n",
    "    >>> Threshold: 0 \n",
    "    >>> Number of species in sample: 0 \n",
    "          >> Number of species with reads > threshold: 0 \n",
    "          >> Number of species with reads < threshold: 0 \n",
    "    >>> Total reads in sample: 34302369\n",
    "          >> Total reads kept at species level (reads > threshold): 0\n",
    "          >> Total reads discarded (species reads < threshold): 0\n",
    "          >> Reads distributed: 0\n",
    "          >> Reads not distributed (eg. no species above threshold): 116886\n",
    "          >> Unclassified reads: 34185483\n",
    "BRACKEN OUTPUT PRODUCED: //project/pi_sarah_gignouxwolfsohn_uml_edu/brooke/working/mcav_assembly_redo/taxonomy/kraken/bracken/T2_16_2019_species.bracken\n",
    "PROGRAM END TIME: 12-14-2023 02:40:30\n",
    "  Bracken complete.\n",
    " >> Checking for Valid Options...\n",
    " >> Running Bracken \n",
    "      >> python src/est_abundance.py -i T3_13_2022.kreport2 -o //project/pi_sarah_gignouxwolfsohn_uml_edu/brooke/working/mcav_assembly_redo/taxonomy/kraken/bracken/T3_13_2022_species.bracken -k //project/pi_sarah_gignouxwolfsohn_uml_edu/Reference_genomes/ref_databases/16S_SILVA132_k2db/database250mers.kmer_distrib -l S -t 0\n",
    "PROGRAM START TIME: 12-14-2023 02:40:30\n",
    "BRACKEN SUMMARY (Kraken report: T3_13_2022.kreport2)\n",
    "    >>> Threshold: 0 \n",
    "    >>> Number of species in sample: 0 \n",
    "          >> Number of species with reads > threshold: 0 \n",
    "          >> Number of species with reads < threshold: 0 \n",
    "    >>> Total reads in sample: 13449647\n",
    "          >> Total reads kept at species level (reads > threshold): 0\n",
    "          >> Total reads discarded (species reads < threshold): 0\n",
    "          >> Reads distributed: 0\n",
    "          >> Reads not distributed (eg. no species above threshold): 41610\n",
    "          >> Unclassified reads: 13408037\n",
    "BRACKEN OUTPUT PRODUCED: //project/pi_sarah_gignouxwolfsohn_uml_edu/brooke/working/mcav_assembly_redo/taxonomy/kraken/bracken/T3_13_2022_species.bracken\n",
    "PROGRAM END TIME: 12-14-2023 02:40:30\n",
    "  Bracken complete.\n",
    " >> Checking for Valid Options...\n",
    " >> Running Bracken \n",
    "      >> python src/est_abundance.py -i T3_14_2019.kreport2 -o //project/pi_sarah_gignouxwolfsohn_uml_edu/brooke/working/mcav_assembly_redo/taxonomy/kraken/bracken/T3_14_2019_species.bracken -k //project/pi_sarah_gignouxwolfsohn_uml_edu/Reference_genomes/ref_databases/16S_SILVA132_k2db/database250mers.kmer_distrib -l S -t 0\n",
    "PROGRAM START TIME: 12-14-2023 02:40:30\n",
    "BRACKEN SUMMARY (Kraken report: T3_14_2019.kreport2)\n",
    "    >>> Threshold: 0 \n",
    "    >>> Number of species in sample: 0 \n",
    "          >> Number of species with reads > threshold: 0 \n",
    "          >> Number of species with reads < threshold: 0 \n",
    "    >>> Total reads in sample: 21400877\n",
    "          >> Total reads kept at species level (reads > threshold): 0\n",
    "          >> Total reads discarded (species reads < threshold): 0\n",
    "          >> Reads distributed: 0\n",
    "          >> Reads not distributed (eg. no species above threshold): 91615\n",
    "          >> Unclassified reads: 21309262\n",
    "BRACKEN OUTPUT PRODUCED: //project/pi_sarah_gignouxwolfsohn_uml_edu/brooke/working/mcav_assembly_redo/taxonomy/kraken/bracken/T3_14_2019_species.bracken\n",
    "PROGRAM END TIME: 12-14-2023 02:40:31\n",
    "  Bracken complete.\n",
    " >> Checking for Valid Options...\n",
    " >> Running Bracken \n",
    "      >> python src/est_abundance.py -i T3_15_2019.kreport2 -o //project/pi_sarah_gignouxwolfsohn_uml_edu/brooke/working/mcav_assembly_redo/taxonomy/kraken/bracken/T3_15_2019_species.bracken -k //project/pi_sarah_gignouxwolfsohn_uml_edu/Reference_genomes/ref_databases/16S_SILVA132_k2db/database250mers.kmer_distrib -l S -t 0\n",
    "PROGRAM START TIME: 12-14-2023 02:40:31\n",
    "BRACKEN SUMMARY (Kraken report: T3_15_2019.kreport2)\n",
    "    >>> Threshold: 0 \n",
    "    >>> Number of species in sample: 0 \n",
    "          >> Number of species with reads > threshold: 0 \n",
    "          >> Number of species with reads < threshold: 0 \n",
    "    >>> Total reads in sample: 19854618\n",
    "          >> Total reads kept at species level (reads > threshold): 0\n",
    "          >> Total reads discarded (species reads < threshold): 0\n",
    "          >> Reads distributed: 0\n",
    "          >> Reads not distributed (eg. no species above threshold): 51464\n",
    "          >> Unclassified reads: 19803154\n",
    "BRACKEN OUTPUT PRODUCED: //project/pi_sarah_gignouxwolfsohn_uml_edu/brooke/working/mcav_assembly_redo/taxonomy/kraken/bracken/T3_15_2019_species.bracken\n",
    "PROGRAM END TIME: 12-14-2023 02:40:31\n",
    "  Bracken complete.\n",
    " >> Checking for Valid Options...\n",
    " >> Running Bracken \n",
    "      >> python src/est_abundance.py -i T3_19_2022.kreport2 -o //project/pi_sarah_gignouxwolfsohn_uml_edu/brooke/working/mcav_assembly_redo/taxonomy/kraken/bracken/T3_19_2022_species.bracken -k //project/pi_sarah_gignouxwolfsohn_uml_edu/Reference_genomes/ref_databases/16S_SILVA132_k2db/database250mers.kmer_distrib -l S -t 0\n",
    "PROGRAM START TIME: 12-14-2023 02:40:31\n",
    "BRACKEN SUMMARY (Kraken report: T3_19_2022.kreport2)\n",
    "    >>> Threshold: 0 \n",
    "    >>> Number of species in sample: 0 \n",
    "          >> Number of species with reads > threshold: 0 \n",
    "          >> Number of species with reads < threshold: 0 \n",
    "    >>> Total reads in sample: 9044371\n",
    "          >> Total reads kept at species level (reads > threshold): 0\n",
    "          >> Total reads discarded (species reads < threshold): 0\n",
    "          >> Reads distributed: 0\n",
    "          >> Reads not distributed (eg. no species above threshold): 93872\n",
    "          >> Unclassified reads: 8950499\n",
    "BRACKEN OUTPUT PRODUCED: //project/pi_sarah_gignouxwolfsohn_uml_edu/brooke/working/mcav_assembly_redo/taxonomy/kraken/bracken/T3_19_2022_species.bracken\n",
    "PROGRAM END TIME: 12-14-2023 02:40:31\n",
    "  Bracken complete.\n",
    " >> Checking for Valid Options...\n",
    " >> Running Bracken \n",
    "      >> python src/est_abundance.py -i T3_1_2019.kreport2 -o //project/pi_sarah_gignouxwolfsohn_uml_edu/brooke/working/mcav_assembly_redo/taxonomy/kraken/bracken/T3_1_2019_species.bracken -k //project/pi_sarah_gignouxwolfsohn_uml_edu/Reference_genomes/ref_databases/16S_SILVA132_k2db/database250mers.kmer_distrib -l S -t 0\n",
    "PROGRAM START TIME: 12-14-2023 02:40:31\n",
    "BRACKEN SUMMARY (Kraken report: T3_1_2019.kreport2)\n",
    "    >>> Threshold: 0 \n",
    "    >>> Number of species in sample: 0 \n",
    "          >> Number of species with reads > threshold: 0 \n",
    "          >> Number of species with reads < threshold: 0 \n",
    "    >>> Total reads in sample: 29977945\n",
    "          >> Total reads kept at species level (reads > threshold): 0\n",
    "          >> Total reads discarded (species reads < threshold): 0\n",
    "          >> Reads distributed: 0\n",
    "          >> Reads not distributed (eg. no species above threshold): 99522\n",
    "          >> Unclassified reads: 29878423\n",
    "BRACKEN OUTPUT PRODUCED: //project/pi_sarah_gignouxwolfsohn_uml_edu/brooke/working/mcav_assembly_redo/taxonomy/kraken/bracken/T3_1_2019_species.bracken\n",
    "PROGRAM END TIME: 12-14-2023 02:40:32\n",
    "  Bracken complete.\n",
    " >> Checking for Valid Options...\n",
    " >> Running Bracken \n",
    "      >> python src/est_abundance.py -i T3_40_2022.kreport2 -o //project/pi_sarah_gignouxwolfsohn_uml_edu/brooke/working/mcav_assembly_redo/taxonomy/kraken/bracken/T3_40_2022_species.bracken -k //project/pi_sarah_gignouxwolfsohn_uml_edu/Reference_genomes/ref_databases/16S_SILVA132_k2db/database250mers.kmer_distrib -l S -t 0\n",
    "PROGRAM START TIME: 12-14-2023 02:40:32\n",
    "BRACKEN SUMMARY (Kraken report: T3_40_2022.kreport2)\n",
    "    >>> Threshold: 0 \n",
    "    >>> Number of species in sample: 0 \n",
    "          >> Number of species with reads > threshold: 0 \n",
    "          >> Number of species with reads < threshold: 0 \n",
    "    >>> Total reads in sample: 26755227\n",
    "          >> Total reads kept at species level (reads > threshold): 0\n",
    "          >> Total reads discarded (species reads < threshold): 0\n",
    "          >> Reads distributed: 0\n",
    "          >> Reads not distributed (eg. no species above threshold): 77871\n",
    "          >> Unclassified reads: 26677356\n",
    "BRACKEN OUTPUT PRODUCED: //project/pi_sarah_gignouxwolfsohn_uml_edu/brooke/working/mcav_assembly_redo/taxonomy/kraken/bracken/T3_40_2022_species.bracken\n",
    "PROGRAM END TIME: 12-14-2023 02:40:32\n",
    "  Bracken complete.\n",
    " >> Checking for Valid Options...\n",
    " >> Running Bracken \n",
    "      >> python src/est_abundance.py -i T3_48_2022.kreport2 -o //project/pi_sarah_gignouxwolfsohn_uml_edu/brooke/working/mcav_assembly_redo/taxonomy/kraken/bracken/T3_48_2022_species.bracken -k //project/pi_sarah_gignouxwolfsohn_uml_edu/Reference_genomes/ref_databases/16S_SILVA132_k2db/database250mers.kmer_distrib -l S -t 0\n",
    "PROGRAM START TIME: 12-14-2023 02:40:32\n",
    "BRACKEN SUMMARY (Kraken report: T3_48_2022.kreport2)\n",
    "    >>> Threshold: 0 \n",
    "    >>> Number of species in sample: 0 \n",
    "          >> Number of species with reads > threshold: 0 \n",
    "          >> Number of species with reads < threshold: 0 \n",
    "    >>> Total reads in sample: 13255150\n",
    "          >> Total reads kept at species level (reads > threshold): 0\n",
    "          >> Total reads discarded (species reads < threshold): 0\n",
    "          >> Reads distributed: 0\n",
    "          >> Reads not distributed (eg. no species above threshold): 133945\n",
    "          >> Unclassified reads: 13121205\n",
    "BRACKEN OUTPUT PRODUCED: //project/pi_sarah_gignouxwolfsohn_uml_edu/brooke/working/mcav_assembly_redo/taxonomy/kraken/bracken/T3_48_2022_species.bracken\n",
    "PROGRAM END TIME: 12-14-2023 02:40:33\n",
    "  Bracken complete.\n",
    " >> Checking for Valid Options...\n",
    " >> Running Bracken \n",
    "      >> python src/est_abundance.py -i T3_49_2022.kreport2 -o //project/pi_sarah_gignouxwolfsohn_uml_edu/brooke/working/mcav_assembly_redo/taxonomy/kraken/bracken/T3_49_2022_species.bracken -k //project/pi_sarah_gignouxwolfsohn_uml_edu/Reference_genomes/ref_databases/16S_SILVA132_k2db/database250mers.kmer_distrib -l S -t 0\n",
    "PROGRAM START TIME: 12-14-2023 02:40:33\n",
    "BRACKEN SUMMARY (Kraken report: T3_49_2022.kreport2)\n",
    "    >>> Threshold: 0 \n",
    "    >>> Number of species in sample: 0 \n",
    "          >> Number of species with reads > threshold: 0 \n",
    "          >> Number of species with reads < threshold: 0 \n",
    "    >>> Total reads in sample: 15749618\n",
    "          >> Total reads kept at species level (reads > threshold): 0\n",
    "          >> Total reads discarded (species reads < threshold): 0\n",
    "          >> Reads distributed: 0\n",
    "          >> Reads not distributed (eg. no species above threshold): 149084\n",
    "          >> Unclassified reads: 15600534\n",
    "BRACKEN OUTPUT PRODUCED: //project/pi_sarah_gignouxwolfsohn_uml_edu/brooke/working/mcav_assembly_redo/taxonomy/kraken/bracken/T3_49_2022_species.bracken\n",
    "PROGRAM END TIME: 12-14-2023 02:40:33\n",
    "  Bracken complete.\n",
    " >> Checking for Valid Options...\n",
    " >> Running Bracken \n",
    "      >> python src/est_abundance.py -i T3_51_2022.kreport2 -o //project/pi_sarah_gignouxwolfsohn_uml_edu/brooke/working/mcav_assembly_redo/taxonomy/kraken/bracken/T3_51_2022_species.bracken -k //project/pi_sarah_gignouxwolfsohn_uml_edu/Reference_genomes/ref_databases/16S_SILVA132_k2db/database250mers.kmer_distrib -l S -t 0\n",
    "PROGRAM START TIME: 12-14-2023 02:40:33\n",
    "BRACKEN SUMMARY (Kraken report: T3_51_2022.kreport2)\n",
    "    >>> Threshold: 0 \n",
    "    >>> Number of species in sample: 0 \n",
    "          >> Number of species with reads > threshold: 0 \n",
    "          >> Number of species with reads < threshold: 0 \n",
    "    >>> Total reads in sample: 13877944\n",
    "          >> Total reads kept at species level (reads > threshold): 0\n",
    "          >> Total reads discarded (species reads < threshold): 0\n",
    "          >> Reads distributed: 0\n",
    "          >> Reads not distributed (eg. no species above threshold): 50639\n",
    "          >> Unclassified reads: 13827305\n",
    "BRACKEN OUTPUT PRODUCED: //project/pi_sarah_gignouxwolfsohn_uml_edu/brooke/working/mcav_assembly_redo/taxonomy/kraken/bracken/T3_51_2022_species.bracken\n",
    "PROGRAM END TIME: 12-14-2023 02:40:34\n",
    "  Bracken complete.\n",
    " >> Checking for Valid Options...\n",
    " >> Running Bracken \n",
    "      >> python src/est_abundance.py -i T3_8_2019.kreport2 -o //project/pi_sarah_gignouxwolfsohn_uml_edu/brooke/working/mcav_assembly_redo/taxonomy/kraken/bracken/T3_8_2019_species.bracken -k //project/pi_sarah_gignouxwolfsohn_uml_edu/Reference_genomes/ref_databases/16S_SILVA132_k2db/database250mers.kmer_distrib -l S -t 0\n",
    "PROGRAM START TIME: 12-14-2023 02:40:34\n",
    "BRACKEN SUMMARY (Kraken report: T3_8_2019.kreport2)\n",
    "    >>> Threshold: 0 \n",
    "    >>> Number of species in sample: 0 \n",
    "          >> Number of species with reads > threshold: 0 \n",
    "          >> Number of species with reads < threshold: 0 \n",
    "    >>> Total reads in sample: 16453406\n",
    "          >> Total reads kept at species level (reads > threshold): 0\n",
    "          >> Total reads discarded (species reads < threshold): 0\n",
    "          >> Reads distributed: 0\n",
    "          >> Reads not distributed (eg. no species above threshold): 94662\n",
    "          >> Unclassified reads: 16358744\n",
    "BRACKEN OUTPUT PRODUCED: //project/pi_sarah_gignouxwolfsohn_uml_edu/brooke/working/mcav_assembly_redo/taxonomy/kraken/bracken/T3_8_2019_species.bracken\n",
    "PROGRAM END TIME: 12-14-2023 02:40:34\n",
    "  Bracken complete.\n",
    " >> Checking for Valid Options...\n",
    " >> Running Bracken \n",
    "      >> python src/est_abundance.py -i T3_9_2019.kreport2 -o //project/pi_sarah_gignouxwolfsohn_uml_edu/brooke/working/mcav_assembly_redo/taxonomy/kraken/bracken/T3_9_2019_species.bracken -k //project/pi_sarah_gignouxwolfsohn_uml_edu/Reference_genomes/ref_databases/16S_SILVA132_k2db/database250mers.kmer_distrib -l S -t 0\n",
    "PROGRAM START TIME: 12-14-2023 02:40:34\n",
    "BRACKEN SUMMARY (Kraken report: T3_9_2019.kreport2)\n",
    "    >>> Threshold: 0 \n",
    "    >>> Number of species in sample: 0 \n",
    "          >> Number of species with reads > threshold: 0 \n",
    "          >> Number of species with reads < threshold: 0 \n",
    "    >>> Total reads in sample: 14032207\n",
    "          >> Total reads kept at species level (reads > threshold): 0\n",
    "          >> Total reads discarded (species reads < threshold): 0\n",
    "          >> Reads distributed: 0\n",
    "          >> Reads not distributed (eg. no species above threshold): 69979\n",
    "          >> Unclassified reads: 13962228\n",
    "BRACKEN OUTPUT PRODUCED: //project/pi_sarah_gignouxwolfsohn_uml_edu/brooke/working/mcav_assembly_redo/taxonomy/kraken/bracken/T3_9_2019_species.bracken\n",
    "PROGRAM END TIME: 12-14-2023 02:40:34\n",
    "  Bracken complete."
   ]
  },
  {
   "cell_type": "markdown",
   "id": "71c5adf3-7bfe-4980-bf82-112b77830a76",
   "metadata": {
    "jp-MarkdownHeadingCollapsed": true,
    "tags": []
   },
   "source": [
    "Species output - NO SPECIES OUTPUT IN ANY SAMPLES\n",
    "ROGRAM START TIME: 12-14-2023 02:40:25\n",
    "BRACKEN SUMMARY (Kraken report: T1_12_2022.kreport2)\n",
    "    >>> Threshold: 0 \n",
    "    >>> Number of species in sample: 0 \n",
    "          >> Number of species with reads > threshold: 0 \n",
    "          >> Number of species with reads < threshold: 0 \n",
    "    >>> Total reads in sample: 20742228\n",
    "          >> Total reads kept at species level (reads > threshold): 0\n",
    "          >> Total reads discarded (species reads < threshold): 0\n",
    "          >> Reads distributed: 0\n",
    "          >> Reads not distributed (eg. no species above threshold): 48519\n",
    "          >> Unclassified reads: 20693709\n",
    "BRACKEN OUTPUT PRODUCED: //project/pi_sarah_gignouxwolfsohn_uml_edu/brooke/working/mcav_assembly_redo/taxonomy/kraken/bracken/T1_12_2022_species.bracken\n",
    "PROGRAM END TIME: 12-14-2023 02:40:26\n",
    "  Bracken complete.\n",
    " >> Checking for Valid Options...\n",
    " >> Running Bracken \n",
    "      >> python src/est_abundance.py -i T1_13_2022.kreport2 -o //project/pi_sarah_gignouxwolfsohn_uml_edu/brooke/working/mcav_assembly_redo/taxonomy/kraken/bracken/T1_13_2022_species.bracken -k //project/pi_sarah_gignouxwolfsohn_uml_edu/Reference_genomes/ref_databases/16S_SILVA132_k2db/database250mers.kmer_distrib -l S -t 0\n",
    "PROGRAM START TIME: 12-14-2023 02:40:26\n",
    "BRACKEN SUMMARY (Kraken report: T1_13_2022.kreport2)\n",
    "    >>> Threshold: 0 \n",
    "    >>> Number of species in sample: 0 \n",
    "          >> Number of species with reads > threshold: 0 \n",
    "          >> Number of species with reads < threshold: 0 \n",
    "    >>> Total reads in sample: 20861960\n",
    "          >> Total reads kept at species level (reads > threshold): 0\n",
    "          >> Total reads discarded (species reads < threshold): 0\n",
    "          >> Reads distributed: 0\n",
    "          >> Reads not distributed (eg. no species above threshold): 50615\n",
    "          >> Unclassified reads: 20811345\n",
    "BRACKEN OUTPUT PRODUCED: //project/pi_sarah_gignouxwolfsohn_uml_edu/brooke/working/mcav_assembly_redo/taxonomy/kraken/bracken/T1_13_2022_species.bracken\n",
    "PROGRAM END TIME: 12-14-2023 02:40:26\n",
    "  Bracken complete.\n",
    " >> Checking for Valid Options...\n",
    " >> Running Bracken \n",
    "      >> python src/est_abundance.py -i T1_16_2019.kreport2 -o //project/pi_sarah_gignouxwolfsohn_uml_edu/brooke/working/mcav_assembly_redo/taxonomy/kraken/bracken/T1_16_2019_species.bracken -k //project/pi_sarah_gignouxwolfsohn_uml_edu/Reference_genomes/ref_databases/16S_SILVA132_k2db/database250mers.kmer_distrib -l S -t 0\n",
    "PROGRAM START TIME: 12-14-2023 02:40:26\n",
    "BRACKEN SUMMARY (Kraken report: T1_16_2019.kreport2)\n",
    "    >>> Threshold: 0 \n",
    "    >>> Number of species in sample: 0 \n",
    "          >> Number of species with reads > threshold: 0 \n",
    "          >> Number of species with reads < threshold: 0 \n",
    "    >>> Total reads in sample: 25807031\n",
    "          >> Total reads kept at species level (reads > threshold): 0\n",
    "          >> Total reads discarded (species reads < threshold): 0\n",
    "          >> Reads distributed: 0\n",
    "          >> Reads not distributed (eg. no species above threshold): 71664\n",
    "          >> Unclassified reads: 25735367\n",
    "BRACKEN OUTPUT PRODUCED: //project/pi_sarah_gignouxwolfsohn_uml_edu/brooke/working/mcav_assembly_redo/taxonomy/kraken/bracken/T1_16_2019_species.bracken\n",
    "PROGRAM END TIME: 12-14-2023 02:40:26\n",
    "  Bracken complete.\n",
    " >> Checking for Valid Options...\n",
    " >> Running Bracken \n",
    "      >> python src/est_abundance.py -i T1_20_2019.kreport2 -o //project/pi_sarah_gignouxwolfsohn_uml_edu/brooke/working/mcav_assembly_redo/taxonomy/kraken/bracken/T1_20_2019_species.bracken -k //project/pi_sarah_gignouxwolfsohn_uml_edu/Reference_genomes/ref_databases/16S_SILVA132_k2db/database250mers.kmer_distrib -l S -t 0\n",
    "PROGRAM START TIME: 12-14-2023 02:40:27\n",
    "BRACKEN SUMMARY (Kraken report: T1_20_2019.kreport2)\n",
    "    >>> Threshold: 0 \n",
    "    >>> Number of species in sample: 0 \n",
    "          >> Number of species with reads > threshold: 0 \n",
    "          >> Number of species with reads < threshold: 0 \n",
    "    >>> Total reads in sample: 8517445\n",
    "          >> Total reads kept at species level (reads > threshold): 0\n",
    "          >> Total reads discarded (species reads < threshold): 0\n",
    "          >> Reads distributed: 0\n",
    "          >> Reads not distributed (eg. no species above threshold): 44133\n",
    "          >> Unclassified reads: 8473312\n",
    "BRACKEN OUTPUT PRODUCED: //project/pi_sarah_gignouxwolfsohn_uml_edu/brooke/working/mcav_assembly_redo/taxonomy/kraken/bracken/T1_20_2019_species.bracken\n",
    "PROGRAM END TIME: 12-14-2023 02:40:27\n",
    "  Bracken complete.\n",
    " >> Checking for Valid Options...\n",
    " >> Running Bracken \n",
    "      >> python src/est_abundance.py -i T1_24_2019.kreport2 -o //project/pi_sarah_gignouxwolfsohn_uml_edu/brooke/working/mcav_assembly_redo/taxonomy/kraken/bracken/T1_24_2019_species.bracken -k //project/pi_sarah_gignouxwolfsohn_uml_edu/Reference_genomes/ref_databases/16S_SILVA132_k2db/database250mers.kmer_distrib -l S -t 0\n",
    "PROGRAM START TIME: 12-14-2023 02:40:27\n",
    "BRACKEN SUMMARY (Kraken report: T1_24_2019.kreport2)\n",
    "    >>> Threshold: 0 \n",
    "    >>> Number of species in sample: 0 \n",
    "          >> Number of species with reads > threshold: 0 \n",
    "          >> Number of species with reads < threshold: 0 \n",
    "    >>> Total reads in sample: 17135017\n",
    "          >> Total reads kept at species level (reads > threshold): 0\n",
    "          >> Total reads discarded (species reads < threshold): 0\n",
    "          >> Reads distributed: 0\n",
    "          >> Reads not distributed (eg. no species above threshold): 59096\n",
    "          >> Unclassified reads: 17075921\n",
    "BRACKEN OUTPUT PRODUCED: //project/pi_sarah_gignouxwolfsohn_uml_edu/brooke/working/mcav_assembly_redo/taxonomy/kraken/bracken/T1_24_2019_species.bracken\n",
    "PROGRAM END TIME: 12-14-2023 02:40:27\n",
    "  Bracken complete.\n",
    " >> Checking for Valid Options...\n",
    " >> Running Bracken \n",
    "      >> python src/est_abundance.py -i T1_40_2022.kreport2 -o //project/pi_sarah_gignouxwolfsohn_uml_edu/brooke/working/mcav_assembly_redo/taxonomy/kraken/bracken/T1_40_2022_species.bracken -k //project/pi_sarah_gignouxwolfsohn_uml_edu/Reference_genomes/ref_databases/16S_SILVA132_k2db/database250mers.kmer_distrib -l S -t 0\n",
    "PROGRAM START TIME: 12-14-2023 02:40:27\n",
    "BRACKEN SUMMARY (Kraken report: T1_40_2022.kreport2)\n",
    "    >>> Threshold: 0 \n",
    "    >>> Number of species in sample: 0 \n",
    "          >> Number of species with reads > threshold: 0 \n",
    "          >> Number of species with reads < threshold: 0 \n",
    "    >>> Total reads in sample: 15419779\n",
    "          >> Total reads kept at species level (reads > threshold): 0\n",
    "          >> Total reads discarded (species reads < threshold): 0\n",
    "          >> Reads distributed: 0\n",
    "          >> Reads not distributed (eg. no species above threshold): 110723\n",
    "          >> Unclassified reads: 15309056\n",
    "BRACKEN OUTPUT PRODUCED: //project/pi_sarah_gignouxwolfsohn_uml_edu/brooke/working/mcav_assembly_redo/taxonomy/kraken/bracken/T1_40_2022_species.bracken\n",
    "PROGRAM END TIME: 12-14-2023 02:40:28\n",
    "  Bracken complete.\n",
    " >> Checking for Valid Options...\n",
    " >> Running Bracken \n",
    "      >> python src/est_abundance.py -i T1_57_2022.kreport2 -o //project/pi_sarah_gignouxwolfsohn_uml_edu/brooke/working/mcav_assembly_redo/taxonomy/kraken/bracken/T1_57_2022_species.bracken -k //project/pi_sarah_gignouxwolfsohn_uml_edu/Reference_genomes/ref_databases/16S_SILVA132_k2db/database250mers.kmer_distrib -l S -t 0\n",
    "PROGRAM START TIME: 12-14-2023 02:40:28\n",
    "BRACKEN SUMMARY (Kraken report: T1_57_2022.kreport2)\n",
    "    >>> Threshold: 0 \n",
    "    >>> Number of species in sample: 0 \n",
    "          >> Number of species with reads > threshold: 0 \n",
    "          >> Number of species with reads < threshold: 0 \n",
    "    >>> Total reads in sample: 31964707\n",
    "          >> Total reads kept at species level (reads > threshold): 0\n",
    "          >> Total reads discarded (species reads < threshold): 0\n",
    "          >> Reads distributed: 0\n",
    "          >> Reads not distributed (eg. no species above threshold): 157041\n",
    "          >> Unclassified reads: 31807666\n",
    "BRACKEN OUTPUT PRODUCED: //project/pi_sarah_gignouxwolfsohn_uml_edu/brooke/working/mcav_assembly_redo/taxonomy/kraken/bracken/T1_57_2022_species.bracken\n",
    "PROGRAM END TIME: 12-14-2023 02:40:28\n",
    "  Bracken complete.\n",
    " >> Checking for Valid Options...\n",
    " >> Running Bracken \n",
    "      >> python src/est_abundance.py -i T1_70_2022.kreport2 -o //project/pi_sarah_gignouxwolfsohn_uml_edu/brooke/working/mcav_assembly_redo/taxonomy/kraken/bracken/T1_70_2022_species.bracken -k //project/pi_sarah_gignouxwolfsohn_uml_edu/Reference_genomes/ref_databases/16S_SILVA132_k2db/database250mers.kmer_distrib -l S -t 0\n",
    "PROGRAM START TIME: 12-14-2023 02:40:28\n",
    "BRACKEN SUMMARY (Kraken report: T1_70_2022.kreport2)\n",
    "    >>> Threshold: 0 \n",
    "    >>> Number of species in sample: 0 \n",
    "          >> Number of species with reads > threshold: 0 \n",
    "          >> Number of species with reads < threshold: 0 \n",
    "    >>> Total reads in sample: 15776674\n",
    "          >> Total reads kept at species level (reads > threshold): 0\n",
    "          >> Total reads discarded (species reads < threshold): 0\n",
    "          >> Reads distributed: 0\n",
    "          >> Reads not distributed (eg. no species above threshold): 121001\n",
    "          >> Unclassified reads: 15655673\n",
    "BRACKEN OUTPUT PRODUCED: //project/pi_sarah_gignouxwolfsohn_uml_edu/brooke/working/mcav_assembly_redo/taxonomy/kraken/bracken/T1_70_2022_species.bracken\n",
    "PROGRAM END TIME: 12-14-2023 02:40:29\n",
    "  Bracken complete.\n",
    " >> Checking for Valid Options...\n",
    " >> Running Bracken \n",
    "      >> python src/est_abundance.py -i T2_10_2022.kreport2 -o //project/pi_sarah_gignouxwolfsohn_uml_edu/brooke/working/mcav_assembly_redo/taxonomy/kraken/bracken/T2_10_2022_species.bracken -k //project/pi_sarah_gignouxwolfsohn_uml_edu/Reference_genomes/ref_databases/16S_SILVA132_k2db/database250mers.kmer_distrib -l S -t 0\n",
    "PROGRAM START TIME: 12-14-2023 02:40:29\n",
    "BRACKEN SUMMARY (Kraken report: T2_10_2022.kreport2)\n",
    "    >>> Threshold: 0 \n",
    "    >>> Number of species in sample: 0 \n",
    "          >> Number of species with reads > threshold: 0 \n",
    "          >> Number of species with reads < threshold: 0 \n",
    "    >>> Total reads in sample: 16876812\n",
    "          >> Total reads kept at species level (reads > threshold): 0\n",
    "          >> Total reads discarded (species reads < threshold): 0\n",
    "          >> Reads distributed: 0\n",
    "          >> Reads not distributed (eg. no species above threshold): 57221\n",
    "          >> Unclassified reads: 16819591\n",
    "BRACKEN OUTPUT PRODUCED: //project/pi_sarah_gignouxwolfsohn_uml_edu/brooke/working/mcav_assembly_redo/taxonomy/kraken/bracken/T2_10_2022_species.bracken\n",
    "PROGRAM END TIME: 12-14-2023 02:40:29\n",
    "  Bracken complete.\n",
    " >> Checking for Valid Options...\n",
    " >> Running Bracken \n",
    "      >> python src/est_abundance.py -i T2_16_2019.kreport2 -o //project/pi_sarah_gignouxwolfsohn_uml_edu/brooke/working/mcav_assembly_redo/taxonomy/kraken/bracken/T2_16_2019_species.bracken -k //project/pi_sarah_gignouxwolfsohn_uml_edu/Reference_genomes/ref_databases/16S_SILVA132_k2db/database250mers.kmer_distrib -l S -t 0\n",
    "PROGRAM START TIME: 12-14-2023 02:40:29\n",
    "BRACKEN SUMMARY (Kraken report: T2_16_2019.kreport2)\n",
    "    >>> Threshold: 0 \n",
    "    >>> Number of species in sample: 0 \n",
    "          >> Number of species with reads > threshold: 0 \n",
    "          >> Number of species with reads < threshold: 0 \n",
    "    >>> Total reads in sample: 34302369\n",
    "          >> Total reads kept at species level (reads > threshold): 0\n",
    "          >> Total reads discarded (species reads < threshold): 0\n",
    "          >> Reads distributed: 0\n",
    "          >> Reads not distributed (eg. no species above threshold): 116886\n",
    "          >> Unclassified reads: 34185483\n",
    "BRACKEN OUTPUT PRODUCED: //project/pi_sarah_gignouxwolfsohn_uml_edu/brooke/working/mcav_assembly_redo/taxonomy/kraken/bracken/T2_16_2019_species.bracken\n",
    "PROGRAM END TIME: 12-14-2023 02:40:30\n",
    "  Bracken complete.\n",
    " >> Checking for Valid Options...\n",
    " >> Running Bracken \n",
    "      >> python src/est_abundance.py -i T3_13_2022.kreport2 -o //project/pi_sarah_gignouxwolfsohn_uml_edu/brooke/working/mcav_assembly_redo/taxonomy/kraken/bracken/T3_13_2022_species.bracken -k //project/pi_sarah_gignouxwolfsohn_uml_edu/Reference_genomes/ref_databases/16S_SILVA132_k2db/database250mers.kmer_distrib -l S -t 0\n",
    "PROGRAM START TIME: 12-14-2023 02:40:30\n",
    "BRACKEN SUMMARY (Kraken report: T3_13_2022.kreport2)\n",
    "    >>> Threshold: 0 \n",
    "    >>> Number of species in sample: 0 \n",
    "          >> Number of species with reads > threshold: 0 \n",
    "          >> Number of species with reads < threshold: 0 \n",
    "    >>> Total reads in sample: 13449647\n",
    "          >> Total reads kept at species level (reads > threshold): 0\n",
    "          >> Total reads discarded (species reads < threshold): 0\n",
    "          >> Reads distributed: 0\n",
    "          >> Reads not distributed (eg. no species above threshold): 41610\n",
    "          >> Unclassified reads: 13408037\n",
    "BRACKEN OUTPUT PRODUCED: //project/pi_sarah_gignouxwolfsohn_uml_edu/brooke/working/mcav_assembly_redo/taxonomy/kraken/bracken/T3_13_2022_species.bracken\n",
    "PROGRAM END TIME: 12-14-2023 02:40:30\n",
    "  Bracken complete.\n",
    " >> Checking for Valid Options...\n",
    " >> Running Bracken \n",
    "      >> python src/est_abundance.py -i T3_14_2019.kreport2 -o //project/pi_sarah_gignouxwolfsohn_uml_edu/brooke/working/mcav_assembly_redo/taxonomy/kraken/bracken/T3_14_2019_species.bracken -k //project/pi_sarah_gignouxwolfsohn_uml_edu/Reference_genomes/ref_databases/16S_SILVA132_k2db/database250mers.kmer_distrib -l S -t 0\n",
    "PROGRAM START TIME: 12-14-2023 02:40:30\n",
    "BRACKEN SUMMARY (Kraken report: T3_14_2019.kreport2)\n",
    "    >>> Threshold: 0 \n",
    "    >>> Number of species in sample: 0 \n",
    "          >> Number of species with reads > threshold: 0 \n",
    "          >> Number of species with reads < threshold: 0 \n",
    "    >>> Total reads in sample: 21400877\n",
    "          >> Total reads kept at species level (reads > threshold): 0\n",
    "          >> Total reads discarded (species reads < threshold): 0\n",
    "          >> Reads distributed: 0\n",
    "          >> Reads not distributed (eg. no species above threshold): 91615\n",
    "          >> Unclassified reads: 21309262\n",
    "BRACKEN OUTPUT PRODUCED: //project/pi_sarah_gignouxwolfsohn_uml_edu/brooke/working/mcav_assembly_redo/taxonomy/kraken/bracken/T3_14_2019_species.bracken\n",
    "PROGRAM END TIME: 12-14-2023 02:40:31\n",
    "  Bracken complete.\n",
    " >> Checking for Valid Options...\n",
    " >> Running Bracken \n",
    "      >> python src/est_abundance.py -i T3_15_2019.kreport2 -o //project/pi_sarah_gignouxwolfsohn_uml_edu/brooke/working/mcav_assembly_redo/taxonomy/kraken/bracken/T3_15_2019_species.bracken -k //project/pi_sarah_gignouxwolfsohn_uml_edu/Reference_genomes/ref_databases/16S_SILVA132_k2db/database250mers.kmer_distrib -l S -t 0\n",
    "PROGRAM START TIME: 12-14-2023 02:40:31\n",
    "BRACKEN SUMMARY (Kraken report: T3_15_2019.kreport2)\n",
    "    >>> Threshold: 0 \n",
    "    >>> Number of species in sample: 0 \n",
    "          >> Number of species with reads > threshold: 0 \n",
    "          >> Number of species with reads < threshold: 0 \n",
    "    >>> Total reads in sample: 19854618\n",
    "          >> Total reads kept at species level (reads > threshold): 0\n",
    "          >> Total reads discarded (species reads < threshold): 0\n",
    "          >> Reads distributed: 0\n",
    "          >> Reads not distributed (eg. no species above threshold): 51464\n",
    "          >> Unclassified reads: 19803154\n",
    "BRACKEN OUTPUT PRODUCED: //project/pi_sarah_gignouxwolfsohn_uml_edu/brooke/working/mcav_assembly_redo/taxonomy/kraken/bracken/T3_15_2019_species.bracken\n",
    "PROGRAM END TIME: 12-14-2023 02:40:31\n",
    "  Bracken complete.\n",
    " >> Checking for Valid Options...\n",
    " >> Running Bracken \n",
    "      >> python src/est_abundance.py -i T3_19_2022.kreport2 -o //project/pi_sarah_gignouxwolfsohn_uml_edu/brooke/working/mcav_assembly_redo/taxonomy/kraken/bracken/T3_19_2022_species.bracken -k //project/pi_sarah_gignouxwolfsohn_uml_edu/Reference_genomes/ref_databases/16S_SILVA132_k2db/database250mers.kmer_distrib -l S -t 0\n",
    "PROGRAM START TIME: 12-14-2023 02:40:31\n",
    "BRACKEN SUMMARY (Kraken report: T3_19_2022.kreport2)\n",
    "    >>> Threshold: 0 \n",
    "    >>> Number of species in sample: 0 \n",
    "          >> Number of species with reads > threshold: 0 \n",
    "          >> Number of species with reads < threshold: 0 \n",
    "    >>> Total reads in sample: 9044371\n",
    "          >> Total reads kept at species level (reads > threshold): 0\n",
    "          >> Total reads discarded (species reads < threshold): 0\n",
    "          >> Reads distributed: 0\n",
    "          >> Reads not distributed (eg. no species above threshold): 93872\n",
    "          >> Unclassified reads: 8950499\n",
    "BRACKEN OUTPUT PRODUCED: //project/pi_sarah_gignouxwolfsohn_uml_edu/brooke/working/mcav_assembly_redo/taxonomy/kraken/bracken/T3_19_2022_species.bracken\n",
    "PROGRAM END TIME: 12-14-2023 02:40:31\n",
    "  Bracken complete.\n",
    " >> Checking for Valid Options...\n",
    " >> Running Bracken \n",
    "      >> python src/est_abundance.py -i T3_1_2019.kreport2 -o //project/pi_sarah_gignouxwolfsohn_uml_edu/brooke/working/mcav_assembly_redo/taxonomy/kraken/bracken/T3_1_2019_species.bracken -k //project/pi_sarah_gignouxwolfsohn_uml_edu/Reference_genomes/ref_databases/16S_SILVA132_k2db/database250mers.kmer_distrib -l S -t 0\n",
    "PROGRAM START TIME: 12-14-2023 02:40:31\n",
    "BRACKEN SUMMARY (Kraken report: T3_1_2019.kreport2)\n",
    "    >>> Threshold: 0 \n",
    "    >>> Number of species in sample: 0 \n",
    "          >> Number of species with reads > threshold: 0 \n",
    "          >> Number of species with reads < threshold: 0 \n",
    "    >>> Total reads in sample: 29977945\n",
    "          >> Total reads kept at species level (reads > threshold): 0\n",
    "          >> Total reads discarded (species reads < threshold): 0\n",
    "          >> Reads distributed: 0\n",
    "          >> Reads not distributed (eg. no species above threshold): 99522\n",
    "          >> Unclassified reads: 29878423\n",
    "BRACKEN OUTPUT PRODUCED: //project/pi_sarah_gignouxwolfsohn_uml_edu/brooke/working/mcav_assembly_redo/taxonomy/kraken/bracken/T3_1_2019_species.bracken\n",
    "PROGRAM END TIME: 12-14-2023 02:40:32\n",
    "  Bracken complete.\n",
    " >> Checking for Valid Options...\n",
    " >> Running Bracken \n",
    "      >> python src/est_abundance.py -i T3_40_2022.kreport2 -o //project/pi_sarah_gignouxwolfsohn_uml_edu/brooke/working/mcav_assembly_redo/taxonomy/kraken/bracken/T3_40_2022_species.bracken -k //project/pi_sarah_gignouxwolfsohn_uml_edu/Reference_genomes/ref_databases/16S_SILVA132_k2db/database250mers.kmer_distrib -l S -t 0\n",
    "PROGRAM START TIME: 12-14-2023 02:40:32\n",
    "BRACKEN SUMMARY (Kraken report: T3_40_2022.kreport2)\n",
    "    >>> Threshold: 0 \n",
    "    >>> Number of species in sample: 0 \n",
    "          >> Number of species with reads > threshold: 0 \n",
    "          >> Number of species with reads < threshold: 0 \n",
    "    >>> Total reads in sample: 26755227\n",
    "          >> Total reads kept at species level (reads > threshold): 0\n",
    "          >> Total reads discarded (species reads < threshold): 0\n",
    "          >> Reads distributed: 0\n",
    "          >> Reads not distributed (eg. no species above threshold): 77871\n",
    "          >> Unclassified reads: 26677356\n",
    "BRACKEN OUTPUT PRODUCED: //project/pi_sarah_gignouxwolfsohn_uml_edu/brooke/working/mcav_assembly_redo/taxonomy/kraken/bracken/T3_40_2022_species.bracken\n",
    "PROGRAM END TIME: 12-14-2023 02:40:32\n",
    "  Bracken complete.\n",
    " >> Checking for Valid Options...\n",
    " >> Running Bracken \n",
    "      >> python src/est_abundance.py -i T3_48_2022.kreport2 -o //project/pi_sarah_gignouxwolfsohn_uml_edu/brooke/working/mcav_assembly_redo/taxonomy/kraken/bracken/T3_48_2022_species.bracken -k //project/pi_sarah_gignouxwolfsohn_uml_edu/Reference_genomes/ref_databases/16S_SILVA132_k2db/database250mers.kmer_distrib -l S -t 0\n",
    "PROGRAM START TIME: 12-14-2023 02:40:32\n",
    "BRACKEN SUMMARY (Kraken report: T3_48_2022.kreport2)\n",
    "    >>> Threshold: 0 \n",
    "    >>> Number of species in sample: 0 \n",
    "          >> Number of species with reads > threshold: 0 \n",
    "          >> Number of species with reads < threshold: 0 \n",
    "    >>> Total reads in sample: 13255150\n",
    "          >> Total reads kept at species level (reads > threshold): 0\n",
    "          >> Total reads discarded (species reads < threshold): 0\n",
    "          >> Reads distributed: 0\n",
    "          >> Reads not distributed (eg. no species above threshold): 133945\n",
    "          >> Unclassified reads: 13121205\n",
    "BRACKEN OUTPUT PRODUCED: //project/pi_sarah_gignouxwolfsohn_uml_edu/brooke/working/mcav_assembly_redo/taxonomy/kraken/bracken/T3_48_2022_species.bracken\n",
    "PROGRAM END TIME: 12-14-2023 02:40:33\n",
    "  Bracken complete.\n",
    " >> Checking for Valid Options...\n",
    " >> Running Bracken \n",
    "      >> python src/est_abundance.py -i T3_49_2022.kreport2 -o //project/pi_sarah_gignouxwolfsohn_uml_edu/brooke/working/mcav_assembly_redo/taxonomy/kraken/bracken/T3_49_2022_species.bracken -k //project/pi_sarah_gignouxwolfsohn_uml_edu/Reference_genomes/ref_databases/16S_SILVA132_k2db/database250mers.kmer_distrib -l S -t 0\n",
    "PROGRAM START TIME: 12-14-2023 02:40:33\n",
    "BRACKEN SUMMARY (Kraken report: T3_49_2022.kreport2)\n",
    "    >>> Threshold: 0 \n",
    "    >>> Number of species in sample: 0 \n",
    "          >> Number of species with reads > threshold: 0 \n",
    "          >> Number of species with reads < threshold: 0 \n",
    "    >>> Total reads in sample: 15749618\n",
    "          >> Total reads kept at species level (reads > threshold): 0\n",
    "          >> Total reads discarded (species reads < threshold): 0\n",
    "          >> Reads distributed: 0\n",
    "          >> Reads not distributed (eg. no species above threshold): 149084\n",
    "          >> Unclassified reads: 15600534\n",
    "BRACKEN OUTPUT PRODUCED: //project/pi_sarah_gignouxwolfsohn_uml_edu/brooke/working/mcav_assembly_redo/taxonomy/kraken/bracken/T3_49_2022_species.bracken\n",
    "PROGRAM END TIME: 12-14-2023 02:40:33\n",
    "  Bracken complete.\n",
    " >> Checking for Valid Options...\n",
    " >> Running Bracken \n",
    "      >> python src/est_abundance.py -i T3_51_2022.kreport2 -o //project/pi_sarah_gignouxwolfsohn_uml_edu/brooke/working/mcav_assembly_redo/taxonomy/kraken/bracken/T3_51_2022_species.bracken -k //project/pi_sarah_gignouxwolfsohn_uml_edu/Reference_genomes/ref_databases/16S_SILVA132_k2db/database250mers.kmer_distrib -l S -t 0\n",
    "PROGRAM START TIME: 12-14-2023 02:40:33\n",
    "BRACKEN SUMMARY (Kraken report: T3_51_2022.kreport2)\n",
    "    >>> Threshold: 0 \n",
    "    >>> Number of species in sample: 0 \n",
    "          >> Number of species with reads > threshold: 0 \n",
    "          >> Number of species with reads < threshold: 0 \n",
    "    >>> Total reads in sample: 13877944\n",
    "          >> Total reads kept at species level (reads > threshold): 0\n",
    "          >> Total reads discarded (species reads < threshold): 0\n",
    "          >> Reads distributed: 0\n",
    "          >> Reads not distributed (eg. no species above threshold): 50639\n",
    "          >> Unclassified reads: 13827305\n",
    "BRACKEN OUTPUT PRODUCED: //project/pi_sarah_gignouxwolfsohn_uml_edu/brooke/working/mcav_assembly_redo/taxonomy/kraken/bracken/T3_51_2022_species.bracken\n",
    "PROGRAM END TIME: 12-14-2023 02:40:34\n",
    "  Bracken complete.\n",
    " >> Checking for Valid Options...\n",
    " >> Running Bracken \n",
    "      >> python src/est_abundance.py -i T3_8_2019.kreport2 -o //project/pi_sarah_gignouxwolfsohn_uml_edu/brooke/working/mcav_assembly_redo/taxonomy/kraken/bracken/T3_8_2019_species.bracken -k //project/pi_sarah_gignouxwolfsohn_uml_edu/Reference_genomes/ref_databases/16S_SILVA132_k2db/database250mers.kmer_distrib -l S -t 0\n",
    "PROGRAM START TIME: 12-14-2023 02:40:34\n",
    "BRACKEN SUMMARY (Kraken report: T3_8_2019.kreport2)\n",
    "    >>> Threshold: 0 \n",
    "    >>> Number of species in sample: 0 \n",
    "          >> Number of species with reads > threshold: 0 \n",
    "          >> Number of species with reads < threshold: 0 \n",
    "    >>> Total reads in sample: 16453406\n",
    "          >> Total reads kept at species level (reads > threshold): 0\n",
    "          >> Total reads discarded (species reads < threshold): 0\n",
    "          >> Reads distributed: 0\n",
    "          >> Reads not distributed (eg. no species above threshold): 94662\n",
    "          >> Unclassified reads: 16358744\n",
    "BRACKEN OUTPUT PRODUCED: //project/pi_sarah_gignouxwolfsohn_uml_edu/brooke/working/mcav_assembly_redo/taxonomy/kraken/bracken/T3_8_2019_species.bracken\n",
    "PROGRAM END TIME: 12-14-2023 02:40:34\n",
    "  Bracken complete.\n",
    " >> Checking for Valid Options...\n",
    " >> Running Bracken \n",
    "      >> python src/est_abundance.py -i T3_9_2019.kreport2 -o //project/pi_sarah_gignouxwolfsohn_uml_edu/brooke/working/mcav_assembly_redo/taxonomy/kraken/bracken/T3_9_2019_species.bracken -k //project/pi_sarah_gignouxwolfsohn_uml_edu/Reference_genomes/ref_databases/16S_SILVA132_k2db/database250mers.kmer_distrib -l S -t 0\n",
    "PROGRAM START TIME: 12-14-2023 02:40:34\n",
    "BRACKEN SUMMARY (Kraken report: T3_9_2019.kreport2)\n",
    "    >>> Threshold: 0 \n",
    "    >>> Number of species in sample: 0 \n",
    "          >> Number of species with reads > threshold: 0 \n",
    "          >> Number of species with reads < threshold: 0 \n",
    "    >>> Total reads in sample: 14032207\n",
    "          >> Total reads kept at species level (reads > threshold): 0\n",
    "          >> Total reads discarded (species reads < threshold): 0\n",
    "          >> Reads distributed: 0\n",
    "          >> Reads not distributed (eg. no species above threshold): 69979\n",
    "          >> Unclassified reads: 13962228\n",
    "BRACKEN OUTPUT PRODUCED: //project/pi_sarah_gignouxwolfsohn_uml_edu/brooke/working/mcav_assembly_redo/taxonomy/kraken/bracken/T3_9_2019_species.bracken\n",
    "PROGRAM END TIME: 12-14-2023 02:40:34\n",
    "  Bracken complete."
   ]
  },
  {
   "cell_type": "markdown",
   "id": "24fddd76-2c63-40f7-8c47-ec8ae46b3b9a",
   "metadata": {
    "tags": []
   },
   "source": [
    "##### KrakenTools \n",
    "- downstream analysis using kraken results\n",
    "- https://ccb.jhu.edu/software/krakentools/index.shtml\n",
    "- download zip file "
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "bf669216-e56f-42fa-9de0-ec1bf184b74a",
   "metadata": {},
   "outputs": [],
   "source": [
    "SCRIPTPATH=/project/pi_sarah_gignouxwolfsohn_uml_edu/brooke/bash_scripts/KrakenTools-master/DiversityTools\n",
    "cd $SCRIPTPATH\n",
    "# make scripts executable \n",
    "chmod +x alpha\\_diversity.py\n",
    "./alpha\\_diversity.py -h\n",
    "# input file must be bracken\n",
    "\n",
    "chmod +x beta\\_diversity.py\n",
    "./beta\\_diversity.py -h\n",
    "# Bray-Curtis dissimilarity"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "f9ccac70-8898-4bec-9092-9fe9548e2ad1",
   "metadata": {},
   "outputs": [],
   "source": [
    "# Alpha Diversity\n",
    "# Shannon diversity - genus level \n",
    "\n",
    "SCRIPTPATH=/project/pi_sarah_gignouxwolfsohn_uml_edu/brooke/bash_scripts/KrakenTools-master/DiversityTools\n",
    "FILEPATH=//project/pi_sarah_gignouxwolfsohn_uml_edu/brooke/mcav_taxonomy/kraken\n",
    "#ls $FILEPATH | find *.kreport2 > kraken_list.txt\n",
    "cat kraken_list.txt| sed 's/.kreport2/''/' > SAMPLEID\n",
    "\n",
    "#this is bash script to be used in terminal..couldn't get bracken scripts to work in nb\n",
    "while IFS= read -r line; do\n",
    "    python $SCRIPTPATH/alpha_diversity.py -f ${line}.bracken \n",
    "done < SAMPLEID\n",
    "\n",
    "# don't know if it does multiple at a time or just one at a time...and what is output?\n",
    "#-f, --filename MYFILE.BRACKEN..........Bracken output file\n",
    "#-a, --alpha TYPE.......................Single letter alpha diversity type (Sh, BP, Si, ISi, F)"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "4a234bab-162f-4454-aaec-599f30ac4f77",
   "metadata": {},
   "source": [
    "Shannon's diversity: 2.8405589418696024\n",
    "Shannon's diversity: 3.747956479976194\n",
    "Shannon's diversity: 3.240773802595393\n",
    "Shannon's diversity: 3.357943870893113\n",
    "Shannon's diversity: 3.127505271223385\n",
    "Shannon's diversity: 2.948836172095586\n",
    "Shannon's diversity: 3.155355846512862\n",
    "Shannon's diversity: 2.9534230075080843\n",
    "Shannon's diversity: 3.837848266676804\n",
    "Shannon's diversity: 3.017049434243576\n",
    "Shannon's diversity: 3.3014806325360837\n",
    "Shannon's diversity: 3.1828393709134533\n",
    "Shannon's diversity: 2.673104478260622\n",
    "Shannon's diversity: 2.8628687828400676\n",
    "Shannon's diversity: 3.1197450036741374\n",
    "Shannon's diversity: 3.4793331275395554\n",
    "Shannon's diversity: 2.842877602269089\n",
    "Shannon's diversity: 2.9019378226014787\n",
    "Shannon's diversity: 3.1864428196047045\n",
    "Shannon's diversity: 3.1428692882448113\n",
    "Shannon's diversity: 3.0649759367640863"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "c19febbb-b03e-4482-a375-a9ce197b5ddd",
   "metadata": {},
   "outputs": [],
   "source": [
    "cat SAMPLEID"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "51444fbc-1c4e-4738-b69d-ca14e192e8b2",
   "metadata": {},
   "source": [
    "T1_12_2022\n",
    "T1_13_2022\n",
    "T1_16_2019\n",
    "T1_20_2019\n",
    "T1_24_2019\n",
    "T1_40_2022\n",
    "T1_57_2022\n",
    "T1_70_2022\n",
    "T2_10_2022\n",
    "T2_16_2019\n",
    "T3_13_2022\n",
    "T3_14_2019\n",
    "T3_15_2019\n",
    "T3_19_2022\n",
    "T3_1_2019\n",
    "T3_40_2022\n",
    "T3_48_2022\n",
    "T3_49_2022\n",
    "T3_51_2022\n",
    "T3_8_2019\n",
    "T3_9_2019"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "a5f33441-bc80-49ec-8590-e2b93df876af",
   "metadata": {},
   "outputs": [],
   "source": [
    "# Beta Diversity\n",
    "\n",
    "# collect list of input files\n",
    "FILEPATH=//project/pi_sarah_gignouxwolfsohn_uml_edu/brooke/mcav_taxonomy/kraken\n",
    "mv *_species.bracken species\n",
    "ls $FILEPATH | find *.bracken > bracken_list.txt\n",
    "FILES=xargs -a bracken_list.txt -d '\\n'"
   ]
  },
  {
   "cell_type": "raw",
   "id": "d00afef5-6612-46c6-b524-ec81c87ad20b",
   "metadata": {},
   "source": [
    "T1_12_2022.bracken T1_13_2022.bracken T1_16_2019.bracken T1_20_2019.bracken T1_24_2019.bracken T1_40_2022.bracken T1_57_2022.bracken T1_70_2022.bracken T2_10_2022.bracken T2_16_2019.bracken T3_13_2022.bracken T3_14_2019.bracken T3_15_2019.bracken T3_19_2022.bracken T3_1_2019.bracken T3_40_2022.bracken T3_48_2022.bracken T3_49_2022.bracken T3_51_2022.bracken T3_8_2019.bracken T3_9_2019.bracken"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "352dbc09-b330-4422-8d8c-a98a6cbadc1a",
   "metadata": {},
   "outputs": [],
   "source": [
    "python $SCRIPTPATH/beta_diversity.py --type bracken --input-files T1_12_2022.bracken T1_13_2022.bracken T1_16_2019.bracken T1_20_2019.bracken T1_24_2019.bracken T1_40_2022.bracken T1_57_2022.bracken T1_70_2022.bracken T2_10_2022.bracken T2_16_2019.bracken T3_13_2022.bracken T3_14_2019.bracken T3_15_2019.bracken T3_19_2022.bracken T3_1_2019.bracken T3_40_2022.bracken T3_48_2022.bracken T3_49_2022.bracken T3_51_2022.bracken T3_8_2019.bracken T3_9_2019.bracken > b_diversity.txt\n"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "6112b155-1293-4ee0-98c4-3f3664c3e47a",
   "metadata": {
    "jp-MarkdownHeadingCollapsed": true,
    "tags": []
   },
   "source": [
    "##### Pavian"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 3,
   "id": "67490870-4e30-485a-8ace-212819f6e70a",
   "metadata": {},
   "outputs": [
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "Loading required package: remotes\n",
      "\n",
      "Skipping install of 'pavian' from a github remote, the SHA1 (3cf9d6f8) has not changed since last install.\n",
      "  Use `force = TRUE` to force installation\n",
      "\n"
     ]
    }
   ],
   "source": [
    "if (!require(remotes)) { install.packages(\"remotes\") }\n",
    "remotes::install_github(\"fbreitwieser/pavian\")"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "a160f15f-c856-4e5f-be28-4c7b6bd5d7bb",
   "metadata": {
    "jp-MarkdownHeadingCollapsed": true,
    "tags": []
   },
   "source": [
    "#### Skbio \n",
    "- to create pcoa plot from b diversity matrix"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 1,
   "id": "66468571-f8ec-4720-9bed-52a026115def",
   "metadata": {},
   "outputs": [],
   "source": [
    "from skbio.diversity import beta_diversity\n",
    "from skbio.stats.distance import mantel"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "697c21c1-70f8-44fc-858f-c536bcc1fa69",
   "metadata": {},
   "outputs": [],
   "source": [
    "#view b diversity file \n",
    "file_path = '//project/pi_sarah_gignouxwolfsohn_uml_edu/brooke/mcav/taxonomy/kraken/b_diversity.txt'\n",
    "\n",
    "with open(file_path, 'r') as file:\n",
    "    content = file.read()\n",
    "    print(content)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 4,
   "id": "fd981208-1a18-4b97-9b67-0add727a5d46",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "'/project/pi_sarah_gignouxwolfsohn_uml_edu/brooke/CBC_metagenomics/mcav1'"
      ]
     },
     "execution_count": 4,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "# Plan\n",
    "# load in first part as a seperate file so i can keep track of which sample goes with which\n",
    "import os\n",
    "os.getcwd()\n",
    "# load in second part as compatible distance matrix "
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 50,
   "id": "32d62dc8-a72d-44a3-8df9-612f085fdf90",
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "[[   nan  0.     1.     2.     3.     4.     5.     6.     7.     8.\n",
      "   9.    10.    11.    12.    13.    14.    15.    16.    17.    18.\n",
      "  19.    20.   ]\n",
      " [ 0.     0.     0.453  0.225  0.426  0.18   0.414  0.5    0.466  0.33\n",
      "   0.471  0.262  0.31   0.163  0.413  0.345  0.39   0.501  0.5    0.243\n",
      "   0.459  0.251]\n",
      " [ 1.     0.     0.     0.437  0.628  0.472  0.669  0.625  0.692  0.394\n",
      "   0.63   0.523  0.561  0.506  0.688  0.513  0.634  0.73   0.715  0.563\n",
      "   0.677  0.524]\n",
      " [ 2.     0.     0.     0.     0.524  0.249  0.372  0.364  0.407  0.343\n",
      "   0.369  0.378  0.227  0.231  0.44   0.201  0.384  0.454  0.439  0.35\n",
      "   0.425  0.291]\n",
      " [ 3.     0.     0.     0.     0.     0.439  0.588  0.667  0.636  0.399\n",
      "   0.588  0.453  0.534  0.428  0.549  0.585  0.489  0.633  0.657  0.373\n",
      "   0.449  0.48 ]\n",
      " [ 4.     0.     0.     0.     0.     0.     0.35   0.451  0.379  0.328\n",
      "   0.436  0.279  0.233  0.215  0.338  0.273  0.33   0.444  0.435  0.19\n",
      "   0.415  0.157]\n",
      " [ 5.     0.     0.     0.     0.     0.     0.     0.238  0.126  0.54\n",
      "   0.432  0.454  0.23   0.376  0.118  0.327  0.363  0.137  0.171  0.381\n",
      "   0.351  0.287]\n",
      " [ 6.     0.     0.     0.     0.     0.     0.     0.     0.217  0.549\n",
      "   0.398  0.583  0.293  0.494  0.312  0.278  0.494  0.265  0.218  0.528\n",
      "   0.427  0.387]\n",
      " [ 7.     0.     0.     0.     0.     0.     0.     0.     0.     0.57\n",
      "   0.407  0.504  0.238  0.433  0.201  0.272  0.423  0.2    0.15   0.45\n",
      "   0.355  0.32 ]\n",
      " [ 8.     0.     0.     0.     0.     0.     0.     0.     0.     0.\n",
      "   0.535  0.381  0.426  0.338  0.549  0.429  0.414  0.608  0.599  0.394\n",
      "   0.528  0.393]\n",
      " [ 9.     0.     0.     0.     0.     0.     0.     0.     0.     0.\n",
      "   0.     0.583  0.34   0.451  0.48   0.294  0.473  0.482  0.408  0.522\n",
      "   0.315  0.45 ]\n",
      " [10.     0.     0.     0.     0.     0.     0.     0.     0.     0.\n",
      "   0.     0.     0.389  0.319  0.452  0.439  0.45   0.53   0.536  0.205\n",
      "   0.519  0.301]\n",
      " [11.     0.     0.     0.     0.     0.     0.     0.     0.     0.\n",
      "   0.     0.     0.     0.289  0.289  0.175  0.31   0.324  0.281  0.302\n",
      "   0.309  0.206]\n",
      " [12.     0.     0.     0.     0.     0.     0.     0.     0.     0.\n",
      "   0.     0.     0.     0.     0.392  0.344  0.309  0.472  0.472  0.288\n",
      "   0.411  0.279]\n",
      " [13.     0.     0.     0.     0.     0.     0.     0.     0.     0.\n",
      "   0.     0.     0.     0.     0.     0.401  0.357  0.194  0.241  0.343\n",
      "   0.39   0.303]\n",
      " [14.     0.     0.     0.     0.     0.     0.     0.     0.     0.\n",
      "   0.     0.     0.     0.     0.     0.     0.372  0.394  0.337  0.39\n",
      "   0.373  0.261]\n",
      " [15.     0.     0.     0.     0.     0.     0.     0.     0.     0.\n",
      "   0.     0.     0.     0.     0.     0.     0.     0.426  0.46   0.34\n",
      "   0.408  0.316]\n",
      " [16.     0.     0.     0.     0.     0.     0.     0.     0.     0.\n",
      "   0.     0.     0.     0.     0.     0.     0.     0.     0.122  0.462\n",
      "   0.399  0.372]\n",
      " [17.     0.     0.     0.     0.     0.     0.     0.     0.     0.\n",
      "   0.     0.     0.     0.     0.     0.     0.     0.     0.     0.485\n",
      "   0.38   0.365]\n",
      " [18.     0.     0.     0.     0.     0.     0.     0.     0.     0.\n",
      "   0.     0.     0.     0.     0.     0.     0.     0.     0.     0.\n",
      "   0.434  0.224]\n",
      " [19.     0.     0.     0.     0.     0.     0.     0.     0.     0.\n",
      "   0.     0.     0.     0.     0.     0.     0.     0.     0.     0.\n",
      "   0.     0.414]\n",
      " [20.     0.     0.     0.     0.     0.     0.     0.     0.     0.\n",
      "   0.     0.     0.     0.     0.     0.     0.     0.     0.     0.\n",
      "   0.     0.   ]]\n"
     ]
    },
    {
     "data": {
      "text/plain": [
       "(22, 22)"
      ]
     },
     "execution_count": 50,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "import pandas as pd\n",
    "import numpy as np\n",
    "\n",
    "# Read the first 20 lines separately\n",
    "with open(file_path, 'r') as file:\n",
    "    first_20_lines = [file.readline().strip() for _ in range(21)]\n",
    "\n",
    "# Define a function to convert \"x.xxx\" to float, replacing with NaN\n",
    "def convert_xxxx_to_float(s):\n",
    "    if s == 'x.xxx':\n",
    "        return 0.0\n",
    "    try:\n",
    "        return float(s)\n",
    "    except (ValueError, TypeError):\n",
    "        return 0.0\n",
    "\n",
    "# Specify the data type and the converter for the columns\n",
    "dtype = float\n",
    "converters = {i: convert_xxxx_to_float for i in range(1, 22)}  # Assuming you want to apply the converter to columns 1 to 21\n",
    "\n",
    "# Skip the first 21 + 1 lines (header lines + \"x\" line)\n",
    "skip_rows = 21\n",
    "\n",
    "# Load the data using numpy\n",
    "bd = np.genfromtxt(file_path, delimiter='\\t', skip_header=skip_rows, dtype=dtype, converters=converters)\n",
    "\n",
    "# Print the modified data array\n",
    "print(bd)\n",
    "bd.shape"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 110,
   "id": "0abc08d7-c360-4c89-9415-3e93ec412a28",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "(22, 22)"
      ]
     },
     "execution_count": 110,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "#transpose bd and add it to itself to make symmetrical (use transposed values to replace 0s)\n",
    "bd3 = bd + bd.T\n",
    "bd3.shape"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 118,
   "id": "72eed2e2-4231-4af3-a3d0-d1f8f17b6c97",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>1</th>\n",
       "      <th>2</th>\n",
       "      <th>3</th>\n",
       "      <th>4</th>\n",
       "      <th>5</th>\n",
       "      <th>6</th>\n",
       "      <th>7</th>\n",
       "      <th>8</th>\n",
       "      <th>9</th>\n",
       "      <th>10</th>\n",
       "      <th>...</th>\n",
       "      <th>12</th>\n",
       "      <th>13</th>\n",
       "      <th>14</th>\n",
       "      <th>15</th>\n",
       "      <th>16</th>\n",
       "      <th>17</th>\n",
       "      <th>18</th>\n",
       "      <th>19</th>\n",
       "      <th>20</th>\n",
       "      <th>21</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>0.000</td>\n",
       "      <td>0.453</td>\n",
       "      <td>0.225</td>\n",
       "      <td>0.426</td>\n",
       "      <td>0.180</td>\n",
       "      <td>0.414</td>\n",
       "      <td>0.500</td>\n",
       "      <td>0.466</td>\n",
       "      <td>0.330</td>\n",
       "      <td>0.471</td>\n",
       "      <td>...</td>\n",
       "      <td>0.310</td>\n",
       "      <td>0.163</td>\n",
       "      <td>0.413</td>\n",
       "      <td>0.345</td>\n",
       "      <td>0.390</td>\n",
       "      <td>0.501</td>\n",
       "      <td>0.500</td>\n",
       "      <td>0.243</td>\n",
       "      <td>0.459</td>\n",
       "      <td>0.251</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>0.453</td>\n",
       "      <td>0.000</td>\n",
       "      <td>0.437</td>\n",
       "      <td>0.628</td>\n",
       "      <td>0.472</td>\n",
       "      <td>0.669</td>\n",
       "      <td>0.625</td>\n",
       "      <td>0.692</td>\n",
       "      <td>0.394</td>\n",
       "      <td>0.630</td>\n",
       "      <td>...</td>\n",
       "      <td>0.561</td>\n",
       "      <td>0.506</td>\n",
       "      <td>0.688</td>\n",
       "      <td>0.513</td>\n",
       "      <td>0.634</td>\n",
       "      <td>0.730</td>\n",
       "      <td>0.715</td>\n",
       "      <td>0.563</td>\n",
       "      <td>0.677</td>\n",
       "      <td>0.524</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>0.225</td>\n",
       "      <td>0.437</td>\n",
       "      <td>0.000</td>\n",
       "      <td>0.524</td>\n",
       "      <td>0.249</td>\n",
       "      <td>0.372</td>\n",
       "      <td>0.364</td>\n",
       "      <td>0.407</td>\n",
       "      <td>0.343</td>\n",
       "      <td>0.369</td>\n",
       "      <td>...</td>\n",
       "      <td>0.227</td>\n",
       "      <td>0.231</td>\n",
       "      <td>0.440</td>\n",
       "      <td>0.201</td>\n",
       "      <td>0.384</td>\n",
       "      <td>0.454</td>\n",
       "      <td>0.439</td>\n",
       "      <td>0.350</td>\n",
       "      <td>0.425</td>\n",
       "      <td>0.291</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>0.426</td>\n",
       "      <td>0.628</td>\n",
       "      <td>0.524</td>\n",
       "      <td>0.000</td>\n",
       "      <td>0.439</td>\n",
       "      <td>0.588</td>\n",
       "      <td>0.667</td>\n",
       "      <td>0.636</td>\n",
       "      <td>0.399</td>\n",
       "      <td>0.588</td>\n",
       "      <td>...</td>\n",
       "      <td>0.534</td>\n",
       "      <td>0.428</td>\n",
       "      <td>0.549</td>\n",
       "      <td>0.585</td>\n",
       "      <td>0.489</td>\n",
       "      <td>0.633</td>\n",
       "      <td>0.657</td>\n",
       "      <td>0.373</td>\n",
       "      <td>0.449</td>\n",
       "      <td>0.480</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>5</th>\n",
       "      <td>0.180</td>\n",
       "      <td>0.472</td>\n",
       "      <td>0.249</td>\n",
       "      <td>0.439</td>\n",
       "      <td>0.000</td>\n",
       "      <td>0.350</td>\n",
       "      <td>0.451</td>\n",
       "      <td>0.379</td>\n",
       "      <td>0.328</td>\n",
       "      <td>0.436</td>\n",
       "      <td>...</td>\n",
       "      <td>0.233</td>\n",
       "      <td>0.215</td>\n",
       "      <td>0.338</td>\n",
       "      <td>0.273</td>\n",
       "      <td>0.330</td>\n",
       "      <td>0.444</td>\n",
       "      <td>0.435</td>\n",
       "      <td>0.190</td>\n",
       "      <td>0.415</td>\n",
       "      <td>0.157</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>6</th>\n",
       "      <td>0.414</td>\n",
       "      <td>0.669</td>\n",
       "      <td>0.372</td>\n",
       "      <td>0.588</td>\n",
       "      <td>0.350</td>\n",
       "      <td>0.000</td>\n",
       "      <td>0.238</td>\n",
       "      <td>0.126</td>\n",
       "      <td>0.540</td>\n",
       "      <td>0.432</td>\n",
       "      <td>...</td>\n",
       "      <td>0.230</td>\n",
       "      <td>0.376</td>\n",
       "      <td>0.118</td>\n",
       "      <td>0.327</td>\n",
       "      <td>0.363</td>\n",
       "      <td>0.137</td>\n",
       "      <td>0.171</td>\n",
       "      <td>0.381</td>\n",
       "      <td>0.351</td>\n",
       "      <td>0.287</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>7</th>\n",
       "      <td>0.500</td>\n",
       "      <td>0.625</td>\n",
       "      <td>0.364</td>\n",
       "      <td>0.667</td>\n",
       "      <td>0.451</td>\n",
       "      <td>0.238</td>\n",
       "      <td>0.000</td>\n",
       "      <td>0.217</td>\n",
       "      <td>0.549</td>\n",
       "      <td>0.398</td>\n",
       "      <td>...</td>\n",
       "      <td>0.293</td>\n",
       "      <td>0.494</td>\n",
       "      <td>0.312</td>\n",
       "      <td>0.278</td>\n",
       "      <td>0.494</td>\n",
       "      <td>0.265</td>\n",
       "      <td>0.218</td>\n",
       "      <td>0.528</td>\n",
       "      <td>0.427</td>\n",
       "      <td>0.387</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>8</th>\n",
       "      <td>0.466</td>\n",
       "      <td>0.692</td>\n",
       "      <td>0.407</td>\n",
       "      <td>0.636</td>\n",
       "      <td>0.379</td>\n",
       "      <td>0.126</td>\n",
       "      <td>0.217</td>\n",
       "      <td>0.000</td>\n",
       "      <td>0.570</td>\n",
       "      <td>0.407</td>\n",
       "      <td>...</td>\n",
       "      <td>0.238</td>\n",
       "      <td>0.433</td>\n",
       "      <td>0.201</td>\n",
       "      <td>0.272</td>\n",
       "      <td>0.423</td>\n",
       "      <td>0.200</td>\n",
       "      <td>0.150</td>\n",
       "      <td>0.450</td>\n",
       "      <td>0.355</td>\n",
       "      <td>0.320</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>9</th>\n",
       "      <td>0.330</td>\n",
       "      <td>0.394</td>\n",
       "      <td>0.343</td>\n",
       "      <td>0.399</td>\n",
       "      <td>0.328</td>\n",
       "      <td>0.540</td>\n",
       "      <td>0.549</td>\n",
       "      <td>0.570</td>\n",
       "      <td>0.000</td>\n",
       "      <td>0.535</td>\n",
       "      <td>...</td>\n",
       "      <td>0.426</td>\n",
       "      <td>0.338</td>\n",
       "      <td>0.549</td>\n",
       "      <td>0.429</td>\n",
       "      <td>0.414</td>\n",
       "      <td>0.608</td>\n",
       "      <td>0.599</td>\n",
       "      <td>0.394</td>\n",
       "      <td>0.528</td>\n",
       "      <td>0.393</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>10</th>\n",
       "      <td>0.471</td>\n",
       "      <td>0.630</td>\n",
       "      <td>0.369</td>\n",
       "      <td>0.588</td>\n",
       "      <td>0.436</td>\n",
       "      <td>0.432</td>\n",
       "      <td>0.398</td>\n",
       "      <td>0.407</td>\n",
       "      <td>0.535</td>\n",
       "      <td>0.000</td>\n",
       "      <td>...</td>\n",
       "      <td>0.340</td>\n",
       "      <td>0.451</td>\n",
       "      <td>0.480</td>\n",
       "      <td>0.294</td>\n",
       "      <td>0.473</td>\n",
       "      <td>0.482</td>\n",
       "      <td>0.408</td>\n",
       "      <td>0.522</td>\n",
       "      <td>0.315</td>\n",
       "      <td>0.450</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>11</th>\n",
       "      <td>0.262</td>\n",
       "      <td>0.523</td>\n",
       "      <td>0.378</td>\n",
       "      <td>0.453</td>\n",
       "      <td>0.279</td>\n",
       "      <td>0.454</td>\n",
       "      <td>0.583</td>\n",
       "      <td>0.504</td>\n",
       "      <td>0.381</td>\n",
       "      <td>0.583</td>\n",
       "      <td>...</td>\n",
       "      <td>0.389</td>\n",
       "      <td>0.319</td>\n",
       "      <td>0.452</td>\n",
       "      <td>0.439</td>\n",
       "      <td>0.450</td>\n",
       "      <td>0.530</td>\n",
       "      <td>0.536</td>\n",
       "      <td>0.205</td>\n",
       "      <td>0.519</td>\n",
       "      <td>0.301</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>12</th>\n",
       "      <td>0.310</td>\n",
       "      <td>0.561</td>\n",
       "      <td>0.227</td>\n",
       "      <td>0.534</td>\n",
       "      <td>0.233</td>\n",
       "      <td>0.230</td>\n",
       "      <td>0.293</td>\n",
       "      <td>0.238</td>\n",
       "      <td>0.426</td>\n",
       "      <td>0.340</td>\n",
       "      <td>...</td>\n",
       "      <td>0.000</td>\n",
       "      <td>0.289</td>\n",
       "      <td>0.289</td>\n",
       "      <td>0.175</td>\n",
       "      <td>0.310</td>\n",
       "      <td>0.324</td>\n",
       "      <td>0.281</td>\n",
       "      <td>0.302</td>\n",
       "      <td>0.309</td>\n",
       "      <td>0.206</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>13</th>\n",
       "      <td>0.163</td>\n",
       "      <td>0.506</td>\n",
       "      <td>0.231</td>\n",
       "      <td>0.428</td>\n",
       "      <td>0.215</td>\n",
       "      <td>0.376</td>\n",
       "      <td>0.494</td>\n",
       "      <td>0.433</td>\n",
       "      <td>0.338</td>\n",
       "      <td>0.451</td>\n",
       "      <td>...</td>\n",
       "      <td>0.289</td>\n",
       "      <td>0.000</td>\n",
       "      <td>0.392</td>\n",
       "      <td>0.344</td>\n",
       "      <td>0.309</td>\n",
       "      <td>0.472</td>\n",
       "      <td>0.472</td>\n",
       "      <td>0.288</td>\n",
       "      <td>0.411</td>\n",
       "      <td>0.279</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>14</th>\n",
       "      <td>0.413</td>\n",
       "      <td>0.688</td>\n",
       "      <td>0.440</td>\n",
       "      <td>0.549</td>\n",
       "      <td>0.338</td>\n",
       "      <td>0.118</td>\n",
       "      <td>0.312</td>\n",
       "      <td>0.201</td>\n",
       "      <td>0.549</td>\n",
       "      <td>0.480</td>\n",
       "      <td>...</td>\n",
       "      <td>0.289</td>\n",
       "      <td>0.392</td>\n",
       "      <td>0.000</td>\n",
       "      <td>0.401</td>\n",
       "      <td>0.357</td>\n",
       "      <td>0.194</td>\n",
       "      <td>0.241</td>\n",
       "      <td>0.343</td>\n",
       "      <td>0.390</td>\n",
       "      <td>0.303</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>15</th>\n",
       "      <td>0.345</td>\n",
       "      <td>0.513</td>\n",
       "      <td>0.201</td>\n",
       "      <td>0.585</td>\n",
       "      <td>0.273</td>\n",
       "      <td>0.327</td>\n",
       "      <td>0.278</td>\n",
       "      <td>0.272</td>\n",
       "      <td>0.429</td>\n",
       "      <td>0.294</td>\n",
       "      <td>...</td>\n",
       "      <td>0.175</td>\n",
       "      <td>0.344</td>\n",
       "      <td>0.401</td>\n",
       "      <td>0.000</td>\n",
       "      <td>0.372</td>\n",
       "      <td>0.394</td>\n",
       "      <td>0.337</td>\n",
       "      <td>0.390</td>\n",
       "      <td>0.373</td>\n",
       "      <td>0.261</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>16</th>\n",
       "      <td>0.390</td>\n",
       "      <td>0.634</td>\n",
       "      <td>0.384</td>\n",
       "      <td>0.489</td>\n",
       "      <td>0.330</td>\n",
       "      <td>0.363</td>\n",
       "      <td>0.494</td>\n",
       "      <td>0.423</td>\n",
       "      <td>0.414</td>\n",
       "      <td>0.473</td>\n",
       "      <td>...</td>\n",
       "      <td>0.310</td>\n",
       "      <td>0.309</td>\n",
       "      <td>0.357</td>\n",
       "      <td>0.372</td>\n",
       "      <td>0.000</td>\n",
       "      <td>0.426</td>\n",
       "      <td>0.460</td>\n",
       "      <td>0.340</td>\n",
       "      <td>0.408</td>\n",
       "      <td>0.316</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>17</th>\n",
       "      <td>0.501</td>\n",
       "      <td>0.730</td>\n",
       "      <td>0.454</td>\n",
       "      <td>0.633</td>\n",
       "      <td>0.444</td>\n",
       "      <td>0.137</td>\n",
       "      <td>0.265</td>\n",
       "      <td>0.200</td>\n",
       "      <td>0.608</td>\n",
       "      <td>0.482</td>\n",
       "      <td>...</td>\n",
       "      <td>0.324</td>\n",
       "      <td>0.472</td>\n",
       "      <td>0.194</td>\n",
       "      <td>0.394</td>\n",
       "      <td>0.426</td>\n",
       "      <td>0.000</td>\n",
       "      <td>0.122</td>\n",
       "      <td>0.462</td>\n",
       "      <td>0.399</td>\n",
       "      <td>0.372</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>18</th>\n",
       "      <td>0.500</td>\n",
       "      <td>0.715</td>\n",
       "      <td>0.439</td>\n",
       "      <td>0.657</td>\n",
       "      <td>0.435</td>\n",
       "      <td>0.171</td>\n",
       "      <td>0.218</td>\n",
       "      <td>0.150</td>\n",
       "      <td>0.599</td>\n",
       "      <td>0.408</td>\n",
       "      <td>...</td>\n",
       "      <td>0.281</td>\n",
       "      <td>0.472</td>\n",
       "      <td>0.241</td>\n",
       "      <td>0.337</td>\n",
       "      <td>0.460</td>\n",
       "      <td>0.122</td>\n",
       "      <td>0.000</td>\n",
       "      <td>0.485</td>\n",
       "      <td>0.380</td>\n",
       "      <td>0.365</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>19</th>\n",
       "      <td>0.243</td>\n",
       "      <td>0.563</td>\n",
       "      <td>0.350</td>\n",
       "      <td>0.373</td>\n",
       "      <td>0.190</td>\n",
       "      <td>0.381</td>\n",
       "      <td>0.528</td>\n",
       "      <td>0.450</td>\n",
       "      <td>0.394</td>\n",
       "      <td>0.522</td>\n",
       "      <td>...</td>\n",
       "      <td>0.302</td>\n",
       "      <td>0.288</td>\n",
       "      <td>0.343</td>\n",
       "      <td>0.390</td>\n",
       "      <td>0.340</td>\n",
       "      <td>0.462</td>\n",
       "      <td>0.485</td>\n",
       "      <td>0.000</td>\n",
       "      <td>0.434</td>\n",
       "      <td>0.224</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>20</th>\n",
       "      <td>0.459</td>\n",
       "      <td>0.677</td>\n",
       "      <td>0.425</td>\n",
       "      <td>0.449</td>\n",
       "      <td>0.415</td>\n",
       "      <td>0.351</td>\n",
       "      <td>0.427</td>\n",
       "      <td>0.355</td>\n",
       "      <td>0.528</td>\n",
       "      <td>0.315</td>\n",
       "      <td>...</td>\n",
       "      <td>0.309</td>\n",
       "      <td>0.411</td>\n",
       "      <td>0.390</td>\n",
       "      <td>0.373</td>\n",
       "      <td>0.408</td>\n",
       "      <td>0.399</td>\n",
       "      <td>0.380</td>\n",
       "      <td>0.434</td>\n",
       "      <td>0.000</td>\n",
       "      <td>0.414</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>21</th>\n",
       "      <td>0.251</td>\n",
       "      <td>0.524</td>\n",
       "      <td>0.291</td>\n",
       "      <td>0.480</td>\n",
       "      <td>0.157</td>\n",
       "      <td>0.287</td>\n",
       "      <td>0.387</td>\n",
       "      <td>0.320</td>\n",
       "      <td>0.393</td>\n",
       "      <td>0.450</td>\n",
       "      <td>...</td>\n",
       "      <td>0.206</td>\n",
       "      <td>0.279</td>\n",
       "      <td>0.303</td>\n",
       "      <td>0.261</td>\n",
       "      <td>0.316</td>\n",
       "      <td>0.372</td>\n",
       "      <td>0.365</td>\n",
       "      <td>0.224</td>\n",
       "      <td>0.414</td>\n",
       "      <td>0.000</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>21 rows × 21 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "       1      2      3      4      5      6      7      8      9      10  ...  \\\n",
       "1   0.000  0.453  0.225  0.426  0.180  0.414  0.500  0.466  0.330  0.471  ...   \n",
       "2   0.453  0.000  0.437  0.628  0.472  0.669  0.625  0.692  0.394  0.630  ...   \n",
       "3   0.225  0.437  0.000  0.524  0.249  0.372  0.364  0.407  0.343  0.369  ...   \n",
       "4   0.426  0.628  0.524  0.000  0.439  0.588  0.667  0.636  0.399  0.588  ...   \n",
       "5   0.180  0.472  0.249  0.439  0.000  0.350  0.451  0.379  0.328  0.436  ...   \n",
       "6   0.414  0.669  0.372  0.588  0.350  0.000  0.238  0.126  0.540  0.432  ...   \n",
       "7   0.500  0.625  0.364  0.667  0.451  0.238  0.000  0.217  0.549  0.398  ...   \n",
       "8   0.466  0.692  0.407  0.636  0.379  0.126  0.217  0.000  0.570  0.407  ...   \n",
       "9   0.330  0.394  0.343  0.399  0.328  0.540  0.549  0.570  0.000  0.535  ...   \n",
       "10  0.471  0.630  0.369  0.588  0.436  0.432  0.398  0.407  0.535  0.000  ...   \n",
       "11  0.262  0.523  0.378  0.453  0.279  0.454  0.583  0.504  0.381  0.583  ...   \n",
       "12  0.310  0.561  0.227  0.534  0.233  0.230  0.293  0.238  0.426  0.340  ...   \n",
       "13  0.163  0.506  0.231  0.428  0.215  0.376  0.494  0.433  0.338  0.451  ...   \n",
       "14  0.413  0.688  0.440  0.549  0.338  0.118  0.312  0.201  0.549  0.480  ...   \n",
       "15  0.345  0.513  0.201  0.585  0.273  0.327  0.278  0.272  0.429  0.294  ...   \n",
       "16  0.390  0.634  0.384  0.489  0.330  0.363  0.494  0.423  0.414  0.473  ...   \n",
       "17  0.501  0.730  0.454  0.633  0.444  0.137  0.265  0.200  0.608  0.482  ...   \n",
       "18  0.500  0.715  0.439  0.657  0.435  0.171  0.218  0.150  0.599  0.408  ...   \n",
       "19  0.243  0.563  0.350  0.373  0.190  0.381  0.528  0.450  0.394  0.522  ...   \n",
       "20  0.459  0.677  0.425  0.449  0.415  0.351  0.427  0.355  0.528  0.315  ...   \n",
       "21  0.251  0.524  0.291  0.480  0.157  0.287  0.387  0.320  0.393  0.450  ...   \n",
       "\n",
       "       12     13     14     15     16     17     18     19     20     21  \n",
       "1   0.310  0.163  0.413  0.345  0.390  0.501  0.500  0.243  0.459  0.251  \n",
       "2   0.561  0.506  0.688  0.513  0.634  0.730  0.715  0.563  0.677  0.524  \n",
       "3   0.227  0.231  0.440  0.201  0.384  0.454  0.439  0.350  0.425  0.291  \n",
       "4   0.534  0.428  0.549  0.585  0.489  0.633  0.657  0.373  0.449  0.480  \n",
       "5   0.233  0.215  0.338  0.273  0.330  0.444  0.435  0.190  0.415  0.157  \n",
       "6   0.230  0.376  0.118  0.327  0.363  0.137  0.171  0.381  0.351  0.287  \n",
       "7   0.293  0.494  0.312  0.278  0.494  0.265  0.218  0.528  0.427  0.387  \n",
       "8   0.238  0.433  0.201  0.272  0.423  0.200  0.150  0.450  0.355  0.320  \n",
       "9   0.426  0.338  0.549  0.429  0.414  0.608  0.599  0.394  0.528  0.393  \n",
       "10  0.340  0.451  0.480  0.294  0.473  0.482  0.408  0.522  0.315  0.450  \n",
       "11  0.389  0.319  0.452  0.439  0.450  0.530  0.536  0.205  0.519  0.301  \n",
       "12  0.000  0.289  0.289  0.175  0.310  0.324  0.281  0.302  0.309  0.206  \n",
       "13  0.289  0.000  0.392  0.344  0.309  0.472  0.472  0.288  0.411  0.279  \n",
       "14  0.289  0.392  0.000  0.401  0.357  0.194  0.241  0.343  0.390  0.303  \n",
       "15  0.175  0.344  0.401  0.000  0.372  0.394  0.337  0.390  0.373  0.261  \n",
       "16  0.310  0.309  0.357  0.372  0.000  0.426  0.460  0.340  0.408  0.316  \n",
       "17  0.324  0.472  0.194  0.394  0.426  0.000  0.122  0.462  0.399  0.372  \n",
       "18  0.281  0.472  0.241  0.337  0.460  0.122  0.000  0.485  0.380  0.365  \n",
       "19  0.302  0.288  0.343  0.390  0.340  0.462  0.485  0.000  0.434  0.224  \n",
       "20  0.309  0.411  0.390  0.373  0.408  0.399  0.380  0.434  0.000  0.414  \n",
       "21  0.206  0.279  0.303  0.261  0.316  0.372  0.365  0.224  0.414  0.000  \n",
       "\n",
       "[21 rows x 21 columns]"
      ]
     },
     "execution_count": 118,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "#convert to dataframe mainly to view better\n",
    "bd_df=pd.DataFrame(bd3)\n",
    "\n",
    "#column and row headings were doubled because of the vectorized addition, so dropping those first columns and rows \n",
    "df=bd_df.drop(columns=[0])\n",
    "df2=df.drop([0])\n",
    "df2"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 9,
   "id": "d62e4c02-1464-44d9-88ce-7ce470926d4f",
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "['T1_12_2022', 'T1_13_2022', 'T1_16_2019', 'T1_20_2019', 'T1_24_2019', 'T1_40_2022', 'T1_57_2022', 'T1_70_2022', 'T2_10_2022', 'T2_16_2019', 'T3_13_2022', 'T3_14_2019', 'T3_15_2019', 'T3_19_2022', 'T3_1_2019', 'T3_40_2022', 'T3_48_2022', 'T3_49_2022', 'T3_51_2022', 'T3_8_2019', 'T3_9_2019']\n"
     ]
    }
   ],
   "source": [
    "# Need to rename column names with sample IDs\n",
    "\n",
    "#read file and seperate lines into a list of values\n",
    "FILEPATH = '//project/pi_sarah_gignouxwolfsohn_uml_edu/brooke/mcav/taxonomy/kraken/SAMPLEID'\n",
    "with open(FILEPATH, 'r') as file:\n",
    "    sampleid = file.read().split()\n",
    "\n",
    "# SAMPLEID contains list of sample ids \n",
    "print(sampleid)\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 165,
   "id": "3cc5fb18-2578-494f-80c2-1f05bf6ac602",
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "    T1_12_2022  T1_13_2022  T1_16_2019  T1_20_2019  T1_24_2019  T1_40_2022  \\\n",
      "1        0.000       0.453       0.225       0.426       0.180       0.414   \n",
      "2        0.453       0.000       0.437       0.628       0.472       0.669   \n",
      "3        0.225       0.437       0.000       0.524       0.249       0.372   \n",
      "4        0.426       0.628       0.524       0.000       0.439       0.588   \n",
      "5        0.180       0.472       0.249       0.439       0.000       0.350   \n",
      "6        0.414       0.669       0.372       0.588       0.350       0.000   \n",
      "7        0.500       0.625       0.364       0.667       0.451       0.238   \n",
      "8        0.466       0.692       0.407       0.636       0.379       0.126   \n",
      "9        0.330       0.394       0.343       0.399       0.328       0.540   \n",
      "10       0.471       0.630       0.369       0.588       0.436       0.432   \n",
      "11       0.262       0.523       0.378       0.453       0.279       0.454   \n",
      "12       0.310       0.561       0.227       0.534       0.233       0.230   \n",
      "13       0.163       0.506       0.231       0.428       0.215       0.376   \n",
      "14       0.413       0.688       0.440       0.549       0.338       0.118   \n",
      "15       0.345       0.513       0.201       0.585       0.273       0.327   \n",
      "16       0.390       0.634       0.384       0.489       0.330       0.363   \n",
      "17       0.501       0.730       0.454       0.633       0.444       0.137   \n",
      "18       0.500       0.715       0.439       0.657       0.435       0.171   \n",
      "19       0.243       0.563       0.350       0.373       0.190       0.381   \n",
      "20       0.459       0.677       0.425       0.449       0.415       0.351   \n",
      "21       0.251       0.524       0.291       0.480       0.157       0.287   \n",
      "\n",
      "    T1_57_2022  T1_70_2022  T2_10_2022  T2_16_2019  ...  T3_14_2019  \\\n",
      "1        0.500       0.466       0.330       0.471  ...       0.310   \n",
      "2        0.625       0.692       0.394       0.630  ...       0.561   \n",
      "3        0.364       0.407       0.343       0.369  ...       0.227   \n",
      "4        0.667       0.636       0.399       0.588  ...       0.534   \n",
      "5        0.451       0.379       0.328       0.436  ...       0.233   \n",
      "6        0.238       0.126       0.540       0.432  ...       0.230   \n",
      "7        0.000       0.217       0.549       0.398  ...       0.293   \n",
      "8        0.217       0.000       0.570       0.407  ...       0.238   \n",
      "9        0.549       0.570       0.000       0.535  ...       0.426   \n",
      "10       0.398       0.407       0.535       0.000  ...       0.340   \n",
      "11       0.583       0.504       0.381       0.583  ...       0.389   \n",
      "12       0.293       0.238       0.426       0.340  ...       0.000   \n",
      "13       0.494       0.433       0.338       0.451  ...       0.289   \n",
      "14       0.312       0.201       0.549       0.480  ...       0.289   \n",
      "15       0.278       0.272       0.429       0.294  ...       0.175   \n",
      "16       0.494       0.423       0.414       0.473  ...       0.310   \n",
      "17       0.265       0.200       0.608       0.482  ...       0.324   \n",
      "18       0.218       0.150       0.599       0.408  ...       0.281   \n",
      "19       0.528       0.450       0.394       0.522  ...       0.302   \n",
      "20       0.427       0.355       0.528       0.315  ...       0.309   \n",
      "21       0.387       0.320       0.393       0.450  ...       0.206   \n",
      "\n",
      "    T3_15_2019  T3_19_2022  T3_1_2019  T3_40_2022  T3_48_2022  T3_49_2022  \\\n",
      "1        0.163       0.413      0.345       0.390       0.501       0.500   \n",
      "2        0.506       0.688      0.513       0.634       0.730       0.715   \n",
      "3        0.231       0.440      0.201       0.384       0.454       0.439   \n",
      "4        0.428       0.549      0.585       0.489       0.633       0.657   \n",
      "5        0.215       0.338      0.273       0.330       0.444       0.435   \n",
      "6        0.376       0.118      0.327       0.363       0.137       0.171   \n",
      "7        0.494       0.312      0.278       0.494       0.265       0.218   \n",
      "8        0.433       0.201      0.272       0.423       0.200       0.150   \n",
      "9        0.338       0.549      0.429       0.414       0.608       0.599   \n",
      "10       0.451       0.480      0.294       0.473       0.482       0.408   \n",
      "11       0.319       0.452      0.439       0.450       0.530       0.536   \n",
      "12       0.289       0.289      0.175       0.310       0.324       0.281   \n",
      "13       0.000       0.392      0.344       0.309       0.472       0.472   \n",
      "14       0.392       0.000      0.401       0.357       0.194       0.241   \n",
      "15       0.344       0.401      0.000       0.372       0.394       0.337   \n",
      "16       0.309       0.357      0.372       0.000       0.426       0.460   \n",
      "17       0.472       0.194      0.394       0.426       0.000       0.122   \n",
      "18       0.472       0.241      0.337       0.460       0.122       0.000   \n",
      "19       0.288       0.343      0.390       0.340       0.462       0.485   \n",
      "20       0.411       0.390      0.373       0.408       0.399       0.380   \n",
      "21       0.279       0.303      0.261       0.316       0.372       0.365   \n",
      "\n",
      "    T3_51_2022  T3_8_2019  T3_9_2019  \n",
      "1        0.243      0.459      0.251  \n",
      "2        0.563      0.677      0.524  \n",
      "3        0.350      0.425      0.291  \n",
      "4        0.373      0.449      0.480  \n",
      "5        0.190      0.415      0.157  \n",
      "6        0.381      0.351      0.287  \n",
      "7        0.528      0.427      0.387  \n",
      "8        0.450      0.355      0.320  \n",
      "9        0.394      0.528      0.393  \n",
      "10       0.522      0.315      0.450  \n",
      "11       0.205      0.519      0.301  \n",
      "12       0.302      0.309      0.206  \n",
      "13       0.288      0.411      0.279  \n",
      "14       0.343      0.390      0.303  \n",
      "15       0.390      0.373      0.261  \n",
      "16       0.340      0.408      0.316  \n",
      "17       0.462      0.399      0.372  \n",
      "18       0.485      0.380      0.365  \n",
      "19       0.000      0.434      0.224  \n",
      "20       0.434      0.000      0.414  \n",
      "21       0.224      0.414      0.000  \n",
      "\n",
      "[21 rows x 21 columns]\n"
     ]
    }
   ],
   "source": [
    "# Rename columns using the 'sampleid' list\n",
    "df2.columns = sampleid\n",
    "print(df2)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 168,
   "id": "b7f16fc1-7d11-401e-8da2-29d4edb2af4d",
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "21x21 distance matrix\n",
      "IDs:\n",
      "'T1_12_2022', 'T1_13_2022', 'T1_16_2019', 'T1_20_2019', 'T1_24_2019', ...\n",
      "Data:\n",
      "[[0.    0.453 0.225 0.426 0.18  0.414 0.5   0.466 0.33  0.471 0.262 0.31\n",
      "  0.163 0.413 0.345 0.39  0.501 0.5   0.243 0.459 0.251]\n",
      " [0.453 0.    0.437 0.628 0.472 0.669 0.625 0.692 0.394 0.63  0.523 0.561\n",
      "  0.506 0.688 0.513 0.634 0.73  0.715 0.563 0.677 0.524]\n",
      " [0.225 0.437 0.    0.524 0.249 0.372 0.364 0.407 0.343 0.369 0.378 0.227\n",
      "  0.231 0.44  0.201 0.384 0.454 0.439 0.35  0.425 0.291]\n",
      " [0.426 0.628 0.524 0.    0.439 0.588 0.667 0.636 0.399 0.588 0.453 0.534\n",
      "  0.428 0.549 0.585 0.489 0.633 0.657 0.373 0.449 0.48 ]\n",
      " [0.18  0.472 0.249 0.439 0.    0.35  0.451 0.379 0.328 0.436 0.279 0.233\n",
      "  0.215 0.338 0.273 0.33  0.444 0.435 0.19  0.415 0.157]\n",
      " [0.414 0.669 0.372 0.588 0.35  0.    0.238 0.126 0.54  0.432 0.454 0.23\n",
      "  0.376 0.118 0.327 0.363 0.137 0.171 0.381 0.351 0.287]\n",
      " [0.5   0.625 0.364 0.667 0.451 0.238 0.    0.217 0.549 0.398 0.583 0.293\n",
      "  0.494 0.312 0.278 0.494 0.265 0.218 0.528 0.427 0.387]\n",
      " [0.466 0.692 0.407 0.636 0.379 0.126 0.217 0.    0.57  0.407 0.504 0.238\n",
      "  0.433 0.201 0.272 0.423 0.2   0.15  0.45  0.355 0.32 ]\n",
      " [0.33  0.394 0.343 0.399 0.328 0.54  0.549 0.57  0.    0.535 0.381 0.426\n",
      "  0.338 0.549 0.429 0.414 0.608 0.599 0.394 0.528 0.393]\n",
      " [0.471 0.63  0.369 0.588 0.436 0.432 0.398 0.407 0.535 0.    0.583 0.34\n",
      "  0.451 0.48  0.294 0.473 0.482 0.408 0.522 0.315 0.45 ]\n",
      " [0.262 0.523 0.378 0.453 0.279 0.454 0.583 0.504 0.381 0.583 0.    0.389\n",
      "  0.319 0.452 0.439 0.45  0.53  0.536 0.205 0.519 0.301]\n",
      " [0.31  0.561 0.227 0.534 0.233 0.23  0.293 0.238 0.426 0.34  0.389 0.\n",
      "  0.289 0.289 0.175 0.31  0.324 0.281 0.302 0.309 0.206]\n",
      " [0.163 0.506 0.231 0.428 0.215 0.376 0.494 0.433 0.338 0.451 0.319 0.289\n",
      "  0.    0.392 0.344 0.309 0.472 0.472 0.288 0.411 0.279]\n",
      " [0.413 0.688 0.44  0.549 0.338 0.118 0.312 0.201 0.549 0.48  0.452 0.289\n",
      "  0.392 0.    0.401 0.357 0.194 0.241 0.343 0.39  0.303]\n",
      " [0.345 0.513 0.201 0.585 0.273 0.327 0.278 0.272 0.429 0.294 0.439 0.175\n",
      "  0.344 0.401 0.    0.372 0.394 0.337 0.39  0.373 0.261]\n",
      " [0.39  0.634 0.384 0.489 0.33  0.363 0.494 0.423 0.414 0.473 0.45  0.31\n",
      "  0.309 0.357 0.372 0.    0.426 0.46  0.34  0.408 0.316]\n",
      " [0.501 0.73  0.454 0.633 0.444 0.137 0.265 0.2   0.608 0.482 0.53  0.324\n",
      "  0.472 0.194 0.394 0.426 0.    0.122 0.462 0.399 0.372]\n",
      " [0.5   0.715 0.439 0.657 0.435 0.171 0.218 0.15  0.599 0.408 0.536 0.281\n",
      "  0.472 0.241 0.337 0.46  0.122 0.    0.485 0.38  0.365]\n",
      " [0.243 0.563 0.35  0.373 0.19  0.381 0.528 0.45  0.394 0.522 0.205 0.302\n",
      "  0.288 0.343 0.39  0.34  0.462 0.485 0.    0.434 0.224]\n",
      " [0.459 0.677 0.425 0.449 0.415 0.351 0.427 0.355 0.528 0.315 0.519 0.309\n",
      "  0.411 0.39  0.373 0.408 0.399 0.38  0.434 0.    0.414]\n",
      " [0.251 0.524 0.291 0.48  0.157 0.287 0.387 0.32  0.393 0.45  0.301 0.206\n",
      "  0.279 0.303 0.261 0.316 0.372 0.365 0.224 0.414 0.   ]]\n"
     ]
    }
   ],
   "source": [
    "from skbio import DistanceMatrix\n",
    "# Convert dataframe to numpy array\n",
    "bdist = df2.to_numpy()\n",
    "\n",
    "# Create a DistanceMatrix object and keep column names as headings in matrix \n",
    "distance_matrix = DistanceMatrix(bdist, ids=df2.columns)\n",
    "print(distance_matrix)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 141,
   "id": "607938e1-3e03-4ea0-97ad-7c297198781b",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>Month_year</th>\n",
       "      <th>Country</th>\n",
       "      <th>Location</th>\n",
       "      <th>CollectionDate</th>\n",
       "      <th>Transect</th>\n",
       "      <th>TransectNum</th>\n",
       "      <th>OldTagNum</th>\n",
       "      <th>NewTagNum</th>\n",
       "      <th>SampleNum</th>\n",
       "      <th>Species</th>\n",
       "      <th>Sample_type</th>\n",
       "      <th>Health status</th>\n",
       "      <th>Sampling_notes</th>\n",
       "      <th>Sample_physical_location</th>\n",
       "      <th>Extraction_physical_location</th>\n",
       "      <th>Date_sequenced</th>\n",
       "      <th>Notes</th>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>Tubelabel_species</th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>122022_BEL_CBC_T4_2_OFAV</th>\n",
       "      <td>122022.0</td>\n",
       "      <td>BEL</td>\n",
       "      <td>CBC</td>\n",
       "      <td>12/5/22</td>\n",
       "      <td>CURLEW</td>\n",
       "      <td>4.0</td>\n",
       "      <td>NaN</td>\n",
       "      <td>77</td>\n",
       "      <td>2</td>\n",
       "      <td>OFAV</td>\n",
       "      <td>Core_EtOH</td>\n",
       "      <td>Healthy</td>\n",
       "      <td>NaN</td>\n",
       "      <td>UML_F1_R1_B5</td>\n",
       "      <td>UML_F1_R2_B3</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>92023_BEL_CBC_T1_171_PAST</th>\n",
       "      <td>92023.0</td>\n",
       "      <td>BEL</td>\n",
       "      <td>CBC</td>\n",
       "      <td>9/25/23</td>\n",
       "      <td>CBC30N</td>\n",
       "      <td>1.0</td>\n",
       "      <td>NaN</td>\n",
       "      <td>2</td>\n",
       "      <td>171</td>\n",
       "      <td>PAST</td>\n",
       "      <td>Core_RNAlater</td>\n",
       "      <td>Healthy</td>\n",
       "      <td>CLP 90%</td>\n",
       "      <td>UML_F1_R1_B10</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>92023_BEL_CBC_T1_172_PAST</th>\n",
       "      <td>92023.0</td>\n",
       "      <td>BEL</td>\n",
       "      <td>CBC</td>\n",
       "      <td>9/25/23</td>\n",
       "      <td>CBC30N</td>\n",
       "      <td>1.0</td>\n",
       "      <td>NaN</td>\n",
       "      <td>21</td>\n",
       "      <td>172</td>\n",
       "      <td>PAST</td>\n",
       "      <td>Core_RNAlater</td>\n",
       "      <td>Healthy</td>\n",
       "      <td>No CL</td>\n",
       "      <td>UML_F1_R1_B10</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>92023_BEL_CBC_T1_173_SSID</th>\n",
       "      <td>92023.0</td>\n",
       "      <td>BEL</td>\n",
       "      <td>CBC</td>\n",
       "      <td>9/25/23</td>\n",
       "      <td>CBC30N</td>\n",
       "      <td>1.0</td>\n",
       "      <td>NaN</td>\n",
       "      <td>3</td>\n",
       "      <td>173</td>\n",
       "      <td>SSID</td>\n",
       "      <td>Core_RNAlater</td>\n",
       "      <td>Healthy</td>\n",
       "      <td>CLP 80%; DC 20%</td>\n",
       "      <td>UML_F1_R1_B10</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>92023_BEL_CBC_T1_174_MCAV</th>\n",
       "      <td>92023.0</td>\n",
       "      <td>BEL</td>\n",
       "      <td>CBC</td>\n",
       "      <td>9/25/23</td>\n",
       "      <td>CBC30N</td>\n",
       "      <td>1.0</td>\n",
       "      <td>NaN</td>\n",
       "      <td>24</td>\n",
       "      <td>174</td>\n",
       "      <td>MCAV</td>\n",
       "      <td>Core_RNAlater</td>\n",
       "      <td>Healthy</td>\n",
       "      <td>CLP 10%</td>\n",
       "      <td>UML_F1_R1_B10</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>11_2023_CBC-BEL_T1_285_PSTR</th>\n",
       "      <td>112023.0</td>\n",
       "      <td>BEL</td>\n",
       "      <td>CBC</td>\n",
       "      <td>11/14/23</td>\n",
       "      <td>CBC30N</td>\n",
       "      <td>1.0</td>\n",
       "      <td>NaN</td>\n",
       "      <td>12</td>\n",
       "      <td>285</td>\n",
       "      <td>PSTR</td>\n",
       "      <td>Core_frozen</td>\n",
       "      <td>Healthy</td>\n",
       "      <td>100% CLP</td>\n",
       "      <td>TXST -80</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>11_2023_CBC-BEL_T1_292_PSTR</th>\n",
       "      <td>112023.0</td>\n",
       "      <td>BEL</td>\n",
       "      <td>CBC</td>\n",
       "      <td>11/14/23</td>\n",
       "      <td>CBC30N</td>\n",
       "      <td>1.0</td>\n",
       "      <td>NaN</td>\n",
       "      <td>12</td>\n",
       "      <td>292</td>\n",
       "      <td>PSTR</td>\n",
       "      <td>Immune</td>\n",
       "      <td>Healthy</td>\n",
       "      <td>100% CLP</td>\n",
       "      <td>TXST -80</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>11_2023_CBC-BEL_T1_277_OFAV/OANN</th>\n",
       "      <td>112023.0</td>\n",
       "      <td>BEL</td>\n",
       "      <td>CBC</td>\n",
       "      <td>11/14/23</td>\n",
       "      <td>CBC30N</td>\n",
       "      <td>1.0</td>\n",
       "      <td>NaN</td>\n",
       "      <td>25</td>\n",
       "      <td>277</td>\n",
       "      <td>OFAV/OANN</td>\n",
       "      <td>Core_RNAlater</td>\n",
       "      <td>Healthy</td>\n",
       "      <td>5% CLB, 5% CLP, 2% TL; TL likely from predatio...</td>\n",
       "      <td>TXST -20</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>11_2023_CBC-BEL_T1_284_OFAV/OANN</th>\n",
       "      <td>112023.0</td>\n",
       "      <td>BEL</td>\n",
       "      <td>CBC</td>\n",
       "      <td>11/14/23</td>\n",
       "      <td>CBC30N</td>\n",
       "      <td>1.0</td>\n",
       "      <td>NaN</td>\n",
       "      <td>25</td>\n",
       "      <td>284</td>\n",
       "      <td>OFAV/OANN</td>\n",
       "      <td>Core_frozen</td>\n",
       "      <td>Healthy</td>\n",
       "      <td>5% CLB, 5% CLP, 2% TL; TL likely from predatio...</td>\n",
       "      <td>TXST -80</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>11_2023_CBC-BEL_T1_291_OFAV/OANN</th>\n",
       "      <td>112023.0</td>\n",
       "      <td>BEL</td>\n",
       "      <td>CBC</td>\n",
       "      <td>11/14/23</td>\n",
       "      <td>CBC30N</td>\n",
       "      <td>1.0</td>\n",
       "      <td>NaN</td>\n",
       "      <td>25</td>\n",
       "      <td>291</td>\n",
       "      <td>OFAV/OANN</td>\n",
       "      <td>Immune</td>\n",
       "      <td>Healthy</td>\n",
       "      <td>5% CLB, 5% CLP, 2% TL; TL likely from predatio...</td>\n",
       "      <td>TXST -80</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>1269 rows × 17 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "                                  Month_year Country Location CollectionDate  \\\n",
       "Tubelabel_species                                                              \n",
       "122022_BEL_CBC_T4_2_OFAV            122022.0     BEL      CBC        12/5/22   \n",
       "92023_BEL_CBC_T1_171_PAST            92023.0     BEL      CBC        9/25/23   \n",
       "92023_BEL_CBC_T1_172_PAST            92023.0     BEL      CBC        9/25/23   \n",
       "92023_BEL_CBC_T1_173_SSID            92023.0     BEL      CBC        9/25/23   \n",
       "92023_BEL_CBC_T1_174_MCAV            92023.0     BEL      CBC        9/25/23   \n",
       "...                                      ...     ...      ...            ...   \n",
       "11_2023_CBC-BEL_T1_285_PSTR         112023.0     BEL      CBC       11/14/23   \n",
       "11_2023_CBC-BEL_T1_292_PSTR         112023.0     BEL      CBC       11/14/23   \n",
       "11_2023_CBC-BEL_T1_277_OFAV/OANN    112023.0     BEL      CBC       11/14/23   \n",
       "11_2023_CBC-BEL_T1_284_OFAV/OANN    112023.0     BEL      CBC       11/14/23   \n",
       "11_2023_CBC-BEL_T1_291_OFAV/OANN    112023.0     BEL      CBC       11/14/23   \n",
       "\n",
       "                                 Transect   TransectNum OldTagNum NewTagNum  \\\n",
       "Tubelabel_species                                                             \n",
       "122022_BEL_CBC_T4_2_OFAV            CURLEW          4.0       NaN        77   \n",
       "92023_BEL_CBC_T1_171_PAST           CBC30N          1.0       NaN         2   \n",
       "92023_BEL_CBC_T1_172_PAST           CBC30N          1.0       NaN        21   \n",
       "92023_BEL_CBC_T1_173_SSID           CBC30N          1.0       NaN         3   \n",
       "92023_BEL_CBC_T1_174_MCAV           CBC30N          1.0       NaN        24   \n",
       "...                                    ...          ...       ...       ...   \n",
       "11_2023_CBC-BEL_T1_285_PSTR         CBC30N          1.0       NaN        12   \n",
       "11_2023_CBC-BEL_T1_292_PSTR         CBC30N          1.0       NaN        12   \n",
       "11_2023_CBC-BEL_T1_277_OFAV/OANN    CBC30N          1.0       NaN        25   \n",
       "11_2023_CBC-BEL_T1_284_OFAV/OANN    CBC30N          1.0       NaN        25   \n",
       "11_2023_CBC-BEL_T1_291_OFAV/OANN    CBC30N          1.0       NaN        25   \n",
       "\n",
       "                                 SampleNum    Species    Sample_type  \\\n",
       "Tubelabel_species                                                      \n",
       "122022_BEL_CBC_T4_2_OFAV                 2       OFAV      Core_EtOH   \n",
       "92023_BEL_CBC_T1_171_PAST              171       PAST  Core_RNAlater   \n",
       "92023_BEL_CBC_T1_172_PAST              172       PAST  Core_RNAlater   \n",
       "92023_BEL_CBC_T1_173_SSID              173       SSID  Core_RNAlater   \n",
       "92023_BEL_CBC_T1_174_MCAV              174       MCAV  Core_RNAlater   \n",
       "...                                    ...        ...            ...   \n",
       "11_2023_CBC-BEL_T1_285_PSTR            285       PSTR    Core_frozen   \n",
       "11_2023_CBC-BEL_T1_292_PSTR            292       PSTR         Immune   \n",
       "11_2023_CBC-BEL_T1_277_OFAV/OANN       277  OFAV/OANN  Core_RNAlater   \n",
       "11_2023_CBC-BEL_T1_284_OFAV/OANN       284  OFAV/OANN    Core_frozen   \n",
       "11_2023_CBC-BEL_T1_291_OFAV/OANN       291  OFAV/OANN         Immune   \n",
       "\n",
       "                                 Health status  \\\n",
       "Tubelabel_species                                \n",
       "122022_BEL_CBC_T4_2_OFAV               Healthy   \n",
       "92023_BEL_CBC_T1_171_PAST              Healthy   \n",
       "92023_BEL_CBC_T1_172_PAST              Healthy   \n",
       "92023_BEL_CBC_T1_173_SSID              Healthy   \n",
       "92023_BEL_CBC_T1_174_MCAV              Healthy   \n",
       "...                                        ...   \n",
       "11_2023_CBC-BEL_T1_285_PSTR            Healthy   \n",
       "11_2023_CBC-BEL_T1_292_PSTR            Healthy   \n",
       "11_2023_CBC-BEL_T1_277_OFAV/OANN       Healthy   \n",
       "11_2023_CBC-BEL_T1_284_OFAV/OANN       Healthy   \n",
       "11_2023_CBC-BEL_T1_291_OFAV/OANN       Healthy   \n",
       "\n",
       "                                                                     Sampling_notes  \\\n",
       "Tubelabel_species                                                                     \n",
       "122022_BEL_CBC_T4_2_OFAV                                                        NaN   \n",
       "92023_BEL_CBC_T1_171_PAST                                                   CLP 90%   \n",
       "92023_BEL_CBC_T1_172_PAST                                                     No CL   \n",
       "92023_BEL_CBC_T1_173_SSID                                           CLP 80%; DC 20%   \n",
       "92023_BEL_CBC_T1_174_MCAV                                                   CLP 10%   \n",
       "...                                                                             ...   \n",
       "11_2023_CBC-BEL_T1_285_PSTR                                                100% CLP   \n",
       "11_2023_CBC-BEL_T1_292_PSTR                                                100% CLP   \n",
       "11_2023_CBC-BEL_T1_277_OFAV/OANN  5% CLB, 5% CLP, 2% TL; TL likely from predatio...   \n",
       "11_2023_CBC-BEL_T1_284_OFAV/OANN  5% CLB, 5% CLP, 2% TL; TL likely from predatio...   \n",
       "11_2023_CBC-BEL_T1_291_OFAV/OANN  5% CLB, 5% CLP, 2% TL; TL likely from predatio...   \n",
       "\n",
       "                                 Sample_physical_location  \\\n",
       "Tubelabel_species                                           \n",
       "122022_BEL_CBC_T4_2_OFAV                     UML_F1_R1_B5   \n",
       "92023_BEL_CBC_T1_171_PAST                   UML_F1_R1_B10   \n",
       "92023_BEL_CBC_T1_172_PAST                   UML_F1_R1_B10   \n",
       "92023_BEL_CBC_T1_173_SSID                   UML_F1_R1_B10   \n",
       "92023_BEL_CBC_T1_174_MCAV                   UML_F1_R1_B10   \n",
       "...                                                   ...   \n",
       "11_2023_CBC-BEL_T1_285_PSTR                      TXST -80   \n",
       "11_2023_CBC-BEL_T1_292_PSTR                      TXST -80   \n",
       "11_2023_CBC-BEL_T1_277_OFAV/OANN                 TXST -20   \n",
       "11_2023_CBC-BEL_T1_284_OFAV/OANN                 TXST -80   \n",
       "11_2023_CBC-BEL_T1_291_OFAV/OANN                 TXST -80   \n",
       "\n",
       "                                 Extraction_physical_location Date_sequenced  \\\n",
       "Tubelabel_species                                                              \n",
       "122022_BEL_CBC_T4_2_OFAV                         UML_F1_R2_B3            NaN   \n",
       "92023_BEL_CBC_T1_171_PAST                                 NaN            NaN   \n",
       "92023_BEL_CBC_T1_172_PAST                                 NaN            NaN   \n",
       "92023_BEL_CBC_T1_173_SSID                                 NaN            NaN   \n",
       "92023_BEL_CBC_T1_174_MCAV                                 NaN            NaN   \n",
       "...                                                       ...            ...   \n",
       "11_2023_CBC-BEL_T1_285_PSTR                               NaN            NaN   \n",
       "11_2023_CBC-BEL_T1_292_PSTR                               NaN            NaN   \n",
       "11_2023_CBC-BEL_T1_277_OFAV/OANN                          NaN            NaN   \n",
       "11_2023_CBC-BEL_T1_284_OFAV/OANN                          NaN            NaN   \n",
       "11_2023_CBC-BEL_T1_291_OFAV/OANN                          NaN            NaN   \n",
       "\n",
       "                                 Notes  \n",
       "Tubelabel_species                       \n",
       "122022_BEL_CBC_T4_2_OFAV           NaN  \n",
       "92023_BEL_CBC_T1_171_PAST          NaN  \n",
       "92023_BEL_CBC_T1_172_PAST          NaN  \n",
       "92023_BEL_CBC_T1_173_SSID          NaN  \n",
       "92023_BEL_CBC_T1_174_MCAV          NaN  \n",
       "...                                ...  \n",
       "11_2023_CBC-BEL_T1_285_PSTR        NaN  \n",
       "11_2023_CBC-BEL_T1_292_PSTR        NaN  \n",
       "11_2023_CBC-BEL_T1_277_OFAV/OANN   NaN  \n",
       "11_2023_CBC-BEL_T1_284_OFAV/OANN   NaN  \n",
       "11_2023_CBC-BEL_T1_291_OFAV/OANN   NaN  \n",
       "\n",
       "[1269 rows x 17 columns]"
      ]
     },
     "execution_count": 141,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "# load in CBC_Samples metadata \n",
    "metapath= '//project/pi_sarah_gignouxwolfsohn_uml_edu/brooke/metadata/CBC_samples.csv'\n",
    "meta=pd.read_csv(metapath, index_col='Tubelabel_species')\n",
    "meta"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 143,
   "id": "41d021ee-807c-4959-84b1-ed15a52f516a",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>Unnamed: 0</th>\n",
       "      <th>Health_Status</th>\n",
       "      <th>Starting_Weight</th>\n",
       "      <th>Extracted</th>\n",
       "      <th>Raw_ng_ul</th>\n",
       "      <th>Date_Enriched</th>\n",
       "      <th>Microbe_ng_ul</th>\n",
       "      <th>Micobe_Location</th>\n",
       "      <th>Microbe_clean_date/n</th>\n",
       "      <th>Host_ng_ul</th>\n",
       "      <th>Host_Location</th>\n",
       "      <th>Host_clean_y/n</th>\n",
       "      <th>Date_Libprep</th>\n",
       "      <th>Status</th>\n",
       "      <th>Notes</th>\n",
       "      <th>Seq_date</th>\n",
       "      <th>Host_Seq_date</th>\n",
       "      <th>Seq_Location (in Unity)</th>\n",
       "      <th>Sample_Code = datecollected_tagnumber_transect_samplenumber_species</th>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>Seq_file</th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>NaN</th>\n",
       "      <td>052022_T2_45_PSTR</td>\n",
       "      <td>Diseased_Margin</td>\n",
       "      <td>181</td>\n",
       "      <td>10_10_2023</td>\n",
       "      <td>22.8</td>\n",
       "      <td>10_17_2023</td>\n",
       "      <td>7.5</td>\n",
       "      <td>UML_F1_R2_B1</td>\n",
       "      <td>y</td>\n",
       "      <td>0.772</td>\n",
       "      <td>UML_F1_R2_B2</td>\n",
       "      <td>n</td>\n",
       "      <td>NaN</td>\n",
       "      <td>sequenced</td>\n",
       "      <td>NaN</td>\n",
       "      <td>10_19_23</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>NaN</th>\n",
       "      <td>052022_T2_46_PSTR</td>\n",
       "      <td>Diseased_Tissue</td>\n",
       "      <td>162</td>\n",
       "      <td>10_10_2023</td>\n",
       "      <td>63.9</td>\n",
       "      <td>10_17_2023</td>\n",
       "      <td>18.5</td>\n",
       "      <td>UML_F1_R2_B1</td>\n",
       "      <td>y</td>\n",
       "      <td>too low</td>\n",
       "      <td>UML_F1_R2_B2</td>\n",
       "      <td>n</td>\n",
       "      <td>NaN</td>\n",
       "      <td>sequenced</td>\n",
       "      <td>NaN</td>\n",
       "      <td>10_19_23</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>NaN</th>\n",
       "      <td>52022_BEL_CBC_T2_59_OFAV</td>\n",
       "      <td>Healthy</td>\n",
       "      <td>174</td>\n",
       "      <td>10_3_2023</td>\n",
       "      <td>73.7</td>\n",
       "      <td>10_16_2023</td>\n",
       "      <td>11.7</td>\n",
       "      <td>UML_F1_R2_B1</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>sequenced</td>\n",
       "      <td>NaN</td>\n",
       "      <td>10_19_23</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>NaN</th>\n",
       "      <td>52022_BEL_CBC_T2_69_OANN</td>\n",
       "      <td>Healthy</td>\n",
       "      <td>134</td>\n",
       "      <td>10_3_2023</td>\n",
       "      <td>133.1</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>NaN</th>\n",
       "      <td>52022_BEL_CBC_T2_72_OFAV</td>\n",
       "      <td>Healthy</td>\n",
       "      <td>75</td>\n",
       "      <td>10_3_2023</td>\n",
       "      <td>47.6</td>\n",
       "      <td>10_11_23</td>\n",
       "      <td>3.96</td>\n",
       "      <td>UML_F1_R2_B1</td>\n",
       "      <td>10_12_23</td>\n",
       "      <td>0.522</td>\n",
       "      <td>UML_F1_R2_B2</td>\n",
       "      <td>n</td>\n",
       "      <td>NaN</td>\n",
       "      <td>sequenced</td>\n",
       "      <td>NaN</td>\n",
       "      <td>10_19_23</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>Coral1.fastq</th>\n",
       "      <td>062019_355_T1_3_MCAV</td>\n",
       "      <td>Healthy</td>\n",
       "      <td>NaN</td>\n",
       "      <td>extracted</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>sequenced</td>\n",
       "      <td>NaN</td>\n",
       "      <td>sequenced</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>Coral2.fastq</th>\n",
       "      <td>062019_59_T2_5_MCAV</td>\n",
       "      <td>Healthy</td>\n",
       "      <td>NaN</td>\n",
       "      <td>extracted</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>sequenced</td>\n",
       "      <td>NaN</td>\n",
       "      <td>sequenced</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>T3-4-Past.fastq</th>\n",
       "      <td>062019_6_T3_4_PAST</td>\n",
       "      <td>Healthy</td>\n",
       "      <td>NaN</td>\n",
       "      <td>extracted</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>sequenced</td>\n",
       "      <td>NaN</td>\n",
       "      <td>sequenced</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>Coral4.fastq</th>\n",
       "      <td>062019_9_T3_6_MCAV</td>\n",
       "      <td>Healthy</td>\n",
       "      <td>NaN</td>\n",
       "      <td>extracted</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>sequenced</td>\n",
       "      <td>NaN</td>\n",
       "      <td>sequenced</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>_</th>\n",
       "      <td>102019_T2_35_PSTR</td>\n",
       "      <td>Healthy</td>\n",
       "      <td>_</td>\n",
       "      <td>NaN</td>\n",
       "      <td>_</td>\n",
       "      <td>_</td>\n",
       "      <td>_</td>\n",
       "      <td>_</td>\n",
       "      <td>_</td>\n",
       "      <td>_</td>\n",
       "      <td>_</td>\n",
       "      <td>_</td>\n",
       "      <td>_</td>\n",
       "      <td>_</td>\n",
       "      <td>_</td>\n",
       "      <td>_</td>\n",
       "      <td>_</td>\n",
       "      <td>_</td>\n",
       "      <td>_</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>104 rows × 19 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "                               Unnamed: 0   Health_Status  Starting_Weight  \\\n",
       "Seq_file                                                                     \n",
       "NaN                     052022_T2_45_PSTR  Diseased_Margin             181   \n",
       "NaN                     052022_T2_46_PSTR  Diseased_Tissue             162   \n",
       "NaN              52022_BEL_CBC_T2_59_OFAV          Healthy             174   \n",
       "NaN              52022_BEL_CBC_T2_69_OANN          Healthy             134   \n",
       "NaN              52022_BEL_CBC_T2_72_OFAV          Healthy              75   \n",
       "...                                   ...              ...             ...   \n",
       "Coral1.fastq         062019_355_T1_3_MCAV          Healthy             NaN   \n",
       "Coral2.fastq          062019_59_T2_5_MCAV          Healthy             NaN   \n",
       "T3-4-Past.fastq        062019_6_T3_4_PAST          Healthy             NaN   \n",
       "Coral4.fastq           062019_9_T3_6_MCAV          Healthy             NaN   \n",
       "_                       102019_T2_35_PSTR          Healthy               _   \n",
       "\n",
       "                  Extracted Raw_ng_ul Date_Enriched Microbe_ng_ul  \\\n",
       "Seq_file                                                            \n",
       "NaN              10_10_2023      22.8    10_17_2023           7.5   \n",
       "NaN              10_10_2023      63.9    10_17_2023          18.5   \n",
       "NaN               10_3_2023      73.7    10_16_2023          11.7   \n",
       "NaN               10_3_2023     133.1           NaN           NaN   \n",
       "NaN               10_3_2023      47.6      10_11_23          3.96   \n",
       "...                     ...       ...           ...           ...   \n",
       "Coral1.fastq      extracted       NaN           NaN           NaN   \n",
       "Coral2.fastq      extracted       NaN           NaN           NaN   \n",
       "T3-4-Past.fastq   extracted       NaN           NaN           NaN   \n",
       "Coral4.fastq      extracted       NaN           NaN           NaN   \n",
       "_                       NaN         _             _             _   \n",
       "\n",
       "                Micobe_Location Microbe_clean_date/n Host_ng_ul Host_Location  \\\n",
       "Seq_file                                                                        \n",
       "NaN                UML_F1_R2_B1                    y      0.772  UML_F1_R2_B2   \n",
       "NaN                UML_F1_R2_B1                    y    too low  UML_F1_R2_B2   \n",
       "NaN                UML_F1_R2_B1                  NaN        NaN           NaN   \n",
       "NaN                         NaN                  NaN        NaN           NaN   \n",
       "NaN                UML_F1_R2_B1             10_12_23      0.522  UML_F1_R2_B2   \n",
       "...                         ...                  ...        ...           ...   \n",
       "Coral1.fastq                NaN                  NaN        NaN           NaN   \n",
       "Coral2.fastq                NaN                  NaN        NaN           NaN   \n",
       "T3-4-Past.fastq             NaN                  NaN        NaN           NaN   \n",
       "Coral4.fastq                NaN                  NaN        NaN           NaN   \n",
       "_                             _                    _          _             _   \n",
       "\n",
       "                Host_clean_y/n Date_Libprep    Status  Notes    Seq_date  \\\n",
       "Seq_file                                                                   \n",
       "NaN                          n          NaN  sequenced    NaN   10_19_23   \n",
       "NaN                          n          NaN  sequenced    NaN   10_19_23   \n",
       "NaN                        NaN          NaN  sequenced    NaN   10_19_23   \n",
       "NaN                        NaN          NaN        NaN    NaN        NaN   \n",
       "NaN                          n          NaN  sequenced    NaN   10_19_23   \n",
       "...                        ...          ...        ...    ...        ...   \n",
       "Coral1.fastq               NaN          NaN  sequenced    NaN  sequenced   \n",
       "Coral2.fastq               NaN          NaN  sequenced    NaN  sequenced   \n",
       "T3-4-Past.fastq            NaN          NaN  sequenced    NaN  sequenced   \n",
       "Coral4.fastq               NaN          NaN  sequenced    NaN  sequenced   \n",
       "_                            _            _          _      _          _   \n",
       "\n",
       "                Host_Seq_date Seq_Location (in Unity)  \\\n",
       "Seq_file                                                \n",
       "NaN                       NaN                     NaN   \n",
       "NaN                       NaN                     NaN   \n",
       "NaN                       NaN                     NaN   \n",
       "NaN                       NaN                     NaN   \n",
       "NaN                       NaN                     NaN   \n",
       "...                       ...                     ...   \n",
       "Coral1.fastq              NaN                     NaN   \n",
       "Coral2.fastq              NaN                     NaN   \n",
       "T3-4-Past.fastq           NaN                     NaN   \n",
       "Coral4.fastq              NaN                     NaN   \n",
       "_                           _                       _   \n",
       "\n",
       "                Sample_Code = datecollected_tagnumber_transect_samplenumber_species  \n",
       "Seq_file                                                                             \n",
       "NaN                                                            NaN                   \n",
       "NaN                                                            NaN                   \n",
       "NaN                                                            NaN                   \n",
       "NaN                                                            NaN                   \n",
       "NaN                                                            NaN                   \n",
       "...                                                            ...                   \n",
       "Coral1.fastq                                                   NaN                   \n",
       "Coral2.fastq                                                   NaN                   \n",
       "T3-4-Past.fastq                                                NaN                   \n",
       "Coral4.fastq                                                   NaN                   \n",
       "_                                                                _                   \n",
       "\n",
       "[104 rows x 19 columns]"
      ]
     },
     "execution_count": 143,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "# Load in CBC Metagenomics metadata too \n",
    "meta2path = '//project/pi_sarah_gignouxwolfsohn_uml_edu/brooke/metadata/Metagenomics_Tracker_Belize.csv'\n",
    "meta2=pd.read_csv(meta2path, index_col='Seq_file')\n",
    "meta2"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 169,
   "id": "ae13bb25-01c3-45a6-ab5c-daaaf19501f5",
   "metadata": {},
   "outputs": [
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "/home/brooke_sienkiewicz_student_uml_edu/.conda/envs/skbio/lib/python3.8/site-packages/skbio/stats/ordination/_principal_coordinate_analysis.py:143: RuntimeWarning: The result contains negative eigenvalues. Please compare their magnitude with the magnitude of some of the largest positive eigenvalues. If the negative ones are smaller, it's probably safe to ignore them, but if they are large in magnitude, the results won't be useful. See the Notes section for more details. The smallest eigenvalue is -0.015530436450687855 and the largest is 0.754563580867275.\n",
      "  warn(\n"
     ]
    }
   ],
   "source": [
    "from skbio.stats.ordination import pcoa\n",
    "\n",
    "pcoa_results = pcoa(distance_matrix)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 171,
   "id": "69374fb8-4f5b-4dd6-b47b-65fd92907772",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "Index(['Unnamed: 0', 'Health_Status ', 'Starting_Weight', 'Extracted',\n",
       "       'Raw_ng_ul', 'Date_Enriched', 'Microbe_ng_ul', 'Micobe_Location',\n",
       "       'Microbe_clean_date/n', 'Host_ng_ul', 'Host_Location', 'Host_clean_y/n',\n",
       "       'Date_Libprep', 'Status ', 'Notes ', 'Seq_date', 'Host_Seq_date',\n",
       "       'Seq_Location (in Unity)',\n",
       "       'Sample_Code = datecollected_tagnumber_transect_samplenumber_species'],\n",
       "      dtype='object')"
      ]
     },
     "execution_count": 171,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "meta2.columns"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 173,
   "id": "7edd422f-e6c6-4dad-9d6c-0f6b56b9bf84",
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Duplicate labels: Index([nan, nan, nan, nan, nan, nan, nan, nan, nan, nan, nan, nan, nan, nan,\n",
      "       nan, nan, nan, nan, nan, nan, nan, nan, nan, nan, nan, nan, nan, nan,\n",
      "       nan, nan, nan, nan, nan, nan, nan, nan, nan, nan, nan, nan, nan, nan,\n",
      "       nan, nan, nan, nan, nan, nan, nan, nan, nan, nan, nan, nan, nan, nan,\n",
      "       nan, nan, nan, nan, nan, nan, nan, nan, nan, nan, nan, nan, nan, nan,\n",
      "       nan, nan, nan],\n",
      "      dtype='object', name='Seq_file')\n"
     ]
    }
   ],
   "source": [
    "# Check for duplicate index labels\n",
    "duplicate_labels = meta2.index[meta2.index.duplicated()]\n",
    "print(\"Duplicate labels:\", duplicate_labels)\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 175,
   "id": "4e546d34-7a65-44a7-9a8d-f6b6f8b2c300",
   "metadata": {},
   "outputs": [],
   "source": [
    "# Remove duplicate index labels\n",
    "meta_mcav = meta2[~meta2.index.duplicated()]"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 184,
   "id": "ccec670a-d5f4-40c8-a105-196904825cf9",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "Seq_file\n",
       "NaN                 Diseased_Margin\n",
       "T3_8_2019_MCAV              Healthy\n",
       "T2_16_2019_MCAV             Healthy\n",
       "T1_20_2019_MCAV             Healthy\n",
       "T1_12_2022_MCAV              Margin\n",
       "T1_24_2019_MCAV             Healthy\n",
       "T3_9_2019_MCAV              Healthy\n",
       "T3_14_2019_MCAV             Healthy\n",
       "T3_1_2019_MCAV              Healthy\n",
       "T3_15_2019_MCAV             Healthy\n",
       "T1_16_2019_MCAV             Healthy\n",
       "T1_40_2022_MCAV              Margin\n",
       "T1_70_2022_MCAV            Diseased\n",
       "T3_48_2022_MCAV              Margin\n",
       "T3_49_2022_MCAV            Diseased\n",
       "T3_51_2022_MCAV            Diseased\n",
       "T3_40_2022_MCAV              Margin\n",
       "T3_60_2022_MCAV            Diseased\n",
       "T1_13_2022_MCAV            Diseased\n",
       "T3_13_2022_MCAV             Healthy\n",
       "T3_1_2022_MCAV              Healthy\n",
       "T3_19_2022_MCAV             Healthy\n",
       "T2_10_2022_MCAV             Healthy\n",
       "T1_57_2022_MCAV             Healthy\n",
       "Coral5.fastq                Healthy\n",
       "T3-21-Mmea.fastq            Healthy\n",
       "Coral1.fastq                Healthy\n",
       "Coral2.fastq                Healthy\n",
       "T3-4-Past.fastq             Healthy\n",
       "Coral4.fastq                Healthy\n",
       "_                           Healthy\n",
       "Name: Health_Status , dtype: object"
      ]
     },
     "execution_count": 184,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "meta_mcav['Health_Status ']"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 185,
   "id": "7fab59e4-a15e-433f-922f-86d7d44e92a8",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "Seq_file\n",
       "NaN                 Diseased_Margin\n",
       "T3_8_2019                   Healthy\n",
       "T2_16_2019                  Healthy\n",
       "T1_20_2019                  Healthy\n",
       "T1_12_2022                   Margin\n",
       "T1_24_2019                  Healthy\n",
       "T3_9_2019                   Healthy\n",
       "T3_14_2019                  Healthy\n",
       "T3_1_2019                   Healthy\n",
       "T3_15_2019                  Healthy\n",
       "T1_16_2019                  Healthy\n",
       "T1_40_2022                   Margin\n",
       "T1_70_2022                 Diseased\n",
       "T3_48_2022                   Margin\n",
       "T3_49_2022                 Diseased\n",
       "T3_51_2022                 Diseased\n",
       "T3_40_2022                   Margin\n",
       "T3_60_2022                 Diseased\n",
       "T1_13_2022                 Diseased\n",
       "T3_13_2022                  Healthy\n",
       "T3_1_2022                   Healthy\n",
       "T3_19_2022                  Healthy\n",
       "T2_10_2022                  Healthy\n",
       "T1_57_2022                  Healthy\n",
       "Coral5.fastq                Healthy\n",
       "T3-21-Mmea.fastq            Healthy\n",
       "Coral1.fastq                Healthy\n",
       "Coral2.fastq                Healthy\n",
       "T3-4-Past.fastq             Healthy\n",
       "Coral4.fastq                Healthy\n",
       "_                           Healthy\n",
       "Name: Health_Status , dtype: object"
      ]
     },
     "execution_count": 185,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "meta_mcav['Health_Status ']\n",
    "# Remove \"_MCAV\" in sampleid in meta df\n",
    "meta_mcav.index = meta_mcav.index.str.replace('_MCAV', '')\n",
    "\n",
    "# Display the modified DataFrame\n",
    "meta_mcav['Health_Status ']\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "61d284a3-a8b8-428b-a271-68668e00ce95",
   "metadata": {},
   "outputs": [],
   "source": [
    "# Assuming `pcoa_results` contains your ordination results\n",
    "ids_in_ordination = set(pcoa_results.samples.index)\n",
    "ids_in_dataframe = set(meta_mcav.index)  \n",
    "\n",
    "# Check for any IDs in the ordination results that are not in the data frame\n",
    "missing_ids = ids_in_ordination - ids_in_dataframe\n",
    "\n",
    "if missing_ids:\n",
    "    print(f\"Warning: IDs {missing_ids} are present in the ordination results but not in the data frame.\")\n",
    "\n",
    "# Continue with plotting code\n",
    "fig = pcoa_results.plot(df=meta_mcav,\n",
    "                        column='Health_Status ',\n",
    "                        axis_labels=['PC 1', 'PC 2', 'PC 3'],\n",
    "                        title='Samples colored by health status',\n",
    "                        cmap='Reds',\n",
    "                        s=35)\n",
    "# was getting error that sampleids didnt match df and pcoa results..."
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 188,
   "id": "95ba6e08-32d5-47f4-af99-62e33d23cbb5",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "Index([               nan,       'T3_8_2019 ',      'T2_16_2019 ',\n",
       "            'T1_20_2019 ',      'T1_12_2022 ',      'T1_24_2019 ',\n",
       "             'T3_9_2019 ',      'T3_14_2019 ',       'T3_1_2019 ',\n",
       "            'T3_15_2019 ',      'T1_16_2019 ',      'T1_40_2022 ',\n",
       "            'T1_70_2022 ',      'T3_48_2022 ',      'T3_49_2022 ',\n",
       "            'T3_51_2022 ',      'T3_40_2022 ',      'T3_60_2022 ',\n",
       "            'T1_13_2022 ',      'T3_13_2022 ',       'T3_1_2022 ',\n",
       "            'T3_19_2022 ',      'T2_10_2022 ',       'T1_57_2022',\n",
       "           'Coral5.fastq', 'T3-21-Mmea.fastq',     'Coral1.fastq',\n",
       "           'Coral2.fastq',  'T3-4-Past.fastq',     'Coral4.fastq',\n",
       "                      '_'],\n",
       "      dtype='object', name='Seq_file')"
      ]
     },
     "execution_count": 188,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "# check sample ids in meta df\n",
    "meta_mcav.index"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 190,
   "id": "a57d4e01-57a8-4d82-8997-f1e1042fc397",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "Index([               nan,        'T3_8_2019',       'T2_16_2019',\n",
       "             'T1_20_2019',       'T1_12_2022',       'T1_24_2019',\n",
       "              'T3_9_2019',       'T3_14_2019',        'T3_1_2019',\n",
       "             'T3_15_2019',       'T1_16_2019',       'T1_40_2022',\n",
       "             'T1_70_2022',       'T3_48_2022',       'T3_49_2022',\n",
       "             'T3_51_2022',       'T3_40_2022',       'T3_60_2022',\n",
       "             'T1_13_2022',       'T3_13_2022',        'T3_1_2022',\n",
       "             'T3_19_2022',       'T2_10_2022',       'T1_57_2022',\n",
       "           'Coral5.fastq', 'T3-21-Mmea.fastq',     'Coral1.fastq',\n",
       "           'Coral2.fastq',  'T3-4-Past.fastq',     'Coral4.fastq',\n",
       "                      '_'],\n",
       "      dtype='object', name='Seq_file')"
      ]
     },
     "execution_count": 190,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "# Strip spaces from indices in meta_mcav\n",
    "meta_mcav.index = meta_mcav.index.str.strip()\n",
    "# Check one more time\n",
    "meta_mcav.index"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 191,
   "id": "3067f9fb-33be-40e6-a11e-ce8ed04c599d",
   "metadata": {},
   "outputs": [
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "/home/brooke_sienkiewicz_student_uml_edu/.conda/envs/skbio/lib/python3.8/site-packages/skbio/stats/ordination/_ordination_results.py:285: UserWarning: Tight layout not applied. The left and right margins cannot be made large enough to accommodate all axes decorations. \n",
      "  fig.tight_layout()\n"
     ]
    },
    {
     "data": {
      "image/png": "iVBORw0KGgoAAAANSUhEUgAAAgkAAAGZCAYAAAAD7KImAAAAOXRFWHRTb2Z0d2FyZQBNYXRwbG90bGliIHZlcnNpb24zLjQuMywgaHR0cHM6Ly9tYXRwbG90bGliLm9yZy/MnkTPAAAACXBIWXMAAA9hAAAPYQGoP6dpAADdnElEQVR4nOydd3xj1Z32n6tqyVZ1t8d9PLanF0+x6Z0lkNACgZBQlhCSkARIdknIuwECG0ICCwQCJJtQkoVAIITNEgKEMmRgBpgZ9+5x70WSiySr3vP+4bkXSVbXVbHnfD+f2SWydM7RlXTPc36VIYQQUCgUCoVCofggSvYCKBQKhUKhpCZUJFAoFAqFQvELFQkUCoVCoVD8QkUChUKhUCgUv1CRQKFQKBQKxS9UJFAoFAqFQvELFQkUCoVCoVD8QkUChUKhUCgUv1CRQKFQKBQKxS8nhEj45JNPcMkll6C4uBhyuRy5ubmoq6vD9773vWQvLSTXXXcdSktLk72MqHn22WfBMAwGBweTvRQvSktLcd111wV9zuDgIBiGwYMPPhjXtXDX6MiRI3Gdxx/79+8HwzDYv38//9gbb7yBu+++2+/zGYbBLbfcErf1HDx4EHfffTfm5uaiHiPY+ikUSmSseZHwt7/9DfX19VhYWMDPf/5zvP3223j00Udx0kkn4aWXXkr28iiUlOONN97APffck5S5Dx48iHvuuSdmkZCs9VMoaw1JshcQb37+85+jrKwMb731FiSSz97ul770Jfz85z9P4soo0WC1WqFUKpO9DAqFQjkhWPOWBIPBgKysLC+BwCESeb/9l156Ceeeey7y8/OhUChQU1ODH/zgB7BYLF7Pu+6665CRkYGuri6cd955SE9PR35+Pn72s58BAD7++GOcfPLJSE9Px4YNG/Dcc895vZ4zL//jH//A9ddfD71ej/T0dFx00UXo7+8P+Z4IIXjiiSewfft2KBQK6HQ6XH755Ste29jYiAsvvBA5OTmQy+UoKCjA5z73OYyOjoac480338RZZ50FjUYDpVKJmpoa3H///V7P+etf/4q6ujoolUqoVCqcc845OHToUMixAeDpp5/Gtm3bkJaWBr1ej0suuQSdnZ1ez+Guc2trK84991yoVCqcddZZAACHw4H77rsP1dXVkMvlyM7OxvXXX4+ZmRmvMZxOJ/793/8deXl5UCqVOPnkk/Hpp5+GtUYOlmXxn//5nyguLkZaWhpqa2vx7rvv8n8/cOAAGIbBH//4xxWv/f3vfw+GYXD48OGQ8ywuLuIb3/gGsrKykJmZiUsvvRTj4+MrnvfSSy+hrq4O6enpyMjIwHnnnYfGxkav5xw5cgRf+tKXUFpaCoVCgdLSUlx11VUYGhoKuobrrrsOv/rVrwAsuxa4f77uoj/84Q+oqamBUqnEtm3b8Prrr4d8fyzL4r777kNVVRUUCgW0Wi22bt2KRx99FABw991349/+7d8AAGVlZfzcnCsknN9nsPVz7qNnn312xdoYhvFyUczMzOCmm25CUVER//066aST8M4774R8nxTKmoKscW688UYCgHz7298mH3/8MXE4HAGfe++995KHH36Y/O1vfyP79+8nTz31FCkrKyNnnHGG1/OuvfZaIpPJSE1NDXn00UfJP/7xD3L99dcTAOSHP/wh2bBhA/nd735H3nrrLXLhhRcSAOTIkSP865955hkCgBQVFZEbbriB/P3vfye/+c1vSE5ODikqKiImk8lrrpKSEq/5v/a1rxGpVEq+973vkTfffJO88MILpLq6muTm5pLJyUlCCCFms5lkZmaS2tpa8qc//Yl88MEH5KWXXiI333wz6ejoCHrNfvvb3xKGYcjpp59OXnjhBfLOO++QJ554gnzzm9/kn/P8888TAOTcc88lr732GnnppZfIrl27iEwmIwcOHFjxXgcGBvjHfvrTnxIA5KqrriJ/+9vfyO9//3tSXl5ONBoN6enp8XrvUqmUlJaWkvvvv5+8++675K233iJut5ucf/75JD09ndxzzz3kH//4B/ntb39LCgsLycaNG4nVavUag2EY8m//9m/k7bffJv/1X/9FCgsLiVqtJtdee23Q6zAwMMB/TieffDL585//TF5++WWye/duIpVKycGDB/nn7tixg5x00kkrxti9ezfZvXt30Hm4a1ReXk6+/e1vk7feeov89re/JTqdbsV37z//8z8JwzDkhhtuIK+//jp59dVXSV1dHUlPTyft7e38815++WXy4x//mPzlL38hH3zwAXnxxRfJaaedRrKzs8nMzAz/vPfff58AIO+//z4hhJBjx46Ryy+/nAAghw4d4v/ZbDZCCCEASGlpKdmzZw/505/+RN544w1y+umnE4lEQvr6+oK+z/vvv5+IxWJy1113kXfffZe8+eab5JFHHiF33303IYSQkZER8u1vf5sAIK+++io/9/z8PCEkvN9nsPVzn+czzzyzYm0AyF133cX/7/POO49kZ2eT3/zmN2T//v3ktddeIz/+8Y/Jiy++GPQ9UihrjTUvEmZnZ8nJJ59MABAARCqVkvr6enL//feTxcXFgK9jWZY4nU7ywQcfEACkubmZ/9u1115LAJA///nP/GNOp5NkZ2cTAKShoYF/3GAwELFYTG6//Xb+MW5TuOSSS7zm/OijjwgAct9993nN5SkSDh06RACQhx56yOu1IyMjRKFQkH//938nhBBy5MgRAoC89tprYV6pZRYXF4larSYnn3wyYVnW73PcbjcpKCggW7ZsIW632+u1OTk5pL6+fsV75USCyWQiCoWCXHDBBV5jDg8PE7lcTq6++mqv9w6APP30017P/eMf/7ji+hNCyOHDhwkA8sQTTxBCCOns7CQAyG233eb1PE7ghCsSCgoKyNLSEv/4wsIC0ev15Oyzz17xPhsbG/nHPv30UwKAPPfcc0Hn4V7rKcIIIeTnP/85AUAmJiYIIcvXSCKRkG9/+9tez1tcXCR5eXnkiiuuCDiHy+UiZrOZpKenk0cffZR/3FckEELIt771LRLo/ACA5ObmkoWFBf6xyclJIhKJyP333x/0fV544YVk+/btQZ/zi1/8YoWo9Eew32eg9UciEjIyMsitt94adA0UyonAmnc3ZGZm4sCBAzh8+DB+9rOf4Qtf+AJ6enrwwx/+EFu2bMHs7Cz/3P7+flx99dXIy8uDWCyGVCrFaaedBgArTOEMw+CCCy7g/7dEIsH69euRn5+PHTt28I/r9Xrk5OT4NfN++ctf9vrf9fX1KCkpwfvvvx/w/bz++utgGAbXXHMNXC4X/y8vLw/btm3jTbPr16+HTqfDHXfcgaeeegodHR1hXa+DBw9iYWEB3/zmN8EwjN/ndHd3Y3x8HF/5yle8XDYZGRm47LLL8PHHH8Nqtfp97aFDh7C0tLQis6CoqAhnnnmmlxmf47LLLltxDbRaLS666CKva7B9+3bk5eXx14C7jr7X+YorrvDrfgrEpZdeirS0NP5/q1QqXHTRRfjnP/8Jt9sNALjqqquQk5PDm7oB4LHHHkN2djauvPLKsOb5/Oc/7/W/t27dCgD8d+ett96Cy+XCV7/6Va/3nZaWhtNOO80rQ8FsNuOOO+7A+vXrIZFIIJFIkJGRAYvFsuK7HClnnHEGVCoV/79zc3MDfsc92bNnD5qbm/HNb34Tb731FhYWFiKaN5LfZ6zs2bMHzz77LO677z58/PHHcDqdgo5PoawW1rxI4KitrcUdd9yBl19+GePj47jtttswODjIBy+azWaccsop+OSTT3Dfffdh//79OHz4MF599VUAwNLSktd4SqXSa+MAAJlMBr1ev2JumUwGm8224vG8vDy/jxkMhoDvY2pqCoQQ5ObmQiqVev37+OOPedGj0WjwwQcfYPv27bjzzjuxadMmFBQU4K677gp6w+N8+uvWrQv4HG59+fn5K/5WUFAAlmVhMpmieq3ve1cqlVCr1V6PTU1NYW5uDjKZbMU1mJyc5K8BN5bvdZZIJMjMzAz4/nwJ9Dk5HA6YzWYAgFwux9e//nW88MILmJubw8zMDP70pz/hxhtvhFwuD2se3zVxr+O+e1NTUwCA3bt3r3jfL730kpfgvfrqq/H444/jxhtvxFtvvYVPP/0Uhw8fRnZ29orvcqT4u3ZyuTzkuD/84Q/x4IMP4uOPP8a//Mu/IDMzE2eddVZYqZ+R/j5j5aWXXsK1116L3/72t6irq4Ner8dXv/pVTE5OCjoPhZLqrPnsBn9IpVLcddddePjhh9HW1gYAeO+99zA+Po79+/fzpxMAMaVihcLfDWdychLr168P+JqsrCwwDIMDBw743Xw8H9uyZQtefPFFEELQ0tKCZ599Fj/5yU+gUCjwgx/8wO/42dnZABA0uJHbJCYmJlb8bXx8HCKRCDqdLqrXZmVleT3mz5rBBfa9+eabfufgTrncXJOTkygsLOT/7nK5ggoxXwJ9TjKZDBkZGfxj3/jGN/Czn/0MTz/9NGw2G1wuF26++eaw5wkFd21eeeUVlJSUBHze/Pw8Xn/9ddx1111en7PdbofRaBRsPZEikUhw++234/bbb8fc3Bzeeecd3HnnnTjvvPMwMjISNGtFiN8nJ+rtdrvX4/6+C1lZWXjkkUfwyCOPYHh4GH/961/xgx/8ANPT0wG/dxTKWmTNWxL8bUbAZ+bJgoICAJ9tRr4b769//eu4re3555/3+t8HDx7E0NAQTj/99ICvufDCC0EIwdjYGGpra1f827Jly4rXMAyDbdu24eGHH4ZWq0VDQ0PA8evr66HRaPDUU0+BEOL3OVVVVSgsLMQLL7zg9RyLxYI///nPfMaDP+rq6qBQKPA///M/Xo+Pjo7ivffe47MXgnHhhRfCYDDA7Xb7vQZVVVUAwF9H3+v8pz/9CS6XK+Q8HK+++qqXJWhxcRH/93//h1NOOQVisZh/PD8/H1/84hfxxBNP4KmnnsJFF12E4uLisOcJxXnnnQeJRIK+vj6/77u2thbA8udNCFnxXf7tb3/Lu0eC4WvBiAdarRaXX345vvWtb8FoNPLZE4HmjuT3GWiM3NxcpKWloaWlxevx//3f/w261uLiYtxyyy0455xzgv52KJS1yJq3JJx33nlYt24dLrroIlRXV4NlWTQ1NeGhhx5CRkYGvvvd7wJY3hx1Oh1uvvlm3HXXXZBKpXj++efR3Nwct7UdOXIEN954I774xS9iZGQEP/rRj1BYWIhvfvObAV9z0kkn4aabbsL111+PI0eO4NRTT0V6ejomJibw4YcfYsuWLfjGN76B119/HU888QQuvvhilJeXgxCCV199FXNzczjnnHMCjp+RkYGHHnoIN954I84++2x87WtfQ25uLo4dO4bm5mY8/vjjEIlE+PnPf44vf/nLuPDCC/H1r38ddrsdv/jFLzA3N8engvpDq9XiP/7jP3DnnXfiq1/9Kq666ioYDAbcc889SEtLw1133RXyun3pS1/C888/jwsuuADf/e53sWfPHkilUoyOjuL999/HF77wBVxyySWoqanBNddcg0ceeQRSqRRnn3022tra8OCDD65wYQRDLBbjnHPOwe233w6WZfHAAw9gYWHBb8Ge7373u9i7dy8A4Jlnngl7jnAoLS3FT37yE/zoRz9Cf38/zj//fOh0OkxNTeHTTz9Feno67rnnHqjVapx66qn4xS9+gaysLJSWluKDDz7A7373O2i12pDzcELzgQcewL/8y79ALBZj69atkMlkMa3/oosuwubNm1FbW4vs7GwMDQ3hkUceQUlJCSorK73mfvTRR3HttddCKpWiqqoqot9nsPVfc801ePrpp1FRUYFt27bh008/xQsvvOD1+vn5eZxxxhm4+uqrUV1dDZVKhcOHD+PNN9/EpZdeGtM1oFBWHUkMmkwIL730Ern66qtJZWUlycjIIFKplBQXF5OvfOUrK1IBDx48SOrq6ohSqSTZ2dnkxhtvJA0NDSsioq+99lqSnp6+Yq7TTjuNbNq0acXjJSUl5HOf+xz/v7lo9rfffpt85StfIVqtlo/47+3t9XqtvxRIQgh5+umnyd69e0l6ejpRKBSkoqKCfPWrX+VTLbu6ushVV11FKioqiEKhIBqNhuzZs4c8++yzYV23N954g5x22mkkPT2dKJVKsnHjRvLAAw94Pee1114je/fuJWlpaSQ9PZ2cddZZ5KOPPvJ6jr8USEKW0yy3bt1KZDIZ0Wg05Atf+IJXCh/33v1dZ0KWs0kefPBBsm3bNpKWlkYyMjJIdXU1+frXv+51De12O/ne975HcnJySFpaGtm3bx85dOgQKSkpCTu74YEHHiD33HMPWbduHZHJZGTHjh3krbfeCvi60tJSUlNTE3RsT7hrdPjwYa/H/WUeELJ83c844wyiVquJXC4nJSUl5PLLLyfvvPMO/5zR0VFy2WWXEZ1OR1QqFTn//PNJW1vbivftbw673U5uvPFGkp2dTRiG8fr8AJBvfetbK95DONfzoYceIvX19SQrK4vIZDJSXFxM/vVf/5UMDg56Pe+HP/whKSgoICKRyGtt4f4+g61/fn6e3HjjjSQ3N5ekp6eTiy66iAwODnplN9hsNnLzzTeTrVu3ErVaTRQKBamqqiJ33XUXsVgsQd8jhbLWYAgJYFOmxI1nn30W119/PQ4fPsybiClrg5aWFmzbtg2/+tWvglqEKBQKZTWw5t0NFEoi6Ovrw9DQEO68807k5+eHbB5FoVAoq4E1H7hIoSSCe++9F+eccw7MZjNefvll2l+CQqGsCai7gUKhUCgUil+oJYFCoVAoFIpfqEigUCgUCoXiFyoSKBQKhUKh+IWKBAqFQqFQKH6hIoFCoVAoFIpfqEigUCgUCoXiFyoSKBQKhUKh+IWKBAqFQqFQKH6hIoFCoVAoFIpfqEigUCgUCoXiFyoSKBQKhUKh+IWKBAqFQqFQKH6hIoFCoVAoFIpfqEigUCgUCoXiFyoSKBQKhUKh+IWKBAqFQqFQKH6hIoFCoVAoFIpfqEigUCgUCoXiFyoSKBQKhUKh+IWKBAqFQqFQKH6hIoFCoVAoFIpfqEigUCgUCoXiFyoSKBQKhUKh+IWKBAqFQqFQKH6hIoFCoVAoFIpfqEigUCgUCoXiFyoSKBQKhUKh+IWKBAqFQqFQKH6hIoFCoVAoFIpfqEigUCgUCoXiFyoSKBQKhUKh+EWS7AVQTjxYloXT6YRIJIJYLAbDMGAYJtnLolAoFIoPVCRQEgYhBG63G06nE0tLS7w4EIvFkEqlEIvFEIvFEImogYtCoVBSAYYQQpK9CMrahxACp9MJt9vN/zfDMCCEgGVZEEKoaKBQKJQUg1oSKHGFEwFOp5MXAhycKOBEACEEhBC4XC44nU4AgEgkgkQigUQioaKBQqFQEgy1JFDiBrfhu1wuAJ+JAk40cI8Fe70/SwMVDRQKhZIYqEigxAVOCLAsCwBeGznLsnA4HBEHLHoKBg5f0SCRSGgQJIVCoQgEFQkUQSGEwOFwwGw2Q6FQQCQSrdi0oxUJ/ubiRIOnpQEAZDIZZDIZnz1BoVAolMihdlqKYHABibOzszh69KhfgSAkXJCjRCLhAx0B4NChQ5iZmcHi4iIWFhZgNptht9vhcrlANTGFQqGEDw1cpAgC515wu90QiUQrghQTAScaAPDxCty6OMuFv0BIammgUCgU/1CRQIkJz9oHhJC4Ww8iwVM0cBYEztVht9upaKBQKJQQUJFAiRrP2gcAeIHA1T8IRLw3YX/jc49R0UChUCjhQ0UCJSo46wHLsiusB6FEAveceBLu/J6igftnt9vhcDgA+K/TQEUDhUI5UaAigRIRvrUP/LkXwhEJqYZnZoRYLF4hGjwtDVyQpEQiSSn3CoVCoQgNFQmUsPGtfRAohTHZIkGITTuYaLDZbF7zSCQSyOVyvrATFQ0UCmWtQEUCJSSepZX9uRd8SbZIAEK7GyIlkGgwGo3o7e1FbW2tV0om556gooFCoaxmqEigBCVQcGIwUkEkxBvfvhMSiYQXU5ylQSQSrYhpoKKBQqGsJqhIoATEt/ZBuJtbskVCMuozBLI0sCwLu90Om81GRQOFQll1UJFAWQFX+8DlcoXlXvCFEwnJKKjEkWxLhm+8Bnc93G433G53wEDIWEtVUygUipBQkUDxIhr3gi/J3uSSPb8/ArXF5sQY93ffmAYqGigUSjKhIoHCE6z2QSRwr0umJSFRRGuxCCQaXC4XnE5nQNFA22JTKJREQkUCxav2gRCllT1FQrJItrshUiIRDZx7gooGCoUSb+gd5gSHC048dOgQFhcXBQmkS7ZIWAvWC88S0ZwoYBgGRqMR//znP706XNpsNq/6FRQKhSIU1JJwguJZ+4AQAqvVCpZlBStExM0R7Dmr7bQfiESIEs8MCkIIn3Lp2eGSYZgVJaSppYFCocQCFQknIL7BieE0ZYqEcETC0tISZDIZ3ztBaNaKAAmEP/cEJ/qcTif/HE/RwGVPUCgUSrhQkXCCEaj2gUgkSohIYFkWXV1dGB4ehlgshkajgU6ng16vR0ZGhqCWjLVIoGBQz7bY3PO4z9qfpcEze4JCoVACQUXCCUKo2gcMwwjm0w4kEiwWC5qbmwEAdXV1cLlcmJubg8lkwuDgIBiGgU6n4/8plcqU38QSbbEIN2MkHNEgEolWBEKm+vWmUCiJhYqEE4Bwah/E25IwMTGB9vZ2FBYWYsOGDXC73SCEQKVSoaioCCzLwmw2w2g0YmZmBseOHYNEIvESDQqFIuw1rGV3QzQbebiigbbFplAonlCRsMZhWRYOhyNk7QOhAwm58dxuN7q6ujA5OYktW7YgNzeXf9wTkUgEtVoNtVqN0tJSsCyL+fl5mEwmTExMoLu7G3K5nHdN6HQ6yGSygHOvVYQUcpxo4Mbkviue1SCpaKBQTmyoSFijcBsxl70QTudGIVPoGIaBxWJBY2MjRCIR6uvrI7IEiEQi3oIAAC6XixcNQ0NDaG9vR3p6Ov8crVYLqVQq2PojIZEbZzwKVHn2nODmAD4TDW1tbcjNzYVOp6OigUI5waAiYQ0STWllId0N3BqamppQVFSEDRs2xJyKJ5FIkJmZiczMTACA0+mEyWSCyWRCX18frFYrVCoV9Ho9XC7XCkvFWiLeG7OvaFhcXER2djYIIbDb7XA4HABALQ0UygkAFQlrCN/aB5HU/RfKkuB2u9HZ2QlCCDZs2ICSkpKYx/SHVCpFTk4OcnJyAAB2u50XDTabDZ2dnRgfH+ctDWq1ek3UDEhGrAVnieKEAFcN0p9o8GxWRTtcUiirHyoS1giepZWBlV0IQyGEJcFsNqOpqYk/XXKugmBrFmoTkcvlyMvLQ15eHsxmM/Ly8iASiWAymTA6OgqWZaHRaPh4BqHSLVM1uyGecwZqi00Igc1m45/DiQbaFptCWb1QkbAG4KwHnCUgmhNzrIGL4+PjaG9vR3FxMSorK7F///6klmWWyWTIzc1FYWEhCCGwWCy8pWFgYGBVpltyJEMkBPtOUdFAoaxdqEhYxXi6F2Lt3CgSiaJyN7jdbnR0dGB6ehrbt29HdnY2gNQqu8wwDDIyMpCRkRGXdMtEkoxrGmm57nBFg78Ol1Q0UCipBRUJq5RoghODEc2mzrkXpFIpTjrpJKSlpcU0nlCEE6TpmW7pdruxsLDglW6ZlpbmJRoCpVuGM5+QpIK7IVICiQaWZWG322Gz2SASiVYEQlLRQKEkHyoSViFutxtWq1XQG2mkgYujo6Po7OxESUkJ1q9fv8IcnWxLQiRzi8XiFemWXCVIf+mWXCpgslhtIsEX33gZTjS43W643W7Y7XaMj48jLy8PCoWCigYKJYlQkbCK4G6kCwsL+PDDD3HuuecKdtMMN3DR5XKho6MDs7OzXu4FX5ItEmJBIpEgKysLWVlZAFamWy4tLUGlUkGn0wmeOhqKZGY3xAt/zaqOHTsGvV7v1f3S1z0RaXAuhUKJHCoSVgme7oV4bMDhjLm4uIimpibIZDLU19d7uReiGS9eCL1x+Eu3NBqNMJlMMBgMcDqdaGhoSEi65Wp0N0QzH7B83aVSKW9pcLlccDqdVDRQKAmEioRVAFc5kQtO5EzdLMsK1mo5mLuBEIKxsTF0dnaitLQUFRUVITfBZFsS4jm3XC5Hfn4+8vPzMTU1hcHBQeTm5nqlW2q1Wl40CJVuyXGiiATuO+bP0hBINHg2q1oLdTEolGRDRUIK41v7gPPJcjc/IUVCILO5y+VCe3s7DAYDduzYwZvgQxFMJKylEx/3eRQWFiYk3TIZG3ai5wyVyhuuaPCtBklFA4USOVQkpCi+tQ88N1ZPkSAU/jb1xcVFNDY2Ii0tLaR7IZzxEkWiBYhvoSHfdMvFxUWYTKYV6ZZcYadIrmsyijcB0dXeiBbP73w4BBINXIdLwH8JaSoaKJTQUJGQYoRT+8BfK+ZYYRiGT6ckhGB0dBRdXV0oKytDRUVFxBtv2CKBZcHMzYIxTAFLFkAmA8nMA9HlADFkEKRK0KRIJIJGo4FGo1mRbjk+Po6urq6I0i2BxKdcJnrOSEWCL8FEg9Pp5J9DRQOFEhoqElKIcGsfcDdBIS0JnLvB5XKhra0NJpMJO3fu5BsqRUpYIoFlIRruATM+CBACIpWDWXBCNDUONicfbMVmQCaPav5UJZJ0S71eD61W65Vumaz4gETPKWS6oz/RwAlxT9HgWRFSIpGsGZcYhRILVCSkCNxNy+12h921UWh3g91ux8GDB6FQKFBfXw+5PPoNOhyRwMyMQzTaD1alA+TLJncCgLhdYKbGIJIrwJZvjGruRBGrxcI33dLhcPCi4dixY17pljqdLuHdLWM91Uc7Z7xTLj1jeTjR0NjYiOzsbOTn5/MNrTwDIalooJyIUJGQZLjaBy6XK6LSykKKBEIIFhYWMDs7i/Xr16O8vDzmG2IokcAQAtHUKIhUxgsEHrEEJEMDZmYCKCgF0pQRz58q7oZIkclkXumWNpuND4Ls7OyE3W6HVCrFwMAA9Ho9VCpVXDfUZLkbEjkfJxpYloVMJuP/m4tp4KwMtC025USEioQkEktpZaFEgtPp5LMXdDodKioqYh4TCC0SiH0JzJIZJJAAUKQDhkkwS9bAzwkydyKJ53xpaWl8uiVXZGh+fh4WiyUh6ZbJdDckGk+RzlkauPfPsiwcDgfsdjsVDZQTCioSkoRv7YO4BQYGYX5+Hs3NzVAoFCgrK8PCwkJM43kScn2MCGAYINBzCAswIhB64+Xhgu2USiU2btzIp1tyhZ3i0d2Si4FYS+6GSOb17DkBUNFAOfGgIiHBeNY+iCVAKxZLAiEEw8PD6OnpQXl5OcrLyzEyMiJ4tkTQ8WRysJpMiGcn/FsKzAsg6SogQx3V/IlyNyQjJZH7vnimWxYXF/tNt5RKpV6iIZJ0SyDxpn9uzmSJhFB1R4KJBrvdHjTlkooGymqEioQEwrIsXC6XIJ0boxUJTqcTbW1tmJubQ21tLR9lL3S2RGhLAgOSuw6YmwUzbwRRaQGRaNmyYDWDcdjAFq8HJNKo5l7LBHp/gdItjUYjxsbGokq3TEYZ6GQIEwB80HAkeIoG37bYvqKBC4KUSCS0WRVl1UBFQgLwTLkSynwbjUiYn59HU1MTMjIycNJJJ3ltEEI3KgrHHUK0WWDXbwYz1APGMPnZH9IUYEurQfKKBVvPWkHo7pYZGRn8c3zTLbn5TiRLQqzzev62fUWDzWbjn+OZbkk7XFJSGSoS4oxvcKJQ/t1ITv6EEAwNDaG3txcVFRUoKyvzW6Ap0SIBAEh2AYhGD2bOADgdy5kNGv1y4GIMrNbshlDEsmlHmm6p0WiSEkSYzMBFocqcc4QSDW63G6OjoygtLYVcLqeigZJyUJEQRzjrwYEDB7B9+3aoVCrBxg735O90OtHa2oqFhQUv94IviXY3eN0AZWkgOYWCzZ1oVms2Rah0S6fTifT0dLjdbszPz8c93ZIjWXEQiRAnvqLB7XZjeHgYJSUlXpYG3w6XVDRQkgUVCXHAt/aBZw8GoQjH3TA3N4empiaoVCrU19cH9T8nw90QL9byzTSe19Q33XJpaQljY2OwWCxoaWmJe7olRzLcDaGaSsVzXoZhIJUux95wVgYuENJms0EkEuEnP/kJ/uVf/gXnnHNOQtdHoVCRIDD+ah8IXR2RGzdYa+fBwUEcO3YM69evR2lpacibecIDF+PMiZDdEE8YhoFSqURmZiZmZ2exb98+mM1mr+6WIpGIFw16vR4KhUKQtZ1IIsHtdnu5OHzdkZxo+OCDD7Bnz56Ero1CAahIEBQud9q39kEiRYLD4UBraysWFxexe/duaLXasMZLVkwCJXKS1UdBpVJBpVLFLd3Sd85E4inqE0moOAhONFgsFqSnxxanQ6FEAxUJAsC5F7jsBV//YTxEgr+Tv8lkQnNzM9RqNU466STehBkOiXY3xHOjW+vuhkSLBH/z+Uu3nJ+fh8lkijrdkiOZtRkSPa+vJcEfhBBYrVZBY5oolHChIiFGwimtzAUoCYnnpk4IwcDAAPr6+lBZWYmSkpKoKjhSd0Pqk6ruDbFYDL1eD71eDyC6dEuOZLkbkpFREW5tBovFAqUy8h4mFEqsUJEQJZHUPoinu8HhcKClpQUWiwV79uyBRqOJerxEWRJYlkVvby9sNhsyMzMjOmWmIqlwso8X0Z7qg6Vbcp+9b7old6JOlkgQOv0xHMKxJADLIoFaEijJgIqEKPAsrQyErn0QL5FgtVrx0UcfQavVor6+PiL3gi+JiklYWlpCU1MTWJaFRqPxOmXq9Xr+lBnLDXstuxuA1SlKwkm3VKvV0Ov1sNlsUcczREs01RaFmjfUd93lcsFutyMjIyNBq6JQPoOKhAjxTWkM58YitEjgWjvPz8+juroaxcXFSangGAx/7ovZ2Vk0NzcjNzcXlZWV/OnN4XDAZDLBaDSiq6sLTqcTGo2Gj5pXqVQRv7+1nN2Q6PniIUr8pVty3wGDwcD74eOdbsmRTHdDKJFgNpsBgIoESlKgIiFMPN0LkQY5CRmTYLfb0draCrPZjJycHJSUlAgybjwtCYQQ9PX1YWBgADU1NVi3bh1fQwJYPmXm5uYiNzeX3zCMRiOMRiOGhob4zoacz1uhUAi2ztVGMgIXE1FgSKlUQqlUorCwEO3t7RCLxVAqlV7plp5BkEKlW3KksrvBYrEAoCKBkhyoSAiDcIITgyHUKd1oNKK5uRk6nY7faIUiXoGLXMyE1WrF3r17oVYH7+rouWGsW7eOT7UzGo2YmppCT08P5HI5Lxh0Ot0KNwt1NwhHMjINCCFQKBQoLi5ekW45NTWF3t5ewdItOVLZ3WCxWKBQKJIiYigUKhJCwFkPuJtIots6A8s3zf7+fvT396OqqgpFRUXo7+/nO8wJQTwCF+12Ow4ePAi1Wo26urqoYiY8U+3Kysq8ouYHBgbQ1tYGlUrlFc8A0OwGIedLhkjw3LDDTbf0/A5EGgibyk2luBoJa138UlITKhICEKr2QSTEIhLsdjtaWlqwtLTkdRKPRwwBIMymwMVMzM7OorKy0m9DqWjxjZq32+28L7ujowMulwsSiQQZGRlYXFyMuy8bWJ2BhKk6HxDaehEs3XJwcBBmsznsdEvPOZNlSZDL5UGfYzabafojJWlQkeCHWN0LvohEoqhiEgwGA1paWqDX67Fjxw6vG53QIoG7Qcbqm3W73Whvb4fBYIBOp0N5eblQS/SLXC5HXl4e8vLy+GC3zs5OLC0toaGhASKRiD9h6vX6hEfNx4O1FpPgS6Qbtr90Sy5zIlS6JUe4qYhCE667IRFil0LxBxUJPnDWAyErsInF4ojiBzwD/aqrq7Fu3Tq/rZ3jZUmIFovFgsbGRkilUpSWlvIBV8HmExKGYZCeno709HTI5XKUlpZiYWEBRqMRExMT6O7uhkKh4AWDTqcLecIMBc1uEJ5YT/WegbCAd7olZ23ismd0Oh1UKlVKZzfQksyUZEJFwnEIIbBYLHz5UyFLtEZy6rfb7WhubobNZsO+ffsCFlCJRwwBgKiFx+TkJNra2rBu3Tps2LABw8PDSY8L4BoQcXEKLpeL3yz6+vqwtLTExzPo9Xqo1eqkbBSRsFqKKcWC0NYL33RLq9XKfw9GRkbAsiykUinkcjnMZnNC/f9UJFBSHSoS8Flw4tTUFMbGxrBnzx5BbxLhioTZ2Vm0tLQgKysLO3fuDHrKjZe7IdKNnWVZ9PT0YHR0FJs3b0ZeXh7/t1RrFS2RSJCdnY3s7GwAyydMo9EIk8mE1tZWvhUyZ2VI1WCxEyEmIV5izdPatG7dOhBCYDab0dPTA7vdjqNHj8Y93dITKhIoqc4JLRJ8ax+IxeK4nJxCbegsy6Kvrw+Dg4OoqalBYWFhyDXEM3AxXGw2G5qbm+F0OlFXV+d1Iwtl6Yj3xhPO+0hLS0NBQQEKCgp4SxJXzKevrw8SicQrniFQgFmiN+1EkqyYhERdU667pVKphE6nQ2lp6YqUW5lMJmi6pSfhxABxMQkUSjI4YUWCv+BEiUQieCMmIHgxJW6jdTgcQd0LvsRLJIQ7ptFoRFNTE7KysrBr1y6/Vo9UsySEek1GRgYyMjL43Pz5+XkYjUaMjo6is7MT6enpvGAIJ2I+Hpwo2Q3JCpb0Tbn1l27JxbVEm27pSTj1GahIoCSTE1IkBKp9EI8eC8HGnZmZQWtra9CNNhBCBy4C4cU5eHac5Go2BGolHGqsVDTlc3ianCsqKuB0OldEzKvVakgkEq9A13izFisupsKcgU70wdItBwYG+A08knRLT6i7gZLqnFAigat9wJUE9g1OjEdLZ2ClSGBZFseOHcPQ0BBfpjiaMYU+qYcqzex0OtHa2oqFhYWwOk6upVbRUqnUq0ERVzp6bGwMFosFBw4c8MqaUCqVKS2CwiUZ5YpTuVV0uOmW3PdArVYHvX7higROpFAoieaEEQnh1D5IhEjw9eNHa0aMZ/tpfywsLKCpqQlKpRL19fUhTazxEDGphEKhQGFhIQghmJ2dRXl5OUwmE2ZmZnDs2DFIpVKveAahWmGfKO6GRM8ZbVnmYOmW7e3tftMtPYOEaUwCJdU5IURCuLUPoi16FApOfMzMzKClpQU5OTnYuHFjTCe0eIiEQJYEzidfVlaGioqKsG/gqykmIRZEIhHUajXUajVKSkp4P7bRaMTw8DA6OjoEa4V9IoiE1VDAKRDB0i25tGAug4aLPwr1XeDSsimUZLCmRQIhBC6XCy6XK6zSymKxGISQuNwY7XY7mpqasHHjRhQWFsY8XrxEgueYbrcbnZ2dmJqawo4dO3gTa7jrW0vuhkjw9WP7tsJ2OBzQaDT8cyJthb2WYxK403WqxCTEQqB0S+670NfXBwDo6uriBaS/dEuLxULLMlOSxpoVCSzLwuVyRVRambtJuN1uwSLXl5aW0NXVBZZlcdJJJwlmNox34KLVakVTUxMYhkF9fX1U7ZnXsrshEgK1wuZOlwBWxDMEItHXNNGmf+79JVokJKILJJduqVKpUFxcDIvFgk8++QQZGRl+0y3VajXS09NhsVioJYGSNNacSPCsfcBZBMK9yXEiQajNd3p6Gq2trdDr9bBarYL6FeMZuMitOz8/H9XV1VHdPJNpSUjlgMFArbC5NsihWmGvdXcD99s7EdIuCSEQi8UoKyvzm275ve99D01NTVCr1WhqasJZZ52FzMzMhK6RQllTIsHTvQAgIoHAPR9AzHEJXBXCkZERbNq0CRqNBtPT0zGN6Qu3CQt5E2cYBiMjI5iZmcGmTZtQUFAQ03gngrsh1nl82yC7XC4+nsFfK+wTRSSsBXdDKHwzG3zdVI8//jjeeust/OxnP8OLL76IBx54ANu2bcP111+Pb3/72wldK+XEZc2IhEC1DyKBYZiYMxyWlpbQ1NQElmVRX1+P9PR02Gw2wTd0obo2ctjtdthsNrhcroiKOgUiVDolxT8SiQSZmZn8idG3FbbD4UB/fz+fFhfv7oDJiEkAkiMSkuHiCPbb1ev1+NKXvoQ77rgD//jHP1BUVIT9+/cnpVsl5cRl1YuEULUPIiWWDIepqSm0tbUhLy8P1dXV/I9Z6A1d6DFNJhMff7BhwwZB/J/JFAmJnjuem7RvK+xDhw5BrVZjfn4eg4ODfNEn7gQqdCvsZDWUSrS1JBExCb6E256ai0nIzc3FlVdemYCVUSifsapFQji1DyKF698QCSzLoru7G2NjY9i0aRPy8/O9/h5vkRAthBAMDQ2ht7cXlZWVGB8fF+xGmWxLwlq0YjAMA5FIhJycHOj1erAsu6IVdlpaGi8YtFqtVzxDNCQ6cDFZXSeB0KmIQhOOSHA4HHA6nTRwkZI0Vq1IYFkWDodDEOuBJ5G6G6xWK5qbm0EIWdHkiEOIDd2XaBoyeeJyudDW1gaTyYTa2lrodDpMTU0JtrkmWySsVTyvqb9W2HNzc3x6nWcrbJ1OB41GE7EITIYlIRmuBiA1MyosFgsA0LLMlKSx6kQCIQQ2mw12ux0ymUxQgQBEZkmYnJxEW1sbCgoKUFVVFfBUwK1RyEJNnEk2GuFhNpvR2NgIuVyO+vp6vruhkGmVyXY3rFWCbdq+JYO56n9GoxHj4+Nwu93QarW8eyKcVtjJiElIxmYNpGawpNlsBkBFAiV5rCqRwLkXBgcHMT8/j+3bt8elrXOozZxlWXR1dWF8fBybN29GXl5eWOMmsoxyIMbHx9He3o6SkhJUVlZ6XT8h0xaTvVGvluyGaOYL99r6Vv/jWmGbTCb09/dDIpF4xTP4a4WdrJiERJKstMtImjslWsBQKByrQiT41j7gWjrH40cdyt3AFRkCgPr6+rAroSVbJHgKm23btvGNijwR2pIQj46aFOFbYY+NjQVshX2iuBvEYnHKioS10iyMsjpJeZHgr/YBJxLiQTBLAudeKCwsRFVVVUQ3s2SKBC4tkxASVNisFUtCoudORkXCWPFshQ0gYCtsu90Oq9UKnU6XkM07WamIyTip0zbRlNVASosEznrgG1gUr26N3Ni+G6/b7UZXVxcmJiawZcsWvuNbpOMKveZw/P6zs7Nobm5Gbm4uampqgt6UhIwjSLYlYa0GTcbrZO+vFbbJZEJPTw/6+/vR39+fkFbYq7m5U6S43W6/Lh5PzGZzWLEjFEq8SEmR4Ole8Je9EG+R4Dm2xWJBU1MTRCJRRO4FXxJtSSCEoK+vDwMDA6ipqcG6detCjie0u4ESHxJxbRUKBRQKBQYGBrBp0yaIxWIYjUavVtie8QxCtcJOVkxCMgoUhWNJELqcO4USKSknEsKpfZAokTAxMYG2tjYUFRVhw4YNMZ02EikSHA4HWlpaYLVasXfvXqjV6rDHWwuWhNVo/k/l+UQiEd+YyLcV9sjICN8K2zOeIdpNN1kxCcmwJISb3UA7QFKSSUqJhHBLK8dTJIhEIr6GwNTUVMAgv2jGTYRImJ+fR8PRo7CPjMHZP4x3X/8H9CVF2PaFC6FdF7wXg9DuhmRC3Q3C4O9kH6wVdnd3N+x2e9StsE+0mIRw6iRQSwIlmaSESODKonLZC6FqH4jFYj6QUWjcbjcmJyehVCqjbpHsj1jKPQfC87ROCMHIyAjam5ox8PuXMHGkCW6nk3/u/sd+jfN++D3su/bqoGsUuk6Cv02Nq/Q4MjICrVbrt9shJTCp1uDJXytsTjT4a4WtUCgCjnmixSSEE7hIRQIlmSRdJERTWlksFvNxC0L+uMfHxzE2NgaFQoG9e/cKOnY05Z5DwbkH3G432tvbMTs7i9m/vYPRQ4eRplZBqdctb9YsC4vRhDfu+Rm0BfmoPucMv+MlwpLArdVgMKCsrAxmsxn9/f1YWlqCWq3mT59qtTrqzZBmNwg7X6SdVLlW2IWFhSCEYHFxEUajcUUrbE44eIpDGpPgDY1JoCSbpIoEznoQaWlliUTCv16IjdztdqOzsxNTU1MoKCiA0+kU/GQRL3fD0tISDh06BKlUiqqCQnzw7geQpSsh8/BjMiIR0jP1WJyaxsGn/xBUJAgduOi5yXA1JkQiEfbt28dXjWQYBjabDUajkfdzA+AFQzSNi6i7Qbj5YvktMAwDtVoNtVqN0tJSuN1uvnT00NAQ2tvboVKpeMHgcrlOKHdDODEJNAWSkkySIhJ8ax9EWlqZ+2G53e6YTdRmsxlNTU2QSCSor6/H7OwspqamYhrTH/EQCU6nE319fSguLsaGDRtw5IWX4VxagipAoSRZejqGjzbCOjcPpVbj9zlCWxK48bhUzPz8fFRXVwNY9mVzpKWloaCgAAUFBSCErGhcpFAokJmZGXNgnNAkQ4wkSiQI3d4cWP7terbCdjgcvDjs7OyEw+GAVCrF0NBQQlphA6nvbhAiJopCiZakiATPE2s0bWG5bnix+vjHxsbQ0dGB4uJiVFZWQiQSxS0oUkiRwLIsenp6sLi46LXpsi4XwDBAgMvJiBiwbhasR6yC7xqFFgksy2J4eBjHjh3zSsUMdi0YhoFGo4FGo0FZWRlf6MczMI6LZcjMzFyRR57soMl4kkhRws0Vz+spk8m8WmF3d3fDYrH4bYXNxTMITSpnN9CYBEqySZq7IdaTYCybucvlQmdnJ6anp7F9+3ZkZ2fzf4vHiR8QLiPDZrOhubkZTqcTmZmZXjeQ/M0bIZZK4LTZIPNzM3VYrMitqoQyU+937Hi4G1pbW7GwsIDdu3fz3QojxbPQDxcYZzAYYDQaMTg46BVtz0Xcr0V3QyI27WTOxzAMxGIx1Go1KisrwbIsH8/g2wqbqxYpRLBrOCd6oeFiiUKJExqTQEk2SQ9cjJZoN93FxUU0NzdDKpXipJNOWuHrTmVLgtFoRFNTE7KysrBr1y50dHR4jVm8azsKt27B8OGjkEhlEEk+u/HZj7ec3XPNlQFvTEJaEpaWlgAsm5Pr6upCVpYLF8/AuKKiIq8eBMPDw+jo6IBMJoNEIoHJZIqqPXKqk2iRkMjr53m6FolEXhYlz1bYAwMDaGtrg1qt5i0N0X7WyQhc5H63tCwzJdVZ1SIhkjRIQgjfyKa0tBQVFRV+byjxSFXkxnUGMPOHghCCgYEB9PX1oaqqCkVFRbzLxXNTZxgGlz98P35/3c2Y7RsAIxJBJJHAfTwQc8flX8Cer3wp4DxCWRJmZmbQ3NwMANi2bZtfgSDURufZg6CiogIOhwO9vb2Yn59He3s73G63V2XAYOl30bJWT/aJng9Y3jwDWQd8W2Hb7XY+nqG9vR0ul4v/LoTbCjvUnPGCu8dQdwMl1UmaSIj1xhPJid/lcqGjowOzs7PYsWMHf5MJNG483A3RWhKcTidvst+zZw80ms8CDv2NmVVeiptefR6Nr7yGlv99A7bFRWSvL8euL12GmnPPCnrSijVwkRDC1/nfuHEjWltbg37O8dh8ZDIZVCoVWJbF5s2bYTab+XLCvb29kMvlfACkTqfjM2VWA8mq7piq2RRyuTxgK+yBgQGIxeKQrbCB5FgSPFO+A0EIoe4GStJZPXdIH8IVCYuLi2hqaoJcLkd9fX3IVLp4uRuiGXdhYQFNTU18YSff+viBhEdGViZOuflfccrN/xrRfLG4G1wuF1pbWzE/P8+LmdbW1qTGBjAMs6KcMBcA2dfX51WbITMzM6LKgBxrObvBM7g4UUQbRBioFbbJZOItiEqlkhcMXCtsIDkpkFwcRKhrazabqUigJJVVKxJCtYsmhGB0dBRdXV0oLS3F+vXrw7rZpUpMwujoKDo7O1FWVoaKigq/a2cYRtC1RutuMJvNaGxsRFpampeYETKlMhKClfP2NFcvLS0JWpsh3pwo7gYh5vN0Q5WXl8PpdPLxDJ6tsHU6Hex2e8IzYsINlqQxCZRksybdDS6Xi6/qt3PnTj4nOxy407TQ+eHhigTPwk6hXCNCZ2JEY0mYnp5GS0sLioqK+DRSjmSJBCC8E75CoUBhYSEKCwu9IunHx8fR3d294uSZ7NoMyWrutBbmlEqlyM7O5jOZPEtHm81mHDt2DEajkXdPxKsVNkc41gvOhaJSqeK2DgolFKvWkhBIJHAmeoVCgZNOOiniqHrPQk1C+qvD2dC5ioQMw4TVN0LIbAQgsk3dsxX15s2bkZ+fH9N4ycY3kt6zNkNXVxecTic0Gg0fzxBuUJyQJMOSkIwSyYkQJlwr7IKCAnz66afIzc0FsFz0q6+vz6sVtk6nEyw7hyOcOAi73Q63203dDZSksmZEAtfgqLu7O6iJPpxxgfiIhGCugenpabS2tvLFkcK5UcbDkhDOeE6nEy0tLTCbzdi3bx9UGRkAJwZ8ihqlkrshEnxrM1itVt410d/fD4lEAr1enxR/diJjEpIhEpIhvtLT05GVleXVCttkMvGtsNPT0wW1KoVbbREAFQmUpLIm3A1ca2eTyRSxeyHQuoTOcAiUNUEIQW9vL4aGhrBp0yYUFARv5+yJ0CIhnE3dbDajoaEBSqUSdXX7kCaRQMQ6wWD5dSwjAsuIQRhRyrsbwoV1uWDs6sVERxdcNhuy8vOgqSyHXSzC5OQkHA4HrFYrv4nEqzZDqjd3EoJkVD/0ndOzOFdFRYXfip8ajYa3NKhUqojXHG7fBq4uCIWSLFa1JcFut2N+fh7Nzc1QKBSor6+P2SzIVX0TOnjR34bucDjQ3NwMm822fCKP0PcoZIXEcMabnJxEa2srSkpKULl+PSTEDRFZrlXBbV8iwoIhBG6RJGXcDYvTM5jtH4REKkX+lo2Q+GSJBMNqmsOHTz2NsebW5XoTYjFYlwsZOdnYfc2VKCgogNVqRWZmJoxGI9ra2sCyrFfqnVA3eepuiA+hNmxPqxKw7BbkRAMX8OpZnyGcWhyRdIBcy2XGKanPqhUJIpEIi4uL+PTTT1FeXo7y8nLBfkyJEAkmkwlNTU3Q6XTYsWNHVK6NRFkSPK0dW7ZsQV5eHkSsGyKwx8XBZ9edgIABgZh1CR4zES7c98A8a8B7Dz2GzrfehcNiBSMSQVOQh303XINdV30xrMCxT559HsOHjyKzvAwy5XKMCGFZGIdH8ckzz6Pq6sugLMjzytc3m80wGAyYnp6OS22GRIqEtRK4GIxIhUmgVtien7dnPINv6jIQviUh3gGUFEooVqVIcDqdmJiYgMViQW1tLV+vXyji1daZZVkQQjA0NITe3l5UVlaipKQk6puA0Juwv/GcTieam5thtVpRV1e37B8lBCLCiSjftTO8UNCqMpJmSbAvmvHiTd/BWEs75OlKpGfqwbrdmB8bx5s/+TmspjmcdsvXg44x2zeA0cYWaIvW8QIBWG69rS8pwmRHN8Y+bUDlxRd89jeP2gylpaVepYQ9azNwoiGS2gzJaBN9IsQkxGK9CKcVdkZGhpcrijuE0JLMlNXAqotJmJ+f51s7q1QqwQUCEB9LAjdmc3MzTCYTamtrodPpYhozHpYEz/EWFxfR0NCAjIwM1NXVeZWuXZYCAUcCQJAmlydNJAy9+0+Mt3ZAlZMF8fF1i6USSNNyYDGa8PHv/oBtl1wEbWHgGBBD/yDsVgt0JUUr/sYwDJQ6LWa7erCePT/gGL6lhD1rMwwPD4NhGP7UmZmZGdRdluhrudo27GgghAhacTFQK2yTyYTOzk4+S4ZlWaSlpQUVYmazOSlZNBSKJ0m1JESacsedwCsqKqBUKtHf3x+XdcVDJCwtLcHlcsHhcAgSOwHEt07CxMQE2traoswUOR7ESNign6/v528xGDHd2weGAXKrq6DQqKN6HwzDYOj9A2BEIl4geKLUarAwOY2ut9/DvuuvCfwuCAFI4Fbm3PojuTb+ajMYDAaMj4+jq6srZBT9iWBJSHRDKSB+Tax8W2Fz8QzDw8NYWFjgazN4xjNwWK1WakmgJJ1V4W7w7F/AncBnZ2fjUhkREL7J0/j4ONrb2wEAtbW1gt2Q4hG46Ha70dXVhdHRUWzbto0P1vJ5IlhGBBFhj9sTVm4kBAzmzRbkhCEC7WYLPnzqd+h4420szS8AAJR6HbZefCHqb/wqJFEIKpvRBIncf4AiIxIBDAPLrCHoGPqSIkgVabAtmpGmWpmGZjWZkLlt0/J4UeBZm4GrCmgymWAwGPjaDFqt1qsFdqr2UVitc8ZbJHjCMAzS09ORnp6O+fl5vliX0WjE5OQkenp6kJaWBpVKhYaGBiwsLFCRQEk6KS8S5ubm0NzcjIyMDK+Sv/Eqn8yNLcTmy7Isurq6MD4+zjc8EhKhLQlutxtutxszMzOoq6sLeoNyM2IwhBx3O3BC4bMwRjcjgtsd3JIAAG6nE6//v3vR+/4/IVepoM7PAwBYjSYc/O/nsDg5hX+5586Ib+Jpeh2WJqb8/o2wLMCySM8KniqbvWE98jfVYPhwA7I3VPBZEYQQzI9PQpaejvxdOyJaVzBC1WbgvpdTU1PQ6/Vx71x4IsQkJFIkeMLVYfHXCru9vR0PPvgghoeHodPp8KMf/Qhnn322YBZICiUSUtbdQAjB4OAgjh07hvXr16O0tNTr5hFvkRDr2EtLS2hqagIhhBc3ra2tgppThQxcXFhYQHNzMwghqKurCx2Bz4jgFkkgZl3HayQsr4OAgft4rYRw3El9Bw6h78BBqPJyIfNIFVTl5sC2sIjOt9/Dli98DsW14W/GDMOg5IxT0Pk/f4Lb4YRY5r2ZWk3zkKsyUHPeWUHHEYlE2HfDNXDabJjs6ILoeOtt59ISFDotaq/+IqQVJRG1LI/kPXCnzqKiIrjdbv60yQXEqVQqPgBSrVYLvtElesPmyqEncsN2u9182/VE4i9wkYtfOe2009DS0oK77roLH3/8MUZGRnD11VdjYWEBY2Nj0Gq1CV0r5cQmJS0JDocDra2tWFxcxO7du/3+KMRicVxuzkDsJ/TZ2Vk0NzcjNzcXNTU1XpYJISs5CmVJGBsbQ0dHB4qKijA4OBj2+ggjgkskBQPCBzISiPiqi+GIhJ5394N1u70EAodclQHL7Cx63z8QkUgAgOIzT8FCawdGm1ogUyohz0gH63ZjaW4BYIBTvnkjNAUrS0n7os7NwTn/fitGGpow2tQK55INupJ1KNu7G/rSYvT19SVkIxWLxVCr1ZBIJNizZw/sdjtvZeDEp2cAZKiS3uGQaEtCMk71yajLAISXAul2u7Flyxb8+te/5tOQqUCgJJqUEwkmkwnNzc1Qq9V+2yNziMViPjJZ6B95tJYEz34GNTU1WLduHf+3eFRyjFUksCyL7u5ujI+PY/v27cjIyMDg4GBkmwPDSQR/fwqjguPMrN/gQu71jEgEiyF47IA/pOlKfOnXj+K9/3ocnX//ByxGExhGBF3xOuy7/hrsuurysMeSpStRcUo9Kk6pj3gdQuJ5LeVyedDaDGlpaXwsQ7S1GZKRcgmcGCIhnIwKi8UCjUYDYPm3sGHDhkQsjULxImVEgqd7IZz6AfHsBR+NSHA4HGhpaYHVasXevXuhVntH5nMmzURWSAyG3W5HU1MTnE4n6urqoFQqYbPZAAi3OYQjEtQF+XAfbvD7N0IICMtClesneDLEvACQnqnHRf/5Y5xx6zeXKy7K5cjbWBVRxcVUI1DLcH+1GQwGA44dOwabzQaNRsOLhnBrMyQriDCRwiTcls3xmDfUtbVarSgsLEzQiigU/yQ9JgH4zL1gNpsDuhd88WzEJHQAV6TZDfPz82hsbIRarV5RT8B33FSwJHDr1Wq12LVrFyQSCViWxfzYBMxjE1icnoEmLzfm9YUjEqrPOQMdb7wN2+Ii0nzKUi/NzUOmVKLq7NMjnttz3ozsLGRkB265HQuJrF0QrngLVpthaGgIIpGIFwx6vT5gMBx1N8QPWkyJslpIuiWBcy9oNBrU19eHveFzJ/N4BC+GG+/g2XmyoqICZWVlQW+qQgdbcje3SG7mo6Oj6Ozs9AoGHWtpQ9sbb2Oq5xjGhkfg+LQJJbt3YsuF50Mdg1gIRySU7tuNjRecg7a//h12swUKrQYgBFbTHIjLhV1XXY78zRujXsNaIlpB4lubgcvPHxsbQ2dnJ9LT0/kASK4iIDdfokUCwwSuSxGvOalIoFACk/hfhwfDw8M4cuQIysrKsH379ogtAvHKcAhnXLfbjdbWVhw7dgw7d+4Mq3dEPCwJQHhxDizLor29Hd3d3dixYwcvaEYam/HPJ36L8ZZ2pOu0UOZkQSSVovPt97D/8d/APDMb9frCEQmMSITzfvRvOO07N0OVnQnb3Bxs8wvQFRbgzO99G2fcfgutOHccITZtkUgErVaL8vJy1NbW4pRTTkFpaSmcTic6Oztx4MABNDU1YWRkhHc/JYpk9W1ItLuBEBK2SFgNbaKvu+46XtxJpVKUl5fj+9//Pt/qmuPPf/4zTj/9dGg0GmRkZGDr1q34yU9+AqPRGHDsz3/+8yguLkZaWhry8/Pxla98BePj4/F+SxQPkmpJ0Gg02LNnDx+cEynxynAItZlbLBY0NjZCKpWivr4eaWlpgowbKZ4iIdgNx2azoampCSzL8vEHwHKNgqa/vA672YLc6uWgKNHsLNKz9FBl6jHR2Y3u9/+JXVdcGtX6wq2oKZHJloMJv3QZjEMjy70RSoujjh1IdPfJRIqYQHMRQmAcHIbVaIJUqUD2+vKAAaGeSKVS5ObmIjc3F4QQWCwWGI1GGAwGmEwmMAyDzs5O3jURz9oMySgDHY+YplBw94C1IhIA4Pzzz8czzzwDp9OJAwcO4MYbb4TFYsGTTz4JAPjRj36EBx54ALfddht++tOfoqCgAL29vXjqqafwhz/8Ad/97nf9jnvGGWfgzjvvRH5+PsbGxvD9738fl19+OQ4ePJjIt3dCk1SRoNVqY9rkk2FJmJycRFtbG9atW4cNGzZEdINJhiWB6zaZmZmJTZs2ed2Ypnv7YBgYgq543fLGyuUoEEAkkSBdr8fAoU+x9fMXQBqmEPIk0s1aqlDwYoWykkCWBMPAEI788WVMtHbAYbFCLJNCX1qMbZd+HuX1e8Men2EYZGRkICMjA8XFy+mdi4uLkEgkGBwcRHt7O9RqNS8YhK7NkAzTf7LmBIKLBK6Y1moRCXK5HHl5y4XQrr76arz//vt47bXX8OSTT+LTTz/FT3/6UzzyyCNeYqC0tBTnnHMO5ubmAo5722238f9dUlKCH/zgB7j44ovhdDrjXkyMskzSYxJiIZEigWVZ9PT0YHR0FJs3b+Z/EJEQj+wGbm2+eMZLBMoWsS+awTqdkB3PqWeOl1fmNnZZuhJ2sxkOizUhIkEo1qp7wt+1nBsdw7u/eBTG4RFoCwqgKSiAy26HoX8Q/3zs1yAsi4qT66Kaj2EYpKWlobKyEgD42gwGgwFjY2NgWdYrADLW2gwnikjg7i2h5jWbzatGJPiiUCjgdDoBAM8//zwyMjLwzW9+0+9zw639YDQa8fzzz0cUu0aJnZTIboiWeIkE34BIm82G5uZmPl0w2mAiodfL+QF9Nw+3243Ozk5MT09j165dATtlylUZEMukcFiX+FbInhYFh8UCmVIJWfrKQkfhri+QSHC5XDh27BgfiS90t7tEiZNEiyDfa9Tx93/AMDiMvJpqiMTLm45MqUD2+grM9g2g6eXXULJnV1SuG1/zv29thsXFRRiNRkxNTfF9B7gASK1WG3FthmTEJCQjBZJzcYT6vq8md4Mnn376KV544QWcddZyRdPe3l6Ul5dHvbHfcccdePzxx2G1WrFv3z68/vrrQi6XEoJVbUmQSCRxsyRwp3Oj0YimpiZkZWXx6YLRIrQlwd+YNpsNjY2NAIC6urqgp7ucygpklpVgtm8QORvWL9+0mOOBVS4XLEYTdl5xelRWBCCwSLBarTh69CikUilEIhEGBwchlUr5FrvRFv9Z6/i6GxwWKwYOHUZGViYvEDzRFBbAODyKyY5urNu+Jeb5PGEYBmq1Gmq1mq/NYDKZYDQa0dvb61WbITMzExkZGSE3xVAxCUtz81icnoFUoYB2XYEgojJZloRQwmQ1uBsIIXC5XNBqtThw4AA2btwIl8sFl8uFSy65BHfffTdsNhsyMzPBsmzUgbC33norrrvuOoyPj+NXv/oVbr/9djz11FNr1mKYCKRSadjieFXfiePpbnC5XBgYGMCxY8dQVVWFoqIiQSLL4ykSOEGTnZ2NjRs3hvwSiKVSbL/08zjw1NOY7OyGtjAfrNMF8/QsbEYTcqs3YMPpp0S9Nn8iwWAwoKmpCQUFBSgrKwOwfLOZm5uD0WhEX18flpaWoNFoeNEQqZVhrd48fK+lw2qFy26HPMBGIk2Tg3U54fCJMo9kvnA3UIlEguzsbGRnZwOAV3OqcGszBNqwzbMGHP7Di+h+dz/sZgvEEinyN1Vj19VfROne2qjeW6g540k4ImFpaQksy6asSHA4HJiYmIDVasVll12Giy++GHq9HgzDQCxe7tuysLCAhYUF3HzzzTCbzejv74/pt1lQUIB77rkHY2Nj6O7ups2uYoBhGKxbty6s7xd1N/iBZVk4HA4MDQ3FlH3hS7xEgtvtxtDQEHp6eiIWNOu2bcFpt9yE9r//A9PdvbDNGsCmKbDxgnOw5XPnIyNEp8RgeIoEQgiGh4fR09ODmpoaFBYWwul08pkZnCCorKzE0tISDAYDjEYjBgcHvf6u0+nCMlueCNkN8ox0yDPSYTebodCoVzzXbrFAIpdDqdNGNVcsKZdKpRJKpRLr1q3jazMYDAa+TkdGRgYvGLRaLf/b8N2wrUYT/vqDuzDe0g6FToOMLD1cDicGPz2Cyc5unHvn91AZg5BlWTbhVqtw0x8BpKRIYFkWAwMDEIvFKCgo4IuxlZSU+H1+Tk4OBgYGoFar+QJfnrhcrrA/A4fDAbvdjqKiIlpDIkoIIZiZmcHo6CgqKytDfhdXvSVB6BTIhYUFtLW1eXVvFIp4iZpjx47BbDajtrYWOp0u4tcXbKpB/sZqzI9P4sD+/di5by8KK8pjXhcnEliWRUdHB6anp73WGGgDUigUWLduHb/BcCWGBwYG+Ah7TjSEY8ZeK/hu2lKFAhWnnoQjz/8Jqpxsr5RHQgjmRsaQu7EKOVWVUc0nVEoiV5tBq9WioqICDocDJpMJBoMBHR0dcLlcvIuJ6wTJzdv82usYb+1AZlkJ381TqlAgTa2CcXAYHz71NEr37Y7aJeZ2uwX9jYc7Z6gbs9lshlgsDju9OpE4HA6wLIuioiIolUoYjUa43e6Aa01LS8PS0hImJycBLAcqSqVS2O12zMzMICMjA7m5K4u2WSwWPi5DLBbDbrdjfHwccrkcOp0uKUWw1grZ2dkYHByE0+mkIiESuG6I69atw9DQkOA3D6EtCUtLS3A4HLDZbKirq4vphsIwDLSF+VCvK4AiypOnvzGdTicOHz4Mt9sdMkbCH55mamA55sJgMMBgMGBoaAhisZj3e3N5/GtVNPizjmw8/2yMt7RjvK0D6Xod0lQquOx2LExNQ5WThd1fvgKiKAPz4lVxUSaT+a3NMD4+DqvVioMHDy5bGNRqtP/tbcjSlSvafTMMA01BPuZGxjB8pDHqDI5UdTdYrVYolcqU3ggjWdu6deugVCoxPT2N6elpAOA3+8xM/9ZKhmFgMpkwPj7Ol9/XaDQoLy9P6euyGojkd73q3Q12uz3mdXDZAFNTU9ixYwcyMjIwNDQk+A1ESJHA+fbFYjE2bNgg2IkjlqZRvjidTkxOTiIrKwubN28WJIo8LS3Nq8Tw/Pw8DAYDBgcH0dHRAbVajbS0NLhcroSUFU5k6WJ/c6Vn6nH2v38XrX99A30fHoLFYIRYJkXl6adg80XnIzdKK0Kg+YTGszYDwzAwGo1Yt24djEYjjnV1Y3psDFK5DBaLBTKZHBKJhOtEDolcBsKysMxG3iWUI1lpl+FYElLR1eAPLrYoFJ5iPxyUSiWqqqqiXRZFIFa1JUGI7Aar1YqmpiYwDIP6+nqv/N54iIRY1+vZLbOmpgbDw8PRDTRvBNP4ITA9DsjkINXbgQ1bIRKJBPHnT05OLp8YjHOY+LgBH915L4ibxbodW7H18xegrG5PzHOIRCLodDrodDqsX78eNpsNRqMRExMTsNls+PDDD3kLQ2Zm5prIrfa3aadn6rHv+muw/fKLsWSag1SpiCmWhCPRKYmEEK/4k7KSErTnZMMyNw+3y4156xwAQCqTQSaTQcwAEDFIU6+MxwiXZKZABoP2baCkCkkXCbEU3InVxz89PY3W1lbk5+ejurqa/+F6dpgUMqhJJBLB4XBE/XqXy4W2tjaYTCa+W+bIyEjEJ3+m4QCYvz4H2G0AIQDDgPn0PaB4PWSV9TGJBEIIjh07hsHBQdj7BtH27B8Bu2M5Al/EoOPNd9D7wYc45eZ/xc6rLo96Hn+kpaWhoKAAcrkcvb29qKqqgtFoxPDwMG9l4DagcNslpxKhPpc0VQbSVMKdPpPR4Mlz85TIZNh43tn49LkXkJGRDpFIDafLxbvYFscnkKbVwKnXYHZ2FjqdLuINP1XdDZxIWG3f0YghBGDdAGEBRgSIxMBaf88ATj/9dGzfvh2PPPJIspcSkqSLhFiIViQQQtDb24uhoSFs2rQJBQUFXn/nfphCBxl61l+IFKvVisbGRkgkEtTX1/PpPxG7MPo7wfzlacDtApQZn/0gXU5gqBcbFsywbdwc1RpdLhdaWlqwuLiIzeUVeO7uB+BasiGPq8EAALk5WJiaxke/eQYFWzchpyY+5kSGYXgrQ0VFBex2O58xMTIyAoZhvGIZYok/Saa7YS3N5y9QctulF6H/w0OYPTaAjJwspKlVkIkY2OcXoFKrsev6ayBRpKGnpwd2ux1arZY3a4dbmyGVRcKaxuUEnEvLQoGDYQCpApDEx+p33XXX4bnnnsPXv/51PPXUU15/++Y3v4knn3wS1157LZ599tm4zM/x6quvrhrL5gknEhwOB5qbm2Gz2bBv3z6oVKoVz+FyfeNd+ChcZmZm0NLSgoKCAlRVVXnd1CJ1DzCH3l7+cXoKBGD5R0kI1IZxuCaGgPz8iNZotVrR0NAAmUyGuro6NL/8GpaMc1D7KXqjysnGTG8f2t94O24iwfeayOVyFBQUoKCgwCslb2RkBB0dHVCpVLyVQa1Wp+wJLtmbdrzn892wNfl5+PzP7sGHT/0Ow4cbYTSOQCSRQF9SjNovX4Ga88/mrZFLS0t8bQYudZYLjAskBJPRBZILwgvGaq22GDYuJ+CwrnyckOOPK+MmFIqKivDiiy/i4Ycf5gOpbTYb/vjHP6K4uDimscPtKRFJbEaySbpIiNXdEEl2A9fsSKfTYceOHUFdCfFIV4xUJBBCMDAwgL6+PmzcuBGFhYWxjcm6wfS2AhKJf5OeRAqGWCEf7gF27gt7nVwQpafbZnZgCMByK2hfGIaBVKHAVHdv2HNEQjgtu31T8riMidHRUQDgrQyZmZkJT5ELxIlgSfB3qteXFOHz998N4+Aw5ienIE1LQ/6maq+UT4ZhVtRmmJ+f5y1HHR0dfG2GzMxMaDQaPkYoGZaEUIHGZrN57VoSCFm2IATDuQSIA9ynYmTnzp3o7+/Hq6++ii9/+csAlk/2RUVFKC//LPX7zTffxH333Ye2tjaIxWLU1dXh0UcfRUVFBQBgcHAQZWVleOmll/DEE0/g448/xpNPPomvfOUruP322/H73/8eYrEYN954IyYnJzE/P4/XXnsNwEp3Q2lpKW666SYcO3YML7/8MnQ6Hf7f//t/uOmmmwR//5GSdJEQC+Fu5IQQDA0Nobe3N2CzI1+ECDL0N2a4G7rL5UJrayvm5+eDFnSKSCQQHDftBXjvx68JieB9Dw8Po7u7G9XV1SgqKuIfl6bJPeZbCety8f0iko1MJvPqSeBb+EelUvGbi2/nw1jjN4YPN6D9729jpqcPsnQlKk8/BRvPPxtK/cp6F4nuE5GswMVA6EuLoS8N76TnGdTKCUHOytDe3g632w2tVgun0wmHw5FQQRSO9SLVSzLHBOsOeF/g4WIVxPHZoq6//no888wzvEh4+umnccMNN2D//v38cywWC26//XZs2bIFFosFP/7xj3HJJZegqanJ63dxxx134KGHHsIzzzwDuVyOBx54AM8//zyeeeYZ1NTU4NFHH8Vrr72GM844I+iaHnroIdx7772488478corr+Ab3/gGTj31VFRXV8flGoTLmhcJnsF+kRQbioe7IVxRY7FY0NjYCJlMFrKgU0Qpi2IxUFAKDHYDcj8nGZcLYBg4sldaLHxhWZZPG/XXRKpk9y4cfO4FOG02wCf63O1ygXW5UHFyfXjrjoJoN1SGYaDRaPh8bG5zMRgMaG1tBSGE93kHyu8Od30HnvgtjvzPS3AuLUGSlga3y4Xhww1ofvX/cPEv7kVm2coKdmvdkhCv+WQyGfLy8pCXl+dVm8FoNKKjowN9fX385xpuVc9oOeGzG0i4hxph77+efOUrX8EPf/hDDA4OgmEYfPTRR3jxxRe9RMJll13m9Zrf/e53yMnJQUdHBzZv/ixu69Zbb8Wll17K/+/HHnsMP/zhD3HJJZcAAB5//HG88cYbIdd0wQUX8J0y77jjDjz88MPYv38/FQmx3BRCpUCazWY0NjZCLpd7BfuFQ7LcDdPT02hpacG6deuwYcOGkDeTSF0Y7L6zIRruXc5s8BQKLAvYl2BT6WApWo+VxVM/w+FwoLGxES6XK2CBpLK63cjZVI3xxhakK5WQHY/Udi7ZYBodQ876clSdfXrY644EITca381lcXERBoMB4+Pj6OrqglQqhUKhgMlk4k3Y4dDz7gf49LkXIFUooM7/rO242+XCTO8xvPmTn+Gq3/1qhdVirWzageZLhOXCszZDf38/du3axYvB/v5+LC0tQa1W86JB6BiVcCsuxiJCUxomzM843OdFQVZWFj73uc/hueeeAyEEn/vc51aUjO7r68N//Md/4OOPP8bs7Cx/nx0eHvYSCbW1n/UPmZ+fx9TUFPbs+SzFWywWY9euXSHv01u3buX/m2EY5OXl8YWnkknSRUIsiMVivuyv781lYmICbW1tKCkpQWVlZcQ/8kS7Gwgh6Ovrw8DAADZv3oz8MAMHI65rsHUfyEgfmEP/ACyLgEi0bNojBNDoMbjnc1AE+XEuLi7i6NGj0Gg0QbtiSmQynPy9W/DPhx6HZXAYC5PLX3aRRILCzRtxwT0/RHqmPi5lqoH4mOY9Ox+WlZXB6XSira0NDocDbW1tYFnWK2MimN+5+S//B7fLBW2mtwVGLJFAnZ+HiY4ujBxpRMmeXXF9T8FIlZiEeMHdO6RSKR+4WllZydfb4OIZAHgFQMZauCzc7IZAvRBWPVyaY7DvM8MsPy+O3HDDDbjlllsAAL/61a9W/P2iiy5CUVER/vu//5sPeN68efOKNHZ/Fh/f3004v11f65WQhe1iYdWLBMDbfMeyLLq6ujA+Po5t27YhJycn6rETld3gdDrR2tqKxcVF7N27F+oIisNEnDHBMCCf+zJI5RYwRz4AMzEEyNJAtuwG2XUa7P1DSAvwhZ6cnERrayvKyspQUVERcgNJz8pE3R3fRRZEGG1qBWEJcqvWo3TfboglEr5o1WpFKpVCqVRCrVajvLwci4uLfDGn7u5uKJVKPvjR08rgcjgw2dmNND+ZNQAgUyqxOD2D6d5jXiIBiMVKQiDG8r2XJcDyNyb4WMmISUjkfNzvxndOrt5GQUEBH6Pi+bkqFApeDGq12oizIyIpy7wm4dIc/WU3cEgVca+XcP755/Mb/nnnnef1N4PBgM7OTvz617/GKacsNxD78MMPQ46p0WiQm5uLTz/9lH+d2+1GY2Mjtm/fLuwbSBBrRiRIpVIsLS2hqamJb84Uy48sHu4Gf2NyLpG0tDTU1dVFHEkfVVolwwBV20CqtsFXDjDM8IrxPK0cW7du9duMxf80yz/yoh3bULRjW8C/rwU8rQylpaVwOp0rAuW406hWo1k+JQTwuZLjlh2Rz0kqupM9gZQBZAz5TBIwyyLBxgJsEKFwIrg3AATdsD1jVDjrkclkgtFoRHd3d1S1GcK1JKzZwEXgeHqjMuF1EjwRi8Xo7Ozk/9sT7rf6m9/8Bvn5+RgeHsYPfvCDsMb99re/jfvvvx/r169HdXU1HnvsMZhMplV7v0u6SIjlwjEMA5FIBJfLhdnZWTQ3NyM3Nxc1NTUx5z4nwt0wNTWFlpaWqF0iwPI1EHKdvu4LLstiYWEhYF2JYCTaRA7EllYbKYHmkUqlXk2MzGYzDAYDJicn0dPTA3nJOkx9chQytQpSmQyMx2ZtX1yEVKHAuh1bvcaM5j1JGUDOHG/X7fG4CIBCRLAURCis1ZgEDu53E8mcUqkUOTk5yMnJ8arNwPUP4RqOcf+irc2w5kUCsCwExJKkVlwMZLUViUR48cUX8Z3vfAebN29GVVUVfvnLX+L0008POeYdd9yByclJfPWrX4VYLMZNN92E8847L+H1OIQi6SIhVkQiEYaGhjA+Po6amhqsW7dOkHHj6W7wLF28ZcsW5OXlhX5xiDGFwtMPtrS0hIaGBkgkkqitHMkQCakGwzBQqVRQqVQoLS2Fy+VCJgu80dqJ2aFhyHU6yORySKVSEKcTllkDqs85c0WL58hP9gQyhqywFi3/ZdnZIGMIbMT/mGs9JoETQdG+x2C1GbhS4Fz6rF6vj6g2w5pOgfSEYeKW5uiPUJUUuToGAHD22Wejo6PD6++e97PS0lK/9zeJRILHHnsMjz32GIDl71lNTQ2uuOIK/jmeWRTAcs0FX5qamoKuNVGsapHA9TWfnp6O2JcfinhlN7jdbhw9ehRWqzWqk7m/MYXciLnxjEYjGhsbkZeXh5qamqhv3sHWFq8NKNXNehKJBFvPOgPie5x476HHsDhrgMXlhpt1AyIRsjZVo+LKi2E0Glf4vCNq8cosC4FAnwA5/pxAtTMiEgkOO8QdRyDqbgacdpDcIri314NkF4R+rcd8iRYJQp7uwq3NQAiBw+GAQqHwe325FM0TQiSsQYaGhvD222/jtNNOg91ux+OPP46BgQFcffXVyV5aVCRdJER7Q5+fn0djYyMYhkF1dbWgAgGIj7vBav0sUKeurk6QXOx4WBLm5+cxPDyMqqqqmMqUJtOSkMh5o/0Ob7rgXBTt2o7ud96HaXgMUkUaimp3QLW+DKa5OXR1dcHpdEKn00Gv10fcHCzcVQUSEuFu2oxpBtIXfgnRzPjxhmEAetsg/uRduM65HO69Z4W1jkS7N+JdbdE3fdZsNmNmZgYGg4FPzeYCIHU6nVemEBUJqxeRSIRnn30W3//+90EIwebNm/HOO++gpqYm2UuLiqSLhEghhGBkZATd3d2oqKjA5ORkXG4skZZ8DgWXkgkA27ZtE6xYi5AigWVZGI1GWCwW1NbWxlxfPJGxAcki1venzs3B7i9fueLx3OMbi9VqhcFgwOzsLEwmE8RiMe/3DtX1kA1SXJNfP4JYGsKxJLAspC8/BdH0GIgifblg1/KLAZsV0rf/BJKdD7Z8Y/BxkBx3Q6Lm41xOUqkUg4ODOOWUUzA3Nwej0Yi+vj6+NkNfXx+ys7OpSFjFFBUV4aOPPkr2MgRjVYkEt9uN9vZ2zM7OYufOncjMzMTs7Gxccu2FcjcQQtDT04ORkRFs3rwZzc3Ngm6cQuXSOhwONDU1wW63Izc3V5AGJKFEAiEkbvUMVhus243hI41YmptHZnkJciqXO2emp6cjPT0dxcXF6O3thcVi4b9TDocDWq2Wz99XKpVe792Nz2IP/F1lBoCTcP+1knBEgmiwC6KJYZA0xWcCAVj2NacpAasZ4k/eTVmRkIzmTiKRCBKJBFlZWXwBHy4A8tlnn8Wf//xnuFwu/OAHP8Bll12Gc889V7BYKwolUhLb2cQP4d7QLRYLPv74YywtLaG+vp6vRhaP2AFAGHeDw+HAkSNHMD09jX379vEBikK6B4SwJCwuLuLQoUOQSqUoLCwU7MaZTEvCarJgNL78Gu7fvA+/OucLePqLX8Uvdp2Gx8/5AsbbvIOmuEC5qqoq1NXVYffu3dDr9TAYDDh8+DAOHTqE7u5uD+HMwMYu/758f2UMltMgHQGCFoHwzP+iod7lyHR/wWcMAyKWQDTYtVzRMwTJiElIlTbRCoUChYWFeOyxx9De3g6GYbBp0yb87ne/Q2lpKR599NGErpNC4VgVloSpqSm0trb6LVUcL5EQa3bDwsICGhsboVKpUFdXx/sbha6iFavfn0vDLC0txfr169Hf3w+73S7I2sIRCck4zaUSn/7hRbz8re+teHz4SAN+dc7F+M7+vyH3eJaD57X0tTK43W6YTCYYDAb09PTw+fuZmZnIy8mGWiFfDlLEslXBSZYFAom5TsJxW0Ww54X5/VxrMQmB5gz1fbfb7bDZbLjnnnugUCgwNzcnqOuTQomElBYJLMuip6cHo6Oj2Lx5s99UQaFjBzzHjVZ8jI+Po729HeXl5SgvL/e68QktaqK1JBBC0N/fj/7+fq80TCFP/8HGIoTA7Xbzkd4ikYj/J8S8qwHn0hL+74d3+/0bcbNw2Wx48ycP4Nrnf8s/Hui9icViL/M1F8tgMBjQ398PmUyG3JwcZOr1UKnVEAdpkw58Jkj8fx4EEgYQgYDZtg/MsTaQ2Qm/BXAYtwvusprl8t+ej89OQNx0EMz0GCCTw71hG+B0rHlLQrgdIKVSKd9rRqvVJmBlFIp/ki4SAt30bDYbmpub4XQ6UVdXF7AjWiq5G1iWRXd3N8bGxrB9+3ZkZ2f7HTfZ7ga3243W1lbMzc2tSB0V0tIRSCRwAgH4rP8GIYQXe1yRLO7/R8NqyG7oePMd2BYWA/6ddbvR/re3YDXNQanTRmSO5/L3i4qK4Ha7MTc3t1xqtrsbNpsNGo2GLxmdfrz5lifc9fN9XAyCNJGH/SErB/jav0PU3w3X//0B4DIwCAFsS4BIDPeeM73H+OQdSP/xCuB0ACAAYSBu/QS1CjVEmzcCCep+mMyYhGCYzeYV8SUUSrJIukjwh9FoRFNTE7KysoI2EQLia0mIZLO02+1obm6Gw+EIKmriWfwoHJaWltDY2AixWIy6uroVnTHjaUngxADLsvwNmvtsuce4f54CTUgrg9DEcq0WJ6fBiEQgQT4/wrKwzBp4kRANYrGYFwTA8kmVqxI4MDAAqVTq1ZhKIpHw3ynPjYoBgULkZw0iMZjyKki+cC3cz/zX8UhJAkilcJ7+BbAbPqscKTrWBulbfwIhLKDM+MxN4XZBvWgE/vdpuL72o4RU3UtVd4PZbF67baIpq46UEAncZkIIweDgII4dO4aqqioUFRWFVNMSiUQwH7onkVgouJoNWq0WO3fuDCpqkmlJMJlMaGxsRE5ODjZu3Oj3Bil0bQNuLE9xwM3j+dl6igCuKiX3/EitDKvlBKbKywkqEACAEYmQnvVZy2Ah3ptnlUC32435+XneLdHe3g6NRuPXxC1jAn0vljv2MWVVYOvPAUb7weYVw73jZJCCUq9nij95F3C7vAUCAIgkcImlkE8Mgh3qAVtaFfP7DEUqBS56ciKlPxLWDXKsHZg3Aho9mPWbwMS5+2M4MAyDv/zlL7j44ouTvZSo2L9/P8444wyYTKaY3VUpIRIA8G135+fnsWfPHmg0mrBeF8/AxXDGHR0dRWdnJ9avX4/S0tKQN/F4xCSEs6lz69ywYQOKi4sDrlNIdwO3Ns/4A+7xUK8DPmu6wlkWOOEQjpVhNWQ3bDz/bKSpVQFdDiKxGBsvOBdKnRZAfMoke/YaqKys5FPxZmZmAACffPIJb4Uoy8tE4MILy8GLzPlXwE4CfL4sC/FgF4hE4sdSQMAyIjCsG6IEioRkuBvC7QC5WsRutLCNH4F9+TfA3OxnD2qzIPriTRDtOCkuc1533XV47rnnACwfMPV6PbZu3YqrrroK1113HX8fmZiYgE6ni8saVhspYb9dWFjAoUOH4Ha7UV9fH7ZAAOIbkxBss2RZFh0dHeju7saOHTtQVlYW1o860ZYElmXR2dmJ7u5u7Ny5EyUlJUHXGQ9Lgtvt5iPXozm5iUQiSKVSyGQy/p9YLOYFjcvlgsPhgMvl4udaDUgVClx0/91+/8aIxZCkpeH8H9/BP5YI4cOl4nHV4WpqaiCVSjEwMAC3yw1C2OV/AUswBYGQ4wUb/JQiBpaFA8OEnQ0RK6nsbljrlgS28SOw//1Tb4EAAHOzYP/7p2Ab41eM6Pzzz8fExAQGBwfx97//HWeccQa++93v4sILL+Qtlnl5eStcsScqSRcJhBC0traioKAAu3btiriJUDIsCXa7HYcPH4bJZEJdXR0fUR4OiRQJDocDR48ehcFgQF1dHe+TDobQgYtOpxOzs7OCnYJFIhHEYjFkMhnS0tIgl8shkUh4ceN2u/kfeqIEQyzva89XvoSrf/s4tIXePQ5Kdu/Et/7xGp/+KMRckcBdt8zMTKxfvx579+4Fjrt5CDlu3WHdYAl7XLwsb+zBWk9DLAa7rhyMy+n3zwzrBhgG7Lpyod+OX1I1u8FisazpmATCupctCEFgX/kNCCv8fR0A5HI58vLyUFhYiJ07d+LOO+/E//7v/+Lvf/873wCKYRi+2ZPD4cAtt9yC/Px8pKWlobS0FPfffz8/3vz8PG666Sbk5ORArVbjzDPPRHNzM//3vr4+fOELX0Bubi4yMjKwe/duvPPOO15reuKJJ1BZWYm0tDTk5ubi8ssv5/9GCMHPf/5zlJeXQ6FQYNu2bXjllVe8Xv/GG29gw4YNUCgUOOOMM/w2jIqWpLsbGIZBXV1d1De/eFoSOPO2541kbm4OjY2N0Ov12Lx5c8TmykQFLprNZjQ0NCAjIwP79u0LGifhuz4hTqwsy0KpVCI3NxddXV1wOBzQ6/XIzs5GVlYW0tLSYp4DWBnLYLVa0dPTA61W69ctwf13KrHjikuw7fIvYPhwg1fFRV8S2ZXR31wuRgwJCESi5ccJyHF3EguAAcu60Tc8Aa1OB5VK5Xetrj1nQjbUs5z5IE/zCFxkIXU7weaXh1WdUQhSNSZhrVsSyLH2lRYEX0yzIMfawXgEvcaTM888E9u2bcOrr76KG2+80etvv/zlL/HXv/4Vf/rTn1BcXIyRkRGMjIwAWP6dfO5zn4Ner8cbb7wBjUaDX//61zjrrLPQ09MDvV4Ps9mMCy64APfddx/S0tLw3HPP4aKLLkJ3dzeKi4tx5MgRfOc738Ef/vAH1NfXw2g04sCBA/z8/+///T+8+uqrePLJJ1FZWYl//vOfuOaaa5CdnY3TTjsNIyMjuPTSS3HzzTfjG9/4Bo4cOYLvfW9l7ZVoSbpIAGIrXBTP7AbA+0YyMjKCrq4uVFZWhjTbB0LoxlH+RMf09DRaWlpQUlKC9evXR7TOWLMbfLMXOLO1xWLB7OwsJiYm0NXVhfT0dD6vn2uhGyuLi4toampCTk4OqqqWfdqcNUHoFEsOoVwAIpEIpXtrEzJXOPgVCQRwAXxRJgbH2ywzAEAwNbeI+YUFDAwOQiQSeWVMcBZCtmYnXKd+DpIDbwBWMwjDgCEEIARWpRryL359RU2FeJGsmIRQ1lKr1bqmLQmYNwr7PIGorq5GS0vLiseHh4dRWVmJk08+GQzDoKSkhP/b+++/j9bWVkxPT/PuiQcffBCvvfYaXnnlFdx0003Ytm0btm3bxr/mvvvuw1/+8hf89a9/xS233ILh4WGkp6fjwgsvhEqlQklJCXbs2AFg+b75X//1X3jvvfdQV1cHACgvL8eHH36IX//61zjttNPw5JNPory8HA8//DAYhkFVVRVaW1vxwAMPCHJdUkIkxHI6iqe7AfjMb9nR0YHp6Wm+Z0Qs4wrtbgA+20AGBgbQ19eHzZs3Iz8/P+LxYnE3BAtQzMjIQEZGBkpLS+F0OvmmRZxZLjMzE1lZWcjMzIzY5QQsV45sb29HRUWFV2Cmp5VhNaZYepJoS8LK68HARgApAClDeF+liwAOIoJal4ktukywLIuFhQUYDAYMDw+jo6MDarWaFw3q078A9/rNkDQcADM5AiJPg6W0BkdZBeqyIv/ORkuqxiSsdXcDNGH2hQn3eQIR6Pd13XXX4ZxzzkFVVRXOP/98XHjhhTj33HMBAEePHoXZbF6xJywtLaGvrw/A8ud5zz334PXXX8f4+DhcLheWlpYwPDwMADjnnHNQUlKC8vJynH/++Tj//PNxySWXQKlUoqOjAzabDeecc47X+A6HgxcSnZ2d2Ldvn9faOUEhBCkhEmJBIpHERSQwzPIpyWq1oqurC4QQ1NXVQaFQxDRuPGISgOXskM7OTphMpoiyQ/yNF82J1dOC4Jve6ItUKvVqoTs/P4/Z2VkMDQ2hvb0darWad0tkZGQEHYsQgqGhIfT392Pz5s3IyckJ+L6ESrFMJsl0NxxfAZwEcBLPtlHezxOJRNBqtdBqtaioqIDdbufrMoyOjoJhmOWMij3n86LQOj8P9niX1ESRqu4Gi8XitxDbWoFZvwnQZgV3Oeiylp+XQDo7O1FWVrbi8Z07d2JgYAB///vf8c477+CKK67A2WefjVdeeQUsyyI/Px/79+9f8Tou9fDf/u3f8NZbb+HBBx/E+vXroVAocPnll/Ot31UqFRoaGrB//368/fbb+PGPf4y7774bhw8f5veKv/3tbygsLPQan7NcxNvCuOpFQrwsCZxIaGho4OsKCGGajJdIOHz4cMACSZEQjSXBM4MhlEDwNx+3oaxfvx42mw2zs7OYnZ3FwMCAV7e8zMxMr8+AZVl0dXVhZmYGtbW1XpUjg+EvxXI1WBmS7W5YSXifs1wuR35+PvLz88GyLBYXF3nB0NnZCZVKBaVSyafLJjI4M1VFQnl5YoI3kwEjEkP0xZuWsxsCILr8poTWS3jvvffQ2tqK2267ze/f1Wo1rrzySlx55ZW4/PLLcf7558NoNGLnzp2YnJyERCJBaWmp39ceOHAA1113HS655BIAyzEnvoGFEokEZ599Ns4++2zcdddd0Gq1eO+993DOOedALpdjeHgYp512mt/xN27cyAdZcnz88ccRvf9gpIRIiNXd4C/AMBYIIRgeHgbLsigpKcGGDRsEu3EJLRIWFhYALJvzt2zZEvM1iNSS4Lm5RioQ/JGWloZ169Zh3bp1YFkWJpMJs7OzfNMinU6HrKwsaLVa9Pb2wuFwYO/evTEFQvqzMnBuk3CsDMk/3QtPvJotiUQiaDQaaDQalJeXw+FwwGAwYGJiAg6HAwcOHPCKZYhnGlo4G7bQhNu7QalUJmhFyUG04yTga3eurJOgy4Lo8vjVSQCWs9MmJyfhdrsxNTWFN998E/fffz8uvPBCfPWrX13x/Icffhj5+fnYvn07RCIRXn75ZeTl5UGr1eLss89GXV0dLr74YjzwwAOoqqrC+Pg43njjDVx88cWora3F+vXr8eqrr+Kiiy4CwzD4j//4D6894PXXX0d/fz9OPfVU6HQ6vPHGG2BZFlVVVVCpVPj+97+P2267DSzL4uSTT8bCwgIOHjyIjIwMXHvttbj55pvx0EMP4fbbb8fXv/51HD16lM/SEIKUEAmx4Bs7ECtutxvt7e2YnZ2FTCZDdna2oDdLIS0f3EkMACorKwVrjhSOSPAskgSsrKAoBCKRiC/kU1VVxQc/Tk5Ooru7G2KxGAUFBbBarZDJZIK8/0gLOSW6aNNaEyQymQz5+fmQSCRwOp2orq6GwWDA2NgYOjs7kZGRwX8H1Gq1oCf/ZFkSwundsJazGzhEO04Cs21fwisuvvnmm/x3TqfTYdu2bfjlL3+Ja6+91u9nk5GRgQceeAC9vb0Qi8XYvXs33njjDf65b7zxBn70ox/hhhtuwMzMDPLy8nDqqaciNzcXwLLIuOGGG1BfX4+srCzccccd/OEOWHZLvPrqq7j77rths9lQWVmJP/7xj9i0adndcu+99yInJwf3338/+vv7+cq+d955JwCguLgYf/7zn3HbbbfhiSeewJ49e/DTn/4UN9xwgyDXiyEpUJrOM7c9UggheOutt3D66afHnFbH9TVgGAY7duzA4cOHUV1dLah/sLe3F3a7HZs3b456DK6R1Pj4OLZt24ajR4/ilFNOEeT0MT8/jyNHjuCss84K+BzfAEXONZMI5ufn+b4emZmZfAAky7J8imVmZmZcTqCeVgbOetLa2gq9Xs8HS8YzlqG5uRmZmZlYt25dXMb3xGg0oqenB/v27Yv7XMByRs7w8DBqaz/L8HA4HHwsg9FoBCEEOp2OFw2xfsYHDx5ETU1NQivrffTRR9i0aVPQUrlnnnkmbrvtNlx99dUJW1ck2Gw2DAwMoKysTLBUZkpiieQzTAlLQiwbDHdTjjUN0mAwoKmpCXl5eaipqeGL9ghdjCdWd4PT6URTUxNsNhv27duH9PR0QV0YoU7HkQQoCo2/DAYu+HFxcREzMzMYGRlBR0cHVCoVH8ugVqsFK+QEfJZ229LSArfbjdzcXP66APELfkx2nYR44s+9IZPJvAJcuViG8fFxdHd3Iz09nXdNRJNGmwx3QyRlmSmUVCAlREKsxGLC56Lje3t7UV1djaKiIkHGDUQsGzpXICk9PR11dXV8gSQhRUIwd0MsAYqxECqDgWEYqNVqqNVqVFRUwOFw8MGPw8PDEIlEXsGP4RaWCoTdbkdjYyMkEgl2794NqVS6wsoQr0JOqz0mIdh8wa6N52dcVlYGp9PJWxna2tp4SxInGsI54aZi4CIhBBaLBSqVKoGrolACc0KLBLfbjba2NhiNRuzevXuFCVDoIEMg+rXOzMygubkZRUVFKwIphRYJ/sYSOkAxXDwzGPbs3g2tSglg+fq5IQLxE2Evk8lQUFCAgoICsCyLubk5zM7Ooq+vD62trdBqtcjKykJ2dnbEjXQsFgsaGhqg1WqxadOmFZu/byyDkCmWqZfdIByRbthSqRS5ubm8FcdsNsNgMGBychI9PT1QKpV88KNWq/U7dqJFAvddoF0gKauJNSESoqmVYLVa+dNgfX29X/9mKlgSPNtnb9q0CQUFBSueE6/OjZ4tvOMZoBgIp9OJlpYWOBwOnFq/DwqpCMBn71MCFi6I4IIIgdLxuMp/er0eGzZswNLSEmZnZzEzM4O+vj7I5XLeyqDT6YLewOfm5tDU1ITCwsKQlSx9MyaESLFMfjGl1JyPYRioVCqoVCq+WJfJZILBYEBHRwfcbjcfy6DX66FQKMLesIWE+42G426gIoGSKqSESIj1xhfpZs5V+svPz0d1dXXAm5PQJZS5McPd0LlMC4PBELRAkpBR9txnwY2XrABFLog0LS0N9XtqIT9+Xz2+EuC4DUECFgQM3GHm7CsUChQVFaGoqAhutxtGoxGzs7Po7OyEw+HgKz/69pfg4iE2bNgQceCgkIWcTqSYhGiRSqXIyclBTk6Ol5VhamoKPT09UCgUSWkD7Cm0gz1nzZdlpqwqUkIkxEq4IoEQwpct3rhx44oKVtGOGwnhigSbzYbGxkYAyyU2g/lYhQ5cBD479SQjQJHLYFjuwbABMtFyl0Fv1wIDcvz/SsAeFwmRrVEsFiM7OxvZ2dn8ZuLbXyI7OxsulwtjY2PYunVrzJkukaZYegqMtVAnIdh88bBc+FoZXC4XjEYjZmZmAACHDh3yypiItaJqMLj0x2Dv02KxAACNSaCkDCeMSHC5XGhra8Pc3FzYZYvjld0Qaq1cp8nMzExs2rQppHlS6JgEYPl6cTe0ZGYwiBiAgQuB7CScUBCBBG9THALPzYQLjJudnUV/fz+sViskEgkmJyfhcrmQlZUFqVQa9VyeRFLIift7Ikj1mIRokUgkfEvfqakp7Nq1C0ajEdPT0+jt7YVCoeCDH7VaraDuiHBqJHAigbobKKlCSoiEeLsbLBYLGhsbIZPJUF9fH3YDoXhYEkIJj7GxMXR0dETUaVLImASO3t5e5OTkQK/XJ2SzCJTBwEDY9xUuIpEI09PTAID6+npeNAwODqK9vR0ajYZ3S4TqLxHJnIB/K4PVaoXNZgOwXD8g3uWiEy0SEh0DwW3YnDAsKSmBy+XiYxm6urrgdDqh0+l40RBrWmK46Y8ymUwwEUqhxEpKiAQgthbFwdpFc1kB69atw4YNGyK6EYlEIjidzqjWFGxMfxs6IQQ9PT0YGRnB9u3bIzJrC2FJ4E6uALB9+3bMzMygo6MDLpcLmZmZfMOlaDo0hiJYDwbOOuDZTsjvGDFYEXxxOBxoamoCAOzevZt/z/76S/T390Mmk/GCQa/XC3b65ESA2WxGS0sLcnJykJWV5RXTwD1P6EJOid60U8G9IZFIvNxPVquVL9Z17NgxpKWl8cGPoYJc/RGOSDCbzUhPT0/otaBQgpEyIiEW/J34CSHo7+9Hf39/wKyAaMaNFX8butPpRHNzM5aWllBXVxdx0FKsgYu+mw6XHsj56T2LFEXSoTEcPDMY/PdgWA5KFGM5LsE77mA5SiGaeIRAcFkvGRkZ2Lx5s9+bumd/CbfbzfeX6O7u5vtLcNcoVh+3yWRCU1MTiouLUV5ezl9v3xRLoQs5JcPdkMjTczh1GdLT05Geno7i4mK4XC7Mzc3BYDCgp6cHDocDWq2WFw3hpNJGIhJOFFi3G70HDmJhYhLq/DxUnlIPUYILXMWD/fv344wzzoDJZApaXbO0tBS33norbr311oStLVLWjEjwtCRw1fAWFxexd+/esLsD+pKI7AYu716hUGDfvn1R3ShjsSR4FgDy17iIM8eWl5fDbrd7dWiUSqW8oIjmZOWZwbB79+6ARY6cEIM5XhXB157AgoETwtxUuIDJvLy8sJt6icVi3orAnT5nZmYwNTWF7u5uKJVK/u+B8vUDMT09jba2Nr8ZFYFiGYTqYpkKJ/t4Emm1Rc9upNznzBVz6uvrg0wm44MfA/0Wwu0AKZT7KtVpfPWveOm7/4650XH+Me26Alz56M+x49LPx2XO6667DnNzcyu6Joa7qUfLs88+i1tvvRVzc3OCjx1v1oRIkEgksNvtAJaVOLfx1NXVxWQej1dMAjdmsAJJkRCtSIi0gqJcLkdhYSEKCwu9TtC+6YPZ2dkh6+p7ZzBUhdggGDgghvh4JgNzPNPBBZFgVoSZmRm0traioqICJSUlUY3hefrkIuk5c3VraytYlvVKsQz23RwdHUVPT4/fCpO+xKOQUzJiElaLKPH8nLlUWpPJBKPRiN7eXthsNt7KwMUycHFDtCTzMo2v/hW/vvwrgI8FdG5sAr++/Cv4+it/iJtQoERGYmuSBiHWdtFc28+PP/4YOTk5qK2tjdl/Hq/sBi4Vs6mpCRs3bkRVVVXM/SsiXWesJZa5E3R1dTVOPvlk7N27FxqNBuPj4zhw4AA++eQT9PX1YWFhYYUrZGpqCkePHkVpaWnQOhXeLNsS7JDABinskMAdpIhSJIyOjqKlpQWbNm2KWiD4QyKRIDc3F5s2bcKpp56KnTt3Ij09HSMjI/jnP//p9xpxbrLe3l7s2LEjpEDwh0gkglQqhVwu54PgPLululwuOBwOuFyugN+b1bRpJ3s+7rewYcMG1NXVYe/evcjKyoLJZMLhw4dx8OBBdHV1YX5+PuRYJ0IHSNbtxkvf/fcVAgEA/9ifbr0DrMAHtEg4ePAgTj31VL6uyne+8x0+8wQA/ud//ge1tbVQqVTIy8vD1VdfzQc6+7J//35cf/31mJ+f52OH7r77bv7vVqsVN9xwA1QqFYqLi/Gb3/yG/9uZZ56JW265xWs8g8EAuVyO9957T9g3HYA1YUkQiURYWFjA7OwstmzZgry8PEHGjYclgdsMBgYG/JaCjoZILAlcgKKQJZYZhkFGRgYyMjJQVla2oneCp0nebDZjcHAwrBNyvCGE4NixYxgdHcXOnTvjWmCHYRhoNBpoNBpUVFTAbrfDYDBgZmbGq7+E3W7H4uIifwOKlWgLOSUjcHGtzKdUKqFUKnkrAxfLMDMzA6fTiYaGBt7K4BukaLFY1nxMQu+Bg14uhhUQAtPIGHoPHETV6ackbmHHaW1txXnnnYd7770Xv/vd7zAzM4NbbrkFt9xyC5555hkAy8HN9957L6qqqjA9PY3bbrsN1113Hd54440V49XX1+ORRx7Bj3/8Y3R3dwPwTnF96KGHcO+99+LOO+/EK6+8gm984xs49dRTUV1djRtvvBG33HILHnroId46+/zzz6OgoABnnHFGAq7GGhAJTqcTIyMjsNvtqKurE7QIidC9G2w2Gx81v2fPHsFODOEGLgaKihcaf70TOP+62+2GVquF3W6HzWZLWqtZlmXR0dEBk8mEPXv2JPzGLJfLva6R0WhEV1cXn+bY09PDCyuh1hZJISehxXEoUj0mIVrEYjEvCEQiEex2OzQaDQwGAx/Xw2VLyGSyE6Ik88LEpKDPi5TXX399xTX2/L7/4he/wNVXX80HE1ZWVuKXv/wlTjvtNDz55JNIS0vDDTfcwD+/vLwcv/zlL7Fnzx6/liCZTAaNRsN3rfXlggsuwDe/+U0AwB133IGHH34Y+/fvR3V1NS677DJ8+9vfxv/+7//iiiuuAAA888wzuO666xJm6UsZkRDNG15cXOT7L6SnpwtepUxIS8L8/DwaGhqg1+sxPz8fcydCT8IRM55FerjXJAIuF31gYABpaWmorq7GwsICH9iXkZHBxzEI1dI5FFxGhdPpxJ49e0LGT8Qbt9uNwcFByGQy7NmzBy6Xi7fEHDt2DHK5nM+W0Ol0gn12gYIfHQ4H5ubmoNPp+JoMQqdY+rLWLRfA8ucsl8u9MmPm5+dhMBjw7rvv4pZbbsH69euhVqvR3NyMrVu3rskARnV+eJbecJ8XKWeccQaefPJJr8c++eQTXHPNNQCAo0eP4tixY3j++ef5v3NCemBgADU1NWhsbMTdd9+NpqYmGI1G/v47PDyMjRs3RrSerVu38v/NCQnOdSGXy3HNNdfg6aefxhVXXIGmpiY0NzevCLyMJykjEiJlcnISra2tKC0thVarRWdnp+BzCCUSxsfH0d7ejvXr16O0tBSTk5OCWihCiYRktXgGvDMY9uzZA4lEAr1ezzfi4TbDhoYG3uQuVEtnf3DlrtPS0lBbWxuXOSLBbrejoaEBcrkcO3bsgFgshkwmQ3FxMYqLi+F2u/ngx/b2drhcLuj1er/9JWKB2zC5zqhyuRylpaV8/RKhUyx9WevZFMDytfWMkxKLxXzzscrKStTW1uInP/kJ2tracPLJJ0OtVuPCCy/EU089tabEQuUp9dCuK8Dc2IT/uASGgW5dASpPqY/L/Onp6Vi/fr3XY6Ojo/x/syyLr3/96/jOd76z4rXFxcWwWCw499xzce655+J//ud/kJ2djeHhYZx33nlwOBwRr8c3o803xuzGG2/E9u3bMTo6iqeffhpnnXWWoLFToVh1IsGz6NDWrVuRm5sLk8kUF/NorCmQgQokCe3GYBgm4Do9LQjJ6sGQm5vrt5CVVCpFfn4+8vPzwbIs5ufn+e6Mra2t/GaYnZ0tSE19zvLEBVwmepPwhUt/1el02Lhxo9/1iMXiFc2KZmdnMT4+jq6uLt4Sk5WVxZs0o2VpaQkNDQ1Qq9VebbDDSbHk/jta1qq7wZNQ2Q1VVVXQ6XS44oor8JOf/AQffvghOjs715RAAACRWIwrH/35cnYDw3gLhePv9YpHHkhavYSdO3fyhzp/tLa2YnZ2Fj/72c9QVFQEADhy5EjQMWUyWdR7yZYtW1BbW4v//u//xgsvvIDHHnssqnGiJWVEQjg/BIfDgebmZthsNuzbt4/3/UTTKjocYslucLlcaG5uhsVi8VorILxI8DdePAIUI4HrwbB+/XoUFRWFnFskEkGn00Gn02HDhg2wWq18S+eenh4olUq+Gl40m6HBYEBLSwtKSkpQVlaW9Bvv/Pw8Ghsbw2o7zeHbX8LhcPBWhsbGRjAM42WJiaTmhsViwdGjR5GdnY3q6mqv9cQjxdKXE8XdEGpOrgOkTCbDmWeeiTPPPDNBq0ssOy79PL7+yh9W1EnQrSvAFY88kNT0xzvuuAP79u3Dt771LXzta19Deno6Ojs78Y9//AOPPfYYiouLIZPJ8Nhjj+Hmm29GW1sb7r333qBjlpaWwmw2491338W2bdv44NZw4QIYlUolLrnkkljfYkSkjEgIxcLCAhobG6FSqVBXV+dlJo5HFgI3LrfRRnJD4U6IXK0G35u10EWafAMXExWg6I9APRgiRalU8iZ3p9PJd+7jAj8j2QzHx8fR2dmJmpqaqCpvCo3BYEBzc3NMNRmA5dOJpyVmYWEBMzMzGBgYQFtbG99fIjs7O2ip34WFBTQ0NGDdunWoqKgIS9B5Whk8/0VbyGktZTcEIpJiSicCOy79PLZ94XMpV3Fx69at+OCDD/CjH/0Ip5xyCgghqKiowJVXXgkAyM7OxrPPPos777wTv/zlL7Fz5048+OCD+PznAwub+vp63HzzzbjyyithMBhw1113eaVBhuKqq67Crbfeiquvvjrhwd6rQiRwPv2ysjK/N7FoN/NQeN4Iwx13dnYWzc3NKCwsDNgrQuj6C56WhGQFKALBezDEglQqRW5uLnJzc0EI4d0S3Gao1Wr5wD7PTACuHsXQ0BC2b9+OzMxMQdYTCxMTE+jo6MDGjRuRn58v2LgikQharRZarRaVlZV8f4mZmZmg/SWMRiOam5tRXl4elWDxFQwAorIyJDomgQsiTCThlmU+UUQCsOx6SGSa47PPPuv38dNPP93roLV79268/fbbAce56qqrcNVVV3k95vl63/EA4Mknn1wRMDk4OLhibO4g5InJZILNZsO//uu/BlxTvEgZkeDvBsGyLHp6ejA6Oopt27YFPJV6FooRclP0HDdUgBt3gu7t7UVNTc2KMrqexMvdkEz3QugeDMLAMIzXZri0tMRvhlyr3+zsbGRmZmJychIGg0GwmgOxMjQ0hL6+voQIllD9JfR6PeRyOSYmJlBdXY3CwsKY5/SNTYjEykAtCctYrdaU+K5SUgOn04mJiQn84Ac/wL59+7Bz586EryFlRIIvXCc+h8MRsumR52YuZJMYbqMN5RpgWRbt7e2YnZ1FbW1tyKI88Qhc9PQTJyuDQaFQBO3BEA+4imhFRUVwuVwwGo2Ynp5GY2MjCCHIzs6G2WxGWlpa0trvEkLQ29uL8fFx7Nq1CxqNJqHz+/aXsFgs6O/vx9jYGIBl8WK1Wvngx3inWHLxMp5WBqErm4YiVUWCxWI5IcoyU8Ljo48+whlnnIENGzbglVdeScoaUlIkcEFdGo0GO3fuDLnpcCbMQO2iYyFU/IDdbkdjYyNYlkVdXV1YJ+h4iASLxYLZ2Vno9fqUymBIJBKJBBqNBv39/dBoNCgvL8fc3ByGhobQ3t4OjUbj5ZZIxHXiijbNzc1h9+7dSa+mxzAMjEYjZmdnsXPnTqjVaj7eo6WlJaL+EpEQrJATl53EMAxflyGaplSRkAyRECq7gRNw1JJA4fDntkg0KSMSuBv22NgYOjo6UFFREVEUejyDFwNt6JyY0el0AdsKBxpTiLVyJzIuK6Cjo4O/yXObYTxPz5FmMMQbLmBUq9XyKXyZmZmoqKjw8tH39fVBLpd7dbCMx4bhdrvR3NwMu92O3bt3J71oE9cXYmRkxMui4RnvwZU3Hx4e5luDc4JBpVIJ9hlzIsBgMKC9vR3V1dVQKpVe7gnuefEo5JSMFMhwLQnJFpIUiicpIxK4E9fExAR27NiBrKysiF4fT5Hgb9yJiQm0tbVFLGYAYSwJnqZbuVyOTZs28Tf5mZkZDA4Oor29HTqdjhcMQpkxhcpgEBKTycQHjPpLKfT10XOnZ65AkaewEuL07HA40NjYCLFYjNra2qS5OjgIIeju7sbU1BRqa2v9Bsf56y/BFbsaHBz0apes1+tjditxpbpramq8gjh9UyzjUcgp0ZYE7n2EE5OwWgIXk33CpURPJJ9dyogEq9WKxcVF1NXVRbWZxatWgq+7gfMvDw8PBw2mDDVmLCIhUICi501+/fr1WFpawszMjGC1BoD4ZTDEAmfR2LBhQ9CAUQ6xWMxfB0IIFhcXMTs7i5GREf70zAmGjIyMiK8TV5QoIyMDW7ZsSXrRJk+Xx549e8IuTOXZGpxlWT74sbe3F0tLSzEJ0MnJSbS3t2PLli0rfkOBYhmELOSUjOJNQPB1ulwu2Gy2lBcJnOC1Wq2CFDmjJB6uMmQ41rSUEQkZGRnYs2dP1ObMRLgbXC4XWlpaYDabVxRIioRYREIkJZYVCgVfa8DlcvGd6BobG/kSyFwmQDhflkRlMEQClzGwZcsWvqJlJDAMA7VaDbVajfLycq/TM9eAx9MtEeo6cVUdc3JyYm4BLgRutxutra1YWlqKyeXBuW4yMzNRVVW1otiVQqHgrQyh3DdjY2Po7u7Gtm3bQloM41XIKZxTvZBw96Zgc5rNZgBI+ZgEsVgMrVbL9xdQKpVJ/55TwodlWczMzECpVIZlDUwZkcD5HaMl3u4Gq9XK19jft29fTCbpaNfq6a+NNINBIpHwvmeuMyN3g+dS4rhTob/NP5kZDP7gSl5PTk4KmjHgeXr2TB3s7OyEw+Hgg/qys7NXbLgmkwlNTU0pU9XR5XKhqakJLMsK7vLwLHbFZZV49pfwDH70vE4jIyPo7e3F9u3bodfrI55XqEJOQqdLh4ILzAw2p9VqBYCUtyQA4LsZckKBsroQiUQoLi4O6x6VMiIBAN9MJhriJRJEIhHm5+fR2dmJgoICVFVVxXxzidSSwAUoeposY9mARCIR31iGK4E8MzODiYkJdHV1QaVS8eb4jIwMLCwspEwGA/BZEyKz2Yzdu3fHLWXMM3WwqqoKFosFMzMzfN8ElUrFC4alpaWIXB7xhouJkEqlfOOoeCGRSFb0l5iZmcHY2Bg6Ozv5/hIulwvj4+PYuXMntFptzPP6c0twgiGUlSEVe0VYLBakpaUlPKAyGhiGQX5+PnJycuB0OpO9HEqEyGSysL//KSUSYkEsFgueAkkIgd1uh9FoxKZNmwS7+UciEnwrKMZqcfGFYRikp6cjPT0dpaWlcDgcvBl5cHCQTy0NVkEykXD1M4DlqmhCpeiFgmEYZGRkICMjg++b4BnUx7IsX6AoGZHznthsNj4mYvPmzQn9zDz7S5SXl/PXaXBwEBaLBRKJBKOjo7DZbBH3lwhGsBRLf1aGRH9G4bg3zGZzwlJzhUIsFq8KUUOJnjUlEoS0JHDBXlarlY+KFwqRSBSW+k5GBUWZTIaCggLk5+ejv78fg4OD0Ov1vKWBOznHO73SH1arFY2Njfzml8ybE9c3gav4WFVVhaWlJa+Khty1SmTsBpcGmpmZiZqamqRvOFKpFGazGU6nE/v27YPL5eLjPbiS2py1RsgNMlghJ7PZzP+unE6nYBkTwQi3JDNNf6SkGiklElLF3WC329HU1AS3243s7GzBT6vhNHiKJEBRaLgMhtnZWezevRtqtZrPApiZmeGLEwXqmRAPuKJNeXl52LBhQ9I3P0IIOjs7MTs7iz179vB+ZE/3zdTUFLq7u3lze3Z2NtRqddzWvri4iIaGBhQUFITdWTKeEEK8vkecW0in03mV1J6dnUVfXx9kMhn/fQonSDRcPK0MZrMZra2tKCkp4esyRNuUKhLC7QAZTTYNhRJPUkokxIJQ7gauI55Wq8WWLVvQ3d0teMnYUA2eYglQjBXPDIY9e/bwp2DPLACuOBGXXtnb2ytIemUgZmZm0NraGnPXRKHgYiIsFgt2797tlQbm675xOp38RtjQ0MBnlXAdLIUKAOWCJktLS1FWVibImLFACEFHRwdMJhNqa2v9psp5ltTmald4Boly1pisrCxBUu0WFxdx9OhRFBUVoaKiAsDKFMt4FXKiJZkpq5U1IxIkEgnsdntMY0xOTqK1tRXl5eUoLy8HwzBxCYgMFJMgdIBipESSwZCWlubVM4FLr+TiBbgTYawb4ejoKLq7u7F582bk5uZGPY5QOJ1ONDU1gRCC2trakFYmqVTq1c6Z62DZ19eH1tZWL7dEtBshJ6JSJWiSZVmvwNJw0i59a1dwZcYnJyfR3d2N9PR0XjBE01+CEwjFxcUoLy/nHw+VYilUISfaAZKyWlkzIiGWzZwQgmPHjmFwcBBbt2712oxEIhFfeEIo/IkEz9MMd3pZLT0YPNMrCSF8euWxY8fQ1tbGF92JxD/PfSajo6PYuXNnyKZZicBms6GxsRFpaWnYunVrxOZwkUjEl9D2dEvEUuyKaz29adMmPi0tmbjdbrS0tMBut4clovzhGSTKWWMMBgPfhp0QwlfIzMzMDDnHwsICjh49GpaVJdJCTuH+TmhJZspqJaVEQjLqJLhcLrS2tmJhYQH79u1bUcgklGsgGnxFgq9ASHQGgWcPhuLi4pjGYhjGayPk0ga5E2FGRga/EQbqBcAFjZpMJuzZsyclbpxcQKBer0dNTY0gn5FSqURJSQlKSkrgdDr5UtGcNcbTLeEvSJSrORBOUaJE4Ha7+VieXbt2CRbYKpVKkZeXh7y8PBBCMD8/j9nZWT42husvwaXsen6n5ufn0dDQgLKyMpSWlkY0r5CFnMJtE50K33UKxZOUEgmxEE1MAlcgSSaToa6uzu+JJB7uBs8xkxmgSAjB4OAgBgYGoq5YGAp/6ZXcDV4ikfCCQa/X81kfLS0tcDqd2LNnT9KbIgGfNfIK1BdCCKRSqZc1hnNLeGYBeJZAHhwcxODgoGA1B2LF5XKhsbERDMOE1bk1WhiGgVarhVarxfr16/nGXf76S0gkEjQ3N6O8vFyQWJZYCjmFkwJpsViou4GScqwpkRDJZm40GtHY2Ij8/HxUV1cHPAGEk4kQKZwlIdkZDJ2dnTAYDAnrwcClVxYUFPC9AGZmZtDZ2Qmn0wmtVsungdXW1ia9qiMAzM7OoqWlRRArS7h4boSeWQBckChn3aqqqkqJ3hlOpxMNDQ2QSqXYtm1bQlNTPRt3efaX6Orqgt1u50sGW61WQYMCIy3k5HK5QlqfaEwCJRVJ/l3Yg0S5G4aHh9Hd3Y3q6moUFRWFHFdodwPDMGBZlr+RpEoGQyLx7QUwNTWFjo4OiEQimEwmNDY28laGZJlgx8fH0dnZmXR/P5cFUFhYiPb2dhgMBmRlZaGvrw+9vb0Jaw3uD4fDgaNHj0KhUGDr1q1JLbbFfadEIhHGx8dRXl4OiUSyor9EdnY2tFqtYGsNp5CT3W6HQqHgxYK/ua1Wa1SlqimUeJJSIiEWwhEJ3OmZq/cfzg9SaHcDIQRisRhWqxXt7e38zT1Rp+ZU68EALFt1Ojs7+cAyrskSlwWQlpbmFdCXiI2Ic8Ns374dmZmZcZ8vFCzLorW1FRaLBfv27UNaWppXa3DOP6/RaLxqV8RTfNpsNhw9ehRqtRqbNm1KejVOALzA9Mz0KCkp4ftLcJkgbrc7YH+JWPG1MgwNDWFubg4lJSVBUyxp4CIlFUn+DiEQoVpFc7XsXS4X6uvrw043E9LdwLkXMjIysGvXLszOzqK/vx9tbW38adBf4yChiCWDIV5wp/WamhoUFBQA8DYhu91uPr2yubkZALy6VwotcnwbR6WCOd/lcqG5uRkul8srY8C3NTjnn+fElVwu9+pgKRKJ4Ha5MHusH6aRMRCWhTo/D7nVlZBEmIWwtLSEo0ePQqfTYePGjSlRAMhoNKKpqQlVVVUoLCz0+ptvfwmuPTjXX4Lrw5GVlSVowavR0VEMDg7y3yVfK4NniqXRaKQigZJypJRIEMLdQAhZMQ5XiU6tVmPXrl0RbSxCuRt84w+4DIDKysoVjYPUarWXqV2IG5aQGQxCQAjBwMAAhoaGgp7WxWKx183dX50B7uQca8EdlmXR3t6O+fn5uDaOigSn04nGxkaIxeKQ311fccWdnLnOjOkSKWYOfgrL6DhY1/GuhGIRssrLsPPKS6EtLAhrTRaLBUePHk2ZdtgAYDAY0NzcjOrqal5sBsK3PbhnQO3w8LBXwSu9Xh+1C2dkZAR9fX3YuXMnLzYDpViOjIzg0KFDqKysjGouCiVeMCTaOshxgKulHg12ux3vv/8+zj33XK8T8tTUFFpaWlBWVoaKioqIb2jz8/M4evQozjzzzKjWBUSWwcCZ2qenp2E0GnlTe05OTlSVDBORwRApnmWfd+zYsSLtNFw86wzMzc3FVP7Y5XLxcRo7duxIiawKrlGTUqnEli1bog4IJIRgzmjE/l/9BiNNrZBl6ZGu0SA9Ix1yiQwLI6PIKi/FKd+4EWmq4IFzqVb6GYhMIISCa6POiQar1RpVf4nR0VH09vZix44dIbNPJiYmcN555+Hkk0/GU089lZQYIQolECllSYgF7gbqcrkgk8lACEFfXx+/OUYbeBaLu4GroBhJiWW5XI7CwkIUFhbypvbp6WmvSoY5OTnQ6/UhN41kZDCEgqtLYbPZYg6a9K0zwJnaGxoa+DbPXHplsGvFuaIkEknKZFVw6bk6nS7mugwMw2BpfArOaQM21u0FI5HAYrHAbDbDaDWCkUnQ19CMrA8PYuu5ZwW8VlwqqG/VwmTCZZ/U1NQgPz8/5vF826h7ZpZ4unCC9ZcYGxtDT09PWAJhcnISF1xwAerr6/G73/2OdlSkpBzJvxt6EKu7AVguWsJtRPPz834LJEU6bjTuBl+fYzQZDJ6mdu6EMzMzw3cazMzMRE5ODrKyslbUeEiFDAZf7Ha712YsZCS+b/ljLr2yq6sLDofDK+bD81otLS2hoaEBKpUq4W2VA8Gd1oVsZmUYHILb5YL0+PeAi2VgWRZLS0sYnVtA64GPYJSJ+YA+z/iYubk5NDY2ClZzQAi4IMSNGzfGLfsknP4SnLsrLS0N4+Pj6O7uxo4dO0JWCZ2ZmcFFF12EHTt24Omnn6YCgZKSpJRIiAXPCOHu7m5IJBLU19fH3MFRLBbzG364G4hnW1oAgmw8viccs9mMmZkZjIyMoKOjAxqNBjk5OcjOzgbDMCmXwcBVLNRqtXGPhPdNrzSbzV5BalzMh1KpRGdnJ/Ly8lLGt85txiUlJSgrKxNsTVwMgi8ikQjp6enIzMxEbkkJtu/d6xUfo1KpkJ6ejqmpKWzYsCFkynCimJ6eRmtra0J7evj2l+C+VxMTE+jq6oJcLofdbseGDRtCWhAMBgMuuugiVFVV4Q9/+ENK/EYpFH+sqW8mwzBoampCfn6+YKVzPS0U4YyXiAJJDMNApVJBpVKhvLx8RUdGAFCpVCgrK0uJ04nJZEJzc3NcKxYGwvNacemV3CY4Pz8PqVQKhmFgMpkEzZ2PBs63XllZKfhmrM7PBQjB/2/vzeOjLM/9/89kI2RPCBMIgWwsQiDJLAqugAtlSyYpVo9aK2rpafG4Va2tttWfVbTf1mPPsQLWpWqttUoWRAQRCQioQDLZA0kghOyZyT7J7PM8vz8898NMmDAzyczkAa/36+UfkmRyz5PkuT/PfV3X58PZbAgY9TvB8zzMBgOmpSYLmQmpqakwm804e/YsWltbIZFIcPbsWeh0OrdKOL6ECYQlS5ZAKpVOyhpG/14xARobG4umpiY0NTU5jFjan5oNDAxApVIhOTkZH374od+9LQjCE0QlEiayebS2tsJmsyE5ORkLFy702prsO5FdYe+05k+DJJbIGBISgv7+fiQkJIDjOKErfrT1sT9hUxViSSicMmUKgoODodPpcMUVVyA0NFQ4tuY4zmG80p837+7ubtTU1GDRokVeqa2PZmbGQsQkJaLnzFlMn5sGid3vQX9LK8KnxWHWksUOX9Pf34+2tjah4ZW5GbJyl32Cpb/KWew6TaZAGE13dzfq6+uRnZ2N+Ph4p/kSJpMJX3/9NVatWoVnnnkGUqkUH3/88YRPOgnC14hqugGAx3HPrFO+s7MTgYGByMjI8HoH/+eff47rr7/+oiNx9j7uYslgYLV5jUYDrVYrGMj4y53v3LlzOHPmjGimKoDzoUij12RvTKTVajEyMuKQXjnR8cqL0dbWhoaGBp9fp+76RpR+8DEGu7oR+n9BSEadDuFxcci+NRdzFDLhc1m5wdmaeJ53mCwZHByc0GSJu3R1daGurk5Uv0/sVCMzM3PMNRmNRnz77bfYsmULTpw4AYlEgh//+MdQqVS46aabRDFqSxBjITqRYDab4e6SzGYzKioqYDabIZfLUVlZidTUVK83Me3fvx9Lly512gDpzBRlsjIYsrOzx5xgGGsTZH0M3nwStDckys7ORnR0tNdeeyJrOnPmDNra2pCdne2yZmwwGIRr1d/fj/DwcEEweHMTtHd29Ecctk6jRVtFFbrq6sFxHKTz0pEky0Rs0nnzIft0SXfcJtlkCfvP3mfAW4ZXnZ2dOHnyJDIzM0WReAl4VvbQ6/XYsGEDeJ7HL3/5Sxw4cAC7d+9GdHQ01Gq1n1ZMEJ5zyYoEe4OkJUuWICgoCMePH0diYqLXj7UPHDgAhUJxwWbniwZFT7CfYJDJZB5t9M48BphgGB236wk2mw01NTUYHh6GTCYTxVOSvS+DXC73OETHYrEIro9sE7Qv4YynNs/zPE6fPo329nYHs53J5ty5c2hqanJrfM8ZHMcJhlfMZ8C+LDGeExl2quGuaPEHWq0WVVVVbgkEg8GA22+/HXq9Hnv37hV+1jzPo7+/n/IaCFFzSYoEjUaDyspKpKSkODTClZWVIT4+3usjWocOHcKSJUsc/pjtw1v8XV4AHDMYmEgaL8xxjt3Yp0yZImyCnjTzsZMdAMjOzhZFvdVms6G6uhp6vR5yuXzCJyb2o6harVaozXtiqc3zvHD6I5fLRWHFyxwwW1pavCpamBjt6elBf38/wsLCHHI4XP3diFEgMG+GjIwMl5MVJpMJd955J3p7e7Fv3z5RxHoThCeITiRYLJYxmwR5nhc6h50ZJFVUVAhWq97kyJEjWLBggVBznMyIZ+C77uiKigphjt6bJxhsFlyj0aCnpwc8zwtPgfHx8WM+Nev1epSXlyMiIgKLFy8WxVSFxWJxEC3e7sHgeV6w1NZqtRgaGkJUVJRwvZydyHAch5qaGuh0OigUClH4V7BTjY6ODsjl8gn5ilwMi8UiWEX39PQAgENZYvTPh/VqZGdni+Zpm02guOPNYDabcffdd6O9vR379+8XzXsgCE+4ZEQCeyIcGBgY80mnpqYGU6ZM8br/+ddff4309HRhamCyGhQB/2Yw2GclaDQaGI1Gp0/NLDjKm+Y/E8VoNKK8vByhoaHIzMz0i2gxm82CYOjt7UVISIhwrWJjY8HzPCorK4UeGjGctPA8j/r6emg0GigUCr+datj/bvX09Ag9Mkw09Pf3C66F/ujVcAcWIOWOu6PFYsF9992HxsZGHDhwQDR9FAThKZeESGBH64GBgcjOzh7zSPfkyZOQSCS44oorvLqmY8eOISkpCTNnzhQsmv3doCiGDIaRkRFhUoI9NYeFhaG7uxtz584VjRMfM26Ki4vzml+Gp9gHLLHJkoCAAISEhEAmk/l0WsJdeJ5HXV0d+vv7oVAoJnVN9vbHfX194HkeCQkJSEpKmnT/CuB8BLU7+RBWqxU/+9nPUFVVhZKSEr+ZPRGELxCdSLBarQ5ZCeyPUyqVYtGiRRe9WTQ0NMBisSAjI8Orazpx4gSkUqlwcxDrBIM/MZlMwgSDRCLxuNbsK1gpJikpaVyBXr7AaDSitLQUwHfmXCMjI4iJiRGu12Q0d7LES51O55VeDW/R0tKC06dPIzU1FXq9Hj09PeA4zq+ju6MZGBiAWq12GkE9GpvNhgceeADffvstDh48OOHAKYKYbERlpjSatrY2nDx5EvPnz8ecOXNc3vADAwNhNBq9ugae5xEUFIT29nahq92fNyk2wWCxWESTwcDzPFpaWtDT0wOlUonIyEih+7+8vNwr3f/jgZkiiSUOG/juCbmsrAwxMTGCyLV/am5sbPS7wOI4DlVVVTAYDFAqlaIoewDnJyvsJ4nsR3eZMVF0dLQgGLwVpT4WzCZ7/vz5LgUCx3F45JFHcPToUZSUlJBAIC4LRHmSYLFYUF9fj46ODmRnZ7vd1dzc3Iz+/n7IZDLXn+wGrEHRYDCgs7MTWq0Ww8PDPvMXGI03Jxi8BXsCZb0ho2vY9t3/Go0GFovFIVzJVwKro6MDJ0+eREZGhs/CfjxleHgYarUaUql0zGwIq9XqMF4JQLhW06ZN87rAstlsqKyshMVigVwuF40lMCulyeXyi/pqGI1Gh7IES2VkfR/eLEsMDg5CrVZj7ty5Lm2yOY7DE088gT179qCkpASpqaleWwdBTCaiEwksItdkMkEul3t0FNva2oquri5ceeWVE17HWA2KBoMBGo0GGo0Gg4ODQliQVCr1atOX/QSDWMKH7E81ZDKZy3E/FoLD+hiGh4cRExMjCCxv1MBZr0ZzczOysrJE00HOYpVnz56NtLQ0t35+9h4DWq3WoVGUpQxOBKvVioqKCvA8D5lMJgrRCQBnz57FuXPnPB69tO/76OnpgdVqdShLTOSEZGhoCGVlZUhPT3d5KsVxHJ566ikUFhbi4MGDmDt37ri/L0GIDdGJhLq6OgwODiIzM9Pjm1hHRwdaWlqwbNmycX9/nueFEwTg4v0HrJtdo9Ggr68PU6dOFTbAibjy+XOCwV3spwXGe6rhzMWQXa/IyEiPr5e9s6NMJhNFrwZwvgt+oj8/+/HKwcFBREZGCqcMnhpeWSwWh+ZfMYyoAkBTUxNaWlqgUCgmNHrJ8zx0Op1wyqDT6QQBHx8f79H1YgLBnVhsnufx7LPP4v3338fBgwexYMGCcb8HghAjohMJFosFNpvzWFtXaDQaNDY24tprrx3X9x7toOhJgyI7Nmb+AkFBQcIJg7vd2WKYYHCGTqdDeXk54uPjccUVV3jlSJdZ+bKnwODgYIdxQVffg5U9BgcHPT5x8iUajQY1NTVudcF7gr3hVW9vr0fXy2w2Q61WCwJPLALhzJkzaG1tnbBAcIbJZHIoSwQHBzuUJca6BjqdDmVlZUhJSUFKSspFvwfP83jxxRfxt7/9DSUlJV5vmCYIMSA6kWCz2YQkRU/p7e1FbW0tbrjhBo+/1psGSRzHCYZEWq1WMCSSSqVj1pnFOMEAfHdNq6qqkJycjNTUVJ+UPdj1sh8XtDdwGn1qYbVaHerqYmm8Y+6Aixcv9mlCoc1mQ39/v3C9LnbMbjQaoVarBZOryR4lBM7naLS3t0OhUHhsk+0p7Hox0WA2mx2uFyubDQ8Po7S0VPhdd/UeXn75ZfzP//wPDhw4gKysLJ++B4KYLEQnEjiOg8ViGdfXsk7klStXevR14xUIwz29aCw5jIH2DgSFTkHKVQrMys50uBEz0xjWx2AymYQNkDXy2df6s7OzRTHBAJxvBly4cKHfOrXtu9k1Go3g/c+uV0BAANRqNYKDg5GVlSWaujpLvPS3OyA7ZmeCYXh4WOj+j4qKQl1dHWJjY7Fo0SJR9LXYuzv6QyA4+/7Dw8OCYBgaGkJkZCSioqLQ1dWFOXPmID093eVr/O///i/+9Kc/Yd++fVAqlX5aPUH4n8tKJOh0Ohw7dgw333yzR99vPA6Kp74oQcn/bIWuWwvgu0sYHBqK1GuWYvXTTyA0ynli5OhGvujoaOj1eoSHhyM7O1sUmx7z8T937hwyMzMn1TNfr9cL12tgYAASiQTh4eHIyMgYVx+Dt7FPl5TJZJOeeGk0GqHVatHV1YWBgQEEBQVh1qxZQg7HZF4ve4GgVCpFkVlhNpvR3t6OpqYm8DyPkJAQQcQ7G9/leR7bt2/HH/7wB+zdu3dC/U8EcSlwWYkEvV6Pw4cPY9WqVS5vhp40KI6mraIaRY8/DYvRiJikRAT8343EqBvGUEcXFq1dhXX/31MuX6+7uxs1NTUIDg6G2Wz22aSEJ9gnJspkMp/5+HvK0NAQ1Go1IiMjERgYiN7eXiGIivV9+HsDtLc0Hk+6pK8YHh5GWVkZEhISEBsbe0FWAhuv9KcgZU2m3d3dfrV/dsXIyAhKS0uRmJiItLQ0DAwMCKcMLLyLlf7S0tLw9ttv47e//S12796N6667bpJXTxC+Z/IfW0cxkRt9YGCgsPlf7HVYgiOzf/bUQbH6k89gGBjE9PmOjn6hkRHgpPE4c/gb9DQ1Y3r62HVNNsEwb948zJkzx2FSoqmpyWuTEp5gtVpRXV0No9EoGuMm4LtpgcrKSqSmpiI5ORkSiQQ2m03wF6isrATgW3+B0dg3Tl555ZWisFkGzosp+9HLhIQEh6yEM2fOoLq62qGM48ufNRMIGo0GSqVSNE2mer0eZWVlmDlzppAmO23aNEybNg3z588XEiy/+uorPPDAA5gzZw66u7vx5z//Gddcc81kL58g/ILoThJ4nofZbB7X11qtVuzfvx833njjmM1s9v0HEonE40Yum8WC11X/AZvFgkjphdMHPM9De7oJq558FNkbVE4/7mqCYfSkRGBgIKRSqUeTEp5iMplQXl6OoKAgZGVlicZkp6urC7W1tRfti+B53sHAyWQyORg4ebux0WazoaqqCkajEXK53K14aH/AenJSU1NdduaPjIwIT8wDAwOIiIgQrpc3yzjstEWr1UKpVIpGTBkMBpSWlkIqlboMJeN5Hm+99Ra2bduGmTNnory8HCEhIVCpVHj99dcnveRFEL5EdCcJE4E9PdpnP9jjjQkGnuPAczwkY2zUEokEEkjA2S5MsrSfYFAqlWNOMAQFBSEhIUFInWSTEtXV1W5NSngKC0SKiYlBRkaGKDrggfM+/pmZmRcdB5VIJIiNjUVsbCzmzZsn+AswW2/WyDd9+vQJH3Pbx08rlUrRiCnmzTBv3jyX7oAAEB4ejvDwcCQnJ8NsNgunMufOnRPGd12NC7qC53mcOnVK+H0Xm0CYPn26W6mlhYWFeOqpp/DRRx9h7dq1sFgs+Prrr1FbW0sCgbjsEZ1ImMgfnUQiQWBgoFOR4K0Rx8CQECQsmIfmYycQEX9hQ59pZASBISGIT09x+PfxZjAEBAQI8bn2kxINDQ1OJyU8pb+/H5WVlZg1a5Zw5DrZ2DcDyuVyxMTEuP21EokEERERiIiIQGpqqmDjq9FocPr0aSEnQSqVelzGYX4DISEhyMrKEo3fQE9PD6qqqsbtzRASEoKZM2di5syZ4DhOGK88efKkg622Jy6GPM/j5MmT6Ovrm/SESXuMRiPKysoQHx/vlpPpJ598gp///Of44IMPsHbtWgBAcHAwli9fjuXLl/tjyQQxqYiu3AB8d/Q9Xg4cOHBBQIx9D8JEPRAAoH7/Qex+dgumRIQjfNr5cTeb1YrepmbMlmfj9m2vCE/kvshgYJMS7Ih9PJkSrC9i/vz5SEpKmvCavAE7benr63OaDTERRpdxWBCVVCpFXFzcRU9QDAaD0DgpFr8B4Hzzqy8yK+x/x5iLof2pTFhYmNO/JfsIaqVSKZreFpbGySLEXd0HPvvsM9xzzz149913ceutt/pplQQhLkQpEsxmM8a7rEOHDmHJkiWIi4ubcIPiWPA8j8Pb3kTZhwWwGIwICZ8Km9kCm8UK6fx05Lzwe0xL+c7O1V8ZDJ5mSrC5fjE5O9rX+mUymU83F/snZq1WC4vF4mDgZH8qw8oxzHFSDKctANDZ2Ym6ujqX5Rhv4SxciQkG1ivD87zQ0KlQKEQjEEwmE0pLS4U0Tlc/w/379+POO+/EG2+8gTvuuMNPqyQI8XHZiYQjR45gwYIFiI+Pd7BY9vaTH8/zOHdCjZOffwlt4xmEhIdh3orrccXNK4TThcnKYLhYpkRkZCQaGxvR2dkpirl+BssWkEgkyM7O9mut35khUWxsrBBCVVtbi6SkJKSnp4tGILS1taGhoQFZWVmT4mNhP13S09MDjuMQHx8Po9EIs9kMpVIpmoZOk8mEsrIyREVFISMjw+XP8NChQ7jtttvw17/+FT/5yU9E8zMniMngshMJ33zzDVJSUjB9+nSvlRc8RUwZDPZH7MwiOiAgAAsWLMCMGTNEcWzOrIPDwsJEkS3Agqg6Ojqg0+kwZcoUzJo1C1Kp1ONgJV/Q0tIiuDvGxsZO6lqA89MldXV1MBgMACCILG+lfY4Xs9mM0tJSoUzk6md39OhRbNiwAS+//DJ++tOfTvrPmiAmG9E1LgLflQXGKxICAwMFkTEZAsHdCQZ/wSYlYmNjodfrYbVaER0djcbGRjQ0NHh9UsJThoeHHY7yxSBapk6diqlTp0Kv12PBggUIDg6GRqPBuXPnhGAlX46jXoyzZ8+iubkZcrlcNKdAPM+jtbUVAQEBuOGGG2C1WoWyRENDA8LDwwXB4C/PD+A7gVBWVoaIiAi3ThCOHTuGW2+9FS+++CIJBIL4P0R5kmCxWIQ+AndhDYr19fVoaWlBXFyc4C3grwAgsWYw6PV6lJeXCyE/zHTKVaaEr2Fz/XPmzBGMf8QAq/UvXrwYCQkJwr/bbDaHICp2xM5Eli8dDO0nPnyRmjheOI5DdXU19Ho9FArFBX9r9mmfvb29QrPoWLbH3sJisaCsrExoFnYl5srKypCbm4tnnnkGDz/8sGh+FwlisrksRMLoBkV2XKzRaDA0NITo6GgkJCRAKpX6bOP2xQSDNxgcHBQaJ8eaCed5HiMjI4JgGM+khKdotVpUV1e7PdfvL5g3g6taPwuiYmUcFkTFrpk36/FitTTmOM7BVMqVGOc4TjC9YrbHztIYJ4qnAqGyshLr1q3Dr3/9azzxxBMkEAjCDlGKBKvVOqYh0miY/8FYDYos8Eaj0aC/vx+RkZHCCYO3brb+mmDwFLYRp6enIzk52e2vGz0p4e1r1t7eLkQq2z+pTyb2oVYymcwjbwYAgoGTVqt1mC5hBk7j/Z1gfgO9vb1QKBSisTTmOA6VlZUwm82Qy+UenzwxYcqu2dDQkFeumcVicfCycCUQamtrsWbNGjz88MP47W9/K5q/XYIQC5e0SGAnCDabza3+A/uu/97eXoSHhwub33gb0iZrgsEVbW1tqK+vn/BGfLFJCU/ry6yhs7m5GVlZWX6NVL4Y7Em9q6vLK/HF+uFhlH+6B7Wff4mB1naExkRj3sobIFethXTWLLevGcdxqKurE904oc1mQ2VlJSwWy7gEgjNMJpNDWcLZeKUrrFarQ4y4q685deoU1qxZg02bNuEPf/gDCQSCcMIlKxIm6qDIaqXMWGfKlClCScKdzU9MEwz2sDjetrY2r3e/TyRTgvWLdHd3Qy6Xi6qmXldXh4GBAcjl8gk/qVuMRny+5WU0HDgEnuMRPHUqjCMjMJuMCE+ejUX33omkuemCgdNYNXn7Wr+Y8iGYQLBarZDJZD7pXRmr94OFdzn7nlarFeXl5QgMDHTLDbOxsRFr1qzBj3/8Y7z00kuiaJglCDEiSpFgs9lgtVrH/DjrP/DWiCOb+Wb1ZfvNLzY29oLXt59gEFOcMksmZBueL2vX9pkSbLRyrEkJjuNQU1MDnU4HuVwuGotem82G6upqGAwGr23Ex//xIb7a+gaiE2dgit2JhNVkRk9TM+betBwZG+9wqMlLpVIHy2NmKmUymdyq9fsLm82GiooKcBwHmUzml74b1vvBBMPIyIgwXhkfH4+wsDDYbDao1WoEBAQgOzvbpUA4e/YsVq9ejR/+8Id45ZVXSCAQxEW4pESCrxwU7bHf/DQaDQAIgiEuLk64gYttgsF+skImk/n1yfNikxKxsbGoq6sTnjzFsuFZrVaHDc8bT8RWkwnv3fNz6DRaxM6edcHHR3r7wFmsuOvt1xCdOFNoFrW3PJ42bRq0Wq2w4YklQMpms6G8vBw8z/tNIDiDNSVrtVr09/cLIiE4OBhKpdLlulpaWvCDH/wAa9euxWuvvUYCgSBccMmIBFcNir6AdWOzzc9qtUIikWDq1Kl+34gvhtFoRHl5OaZMmYLMzMxJnaywn5To7u7G8PAwgoODkZqaioSEBFGIKrPZLMRiu/Pk6S5951rw/n0PYGpMlMMpAoPjOPSePgvV/3sOc6+/xuFjRqMRXV1dOHv2LKxWq9Av429vAWewo3yJRAKZTDbpZlcM5qRo74tiX5YYvc6Ojg784Ac/wMqVK/G3v/2NBAJBuIE45vRGMfqG6O2AJncJCAhAXFwc4uLikJCQIIw4Wq1WHDlyRDhenz59+qRtzDqdDuXl5aIxI2IpjAEBAejo6BAaz7RaLRobG30yXeIJzN0xPDzcrfE4TwgICoIkIMBpTDgA8DYbJAESBDo5HQgICEBXV5eQLcByJdRqNQIDAx28Bfz5M2YCwd2jfH9hs9lQW1uL4OBgXHXVVQgICBDGK1lCalxcHHp7ezF37lyEhoZi3bp1uPbaa/H6669P+t8JQVwqiPIkgeM4WCwWAN6LeJ4IXV1dqKurEyYYWDoeO2EYGRkRasvTp0/325F6b28vqqqqkJycjNTUVNF0Zw8NDaG8vBwzZ87EvHnzhHV5c1JiPLCgpri4OLdCfjyF53l89MBjaK+sQfzc1As+PtjRiSkREfjJu68jNOp8Hwt7InYmXOyDqDQaDWw2m4O3gC/LEWxaIDAwUFQCgY1fsumK0QLdfrzyt7/9LT755BPBJfOdd96BTCYTzd8KQYgdUYsEbzcoeoq7Ewz2RkQ6nU4wIpJKpT4rSXR0dODkyZNYuHAhEhMTffI9xkNvby8qKyuRlpaGlJSUMT9vIpMS40Gn00GtVl8gXLxN/ZeHsPf5/wdJYACiZs4QkhFHevtg6B/Atf95H5bdc6fw+UajEWVlZYiOjsaiRYtcTojodDqhj4E18fnC9IoFbgUFBbk1LeAvPPVn6O3tRX5+vjBS+cUXXyA+Ph5btmzBXXfd5adVE8SliyhFgs1mg9lsFsYgfdGg6IrxTjCwxqru7m7BVIeNVnqjq9/e9CczM3NSEgDHoqurC7W1tR4LF08mJcYDs39OSUlBauqFT/jehOd5VBTsxLdvv4+Rvn5AIgF4DlMiI7EkZw2u/8X9QrlBr9ejrKxs3BHUzPRKq9ViYGAAERERgmCYSBCVvSFRZmamqAQCm0ZRKBQuBcLAwADWr1+PpKQk7NixAyEhITAajSgpKUFCQgLkcrmfVk4Qly6iFAkvvfQStFotcnNzIZfL/X6T8lYGg8lkcjheZzdxZt7kKRzH4dSpU+jp6RHV6CVw3s44MzMT8fHx434db2dK9PT0oKqqCvPnz0dSUtK41+UpQ10anDn8NYZ7ejAlIgJp1y5DfFqK8PHh4WGUlZVd1C7bE8xms4MZUUhIiMdmRMB5S+MpU6a4ZUjkLzwVCENDQ8jNzcW0adNQVFQkioZZgrgUEaVI2Lt3L/7+979jz549iIuLQ25uLvLy8nDllVf6XDDo9XpUVFR4PYPBYrE4uD2yerxUKkVkZKTLTcJqtaK6uhpGoxEymUw0Nz1m3tTe3g6ZTObVZMKJZkqwk42MjAzMmDHDa+uaKDqdDmVlZUhKSkJ6errXT8lGmxGxkxnWxzDW35DZbIZarUZoaCgyMzNFJRBqamowMjLiNERqNMPDw8jLy0NYWBh27dolGl8OgrgUEaVIYOj1euzbtw8FBQX49NNPER4ejpycHOTl5eHqq6/2+kSBvzIYWJQuq8eHhIQIgiE6OvqC72symRzqw2KZnWclmb6+Pp+bNwEXHq9fbFKira0NDQ0NEz7Z8DaDg4NQq9V+KX0A509mmEA1Go2Ii4sTThlYzwyLVQ4LC/P61MdE4HleMOJSKpUuBcLIyAg2bNgAiUSC3bt3T9himyC+74haJNhjNBrx5ZdfoqCgAJ988gmCgoKwfv165Ofn47rrrpvwxjl6gsFfsKc+9rQcEBDg4PZoMBigVqsRExODjIwM0dy8manUZJ1sXGxSore3d9xBTb6kv78fFRUVSE9Pn7ScD9b1zxJSo6KiEBcXh66uLkRFRWHx4sWi+R3jeR61tbUYGhqCQqFw2QRsMBhw2223wWg0Ys+ePYiKivLTSgni8uWSEQn2WCwWHDx4EDt27MDOnTthtVqxfv165OXlYcWKFR6NIIopg4GNuzHBwEY/p0+fjsWLF4umgcxsNqOiogISiUQUroBsUqK7uxsajQY8zyMhIQFJSUk+mZQYD6w3YsGCBZg160I3xsnAZDKhs7MTTU1NsNlsCAsLE4SWsxMtf8LzvJCpoVQqXQoEk8mEO+64A319fdi3b5+oxCFBXMpckiLBHmZstGPHDhQXF2N4eBjr1q2DSqXCzTfffNEnXLFmMADn6+kxMTHQ6/WwWCzCrPfF6sq+xmAwoLy8HOHh4aISLmxT6e3tRVpamuD37+1JifGg0WhQXV2NRYsWYebMmX7//mNhP365YMECoY+hp6cHEonEwcDJn9eNxWP39fVBqVS6PKUym824++670d7ejv3794smXZQgLgcueZFgj81mwzfffIOCggIUFRWhr68Pq1evhkqlwqpVqxzq1iaTCTU1NaLLYACAc+fO4cyZM8LJhv18fHd3N4xGI6ZNm4aEhASfG+rYMzw8DLVajenTp49rZM9XsM73kZERyOVy4Wfp7UmJ8cDE3pIlSyCVSn3+/dyFCQTm8Gj/s2R25KwsYbFYBAMnX183nudx6tQp9Pb2uiUQLBYL7r33Xpw+fRoHDhwQVf8JQVwOXFYiwR6O41BaWoodO3agqKgIHR0dWLVqFVQqFdLT03Hvvffid7/7HfLz8yc168AenufR0NCAzs7OMScFnHX8x8XFCX0MvnJ7ZPX0OXPmIC0tTTQCwWq1OkQXj/X+JzopMR7a29tRX18vuuZJo9GI0tJSxMbGunSeZO6iTDAMDw8jJiZGuG7enBxgceJarRZKpdLla1utVmzatAnV1dWC9wFBEN7lshUJ9jCXth07duCf//wnurq6IJfLcc899yAnJ2fS66/Ad6cgNTU1GB4ehkwmQ1hYmFtfp9frhY1vaGhIuIFLpVKvbXwajQY1NTV+9xpwBXMFDAwMRFZWlkdiz5NJifHAfCOys7NFdfxtMBhQVlaGuLg4LFy40OPfe6PRKAiG/v5+hIeHC2Uwd0Z5x4IJZI1G45ZAsNls2Lx5M44dO4ZDhw6JqoxDEJcT3wuRwPj3v/+N+++/Hw899BBCQkJQVFSEU6dOYcWKFcjLy8P69esRFxfnd8HAGgEBIDs7e9ynAUajURAMbONjbo/uio7RsFHCxYsXi+q43GQyQa1WC34WE6mZeztTgjXCim26wmAwoLS0dNwOj6OxWCyCgVNPTw+Cg4OFkkRsbKzbDaM8z6OxsRFdXV1QKpUuf1c5jsNDDz2Er776CiUlJZg9e/aE3gdBEGPzvREJ3d3dyMrKwltvvYV169YBOP/0UlBQgMLCQlRWVuL6669HXl4ecnJyIJVKfS4Y9Ho9ysvLERER4dVGQPuNr7e3V4geZm6Prt6Xvf1zdnY2YmNjvbIub6DX64WxUFd5B54ykUwJnufR1NSE1tZWyOVyUY3gMQvo6dOn+8QDhFlrMwMnm83mYOA01ikPM+Pq6OiAUql0eYrDcRwef/xx7N27FwcPHrxoPghBEBPneyMSgO9ulGM9pbBNkQmG0tJSXH311VCpVMjNzUViYqLXb6yDg4OCeZM3rHnHgj3xsY0vNDRU2PicPSmz5jGtViu6qQ9v2xlfDE8yJdjTcGdnJxQKhahMfPR6PUpLS5GQkODzawZ8dy3YdAkLorI3cLIvg505cwZtbW1uC4Tf/OY3KC4uRklJCebOnevT90EQxPdMJLgLz/NobW1FYWEhCgsL8c0330CpVAr20HPmzJnwjVar1aK6uhrp6elITk720spdY7PZHARDUFCQw5Mys8AdHh6GXC4XlaUtC2qajGjsi01KxMfH48yZM+jp6YFCoRh3accXjIyMCKLKl+mXF0Ov1wunWoODg4iMjMT06dNhMpnQ3d0NpVLpUlRxHIdnn30WH3zwAUpKSrBgwQI/rZ4gvt+QSHABz/Po7OxEUVERCgsL8dVXXyErKwsqlUqYlPD0xtvW1ob6+nosXrx4Ujuy7Z+UNRoNgO8SN4OCgqBQKEQ1FsoiqP3tiOmM0ZMSOp0OAQEBSElJwaxZs0Rz3UZGRlBaWorExETMnTt30ptzgfNBVM3NzRgZGcGUKVOEvpmYmBina+R5Hlu2bMGbb76JAwcOICMjYxJWThDfT0gkeADP8+jp6REEw4EDB7Bw4UKoVCrk5eW5rPWy+mtbW5vo6vxGoxEnTpwA8J14YE6Pk2lCxOju7kZNTY3ozIjYqcvQ0BASExPR19fnk0mJ8cDKMrNmzfJJiNREaG5uRnNzM7Kzs4XeGa1WCwBCSYL9zvE8jz//+c949dVX8eWXXyIrK2uSV08Q3y9IJIwTnufR39+PnTt3orCwEF988QXS0tKgUqmQn59/QUMdx3Gora3FwMCAX8KQPGFkZARqtdphbt7+aN1sNgu1+Is1ofkC5jUw2ZbZo2HZFSaTCXK5XJhI8fakxHgYHh5GaWkpZs+eLSpPC+A7o7CmpiYoFAqHxk77co5Wq8Wrr76Kvr4+JCcnY/fu3fjyyy+hUCgmceUE8f2ERIKXGBwcxK5du1BYWIjPP/8ciYmJwgnD7Nmzcc8992Djxo1QqVQufej9yeDgIMrLyzFr1iynR9LMTIe5PRoMBsTFxQluj74ybwLOjxJmZWWJymvAZrOhoqICNpsNMplsTAfCiUxKjBcWQ81Mr8RES0sLzpw5c4FAGA3P86isrMTLL7+MkpISDA0NYdmyZVCpVPjhD3+I9PR0P66aIL7fkEjwAcPDw/jss8+EiOvQ0FAkJibipZdewvLly0UROAScr/N70jw5uhbPXAulUqnXxA8ry7S3t4tulNBqtaK8vFwIt3L3VMWTSYnxMjQ0BLVaLTR2ionW1lacPn0acrncqZOoPTzP46233sLvfvc77N69G+np6di1axeKi4shk8nwwgsv+GnVBEGQSPAhVVVVWLNmDRYtWgSpVIrPPvsMERERyMnJQV5eHq6++upJq/V3dnairq5uQnV+5lrIutajo6MFwTDeqQg2ftnT0yO6sozFYoFarUZwcDCysrLG/bPzRabE0NAQysrKkJqaKjrvgLa2NjQ2NrplLsXzPP7xj3/giSeewK5du7BixQq/rJEgCOeQSPARAwMDSE9Px6OPPoqnn34aEokERqMR+/fvR2FhIXbu3ImQkBCsX78e+fn5uPbaa/0W1MQCpLKysjBt2jSvvKbJZBI2vf7+fkRERAhd6+5u9KwRUKfTiW78kjk8hoWFYcmSJV47DfJGpsTg4CDUajXS0tL8Ok7rDqynRCaTuWzU5Xke//rXv/DII4+guLgYN998s59WSRDEWJBI8CGNjY2YN2+e049ZLBaUlJSgoKAAxcXF4DgO69atQ35+PpYvX+6TWj8z/Ono6BgzQMobWCwWB7dH1ryXkJAwptujzWZDZWUlzGazQyOgGGCJiVFRUcjIyPBpuchgMAjXzp1JCeYdkZ6ePumjoaPp6OjAqVOn3BIIALBjxw5s3rwZH330EdauXeuHFRIE4QoSCSLAarXi8OHD2LFjB4qLi6HX67Fu3TqoVCrcdNNNXpm75zgOdXV1GBgYgEwm89sxvtVqdTBvCgkJETY9FqxlsVgcsiv8daLiDhMNRJoIriYlWNPp3LlzRZdf0NnZiZMnT7odcLVz50789Kc/xb/+9S/k5ub6YYUEQbgDiQSRYbPZ8PXXX6OgoABFRUUYGBjA6tWroVKpsGrVqnG5+bGndDauN1nTFTabTej212q1CAwMxLRp09DX14fw8PAJ1fl9AXMr9Jed8cUYPSkhkUhgtVqRlJSE+fPni6YZFgC6urpQV1fndjlr9+7d2LhxI9577z1s2LDBDyskCMJdSCSIGI7jcOLECezYsQNFRUXo6urCLbfcgry8PKxevdqtTAWz2ewQpyyWp3SO49DV1YVTp06B53kEBgY6mDdN9qan0+mgVqtFaUbU09ODyspKREVFQa/X+2RSYrx0d3ejtrYWmZmZiI+Pd/n5X3zxBe666y688cYbuOOOO/ywQoIgPIFEwiUCx3GoqKgQBENzczNuvvlmqFQqrF27Vji6t8dgMECtViMiIsKrzXbeYHh4GGq1GlKpFPPnz3fo9rdYLIJgiI+P9/umx47xxeg10NfXh4qKCixYsACzZs3yyaTEeNFoNKiurkZmZqZbxlcHDx7Ebbfdhq1bt+Luu+8WlRAjCOI7SCRcgvA8j5qaGkEw1NfXY+XKlcjLy8O6desQFxeHEydO4OjRo1izZo1PooEnAtuEk5KSLnhKZwmCbNMzGo0Obo++3vT6+/tRUVEhykkB5mtxxRVXIDEx8YKPe2NSYrx4KhCOHDmCDRs24JVXXsH9998vqt9PgiDOQyLhEofnedTX1wsR19XV1VAqlaisrMS9996Ll156SVQnCOxJ2B0DJ7bpdXd3Q6PRCJHDrPHR2xMQbBOeP38+kpKSvPraE6WnpwdVVVVYuHCh274Wnk5KjBetVouqqiosWbIEUqnU5ed/++23yM/Px5YtW7B582YSCAQhYkgkXEbwPI9t27bh0UcfxaJFi1BTU4Orr74aeXl5yM3NxcyZMyf1hsyeNhcuXOj0SdgVer1eeEoeGhpCTEyMsOlN9CmZRXd7sgn7C7a2RYsWYcaMGeN6DWeTEqykM5FMCSZeMjIy3Eo0LSsrQ05ODp599lk8/PDDJBAIQuSQSLiMePvtt/HQQw/hgw8+QE5ODlpaWlBYWIjCwkJ8++23uPLKK5GbmyvkSfjzBs1m5hcvXuzW06YrjEajIBgGBgYQFRUlCAZPJ0BYyuRkR3c7gwkEdzdhd/BWpgQ7eXFXvFRWVmLdunX49a9/jSeeeIIEAkFcApBIuIw4dOgQgoKCcO211zr8O8/z6OjoECKuDx8+jKysLOTl5UGlUvk8KbClpQWnT592e2beU8xmsyAY2DglM28KDw+/6Htj4kVsKZPA+ZMXX4qX8WZKsLKRuycvNTU1WLt2LR555BHBgZQgCPFDIuF7Bs/z0Gg0KC4uRmFhIUpKSrBw4UJBMHizyZHneTQ1NaG1tdWnDo/2WCwWB/Om0NBQ4Sl59LF6a2srGhsbfSZeJgI73XC3zu8N3J2UYAJhrAbK0Zw8eRJr167Fz372Mzz33HMkEAjiEoJEwvcYnufR39+PnTt3oqCgAPv370d6ejpUKhXy8/OxcOHCcTc9soZKjUYDuVyOiIgIL6/eNTabzUEwBAUFCYJhYGAAzc3NboUO+RvmNTCZpxtjTUpERESgra0NV1xxBWbNmuXydRobG7F69Wr85Cc/wYsvviiqJlqCIFxDIoEQGBgYwK5du1BYWIjPP/8cSUlJUKlUyMvLQ1ZWlts3eHsLaIVCIYqgJo7jhDp8V1cXOI6DVCpFUlISYmNjRbN5MbdCsZU/DAYDzp07h9bWVgBw6AEZa1KiqakJa9aswYYNG/Df//3fornGBEG4D4kEwik6nQ6fffYZCgoKsGfPHsTHxyM3Nxf5+flQKpVj3vBtNhuqqqpgNBon1QLaGTzP4/Tp02hvb8e8efOg0+mg0Whgs9kc3B4nM7775MmTbrsV+hOWNDl37lwkJCSMOSkRGRmJgIAAnDt3DqtXr8a6devw17/+lQQCQVyikEggXKLX67F3714UFBRg9+7diIqKQk5ODvLy8rBs2TJhU+3r60NDQwMCAgIgk8lEYwENOJY/FAqF8PQ7ug5vNpsdzJuCgoL8sj7WQOnN+G5vMTQ0hLKyMqdJk/aTErW1tXjyySdx/fXXo6KiAitXrsQbb7xBAoEgLmFIJBAeYTQa8cUXX6CwsBA7d+7ElClTkJOTg5UrV+KZZ57BqlWr8OKLL4oqqInnedTV1aG/v/+i5Q+e5zE8PCyYNxkMBsTFxSEhIcGnFsft7e2or68XZQMlEwjuOFAajUb8+9//xt/+9jfU19cjNDQUOTk5yM/Px+rVq33q+EgQhG8gkUCMG7PZjJKSErz77rsoKChASkoKli1bhh/+8IdYvny51x0RxwPHcaipqcHw8DDkcrlHGxVr3Ovu7nawOJZKpV4ro7S1taGhoUGUAkGn06GsrAwpKSlISUlx+fkajQZr165FVlYW3n33XZSVlaGoqAg7d+5ESUnJuAy0CIKYXEgkEBOivr4et9xyC1avXo0f/ehHKCoqQnFxMQwGA9avXw+VSoUbb7xxUp4iOY5DVVUVDAYDFArFhESLwWAQShKDg4OIjo4WBMN4GzPZCKZMJkNsbOy41+YLhoeHUVpaiuTkZKSmprr8/N7eXqxbtw7z5s3Dhx9+KKpSE0EQ44dEAjEh7rzzTiQnJ2PLli3C/LvNZsPRo0dRUFCAoqIiDA4OYs2aNVCpVLjllls8dkQcDzabDZWVlbBYLJDL5V7dtEwmkyAY+vv7ERERgYSEBI8yEVpbW3H69GlRjmAygeBuCmZ/fz9ycnKQlJSEHTt2iOIEiSAI70AigZgQZrP5opsCx3E4fvy4kFjZ3d2NVatWIS8vD6tXr/aJf4LVakV5eTkAQCaT+bT50GKxQKvVoru7W+j0Z26PERERTo2DWlpacObMGVEKhJGREZSWlgoJna4YHBxEbm4u4uPjUVxcLKppFoIgJg6JBMJvcByH8vJy7NixA4WFhWhtbcVNN92EvLw8rF27dkJBQwyLxQK1Wo3g4GBkZWX5tYHSarU6mDeFhIQIJYno6GhIJBKcO3cOTU1NkMvlfnGg9AQmEGbNmnVBhLczdDod8vPzERYWhl27donCD4MgCO9CIoGYFHieR01NDT7++GMUFRWhoaEBN954I1QqFdavX4/Y2FiPBYPZbEZZWRmmTp2KzMzMSR29s9lswmigVqtFYGAgQkNDodPpoFQqRScQ9Ho9SktLMWPGDMybN8/ltR8ZGcGGDRsgkUjw2WefeTV6miAI8UAigZh0eJ7HqVOnhJJETU0NbrjhBqhUKuTk5GD69OkuNy2j0Qi1Wo2IiAgsXrxYVLP5HMfh5MmT6OrqQkBAACQSiYN502Sv1WAwoLS0FFKpFPPnz3d5rQ0GA370ox/BbDZjz549iIyM9NNKCYLwN+K5kxLfWyQSCRYuXIjf/e53KCsrQ11dHW655Ra8//77mDdvHtauXYvt27ejo6MDzjQt2+Sio6OxZMmSSd90R9Pc3AytVourrroKK1asQFZWFoKCgnDq1CkcOnQI1dXV6O7uhs1m8/va2LWbPn26WwLBaDTizjvvxMjICHbv3n3ZCoSNGzdCIpFAIpEgODgYaWlpePzxxzEyMuLweQUFBVixYgWio6MRERGBzMxMPPfcc+jr63P6us3Nzbj//vuRmpqKqVOnIj09Hc888wzMZrM/3hZBeAydJBCihed5tLS0oKCgAIWFhTh27Biuuuoq5ObmQqVSYfbs2aiurkZBQQHuuOMOryZYegP7FEylUnlBkybP8xgaGhImJYxGo4Pbo6/HCI1GI0pLSzFt2jRcccUVLq+d2WzGj3/8Y3R2duKLL74Qna+DN9m4cSO6u7vx97//HRaLBYcPH8ZPf/pT3HPPPdi2bRsA4Omnn8Yf//hHPProo8jPz0diYiIaGxuxfft23HDDDXj44YcveN29e/fi3//+N+644w7MnTsXNTU12LRpE+6++278+c9/9vfbJAiXkEggLgl4nkdHRwcKCwtRWFiII0eOICMjA83NzVi3bh1ef/11UZ0g8DyPM2fOoL29HQqFwuUUB0tdZG6PIyMjiIuLExofvT1WyARCXFwcFi5c6FIgWCwWbNy4EU1NTfjyyy9Fly3hbTZu3IiBgQEUFxcL/7Zp0yZ8+umn6OzsxPHjx7F06VL85S9/cSoGBgYG3J5c+dOf/oRt27ahqanJS6snCO/hH2N6gpggEokEs2bNwoMPPoj/+q//wr59+7BhwwakpaXho48+Qm1tLfLy8qBSqdw6NvclLEiqo6MDSqXSraY+iUSCiIgIREREID09HXq9HhqNRsh0iImJEQTDRI2pTCYTysrKEBsb65ZAsFqt2LRpExoaGlBSUnLZC4SxmDp1KiwWCwDgn//8JyIiIrB582ann+vJaOvg4OBlfSpDXNqI59GLINzk2LFjuP322/Hcc8+hsrISXV1deOihh3DixAksW7YMS5cuxQsvvIDa2lpwHOfXtfE8j8bGRnR2drotEJwRFhaGlJQUXHXVVbjuuusglUqh0Whw5MgRHD9+HM3NzdDr9R6/LhMI0dHRWLRokUuBYLPZsHnzZlRWVmL//v2QSqXjej+XOsePH8cHH3yAm266CQDQ2NiItLS0CZeEzpw5g1dffRU///nPvbFMgvA6VG4gLjnq6+tx9OhR3HfffQ7/zhIdP/nkExQWFmLfvn2YPXu2EHHt67FInufR0NCA7u5uKJVKnzhLms1moYehr68P4eHhDm6PF9v0zWYzSktLERkZicWLF7sUCBzH4cEHH8Thw4dRUlKC2bNne/vtiJaNGzfi/fffR2hoKKxWKywWC1QqFbZv3w6pVIo1a9ags7MTFRUV4/4eHR0dWL58OZYvX44333zTe4snCC9CIoG4bNHpdNi9ezcKCgqwZ88eSKVSQTAoFAqvCgYWRa3VaqFQKPxiPW2xWBzMm0JDQ4WSxGhjKuYhER4e7taIKMdxeOyxx7Bv3z6UlJS4FfB0ObFx40a0t7dj27ZtCA4ORmJiosOpwcMPP4y3334bfX194zpN6OjowMqVK7F06VK88847ouqnIQh7SCQQ3wtGRkawd+9eFBQUYPfu3YiOjkZubi7y8vKwdOnSCTkzMp+Hnp4eKJXKSXEetNlsgmDQarUIDg4WBEN4eDjUajWmTp3q1ogox3H4zW9+g+LiYhw8eNAte+bLDWeNi/YcO3YMy5YtG1fjYnt7O1auXAmFQoH3339fVLHqBDEaEgnE9w6DwYAvvvgChYWF+OSTTxAaGoqcnBzk5+fjmmuu8Sjrged5nDx5En19fVAoFKKwJuY4TnB71Gg0sNlsCA0NxYIFC1yaN3Ech9///vf48MMPUVJSggULFvhx5eLBlUgAgCeffBIvv/wyfvnLXwojkKdPn8b27dtx3XXXORUPrMQwZ84cvPfeew4CYcaMGb54KwQxIUgkEN9rzGYzDhw4gIKCAhQXF0MikWD9+vXIy8vDDTfccNHRQ57nUVdXh/7+fiiVykmJw74YFosFZWVlkEgkiIqKgkajAcdxDm6P9psUz/N4/vnn8fbbb6OkpASLFi2axNVPLu6IBAD46KOP8Nprr6G8vBwcxyE9PR233norHnzwQacnCe+88w7uvfdep69Ft2JCjJBIIIj/w2q14tChQ9ixYweKi4thMpmwfv16qFQq3HjjjQ4Jh8xqeWBgAAqFQnQCwWq1OgRdBQQECI2dGo0G3d3dsFgsmDp1Kurq6pCfn48333wTf/3rX3HgwAFkZmZO9lsgCEIEkEggCCfYbDYcPXpUyJPQ6XRYvXo18vLysGLFCtx3333IzMzEk08+Kbp4ZCYQgoKCxkzC5HkeOp0OR48exWOPPYaOjg4EBwfjV7/6FR588EGa2ycIAgCJBIJwCcdxOHbsmBBxPTg4iKlTp+I3v/kNbrvtNpduiv7EZrNBrVYjICAA2dnZLpvieJ7H1q1bsXXrVvzgBz/AiRMnUFVVhZUrV+Kdd95BYmKin1ZOEIQYIZFAEG5isVhw1113Qa1WY+3atdi7dy/a2tpw8803Iy8vD2vWrLlg9NCf2Gw2lJeXAwBkMplbAuHNN9/E73//e3z22We49tprAXwXQrRz505s3rzZ5/kRBEGIGxIJBOEmDzzwAI4cOYL9+/dj+vTp4DgONTU1wgnD6dOnceONN0KlUmHdunWIjY31m2Cw2WyoqKgAx3GQy+VuCYT33nsPv/rVr7Br1y6sWLHCL+skCOLSgkQCQbhJY2Mj4uLiMG3atAs+xrwSmGCora3F8uXLoVKpkJOTg/j4eJ8JBpvNhsrKSthsNshkMpcjnDzP44MPPsCjjz6KnTt3ClbDBEEQoyGRQBBehgU8sYjr8vJyXHvttVCpVMjNzcWMGTO8Jhg4jkNlZSUsFgvkcrlbHg8ff/wxNm/ejB07dmDNmjVeWQdBEJcnJBIIwofwPI9z584JguHYsWNYunQpVCoVVCoVkpKSxi0YmEAwm82Qy+Vu9Q8UFxdj06ZN+Ne//oXc3NxxfV+CIL4/kEggCD/B8zza29tRWFiIwsJCHD16FDKZTIi4TklJcVswcByHqqoqGI1GKBQKtwTCp59+invvvRfvvfceNmzYMNG3QxDE9wASCQQxCfA8j+7ubhQVFaGwsBCHDh1CRkaGIBjmzZs3pmDgOA7V1dUwGAxuC4TPP/8cd999N9588038x3/8h7ffDkEQlykkEghikuF5Hr29vdi5cycKCgrw5ZdfYv78+VCpVMjLy8PChQsFwWA2m1FdXQ2z2QyFQnFR22hGSUkJbr/9dmzduhV33333pI1oEgRx6UEigSBEBM/zGBgYwK5du1BQUIB9+/YhOTkZubm5yMnJwfPPP4/IyEi8+eabbgmEw4cP49Zbb8Vf/vIX3HfffSQQCILwCBIJBCFihoaGsHv3buzYsQN79uxBZGQkbr31Vtx+++2Qy+UXTXT85ptvkJ+fj5deegm/+MUvSCAQBOExFw+WJwhiUomKisLtt9+OmJgYJCUl4fnnn0dfXx9ycnKQkZGBJ598El9//TVsNpvD15WWlmLDhg34wx/+QAKBIIhxQycJBCFytm7dildeeQUHDx7ErFmzAAAGgwH79u1DYWEhdu3ahdDQUOTm5iIvLw/h4eHIzc3FU089hccff5wEAkEQ44ZEAkGIHJPJhL6+PsycOdPpx81mM/bv34/CwkIUFRWhv78fTz75JLZs2UICgSCICUEigSAuIywWC9566y3853/+JwkEgiAmDIkEgiAIgiCcQo2LBEEQBEE4hUQCQRAEQRBOIZFAEARBEIRTSCQQBEEQBOEUEgkEQRAEQTiFRAJBEARBEE4hkUAQBEEQhFNIJIiUjRs3QiKRQCKRIDg4GGlpaXj88ccxMjLi8HkFBQVYsWIFoqOjERERgczMTDz33HPo6+sb87VfeOEFXHPNNQgLC0NMTIyP3wlBEARxqUIiQcSsXr0anZ2daGpqwvPPP4+tW7fi8ccfFz7+9NNP4/bbb8eVV16JPXv2oKamBi+//DIqKyvxj3/8Y8zXNZvN+NGPfoRf/OIX/ngbBEEQxCUKOS6KlI0bN2JgYADFxcXCv23atAmffvopOjs7cfz4cSxduhR/+ctf8PDDD1/w9QMDAy5PCd555x088sgjGBgY8O7iCYIgiMsCOkm4hJg6dSosFgsA4J///CciIiKwefNmp59LZQSCIAhiopBIuEQ4fvw4PvjgA9x0000AgMbGRqSlpSE4OHiSV0YQBEFcrgRN9gKIsfn0008REREBq9UKi8UClUqFV199FQDA8zyl/BEEQRA+hUSCiFm5ciW2bduG4OBgJCYmOpwazJ8/H0eOHIHFYqHTBIIgCMInULlBxISHh2Pu3LlITk6+QAjceeedGB4extatW51+LTUjEgRBEBOFThIuUZYuXYpf/epXeOyxx9De3o78/HwkJibi9OnT2L59O6677jqnUw8A0NLSgr6+PrS0tMBms6GiogIAMHfuXERERPjxXRAEQRBihkTCJcwf//hHKBQKvPbaa9i+fTs4jkN6ejpuvfVW3HPPPWN+3e9//3u8++67wv/LZDIAQElJCVasWOHrZRMEQRCXCOSTQBAEQRCEU6gngSAIgiAIp5BIIAiCIAjCKSQSCIIgCIJwCokEgiAIgiCcQiKBIAiCIAinkEggCIIgCMIpJBIIgiAIgnAKiQSCIAiCIJxCIoEgCIIgCKeQSCAIgiAIwikkEgiCIAiCcAqJBIIgCIIgnEIigSAIgiAIp/z/97qh7b5i5coAAAAASUVORK5CYII=",
      "text/plain": [
       "<Figure size 640x480 with 1 Axes>"
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    }
   ],
   "source": [
    "#plot pcoa results by health status \n",
    "%matplotlib inline\n",
    "\n",
    "fig = pcoa_results.plot(df=meta_mcav,\n",
    "                        column='Health_Status ',\n",
    "                        axis_labels=['PC 1', 'PC 2', 'PC 3'],\n",
    "                        title='Samples colored by health status',\n",
    "                        cmap='Reds',\n",
    "                        s=35)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "89c132bd-c24b-4ba2-9a33-7ca2c9b6219b",
   "metadata": {},
   "outputs": [],
   "source": [
    "%matplotlib inline\n",
    "\n",
    "fig = pcoa_results.plot(df=meta_mcav,\n",
    "                        column='Health_Status ',\n",
    "                        axis_labels=['PC 1', 'PC 2', 'PC 3'],\n",
    "                        title='Samples colored by health status',\n",
    "                        cmap='Reds',\n",
    "                        s=35)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 194,
   "id": "542a4f82-c42a-49d6-9423-591f50ded476",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>Unnamed: 0</th>\n",
       "      <th>Health_Status</th>\n",
       "      <th>Starting_Weight</th>\n",
       "      <th>Extracted</th>\n",
       "      <th>Raw_ng_ul</th>\n",
       "      <th>Date_Enriched</th>\n",
       "      <th>Microbe_ng_ul</th>\n",
       "      <th>Micobe_Location</th>\n",
       "      <th>Microbe_clean_date/n</th>\n",
       "      <th>Host_ng_ul</th>\n",
       "      <th>Host_Location</th>\n",
       "      <th>Host_clean_y/n</th>\n",
       "      <th>Date_Libprep</th>\n",
       "      <th>Status</th>\n",
       "      <th>Notes</th>\n",
       "      <th>Seq_date</th>\n",
       "      <th>Host_Seq_date</th>\n",
       "      <th>Seq_Location (in Unity)</th>\n",
       "      <th>Sample_Code = datecollected_tagnumber_transect_samplenumber_species</th>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>Seq_file</th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>NaN</th>\n",
       "      <td>052022_T2_45_PSTR</td>\n",
       "      <td>Diseased_Margin</td>\n",
       "      <td>181</td>\n",
       "      <td>10_10_2023</td>\n",
       "      <td>22.8</td>\n",
       "      <td>10_17_2023</td>\n",
       "      <td>7.5</td>\n",
       "      <td>UML_F1_R2_B1</td>\n",
       "      <td>y</td>\n",
       "      <td>0.772</td>\n",
       "      <td>UML_F1_R2_B2</td>\n",
       "      <td>n</td>\n",
       "      <td>NaN</td>\n",
       "      <td>sequenced</td>\n",
       "      <td>NaN</td>\n",
       "      <td>10_19_23</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>T3_8_2019</th>\n",
       "      <td>062019_14_T3_8_MCAV</td>\n",
       "      <td>Healthy</td>\n",
       "      <td>_</td>\n",
       "      <td>5_18_2022</td>\n",
       "      <td>7.8</td>\n",
       "      <td>8_8_2022</td>\n",
       "      <td>-16.13</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>_</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>8_30_2022</td>\n",
       "      <td>sequenced</td>\n",
       "      <td>Processed following protocol as written</td>\n",
       "      <td>sequenced</td>\n",
       "      <td>NaN</td>\n",
       "      <td>/project/pi_sarah_gignouxwolfsohn_uml_edu/broo...</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>T2_16_2019</th>\n",
       "      <td>062019_60_T2_16_MCAV</td>\n",
       "      <td>Healthy</td>\n",
       "      <td>_</td>\n",
       "      <td>5_18_2022</td>\n",
       "      <td>6.1</td>\n",
       "      <td>8_8_2022</td>\n",
       "      <td>1.89</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>_</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>8_30_2022</td>\n",
       "      <td>sequenced</td>\n",
       "      <td>Processed following protocol as written</td>\n",
       "      <td>sequenced</td>\n",
       "      <td>NaN</td>\n",
       "      <td>/project/pi_sarah_gignouxwolfsohn_uml_edu/broo...</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>T1_20_2019</th>\n",
       "      <td>062019_7_T1_20_MCAV</td>\n",
       "      <td>Healthy</td>\n",
       "      <td>_</td>\n",
       "      <td>5_18_2022</td>\n",
       "      <td>20.8</td>\n",
       "      <td>8_8_2022</td>\n",
       "      <td>2.41</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>_</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>8_30_2022</td>\n",
       "      <td>sequenced</td>\n",
       "      <td>Processed following protocol as written</td>\n",
       "      <td>sequenced</td>\n",
       "      <td>NaN</td>\n",
       "      <td>/project/pi_sarah_gignouxwolfsohn_uml_edu/broo...</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>T1_12_2022</th>\n",
       "      <td>052022_8_T1_12_MCAV</td>\n",
       "      <td>Margin</td>\n",
       "      <td>_</td>\n",
       "      <td>6_10_2022</td>\n",
       "      <td>6.2</td>\n",
       "      <td>7_6_2022</td>\n",
       "      <td>0.1</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>_</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>7_29_2022</td>\n",
       "      <td>sequenced</td>\n",
       "      <td>(Library Prep) Used 3µl of water to reach init...</td>\n",
       "      <td>sequenced</td>\n",
       "      <td>NaN</td>\n",
       "      <td>/project/pi_sarah_gignouxwolfsohn_uml_edu/broo...</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>T1_24_2019</th>\n",
       "      <td>062019_15_T1_24_MCAV</td>\n",
       "      <td>Healthy</td>\n",
       "      <td>177</td>\n",
       "      <td>6_13_2022</td>\n",
       "      <td>7.1</td>\n",
       "      <td>7_1_2022</td>\n",
       "      <td>1.4</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>_</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>7_20_2022</td>\n",
       "      <td>sequenced</td>\n",
       "      <td>Processed following protocol as written</td>\n",
       "      <td>sequenced</td>\n",
       "      <td>NaN</td>\n",
       "      <td>/project/pi_sarah_gignouxwolfsohn_uml_edu/broo...</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>T3_9_2019</th>\n",
       "      <td>062019_15_T3_9_MCAV</td>\n",
       "      <td>Healthy</td>\n",
       "      <td>162</td>\n",
       "      <td>6_13_2022</td>\n",
       "      <td>0.8</td>\n",
       "      <td>7_1_2022</td>\n",
       "      <td>1.3</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>_</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>7_20_2022</td>\n",
       "      <td>sequenced</td>\n",
       "      <td>Processed following protocol as written</td>\n",
       "      <td>sequenced</td>\n",
       "      <td>NaN</td>\n",
       "      <td>/project/pi_sarah_gignouxwolfsohn_uml_edu/broo...</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>T3_14_2019</th>\n",
       "      <td>062019_17_T3_14_MCAV</td>\n",
       "      <td>Healthy</td>\n",
       "      <td>162</td>\n",
       "      <td>6_13_2022</td>\n",
       "      <td>3.6</td>\n",
       "      <td>7_1_2022</td>\n",
       "      <td>2.1</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>_</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>7_20_2022</td>\n",
       "      <td>sequenced</td>\n",
       "      <td>Processed following protocol as written</td>\n",
       "      <td>sequenced</td>\n",
       "      <td>NaN</td>\n",
       "      <td>/project/pi_sarah_gignouxwolfsohn_uml_edu/broo...</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>T3_1_2019</th>\n",
       "      <td>062019_2_T3_1_MCAV</td>\n",
       "      <td>Healthy</td>\n",
       "      <td>113</td>\n",
       "      <td>6_13_2022</td>\n",
       "      <td>3.9</td>\n",
       "      <td>7_1_2022</td>\n",
       "      <td>0.6</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>_</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>7_20_2022</td>\n",
       "      <td>sequenced</td>\n",
       "      <td>Processed following protocol as written</td>\n",
       "      <td>sequenced</td>\n",
       "      <td>NaN</td>\n",
       "      <td>/project/pi_sarah_gignouxwolfsohn_uml_edu/broo...</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>T3_15_2019</th>\n",
       "      <td>062019_22_T3_15_MCAV</td>\n",
       "      <td>Healthy</td>\n",
       "      <td>199</td>\n",
       "      <td>6_13_2022</td>\n",
       "      <td>7.7</td>\n",
       "      <td>7_1_2022</td>\n",
       "      <td>2.5</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>_</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>7_20_2022</td>\n",
       "      <td>sequenced</td>\n",
       "      <td>Processed following protocol as written</td>\n",
       "      <td>sequenced</td>\n",
       "      <td>NaN</td>\n",
       "      <td>/project/pi_sarah_gignouxwolfsohn_uml_edu/broo...</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>T1_16_2019</th>\n",
       "      <td>062019_8_T1_16_MCAV</td>\n",
       "      <td>Healthy</td>\n",
       "      <td>188</td>\n",
       "      <td>6_13_2022</td>\n",
       "      <td>8.6</td>\n",
       "      <td>7_1_2022</td>\n",
       "      <td>2.7</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>_</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>7_20_2022</td>\n",
       "      <td>sequenced</td>\n",
       "      <td>Processed following protocol as written</td>\n",
       "      <td>sequenced</td>\n",
       "      <td>NaN</td>\n",
       "      <td>/project/pi_sarah_gignouxwolfsohn_uml_edu/broo...</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>T1_40_2022</th>\n",
       "      <td>052022_15_T1_40_MCAV</td>\n",
       "      <td>Margin</td>\n",
       "      <td>70</td>\n",
       "      <td>6_14_2022</td>\n",
       "      <td>18.4</td>\n",
       "      <td>7_6_2022</td>\n",
       "      <td>5.2</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>_</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>7_29_2022</td>\n",
       "      <td>sequenced</td>\n",
       "      <td>Processed following protocol as written</td>\n",
       "      <td>sequenced</td>\n",
       "      <td>NaN</td>\n",
       "      <td>/project/pi_sarah_gignouxwolfsohn_uml_edu/broo...</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>T1_70_2022</th>\n",
       "      <td>052022_15_T1_70_MCAV</td>\n",
       "      <td>Diseased</td>\n",
       "      <td>117</td>\n",
       "      <td>6_14_2022</td>\n",
       "      <td>95</td>\n",
       "      <td>7_6_2022</td>\n",
       "      <td>5.6</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>_</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>7_29_2022</td>\n",
       "      <td>sequenced</td>\n",
       "      <td>Processed following protocol as written</td>\n",
       "      <td>sequenced</td>\n",
       "      <td>NaN</td>\n",
       "      <td>/project/pi_sarah_gignouxwolfsohn_uml_edu/broo...</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>T3_48_2022</th>\n",
       "      <td>052022_15_T3_48_MCAV</td>\n",
       "      <td>Margin</td>\n",
       "      <td>155</td>\n",
       "      <td>6_14_2022</td>\n",
       "      <td>38.3</td>\n",
       "      <td>7_6_2022</td>\n",
       "      <td>6.4</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>_</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>7_29_2022</td>\n",
       "      <td>sequenced</td>\n",
       "      <td>(Library Prep) Used 3µl of water to reach init...</td>\n",
       "      <td>sequenced</td>\n",
       "      <td>NaN</td>\n",
       "      <td>/project/pi_sarah_gignouxwolfsohn_uml_edu/broo...</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>T3_49_2022</th>\n",
       "      <td>052022_15_T3_49_MCAV</td>\n",
       "      <td>Diseased</td>\n",
       "      <td>142</td>\n",
       "      <td>6_14_2022</td>\n",
       "      <td>41.9</td>\n",
       "      <td>7_6_2022</td>\n",
       "      <td>5.8</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>_</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>7_29_2022</td>\n",
       "      <td>sequenced</td>\n",
       "      <td>Processed following protocol as written</td>\n",
       "      <td>sequenced</td>\n",
       "      <td>NaN</td>\n",
       "      <td>/project/pi_sarah_gignouxwolfsohn_uml_edu/broo...</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>T3_51_2022</th>\n",
       "      <td>052022_17_T3_51_MCAV</td>\n",
       "      <td>Diseased</td>\n",
       "      <td>104</td>\n",
       "      <td>6_14_2022</td>\n",
       "      <td>30.8</td>\n",
       "      <td>7_6_2022</td>\n",
       "      <td>1.5</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>_</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>7_29_2022</td>\n",
       "      <td>sequenced</td>\n",
       "      <td>Processed following protocol as written</td>\n",
       "      <td>sequenced</td>\n",
       "      <td>NaN</td>\n",
       "      <td>/project/pi_sarah_gignouxwolfsohn_uml_edu/broo...</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>T3_40_2022</th>\n",
       "      <td>052022_22_T3_40_MCAV</td>\n",
       "      <td>Margin</td>\n",
       "      <td>139</td>\n",
       "      <td>6_14_2022</td>\n",
       "      <td>24.7</td>\n",
       "      <td>7_6_2022</td>\n",
       "      <td>7.5</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>_</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>7_29_2022</td>\n",
       "      <td>sequenced</td>\n",
       "      <td>Used 3µl of water to reach initial volume of  ...</td>\n",
       "      <td>sequenced</td>\n",
       "      <td>NaN</td>\n",
       "      <td>/project/pi_sarah_gignouxwolfsohn_uml_edu/broo...</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>T3_60_2022</th>\n",
       "      <td>052022_22_T3_60_MCAV</td>\n",
       "      <td>Diseased</td>\n",
       "      <td>187</td>\n",
       "      <td>6_14_2022</td>\n",
       "      <td>4.4</td>\n",
       "      <td>8_8_2022</td>\n",
       "      <td>2.72</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>_</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>8_30_2022</td>\n",
       "      <td>sequenced</td>\n",
       "      <td>Processed following protocol as written</td>\n",
       "      <td>sequenced</td>\n",
       "      <td>NaN</td>\n",
       "      <td>/project/pi_sarah_gignouxwolfsohn_uml_edu/broo...</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>T1_13_2022</th>\n",
       "      <td>052022_8_T1_13_MCAV</td>\n",
       "      <td>Diseased</td>\n",
       "      <td>133</td>\n",
       "      <td>6_14_2022</td>\n",
       "      <td>5.8</td>\n",
       "      <td>7_6_2022</td>\n",
       "      <td>1.7</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>_</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>7_29_2022</td>\n",
       "      <td>sequenced</td>\n",
       "      <td>Processed following protocol as written</td>\n",
       "      <td>sequenced</td>\n",
       "      <td>NaN</td>\n",
       "      <td>/project/pi_sarah_gignouxwolfsohn_uml_edu/broo...</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>T3_13_2022</th>\n",
       "      <td>052022_14_T3_13_MCAV</td>\n",
       "      <td>Healthy</td>\n",
       "      <td>_</td>\n",
       "      <td>6_15_2022</td>\n",
       "      <td>13.2</td>\n",
       "      <td>6_22_2022</td>\n",
       "      <td>1</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>0.5</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>6_27_2022</td>\n",
       "      <td>sequenced</td>\n",
       "      <td>Processed following protocol as written</td>\n",
       "      <td>sequenced</td>\n",
       "      <td>NaN</td>\n",
       "      <td>/project/pi_sarah_gignouxwolfsohn_uml_edu/broo...</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>T3_1_2022</th>\n",
       "      <td>052022_2_T3_1_MCAV</td>\n",
       "      <td>Healthy</td>\n",
       "      <td>_</td>\n",
       "      <td>6_15_2022</td>\n",
       "      <td>37</td>\n",
       "      <td>6_22_2022</td>\n",
       "      <td>3.6</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>1.8</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>6_27_2022</td>\n",
       "      <td>sequenced</td>\n",
       "      <td>Processed following protocol as written</td>\n",
       "      <td>sequenced</td>\n",
       "      <td>NaN</td>\n",
       "      <td>/project/pi_sarah_gignouxwolfsohn_uml_edu/broo...</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>T3_19_2022</th>\n",
       "      <td>052022_21_T3_19_MCAV</td>\n",
       "      <td>Healthy</td>\n",
       "      <td>_</td>\n",
       "      <td>6_15_2022</td>\n",
       "      <td>38.2</td>\n",
       "      <td>6_22_2022</td>\n",
       "      <td>6.3</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>0</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>6_27_2022</td>\n",
       "      <td>sequenced</td>\n",
       "      <td>Processed following protocol as written</td>\n",
       "      <td>sequenced</td>\n",
       "      <td>NaN</td>\n",
       "      <td>/project/pi_sarah_gignouxwolfsohn_uml_edu/broo...</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>T2_10_2022</th>\n",
       "      <td>052022_60_T2_10_MCAV</td>\n",
       "      <td>Healthy</td>\n",
       "      <td>_</td>\n",
       "      <td>6_15_2022</td>\n",
       "      <td>22.4</td>\n",
       "      <td>6_22_2022</td>\n",
       "      <td>0.3</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>0.5</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>6_27_2022</td>\n",
       "      <td>sequenced</td>\n",
       "      <td>Processed following protocol as written</td>\n",
       "      <td>sequenced</td>\n",
       "      <td>NaN</td>\n",
       "      <td>/project/pi_sarah_gignouxwolfsohn_uml_edu/broo...</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>T1_57_2022</th>\n",
       "      <td>052022_7_T1_57_MCAV</td>\n",
       "      <td>Healthy</td>\n",
       "      <td>_</td>\n",
       "      <td>6_15_2022</td>\n",
       "      <td>75.5</td>\n",
       "      <td>6_22_2022</td>\n",
       "      <td>1.43</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>2.4</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>6_24_2022</td>\n",
       "      <td>sequenced</td>\n",
       "      <td>Processed following protocol as written</td>\n",
       "      <td>sequenced</td>\n",
       "      <td>NaN</td>\n",
       "      <td>/project/pi_sarah_gignouxwolfsohn_uml_edu/broo...</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>Coral5.fastq</th>\n",
       "      <td>062019_21_T3_16_MCAV</td>\n",
       "      <td>Healthy</td>\n",
       "      <td>NaN</td>\n",
       "      <td>extracted</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>sequenced</td>\n",
       "      <td>NaN</td>\n",
       "      <td>sequenced</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>T3-21-Mmea.fastq</th>\n",
       "      <td>062019_345_T3_21_MMEA</td>\n",
       "      <td>Healthy</td>\n",
       "      <td>NaN</td>\n",
       "      <td>extracted</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>sequenced</td>\n",
       "      <td>NaN</td>\n",
       "      <td>sequenced</td>\n",
       "      <td>NaN</td>\n",
       "      <td>/project/pi_sarah_gignouxwolfsohn_uml_edu/broo...</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>Coral1.fastq</th>\n",
       "      <td>062019_355_T1_3_MCAV</td>\n",
       "      <td>Healthy</td>\n",
       "      <td>NaN</td>\n",
       "      <td>extracted</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>sequenced</td>\n",
       "      <td>NaN</td>\n",
       "      <td>sequenced</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>Coral2.fastq</th>\n",
       "      <td>062019_59_T2_5_MCAV</td>\n",
       "      <td>Healthy</td>\n",
       "      <td>NaN</td>\n",
       "      <td>extracted</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>sequenced</td>\n",
       "      <td>NaN</td>\n",
       "      <td>sequenced</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>T3-4-Past.fastq</th>\n",
       "      <td>062019_6_T3_4_PAST</td>\n",
       "      <td>Healthy</td>\n",
       "      <td>NaN</td>\n",
       "      <td>extracted</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>sequenced</td>\n",
       "      <td>NaN</td>\n",
       "      <td>sequenced</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>Coral4.fastq</th>\n",
       "      <td>062019_9_T3_6_MCAV</td>\n",
       "      <td>Healthy</td>\n",
       "      <td>NaN</td>\n",
       "      <td>extracted</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>sequenced</td>\n",
       "      <td>NaN</td>\n",
       "      <td>sequenced</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>_</th>\n",
       "      <td>102019_T2_35_PSTR</td>\n",
       "      <td>Healthy</td>\n",
       "      <td>_</td>\n",
       "      <td>NaN</td>\n",
       "      <td>_</td>\n",
       "      <td>_</td>\n",
       "      <td>_</td>\n",
       "      <td>_</td>\n",
       "      <td>_</td>\n",
       "      <td>_</td>\n",
       "      <td>_</td>\n",
       "      <td>_</td>\n",
       "      <td>_</td>\n",
       "      <td>_</td>\n",
       "      <td>_</td>\n",
       "      <td>_</td>\n",
       "      <td>_</td>\n",
       "      <td>_</td>\n",
       "      <td>_</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "                             Unnamed: 0   Health_Status  Starting_Weight  \\\n",
       "Seq_file                                                                   \n",
       "NaN                   052022_T2_45_PSTR  Diseased_Margin             181   \n",
       "T3_8_2019           062019_14_T3_8_MCAV          Healthy               _   \n",
       "T2_16_2019         062019_60_T2_16_MCAV          Healthy               _   \n",
       "T1_20_2019          062019_7_T1_20_MCAV          Healthy               _   \n",
       "T1_12_2022          052022_8_T1_12_MCAV           Margin               _   \n",
       "T1_24_2019         062019_15_T1_24_MCAV          Healthy             177   \n",
       "T3_9_2019           062019_15_T3_9_MCAV          Healthy             162   \n",
       "T3_14_2019         062019_17_T3_14_MCAV          Healthy             162   \n",
       "T3_1_2019            062019_2_T3_1_MCAV          Healthy             113   \n",
       "T3_15_2019         062019_22_T3_15_MCAV          Healthy             199   \n",
       "T1_16_2019          062019_8_T1_16_MCAV          Healthy             188   \n",
       "T1_40_2022         052022_15_T1_40_MCAV           Margin              70   \n",
       "T1_70_2022         052022_15_T1_70_MCAV         Diseased             117   \n",
       "T3_48_2022         052022_15_T3_48_MCAV           Margin             155   \n",
       "T3_49_2022         052022_15_T3_49_MCAV         Diseased             142   \n",
       "T3_51_2022         052022_17_T3_51_MCAV         Diseased             104   \n",
       "T3_40_2022         052022_22_T3_40_MCAV           Margin             139   \n",
       "T3_60_2022         052022_22_T3_60_MCAV         Diseased             187   \n",
       "T1_13_2022          052022_8_T1_13_MCAV         Diseased             133   \n",
       "T3_13_2022         052022_14_T3_13_MCAV          Healthy               _   \n",
       "T3_1_2022            052022_2_T3_1_MCAV          Healthy               _   \n",
       "T3_19_2022         052022_21_T3_19_MCAV          Healthy               _   \n",
       "T2_10_2022         052022_60_T2_10_MCAV          Healthy               _   \n",
       "T1_57_2022          052022_7_T1_57_MCAV          Healthy               _   \n",
       "Coral5.fastq       062019_21_T3_16_MCAV          Healthy             NaN   \n",
       "T3-21-Mmea.fastq  062019_345_T3_21_MMEA          Healthy             NaN   \n",
       "Coral1.fastq       062019_355_T1_3_MCAV          Healthy             NaN   \n",
       "Coral2.fastq        062019_59_T2_5_MCAV          Healthy             NaN   \n",
       "T3-4-Past.fastq      062019_6_T3_4_PAST          Healthy             NaN   \n",
       "Coral4.fastq         062019_9_T3_6_MCAV          Healthy             NaN   \n",
       "_                     102019_T2_35_PSTR          Healthy               _   \n",
       "\n",
       "                   Extracted Raw_ng_ul Date_Enriched Microbe_ng_ul  \\\n",
       "Seq_file                                                             \n",
       "NaN               10_10_2023      22.8    10_17_2023           7.5   \n",
       "T3_8_2019          5_18_2022       7.8      8_8_2022        -16.13   \n",
       "T2_16_2019         5_18_2022       6.1      8_8_2022          1.89   \n",
       "T1_20_2019         5_18_2022      20.8      8_8_2022          2.41   \n",
       "T1_12_2022         6_10_2022       6.2      7_6_2022           0.1   \n",
       "T1_24_2019         6_13_2022       7.1      7_1_2022           1.4   \n",
       "T3_9_2019          6_13_2022       0.8      7_1_2022           1.3   \n",
       "T3_14_2019         6_13_2022       3.6      7_1_2022           2.1   \n",
       "T3_1_2019          6_13_2022       3.9      7_1_2022           0.6   \n",
       "T3_15_2019         6_13_2022       7.7      7_1_2022           2.5   \n",
       "T1_16_2019         6_13_2022       8.6      7_1_2022           2.7   \n",
       "T1_40_2022         6_14_2022      18.4      7_6_2022           5.2   \n",
       "T1_70_2022         6_14_2022        95      7_6_2022           5.6   \n",
       "T3_48_2022         6_14_2022      38.3      7_6_2022           6.4   \n",
       "T3_49_2022         6_14_2022      41.9      7_6_2022           5.8   \n",
       "T3_51_2022         6_14_2022      30.8      7_6_2022           1.5   \n",
       "T3_40_2022         6_14_2022      24.7      7_6_2022           7.5   \n",
       "T3_60_2022         6_14_2022       4.4      8_8_2022          2.72   \n",
       "T1_13_2022         6_14_2022       5.8      7_6_2022           1.7   \n",
       "T3_13_2022         6_15_2022      13.2     6_22_2022             1   \n",
       "T3_1_2022          6_15_2022        37     6_22_2022           3.6   \n",
       "T3_19_2022         6_15_2022      38.2     6_22_2022           6.3   \n",
       "T2_10_2022         6_15_2022      22.4     6_22_2022           0.3   \n",
       "T1_57_2022         6_15_2022      75.5     6_22_2022          1.43   \n",
       "Coral5.fastq       extracted       NaN           NaN           NaN   \n",
       "T3-21-Mmea.fastq   extracted       NaN           NaN           NaN   \n",
       "Coral1.fastq       extracted       NaN           NaN           NaN   \n",
       "Coral2.fastq       extracted       NaN           NaN           NaN   \n",
       "T3-4-Past.fastq    extracted       NaN           NaN           NaN   \n",
       "Coral4.fastq       extracted       NaN           NaN           NaN   \n",
       "_                        NaN         _             _             _   \n",
       "\n",
       "                 Micobe_Location Microbe_clean_date/n Host_ng_ul  \\\n",
       "Seq_file                                                           \n",
       "NaN                 UML_F1_R2_B1                    y      0.772   \n",
       "T3_8_2019                    NaN                  NaN          _   \n",
       "T2_16_2019                   NaN                  NaN          _   \n",
       "T1_20_2019                   NaN                  NaN          _   \n",
       "T1_12_2022                   NaN                  NaN          _   \n",
       "T1_24_2019                   NaN                  NaN          _   \n",
       "T3_9_2019                    NaN                  NaN          _   \n",
       "T3_14_2019                   NaN                  NaN          _   \n",
       "T3_1_2019                    NaN                  NaN          _   \n",
       "T3_15_2019                   NaN                  NaN          _   \n",
       "T1_16_2019                   NaN                  NaN          _   \n",
       "T1_40_2022                   NaN                  NaN          _   \n",
       "T1_70_2022                   NaN                  NaN          _   \n",
       "T3_48_2022                   NaN                  NaN          _   \n",
       "T3_49_2022                   NaN                  NaN          _   \n",
       "T3_51_2022                   NaN                  NaN          _   \n",
       "T3_40_2022                   NaN                  NaN          _   \n",
       "T3_60_2022                   NaN                  NaN          _   \n",
       "T1_13_2022                   NaN                  NaN          _   \n",
       "T3_13_2022                   NaN                  NaN        0.5   \n",
       "T3_1_2022                    NaN                  NaN        1.8   \n",
       "T3_19_2022                   NaN                  NaN          0   \n",
       "T2_10_2022                   NaN                  NaN        0.5   \n",
       "T1_57_2022                   NaN                  NaN        2.4   \n",
       "Coral5.fastq                 NaN                  NaN        NaN   \n",
       "T3-21-Mmea.fastq             NaN                  NaN        NaN   \n",
       "Coral1.fastq                 NaN                  NaN        NaN   \n",
       "Coral2.fastq                 NaN                  NaN        NaN   \n",
       "T3-4-Past.fastq              NaN                  NaN        NaN   \n",
       "Coral4.fastq                 NaN                  NaN        NaN   \n",
       "_                              _                    _          _   \n",
       "\n",
       "                 Host_Location Host_clean_y/n Date_Libprep    Status   \\\n",
       "Seq_file                                                                \n",
       "NaN               UML_F1_R2_B2              n          NaN  sequenced   \n",
       "T3_8_2019                  NaN            NaN    8_30_2022  sequenced   \n",
       "T2_16_2019                 NaN            NaN    8_30_2022  sequenced   \n",
       "T1_20_2019                 NaN            NaN    8_30_2022  sequenced   \n",
       "T1_12_2022                 NaN            NaN    7_29_2022  sequenced   \n",
       "T1_24_2019                 NaN            NaN    7_20_2022  sequenced   \n",
       "T3_9_2019                  NaN            NaN    7_20_2022  sequenced   \n",
       "T3_14_2019                 NaN            NaN    7_20_2022  sequenced   \n",
       "T3_1_2019                  NaN            NaN    7_20_2022  sequenced   \n",
       "T3_15_2019                 NaN            NaN    7_20_2022  sequenced   \n",
       "T1_16_2019                 NaN            NaN    7_20_2022  sequenced   \n",
       "T1_40_2022                 NaN            NaN    7_29_2022  sequenced   \n",
       "T1_70_2022                 NaN            NaN    7_29_2022  sequenced   \n",
       "T3_48_2022                 NaN            NaN    7_29_2022  sequenced   \n",
       "T3_49_2022                 NaN            NaN    7_29_2022  sequenced   \n",
       "T3_51_2022                 NaN            NaN    7_29_2022  sequenced   \n",
       "T3_40_2022                 NaN            NaN    7_29_2022  sequenced   \n",
       "T3_60_2022                 NaN            NaN    8_30_2022  sequenced   \n",
       "T1_13_2022                 NaN            NaN    7_29_2022  sequenced   \n",
       "T3_13_2022                 NaN            NaN    6_27_2022  sequenced   \n",
       "T3_1_2022                  NaN            NaN    6_27_2022  sequenced   \n",
       "T3_19_2022                 NaN            NaN    6_27_2022  sequenced   \n",
       "T2_10_2022                 NaN            NaN    6_27_2022  sequenced   \n",
       "T1_57_2022                 NaN            NaN    6_24_2022  sequenced   \n",
       "Coral5.fastq               NaN            NaN          NaN  sequenced   \n",
       "T3-21-Mmea.fastq           NaN            NaN          NaN  sequenced   \n",
       "Coral1.fastq               NaN            NaN          NaN  sequenced   \n",
       "Coral2.fastq               NaN            NaN          NaN  sequenced   \n",
       "T3-4-Past.fastq            NaN            NaN          NaN  sequenced   \n",
       "Coral4.fastq               NaN            NaN          NaN  sequenced   \n",
       "_                            _              _            _          _   \n",
       "\n",
       "                                                             Notes   \\\n",
       "Seq_file                                                              \n",
       "NaN                                                             NaN   \n",
       "T3_8_2019                  Processed following protocol as written    \n",
       "T2_16_2019                 Processed following protocol as written    \n",
       "T1_20_2019                 Processed following protocol as written    \n",
       "T1_12_2022        (Library Prep) Used 3µl of water to reach init...   \n",
       "T1_24_2019                 Processed following protocol as written    \n",
       "T3_9_2019                  Processed following protocol as written    \n",
       "T3_14_2019                 Processed following protocol as written    \n",
       "T3_1_2019                  Processed following protocol as written    \n",
       "T3_15_2019                 Processed following protocol as written    \n",
       "T1_16_2019                 Processed following protocol as written    \n",
       "T1_40_2022                 Processed following protocol as written    \n",
       "T1_70_2022                 Processed following protocol as written    \n",
       "T3_48_2022        (Library Prep) Used 3µl of water to reach init...   \n",
       "T3_49_2022                 Processed following protocol as written    \n",
       "T3_51_2022                 Processed following protocol as written    \n",
       "T3_40_2022        Used 3µl of water to reach initial volume of  ...   \n",
       "T3_60_2022                 Processed following protocol as written    \n",
       "T1_13_2022                 Processed following protocol as written    \n",
       "T3_13_2022                 Processed following protocol as written    \n",
       "T3_1_2022                  Processed following protocol as written    \n",
       "T3_19_2022                 Processed following protocol as written    \n",
       "T2_10_2022                 Processed following protocol as written    \n",
       "T1_57_2022                 Processed following protocol as written    \n",
       "Coral5.fastq                                                    NaN   \n",
       "T3-21-Mmea.fastq                                                NaN   \n",
       "Coral1.fastq                                                    NaN   \n",
       "Coral2.fastq                                                    NaN   \n",
       "T3-4-Past.fastq                                                 NaN   \n",
       "Coral4.fastq                                                    NaN   \n",
       "_                                                                 _   \n",
       "\n",
       "                   Seq_date Host_Seq_date  \\\n",
       "Seq_file                                    \n",
       "NaN                10_19_23           NaN   \n",
       "T3_8_2019         sequenced           NaN   \n",
       "T2_16_2019        sequenced           NaN   \n",
       "T1_20_2019        sequenced           NaN   \n",
       "T1_12_2022        sequenced           NaN   \n",
       "T1_24_2019        sequenced           NaN   \n",
       "T3_9_2019         sequenced           NaN   \n",
       "T3_14_2019        sequenced           NaN   \n",
       "T3_1_2019         sequenced           NaN   \n",
       "T3_15_2019        sequenced           NaN   \n",
       "T1_16_2019        sequenced           NaN   \n",
       "T1_40_2022        sequenced           NaN   \n",
       "T1_70_2022        sequenced           NaN   \n",
       "T3_48_2022        sequenced           NaN   \n",
       "T3_49_2022        sequenced           NaN   \n",
       "T3_51_2022        sequenced           NaN   \n",
       "T3_40_2022        sequenced           NaN   \n",
       "T3_60_2022        sequenced           NaN   \n",
       "T1_13_2022        sequenced           NaN   \n",
       "T3_13_2022        sequenced           NaN   \n",
       "T3_1_2022         sequenced           NaN   \n",
       "T3_19_2022        sequenced           NaN   \n",
       "T2_10_2022        sequenced           NaN   \n",
       "T1_57_2022        sequenced           NaN   \n",
       "Coral5.fastq      sequenced           NaN   \n",
       "T3-21-Mmea.fastq  sequenced           NaN   \n",
       "Coral1.fastq      sequenced           NaN   \n",
       "Coral2.fastq      sequenced           NaN   \n",
       "T3-4-Past.fastq   sequenced           NaN   \n",
       "Coral4.fastq      sequenced           NaN   \n",
       "_                         _             _   \n",
       "\n",
       "                                            Seq_Location (in Unity)  \\\n",
       "Seq_file                                                              \n",
       "NaN                                                             NaN   \n",
       "T3_8_2019         /project/pi_sarah_gignouxwolfsohn_uml_edu/broo...   \n",
       "T2_16_2019        /project/pi_sarah_gignouxwolfsohn_uml_edu/broo...   \n",
       "T1_20_2019        /project/pi_sarah_gignouxwolfsohn_uml_edu/broo...   \n",
       "T1_12_2022        /project/pi_sarah_gignouxwolfsohn_uml_edu/broo...   \n",
       "T1_24_2019        /project/pi_sarah_gignouxwolfsohn_uml_edu/broo...   \n",
       "T3_9_2019         /project/pi_sarah_gignouxwolfsohn_uml_edu/broo...   \n",
       "T3_14_2019        /project/pi_sarah_gignouxwolfsohn_uml_edu/broo...   \n",
       "T3_1_2019         /project/pi_sarah_gignouxwolfsohn_uml_edu/broo...   \n",
       "T3_15_2019        /project/pi_sarah_gignouxwolfsohn_uml_edu/broo...   \n",
       "T1_16_2019        /project/pi_sarah_gignouxwolfsohn_uml_edu/broo...   \n",
       "T1_40_2022        /project/pi_sarah_gignouxwolfsohn_uml_edu/broo...   \n",
       "T1_70_2022        /project/pi_sarah_gignouxwolfsohn_uml_edu/broo...   \n",
       "T3_48_2022        /project/pi_sarah_gignouxwolfsohn_uml_edu/broo...   \n",
       "T3_49_2022        /project/pi_sarah_gignouxwolfsohn_uml_edu/broo...   \n",
       "T3_51_2022        /project/pi_sarah_gignouxwolfsohn_uml_edu/broo...   \n",
       "T3_40_2022        /project/pi_sarah_gignouxwolfsohn_uml_edu/broo...   \n",
       "T3_60_2022        /project/pi_sarah_gignouxwolfsohn_uml_edu/broo...   \n",
       "T1_13_2022        /project/pi_sarah_gignouxwolfsohn_uml_edu/broo...   \n",
       "T3_13_2022        /project/pi_sarah_gignouxwolfsohn_uml_edu/broo...   \n",
       "T3_1_2022         /project/pi_sarah_gignouxwolfsohn_uml_edu/broo...   \n",
       "T3_19_2022        /project/pi_sarah_gignouxwolfsohn_uml_edu/broo...   \n",
       "T2_10_2022        /project/pi_sarah_gignouxwolfsohn_uml_edu/broo...   \n",
       "T1_57_2022        /project/pi_sarah_gignouxwolfsohn_uml_edu/broo...   \n",
       "Coral5.fastq                                                    NaN   \n",
       "T3-21-Mmea.fastq  /project/pi_sarah_gignouxwolfsohn_uml_edu/broo...   \n",
       "Coral1.fastq                                                    NaN   \n",
       "Coral2.fastq                                                    NaN   \n",
       "T3-4-Past.fastq                                                 NaN   \n",
       "Coral4.fastq                                                    NaN   \n",
       "_                                                                 _   \n",
       "\n",
       "                 Sample_Code = datecollected_tagnumber_transect_samplenumber_species  \n",
       "Seq_file                                                                              \n",
       "NaN                                                             NaN                   \n",
       "T3_8_2019                                                       NaN                   \n",
       "T2_16_2019                                                      NaN                   \n",
       "T1_20_2019                                                      NaN                   \n",
       "T1_12_2022                                                      NaN                   \n",
       "T1_24_2019                                                      NaN                   \n",
       "T3_9_2019                                                       NaN                   \n",
       "T3_14_2019                                                      NaN                   \n",
       "T3_1_2019                                                       NaN                   \n",
       "T3_15_2019                                                      NaN                   \n",
       "T1_16_2019                                                      NaN                   \n",
       "T1_40_2022                                                      NaN                   \n",
       "T1_70_2022                                                      NaN                   \n",
       "T3_48_2022                                                      NaN                   \n",
       "T3_49_2022                                                      NaN                   \n",
       "T3_51_2022                                                      NaN                   \n",
       "T3_40_2022                                                      NaN                   \n",
       "T3_60_2022                                                      NaN                   \n",
       "T1_13_2022                                                      NaN                   \n",
       "T3_13_2022                                                      NaN                   \n",
       "T3_1_2022                                                       NaN                   \n",
       "T3_19_2022                                                      NaN                   \n",
       "T2_10_2022                                                      NaN                   \n",
       "T1_57_2022                                                      NaN                   \n",
       "Coral5.fastq                                                    NaN                   \n",
       "T3-21-Mmea.fastq                                                NaN                   \n",
       "Coral1.fastq                                                    NaN                   \n",
       "Coral2.fastq                                                    NaN                   \n",
       "T3-4-Past.fastq                                                 NaN                   \n",
       "Coral4.fastq                                                    NaN                   \n",
       "_                                                                 _                   "
      ]
     },
     "execution_count": 194,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "meta_mcav"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 192,
   "id": "6270cf66-ab7a-4803-8c59-ce0f9e88ea54",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>Month_year</th>\n",
       "      <th>Country</th>\n",
       "      <th>Location</th>\n",
       "      <th>CollectionDate</th>\n",
       "      <th>Transect</th>\n",
       "      <th>TransectNum</th>\n",
       "      <th>OldTagNum</th>\n",
       "      <th>NewTagNum</th>\n",
       "      <th>SampleNum</th>\n",
       "      <th>Species</th>\n",
       "      <th>Sample_type</th>\n",
       "      <th>Health status</th>\n",
       "      <th>Sampling_notes</th>\n",
       "      <th>Sample_physical_location</th>\n",
       "      <th>Extraction_physical_location</th>\n",
       "      <th>Date_sequenced</th>\n",
       "      <th>Notes</th>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>Tubelabel_species</th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>122022_BEL_CBC_T4_2_OFAV</th>\n",
       "      <td>122022.0</td>\n",
       "      <td>BEL</td>\n",
       "      <td>CBC</td>\n",
       "      <td>12/5/22</td>\n",
       "      <td>CURLEW</td>\n",
       "      <td>4.0</td>\n",
       "      <td>NaN</td>\n",
       "      <td>77</td>\n",
       "      <td>2</td>\n",
       "      <td>OFAV</td>\n",
       "      <td>Core_EtOH</td>\n",
       "      <td>Healthy</td>\n",
       "      <td>NaN</td>\n",
       "      <td>UML_F1_R1_B5</td>\n",
       "      <td>UML_F1_R2_B3</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>92023_BEL_CBC_T1_171_PAST</th>\n",
       "      <td>92023.0</td>\n",
       "      <td>BEL</td>\n",
       "      <td>CBC</td>\n",
       "      <td>9/25/23</td>\n",
       "      <td>CBC30N</td>\n",
       "      <td>1.0</td>\n",
       "      <td>NaN</td>\n",
       "      <td>2</td>\n",
       "      <td>171</td>\n",
       "      <td>PAST</td>\n",
       "      <td>Core_RNAlater</td>\n",
       "      <td>Healthy</td>\n",
       "      <td>CLP 90%</td>\n",
       "      <td>UML_F1_R1_B10</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>92023_BEL_CBC_T1_172_PAST</th>\n",
       "      <td>92023.0</td>\n",
       "      <td>BEL</td>\n",
       "      <td>CBC</td>\n",
       "      <td>9/25/23</td>\n",
       "      <td>CBC30N</td>\n",
       "      <td>1.0</td>\n",
       "      <td>NaN</td>\n",
       "      <td>21</td>\n",
       "      <td>172</td>\n",
       "      <td>PAST</td>\n",
       "      <td>Core_RNAlater</td>\n",
       "      <td>Healthy</td>\n",
       "      <td>No CL</td>\n",
       "      <td>UML_F1_R1_B10</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>92023_BEL_CBC_T1_173_SSID</th>\n",
       "      <td>92023.0</td>\n",
       "      <td>BEL</td>\n",
       "      <td>CBC</td>\n",
       "      <td>9/25/23</td>\n",
       "      <td>CBC30N</td>\n",
       "      <td>1.0</td>\n",
       "      <td>NaN</td>\n",
       "      <td>3</td>\n",
       "      <td>173</td>\n",
       "      <td>SSID</td>\n",
       "      <td>Core_RNAlater</td>\n",
       "      <td>Healthy</td>\n",
       "      <td>CLP 80%; DC 20%</td>\n",
       "      <td>UML_F1_R1_B10</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>92023_BEL_CBC_T1_174_MCAV</th>\n",
       "      <td>92023.0</td>\n",
       "      <td>BEL</td>\n",
       "      <td>CBC</td>\n",
       "      <td>9/25/23</td>\n",
       "      <td>CBC30N</td>\n",
       "      <td>1.0</td>\n",
       "      <td>NaN</td>\n",
       "      <td>24</td>\n",
       "      <td>174</td>\n",
       "      <td>MCAV</td>\n",
       "      <td>Core_RNAlater</td>\n",
       "      <td>Healthy</td>\n",
       "      <td>CLP 10%</td>\n",
       "      <td>UML_F1_R1_B10</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>11_2023_CBC-BEL_T1_285_PSTR</th>\n",
       "      <td>112023.0</td>\n",
       "      <td>BEL</td>\n",
       "      <td>CBC</td>\n",
       "      <td>11/14/23</td>\n",
       "      <td>CBC30N</td>\n",
       "      <td>1.0</td>\n",
       "      <td>NaN</td>\n",
       "      <td>12</td>\n",
       "      <td>285</td>\n",
       "      <td>PSTR</td>\n",
       "      <td>Core_frozen</td>\n",
       "      <td>Healthy</td>\n",
       "      <td>100% CLP</td>\n",
       "      <td>TXST -80</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>11_2023_CBC-BEL_T1_292_PSTR</th>\n",
       "      <td>112023.0</td>\n",
       "      <td>BEL</td>\n",
       "      <td>CBC</td>\n",
       "      <td>11/14/23</td>\n",
       "      <td>CBC30N</td>\n",
       "      <td>1.0</td>\n",
       "      <td>NaN</td>\n",
       "      <td>12</td>\n",
       "      <td>292</td>\n",
       "      <td>PSTR</td>\n",
       "      <td>Immune</td>\n",
       "      <td>Healthy</td>\n",
       "      <td>100% CLP</td>\n",
       "      <td>TXST -80</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>11_2023_CBC-BEL_T1_277_OFAV/OANN</th>\n",
       "      <td>112023.0</td>\n",
       "      <td>BEL</td>\n",
       "      <td>CBC</td>\n",
       "      <td>11/14/23</td>\n",
       "      <td>CBC30N</td>\n",
       "      <td>1.0</td>\n",
       "      <td>NaN</td>\n",
       "      <td>25</td>\n",
       "      <td>277</td>\n",
       "      <td>OFAV/OANN</td>\n",
       "      <td>Core_RNAlater</td>\n",
       "      <td>Healthy</td>\n",
       "      <td>5% CLB, 5% CLP, 2% TL; TL likely from predatio...</td>\n",
       "      <td>TXST -20</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>11_2023_CBC-BEL_T1_284_OFAV/OANN</th>\n",
       "      <td>112023.0</td>\n",
       "      <td>BEL</td>\n",
       "      <td>CBC</td>\n",
       "      <td>11/14/23</td>\n",
       "      <td>CBC30N</td>\n",
       "      <td>1.0</td>\n",
       "      <td>NaN</td>\n",
       "      <td>25</td>\n",
       "      <td>284</td>\n",
       "      <td>OFAV/OANN</td>\n",
       "      <td>Core_frozen</td>\n",
       "      <td>Healthy</td>\n",
       "      <td>5% CLB, 5% CLP, 2% TL; TL likely from predatio...</td>\n",
       "      <td>TXST -80</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>11_2023_CBC-BEL_T1_291_OFAV/OANN</th>\n",
       "      <td>112023.0</td>\n",
       "      <td>BEL</td>\n",
       "      <td>CBC</td>\n",
       "      <td>11/14/23</td>\n",
       "      <td>CBC30N</td>\n",
       "      <td>1.0</td>\n",
       "      <td>NaN</td>\n",
       "      <td>25</td>\n",
       "      <td>291</td>\n",
       "      <td>OFAV/OANN</td>\n",
       "      <td>Immune</td>\n",
       "      <td>Healthy</td>\n",
       "      <td>5% CLB, 5% CLP, 2% TL; TL likely from predatio...</td>\n",
       "      <td>TXST -80</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>1269 rows × 17 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "                                  Month_year Country Location CollectionDate  \\\n",
       "Tubelabel_species                                                              \n",
       "122022_BEL_CBC_T4_2_OFAV            122022.0     BEL      CBC        12/5/22   \n",
       "92023_BEL_CBC_T1_171_PAST            92023.0     BEL      CBC        9/25/23   \n",
       "92023_BEL_CBC_T1_172_PAST            92023.0     BEL      CBC        9/25/23   \n",
       "92023_BEL_CBC_T1_173_SSID            92023.0     BEL      CBC        9/25/23   \n",
       "92023_BEL_CBC_T1_174_MCAV            92023.0     BEL      CBC        9/25/23   \n",
       "...                                      ...     ...      ...            ...   \n",
       "11_2023_CBC-BEL_T1_285_PSTR         112023.0     BEL      CBC       11/14/23   \n",
       "11_2023_CBC-BEL_T1_292_PSTR         112023.0     BEL      CBC       11/14/23   \n",
       "11_2023_CBC-BEL_T1_277_OFAV/OANN    112023.0     BEL      CBC       11/14/23   \n",
       "11_2023_CBC-BEL_T1_284_OFAV/OANN    112023.0     BEL      CBC       11/14/23   \n",
       "11_2023_CBC-BEL_T1_291_OFAV/OANN    112023.0     BEL      CBC       11/14/23   \n",
       "\n",
       "                                 Transect   TransectNum OldTagNum NewTagNum  \\\n",
       "Tubelabel_species                                                             \n",
       "122022_BEL_CBC_T4_2_OFAV            CURLEW          4.0       NaN        77   \n",
       "92023_BEL_CBC_T1_171_PAST           CBC30N          1.0       NaN         2   \n",
       "92023_BEL_CBC_T1_172_PAST           CBC30N          1.0       NaN        21   \n",
       "92023_BEL_CBC_T1_173_SSID           CBC30N          1.0       NaN         3   \n",
       "92023_BEL_CBC_T1_174_MCAV           CBC30N          1.0       NaN        24   \n",
       "...                                    ...          ...       ...       ...   \n",
       "11_2023_CBC-BEL_T1_285_PSTR         CBC30N          1.0       NaN        12   \n",
       "11_2023_CBC-BEL_T1_292_PSTR         CBC30N          1.0       NaN        12   \n",
       "11_2023_CBC-BEL_T1_277_OFAV/OANN    CBC30N          1.0       NaN        25   \n",
       "11_2023_CBC-BEL_T1_284_OFAV/OANN    CBC30N          1.0       NaN        25   \n",
       "11_2023_CBC-BEL_T1_291_OFAV/OANN    CBC30N          1.0       NaN        25   \n",
       "\n",
       "                                 SampleNum    Species    Sample_type  \\\n",
       "Tubelabel_species                                                      \n",
       "122022_BEL_CBC_T4_2_OFAV                 2       OFAV      Core_EtOH   \n",
       "92023_BEL_CBC_T1_171_PAST              171       PAST  Core_RNAlater   \n",
       "92023_BEL_CBC_T1_172_PAST              172       PAST  Core_RNAlater   \n",
       "92023_BEL_CBC_T1_173_SSID              173       SSID  Core_RNAlater   \n",
       "92023_BEL_CBC_T1_174_MCAV              174       MCAV  Core_RNAlater   \n",
       "...                                    ...        ...            ...   \n",
       "11_2023_CBC-BEL_T1_285_PSTR            285       PSTR    Core_frozen   \n",
       "11_2023_CBC-BEL_T1_292_PSTR            292       PSTR         Immune   \n",
       "11_2023_CBC-BEL_T1_277_OFAV/OANN       277  OFAV/OANN  Core_RNAlater   \n",
       "11_2023_CBC-BEL_T1_284_OFAV/OANN       284  OFAV/OANN    Core_frozen   \n",
       "11_2023_CBC-BEL_T1_291_OFAV/OANN       291  OFAV/OANN         Immune   \n",
       "\n",
       "                                 Health status  \\\n",
       "Tubelabel_species                                \n",
       "122022_BEL_CBC_T4_2_OFAV               Healthy   \n",
       "92023_BEL_CBC_T1_171_PAST              Healthy   \n",
       "92023_BEL_CBC_T1_172_PAST              Healthy   \n",
       "92023_BEL_CBC_T1_173_SSID              Healthy   \n",
       "92023_BEL_CBC_T1_174_MCAV              Healthy   \n",
       "...                                        ...   \n",
       "11_2023_CBC-BEL_T1_285_PSTR            Healthy   \n",
       "11_2023_CBC-BEL_T1_292_PSTR            Healthy   \n",
       "11_2023_CBC-BEL_T1_277_OFAV/OANN       Healthy   \n",
       "11_2023_CBC-BEL_T1_284_OFAV/OANN       Healthy   \n",
       "11_2023_CBC-BEL_T1_291_OFAV/OANN       Healthy   \n",
       "\n",
       "                                                                     Sampling_notes  \\\n",
       "Tubelabel_species                                                                     \n",
       "122022_BEL_CBC_T4_2_OFAV                                                        NaN   \n",
       "92023_BEL_CBC_T1_171_PAST                                                   CLP 90%   \n",
       "92023_BEL_CBC_T1_172_PAST                                                     No CL   \n",
       "92023_BEL_CBC_T1_173_SSID                                           CLP 80%; DC 20%   \n",
       "92023_BEL_CBC_T1_174_MCAV                                                   CLP 10%   \n",
       "...                                                                             ...   \n",
       "11_2023_CBC-BEL_T1_285_PSTR                                                100% CLP   \n",
       "11_2023_CBC-BEL_T1_292_PSTR                                                100% CLP   \n",
       "11_2023_CBC-BEL_T1_277_OFAV/OANN  5% CLB, 5% CLP, 2% TL; TL likely from predatio...   \n",
       "11_2023_CBC-BEL_T1_284_OFAV/OANN  5% CLB, 5% CLP, 2% TL; TL likely from predatio...   \n",
       "11_2023_CBC-BEL_T1_291_OFAV/OANN  5% CLB, 5% CLP, 2% TL; TL likely from predatio...   \n",
       "\n",
       "                                 Sample_physical_location  \\\n",
       "Tubelabel_species                                           \n",
       "122022_BEL_CBC_T4_2_OFAV                     UML_F1_R1_B5   \n",
       "92023_BEL_CBC_T1_171_PAST                   UML_F1_R1_B10   \n",
       "92023_BEL_CBC_T1_172_PAST                   UML_F1_R1_B10   \n",
       "92023_BEL_CBC_T1_173_SSID                   UML_F1_R1_B10   \n",
       "92023_BEL_CBC_T1_174_MCAV                   UML_F1_R1_B10   \n",
       "...                                                   ...   \n",
       "11_2023_CBC-BEL_T1_285_PSTR                      TXST -80   \n",
       "11_2023_CBC-BEL_T1_292_PSTR                      TXST -80   \n",
       "11_2023_CBC-BEL_T1_277_OFAV/OANN                 TXST -20   \n",
       "11_2023_CBC-BEL_T1_284_OFAV/OANN                 TXST -80   \n",
       "11_2023_CBC-BEL_T1_291_OFAV/OANN                 TXST -80   \n",
       "\n",
       "                                 Extraction_physical_location Date_sequenced  \\\n",
       "Tubelabel_species                                                              \n",
       "122022_BEL_CBC_T4_2_OFAV                         UML_F1_R2_B3            NaN   \n",
       "92023_BEL_CBC_T1_171_PAST                                 NaN            NaN   \n",
       "92023_BEL_CBC_T1_172_PAST                                 NaN            NaN   \n",
       "92023_BEL_CBC_T1_173_SSID                                 NaN            NaN   \n",
       "92023_BEL_CBC_T1_174_MCAV                                 NaN            NaN   \n",
       "...                                                       ...            ...   \n",
       "11_2023_CBC-BEL_T1_285_PSTR                               NaN            NaN   \n",
       "11_2023_CBC-BEL_T1_292_PSTR                               NaN            NaN   \n",
       "11_2023_CBC-BEL_T1_277_OFAV/OANN                          NaN            NaN   \n",
       "11_2023_CBC-BEL_T1_284_OFAV/OANN                          NaN            NaN   \n",
       "11_2023_CBC-BEL_T1_291_OFAV/OANN                          NaN            NaN   \n",
       "\n",
       "                                 Notes  \n",
       "Tubelabel_species                       \n",
       "122022_BEL_CBC_T4_2_OFAV           NaN  \n",
       "92023_BEL_CBC_T1_171_PAST          NaN  \n",
       "92023_BEL_CBC_T1_172_PAST          NaN  \n",
       "92023_BEL_CBC_T1_173_SSID          NaN  \n",
       "92023_BEL_CBC_T1_174_MCAV          NaN  \n",
       "...                                ...  \n",
       "11_2023_CBC-BEL_T1_285_PSTR        NaN  \n",
       "11_2023_CBC-BEL_T1_292_PSTR        NaN  \n",
       "11_2023_CBC-BEL_T1_277_OFAV/OANN   NaN  \n",
       "11_2023_CBC-BEL_T1_284_OFAV/OANN   NaN  \n",
       "11_2023_CBC-BEL_T1_291_OFAV/OANN   NaN  \n",
       "\n",
       "[1269 rows x 17 columns]"
      ]
     },
     "execution_count": 192,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "meta"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "c783c375-85ac-4644-92b1-a7990bd1247d",
   "metadata": {},
   "source": [
    "#### Merge Genus Level otu tables \n",
    "- goal: create taxa bar plots comparing composition and abundance between samples"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "c2d37641-29f7-4db0-beb4-07302f42a258",
   "metadata": {},
   "outputs": [],
   "source": [
    "# Merge bracken outputs \n",
    "#conda activate kraken\n",
    "\n",
    "otupath='/project/pi_sarah_gignouxwolfsohn_uml_edu/brooke/mcav/taxonomy/kraken'\n",
    "level='G'\n",
    "\n",
    "combine_bracken_outputs.py \\\n",
    "--files $otupath/*_$level.bracken \\\n",
    "--names 'T1_12_2022','T1_13_2022','T1_16_2019','T1_20_2019','T1_24_2019','T1_40_2022','T1_57_2022','T1_70_2022','T2_10_2022','T2_16_2019','T3_13_2022','T3_14_2019','T3_15_2019','T3_19_2022','T3_1_2019','T3_40_2022','T3_48_2022','T3_49_2022','T3_51_2022','T3_8_2019','T3_9_2019' \\\n",
    "--output $otupath/merged_bracken_genus.txt\n",
    "\n",
    "#Keep getting this output:\n",
    "    #PROGRAM START TIME: 12-18-2023 14:47:19\n",
    "#Processing Output File /project/pi_sarah_gignouxwolfsohn_uml_edu/brooke/mcav/taxonomy/kraken/T1_12_2022_G.bracken:: Sample T1_12_2022\n",
    "#Taxonomy IDs not matching for species uncultured: (27579        27604)\n",
    "\n",
    "# merging manually now"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 12,
   "id": "4ace464a-8688-4bf9-b580-a894c5511f4a",
   "metadata": {},
   "outputs": [],
   "source": [
    "import pandas as pd\n",
    "\n",
    "# Create a dictionary to store DataFrames\n",
    "dfs = {}\n",
    "level='G'\n",
    "otupath='/project/pi_sarah_gignouxwolfsohn_uml_edu/brooke/mcav/taxonomy/kraken'\n",
    "# Read files based on sampleid and store each in the dictionary\n",
    "for sample in sampleid:\n",
    "    filename = f\"{otupath}/{sample}_{level}.bracken\"\n",
    "    df = pd.read_csv(filename, sep='\\t', header=0)\n",
    "    \n",
    "    # Store DataFrame in the dictionary with sampleid as the key\n",
    "    dfs[sample] = df\n",
    "#dfs"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 32,
   "id": "c906baef-cd09-47ac-aef8-d0fc244fbc7a",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>name</th>\n",
       "      <th>taxonomy_id</th>\n",
       "      <th>taxonomy_lvl</th>\n",
       "      <th>kraken_assigned_reads</th>\n",
       "      <th>added_reads</th>\n",
       "      <th>new_est_reads</th>\n",
       "      <th>fraction_total_reads</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>6892</th>\n",
       "      <td>1174-901-12</td>\n",
       "      <td>26040</td>\n",
       "      <td>G</td>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "      <td>0.00003</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>9162</th>\n",
       "      <td>1921-2</td>\n",
       "      <td>25191</td>\n",
       "      <td>G</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0.00000</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>9163</th>\n",
       "      <td>1921-3</td>\n",
       "      <td>25192</td>\n",
       "      <td>G</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0.00000</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>9164</th>\n",
       "      <td>1959-1</td>\n",
       "      <td>25193</td>\n",
       "      <td>G</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0.00000</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>6176</th>\n",
       "      <td>2013Ark19i</td>\n",
       "      <td>26350</td>\n",
       "      <td>G</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0.00000</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>787</th>\n",
       "      <td>uncultured</td>\n",
       "      <td>27571</td>\n",
       "      <td>G</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0.00000</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>8562</th>\n",
       "      <td>uncultured</td>\n",
       "      <td>24999</td>\n",
       "      <td>G</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0.00000</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1929</th>\n",
       "      <td>uncultured</td>\n",
       "      <td>28584</td>\n",
       "      <td>G</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0.00000</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>9472</th>\n",
       "      <td>wb1-A12</td>\n",
       "      <td>26896</td>\n",
       "      <td>G</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0.00000</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>6603</th>\n",
       "      <td>wb1-P19</td>\n",
       "      <td>26749</td>\n",
       "      <td>G</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0.00000</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>9740 rows × 7 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "             name  taxonomy_id taxonomy_lvl  kraken_assigned_reads  \\\n",
       "6892  1174-901-12        26040            G                      1   \n",
       "9162       1921-2        25191            G                      0   \n",
       "9163       1921-3        25192            G                      0   \n",
       "9164       1959-1        25193            G                      0   \n",
       "6176   2013Ark19i        26350            G                      0   \n",
       "...           ...          ...          ...                    ...   \n",
       "787    uncultured        27571            G                      0   \n",
       "8562   uncultured        24999            G                      0   \n",
       "1929   uncultured        28584            G                      0   \n",
       "9472      wb1-A12        26896            G                      0   \n",
       "6603      wb1-P19        26749            G                      0   \n",
       "\n",
       "      added_reads  new_est_reads  fraction_total_reads  \n",
       "6892            0              1               0.00003  \n",
       "9162            0              0               0.00000  \n",
       "9163            0              0               0.00000  \n",
       "9164            0              0               0.00000  \n",
       "6176            0              0               0.00000  \n",
       "...           ...            ...                   ...  \n",
       "787             0              0               0.00000  \n",
       "8562            0              0               0.00000  \n",
       "1929            0              0               0.00000  \n",
       "9472            0              0               0.00000  \n",
       "6603            0              0               0.00000  \n",
       "\n",
       "[9740 rows x 7 columns]"
      ]
     },
     "execution_count": 32,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "T1_12_2022=pd.DataFrame(dfs['T1_12_2022']).sort_values(by='name')\n",
    "# can't figure out how to put this in the for loop...want each sample to have its own df and need them to be sorted the same"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 40,
   "id": "489d72e4-af11-49ea-9962-43e9cc8de6bb",
   "metadata": {},
   "outputs": [
    {
     "ename": "NameError",
     "evalue": "name 'T3_9_2019' is not defined",
     "output_type": "error",
     "traceback": [
      "\u001b[0;31m---------------------------------------------------------------------------\u001b[0m",
      "\u001b[0;31mNameError\u001b[0m                                 Traceback (most recent call last)",
      "Cell \u001b[0;32mIn[40], line 3\u001b[0m\n\u001b[1;32m      1\u001b[0m \u001b[38;5;28;01mfor\u001b[39;00m sample \u001b[38;5;129;01min\u001b[39;00m sampleid:\n\u001b[1;32m      2\u001b[0m     sample\u001b[38;5;241m=\u001b[39mpd\u001b[38;5;241m.\u001b[39mDataFrame(dfs[sample])\u001b[38;5;241m.\u001b[39msort_values(by\u001b[38;5;241m=\u001b[39m\u001b[38;5;124m'\u001b[39m\u001b[38;5;124mname\u001b[39m\u001b[38;5;124m'\u001b[39m)\n\u001b[0;32m----> 3\u001b[0m \u001b[43mT3_9_2019\u001b[49m\n",
      "\u001b[0;31mNameError\u001b[0m: name 'T3_9_2019' is not defined"
     ]
    }
   ],
   "source": [
    "for sample in sampleid:\n",
    "    sample=pd.DataFrame(dfs[sample]).sort_values(by='name')\n",
    "T3_9_2019"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 17,
   "id": "8bfc29d5-c50b-4a9d-8d77-96ac8fef4e6f",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>name</th>\n",
       "      <th>taxonomy_id</th>\n",
       "      <th>taxonomy_lvl</th>\n",
       "      <th>kraken_assigned_reads</th>\n",
       "      <th>added_reads</th>\n",
       "      <th>new_est_reads</th>\n",
       "      <th>fraction_total_reads</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>7161</th>\n",
       "      <td>1174-901-12</td>\n",
       "      <td>26040</td>\n",
       "      <td>G</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0.00000</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>8722</th>\n",
       "      <td>1921-2</td>\n",
       "      <td>25191</td>\n",
       "      <td>G</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0.00000</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>8723</th>\n",
       "      <td>1921-3</td>\n",
       "      <td>25192</td>\n",
       "      <td>G</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0.00000</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>8724</th>\n",
       "      <td>1959-1</td>\n",
       "      <td>25193</td>\n",
       "      <td>G</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0.00000</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>6431</th>\n",
       "      <td>2013Ark19i</td>\n",
       "      <td>26350</td>\n",
       "      <td>G</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0.00000</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>9163</th>\n",
       "      <td>uncultured</td>\n",
       "      <td>24539</td>\n",
       "      <td>G</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0.00000</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>7483</th>\n",
       "      <td>uncultured</td>\n",
       "      <td>11484</td>\n",
       "      <td>G</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0.00000</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>9167</th>\n",
       "      <td>uncultured</td>\n",
       "      <td>14229</td>\n",
       "      <td>G</td>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "      <td>0.00003</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>9489</th>\n",
       "      <td>wb1-A12</td>\n",
       "      <td>26896</td>\n",
       "      <td>G</td>\n",
       "      <td>1</td>\n",
       "      <td>4</td>\n",
       "      <td>5</td>\n",
       "      <td>0.00017</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>6787</th>\n",
       "      <td>wb1-P19</td>\n",
       "      <td>26749</td>\n",
       "      <td>G</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0.00000</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>9740 rows × 7 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "             name  taxonomy_id taxonomy_lvl  kraken_assigned_reads  \\\n",
       "7161  1174-901-12        26040            G                      0   \n",
       "8722       1921-2        25191            G                      0   \n",
       "8723       1921-3        25192            G                      0   \n",
       "8724       1959-1        25193            G                      0   \n",
       "6431   2013Ark19i        26350            G                      0   \n",
       "...           ...          ...          ...                    ...   \n",
       "9163   uncultured        24539            G                      0   \n",
       "7483   uncultured        11484            G                      0   \n",
       "9167   uncultured        14229            G                      1   \n",
       "9489      wb1-A12        26896            G                      1   \n",
       "6787      wb1-P19        26749            G                      0   \n",
       "\n",
       "      added_reads  new_est_reads  fraction_total_reads  \n",
       "7161            0              0               0.00000  \n",
       "8722            0              0               0.00000  \n",
       "8723            0              0               0.00000  \n",
       "8724            0              0               0.00000  \n",
       "6431            0              0               0.00000  \n",
       "...           ...            ...                   ...  \n",
       "9163            0              0               0.00000  \n",
       "7483            0              0               0.00000  \n",
       "9167            0              1               0.00003  \n",
       "9489            4              5               0.00017  \n",
       "6787            0              0               0.00000  \n",
       "\n",
       "[9740 rows x 7 columns]"
      ]
     },
     "execution_count": 17,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "dfs['T1_13_2022'].sort_values(by='name')"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 15,
   "id": "150c6fc9-fe82-4253-af94-6cefabe7c3bb",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "['T1_12_2022',\n",
       " 'T1_13_2022',\n",
       " 'T1_16_2019',\n",
       " 'T1_20_2019',\n",
       " 'T1_24_2019',\n",
       " 'T1_40_2022',\n",
       " 'T1_57_2022',\n",
       " 'T1_70_2022',\n",
       " 'T2_10_2022',\n",
       " 'T2_16_2019',\n",
       " 'T3_13_2022',\n",
       " 'T3_14_2019',\n",
       " 'T3_15_2019',\n",
       " 'T3_19_2022',\n",
       " 'T3_1_2019',\n",
       " 'T3_40_2022',\n",
       " 'T3_48_2022',\n",
       " 'T3_49_2022',\n",
       " 'T3_51_2022',\n",
       " 'T3_8_2019',\n",
       " 'T3_9_2019']"
      ]
     },
     "execution_count": 15,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "sampleid"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 19,
   "id": "7c33bc4c-be3a-4d57-a9e9-28b0ae76158a",
   "metadata": {},
   "outputs": [],
   "source": [
    "import matplotlib.pyplot as plt\n",
    "import numpy as np"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "cf22b714-2a29-4d31-ae44-da8d01f28255",
   "metadata": {},
   "outputs": [],
   "source": [
    "# Example:\n",
    "# Data\n",
    "groups = ['G1', 'G2', 'G3', 'G4', 'G5']\n",
    "values1 = [12, 19, 14, 27, 16]\n",
    "values2 = [21, 30, 15, 17, 20]\n",
    "values3 = [15, 23, 12, 11, 15]\n",
    "\n",
    "fig, ax = plt.subplots()\n",
    "\n",
    "# Stacked bar chart\n",
    "ax.bar(groups, values1)\n",
    "ax.bar(groups, values2, bottom = values1)\n",
    "ax.bar(groups, values3, bottom = np.add(values1, values2))\n",
    "\n",
    "# plt.show() \n"
   ]
  }
 ],
 "metadata": {
  "kernelspec": {
   "display_name": "Python [conda env:.conda-skbio]",
   "language": "python",
   "name": "conda-env-.conda-skbio-py"
  },
  "language_info": {
   "codemirror_mode": {
    "name": "ipython",
    "version": 3
   },
   "file_extension": ".py",
   "mimetype": "text/x-python",
   "name": "python",
   "nbconvert_exporter": "python",
   "pygments_lexer": "ipython3",
   "version": "3.8.18"
  }
 },
 "nbformat": 4,
 "nbformat_minor": 5
}
