{
 "cells": [
  {
   "cell_type": "markdown",
   "id": "7b64ba0e-624e-404a-832e-de0ab81dc810",
   "metadata": {},
   "source": [
    "### Taxonomic Profiling on MAGS\n",
    "- Can't figure out on anvio so for now trying other options\n",
    "- option 1: sylph \n",
    "- option 2: kraken"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "131c9931-f55e-41d1-ab97-20fac3a466cc",
   "metadata": {
    "jp-MarkdownHeadingCollapsed": true,
    "tags": []
   },
   "source": [
    "#### Sylph\n",
    "https://github.com/bluenote-1577/sylph\n",
    "- average nucleic identity (ANI) querying or taxonomic profiling"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "3345e46d-9dde-4dea-88cd-4be610cc4260",
   "metadata": {},
   "outputs": [],
   "source": [
    "#activate unity interactive job..need high memory for installation\n",
    "salloc -p cpu --mem 150G"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "a7b025ed-0786-4283-a3be-b5343e54d059",
   "metadata": {},
   "outputs": [],
   "source": [
    "%conda init --all"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "dfb5beeb-938f-4145-897b-99a0c4d22899",
   "metadata": {},
   "outputs": [],
   "source": [
    "#module load miniconda/22.11.1-1\n",
    "#conda create -n taxa\n",
    "%conda activate taxa"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "238f2098-b303-43b1-8ad9-604d8216f719",
   "metadata": {},
   "outputs": [],
   "source": [
    "#install sylph package\n",
    "conda install -c bioconda sylph\n",
    "#if above doesnt work do: \n",
    "conda install -c bioconda -c conda-forge sylph"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "a75cf3d5-7ff1-47db-ba4a-b2faf6cff548",
   "metadata": {},
   "outputs": [],
   "source": [
    "#set parameters\n",
    "SAMPLENAME=mcav\n",
    "CONTIGPATH=/project/pi_sarah_gignouxwolfsohn_uml_edu/brooke/results/mcav_assembly3\n",
    "CONTIGFILE=mcav.contigs-fixed.fa"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "780258a3-bc5d-46fb-acce-3419e19713b3",
   "metadata": {},
   "outputs": [],
   "source": [
    "#sketch paired-end reads\n",
    "sylph sketch -1 ./trimmed/*_R1_001_val_1.fq -2 ./trimmed/*_R2_001_val_2.fq -d \"$SAMPLENAME\"_sketches -t 50 \n",
    "#decrease -c (default =200)..."
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "8e17431b-56de-4ee6-a3d8-9de32ae1c080",
   "metadata": {},
   "outputs": [],
   "source": [
    "#create database if contigs are all in one fasta file\n",
    "sylph sketch $CONTIGPATH/$CONTIGFILE -i -o ./working/\"$SAMPLENAME\"_contig_database"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "6bba0587-108a-410a-9efd-eca97bb47bdb",
   "metadata": {},
   "outputs": [],
   "source": [
    "# query for ANI \n",
    "sylph query \"$SAMPLENAME\"_sketches/*.sylsp ./working/\"$SAMPLENAME\"_contig_database.syldb\n",
    "#output is statistics for each genome against our reads\n",
    "#redirect output??\n",
    "#Thus sylph query is not a profiler. It does not tell you the abundance of the genomes in your sample, just how similar your query genome is to the genomes within the sample."
   ]
  },
  {
   "cell_type": "raw",
   "id": "452b0bcd-b16a-4749-9e2e-88e076317274",
   "metadata": {},
   "source": [
    "Sample_file\tGenome_file\tAdjusted_ANI\tEff_cov\tANI_5-95_percentile\tEff_lambda\tLambda_5-95_percentile\tMedian_cov\tMean_cov_geq1\tContainment_ind\tNaive_ANI\tContig_name\n",
    "./trimmed/T1_12_2022_MCAV_R1_001_val_1.fq\t/project/pi_sarah_gignouxwolfsohn_uml_edu/brooke/results/mcav_assembly3/mcav.contigs-fixed.fa\t100.00\t8.213\tNA-NA\tHIGH\tNA-NA\t8\t8.213\t61/61\t100.00\tc_000000078117\n",
    "./trimmed/T1_12_2022_MCAV_R1_001_val_1.fq\t/project/pi_sarah_gignouxwolfsohn_uml_edu/brooke/results/mcav_assembly3/mcav.contigs-fixed.fa\t100.00\t16.000\tNA-NA\tHIGH\tNA-NA\t16\t15.440\t50/50\t100.00\tc_000000159761\n",
    "./trimmed/T1_12_2022_MCAV_R1_001_val_1.fq\t/project/pi_sarah_gignouxwolfsohn_uml_edu/brooke/results/mcav_assembly3/mcav.contigs-fixed.fa\t100.00\t822.000\tNA-NA\tHIGH\tNA-NA\t822\t697.089\t56/56\t100.00\tc_000000263852\n",
    "./trimmed/T1_12_2022_MCAV_R1_001_val_1.fq\t/project/pi_sarah_gignouxwolfsohn_uml_edu/brooke/results/mcav_assembly3/mcav.contigs-fixed.fa\t100.00\t33.000\tNA-NA\tHIGH\tNA-NA\t33\t32.960\t50/50\t100.00\tc_000000432420\n",
    "./trimmed/T1_12_2022_MCAV_R1_001_val_1.fq\t/project/pi_sarah_gignouxwolfsohn_uml_edu/brooke/results/mcav_assembly3/mcav.contigs-fixed.fa\t100.00\t6.722\tNA-NA\tHIGH\tNA-NA\t7\t6.722\t90/90\t100.00\tc_000000537045\n",
    "./trimmed/T1_12_2022_MCAV_R1_001_val_1.fq\t/project/pi_sarah_gignouxwolfsohn_uml_edu/brooke/results/mcav_assembly3/mcav.contigs-fixed.fa\t100.00\t8.125\tNA-NA\tHIGH\tNA-NA\t8\t8.125\t56/56\t100.00\tc_000000694960\n",
    "./trimmed/T1_12_2022_MCAV_R1_001_val_1.fq\t/project/pi_sarah_gignouxwolfsohn_uml_edu/brooke/results/mcav_assembly3/mcav.contigs-fixed.fa\t100.00\t6.635\tNA-NA\tHIGH\tNA-NA\t7\t6.635\t52/52\t100.00\tc_000000696837\n",
    "./trimmed/T1_12_2022_MCAV_R1_001_val_1.fq\t/project/pi_sarah_gignouxwolfsohn_uml_edu/brooke/results/mcav_assembly3/mcav.contigs-fixed.fa\t100.00\t28.000\tNA-NA\tHIGH\tNA-NA\t28\t27.696\t56/56\t100.00\tc_000000711824\n",
    "./trimmed/T1_12_2022_MCAV_R1_001_val_1.fq\t/project/pi_sarah_gignouxwolfsohn_uml_edu/brooke/results/mcav_assembly3/mcav.contigs-fixed.fa\t99.97\t20.000\tNA-NA\tHIGH\tNA-NA\t20\t17.292\t96/97\t99.97\tc_000000243370\n",
    "./trimmed/T1_12_2022_MCAV_R1_001_val_1.fq\t/project/pi_sarah_gignouxwolfsohn_uml_edu/brooke/results/mcav_assembly3/mcav.contigs-fixed.fa\t99.95\t10.898\tNA-NA\tHIGH\tNA-NA\t11\t10.898\t59/60\t99.95\tc_000000716083\n",
    "./trimmed/T1_12_2022_MCAV_R1_001_val_1.fq\t/project/pi_sarah_gignouxwolfsohn_uml_edu/brooke/results/mcav_assembly3/mcav.contigs-fixed.fa\t99.94\t43.000\tNA-NA\tHIGH\tNA-NA\t43\t43.240\t50/51\t99.94\tc_000000045189\n",
    "./trimmed/T1_12_2022_MCAV_R1_001_val_1.fq\t/project/pi_sarah_gignouxwolfsohn_uml_edu/brooke/results/mcav_assembly3/mcav.contigs-fixed.fa\t99.90\t4.523\tNA-NA\tHIGH\tNA-NA\t4\t4.523\t65/67\t99.90\tc_000000500791\n",
    "./trimmed/T1_12_2022_MCAV_R1_001_val_1.fq\t/project/pi_sarah_gignouxwolfsohn_uml_edu/brooke/results/mcav_assembly3/mcav.contigs-fixed.fa\t99.88\t16.000\tNA-NA\tHIGH\tNA-NA\t16\t15.500\t52/54\t99.88\tc_000000078627"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "bc53dd73-5902-4c2c-a4ee-c3d403a4505d",
   "metadata": {},
   "outputs": [],
   "source": [
    "#Import taxa database: https://github.com/bluenote-1577/sylph/wiki/Pre%E2%80%90built-databases\n",
    "wget https://storage.googleapis.com/sylph-stuff/v0.3-c200-gtdb-r214.syldb -O gtdb_database.syldb"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "6be41ed2-31bd-49af-b653-01e8a859e441",
   "metadata": {},
   "outputs": [],
   "source": [
    "#move database into folder called gtdb_genomes_reps_r214\n",
    "mkdir working/gtdb_genomes_reps_r214\n",
    "mv gtdb_database.syldb working/gtdb_genomes_reps_r214/gtdb_database.syldb\n",
    "\n",
    "#set path to GTDB\n",
    "DB=working/gtdb_genomes_reps_r214/gtdb_database.syldb\n",
    "#set input & output path\n",
    "INDIR=mcav_sketches\n",
    "OUTDIR=results/taxonomy"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "f253d868-e4b9-41f3-99b5-25dcea6cea05",
   "metadata": {},
   "outputs": [],
   "source": [
    "#maps samples against contig database???\n",
    "#read more: https://github.com/bluenote-1577/sylph/wiki/sylph-cookbook#creating-a-database-of-fasta-files\n",
    "sylph profile \"$SAMPLENAME\"_sketches/*.sylsp ./working/\"$SAMPLENAME\"_contig_database.syldb -t 50 -o ./results/\"$SAMPLENAME\"_sylph_profile.tsv"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "db901a55-27b8-4104-b1fa-9ec1df93412e",
   "metadata": {},
   "outputs": [],
   "source": [
    "#map reads against GTDB \n",
    "sylph profile $INDIR/*_MCAV_R1_001_val_1.fq.paired.sylsp \"$DB\" -t 10 -o $OUTDIR/sylph_GTDB.tsv"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "79605a6c-c5c5-4053-ac3f-a186f5a7a49b",
   "metadata": {},
   "outputs": [],
   "source": [
    "#get taxonomic profile \n",
    "#clone repo (in \"working\" dir)\n",
    "git clone https://github.com/bluenote-1577/sylph\n",
    "python working/sylph/scripts/sylph_to_taxprof.py -s $OUTDIR/sylph_GTDB.tsv -m  working/sylph/scripts/gtdb_r214_metadata.tsv.gz -o $OUTDIR/prefix_.tsv\n",
    "\n",
    "#results: only 6 samples have hits on the reference genomes database"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "c3fdbc2b-71ca-4c59-aa2e-7a827289b1fb",
   "metadata": {},
   "outputs": [],
   "source": [
    "#convert to tsv\n",
    "\n",
    "# Loop over all files \n",
    "for input_file in prefix_*_R1_001_val_1.fq.sylphmpa; do\n",
    "    # Extract the unique identifier from the file name\n",
    "    unique_identifier=$(echo \"$input_file\" | sed 's/^prefix_\\(.*\\)_R1_001_val_1\\.fq\\.sylphmpa$/\\1/')\n",
    "\n",
    "    # Generate the output TSV file name\n",
    "    output_file=\"${unique_identifier}.tsv\"\n",
    "\n",
    "    # Process the input file and save the result to the output file\n",
    "    awk -F'\\t' '{for (i=1; i<=NF; i++) printf \"%s%s\", $i, (i<NF ? \"\\t\" : \"\\n\")}' \"$input_file\" > \"$output_file\"\n",
    "\n",
    "    echo \"Processed $input_file and saved as $output_file\"\n",
    "done\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "4e496080-46b1-46bd-a409-430893cd6c7e",
   "metadata": {},
   "outputs": [],
   "source": [
    "#repeating with OceanDNA database\n",
    "wget https://storage.googleapis.com/sylph-stuff/OceanDNA-c200-v0.3.syldb\n",
    "mv OceanDNA-c200-v0.3.syldb working/gtdb_genomes_reps_r214/OceanDNA-c200-v0.3.syldb\n",
    "oceanDB=working/gtdb_genomes_reps_r214/OceanDNA-c200-v0.3.syldb\n",
    "\n",
    "sylph profile $INDIR/*_MCAV_R1_001_val_1.fq.paired.sylsp \"$oceanDB\" -t 10 -o $OUTDIR/sylph_ocean.tsv"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "5a7bde5c-0abf-4a7b-85e9-1b6c8ad68166",
   "metadata": {
    "tags": []
   },
   "source": [
    "#### Kraken\n",
    "- https://ccb.jhu.edu/software/kraken/MANUAL.html"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "8b5db189-87bf-404d-a07b-b0a64c401772",
   "metadata": {},
   "source": [
    "having trouble building \"standard\" databases...using silva anyway\n",
    "- ask unity?\n",
    "https://bioinformaticsworkbook.org/dataAnalysis/Metagenomics/Kraken.html#gsc.tab=0\n",
    "\n",
    "https://bioinformatics.stackexchange.com/questions/20589/kraken2-standard-database-failing-to-build-unexpected-ftp-path?rq=1\n",
    "\n",
    "https://github.com/DerrickWood/kraken2/issues/292"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "e1589f46-cd88-4e3c-8609-e76b7e35ce86",
   "metadata": {},
   "outputs": [],
   "source": [
    "module load kraken2/2.1.2"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "9448ab64-f479-4ca5-9724-389867783026",
   "metadata": {},
   "outputs": [],
   "source": [
    "# download SILVA database\n",
    "wget -c url https://genome-idx.s3.amazonaws.com/kraken/16S_Silva132_20200326.tgz\n",
    "#more db options (thinking about 'standard': https://benlangmead.github.io/aws-indexes/k2\n",
    "# unzip \n",
    "tar -xvzf //project/pi_sarah_gignouxwolfsohn_uml_edu/Reference_genomes/ref_databases/16S_Silva132_20200326.tgz"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "db934654-5e5f-427e-b888-675066ed0c6d",
   "metadata": {},
   "outputs": [],
   "source": [
    "#!/bin/bash\n",
    "#SBATCH -c 24  # Number of Cores per Task\n",
    "#SBATCH --mem=180G  # Requested Memory\n",
    "#SBATCH -p cpu  # Partition\n",
    "#SBATCH -t 24:00:00  # Job time limit\n",
    "#SBATCH --mail-type=ALL\n",
    "#SBATCH -o //project/pi_sarah_gignouxwolfsohn_uml_edu/brooke/bash_scripts/slurm-kraken-%j.out  # %j = job ID\n",
    "\n",
    "module load miniconda/22.11.1-1\n",
    "module load kraken2/2.1.2\n",
    "\n",
    "# i think this specifically downloads NCBI and RefSeq\n",
    "#kraken2-build --build --standard --threads 24 --db $DBNAME\n",
    "#error with std db: kraken2: database (\"//project/pi_sarah_gignouxwolfsohn_uml_edu/Reference_genomes/ref_databases/standard\") does not contain necessary file taxo.k2d\n",
    "#/var/spool/slurm/slurmd/job15982456/slurm_script: line 23: --report: command not found\n",
    "# see error notes above\n",
    "\n",
    "# set variables\n",
    "DBNAME=//project/pi_sarah_gignouxwolfsohn_uml_edu/Reference_genomes/ref_databases/16S_SILVA132_k2db\n",
    "READSPATH=/project/pi_sarah_gignouxwolfsohn_uml_edu/brooke/trimmed\n",
    "OUTDIR=//project/pi_sarah_gignouxwolfsohn_uml_edu/brooke/mcav_taxonomy/kraken\n",
    "\n",
    "#mkdir /project/pi_sarah_gignouxwolfsohn_uml_edu/brooke/mcav_taxonomy\n",
    "#mkdir /project/pi_sarah_gignouxwolfsohn_uml_edu/brooke/mcav_taxonomy/kraken\n",
    "# classify each set of paired end reads against silva taxonomic database\n",
    "for f in T1_12_2022 T1_13_2022 T1_16_2019 T1_20_2019 T1_24_2019 T1_40_2022 T1_57_2022 T1_70_2022 T2_10_2022 T2_16_2019 T3_13_2022 T3_14_2019 T3_15_2019 T3_19_2022 T3_1_2019 T3_40_2022 T3_48_2022 T3_49_2022 T3_51_2022 T3_60_2022 T3_8_2019 T3_9_2019\n",
    "do\n",
    "kraken2 --db $DBNAME --threads 4 --report $OUTDIR/${f}.kreport2 --paired $READSPATH/${f}_MCAV_R1_001_val_1.fq $READSPATH/${f}_MCAV_R2_001_val_2.fq > $OUTDIR/${f}.kraken2\n",
    "done\n",
    "\n",
    "#bash script: kraken\n",
    "#JOB ID: 16083608\n",
    "# SUCCESS!"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "47f2598b-1b3f-470b-9c24-647f1d48f7f0",
   "metadata": {},
   "source": [
    "##### BRACKEN\n",
    "- est species abundances??\n",
    "https://github.com/jenniferlu717/Bracken/blob/master/README.md"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "75ef3909-578a-4ab8-898b-ae38dcf61303",
   "metadata": {},
   "outputs": [],
   "source": [
    "# Installation\n",
    "\n",
    "#create symlink\n",
    "ln -s //project/pi_sarah_gignouxwolfsohn_uml_edu/brooke/bash_scripts  //project/pi_sarah_gignouxwolfsohn_uml_edu/brooke\n",
    "ln -s //project/pi_sarah_gignouxwolfsohn_uml_edu/brooke/bash_scripts //project/pi_sarah_gignouxwolfsohn_uml_edu/brooke\n",
    "#execute bracken scripts from second dir path "
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "8483c2db-36da-49f2-a146-b18f7233e439",
   "metadata": {},
   "outputs": [],
   "source": [
    "#!/bin/bash\n",
    "#SBATCH -c 24  # Number of Cores per Task\n",
    "#SBATCH --mem=180G  # Requested Memory\n",
    "#SBATCH -p cpu  # Partition\n",
    "#SBATCH -t 24:00:00  # Job time limit\n",
    "#SBATCH --mail-type=ALL\n",
    "#SBATCH -o //project/pi_sarah_gignouxwolfsohn_uml_edu/brooke/bash_scripts/slurm-bracken-%j.out  # %j = job ID\n",
    "\n",
    "module load miniconda/22.11.1-1\n",
    "module activate kraken\n",
    "module load kraken2/2.1.2\n",
    "\n",
    "DBNAME=//project/pi_sarah_gignouxwolfsohn_uml_edu/Reference_genomes/ref_databases/16S_SILVA132_k2db\n",
    "THREADS=3\n",
    "KMER_LEN=35\n",
    "READ_LEN=250\n",
    "BRACKEN_BUILD_PATH=/project/pi_sarah_gignouxwolfsohn_uml_edu/brooke/bash_scripts\n",
    "# Generate bracken database - just do once\n",
    "#$BRACKEN_BUILD_PATH/bracken-build.txt -d ${DBNAME} -t ${THREADS} -k ${KMER_LEN} -l ${READ_LEN} \n",
    "#bracken-build -d ${DBNAME} -k ${KMER_LEN} -l ${READ_LEN} \n",
    "#${KRAKEN_DB}`  = location of the built Kraken 1/Kraken 2/KrakenUniq database\n",
    "#${THREADS}`    = number of threads to use with Kraken and the Bracken scripts\n",
    "#${KMER_LEN}`   = length of kmer used to build the Kraken database \n",
    "    #Kraken 1/KrakenUniq default kmer length = 31\n",
    "    #Kraken 2 default kmer length = 35\n",
    "    #Default set in the script is 35. \n",
    "#${READ_LEN}`   = the read length of your data e.g., if you are using 100 bp reads, set it to `100`. \n",
    "#need taxo.k2d, hash.k2d, and opts.k2d. in kraken2 database\n",
    "\n",
    "# Abundance Estimation\n",
    "FILEPATH=//project/pi_sarah_gignouxwolfsohn_uml_edu/brooke/mcav_taxonomy/kraken\n",
    "#ls $FILEPATH | find *.kreport2 > kraken_list.txt\n",
    "#cat kraken_list.txt| sed 's/.kreport2/''/' > SAMPLEID\n",
    "SCRIPTPATH=/project/pi_sarah_gignouxwolfsohn_uml_edu/brooke/bash_scripts/KrakenTools-master/DiversityTools\n",
    "#$BRACKEN_BUILD_PATH/bash install_bracken.sh\n",
    "\n",
    "while IFS= read -r line; do\n",
    "    # Execute bracken command for each file\n",
    "    bracken -d \"$DBNAME\" -i $FILEPATH/\"$line\" -o \"$FILEPATH/${line%.kreport2}.bracken\" -r \"$READ_LEN\" -l 'G'\n",
    "done < $FILEPATH/kraken_list.txt\n",
    "#${SAMPLE}.kreport - the kraken report generated for a given dataset\n",
    "#{BRACKEN_OUTPUT_FILE}.bracken - the desired name of the output file to be generated by the code\n",
    "#The following optional parameters may be specified:\n",
    "\n",
    "#${LEVEL} - Default = 'S'. This specifies that abundance estimation will calculate estimated reads for each species. Other possible options are K (kingdom level), P (phylum), C (class), O (order), F (family), and G (genus).\n",
    "#${THRESHOLD} - Default = 10. For species classification, any species with <= 10 (or otherwise specified) reads will not receive any additional reads from higher taxonomy levels when distributing reads for abundance estimation. \n",
    "#If another classification level is specified, thresholding will occur at that level.\n",
    "\n",
    "#repeating for speceis just for funsies\n",
    "while IFS= read -r line; do\n",
    "    # Execute bracken command for each file\n",
    "    bracken -d \"$DBNAME\" -i $FILEPATH/\"$line\" -o \"$FILEPATH/${line%.kreport2}_species.bracken\" -r \"$READ_LEN\"\n",
    "done < kraken_list.txt\n",
    "\n",
    "#bash script: bracken_script \n",
    "#JOB ID: 16083669"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "839dd097-ecac-4a1e-aa5a-76d1e2363be8",
   "metadata": {
    "jp-MarkdownHeadingCollapsed": true,
    "tags": []
   },
   "source": [
    "###### Genus & Species output"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "bbb2e625-b254-4980-8420-9b77bbd9d44d",
   "metadata": {
    "tags": []
   },
   "source": [
    "\n",
    "ROGRAM START TIME: 12-14-2023 02:40:25\n",
    "BRACKEN SUMMARY (Kraken report: T1_12_2022.kreport2)\n",
    "    >>> Threshold: 0 \n",
    "    >>> Number of species in sample: 0 \n",
    "          >> Number of species with reads > threshold: 0 \n",
    "          >> Number of species with reads < threshold: 0 \n",
    "    >>> Total reads in sample: 20742228\n",
    "          >> Total reads kept at species level (reads > threshold): 0\n",
    "          >> Total reads discarded (species reads < threshold): 0\n",
    "          >> Reads distributed: 0\n",
    "          >> Reads not distributed (eg. no species above threshold): 48519\n",
    "          >> Unclassified reads: 20693709\n",
    "BRACKEN OUTPUT PRODUCED: //project/pi_sarah_gignouxwolfsohn_uml_edu/brooke/working/mcav_assembly_redo/taxonomy/kraken/bracken/T1_12_2022_species.bracken\n",
    "PROGRAM END TIME: 12-14-2023 02:40:26\n",
    "  Bracken complete.\n",
    " >> Checking for Valid Options...\n",
    " >> Running Bracken \n",
    "      >> python src/est_abundance.py -i T1_13_2022.kreport2 -o //project/pi_sarah_gignouxwolfsohn_uml_edu/brooke/working/mcav_assembly_redo/taxonomy/kraken/bracken/T1_13_2022_species.bracken -k //project/pi_sarah_gignouxwolfsohn_uml_edu/Reference_genomes/ref_databases/16S_SILVA132_k2db/database250mers.kmer_distrib -l S -t 0\n",
    "PROGRAM START TIME: 12-14-2023 02:40:26\n",
    "BRACKEN SUMMARY (Kraken report: T1_13_2022.kreport2)\n",
    "    >>> Threshold: 0 \n",
    "    >>> Number of species in sample: 0 \n",
    "          >> Number of species with reads > threshold: 0 \n",
    "          >> Number of species with reads < threshold: 0 \n",
    "    >>> Total reads in sample: 20861960\n",
    "          >> Total reads kept at species level (reads > threshold): 0\n",
    "          >> Total reads discarded (species reads < threshold): 0\n",
    "          >> Reads distributed: 0\n",
    "          >> Reads not distributed (eg. no species above threshold): 50615\n",
    "          >> Unclassified reads: 20811345\n",
    "BRACKEN OUTPUT PRODUCED: //project/pi_sarah_gignouxwolfsohn_uml_edu/brooke/working/mcav_assembly_redo/taxonomy/kraken/bracken/T1_13_2022_species.bracken\n",
    "PROGRAM END TIME: 12-14-2023 02:40:26\n",
    "  Bracken complete.\n",
    " >> Checking for Valid Options...\n",
    " >> Running Bracken \n",
    "      >> python src/est_abundance.py -i T1_16_2019.kreport2 -o //project/pi_sarah_gignouxwolfsohn_uml_edu/brooke/working/mcav_assembly_redo/taxonomy/kraken/bracken/T1_16_2019_species.bracken -k //project/pi_sarah_gignouxwolfsohn_uml_edu/Reference_genomes/ref_databases/16S_SILVA132_k2db/database250mers.kmer_distrib -l S -t 0\n",
    "PROGRAM START TIME: 12-14-2023 02:40:26\n",
    "BRACKEN SUMMARY (Kraken report: T1_16_2019.kreport2)\n",
    "    >>> Threshold: 0 \n",
    "    >>> Number of species in sample: 0 \n",
    "          >> Number of species with reads > threshold: 0 \n",
    "          >> Number of species with reads < threshold: 0 \n",
    "    >>> Total reads in sample: 25807031\n",
    "          >> Total reads kept at species level (reads > threshold): 0\n",
    "          >> Total reads discarded (species reads < threshold): 0\n",
    "          >> Reads distributed: 0\n",
    "          >> Reads not distributed (eg. no species above threshold): 71664\n",
    "          >> Unclassified reads: 25735367\n",
    "BRACKEN OUTPUT PRODUCED: //project/pi_sarah_gignouxwolfsohn_uml_edu/brooke/working/mcav_assembly_redo/taxonomy/kraken/bracken/T1_16_2019_species.bracken\n",
    "PROGRAM END TIME: 12-14-2023 02:40:26\n",
    "  Bracken complete.\n",
    " >> Checking for Valid Options...\n",
    " >> Running Bracken \n",
    "      >> python src/est_abundance.py -i T1_20_2019.kreport2 -o //project/pi_sarah_gignouxwolfsohn_uml_edu/brooke/working/mcav_assembly_redo/taxonomy/kraken/bracken/T1_20_2019_species.bracken -k //project/pi_sarah_gignouxwolfsohn_uml_edu/Reference_genomes/ref_databases/16S_SILVA132_k2db/database250mers.kmer_distrib -l S -t 0\n",
    "PROGRAM START TIME: 12-14-2023 02:40:27\n",
    "BRACKEN SUMMARY (Kraken report: T1_20_2019.kreport2)\n",
    "    >>> Threshold: 0 \n",
    "    >>> Number of species in sample: 0 \n",
    "          >> Number of species with reads > threshold: 0 \n",
    "          >> Number of species with reads < threshold: 0 \n",
    "    >>> Total reads in sample: 8517445\n",
    "          >> Total reads kept at species level (reads > threshold): 0\n",
    "          >> Total reads discarded (species reads < threshold): 0\n",
    "          >> Reads distributed: 0\n",
    "          >> Reads not distributed (eg. no species above threshold): 44133\n",
    "          >> Unclassified reads: 8473312\n",
    "BRACKEN OUTPUT PRODUCED: //project/pi_sarah_gignouxwolfsohn_uml_edu/brooke/working/mcav_assembly_redo/taxonomy/kraken/bracken/T1_20_2019_species.bracken\n",
    "PROGRAM END TIME: 12-14-2023 02:40:27\n",
    "  Bracken complete.\n",
    " >> Checking for Valid Options...\n",
    " >> Running Bracken \n",
    "      >> python src/est_abundance.py -i T1_24_2019.kreport2 -o //project/pi_sarah_gignouxwolfsohn_uml_edu/brooke/working/mcav_assembly_redo/taxonomy/kraken/bracken/T1_24_2019_species.bracken -k //project/pi_sarah_gignouxwolfsohn_uml_edu/Reference_genomes/ref_databases/16S_SILVA132_k2db/database250mers.kmer_distrib -l S -t 0\n",
    "PROGRAM START TIME: 12-14-2023 02:40:27\n",
    "BRACKEN SUMMARY (Kraken report: T1_24_2019.kreport2)\n",
    "    >>> Threshold: 0 \n",
    "    >>> Number of species in sample: 0 \n",
    "          >> Number of species with reads > threshold: 0 \n",
    "          >> Number of species with reads < threshold: 0 \n",
    "    >>> Total reads in sample: 17135017\n",
    "          >> Total reads kept at species level (reads > threshold): 0\n",
    "          >> Total reads discarded (species reads < threshold): 0\n",
    "          >> Reads distributed: 0\n",
    "          >> Reads not distributed (eg. no species above threshold): 59096\n",
    "          >> Unclassified reads: 17075921\n",
    "BRACKEN OUTPUT PRODUCED: //project/pi_sarah_gignouxwolfsohn_uml_edu/brooke/working/mcav_assembly_redo/taxonomy/kraken/bracken/T1_24_2019_species.bracken\n",
    "PROGRAM END TIME: 12-14-2023 02:40:27\n",
    "  Bracken complete.\n",
    " >> Checking for Valid Options...\n",
    " >> Running Bracken \n",
    "      >> python src/est_abundance.py -i T1_40_2022.kreport2 -o //project/pi_sarah_gignouxwolfsohn_uml_edu/brooke/working/mcav_assembly_redo/taxonomy/kraken/bracken/T1_40_2022_species.bracken -k //project/pi_sarah_gignouxwolfsohn_uml_edu/Reference_genomes/ref_databases/16S_SILVA132_k2db/database250mers.kmer_distrib -l S -t 0\n",
    "PROGRAM START TIME: 12-14-2023 02:40:27\n",
    "BRACKEN SUMMARY (Kraken report: T1_40_2022.kreport2)\n",
    "    >>> Threshold: 0 \n",
    "    >>> Number of species in sample: 0 \n",
    "          >> Number of species with reads > threshold: 0 \n",
    "          >> Number of species with reads < threshold: 0 \n",
    "    >>> Total reads in sample: 15419779\n",
    "          >> Total reads kept at species level (reads > threshold): 0\n",
    "          >> Total reads discarded (species reads < threshold): 0\n",
    "          >> Reads distributed: 0\n",
    "          >> Reads not distributed (eg. no species above threshold): 110723\n",
    "          >> Unclassified reads: 15309056\n",
    "BRACKEN OUTPUT PRODUCED: //project/pi_sarah_gignouxwolfsohn_uml_edu/brooke/working/mcav_assembly_redo/taxonomy/kraken/bracken/T1_40_2022_species.bracken\n",
    "PROGRAM END TIME: 12-14-2023 02:40:28\n",
    "  Bracken complete.\n",
    " >> Checking for Valid Options...\n",
    " >> Running Bracken \n",
    "      >> python src/est_abundance.py -i T1_57_2022.kreport2 -o //project/pi_sarah_gignouxwolfsohn_uml_edu/brooke/working/mcav_assembly_redo/taxonomy/kraken/bracken/T1_57_2022_species.bracken -k //project/pi_sarah_gignouxwolfsohn_uml_edu/Reference_genomes/ref_databases/16S_SILVA132_k2db/database250mers.kmer_distrib -l S -t 0\n",
    "PROGRAM START TIME: 12-14-2023 02:40:28\n",
    "BRACKEN SUMMARY (Kraken report: T1_57_2022.kreport2)\n",
    "    >>> Threshold: 0 \n",
    "    >>> Number of species in sample: 0 \n",
    "          >> Number of species with reads > threshold: 0 \n",
    "          >> Number of species with reads < threshold: 0 \n",
    "    >>> Total reads in sample: 31964707\n",
    "          >> Total reads kept at species level (reads > threshold): 0\n",
    "          >> Total reads discarded (species reads < threshold): 0\n",
    "          >> Reads distributed: 0\n",
    "          >> Reads not distributed (eg. no species above threshold): 157041\n",
    "          >> Unclassified reads: 31807666\n",
    "BRACKEN OUTPUT PRODUCED: //project/pi_sarah_gignouxwolfsohn_uml_edu/brooke/working/mcav_assembly_redo/taxonomy/kraken/bracken/T1_57_2022_species.bracken\n",
    "PROGRAM END TIME: 12-14-2023 02:40:28\n",
    "  Bracken complete.\n",
    " >> Checking for Valid Options...\n",
    " >> Running Bracken \n",
    "      >> python src/est_abundance.py -i T1_70_2022.kreport2 -o //project/pi_sarah_gignouxwolfsohn_uml_edu/brooke/working/mcav_assembly_redo/taxonomy/kraken/bracken/T1_70_2022_species.bracken -k //project/pi_sarah_gignouxwolfsohn_uml_edu/Reference_genomes/ref_databases/16S_SILVA132_k2db/database250mers.kmer_distrib -l S -t 0\n",
    "PROGRAM START TIME: 12-14-2023 02:40:28\n",
    "BRACKEN SUMMARY (Kraken report: T1_70_2022.kreport2)\n",
    "    >>> Threshold: 0 \n",
    "    >>> Number of species in sample: 0 \n",
    "          >> Number of species with reads > threshold: 0 \n",
    "          >> Number of species with reads < threshold: 0 \n",
    "    >>> Total reads in sample: 15776674\n",
    "          >> Total reads kept at species level (reads > threshold): 0\n",
    "          >> Total reads discarded (species reads < threshold): 0\n",
    "          >> Reads distributed: 0\n",
    "          >> Reads not distributed (eg. no species above threshold): 121001\n",
    "          >> Unclassified reads: 15655673\n",
    "BRACKEN OUTPUT PRODUCED: //project/pi_sarah_gignouxwolfsohn_uml_edu/brooke/working/mcav_assembly_redo/taxonomy/kraken/bracken/T1_70_2022_species.bracken\n",
    "PROGRAM END TIME: 12-14-2023 02:40:29\n",
    "  Bracken complete.\n",
    " >> Checking for Valid Options...\n",
    " >> Running Bracken \n",
    "      >> python src/est_abundance.py -i T2_10_2022.kreport2 -o //project/pi_sarah_gignouxwolfsohn_uml_edu/brooke/working/mcav_assembly_redo/taxonomy/kraken/bracken/T2_10_2022_species.bracken -k //project/pi_sarah_gignouxwolfsohn_uml_edu/Reference_genomes/ref_databases/16S_SILVA132_k2db/database250mers.kmer_distrib -l S -t 0\n",
    "PROGRAM START TIME: 12-14-2023 02:40:29\n",
    "BRACKEN SUMMARY (Kraken report: T2_10_2022.kreport2)\n",
    "    >>> Threshold: 0 \n",
    "    >>> Number of species in sample: 0 \n",
    "          >> Number of species with reads > threshold: 0 \n",
    "          >> Number of species with reads < threshold: 0 \n",
    "    >>> Total reads in sample: 16876812\n",
    "          >> Total reads kept at species level (reads > threshold): 0\n",
    "          >> Total reads discarded (species reads < threshold): 0\n",
    "          >> Reads distributed: 0\n",
    "          >> Reads not distributed (eg. no species above threshold): 57221\n",
    "          >> Unclassified reads: 16819591\n",
    "BRACKEN OUTPUT PRODUCED: //project/pi_sarah_gignouxwolfsohn_uml_edu/brooke/working/mcav_assembly_redo/taxonomy/kraken/bracken/T2_10_2022_species.bracken\n",
    "PROGRAM END TIME: 12-14-2023 02:40:29\n",
    "  Bracken complete.\n",
    " >> Checking for Valid Options...\n",
    " >> Running Bracken \n",
    "      >> python src/est_abundance.py -i T2_16_2019.kreport2 -o //project/pi_sarah_gignouxwolfsohn_uml_edu/brooke/working/mcav_assembly_redo/taxonomy/kraken/bracken/T2_16_2019_species.bracken -k //project/pi_sarah_gignouxwolfsohn_uml_edu/Reference_genomes/ref_databases/16S_SILVA132_k2db/database250mers.kmer_distrib -l S -t 0\n",
    "PROGRAM START TIME: 12-14-2023 02:40:29\n",
    "BRACKEN SUMMARY (Kraken report: T2_16_2019.kreport2)\n",
    "    >>> Threshold: 0 \n",
    "    >>> Number of species in sample: 0 \n",
    "          >> Number of species with reads > threshold: 0 \n",
    "          >> Number of species with reads < threshold: 0 \n",
    "    >>> Total reads in sample: 34302369\n",
    "          >> Total reads kept at species level (reads > threshold): 0\n",
    "          >> Total reads discarded (species reads < threshold): 0\n",
    "          >> Reads distributed: 0\n",
    "          >> Reads not distributed (eg. no species above threshold): 116886\n",
    "          >> Unclassified reads: 34185483\n",
    "BRACKEN OUTPUT PRODUCED: //project/pi_sarah_gignouxwolfsohn_uml_edu/brooke/working/mcav_assembly_redo/taxonomy/kraken/bracken/T2_16_2019_species.bracken\n",
    "PROGRAM END TIME: 12-14-2023 02:40:30\n",
    "  Bracken complete.\n",
    " >> Checking for Valid Options...\n",
    " >> Running Bracken \n",
    "      >> python src/est_abundance.py -i T3_13_2022.kreport2 -o //project/pi_sarah_gignouxwolfsohn_uml_edu/brooke/working/mcav_assembly_redo/taxonomy/kraken/bracken/T3_13_2022_species.bracken -k //project/pi_sarah_gignouxwolfsohn_uml_edu/Reference_genomes/ref_databases/16S_SILVA132_k2db/database250mers.kmer_distrib -l S -t 0\n",
    "PROGRAM START TIME: 12-14-2023 02:40:30\n",
    "BRACKEN SUMMARY (Kraken report: T3_13_2022.kreport2)\n",
    "    >>> Threshold: 0 \n",
    "    >>> Number of species in sample: 0 \n",
    "          >> Number of species with reads > threshold: 0 \n",
    "          >> Number of species with reads < threshold: 0 \n",
    "    >>> Total reads in sample: 13449647\n",
    "          >> Total reads kept at species level (reads > threshold): 0\n",
    "          >> Total reads discarded (species reads < threshold): 0\n",
    "          >> Reads distributed: 0\n",
    "          >> Reads not distributed (eg. no species above threshold): 41610\n",
    "          >> Unclassified reads: 13408037\n",
    "BRACKEN OUTPUT PRODUCED: //project/pi_sarah_gignouxwolfsohn_uml_edu/brooke/working/mcav_assembly_redo/taxonomy/kraken/bracken/T3_13_2022_species.bracken\n",
    "PROGRAM END TIME: 12-14-2023 02:40:30\n",
    "  Bracken complete.\n",
    " >> Checking for Valid Options...\n",
    " >> Running Bracken \n",
    "      >> python src/est_abundance.py -i T3_14_2019.kreport2 -o //project/pi_sarah_gignouxwolfsohn_uml_edu/brooke/working/mcav_assembly_redo/taxonomy/kraken/bracken/T3_14_2019_species.bracken -k //project/pi_sarah_gignouxwolfsohn_uml_edu/Reference_genomes/ref_databases/16S_SILVA132_k2db/database250mers.kmer_distrib -l S -t 0\n",
    "PROGRAM START TIME: 12-14-2023 02:40:30\n",
    "BRACKEN SUMMARY (Kraken report: T3_14_2019.kreport2)\n",
    "    >>> Threshold: 0 \n",
    "    >>> Number of species in sample: 0 \n",
    "          >> Number of species with reads > threshold: 0 \n",
    "          >> Number of species with reads < threshold: 0 \n",
    "    >>> Total reads in sample: 21400877\n",
    "          >> Total reads kept at species level (reads > threshold): 0\n",
    "          >> Total reads discarded (species reads < threshold): 0\n",
    "          >> Reads distributed: 0\n",
    "          >> Reads not distributed (eg. no species above threshold): 91615\n",
    "          >> Unclassified reads: 21309262\n",
    "BRACKEN OUTPUT PRODUCED: //project/pi_sarah_gignouxwolfsohn_uml_edu/brooke/working/mcav_assembly_redo/taxonomy/kraken/bracken/T3_14_2019_species.bracken\n",
    "PROGRAM END TIME: 12-14-2023 02:40:31\n",
    "  Bracken complete.\n",
    " >> Checking for Valid Options...\n",
    " >> Running Bracken \n",
    "      >> python src/est_abundance.py -i T3_15_2019.kreport2 -o //project/pi_sarah_gignouxwolfsohn_uml_edu/brooke/working/mcav_assembly_redo/taxonomy/kraken/bracken/T3_15_2019_species.bracken -k //project/pi_sarah_gignouxwolfsohn_uml_edu/Reference_genomes/ref_databases/16S_SILVA132_k2db/database250mers.kmer_distrib -l S -t 0\n",
    "PROGRAM START TIME: 12-14-2023 02:40:31\n",
    "BRACKEN SUMMARY (Kraken report: T3_15_2019.kreport2)\n",
    "    >>> Threshold: 0 \n",
    "    >>> Number of species in sample: 0 \n",
    "          >> Number of species with reads > threshold: 0 \n",
    "          >> Number of species with reads < threshold: 0 \n",
    "    >>> Total reads in sample: 19854618\n",
    "          >> Total reads kept at species level (reads > threshold): 0\n",
    "          >> Total reads discarded (species reads < threshold): 0\n",
    "          >> Reads distributed: 0\n",
    "          >> Reads not distributed (eg. no species above threshold): 51464\n",
    "          >> Unclassified reads: 19803154\n",
    "BRACKEN OUTPUT PRODUCED: //project/pi_sarah_gignouxwolfsohn_uml_edu/brooke/working/mcav_assembly_redo/taxonomy/kraken/bracken/T3_15_2019_species.bracken\n",
    "PROGRAM END TIME: 12-14-2023 02:40:31\n",
    "  Bracken complete.\n",
    " >> Checking for Valid Options...\n",
    " >> Running Bracken \n",
    "      >> python src/est_abundance.py -i T3_19_2022.kreport2 -o //project/pi_sarah_gignouxwolfsohn_uml_edu/brooke/working/mcav_assembly_redo/taxonomy/kraken/bracken/T3_19_2022_species.bracken -k //project/pi_sarah_gignouxwolfsohn_uml_edu/Reference_genomes/ref_databases/16S_SILVA132_k2db/database250mers.kmer_distrib -l S -t 0\n",
    "PROGRAM START TIME: 12-14-2023 02:40:31\n",
    "BRACKEN SUMMARY (Kraken report: T3_19_2022.kreport2)\n",
    "    >>> Threshold: 0 \n",
    "    >>> Number of species in sample: 0 \n",
    "          >> Number of species with reads > threshold: 0 \n",
    "          >> Number of species with reads < threshold: 0 \n",
    "    >>> Total reads in sample: 9044371\n",
    "          >> Total reads kept at species level (reads > threshold): 0\n",
    "          >> Total reads discarded (species reads < threshold): 0\n",
    "          >> Reads distributed: 0\n",
    "          >> Reads not distributed (eg. no species above threshold): 93872\n",
    "          >> Unclassified reads: 8950499\n",
    "BRACKEN OUTPUT PRODUCED: //project/pi_sarah_gignouxwolfsohn_uml_edu/brooke/working/mcav_assembly_redo/taxonomy/kraken/bracken/T3_19_2022_species.bracken\n",
    "PROGRAM END TIME: 12-14-2023 02:40:31\n",
    "  Bracken complete.\n",
    " >> Checking for Valid Options...\n",
    " >> Running Bracken \n",
    "      >> python src/est_abundance.py -i T3_1_2019.kreport2 -o //project/pi_sarah_gignouxwolfsohn_uml_edu/brooke/working/mcav_assembly_redo/taxonomy/kraken/bracken/T3_1_2019_species.bracken -k //project/pi_sarah_gignouxwolfsohn_uml_edu/Reference_genomes/ref_databases/16S_SILVA132_k2db/database250mers.kmer_distrib -l S -t 0\n",
    "PROGRAM START TIME: 12-14-2023 02:40:31\n",
    "BRACKEN SUMMARY (Kraken report: T3_1_2019.kreport2)\n",
    "    >>> Threshold: 0 \n",
    "    >>> Number of species in sample: 0 \n",
    "          >> Number of species with reads > threshold: 0 \n",
    "          >> Number of species with reads < threshold: 0 \n",
    "    >>> Total reads in sample: 29977945\n",
    "          >> Total reads kept at species level (reads > threshold): 0\n",
    "          >> Total reads discarded (species reads < threshold): 0\n",
    "          >> Reads distributed: 0\n",
    "          >> Reads not distributed (eg. no species above threshold): 99522\n",
    "          >> Unclassified reads: 29878423\n",
    "BRACKEN OUTPUT PRODUCED: //project/pi_sarah_gignouxwolfsohn_uml_edu/brooke/working/mcav_assembly_redo/taxonomy/kraken/bracken/T3_1_2019_species.bracken\n",
    "PROGRAM END TIME: 12-14-2023 02:40:32\n",
    "  Bracken complete.\n",
    " >> Checking for Valid Options...\n",
    " >> Running Bracken \n",
    "      >> python src/est_abundance.py -i T3_40_2022.kreport2 -o //project/pi_sarah_gignouxwolfsohn_uml_edu/brooke/working/mcav_assembly_redo/taxonomy/kraken/bracken/T3_40_2022_species.bracken -k //project/pi_sarah_gignouxwolfsohn_uml_edu/Reference_genomes/ref_databases/16S_SILVA132_k2db/database250mers.kmer_distrib -l S -t 0\n",
    "PROGRAM START TIME: 12-14-2023 02:40:32\n",
    "BRACKEN SUMMARY (Kraken report: T3_40_2022.kreport2)\n",
    "    >>> Threshold: 0 \n",
    "    >>> Number of species in sample: 0 \n",
    "          >> Number of species with reads > threshold: 0 \n",
    "          >> Number of species with reads < threshold: 0 \n",
    "    >>> Total reads in sample: 26755227\n",
    "          >> Total reads kept at species level (reads > threshold): 0\n",
    "          >> Total reads discarded (species reads < threshold): 0\n",
    "          >> Reads distributed: 0\n",
    "          >> Reads not distributed (eg. no species above threshold): 77871\n",
    "          >> Unclassified reads: 26677356\n",
    "BRACKEN OUTPUT PRODUCED: //project/pi_sarah_gignouxwolfsohn_uml_edu/brooke/working/mcav_assembly_redo/taxonomy/kraken/bracken/T3_40_2022_species.bracken\n",
    "PROGRAM END TIME: 12-14-2023 02:40:32\n",
    "  Bracken complete.\n",
    " >> Checking for Valid Options...\n",
    " >> Running Bracken \n",
    "      >> python src/est_abundance.py -i T3_48_2022.kreport2 -o //project/pi_sarah_gignouxwolfsohn_uml_edu/brooke/working/mcav_assembly_redo/taxonomy/kraken/bracken/T3_48_2022_species.bracken -k //project/pi_sarah_gignouxwolfsohn_uml_edu/Reference_genomes/ref_databases/16S_SILVA132_k2db/database250mers.kmer_distrib -l S -t 0\n",
    "PROGRAM START TIME: 12-14-2023 02:40:32\n",
    "BRACKEN SUMMARY (Kraken report: T3_48_2022.kreport2)\n",
    "    >>> Threshold: 0 \n",
    "    >>> Number of species in sample: 0 \n",
    "          >> Number of species with reads > threshold: 0 \n",
    "          >> Number of species with reads < threshold: 0 \n",
    "    >>> Total reads in sample: 13255150\n",
    "          >> Total reads kept at species level (reads > threshold): 0\n",
    "          >> Total reads discarded (species reads < threshold): 0\n",
    "          >> Reads distributed: 0\n",
    "          >> Reads not distributed (eg. no species above threshold): 133945\n",
    "          >> Unclassified reads: 13121205\n",
    "BRACKEN OUTPUT PRODUCED: //project/pi_sarah_gignouxwolfsohn_uml_edu/brooke/working/mcav_assembly_redo/taxonomy/kraken/bracken/T3_48_2022_species.bracken\n",
    "PROGRAM END TIME: 12-14-2023 02:40:33\n",
    "  Bracken complete.\n",
    " >> Checking for Valid Options...\n",
    " >> Running Bracken \n",
    "      >> python src/est_abundance.py -i T3_49_2022.kreport2 -o //project/pi_sarah_gignouxwolfsohn_uml_edu/brooke/working/mcav_assembly_redo/taxonomy/kraken/bracken/T3_49_2022_species.bracken -k //project/pi_sarah_gignouxwolfsohn_uml_edu/Reference_genomes/ref_databases/16S_SILVA132_k2db/database250mers.kmer_distrib -l S -t 0\n",
    "PROGRAM START TIME: 12-14-2023 02:40:33\n",
    "BRACKEN SUMMARY (Kraken report: T3_49_2022.kreport2)\n",
    "    >>> Threshold: 0 \n",
    "    >>> Number of species in sample: 0 \n",
    "          >> Number of species with reads > threshold: 0 \n",
    "          >> Number of species with reads < threshold: 0 \n",
    "    >>> Total reads in sample: 15749618\n",
    "          >> Total reads kept at species level (reads > threshold): 0\n",
    "          >> Total reads discarded (species reads < threshold): 0\n",
    "          >> Reads distributed: 0\n",
    "          >> Reads not distributed (eg. no species above threshold): 149084\n",
    "          >> Unclassified reads: 15600534\n",
    "BRACKEN OUTPUT PRODUCED: //project/pi_sarah_gignouxwolfsohn_uml_edu/brooke/working/mcav_assembly_redo/taxonomy/kraken/bracken/T3_49_2022_species.bracken\n",
    "PROGRAM END TIME: 12-14-2023 02:40:33\n",
    "  Bracken complete.\n",
    " >> Checking for Valid Options...\n",
    " >> Running Bracken \n",
    "      >> python src/est_abundance.py -i T3_51_2022.kreport2 -o //project/pi_sarah_gignouxwolfsohn_uml_edu/brooke/working/mcav_assembly_redo/taxonomy/kraken/bracken/T3_51_2022_species.bracken -k //project/pi_sarah_gignouxwolfsohn_uml_edu/Reference_genomes/ref_databases/16S_SILVA132_k2db/database250mers.kmer_distrib -l S -t 0\n",
    "PROGRAM START TIME: 12-14-2023 02:40:33\n",
    "BRACKEN SUMMARY (Kraken report: T3_51_2022.kreport2)\n",
    "    >>> Threshold: 0 \n",
    "    >>> Number of species in sample: 0 \n",
    "          >> Number of species with reads > threshold: 0 \n",
    "          >> Number of species with reads < threshold: 0 \n",
    "    >>> Total reads in sample: 13877944\n",
    "          >> Total reads kept at species level (reads > threshold): 0\n",
    "          >> Total reads discarded (species reads < threshold): 0\n",
    "          >> Reads distributed: 0\n",
    "          >> Reads not distributed (eg. no species above threshold): 50639\n",
    "          >> Unclassified reads: 13827305\n",
    "BRACKEN OUTPUT PRODUCED: //project/pi_sarah_gignouxwolfsohn_uml_edu/brooke/working/mcav_assembly_redo/taxonomy/kraken/bracken/T3_51_2022_species.bracken\n",
    "PROGRAM END TIME: 12-14-2023 02:40:34\n",
    "  Bracken complete.\n",
    " >> Checking for Valid Options...\n",
    " >> Running Bracken \n",
    "      >> python src/est_abundance.py -i T3_8_2019.kreport2 -o //project/pi_sarah_gignouxwolfsohn_uml_edu/brooke/working/mcav_assembly_redo/taxonomy/kraken/bracken/T3_8_2019_species.bracken -k //project/pi_sarah_gignouxwolfsohn_uml_edu/Reference_genomes/ref_databases/16S_SILVA132_k2db/database250mers.kmer_distrib -l S -t 0\n",
    "PROGRAM START TIME: 12-14-2023 02:40:34\n",
    "BRACKEN SUMMARY (Kraken report: T3_8_2019.kreport2)\n",
    "    >>> Threshold: 0 \n",
    "    >>> Number of species in sample: 0 \n",
    "          >> Number of species with reads > threshold: 0 \n",
    "          >> Number of species with reads < threshold: 0 \n",
    "    >>> Total reads in sample: 16453406\n",
    "          >> Total reads kept at species level (reads > threshold): 0\n",
    "          >> Total reads discarded (species reads < threshold): 0\n",
    "          >> Reads distributed: 0\n",
    "          >> Reads not distributed (eg. no species above threshold): 94662\n",
    "          >> Unclassified reads: 16358744\n",
    "BRACKEN OUTPUT PRODUCED: //project/pi_sarah_gignouxwolfsohn_uml_edu/brooke/working/mcav_assembly_redo/taxonomy/kraken/bracken/T3_8_2019_species.bracken\n",
    "PROGRAM END TIME: 12-14-2023 02:40:34\n",
    "  Bracken complete.\n",
    " >> Checking for Valid Options...\n",
    " >> Running Bracken \n",
    "      >> python src/est_abundance.py -i T3_9_2019.kreport2 -o //project/pi_sarah_gignouxwolfsohn_uml_edu/brooke/working/mcav_assembly_redo/taxonomy/kraken/bracken/T3_9_2019_species.bracken -k //project/pi_sarah_gignouxwolfsohn_uml_edu/Reference_genomes/ref_databases/16S_SILVA132_k2db/database250mers.kmer_distrib -l S -t 0\n",
    "PROGRAM START TIME: 12-14-2023 02:40:34\n",
    "BRACKEN SUMMARY (Kraken report: T3_9_2019.kreport2)\n",
    "    >>> Threshold: 0 \n",
    "    >>> Number of species in sample: 0 \n",
    "          >> Number of species with reads > threshold: 0 \n",
    "          >> Number of species with reads < threshold: 0 \n",
    "    >>> Total reads in sample: 14032207\n",
    "          >> Total reads kept at species level (reads > threshold): 0\n",
    "          >> Total reads discarded (species reads < threshold): 0\n",
    "          >> Reads distributed: 0\n",
    "          >> Reads not distributed (eg. no species above threshold): 69979\n",
    "          >> Unclassified reads: 13962228\n",
    "BRACKEN OUTPUT PRODUCED: //project/pi_sarah_gignouxwolfsohn_uml_edu/brooke/working/mcav_assembly_redo/taxonomy/kraken/bracken/T3_9_2019_species.bracken\n",
    "PROGRAM END TIME: 12-14-2023 02:40:34\n",
    "  Bracken complete."
   ]
  },
  {
   "cell_type": "markdown",
   "id": "71c5adf3-7bfe-4980-bf82-112b77830a76",
   "metadata": {
    "jp-MarkdownHeadingCollapsed": true,
    "tags": []
   },
   "source": [
    "Species output \n",
    "ROGRAM START TIME: 12-14-2023 02:40:25\n",
    "BRACKEN SUMMARY (Kraken report: T1_12_2022.kreport2)\n",
    "    >>> Threshold: 0 \n",
    "    >>> Number of species in sample: 0 \n",
    "          >> Number of species with reads > threshold: 0 \n",
    "          >> Number of species with reads < threshold: 0 \n",
    "    >>> Total reads in sample: 20742228\n",
    "          >> Total reads kept at species level (reads > threshold): 0\n",
    "          >> Total reads discarded (species reads < threshold): 0\n",
    "          >> Reads distributed: 0\n",
    "          >> Reads not distributed (eg. no species above threshold): 48519\n",
    "          >> Unclassified reads: 20693709\n",
    "BRACKEN OUTPUT PRODUCED: //project/pi_sarah_gignouxwolfsohn_uml_edu/brooke/working/mcav_assembly_redo/taxonomy/kraken/bracken/T1_12_2022_species.bracken\n",
    "PROGRAM END TIME: 12-14-2023 02:40:26\n",
    "  Bracken complete.\n",
    " >> Checking for Valid Options...\n",
    " >> Running Bracken \n",
    "      >> python src/est_abundance.py -i T1_13_2022.kreport2 -o //project/pi_sarah_gignouxwolfsohn_uml_edu/brooke/working/mcav_assembly_redo/taxonomy/kraken/bracken/T1_13_2022_species.bracken -k //project/pi_sarah_gignouxwolfsohn_uml_edu/Reference_genomes/ref_databases/16S_SILVA132_k2db/database250mers.kmer_distrib -l S -t 0\n",
    "PROGRAM START TIME: 12-14-2023 02:40:26\n",
    "BRACKEN SUMMARY (Kraken report: T1_13_2022.kreport2)\n",
    "    >>> Threshold: 0 \n",
    "    >>> Number of species in sample: 0 \n",
    "          >> Number of species with reads > threshold: 0 \n",
    "          >> Number of species with reads < threshold: 0 \n",
    "    >>> Total reads in sample: 20861960\n",
    "          >> Total reads kept at species level (reads > threshold): 0\n",
    "          >> Total reads discarded (species reads < threshold): 0\n",
    "          >> Reads distributed: 0\n",
    "          >> Reads not distributed (eg. no species above threshold): 50615\n",
    "          >> Unclassified reads: 20811345\n",
    "BRACKEN OUTPUT PRODUCED: //project/pi_sarah_gignouxwolfsohn_uml_edu/brooke/working/mcav_assembly_redo/taxonomy/kraken/bracken/T1_13_2022_species.bracken\n",
    "PROGRAM END TIME: 12-14-2023 02:40:26\n",
    "  Bracken complete.\n",
    " >> Checking for Valid Options...\n",
    " >> Running Bracken \n",
    "      >> python src/est_abundance.py -i T1_16_2019.kreport2 -o //project/pi_sarah_gignouxwolfsohn_uml_edu/brooke/working/mcav_assembly_redo/taxonomy/kraken/bracken/T1_16_2019_species.bracken -k //project/pi_sarah_gignouxwolfsohn_uml_edu/Reference_genomes/ref_databases/16S_SILVA132_k2db/database250mers.kmer_distrib -l S -t 0\n",
    "PROGRAM START TIME: 12-14-2023 02:40:26\n",
    "BRACKEN SUMMARY (Kraken report: T1_16_2019.kreport2)\n",
    "    >>> Threshold: 0 \n",
    "    >>> Number of species in sample: 0 \n",
    "          >> Number of species with reads > threshold: 0 \n",
    "          >> Number of species with reads < threshold: 0 \n",
    "    >>> Total reads in sample: 25807031\n",
    "          >> Total reads kept at species level (reads > threshold): 0\n",
    "          >> Total reads discarded (species reads < threshold): 0\n",
    "          >> Reads distributed: 0\n",
    "          >> Reads not distributed (eg. no species above threshold): 71664\n",
    "          >> Unclassified reads: 25735367\n",
    "BRACKEN OUTPUT PRODUCED: //project/pi_sarah_gignouxwolfsohn_uml_edu/brooke/working/mcav_assembly_redo/taxonomy/kraken/bracken/T1_16_2019_species.bracken\n",
    "PROGRAM END TIME: 12-14-2023 02:40:26\n",
    "  Bracken complete.\n",
    " >> Checking for Valid Options...\n",
    " >> Running Bracken \n",
    "      >> python src/est_abundance.py -i T1_20_2019.kreport2 -o //project/pi_sarah_gignouxwolfsohn_uml_edu/brooke/working/mcav_assembly_redo/taxonomy/kraken/bracken/T1_20_2019_species.bracken -k //project/pi_sarah_gignouxwolfsohn_uml_edu/Reference_genomes/ref_databases/16S_SILVA132_k2db/database250mers.kmer_distrib -l S -t 0\n",
    "PROGRAM START TIME: 12-14-2023 02:40:27\n",
    "BRACKEN SUMMARY (Kraken report: T1_20_2019.kreport2)\n",
    "    >>> Threshold: 0 \n",
    "    >>> Number of species in sample: 0 \n",
    "          >> Number of species with reads > threshold: 0 \n",
    "          >> Number of species with reads < threshold: 0 \n",
    "    >>> Total reads in sample: 8517445\n",
    "          >> Total reads kept at species level (reads > threshold): 0\n",
    "          >> Total reads discarded (species reads < threshold): 0\n",
    "          >> Reads distributed: 0\n",
    "          >> Reads not distributed (eg. no species above threshold): 44133\n",
    "          >> Unclassified reads: 8473312\n",
    "BRACKEN OUTPUT PRODUCED: //project/pi_sarah_gignouxwolfsohn_uml_edu/brooke/working/mcav_assembly_redo/taxonomy/kraken/bracken/T1_20_2019_species.bracken\n",
    "PROGRAM END TIME: 12-14-2023 02:40:27\n",
    "  Bracken complete.\n",
    " >> Checking for Valid Options...\n",
    " >> Running Bracken \n",
    "      >> python src/est_abundance.py -i T1_24_2019.kreport2 -o //project/pi_sarah_gignouxwolfsohn_uml_edu/brooke/working/mcav_assembly_redo/taxonomy/kraken/bracken/T1_24_2019_species.bracken -k //project/pi_sarah_gignouxwolfsohn_uml_edu/Reference_genomes/ref_databases/16S_SILVA132_k2db/database250mers.kmer_distrib -l S -t 0\n",
    "PROGRAM START TIME: 12-14-2023 02:40:27\n",
    "BRACKEN SUMMARY (Kraken report: T1_24_2019.kreport2)\n",
    "    >>> Threshold: 0 \n",
    "    >>> Number of species in sample: 0 \n",
    "          >> Number of species with reads > threshold: 0 \n",
    "          >> Number of species with reads < threshold: 0 \n",
    "    >>> Total reads in sample: 17135017\n",
    "          >> Total reads kept at species level (reads > threshold): 0\n",
    "          >> Total reads discarded (species reads < threshold): 0\n",
    "          >> Reads distributed: 0\n",
    "          >> Reads not distributed (eg. no species above threshold): 59096\n",
    "          >> Unclassified reads: 17075921\n",
    "BRACKEN OUTPUT PRODUCED: //project/pi_sarah_gignouxwolfsohn_uml_edu/brooke/working/mcav_assembly_redo/taxonomy/kraken/bracken/T1_24_2019_species.bracken\n",
    "PROGRAM END TIME: 12-14-2023 02:40:27\n",
    "  Bracken complete.\n",
    " >> Checking for Valid Options...\n",
    " >> Running Bracken \n",
    "      >> python src/est_abundance.py -i T1_40_2022.kreport2 -o //project/pi_sarah_gignouxwolfsohn_uml_edu/brooke/working/mcav_assembly_redo/taxonomy/kraken/bracken/T1_40_2022_species.bracken -k //project/pi_sarah_gignouxwolfsohn_uml_edu/Reference_genomes/ref_databases/16S_SILVA132_k2db/database250mers.kmer_distrib -l S -t 0\n",
    "PROGRAM START TIME: 12-14-2023 02:40:27\n",
    "BRACKEN SUMMARY (Kraken report: T1_40_2022.kreport2)\n",
    "    >>> Threshold: 0 \n",
    "    >>> Number of species in sample: 0 \n",
    "          >> Number of species with reads > threshold: 0 \n",
    "          >> Number of species with reads < threshold: 0 \n",
    "    >>> Total reads in sample: 15419779\n",
    "          >> Total reads kept at species level (reads > threshold): 0\n",
    "          >> Total reads discarded (species reads < threshold): 0\n",
    "          >> Reads distributed: 0\n",
    "          >> Reads not distributed (eg. no species above threshold): 110723\n",
    "          >> Unclassified reads: 15309056\n",
    "BRACKEN OUTPUT PRODUCED: //project/pi_sarah_gignouxwolfsohn_uml_edu/brooke/working/mcav_assembly_redo/taxonomy/kraken/bracken/T1_40_2022_species.bracken\n",
    "PROGRAM END TIME: 12-14-2023 02:40:28\n",
    "  Bracken complete.\n",
    " >> Checking for Valid Options...\n",
    " >> Running Bracken \n",
    "      >> python src/est_abundance.py -i T1_57_2022.kreport2 -o //project/pi_sarah_gignouxwolfsohn_uml_edu/brooke/working/mcav_assembly_redo/taxonomy/kraken/bracken/T1_57_2022_species.bracken -k //project/pi_sarah_gignouxwolfsohn_uml_edu/Reference_genomes/ref_databases/16S_SILVA132_k2db/database250mers.kmer_distrib -l S -t 0\n",
    "PROGRAM START TIME: 12-14-2023 02:40:28\n",
    "BRACKEN SUMMARY (Kraken report: T1_57_2022.kreport2)\n",
    "    >>> Threshold: 0 \n",
    "    >>> Number of species in sample: 0 \n",
    "          >> Number of species with reads > threshold: 0 \n",
    "          >> Number of species with reads < threshold: 0 \n",
    "    >>> Total reads in sample: 31964707\n",
    "          >> Total reads kept at species level (reads > threshold): 0\n",
    "          >> Total reads discarded (species reads < threshold): 0\n",
    "          >> Reads distributed: 0\n",
    "          >> Reads not distributed (eg. no species above threshold): 157041\n",
    "          >> Unclassified reads: 31807666\n",
    "BRACKEN OUTPUT PRODUCED: //project/pi_sarah_gignouxwolfsohn_uml_edu/brooke/working/mcav_assembly_redo/taxonomy/kraken/bracken/T1_57_2022_species.bracken\n",
    "PROGRAM END TIME: 12-14-2023 02:40:28\n",
    "  Bracken complete.\n",
    " >> Checking for Valid Options...\n",
    " >> Running Bracken \n",
    "      >> python src/est_abundance.py -i T1_70_2022.kreport2 -o //project/pi_sarah_gignouxwolfsohn_uml_edu/brooke/working/mcav_assembly_redo/taxonomy/kraken/bracken/T1_70_2022_species.bracken -k //project/pi_sarah_gignouxwolfsohn_uml_edu/Reference_genomes/ref_databases/16S_SILVA132_k2db/database250mers.kmer_distrib -l S -t 0\n",
    "PROGRAM START TIME: 12-14-2023 02:40:28\n",
    "BRACKEN SUMMARY (Kraken report: T1_70_2022.kreport2)\n",
    "    >>> Threshold: 0 \n",
    "    >>> Number of species in sample: 0 \n",
    "          >> Number of species with reads > threshold: 0 \n",
    "          >> Number of species with reads < threshold: 0 \n",
    "    >>> Total reads in sample: 15776674\n",
    "          >> Total reads kept at species level (reads > threshold): 0\n",
    "          >> Total reads discarded (species reads < threshold): 0\n",
    "          >> Reads distributed: 0\n",
    "          >> Reads not distributed (eg. no species above threshold): 121001\n",
    "          >> Unclassified reads: 15655673\n",
    "BRACKEN OUTPUT PRODUCED: //project/pi_sarah_gignouxwolfsohn_uml_edu/brooke/working/mcav_assembly_redo/taxonomy/kraken/bracken/T1_70_2022_species.bracken\n",
    "PROGRAM END TIME: 12-14-2023 02:40:29\n",
    "  Bracken complete.\n",
    " >> Checking for Valid Options...\n",
    " >> Running Bracken \n",
    "      >> python src/est_abundance.py -i T2_10_2022.kreport2 -o //project/pi_sarah_gignouxwolfsohn_uml_edu/brooke/working/mcav_assembly_redo/taxonomy/kraken/bracken/T2_10_2022_species.bracken -k //project/pi_sarah_gignouxwolfsohn_uml_edu/Reference_genomes/ref_databases/16S_SILVA132_k2db/database250mers.kmer_distrib -l S -t 0\n",
    "PROGRAM START TIME: 12-14-2023 02:40:29\n",
    "BRACKEN SUMMARY (Kraken report: T2_10_2022.kreport2)\n",
    "    >>> Threshold: 0 \n",
    "    >>> Number of species in sample: 0 \n",
    "          >> Number of species with reads > threshold: 0 \n",
    "          >> Number of species with reads < threshold: 0 \n",
    "    >>> Total reads in sample: 16876812\n",
    "          >> Total reads kept at species level (reads > threshold): 0\n",
    "          >> Total reads discarded (species reads < threshold): 0\n",
    "          >> Reads distributed: 0\n",
    "          >> Reads not distributed (eg. no species above threshold): 57221\n",
    "          >> Unclassified reads: 16819591\n",
    "BRACKEN OUTPUT PRODUCED: //project/pi_sarah_gignouxwolfsohn_uml_edu/brooke/working/mcav_assembly_redo/taxonomy/kraken/bracken/T2_10_2022_species.bracken\n",
    "PROGRAM END TIME: 12-14-2023 02:40:29\n",
    "  Bracken complete.\n",
    " >> Checking for Valid Options...\n",
    " >> Running Bracken \n",
    "      >> python src/est_abundance.py -i T2_16_2019.kreport2 -o //project/pi_sarah_gignouxwolfsohn_uml_edu/brooke/working/mcav_assembly_redo/taxonomy/kraken/bracken/T2_16_2019_species.bracken -k //project/pi_sarah_gignouxwolfsohn_uml_edu/Reference_genomes/ref_databases/16S_SILVA132_k2db/database250mers.kmer_distrib -l S -t 0\n",
    "PROGRAM START TIME: 12-14-2023 02:40:29\n",
    "BRACKEN SUMMARY (Kraken report: T2_16_2019.kreport2)\n",
    "    >>> Threshold: 0 \n",
    "    >>> Number of species in sample: 0 \n",
    "          >> Number of species with reads > threshold: 0 \n",
    "          >> Number of species with reads < threshold: 0 \n",
    "    >>> Total reads in sample: 34302369\n",
    "          >> Total reads kept at species level (reads > threshold): 0\n",
    "          >> Total reads discarded (species reads < threshold): 0\n",
    "          >> Reads distributed: 0\n",
    "          >> Reads not distributed (eg. no species above threshold): 116886\n",
    "          >> Unclassified reads: 34185483\n",
    "BRACKEN OUTPUT PRODUCED: //project/pi_sarah_gignouxwolfsohn_uml_edu/brooke/working/mcav_assembly_redo/taxonomy/kraken/bracken/T2_16_2019_species.bracken\n",
    "PROGRAM END TIME: 12-14-2023 02:40:30\n",
    "  Bracken complete.\n",
    " >> Checking for Valid Options...\n",
    " >> Running Bracken \n",
    "      >> python src/est_abundance.py -i T3_13_2022.kreport2 -o //project/pi_sarah_gignouxwolfsohn_uml_edu/brooke/working/mcav_assembly_redo/taxonomy/kraken/bracken/T3_13_2022_species.bracken -k //project/pi_sarah_gignouxwolfsohn_uml_edu/Reference_genomes/ref_databases/16S_SILVA132_k2db/database250mers.kmer_distrib -l S -t 0\n",
    "PROGRAM START TIME: 12-14-2023 02:40:30\n",
    "BRACKEN SUMMARY (Kraken report: T3_13_2022.kreport2)\n",
    "    >>> Threshold: 0 \n",
    "    >>> Number of species in sample: 0 \n",
    "          >> Number of species with reads > threshold: 0 \n",
    "          >> Number of species with reads < threshold: 0 \n",
    "    >>> Total reads in sample: 13449647\n",
    "          >> Total reads kept at species level (reads > threshold): 0\n",
    "          >> Total reads discarded (species reads < threshold): 0\n",
    "          >> Reads distributed: 0\n",
    "          >> Reads not distributed (eg. no species above threshold): 41610\n",
    "          >> Unclassified reads: 13408037\n",
    "BRACKEN OUTPUT PRODUCED: //project/pi_sarah_gignouxwolfsohn_uml_edu/brooke/working/mcav_assembly_redo/taxonomy/kraken/bracken/T3_13_2022_species.bracken\n",
    "PROGRAM END TIME: 12-14-2023 02:40:30\n",
    "  Bracken complete.\n",
    " >> Checking for Valid Options...\n",
    " >> Running Bracken \n",
    "      >> python src/est_abundance.py -i T3_14_2019.kreport2 -o //project/pi_sarah_gignouxwolfsohn_uml_edu/brooke/working/mcav_assembly_redo/taxonomy/kraken/bracken/T3_14_2019_species.bracken -k //project/pi_sarah_gignouxwolfsohn_uml_edu/Reference_genomes/ref_databases/16S_SILVA132_k2db/database250mers.kmer_distrib -l S -t 0\n",
    "PROGRAM START TIME: 12-14-2023 02:40:30\n",
    "BRACKEN SUMMARY (Kraken report: T3_14_2019.kreport2)\n",
    "    >>> Threshold: 0 \n",
    "    >>> Number of species in sample: 0 \n",
    "          >> Number of species with reads > threshold: 0 \n",
    "          >> Number of species with reads < threshold: 0 \n",
    "    >>> Total reads in sample: 21400877\n",
    "          >> Total reads kept at species level (reads > threshold): 0\n",
    "          >> Total reads discarded (species reads < threshold): 0\n",
    "          >> Reads distributed: 0\n",
    "          >> Reads not distributed (eg. no species above threshold): 91615\n",
    "          >> Unclassified reads: 21309262\n",
    "BRACKEN OUTPUT PRODUCED: //project/pi_sarah_gignouxwolfsohn_uml_edu/brooke/working/mcav_assembly_redo/taxonomy/kraken/bracken/T3_14_2019_species.bracken\n",
    "PROGRAM END TIME: 12-14-2023 02:40:31\n",
    "  Bracken complete.\n",
    " >> Checking for Valid Options...\n",
    " >> Running Bracken \n",
    "      >> python src/est_abundance.py -i T3_15_2019.kreport2 -o //project/pi_sarah_gignouxwolfsohn_uml_edu/brooke/working/mcav_assembly_redo/taxonomy/kraken/bracken/T3_15_2019_species.bracken -k //project/pi_sarah_gignouxwolfsohn_uml_edu/Reference_genomes/ref_databases/16S_SILVA132_k2db/database250mers.kmer_distrib -l S -t 0\n",
    "PROGRAM START TIME: 12-14-2023 02:40:31\n",
    "BRACKEN SUMMARY (Kraken report: T3_15_2019.kreport2)\n",
    "    >>> Threshold: 0 \n",
    "    >>> Number of species in sample: 0 \n",
    "          >> Number of species with reads > threshold: 0 \n",
    "          >> Number of species with reads < threshold: 0 \n",
    "    >>> Total reads in sample: 19854618\n",
    "          >> Total reads kept at species level (reads > threshold): 0\n",
    "          >> Total reads discarded (species reads < threshold): 0\n",
    "          >> Reads distributed: 0\n",
    "          >> Reads not distributed (eg. no species above threshold): 51464\n",
    "          >> Unclassified reads: 19803154\n",
    "BRACKEN OUTPUT PRODUCED: //project/pi_sarah_gignouxwolfsohn_uml_edu/brooke/working/mcav_assembly_redo/taxonomy/kraken/bracken/T3_15_2019_species.bracken\n",
    "PROGRAM END TIME: 12-14-2023 02:40:31\n",
    "  Bracken complete.\n",
    " >> Checking for Valid Options...\n",
    " >> Running Bracken \n",
    "      >> python src/est_abundance.py -i T3_19_2022.kreport2 -o //project/pi_sarah_gignouxwolfsohn_uml_edu/brooke/working/mcav_assembly_redo/taxonomy/kraken/bracken/T3_19_2022_species.bracken -k //project/pi_sarah_gignouxwolfsohn_uml_edu/Reference_genomes/ref_databases/16S_SILVA132_k2db/database250mers.kmer_distrib -l S -t 0\n",
    "PROGRAM START TIME: 12-14-2023 02:40:31\n",
    "BRACKEN SUMMARY (Kraken report: T3_19_2022.kreport2)\n",
    "    >>> Threshold: 0 \n",
    "    >>> Number of species in sample: 0 \n",
    "          >> Number of species with reads > threshold: 0 \n",
    "          >> Number of species with reads < threshold: 0 \n",
    "    >>> Total reads in sample: 9044371\n",
    "          >> Total reads kept at species level (reads > threshold): 0\n",
    "          >> Total reads discarded (species reads < threshold): 0\n",
    "          >> Reads distributed: 0\n",
    "          >> Reads not distributed (eg. no species above threshold): 93872\n",
    "          >> Unclassified reads: 8950499\n",
    "BRACKEN OUTPUT PRODUCED: //project/pi_sarah_gignouxwolfsohn_uml_edu/brooke/working/mcav_assembly_redo/taxonomy/kraken/bracken/T3_19_2022_species.bracken\n",
    "PROGRAM END TIME: 12-14-2023 02:40:31\n",
    "  Bracken complete.\n",
    " >> Checking for Valid Options...\n",
    " >> Running Bracken \n",
    "      >> python src/est_abundance.py -i T3_1_2019.kreport2 -o //project/pi_sarah_gignouxwolfsohn_uml_edu/brooke/working/mcav_assembly_redo/taxonomy/kraken/bracken/T3_1_2019_species.bracken -k //project/pi_sarah_gignouxwolfsohn_uml_edu/Reference_genomes/ref_databases/16S_SILVA132_k2db/database250mers.kmer_distrib -l S -t 0\n",
    "PROGRAM START TIME: 12-14-2023 02:40:31\n",
    "BRACKEN SUMMARY (Kraken report: T3_1_2019.kreport2)\n",
    "    >>> Threshold: 0 \n",
    "    >>> Number of species in sample: 0 \n",
    "          >> Number of species with reads > threshold: 0 \n",
    "          >> Number of species with reads < threshold: 0 \n",
    "    >>> Total reads in sample: 29977945\n",
    "          >> Total reads kept at species level (reads > threshold): 0\n",
    "          >> Total reads discarded (species reads < threshold): 0\n",
    "          >> Reads distributed: 0\n",
    "          >> Reads not distributed (eg. no species above threshold): 99522\n",
    "          >> Unclassified reads: 29878423\n",
    "BRACKEN OUTPUT PRODUCED: //project/pi_sarah_gignouxwolfsohn_uml_edu/brooke/working/mcav_assembly_redo/taxonomy/kraken/bracken/T3_1_2019_species.bracken\n",
    "PROGRAM END TIME: 12-14-2023 02:40:32\n",
    "  Bracken complete.\n",
    " >> Checking for Valid Options...\n",
    " >> Running Bracken \n",
    "      >> python src/est_abundance.py -i T3_40_2022.kreport2 -o //project/pi_sarah_gignouxwolfsohn_uml_edu/brooke/working/mcav_assembly_redo/taxonomy/kraken/bracken/T3_40_2022_species.bracken -k //project/pi_sarah_gignouxwolfsohn_uml_edu/Reference_genomes/ref_databases/16S_SILVA132_k2db/database250mers.kmer_distrib -l S -t 0\n",
    "PROGRAM START TIME: 12-14-2023 02:40:32\n",
    "BRACKEN SUMMARY (Kraken report: T3_40_2022.kreport2)\n",
    "    >>> Threshold: 0 \n",
    "    >>> Number of species in sample: 0 \n",
    "          >> Number of species with reads > threshold: 0 \n",
    "          >> Number of species with reads < threshold: 0 \n",
    "    >>> Total reads in sample: 26755227\n",
    "          >> Total reads kept at species level (reads > threshold): 0\n",
    "          >> Total reads discarded (species reads < threshold): 0\n",
    "          >> Reads distributed: 0\n",
    "          >> Reads not distributed (eg. no species above threshold): 77871\n",
    "          >> Unclassified reads: 26677356\n",
    "BRACKEN OUTPUT PRODUCED: //project/pi_sarah_gignouxwolfsohn_uml_edu/brooke/working/mcav_assembly_redo/taxonomy/kraken/bracken/T3_40_2022_species.bracken\n",
    "PROGRAM END TIME: 12-14-2023 02:40:32\n",
    "  Bracken complete.\n",
    " >> Checking for Valid Options...\n",
    " >> Running Bracken \n",
    "      >> python src/est_abundance.py -i T3_48_2022.kreport2 -o //project/pi_sarah_gignouxwolfsohn_uml_edu/brooke/working/mcav_assembly_redo/taxonomy/kraken/bracken/T3_48_2022_species.bracken -k //project/pi_sarah_gignouxwolfsohn_uml_edu/Reference_genomes/ref_databases/16S_SILVA132_k2db/database250mers.kmer_distrib -l S -t 0\n",
    "PROGRAM START TIME: 12-14-2023 02:40:32\n",
    "BRACKEN SUMMARY (Kraken report: T3_48_2022.kreport2)\n",
    "    >>> Threshold: 0 \n",
    "    >>> Number of species in sample: 0 \n",
    "          >> Number of species with reads > threshold: 0 \n",
    "          >> Number of species with reads < threshold: 0 \n",
    "    >>> Total reads in sample: 13255150\n",
    "          >> Total reads kept at species level (reads > threshold): 0\n",
    "          >> Total reads discarded (species reads < threshold): 0\n",
    "          >> Reads distributed: 0\n",
    "          >> Reads not distributed (eg. no species above threshold): 133945\n",
    "          >> Unclassified reads: 13121205\n",
    "BRACKEN OUTPUT PRODUCED: //project/pi_sarah_gignouxwolfsohn_uml_edu/brooke/working/mcav_assembly_redo/taxonomy/kraken/bracken/T3_48_2022_species.bracken\n",
    "PROGRAM END TIME: 12-14-2023 02:40:33\n",
    "  Bracken complete.\n",
    " >> Checking for Valid Options...\n",
    " >> Running Bracken \n",
    "      >> python src/est_abundance.py -i T3_49_2022.kreport2 -o //project/pi_sarah_gignouxwolfsohn_uml_edu/brooke/working/mcav_assembly_redo/taxonomy/kraken/bracken/T3_49_2022_species.bracken -k //project/pi_sarah_gignouxwolfsohn_uml_edu/Reference_genomes/ref_databases/16S_SILVA132_k2db/database250mers.kmer_distrib -l S -t 0\n",
    "PROGRAM START TIME: 12-14-2023 02:40:33\n",
    "BRACKEN SUMMARY (Kraken report: T3_49_2022.kreport2)\n",
    "    >>> Threshold: 0 \n",
    "    >>> Number of species in sample: 0 \n",
    "          >> Number of species with reads > threshold: 0 \n",
    "          >> Number of species with reads < threshold: 0 \n",
    "    >>> Total reads in sample: 15749618\n",
    "          >> Total reads kept at species level (reads > threshold): 0\n",
    "          >> Total reads discarded (species reads < threshold): 0\n",
    "          >> Reads distributed: 0\n",
    "          >> Reads not distributed (eg. no species above threshold): 149084\n",
    "          >> Unclassified reads: 15600534\n",
    "BRACKEN OUTPUT PRODUCED: //project/pi_sarah_gignouxwolfsohn_uml_edu/brooke/working/mcav_assembly_redo/taxonomy/kraken/bracken/T3_49_2022_species.bracken\n",
    "PROGRAM END TIME: 12-14-2023 02:40:33\n",
    "  Bracken complete.\n",
    " >> Checking for Valid Options...\n",
    " >> Running Bracken \n",
    "      >> python src/est_abundance.py -i T3_51_2022.kreport2 -o //project/pi_sarah_gignouxwolfsohn_uml_edu/brooke/working/mcav_assembly_redo/taxonomy/kraken/bracken/T3_51_2022_species.bracken -k //project/pi_sarah_gignouxwolfsohn_uml_edu/Reference_genomes/ref_databases/16S_SILVA132_k2db/database250mers.kmer_distrib -l S -t 0\n",
    "PROGRAM START TIME: 12-14-2023 02:40:33\n",
    "BRACKEN SUMMARY (Kraken report: T3_51_2022.kreport2)\n",
    "    >>> Threshold: 0 \n",
    "    >>> Number of species in sample: 0 \n",
    "          >> Number of species with reads > threshold: 0 \n",
    "          >> Number of species with reads < threshold: 0 \n",
    "    >>> Total reads in sample: 13877944\n",
    "          >> Total reads kept at species level (reads > threshold): 0\n",
    "          >> Total reads discarded (species reads < threshold): 0\n",
    "          >> Reads distributed: 0\n",
    "          >> Reads not distributed (eg. no species above threshold): 50639\n",
    "          >> Unclassified reads: 13827305\n",
    "BRACKEN OUTPUT PRODUCED: //project/pi_sarah_gignouxwolfsohn_uml_edu/brooke/working/mcav_assembly_redo/taxonomy/kraken/bracken/T3_51_2022_species.bracken\n",
    "PROGRAM END TIME: 12-14-2023 02:40:34\n",
    "  Bracken complete.\n",
    " >> Checking for Valid Options...\n",
    " >> Running Bracken \n",
    "      >> python src/est_abundance.py -i T3_8_2019.kreport2 -o //project/pi_sarah_gignouxwolfsohn_uml_edu/brooke/working/mcav_assembly_redo/taxonomy/kraken/bracken/T3_8_2019_species.bracken -k //project/pi_sarah_gignouxwolfsohn_uml_edu/Reference_genomes/ref_databases/16S_SILVA132_k2db/database250mers.kmer_distrib -l S -t 0\n",
    "PROGRAM START TIME: 12-14-2023 02:40:34\n",
    "BRACKEN SUMMARY (Kraken report: T3_8_2019.kreport2)\n",
    "    >>> Threshold: 0 \n",
    "    >>> Number of species in sample: 0 \n",
    "          >> Number of species with reads > threshold: 0 \n",
    "          >> Number of species with reads < threshold: 0 \n",
    "    >>> Total reads in sample: 16453406\n",
    "          >> Total reads kept at species level (reads > threshold): 0\n",
    "          >> Total reads discarded (species reads < threshold): 0\n",
    "          >> Reads distributed: 0\n",
    "          >> Reads not distributed (eg. no species above threshold): 94662\n",
    "          >> Unclassified reads: 16358744\n",
    "BRACKEN OUTPUT PRODUCED: //project/pi_sarah_gignouxwolfsohn_uml_edu/brooke/working/mcav_assembly_redo/taxonomy/kraken/bracken/T3_8_2019_species.bracken\n",
    "PROGRAM END TIME: 12-14-2023 02:40:34\n",
    "  Bracken complete.\n",
    " >> Checking for Valid Options...\n",
    " >> Running Bracken \n",
    "      >> python src/est_abundance.py -i T3_9_2019.kreport2 -o //project/pi_sarah_gignouxwolfsohn_uml_edu/brooke/working/mcav_assembly_redo/taxonomy/kraken/bracken/T3_9_2019_species.bracken -k //project/pi_sarah_gignouxwolfsohn_uml_edu/Reference_genomes/ref_databases/16S_SILVA132_k2db/database250mers.kmer_distrib -l S -t 0\n",
    "PROGRAM START TIME: 12-14-2023 02:40:34\n",
    "BRACKEN SUMMARY (Kraken report: T3_9_2019.kreport2)\n",
    "    >>> Threshold: 0 \n",
    "    >>> Number of species in sample: 0 \n",
    "          >> Number of species with reads > threshold: 0 \n",
    "          >> Number of species with reads < threshold: 0 \n",
    "    >>> Total reads in sample: 14032207\n",
    "          >> Total reads kept at species level (reads > threshold): 0\n",
    "          >> Total reads discarded (species reads < threshold): 0\n",
    "          >> Reads distributed: 0\n",
    "          >> Reads not distributed (eg. no species above threshold): 69979\n",
    "          >> Unclassified reads: 13962228\n",
    "BRACKEN OUTPUT PRODUCED: //project/pi_sarah_gignouxwolfsohn_uml_edu/brooke/working/mcav_assembly_redo/taxonomy/kraken/bracken/T3_9_2019_species.bracken\n",
    "PROGRAM END TIME: 12-14-2023 02:40:34\n",
    "  Bracken complete."
   ]
  },
  {
   "cell_type": "markdown",
   "id": "24fddd76-2c63-40f7-8c47-ec8ae46b3b9a",
   "metadata": {
    "tags": []
   },
   "source": [
    "##### KrakenTools \n",
    "- downstream analysis using kraken results\n",
    "- https://ccb.jhu.edu/software/krakentools/index.shtml\n",
    "- download zip file "
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "bf669216-e56f-42fa-9de0-ec1bf184b74a",
   "metadata": {},
   "outputs": [],
   "source": [
    "SCRIPTPATH=/project/pi_sarah_gignouxwolfsohn_uml_edu/brooke/bash_scripts/KrakenTools-master/DiversityTools\n",
    "cd $SCRIPTPATH\n",
    "# make scripts executable \n",
    "chmod +x alpha\\_diversity.py\n",
    "./alpha\\_diversity.py -h\n",
    "# input file must be bracken\n",
    "\n",
    "chmod +x beta\\_diversity.py\n",
    "./beta\\_diversity.py -h\n",
    "# Bray-Curtis dissimilarity"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "f9ccac70-8898-4bec-9092-9fe9548e2ad1",
   "metadata": {},
   "outputs": [],
   "source": [
    "# Alpha Diversity\n",
    "# Shannon diversity - genus level \n",
    "\n",
    "SCRIPTPATH=/project/pi_sarah_gignouxwolfsohn_uml_edu/brooke/bash_scripts/KrakenTools-master/DiversityTools\n",
    "FILEPATH=//project/pi_sarah_gignouxwolfsohn_uml_edu/brooke/mcav_taxonomy/kraken\n",
    "#ls $FILEPATH | find *.kreport2 > kraken_list.txt\n",
    "cat kraken_list.txt| sed 's/.kreport2/''/' > SAMPLEID\n",
    "\n",
    "while IFS= read -r line; do\n",
    "    python $SCRIPTPATH/alpha_diversity.py -f ${line}.bracken \n",
    "done < SAMPLEID\n",
    "\n",
    "# don't know if it does multiple at a time or just one at a time...and what is output?\n",
    "#-f, --filename MYFILE.BRACKEN..........Bracken output file\n",
    "#-a, --alpha TYPE.......................Single letter alpha diversity type (Sh, BP, Si, ISi, F)"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "4a234bab-162f-4454-aaec-599f30ac4f77",
   "metadata": {},
   "source": [
    "Shannon's diversity: 2.8405589418696024\n",
    "Shannon's diversity: 3.747956479976194\n",
    "Shannon's diversity: 3.240773802595393\n",
    "Shannon's diversity: 3.357943870893113\n",
    "Shannon's diversity: 3.127505271223385\n",
    "Shannon's diversity: 2.948836172095586\n",
    "Shannon's diversity: 3.155355846512862\n",
    "Shannon's diversity: 2.9534230075080843\n",
    "Shannon's diversity: 3.837848266676804\n",
    "Shannon's diversity: 3.017049434243576\n",
    "Shannon's diversity: 3.3014806325360837\n",
    "Shannon's diversity: 3.1828393709134533\n",
    "Shannon's diversity: 2.673104478260622\n",
    "Shannon's diversity: 2.8628687828400676\n",
    "Shannon's diversity: 3.1197450036741374\n",
    "Shannon's diversity: 3.4793331275395554\n",
    "Shannon's diversity: 2.842877602269089\n",
    "Shannon's diversity: 2.9019378226014787\n",
    "Shannon's diversity: 3.1864428196047045\n",
    "Shannon's diversity: 3.1428692882448113\n",
    "Shannon's diversity: 3.0649759367640863"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "c19febbb-b03e-4482-a375-a9ce197b5ddd",
   "metadata": {},
   "outputs": [],
   "source": [
    "cat SAMPLEID"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "51444fbc-1c4e-4738-b69d-ca14e192e8b2",
   "metadata": {},
   "source": [
    "T1_12_2022\n",
    "T1_13_2022\n",
    "T1_16_2019\n",
    "T1_20_2019\n",
    "T1_24_2019\n",
    "T1_40_2022\n",
    "T1_57_2022\n",
    "T1_70_2022\n",
    "T2_10_2022\n",
    "T2_16_2019\n",
    "T3_13_2022\n",
    "T3_14_2019\n",
    "T3_15_2019\n",
    "T3_19_2022\n",
    "T3_1_2019\n",
    "T3_40_2022\n",
    "T3_48_2022\n",
    "T3_49_2022\n",
    "T3_51_2022\n",
    "T3_8_2019\n",
    "T3_9_2019"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "a5f33441-bc80-49ec-8590-e2b93df876af",
   "metadata": {},
   "outputs": [],
   "source": [
    "# Beta Diversity\n",
    "\n",
    "# collect list of input files\n",
    "FILEPATH=//project/pi_sarah_gignouxwolfsohn_uml_edu/brooke/mcav_taxonomy/kraken\n",
    "mv *_species.bracken species\n",
    "ls $FILEPATH | find *.bracken > bracken_list.txt\n",
    "FILES=xargs -a bracken_list.txt -d '\\n'"
   ]
  },
  {
   "cell_type": "raw",
   "id": "d00afef5-6612-46c6-b524-ec81c87ad20b",
   "metadata": {},
   "source": [
    "T1_12_2022.bracken T1_13_2022.bracken T1_16_2019.bracken T1_20_2019.bracken T1_24_2019.bracken T1_40_2022.bracken T1_57_2022.bracken T1_70_2022.bracken T2_10_2022.bracken T2_16_2019.bracken T3_13_2022.bracken T3_14_2019.bracken T3_15_2019.bracken T3_19_2022.bracken T3_1_2019.bracken T3_40_2022.bracken T3_48_2022.bracken T3_49_2022.bracken T3_51_2022.bracken T3_8_2019.bracken T3_9_2019.bracken"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "352dbc09-b330-4422-8d8c-a98a6cbadc1a",
   "metadata": {},
   "outputs": [],
   "source": [
    "python $SCRIPTPATH/beta_diversity.py --type bracken --input-files T1_12_2022.bracken T1_13_2022.bracken T1_16_2019.bracken T1_20_2019.bracken T1_24_2019.bracken T1_40_2022.bracken T1_57_2022.bracken T1_70_2022.bracken T2_10_2022.bracken T2_16_2019.bracken T3_13_2022.bracken T3_14_2019.bracken T3_15_2019.bracken T3_19_2022.bracken T3_1_2019.bracken T3_40_2022.bracken T3_48_2022.bracken T3_49_2022.bracken T3_51_2022.bracken T3_8_2019.bracken T3_9_2019.bracken > b_diversity.txt\n"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "a160f15f-c856-4e5f-be28-4c7b6bd5d7bb",
   "metadata": {
    "jp-MarkdownHeadingCollapsed": true,
    "tags": []
   },
   "source": [
    "#### Attempting to use Skbio to create pcoa plot from b diversity matrix\n",
    "\n",
    "- matrix is non symmetrical ...something is wrong with the output from krakentools...probably something wrong with the files is my guess"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 7,
   "id": "66468571-f8ec-4720-9bed-52a026115def",
   "metadata": {},
   "outputs": [],
   "source": [
    "from skbio.diversity import beta_diversity\n",
    "from skbio.stats.distance import mantel"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 8,
   "id": "697c21c1-70f8-44fc-858f-c536bcc1fa69",
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "#0\tT1_12_2022.kreport2 (20742228 reads)\n",
      "#1\tT1_13_2022.kreport2 (20861960 reads)\n",
      "#2\tT1_16_2019.kreport2 (25807031 reads)\n",
      "#3\tT1_20_2019.kreport2 (8517445 reads)\n",
      "#4\tT1_24_2019.kreport2 (17135017 reads)\n",
      "#5\tT1_40_2022.kreport2 (15419779 reads)\n",
      "#6\tT1_57_2022.kreport2 (31964707 reads)\n",
      "#7\tT1_70_2022.kreport2 (15776674 reads)\n",
      "#8\tT2_10_2022.kreport2 (16876812 reads)\n",
      "#9\tT2_16_2019.kreport2 (34302369 reads)\n",
      "#10\tT3_13_2022.kreport2 (13449647 reads)\n",
      "#11\tT3_14_2019.kreport2 (21400877 reads)\n",
      "#12\tT3_15_2019.kreport2 (19854618 reads)\n",
      "#13\tT3_19_2022.kreport2 (9044371 reads)\n",
      "#14\tT3_1_2019.kreport2 (29977945 reads)\n",
      "#15\tT3_40_2022.kreport2 (26755227 reads)\n",
      "#16\tT3_48_2022.kreport2 (13255150 reads)\n",
      "#17\tT3_49_2022.kreport2 (15749618 reads)\n",
      "#18\tT3_51_2022.kreport2 (13877944 reads)\n",
      "#19\tT3_8_2019.kreport2 (16453406 reads)\n",
      "#20\tT3_9_2019.kreport2 (14032207 reads)\n",
      "x\t0\t1\t2\t3\t4\t5\t6\t7\t8\t9\t10\t11\t12\t13\t14\t15\t16\t17\t18\t19\t20\n",
      "0\t0.000\t0.004\t0.109\t0.419\t0.096\t0.151\t0.213\t0.140\t0.104\t0.247\t0.214\t0.016\t0.022\t0.397\t0.182\t0.127\t0.226\t0.143\t0.199\t0.118\t0.195\n",
      "1\tx.xxx\t0.000\t0.107\t0.422\t0.100\t0.155\t0.211\t0.144\t0.107\t0.244\t0.217\t0.014\t0.026\t0.400\t0.180\t0.125\t0.230\t0.146\t0.203\t0.122\t0.198\n",
      "2\tx.xxx\tx.xxx\t0.000\t0.505\t0.203\t0.255\t0.107\t0.245\t0.210\t0.142\t0.315\t0.095\t0.131\t0.485\t0.075\t0.019\t0.326\t0.247\t0.301\t0.224\t0.297\n",
      "3\tx.xxx\tx.xxx\tx.xxx\t0.000\t0.337\t0.289\t0.580\t0.300\t0.330\t0.602\t0.226\t0.431\t0.401\t0.032\t0.558\t0.518\t0.219\t0.299\t0.241\t0.318\t0.246\n",
      "4\tx.xxx\tx.xxx\tx.xxx\tx.xxx\t0.000\t0.056\t0.302\t0.045\t0.009\t0.334\t0.121\t0.111\t0.074\t0.313\t0.273\t0.220\t0.133\t0.048\t0.105\t0.023\t0.101\n",
      "5\tx.xxx\tx.xxx\tx.xxx\tx.xxx\tx.xxx\t0.000\t0.350\t0.012\t0.050\t0.382\t0.069\t0.164\t0.130\t0.261\t0.323\t0.271\t0.078\t0.011\t0.053\t0.035\t0.048\n",
      "6\tx.xxx\tx.xxx\tx.xxx\tx.xxx\tx.xxx\tx.xxx\t0.000\t0.340\t0.309\t0.038\t0.408\t0.198\t0.234\t0.559\t0.032\t0.089\t0.415\t0.342\t0.395\t0.321\t0.390\n",
      "7\tx.xxx\tx.xxx\tx.xxx\tx.xxx\tx.xxx\tx.xxx\tx.xxx\t0.000\t0.039\t0.372\t0.080\t0.153\t0.119\t0.272\t0.312\t0.261\t0.089\t0.003\t0.064\t0.024\t0.059\n",
      "8\tx.xxx\tx.xxx\tx.xxx\tx.xxx\tx.xxx\tx.xxx\tx.xxx\tx.xxx\t0.000\t0.341\t0.114\t0.119\t0.082\t0.307\t0.280\t0.227\t0.127\t0.041\t0.099\t0.016\t0.094\n",
      "9\tx.xxx\tx.xxx\tx.xxx\tx.xxx\tx.xxx\tx.xxx\tx.xxx\tx.xxx\tx.xxx\t0.000\t0.437\t0.233\t0.267\t0.585\t0.068\t0.124\t0.446\t0.374\t0.424\t0.352\t0.420\n",
      "10\tx.xxx\tx.xxx\tx.xxx\tx.xxx\tx.xxx\tx.xxx\tx.xxx\tx.xxx\tx.xxx\tx.xxx\t0.000\t0.228\t0.193\t0.201\t0.381\t0.331\t0.015\t0.079\t0.016\t0.101\t0.022\n",
      "11\tx.xxx\tx.xxx\tx.xxx\tx.xxx\tx.xxx\tx.xxx\tx.xxx\tx.xxx\tx.xxx\tx.xxx\tx.xxx\t0.000\t0.038\t0.408\t0.167\t0.112\t0.239\t0.156\t0.213\t0.132\t0.208\n",
      "12\tx.xxx\tx.xxx\tx.xxx\tx.xxx\tx.xxx\tx.xxx\tx.xxx\tx.xxx\tx.xxx\tx.xxx\tx.xxx\tx.xxx\t0.000\t0.378\t0.203\t0.148\t0.205\t0.121\t0.178\t0.097\t0.174\n",
      "13\tx.xxx\tx.xxx\tx.xxx\tx.xxx\tx.xxx\tx.xxx\tx.xxx\tx.xxx\tx.xxx\tx.xxx\tx.xxx\tx.xxx\tx.xxx\t0.000\t0.538\t0.497\t0.189\t0.270\t0.215\t0.293\t0.219\n",
      "14\tx.xxx\tx.xxx\tx.xxx\tx.xxx\tx.xxx\tx.xxx\tx.xxx\tx.xxx\tx.xxx\tx.xxx\tx.xxx\tx.xxx\tx.xxx\tx.xxx\t0.000\t0.057\t0.390\t0.315\t0.367\t0.293\t0.363\n",
      "15\tx.xxx\tx.xxx\tx.xxx\tx.xxx\tx.xxx\tx.xxx\tx.xxx\tx.xxx\tx.xxx\tx.xxx\tx.xxx\tx.xxx\tx.xxx\tx.xxx\tx.xxx\t0.000\t0.341\t0.263\t0.317\t0.240\t0.313\n",
      "16\tx.xxx\tx.xxx\tx.xxx\tx.xxx\tx.xxx\tx.xxx\tx.xxx\tx.xxx\tx.xxx\tx.xxx\tx.xxx\tx.xxx\tx.xxx\tx.xxx\tx.xxx\tx.xxx\t0.000\t0.086\t0.029\t0.112\t0.034\n",
      "17\tx.xxx\tx.xxx\tx.xxx\tx.xxx\tx.xxx\tx.xxx\tx.xxx\tx.xxx\tx.xxx\tx.xxx\tx.xxx\tx.xxx\tx.xxx\tx.xxx\tx.xxx\tx.xxx\tx.xxx\t0.000\t0.063\t0.026\t0.058\n",
      "18\tx.xxx\tx.xxx\tx.xxx\tx.xxx\tx.xxx\tx.xxx\tx.xxx\tx.xxx\tx.xxx\tx.xxx\tx.xxx\tx.xxx\tx.xxx\tx.xxx\tx.xxx\tx.xxx\tx.xxx\tx.xxx\t0.000\t0.085\t0.006\n",
      "19\tx.xxx\tx.xxx\tx.xxx\tx.xxx\tx.xxx\tx.xxx\tx.xxx\tx.xxx\tx.xxx\tx.xxx\tx.xxx\tx.xxx\tx.xxx\tx.xxx\tx.xxx\tx.xxx\tx.xxx\tx.xxx\tx.xxx\t0.000\t0.081\n",
      "20\tx.xxx\tx.xxx\tx.xxx\tx.xxx\tx.xxx\tx.xxx\tx.xxx\tx.xxx\tx.xxx\tx.xxx\tx.xxx\tx.xxx\tx.xxx\tx.xxx\tx.xxx\tx.xxx\tx.xxx\tx.xxx\tx.xxx\tx.xxx\t0.000\n",
      "\n"
     ]
    }
   ],
   "source": [
    "#view b diversity file \n",
    "file_path = '//project/pi_sarah_gignouxwolfsohn_uml_edu/brooke/working/mcav_assembly_redo/taxonomy/kraken/b_diversity.txt'\n",
    "\n",
    "with open(file_path, 'r') as file:\n",
    "    content = file.read()\n",
    "    print(content)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 7,
   "id": "fd981208-1a18-4b97-9b67-0add727a5d46",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "'/project/pi_sarah_gignouxwolfsohn_uml_edu/brooke/CBC_metagenomics/mcav1'"
      ]
     },
     "execution_count": 7,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "# Plan\n",
    "# load in first part as a seperate file so i can keep track of which sample goes with which\n",
    "import os\n",
    "os.getcwd()\n",
    "# load in second part as compatible distance matrix "
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 44,
   "id": "32d62dc8-a72d-44a3-8df9-612f085fdf90",
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "#0\tT1_12_2022.kreport2 (20742228 reads)\n",
      "#1\tT1_13_2022.kreport2 (20861960 reads)\n",
      "#2\tT1_16_2019.kreport2 (25807031 reads)\n",
      "#3\tT1_20_2019.kreport2 (8517445 reads)\n",
      "#4\tT1_24_2019.kreport2 (17135017 reads)\n",
      "#5\tT1_40_2022.kreport2 (15419779 reads)\n",
      "#6\tT1_57_2022.kreport2 (31964707 reads)\n",
      "#7\tT1_70_2022.kreport2 (15776674 reads)\n",
      "#8\tT2_10_2022.kreport2 (16876812 reads)\n",
      "#9\tT2_16_2019.kreport2 (34302369 reads)\n",
      "#10\tT3_13_2022.kreport2 (13449647 reads)\n",
      "#11\tT3_14_2019.kreport2 (21400877 reads)\n",
      "#12\tT3_15_2019.kreport2 (19854618 reads)\n",
      "#13\tT3_19_2022.kreport2 (9044371 reads)\n",
      "#14\tT3_1_2019.kreport2 (29977945 reads)\n",
      "#15\tT3_40_2022.kreport2 (26755227 reads)\n",
      "#16\tT3_48_2022.kreport2 (13255150 reads)\n",
      "#17\tT3_49_2022.kreport2 (15749618 reads)\n",
      "#18\tT3_51_2022.kreport2 (13877944 reads)\n",
      "#19\tT3_8_2019.kreport2 (16453406 reads)\n",
      "#20\tT3_9_2019.kreport2 (14032207 reads)\n",
      "        0      1      2      3      4      5      6      7      8      9  ...  \\\n",
      "x                                                                         ...   \n",
      "0   0.000  0.004  0.109  0.419  0.096  0.151  0.213  0.140  0.104  0.247  ...   \n",
      "1   x.xxx  0.000  0.107  0.422  0.100  0.155  0.211  0.144  0.107  0.244  ...   \n",
      "2   x.xxx  x.xxx  0.000  0.505  0.203  0.255  0.107  0.245  0.210  0.142  ...   \n",
      "3   x.xxx  x.xxx  x.xxx  0.000  0.337  0.289  0.580  0.300  0.330  0.602  ...   \n",
      "4   x.xxx  x.xxx  x.xxx  x.xxx  0.000  0.056  0.302  0.045  0.009  0.334  ...   \n",
      "5   x.xxx  x.xxx  x.xxx  x.xxx  x.xxx  0.000  0.350  0.012  0.050  0.382  ...   \n",
      "6   x.xxx  x.xxx  x.xxx  x.xxx  x.xxx  x.xxx  0.000  0.340  0.309  0.038  ...   \n",
      "7   x.xxx  x.xxx  x.xxx  x.xxx  x.xxx  x.xxx  x.xxx  0.000  0.039  0.372  ...   \n",
      "8   x.xxx  x.xxx  x.xxx  x.xxx  x.xxx  x.xxx  x.xxx  x.xxx  0.000  0.341  ...   \n",
      "9   x.xxx  x.xxx  x.xxx  x.xxx  x.xxx  x.xxx  x.xxx  x.xxx  x.xxx  0.000  ...   \n",
      "10  x.xxx  x.xxx  x.xxx  x.xxx  x.xxx  x.xxx  x.xxx  x.xxx  x.xxx  x.xxx  ...   \n",
      "11  x.xxx  x.xxx  x.xxx  x.xxx  x.xxx  x.xxx  x.xxx  x.xxx  x.xxx  x.xxx  ...   \n",
      "12  x.xxx  x.xxx  x.xxx  x.xxx  x.xxx  x.xxx  x.xxx  x.xxx  x.xxx  x.xxx  ...   \n",
      "13  x.xxx  x.xxx  x.xxx  x.xxx  x.xxx  x.xxx  x.xxx  x.xxx  x.xxx  x.xxx  ...   \n",
      "14  x.xxx  x.xxx  x.xxx  x.xxx  x.xxx  x.xxx  x.xxx  x.xxx  x.xxx  x.xxx  ...   \n",
      "15  x.xxx  x.xxx  x.xxx  x.xxx  x.xxx  x.xxx  x.xxx  x.xxx  x.xxx  x.xxx  ...   \n",
      "16  x.xxx  x.xxx  x.xxx  x.xxx  x.xxx  x.xxx  x.xxx  x.xxx  x.xxx  x.xxx  ...   \n",
      "17  x.xxx  x.xxx  x.xxx  x.xxx  x.xxx  x.xxx  x.xxx  x.xxx  x.xxx  x.xxx  ...   \n",
      "18  x.xxx  x.xxx  x.xxx  x.xxx  x.xxx  x.xxx  x.xxx  x.xxx  x.xxx  x.xxx  ...   \n",
      "19  x.xxx  x.xxx  x.xxx  x.xxx  x.xxx  x.xxx  x.xxx  x.xxx  x.xxx  x.xxx  ...   \n",
      "20  x.xxx  x.xxx  x.xxx  x.xxx  x.xxx  x.xxx  x.xxx  x.xxx  x.xxx  x.xxx  ...   \n",
      "\n",
      "       11     12     13     14     15     16     17     18     19     20  \n",
      "x                                                                         \n",
      "0   0.016  0.022  0.397  0.182  0.127  0.226  0.143  0.199  0.118  0.195  \n",
      "1   0.014  0.026  0.400  0.180  0.125  0.230  0.146  0.203  0.122  0.198  \n",
      "2   0.095  0.131  0.485  0.075  0.019  0.326  0.247  0.301  0.224  0.297  \n",
      "3   0.431  0.401  0.032  0.558  0.518  0.219  0.299  0.241  0.318  0.246  \n",
      "4   0.111  0.074  0.313  0.273  0.220  0.133  0.048  0.105  0.023  0.101  \n",
      "5   0.164  0.130  0.261  0.323  0.271  0.078  0.011  0.053  0.035  0.048  \n",
      "6   0.198  0.234  0.559  0.032  0.089  0.415  0.342  0.395  0.321  0.390  \n",
      "7   0.153  0.119  0.272  0.312  0.261  0.089  0.003  0.064  0.024  0.059  \n",
      "8   0.119  0.082  0.307  0.280  0.227  0.127  0.041  0.099  0.016  0.094  \n",
      "9   0.233  0.267  0.585  0.068  0.124  0.446  0.374  0.424  0.352  0.420  \n",
      "10  0.228  0.193  0.201  0.381  0.331  0.015  0.079  0.016  0.101  0.022  \n",
      "11  0.000  0.038  0.408  0.167  0.112  0.239  0.156  0.213  0.132  0.208  \n",
      "12  x.xxx  0.000  0.378  0.203  0.148  0.205  0.121  0.178  0.097  0.174  \n",
      "13  x.xxx  x.xxx  0.000  0.538  0.497  0.189  0.270  0.215  0.293  0.219  \n",
      "14  x.xxx  x.xxx  x.xxx  0.000  0.057  0.390  0.315  0.367  0.293  0.363  \n",
      "15  x.xxx  x.xxx  x.xxx  x.xxx  0.000  0.341  0.263  0.317  0.240  0.313  \n",
      "16  x.xxx  x.xxx  x.xxx  x.xxx  x.xxx  0.000  0.086  0.029  0.112  0.034  \n",
      "17  x.xxx  x.xxx  x.xxx  x.xxx  x.xxx  x.xxx  0.000  0.063  0.026  0.058  \n",
      "18  x.xxx  x.xxx  x.xxx  x.xxx  x.xxx  x.xxx  x.xxx  0.000  0.085  0.006  \n",
      "19  x.xxx  x.xxx  x.xxx  x.xxx  x.xxx  x.xxx  x.xxx  x.xxx  0.000  0.081  \n",
      "20  x.xxx  x.xxx  x.xxx  x.xxx  x.xxx  x.xxx  x.xxx  x.xxx  x.xxx  0.000  \n",
      "\n",
      "[21 rows x 21 columns]\n"
     ]
    }
   ],
   "source": [
    "import pandas as pd\n",
    "import numpy as np\n",
    "\n",
    "# Read the first 20 lines separately\n",
    "with open(file_path, 'r') as file:\n",
    "    first_20_lines = [file.readline().strip() for _ in range(21)]\n",
    "\n",
    "# Load the distance matrix directly into a NumPy array\n",
    "# change 'x' value to nan so numpy can read it\n",
    "\n",
    "\n",
    "#b_diversity = np.loadtxt(file_path, skiprows=20, dtype=float, converters={0: converter_func, 1: converter_func, 2: converter_func})\n",
    "\n",
    "# Read the remaining lines into a DataFrame\n",
    "b_diversity = pd.read_csv(file_path, sep='\\t', skiprows=21, header=0, index_col=0)\n",
    "\n",
    "# Display the first 20 lines\n",
    "for line in first_20_lines:\n",
    "    print(line)\n",
    "# Display the distance matrix\n",
    "print(b_diversity)\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 46,
   "id": "fd683607-0180-4b9f-bfcc-b112876ef4cb",
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "        0      1      2      3      4      5      6      7      8      9  ...  \\\n",
      "x                                                                         ...   \n",
      "0   0.000  0.004  0.109  0.419  0.096  0.151  0.213  0.140  0.104  0.247  ...   \n",
      "1       0  0.000  0.107  0.422  0.100  0.155  0.211  0.144  0.107  0.244  ...   \n",
      "2       0      0  0.000  0.505  0.203  0.255  0.107  0.245  0.210  0.142  ...   \n",
      "3       0      0      0  0.000  0.337  0.289  0.580  0.300  0.330  0.602  ...   \n",
      "4       0      0      0      0  0.000  0.056  0.302  0.045  0.009  0.334  ...   \n",
      "5       0      0      0      0      0  0.000  0.350  0.012  0.050  0.382  ...   \n",
      "6       0      0      0      0      0      0  0.000  0.340  0.309  0.038  ...   \n",
      "7       0      0      0      0      0      0      0  0.000  0.039  0.372  ...   \n",
      "8       0      0      0      0      0      0      0      0  0.000  0.341  ...   \n",
      "9       0      0      0      0      0      0      0      0      0  0.000  ...   \n",
      "10      0      0      0      0      0      0      0      0      0      0  ...   \n",
      "11      0      0      0      0      0      0      0      0      0      0  ...   \n",
      "12      0      0      0      0      0      0      0      0      0      0  ...   \n",
      "13      0      0      0      0      0      0      0      0      0      0  ...   \n",
      "14      0      0      0      0      0      0      0      0      0      0  ...   \n",
      "15      0      0      0      0      0      0      0      0      0      0  ...   \n",
      "16      0      0      0      0      0      0      0      0      0      0  ...   \n",
      "17      0      0      0      0      0      0      0      0      0      0  ...   \n",
      "18      0      0      0      0      0      0      0      0      0      0  ...   \n",
      "19      0      0      0      0      0      0      0      0      0      0  ...   \n",
      "20      0      0      0      0      0      0      0      0      0      0  ...   \n",
      "\n",
      "       11     12     13     14     15     16     17     18     19     20  \n",
      "x                                                                         \n",
      "0   0.016  0.022  0.397  0.182  0.127  0.226  0.143  0.199  0.118  0.195  \n",
      "1   0.014  0.026  0.400  0.180  0.125  0.230  0.146  0.203  0.122  0.198  \n",
      "2   0.095  0.131  0.485  0.075  0.019  0.326  0.247  0.301  0.224  0.297  \n",
      "3   0.431  0.401  0.032  0.558  0.518  0.219  0.299  0.241  0.318  0.246  \n",
      "4   0.111  0.074  0.313  0.273  0.220  0.133  0.048  0.105  0.023  0.101  \n",
      "5   0.164  0.130  0.261  0.323  0.271  0.078  0.011  0.053  0.035  0.048  \n",
      "6   0.198  0.234  0.559  0.032  0.089  0.415  0.342  0.395  0.321  0.390  \n",
      "7   0.153  0.119  0.272  0.312  0.261  0.089  0.003  0.064  0.024  0.059  \n",
      "8   0.119  0.082  0.307  0.280  0.227  0.127  0.041  0.099  0.016  0.094  \n",
      "9   0.233  0.267  0.585  0.068  0.124  0.446  0.374  0.424  0.352  0.420  \n",
      "10  0.228  0.193  0.201  0.381  0.331  0.015  0.079  0.016  0.101  0.022  \n",
      "11  0.000  0.038  0.408  0.167  0.112  0.239  0.156  0.213  0.132  0.208  \n",
      "12      0  0.000  0.378  0.203  0.148  0.205  0.121  0.178  0.097  0.174  \n",
      "13      0      0  0.000  0.538  0.497  0.189  0.270  0.215  0.293  0.219  \n",
      "14      0      0      0  0.000  0.057  0.390  0.315  0.367  0.293  0.363  \n",
      "15      0      0      0      0  0.000  0.341  0.263  0.317  0.240  0.313  \n",
      "16      0      0      0      0      0  0.000  0.086  0.029  0.112  0.034  \n",
      "17      0      0      0      0      0      0  0.000  0.063  0.026  0.058  \n",
      "18      0      0      0      0      0      0      0  0.000  0.085  0.006  \n",
      "19      0      0      0      0      0      0      0      0  0.000  0.081  \n",
      "20      0      0      0      0      0      0      0      0      0  0.000  \n",
      "\n",
      "[21 rows x 21 columns]\n"
     ]
    }
   ],
   "source": [
    "#replace null values with 'NaN'\n",
    "bd_cleaned=b_diversity.replace('x.xxx', 0)\n",
    "print(bd_cleaned)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 47,
   "id": "f093d88b-1d47-40aa-a2eb-4403ac4bf436",
   "metadata": {},
   "outputs": [
    {
     "ename": "DistanceMatrixError",
     "evalue": "Data must be symmetric and cannot contain NaNs.",
     "output_type": "error",
     "traceback": [
      "\u001b[0;31m---------------------------------------------------------------------------\u001b[0m",
      "\u001b[0;31mDistanceMatrixError\u001b[0m                       Traceback (most recent call last)",
      "Cell \u001b[0;32mIn[47], line 2\u001b[0m\n\u001b[1;32m      1\u001b[0m \u001b[38;5;28;01mfrom\u001b[39;00m \u001b[38;5;21;01mskbio\u001b[39;00m\u001b[38;5;21;01m.\u001b[39;00m\u001b[38;5;21;01mstats\u001b[39;00m\u001b[38;5;21;01m.\u001b[39;00m\u001b[38;5;21;01mordination\u001b[39;00m \u001b[38;5;28;01mimport\u001b[39;00m pcoa\n\u001b[0;32m----> 2\u001b[0m b_pc \u001b[38;5;241m=\u001b[39m \u001b[43mpcoa\u001b[49m\u001b[43m(\u001b[49m\u001b[43mbd_cleaned\u001b[49m\u001b[43m)\u001b[49m\n",
      "File \u001b[0;32m~/.conda/envs/skbio/lib/python3.8/site-packages/skbio/stats/ordination/_principal_coordinate_analysis.py:96\u001b[0m, in \u001b[0;36mpcoa\u001b[0;34m(distance_matrix, method, number_of_dimensions, inplace)\u001b[0m\n\u001b[1;32m     23\u001b[0m \u001b[38;5;129m@experimental\u001b[39m(as_of\u001b[38;5;241m=\u001b[39m\u001b[38;5;124m\"\u001b[39m\u001b[38;5;124m0.4.0\u001b[39m\u001b[38;5;124m\"\u001b[39m)\n\u001b[1;32m     24\u001b[0m \u001b[38;5;28;01mdef\u001b[39;00m \u001b[38;5;21mpcoa\u001b[39m(distance_matrix, method\u001b[38;5;241m=\u001b[39m\u001b[38;5;124m\"\u001b[39m\u001b[38;5;124meigh\u001b[39m\u001b[38;5;124m\"\u001b[39m, number_of_dimensions\u001b[38;5;241m=\u001b[39m\u001b[38;5;241m0\u001b[39m,\n\u001b[1;32m     25\u001b[0m          inplace\u001b[38;5;241m=\u001b[39m\u001b[38;5;28;01mFalse\u001b[39;00m):\n\u001b[1;32m     26\u001b[0m \u001b[38;5;250m    \u001b[39m\u001b[38;5;124mr\u001b[39m\u001b[38;5;124;03m\"\"\"Perform Principal Coordinate Analysis.\u001b[39;00m\n\u001b[1;32m     27\u001b[0m \n\u001b[1;32m     28\u001b[0m \u001b[38;5;124;03m    Principal Coordinate Analysis (PCoA) is a method similar\u001b[39;00m\n\u001b[0;32m   (...)\u001b[0m\n\u001b[1;32m     94\u001b[0m \u001b[38;5;124;03m        ignored.\u001b[39;00m\n\u001b[1;32m     95\u001b[0m \u001b[38;5;124;03m    \"\"\"\u001b[39;00m\n\u001b[0;32m---> 96\u001b[0m     distance_matrix \u001b[38;5;241m=\u001b[39m \u001b[43mDistanceMatrix\u001b[49m\u001b[43m(\u001b[49m\u001b[43mdistance_matrix\u001b[49m\u001b[43m)\u001b[49m\n\u001b[1;32m     98\u001b[0m     \u001b[38;5;66;03m# Center distance matrix, a requirement for PCoA here\u001b[39;00m\n\u001b[1;32m     99\u001b[0m     matrix_data \u001b[38;5;241m=\u001b[39m center_distance_matrix(distance_matrix\u001b[38;5;241m.\u001b[39mdata, inplace\u001b[38;5;241m=\u001b[39minplace)\n",
      "File \u001b[0;32m~/.conda/envs/skbio/lib/python3.8/site-packages/skbio/stats/distance/_base.py:106\u001b[0m, in \u001b[0;36mDissimilarityMatrix.__init__\u001b[0;34m(self, data, ids)\u001b[0m\n\u001b[1;32m    103\u001b[0m     ids \u001b[38;5;241m=\u001b[39m (\u001b[38;5;28mstr\u001b[39m(i) \u001b[38;5;28;01mfor\u001b[39;00m i \u001b[38;5;129;01min\u001b[39;00m \u001b[38;5;28mrange\u001b[39m(data\u001b[38;5;241m.\u001b[39mshape[\u001b[38;5;241m0\u001b[39m]))\n\u001b[1;32m    104\u001b[0m ids \u001b[38;5;241m=\u001b[39m \u001b[38;5;28mtuple\u001b[39m(ids)\n\u001b[0;32m--> 106\u001b[0m \u001b[38;5;28;43mself\u001b[39;49m\u001b[38;5;241;43m.\u001b[39;49m\u001b[43m_validate\u001b[49m\u001b[43m(\u001b[49m\u001b[43mdata\u001b[49m\u001b[43m,\u001b[49m\u001b[43m \u001b[49m\u001b[43mids\u001b[49m\u001b[43m)\u001b[49m\n\u001b[1;32m    108\u001b[0m \u001b[38;5;28mself\u001b[39m\u001b[38;5;241m.\u001b[39m_data \u001b[38;5;241m=\u001b[39m data\n\u001b[1;32m    109\u001b[0m \u001b[38;5;28mself\u001b[39m\u001b[38;5;241m.\u001b[39m_ids \u001b[38;5;241m=\u001b[39m ids\n",
      "File \u001b[0;32m~/.conda/envs/skbio/lib/python3.8/site-packages/skbio/stats/distance/_base.py:1059\u001b[0m, in \u001b[0;36mDistanceMatrix._validate\u001b[0;34m(self, data, ids)\u001b[0m\n\u001b[1;32m   1056\u001b[0m \u001b[38;5;28msuper\u001b[39m(DistanceMatrix, \u001b[38;5;28mself\u001b[39m)\u001b[38;5;241m.\u001b[39m_validate(data, ids)\n\u001b[1;32m   1058\u001b[0m \u001b[38;5;28;01mif\u001b[39;00m (data\u001b[38;5;241m.\u001b[39mT \u001b[38;5;241m!=\u001b[39m data)\u001b[38;5;241m.\u001b[39many():\n\u001b[0;32m-> 1059\u001b[0m     \u001b[38;5;28;01mraise\u001b[39;00m DistanceMatrixError(\n\u001b[1;32m   1060\u001b[0m         \u001b[38;5;124m\"\u001b[39m\u001b[38;5;124mData must be symmetric and cannot contain NaNs.\u001b[39m\u001b[38;5;124m\"\u001b[39m)\n\u001b[1;32m   1062\u001b[0m \u001b[38;5;28;01mif\u001b[39;00m np\u001b[38;5;241m.\u001b[39mtrace(data) \u001b[38;5;241m!=\u001b[39m \u001b[38;5;241m0\u001b[39m:\n\u001b[1;32m   1063\u001b[0m     \u001b[38;5;28;01mraise\u001b[39;00m DistanceMatrixError(\u001b[38;5;124m\"\u001b[39m\u001b[38;5;124mData must be hollow (i.e., the diagonal\u001b[39m\u001b[38;5;124m\"\u001b[39m\n\u001b[1;32m   1064\u001b[0m                               \u001b[38;5;124m\"\u001b[39m\u001b[38;5;124m can only contain zeros).\u001b[39m\u001b[38;5;124m\"\u001b[39m)\n",
      "\u001b[0;31mDistanceMatrixError\u001b[0m: Data must be symmetric and cannot contain NaNs."
     ]
    }
   ],
   "source": [
    "from skbio.stats.ordination import pcoa\n",
    "b_pc = pcoa(bd_cleaned)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "f37c356f-18ab-4c44-85a5-25262b99b8f7",
   "metadata": {},
   "outputs": [],
   "source": []
  }
 ],
 "metadata": {
  "kernelspec": {
   "display_name": "Python (skbio)",
   "language": "python",
   "name": "skbio"
  },
  "language_info": {
   "codemirror_mode": {
    "name": "ipython",
    "version": 3
   },
   "file_extension": ".py",
   "mimetype": "text/x-python",
   "name": "python",
   "nbconvert_exporter": "python",
   "pygments_lexer": "ipython3",
   "version": "3.8.18"
  }
 },
 "nbformat": 4,
 "nbformat_minor": 5
}
